-- MySQL dump 10.13  Distrib 8.0.36, for macos14 (arm64)
--
-- Host: 127.0.0.1    Database: oncokb_demo
-- ------------------------------------------------------
-- Server version	8.0.36

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!50503 SET NAMES utf8mb4 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;

--
-- Table structure for table `alteration`
--

DROP TABLE IF EXISTS `alteration`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `alteration` (
  `id` int NOT NULL AUTO_INCREMENT,
  `alteration` varchar(255) DEFAULT NULL,
  `alteration_type` varchar(255) DEFAULT NULL,
  `name` longtext NOT NULL,
  `protein_end` int DEFAULT NULL,
  `protein_start` int DEFAULT NULL,
  `ref_residues` varchar(255) DEFAULT NULL,
  `uuid` varchar(40) DEFAULT NULL,
  `variant_residues` varchar(255) DEFAULT NULL,
  `consequence` varchar(100) DEFAULT NULL,
  `entrez_gene_id` int DEFAULT NULL,
  `protein_change` longtext,
  `for_germline` bit(1) DEFAULT b'0',
  PRIMARY KEY (`id`),
  KEY `FKD5A6189FC602BC92` (`entrez_gene_id`),
  KEY `FKD5A6189FAEE285DC` (`consequence`),
  CONSTRAINT `FKD5A6189FAEE285DC` FOREIGN KEY (`consequence`) REFERENCES `variant_consequence` (`term`),
  CONSTRAINT `FKD5A6189FC602BC92` FOREIGN KEY (`entrez_gene_id`) REFERENCES `gene` (`entrez_gene_id`)
) ENGINE=InnoDB AUTO_INCREMENT=19400 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `alteration`
--

LOCK TABLES `alteration` WRITE;
/*!40000 ALTER TABLE `alteration` DISABLE KEYS */;
INSERT INTO `alteration` VALUES (1474,'V600E','MUTATION','V600E',600,600,'V',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1475,'L597S','MUTATION','L597S',597,597,'L',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1476,'K601E','MUTATION','K601E',601,601,'K',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1477,'V600','MUTATION','V600',600,600,'V',NULL,NULL,'NA',673,NULL,_binary '\0'),(1478,'V600K','MUTATION','V600K',600,600,'V',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1479,'V600D','MUTATION','V600D',600,600,'V',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1480,'V600R','MUTATION','V600R',600,600,'V',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1481,'V600G','MUTATION','V600G',600,600,'V',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1482,'V600M','MUTATION','V600M',600,600,'V',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1483,'L597R','MUTATION','L597R',597,597,'L',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1484,'L597V','MUTATION','L597V',597,597,'L',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1485,'L597','MUTATION','L597',597,597,'L',NULL,NULL,'NA',673,NULL,_binary '\0'),(1486,'K601','MUTATION','K601',601,601,'K',NULL,NULL,'NA',673,NULL,_binary '\0'),(1487,'K601Q','MUTATION','K601Q',601,601,'K',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1488,'L597Q','MUTATION','L597Q',597,597,'L',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1489,'K601N','MUTATION','K601N',601,601,'K',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1490,'D594A','MUTATION','D594A',594,594,'D',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1491,'D594E','MUTATION','D594E',594,594,'D',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1492,'D594G','MUTATION','D594G',594,594,'D',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1493,'D594N','MUTATION','D594N',594,594,'D',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1494,'D594V','MUTATION','D594V',594,594,'D',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1495,'D594Y','MUTATION','D594Y',594,594,'D',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1496,'G464E','MUTATION','G464E',464,464,'G',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1497,'G464R','MUTATION','G464R',464,464,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1498,'G464V','MUTATION','G464V',464,464,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1499,'G466A','MUTATION','G466A',466,466,'G',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1500,'G466E','MUTATION','G466E',466,466,'G',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1501,'G466R','MUTATION','G466R',466,466,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1502,'G466V','MUTATION','G466V',466,466,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1503,'G469A','MUTATION','G469A',469,469,'G',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1504,'G469E','MUTATION','G469E',469,469,'G',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1505,'G469L','MUTATION','G469L',469,469,'G',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1506,'G469R','MUTATION','G469R',469,469,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1507,'G469V','MUTATION','G469V',469,469,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1508,'G469del','MUTATION','G469del',469,469,'G',NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1509,'S151A','MUTATION','S151A',151,151,'S',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1510,'Q201H','MUTATION','Q201H',201,201,'Q',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1511,'T241P','MUTATION','T241P',241,241,'T',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1512,'A246P','MUTATION','A246P',246,246,'A',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1513,'Q257R','MUTATION','Q257R',257,257,'Q',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1514,'E275K','MUTATION','E275K',275,275,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1515,'D287H','MUTATION','D287H',287,287,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1516,'I326V','MUTATION','I326V',326,326,'I',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1517,'V459L','MUTATION','V459L',459,459,'V',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1518,'R462E','MUTATION','R462E',462,462,'R',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1519,'R462I','MUTATION','R462I',462,462,'R',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1520,'I463S','MUTATION','I463S',463,463,'I',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1521,'S467A','MUTATION','S467A',467,467,'S',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1522,'F468C','MUTATION','F468C',468,468,'F',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1523,'V471F','MUTATION','V471F',471,471,'V',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1524,'Y472C','MUTATION','Y472C',472,472,'Y',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1525,'G478C','MUTATION','G478C',478,478,'G',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1526,'K483E','MUTATION','K483E',483,483,'K',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1527,'K483M','MUTATION','K483M',483,483,'K',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1528,'L485F','MUTATION','L485F',485,485,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1529,'K499E','MUTATION','K499E',499,499,'K',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1530,'E501G','MUTATION','E501G',501,501,'E',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1531,'E501K','MUTATION','E501K',501,501,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1532,'L505H','MUTATION','L505H',505,505,'L',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1533,'R509H','MUTATION','R509H',509,509,'R',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1534,'T529I','MUTATION','T529I',529,529,'T',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1535,'T529M','MUTATION','T529M',529,529,'T',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1536,'T529N','MUTATION','T529N',529,529,'T',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1537,'W531C','MUTATION','W531C',531,531,'W',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1538,'N581D','MUTATION','N581D',581,581,'N',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1539,'N581S','MUTATION','N581S',581,581,'N',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1540,'N581Y','MUTATION','N581Y',581,581,'N',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1541,'L584F','MUTATION','L584F',584,584,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1542,'E586K','MUTATION','E586K',586,586,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1543,'F595L','MUTATION','F595L',595,595,'F',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1544,'G596C','MUTATION','G596C',596,596,'G',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1545,'G596R','MUTATION','G596R',596,596,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1546,'G596V','MUTATION','G596V',596,596,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1547,'A598T','MUTATION','A598T',598,598,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1548,'A598V','MUTATION','A598V',598,598,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1549,'T599I','MUTATION','T599I',599,599,'T',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1550,'T599R','MUTATION','T599R',599,599,'T',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1551,'R671Q','MUTATION','R671Q',671,671,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1552,'A728V','MUTATION','A728V',728,728,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1553,'L485_P490del','MUTATION','L485_P490del',490,485,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1554,'L485_Q494del','MUTATION','L485_Q494del',494,485,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1555,'L485_P490delinsF','MUTATION','L485_P490delinsF',490,485,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1556,'L485_P490delinsY','MUTATION','L485_P490delinsY',490,485,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1557,'N486_P490del','MUTATION','N486_P490del',490,486,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1558,'V487_P492delinsA','MUTATION','V487_P492delinsA',492,487,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1559,'T488_P492del','MUTATION','T488_P492del',492,488,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1560,'P490_Q494del','MUTATION','P490_Q494del',494,490,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1561,'R506_K507insVLR','MUTATION','R506_K507insVLR',507,506,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1562,'T599dup','MUTATION','T599dup',599,599,'T',NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1563,'T599insTT','MUTATION','T599insTT',599,599,'T',NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1564,'T599_V600insV','MUTATION','T599_V600insV',600,599,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1565,'V600delinsYM','MUTATION','V600delinsYM',600,600,'V',NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1566,'V600_K601insDFGLAT','MUTATION','V600_K601insDFGLAT',601,600,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1567,'Fusions','MUTATION','Fusions',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1568,'AGAP3-BRAF Fusion','MUTATION','AGAP3-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1569,'AGK-BRAF Fusion','MUTATION','AGK-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1570,'AKAP9-BRAF Fusion','MUTATION','AKAP9-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1571,'CUL1-BRAF Fusion','MUTATION','CUL1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1572,'FAM131B-BRAF Fusion','MUTATION','FAM131B-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1573,'GTF2I-BRAF Fusion','MUTATION','GTF2I-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1574,'KIAA1549-BRAF Fusion','MUTATION','KIAA1549-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1575,'MKRN1-BRAF Fusion','MUTATION','MKRN1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1576,'PAPSS1-BRAF Fusion','MUTATION','PAPSS1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1577,'PPFIBP2-BRAF Fusion','MUTATION','PPFIBP2-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1578,'SND1-BRAF Fusion','MUTATION','SND1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1579,'TRIM24-BRAF Fusion','MUTATION','TRIM24-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1580,'ZKSCAN1-BRAF Fusion','MUTATION','ZKSCAN1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1581,'Amplification','MUTATION','Amplification',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1582,'D594H','MUTATION','D594H',594,594,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1583,'N581I','MUTATION','N581I',581,581,'N',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1584,'G596D','MUTATION','G596D',596,596,'G',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1585,'P367R','MUTATION','P367R',367,367,'P',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1586,'H574Q','MUTATION','H574Q',574,574,'H',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1587,'W450L','MUTATION','W450L',450,450,'W',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1588,'S467L','MUTATION','S467L',467,467,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1589,'K601T','MUTATION','K601T',601,601,'K',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1590,'G464','MUTATION','G464',464,464,'G',NULL,NULL,'NA',673,NULL,_binary '\0'),(1591,'F247L','MUTATION','F247L',247,247,'F',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1592,'L514V','MUTATION','L514V',514,514,'L',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1593,'PCBP2-BRAF Fusion','MUTATION','PCBP2-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1594,'N486_T491delinsK','MUTATION','N486_T491delinsK',491,486,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1595,'Kinase Domain Duplication','MUTATION','Kinase Domain Duplication',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1596,'ARMC10-BRAF Fusion','MUTATION','ARMC10-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1597,'N486del','MUTATION','N486del',486,486,'N',NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1598,'N486_V487del','MUTATION','N486_V487del',487,486,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1599,'N486_T488del','MUTATION','N486_T488del',488,486,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1600,'N486_A489del','MUTATION','N486_A489del',489,486,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1601,'N486_T491del','MUTATION','N486_T491del',491,486,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1602,'P490del','MUTATION','P490del',490,490,'P',NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1603,'A489_P490del','MUTATION','A489_P490del',490,489,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1604,'T488_P490del','MUTATION','T488_P490del',490,488,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1605,'V487_P490del','MUTATION','V487_P490del',490,487,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1606,'A598_T599insV','MUTATION','A598_T599insV',599,598,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1607,'Oncogenic Mutations','MUTATION','Oncogenic Mutations',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1608,'Oncogenic Mutations {excluding V600}','MUTATION','Oncogenic Mutations (excluding V600)',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1609,'V600 {excluding V600E; V600K}','MUTATION','V600 (excluding V600E and V600K)',600,600,'V',NULL,NULL,'NA',673,NULL,_binary '\0'),(1610,'V169_D380del','MUTATION','p61BRAF-V600E',380,169,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1611,'I592_A598dup','MUTATION','I592_A598dup',598,592,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1612,'L485W','MUTATION','L485W',485,485,'L',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1613,'G469S','MUTATION','G469S',469,469,'G',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1614,'N581T','MUTATION','N581T',581,581,'N',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1615,'G596S','MUTATION','G596S',596,596,'G',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1616,'G606W','MUTATION','G606W',606,606,'G',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1617,'A762V','MUTATION','A762V',762,762,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1618,'R682W','MUTATION','R682W',682,682,'R',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1619,'P367S','MUTATION','P367S',367,367,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1620,'A33T','MUTATION','A33T',33,33,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1621,'R271H','MUTATION','R271H',271,271,'R',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1622,'H725Y','MUTATION','H725Y',725,725,'H',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1623,'F294L','MUTATION','F294L',294,294,'F',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1624,'BRAF-PJA2 Fusion','MUTATION','BRAF-PJA2 Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1625,'I572F','MUTATION','I572F',572,572,'I',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1626,'M620I','MUTATION','M620I',620,620,'M',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1627,'P162S','MUTATION','P162S',162,162,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1628,'R239Q','MUTATION','R239Q',239,239,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1629,'R558Q','MUTATION','R558Q',558,558,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1630,'S363F','MUTATION','S363F',363,363,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1631,'T310I','MUTATION','T310I',310,310,'T',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1632,'L485Y','MUTATION','L485Y',485,485,'L',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1633,'L597P','MUTATION','L597P',597,597,'L',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1634,'V600A','MUTATION','V600A',600,600,'V',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1635,'V600L','MUTATION','V600L',600,600,'V',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1636,'M117L','MUTATION','M117L',117,117,'M',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1637,'N412S','MUTATION','N412S',412,412,'N',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1638,'S616Y','MUTATION','S616Y',616,616,'S',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1639,'T274A','MUTATION','T274A',274,274,'T',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1640,'S607F','MUTATION','S607F',607,607,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1641,'M517V','MUTATION','M517V',517,517,'M',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1642,'W604C','MUTATION','W604C',604,604,'W',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1643,'R389C','MUTATION','R389C',389,389,'R',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1644,'E451K','MUTATION','E451K',451,451,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1645,'S122Y','MUTATION','S122Y',122,122,'S',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1646,'R239L','MUTATION','R239L',239,239,'R',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1647,'ZNF767-BRAF Fusion','MUTATION','ZNF767-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1648,'JHDM1D-BRAF Fusion','MUTATION','JHDM1D-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1649,'E46K','MUTATION','E46K',46,46,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1650,'N49I','MUTATION','N49I',49,49,'N',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1651,'L64I','MUTATION','L64I',64,64,'L',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1652,'K205Q','MUTATION','K205Q',205,205,'K',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1653,'E228V','MUTATION','E228V',228,228,'E',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1654,'L245F','MUTATION','L245F',245,245,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1655,'G258V','MUTATION','G258V',258,258,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1656,'F259L','MUTATION','F259L',259,259,'F',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1657,'A320T','MUTATION','A320T',320,320,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1658,'P348T','MUTATION','P348T',348,348,'P',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1659,'S364L','MUTATION','S364L',364,364,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1660,'D380H','MUTATION','D380H',380,380,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1661,'X380_splice','MUTATION','X380_splice',380,380,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1662,'T401I','MUTATION','T401I',401,401,'T',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1663,'S419Y','MUTATION','S419Y',419,419,'S',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1664,'G421V','MUTATION','G421V',421,421,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1665,'H568D','MUTATION','H568D',568,568,'H',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1666,'L613F','MUTATION','L613F',613,613,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1667,'V639I','MUTATION','V639I',639,639,'V',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1668,'E648Q','MUTATION','E648Q',648,648,'E',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1669,'L678I','MUTATION','L678I',678,678,'L',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1670,'E695K','MUTATION','E695K',695,695,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1671,'K698R','MUTATION','K698R',698,698,'K',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1672,'A712T','MUTATION','A712T',712,712,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1673,'P731S','MUTATION','P731S',731,731,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1674,'A762E','MUTATION','A762E',762,762,'A',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1675,'Q609H','MUTATION','Q609H',609,609,'Q',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1676,'GNAI1-BRAF Fusion','MUTATION','GNAI1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1677,'CLCN6-BRAF Fusion','MUTATION','CLCN6-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1678,'T470P','MUTATION','T470P',470,470,'T',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1679,'M517I','MUTATION','M517I',517,517,'M',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1680,'CSTF3-BRAF Fusion','MUTATION','CSTF3-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1681,'SEPT3-BRAF Fusion','MUTATION','SEPT3-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1682,'SLC45A3-BRAF Fusion','MUTATION','SLC45A3-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1683,'G615R','MUTATION','G615R',615,615,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1684,'L485S','MUTATION','L485S',485,485,'L',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1685,'F595I','MUTATION','F595I',595,595,'F',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1686,'W604G','MUTATION','W604G',604,604,'W',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1687,'P367L','MUTATION','P367L',367,367,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1688,'H574Y','MUTATION','H574Y',574,574,'H',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1689,'H539Y','MUTATION','H539Y',539,539,'H',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1690,'R354Q','MUTATION','R354Q',354,354,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1691,'K205N','MUTATION','K205N',205,205,'K',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1692,'A489_T491del','MUTATION','A489_T491del',491,489,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1693,'BRAF-CDK5RAP2 fusion','MUTATION','BRAF-CDK5RAP2 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1694,'BRAF-LUC7L2 fusion','MUTATION','BRAF-LUC7L2 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1695,'BRAF-NCRNA00244 fusion','MUTATION','BRAF-NCRNA00244 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1696,'BRAF-SCRIB fusion','MUTATION','BRAF-SCRIB fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1697,'BRAF-VIPR2 fusion','MUTATION','BRAF-VIPR2 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1698,'E549Q','MUTATION','E549Q',549,549,'E',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1699,'E715K','MUTATION','E715K',715,715,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1700,'F109I','MUTATION','F109I',109,109,'F',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1701,'G189C','MUTATION','G189C',189,189,'G',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1702,'G8V','MUTATION','G8V',8,8,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1703,'L312P','MUTATION','L312P',312,312,'L',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1704,'L331F','MUTATION','L331F',331,331,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1705,'L382F','MUTATION','L382F',382,382,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1706,'L525P','MUTATION','L525P',525,525,'L',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1707,'L553R','MUTATION','L553R',553,553,'L',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1708,'L577I','MUTATION','L577I',577,577,'L',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1709,'L674Q','MUTATION','L674Q',674,674,'L',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1710,'P321L','MUTATION','P321L',321,321,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1711,'P343L','MUTATION','P343L',343,343,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1712,'P345L','MUTATION','P345L',345,345,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1713,'P708S','MUTATION','P708S',708,708,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1714,'Q302R','MUTATION','Q302R',302,302,'Q',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1715,'R252Q','MUTATION','R252Q',252,252,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1716,'R362L','MUTATION','R362L',362,362,'R',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1717,'S125F','MUTATION','S125F',125,125,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1718,'S136P','MUTATION','S136P',136,136,'S',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1719,'T121I','MUTATION','T121I',121,121,'T',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1720,'T233P','MUTATION','T233P',233,233,'T',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1721,'T332I','MUTATION','T332I',332,332,'T',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1722,'V168L','MUTATION','V168L',168,168,'V',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1723,'Y633C','MUTATION','Y633C',633,633,'Y',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1724,'BRAF-CCDC6 fusion','MUTATION','BRAF-CCDC6 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1725,'Q257H','MUTATION','Q257H',257,257,'Q',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1726,'S614P','MUTATION','S614P',614,614,'S',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1727,'K601del','MUTATION','K601del',601,601,'K',NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1728,'K601I','MUTATION','K601I',601,601,'K',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1729,'R603*','MUTATION','R603*',603,603,'R',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1730,'V600Q','MUTATION','V600Q',600,600,'V',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1731,'X327_splice','MUTATION','X327_splice',327,327,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1732,'T599delinsIP','MUTATION','T599delinsIP',599,599,'T',NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1733,'L65M','MUTATION','L65M',65,65,'L',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1734,'R360Q','MUTATION','R360Q',360,360,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1735,'P403L','MUTATION','P403L',403,403,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1736,'F583C','MUTATION','F583C',583,583,'F',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1737,'R603Q','MUTATION','R603Q',603,603,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1738,'S602Y','MUTATION','S602Y',602,602,'S',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1739,'Q356K','MUTATION','Q356K',356,356,'Q',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1740,'R444Q','MUTATION','R444Q',444,444,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1741,'I457V','MUTATION','I457V',457,457,'I',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1742,'T241M','MUTATION','T241M',241,241,'T',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1743,'A33V','MUTATION','A33V',33,33,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1744,'BRAF-METTL2 fusion','MUTATION','BRAF-METTL2 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1745,'BRAF-METTL2B fusion','MUTATION','BRAF-METTL2B fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1746,'P492T','MUTATION','P492T',492,492,'P',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1747,'S134F','MUTATION','S134F',134,134,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1748,'BRAF-PHTF2 fusion','MUTATION','BRAF-PHTF2 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1749,'P318L','MUTATION','P318L',318,318,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1750,'D385N','MUTATION','D385N',385,385,'D',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1751,'P422A','MUTATION','P422A',422,422,'P',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1752,'G518A','MUTATION','G518A',518,518,'G',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1753,'R362G','MUTATION','R362G',362,362,'R',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1754,'BRAF-PWWP2A fusion','MUTATION','BRAF-PWWP2A fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1755,'S419F','MUTATION','S419F',419,419,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1756,'F256L','MUTATION','F256L',256,256,'F',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1757,'ARL4C-BRAF fusion','MUTATION','ARL4C-BRAF fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1758,'NFIA-BRAF fusion','MUTATION','NFIA-BRAF fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1759,'VK600EI','MUTATION','VK600EI',601,600,'VK',NULL,'EI','missense_variant',673,NULL,_binary '\0'),(1760,'L485_A489delinsFS','MUTATION','L485_A489delinsFS',489,485,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1761,'K483N','MUTATION','K483N',483,483,'K',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1762,'X581_splice','MUTATION','X581_splice',581,581,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1763,'BRAF-MRPS33 fusion','MUTATION','BRAF-MRPS33 fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1764,'X237_splice','MUTATION','X237_splice',237,237,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1765,'PARP12-BRAF fusion','MUTATION','PARP12-BRAF fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1766,'S727T','MUTATION','S727T',727,727,'S',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1767,'D352Y','MUTATION','D352Y',352,352,'D',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1768,'H725Q','MUTATION','H725Q',725,725,'H',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1769,'N105Tfs*77','MUTATION','N105Tfs*77',105,105,'N',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1770,'E695Q','MUTATION','E695Q',695,695,'E',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1771,'S634C','MUTATION','S634C',634,634,'S',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1772,'W48*','MUTATION','W48*',48,48,'W',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1773,'I665V','MUTATION','I665V',665,665,'I',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1774,'L441R','MUTATION','L441R',441,441,'L',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1775,'H353P','MUTATION','H353P',353,353,'H',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1776,'H574L','MUTATION','H574L',574,574,'H',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1777,'A33D','MUTATION','A33D',33,33,'A',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1778,'I714V','MUTATION','I714V',714,714,'I',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1779,'D40Y','MUTATION','D40Y',40,40,'D',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1780,'A404Cfs*9','MUTATION','A404Cfs*9',404,404,'A',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1781,'K183*','MUTATION','K183*',183,183,'K',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1782,'D352H','MUTATION','D352H',352,352,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1783,'S124F','MUTATION','S124F',124,124,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1784,'S337L','MUTATION','S337L',337,337,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1785,'Q562P','MUTATION','Q562P',562,562,'Q',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1786,'L161M','MUTATION','L161M',161,161,'L',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1787,'E13K','MUTATION','E13K',13,13,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1788,'E695D','MUTATION','E695D',695,695,'E',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1789,'R260C','MUTATION','R260C',260,260,'R',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1790,'D555H','MUTATION','D555H',555,555,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1791,'R603L','MUTATION','R603L',603,603,'R',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1792,'L567V','MUTATION','L567V',567,567,'L',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1793,'H608Q','MUTATION','H608Q',608,608,'H',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1794,'I554T','MUTATION','I554T',554,554,'I',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1795,'E45K','MUTATION','E45K',45,45,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1796,'A27T','MUTATION','A27T',27,27,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1797,'A31T','MUTATION','A31T',31,31,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1798,'S37L','MUTATION','S37L',37,37,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1799,'P336L','MUTATION','P336L',336,336,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1800,'T142S','MUTATION','T142S',142,142,'T',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1801,'P622S','MUTATION','P622S',622,622,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1802,'I572S','MUTATION','I572S',572,572,'I',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1803,'E26K','MUTATION','E26K',26,26,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1804,'K473M','MUTATION','K473M',473,473,'K',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1805,'N236K','MUTATION','N236K',236,236,'N',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1806,'G11S','MUTATION','G11S',11,11,'G',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1807,'Q612R','MUTATION','Q612R',612,612,'Q',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1808,'P341T','MUTATION','P341T',341,341,'P',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1809,'S116P','MUTATION','S116P',116,116,'S',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1810,'S428L','MUTATION','S428L',428,428,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1811,'W531S','MUTATION','W531S',531,531,'W',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1812,'A12E','MUTATION','A12E',12,12,'A',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1813,'L382V','MUTATION','L382V',382,382,'L',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1814,'Y82F','MUTATION','Y82F',82,82,'Y',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1815,'S637L','MUTATION','S637L',637,637,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1816,'S634G','MUTATION','S634G',634,634,'S',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1817,'D324H','MUTATION','D324H',324,324,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1818,'R735Q','MUTATION','R735Q',735,735,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1819,'M693T','MUTATION','M693T',693,693,'M',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1820,'A34_A35insPA','MUTATION','A34_A35insPA',35,34,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1821,'K164E','MUTATION','K164E',164,164,'K',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1822,'Q628H','MUTATION','Q628H',628,628,'Q',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1823,'P420T','MUTATION','P420T',420,420,'P',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1824,'V480L','MUTATION','V480L',480,480,'V',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1825,'S535C','MUTATION','S535C',535,535,'S',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1826,'P403S','MUTATION','P403S',403,403,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1827,'A63V','MUTATION','A63V',63,63,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1828,'G28_A35del','MUTATION','G28_A35del',35,28,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1829,'A34E','MUTATION','A34E',34,34,'A',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1830,'R704G','MUTATION','R704G',704,704,'R',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1831,'R682Q','MUTATION','R682Q',682,682,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1832,'E26_A33delinsD','MUTATION','E26_A33delinsD',33,26,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1833,'G32_A33dup','MUTATION','G32_A33dup',33,32,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1834,'G106V','MUTATION','G106V',106,106,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1835,'D391G','MUTATION','D391G',391,391,'D',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1836,'I556F','MUTATION','I556F',556,556,'I',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1837,'P722Q','MUTATION','P722Q',722,722,'P',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1838,'P676A','MUTATION','P676A',676,676,'P',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1839,'P152S','MUTATION','P152S',152,152,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1840,'ZC3HAV1-BRAF Fusion','MUTATION','ZC3HAV1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1841,'CCDC6-BRAF Fusion','MUTATION','CCDC6-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1842,'EPS15-BRAF Fusion','MUTATION','EPS15-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1843,'NUP214-BRAF Fusion','MUTATION','NUP214-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1844,'BTF3L4-BRAF Fusion','MUTATION','BTF3L4-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1845,'GHR-BRAF Fusion','MUTATION','GHR-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1846,'CCDC91-BRAF Fusion','MUTATION','CCDC91-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1847,'DYNC1I2-BRAF Fusion','MUTATION','DYNC1I2-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1848,'MZT1-BRAF Fusion','MUTATION','MZT1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1849,'RAD18-BRAF Fusion','MUTATION','RAD18-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1850,'CUX1-BRAF Fusion','MUTATION','CUX1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1851,'SLC12A7-BRAF Fusion','MUTATION','SLC12A7-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1852,'MYRIP-BRAF Fusion','MUTATION','MYRIP-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1853,'NUB1-BRAF Fusion','MUTATION','NUB1-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1854,'KLHL7-BRAF Fusion','MUTATION','KLHL7-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1855,'TANK-BRAF Fusion','MUTATION','TANK-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1856,'RBMS3-BRAF Fusion','MUTATION','RBMS3-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1857,'STRN3-BRAF Fusion','MUTATION','STRN3-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1858,'STK35-BRAF Fusion','MUTATION','STK35-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1859,'ETFA-BRAF Fusion','MUTATION','ETFA-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1860,'SVOPL-BRAF Fusion','MUTATION','SVOPL-BRAF Fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1861,'G15Afs*39','MUTATION','G15Afs*39',15,15,'G',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1862,'Q16H','MUTATION','Q16H',16,16,'Q',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1863,'L18Q','MUTATION','L18Q',18,18,'L',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1864,'P25L','MUTATION','P25L',25,25,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1865,'A27Rfs*20','MUTATION','A27Rfs*20',27,27,'A',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1866,'G30Pfs*21','MUTATION','G30Pfs*21',30,30,'G',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1867,'X46_splice','MUTATION','X46_splice',46,46,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1868,'E80G','MUTATION','E80G',80,80,'E',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1869,'Q94E','MUTATION','Q94E',94,94,'Q',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1870,'R95T','MUTATION','R95T',95,95,'R',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1871,'S102Y','MUTATION','S102Y',102,102,'S',NULL,'Y','missense_variant',673,NULL,_binary '\0'),(1872,'S110F','MUTATION','S110F',110,110,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1873,'S112C','MUTATION','S112C',112,112,'S',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1874,'S113I','MUTATION','S113I',113,113,'S',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1875,'S114F','MUTATION','S114F',114,114,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1876,'M117I','MUTATION','M117I',117,117,'M',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1877,'S129L','MUTATION','S129L',129,129,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1878,'RNF130-BRAF fusion','MUTATION','RNF130-BRAF fusion',100000,-1,NULL,NULL,NULL,'NA',673,NULL,_binary '\0'),(1879,'P141S','MUTATION','P141S',141,141,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1880,'D143G','MUTATION','D143G',143,143,'D',NULL,'G','missense_variant',673,NULL,_binary '\0'),(1881,'R146W','MUTATION','R146W',146,146,'R',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1882,'N148S','MUTATION','N148S',148,148,'N',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1883,'P155Tfs*4','MUTATION','P155Tfs*4',155,155,'P',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1884,'V157I','MUTATION','V157I',157,157,'V',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1885,'K183E','MUTATION','K183E',183,183,'K',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1886,'P192L','MUTATION','P192L',192,192,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1887,'C194*','MUTATION','C194*',194,194,'C',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1888,'C195W','MUTATION','C195W',195,195,'C',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1889,'I208V','MUTATION','I208V',208,208,'I',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1890,'W210L','MUTATION','W210L',210,210,'W',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1891,'X238_splice','MUTATION','X238_splice',238,238,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1892,'P277Hfs*2','MUTATION','P277Hfs*2',277,277,'P',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1893,'Q285Tfs*3','MUTATION','Q285Tfs*3',285,285,'Q',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1894,'L286V','MUTATION','L286V',286,286,'L',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1895,'L288F','MUTATION','L288F',288,288,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1896,'I300V','MUTATION','I300V',300,300,'I',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1897,'E304K','MUTATION','E304K',304,304,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1898,'A305S','MUTATION','A305S',305,305,'A',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1899,'A305V','MUTATION','A305V',305,305,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1900,'E309*','MUTATION','E309*',309,309,'E',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1901,'P318F','MUTATION','P318F',318,318,'P',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1902,'S337*','MUTATION','S337*',337,337,'S',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1903,'K338E','MUTATION','K338E',338,338,'K',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1904,'P341L','MUTATION','P341L',341,341,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1905,'F346L','MUTATION','F346L',346,346,'F',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1906,'D350N','MUTATION','D350N',350,350,'D',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1907,'R354*','MUTATION','R354*',354,354,'R',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1908,'R360*','MUTATION','R360*',360,360,'R',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1909,'I379Mfs*3','MUTATION','I379Mfs*3',379,379,'I',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1910,'Q386L','MUTATION','Q386L',386,386,'Q',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1911,'X392_splice','MUTATION','X392_splice',392,392,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1912,'G393*','MUTATION','G393*',393,393,'G',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1913,'S394*','MUTATION','S394*',394,394,'S',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1914,'G397S','MUTATION','G397S',397,397,'G',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1915,'P403Lfs*8','MUTATION','P403Lfs*8',403,403,'P',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1916,'S428*','MUTATION','S428*',428,428,'S',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1917,'S428T','MUTATION','S428T',428,428,'S',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1918,'S430L','MUTATION','S430L',430,430,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1919,'S432L','MUTATION','S432L',432,432,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1920,'E433K','MUTATION','E433K',433,433,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1921,'X438_splice','MUTATION','X438_splice',438,438,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1922,'R444W','MUTATION','R444W',444,444,'R',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1923,'D449N','MUTATION','D449N',449,449,'D',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1924,'E451D','MUTATION','E451D',451,451,'E',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1925,'D454N','MUTATION','D454N',454,454,'D',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1926,'Q456H','MUTATION','Q456H',456,456,'Q',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1927,'W476*','MUTATION','W476*',476,476,'W',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1928,'V480M','MUTATION','V480M',480,480,'V',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1929,'V487L','MUTATION','V487L',487,487,'V',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1930,'Q493*','MUTATION','Q493*',493,493,'Q',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1931,'A497V','MUTATION','A497V',497,497,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1932,'G503E','MUTATION','G503E',503,503,'G',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1933,'L537S','MUTATION','L537S',537,537,'L',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1934,'M550I','MUTATION','M550I',550,550,'M',NULL,'I','missense_variant',673,NULL,_binary '\0'),(1935,'A561V','MUTATION','A561V',561,561,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1936,'G563C','MUTATION','G563C',563,563,'G',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1937,'M564T','MUTATION','M564T',564,564,'M',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1938,'X565_splice','MUTATION','X565_splice',565,565,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1939,'A569D','MUTATION','A569D',569,569,'A',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1940,'K591_I592ins*','MUTATION','K591_I592ins*',592,591,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1941,'K591_A598dup','MUTATION','K591_A598dup',598,591,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1942,'L618W','MUTATION','L618W',618,618,'L',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1943,'X621_splice','MUTATION','X621_splice',621,621,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1944,'I625F','MUTATION','I625F',625,625,'I',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1945,'P632L','MUTATION','P632L',632,632,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1946,'Q636E','MUTATION','Q636E',636,636,'Q',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1947,'S637*','MUTATION','S637*',637,637,'S',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1948,'R671*','MUTATION','R671*',671,671,'R',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1949,'P686S','MUTATION','P686S',686,686,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1950,'R701T','MUTATION','R701T',701,701,'R',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1951,'R704Tfs*11','MUTATION','R704Tfs*11',704,704,'R',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1952,'X709_splice','MUTATION','X709_splice',709,709,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1953,'R726C','MUTATION','R726C',726,726,'R',NULL,'C','missense_variant',673,NULL,_binary '\0'),(1954,'L733F','MUTATION','L733F',733,733,'L',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1955,'E741D','MUTATION','E741D',741,741,'E',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1956,'P751A','MUTATION','P751A',751,751,'P',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1957,'G758V','MUTATION','G758V',758,758,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1958,'G759R','MUTATION','G759R',759,759,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1959,'G466*','MUTATION','G466*',466,466,'G',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1960,'L514P','MUTATION','L514P',514,514,'L',NULL,'P','missense_variant',673,NULL,_binary '\0'),(1961,'N581_A598dup','MUTATION','N581_A598dup',598,581,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1962,'Q609E','MUTATION','Q609E',609,609,'Q',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1963,'V600_R603del','MUTATION','V600_R603del',603,600,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1964,'W604L','MUTATION','W604L',604,604,'W',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1965,'Y472dup','MUTATION','Y472dup',472,472,'Y',NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(1966,'S732F','MUTATION','S732F',732,732,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(1967,'G32_A33del','MUTATION','G32_A33del',33,32,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1968,'R384T','MUTATION','R384T',384,384,'R',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1969,'T401A','MUTATION','T401A',401,401,'T',NULL,'A','missense_variant',673,NULL,_binary '\0'),(1970,'R575K','MUTATION','R575K',575,575,'R',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1971,'S136L','MUTATION','S136L',136,136,'S',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1972,'R719H','MUTATION','R719H',719,719,'R',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1973,'X580_splice','MUTATION','X580_splice',580,580,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1974,'E26_A33del','MUTATION','E26_A33del',33,26,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1975,'G30_A33del','MUTATION','G30_A33del',33,30,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1976,'G258D','MUTATION','G258D',258,258,'G',NULL,'D','missense_variant',673,NULL,_binary '\0'),(1977,'P321S','MUTATION','P321S',321,321,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(1978,'S316*','MUTATION','S316*',316,316,'S',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1979,'G606V','MUTATION','G606V',606,606,'G',NULL,'V','missense_variant',673,NULL,_binary '\0'),(1980,'R558*','MUTATION','R558*',558,558,'R',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1981,'R389H','MUTATION','R389H',389,389,'R',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1982,'R362Q','MUTATION','R362Q',362,362,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(1983,'D479N','MUTATION','D479N',479,479,'D',NULL,'N','missense_variant',673,NULL,_binary '\0'),(1984,'D565E','MUTATION','D565E',565,565,'D',NULL,'E','missense_variant',673,NULL,_binary '\0'),(1985,'N105Efs*4','MUTATION','N105Efs*4',105,105,'N',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(1986,'G265R','MUTATION','G265R',265,265,'G',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1987,'P328L','MUTATION','P328L',328,328,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(1988,'H608R','MUTATION','H608R',608,608,'H',NULL,'R','missense_variant',673,NULL,_binary '\0'),(1989,'P523T','MUTATION','P523T',523,523,'P',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1990,'D454H','MUTATION','D454H',454,454,'D',NULL,'H','missense_variant',673,NULL,_binary '\0'),(1991,'X393_splice','MUTATION','X393_splice',393,393,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1992,'R735W','MUTATION','R735W',735,735,'R',NULL,'W','missense_variant',673,NULL,_binary '\0'),(1993,'I543M','MUTATION','I543M',543,543,'I',NULL,'M','missense_variant',673,NULL,_binary '\0'),(1994,'R691T','MUTATION','R691T',691,691,'R',NULL,'T','missense_variant',673,NULL,_binary '\0'),(1995,'S657*','MUTATION','S657*',657,657,'S',NULL,'*','stop_gained',673,NULL,_binary '\0'),(1996,'Q98del','MUTATION','Q98del',98,98,'Q',NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(1997,'X203_splice','MUTATION','X203_splice',203,203,NULL,NULL,NULL,'splice_region_variant',673,NULL,_binary '\0'),(1998,'E83K','MUTATION','E83K',83,83,'E',NULL,'K','missense_variant',673,NULL,_binary '\0'),(1999,'K552N','MUTATION','K552N',552,552,'K',NULL,'N','missense_variant',673,NULL,_binary '\0'),(2000,'E741G','MUTATION','E741G',741,741,'E',NULL,'G','missense_variant',673,NULL,_binary '\0'),(2001,'L613M','MUTATION','L613M',613,613,'L',NULL,'M','missense_variant',673,NULL,_binary '\0'),(2002,'S122F','MUTATION','S122F',122,122,'S',NULL,'F','missense_variant',673,NULL,_binary '\0'),(2003,'L185M','MUTATION','L185M',185,185,'L',NULL,'M','missense_variant',673,NULL,_binary '\0'),(2004,'M564I','MUTATION','M564I',564,564,'M',NULL,'I','missense_variant',673,NULL,_binary '\0'),(2005,'P731L','MUTATION','P731L',731,731,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(2006,'G10_G11del','MUTATION','G10_G11del',11,10,NULL,NULL,NULL,'inframe_deletion',673,NULL,_binary '\0'),(2007,'P170L','MUTATION','P170L',170,170,'P',NULL,'L','missense_variant',673,NULL,_binary '\0'),(2008,'P328S','MUTATION','P328S',328,328,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(2009,'P731A','MUTATION','P731A',731,731,'P',NULL,'A','missense_variant',673,NULL,_binary '\0'),(2010,'V504E','MUTATION','V504E',504,504,'V',NULL,'E','missense_variant',673,NULL,_binary '\0'),(2011,'S607Pfs*43','MUTATION','S607Pfs*43',607,607,'S',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(2012,'Y566*','MUTATION','Y566*',566,566,'Y',NULL,'*','stop_gained',673,NULL,_binary '\0'),(2013,'Q709Pfs*7','MUTATION','Q709Pfs*7',709,709,'Q',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(2014,'C264F','MUTATION','C264F',264,264,'C',NULL,'F','missense_variant',673,NULL,_binary '\0'),(2015,'M484V','MUTATION','M484V',484,484,'M',NULL,'V','missense_variant',673,NULL,_binary '\0'),(2016,'F68C','MUTATION','F68C',68,68,'F',NULL,'C','missense_variant',673,NULL,_binary '\0'),(2017,'Q344*','MUTATION','Q344*',344,344,'Q',NULL,'*','stop_gained',673,NULL,_binary '\0'),(2018,'V369A','MUTATION','V369A',369,369,'V',NULL,'A','missense_variant',673,NULL,_binary '\0'),(2019,'K150Sfs*32','MUTATION','K150Sfs*32',150,150,'K',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(2020,'A569P','MUTATION','A569P',569,569,'A',NULL,'P','missense_variant',673,NULL,_binary '\0'),(2021,'W476R','MUTATION','W476R',476,476,'W',NULL,'R','missense_variant',673,NULL,_binary '\0'),(2022,'R437G','MUTATION','R437G',437,437,'R',NULL,'G','missense_variant',673,NULL,_binary '\0'),(2023,'Q709K','MUTATION','Q709K',709,709,'Q',NULL,'K','missense_variant',673,NULL,_binary '\0'),(2024,'A712V','MUTATION','A712V',712,712,'A',NULL,'V','missense_variant',673,NULL,_binary '\0'),(2025,'A35T','MUTATION','A35T',35,35,'A',NULL,'T','missense_variant',673,NULL,_binary '\0'),(2026,'D179G','MUTATION','D179G',179,179,'D',NULL,'G','missense_variant',673,NULL,_binary '\0'),(2027,'M23I','MUTATION','M23I',23,23,'M',NULL,'I','missense_variant',673,NULL,_binary '\0'),(2028,'P141A','MUTATION','P141A',141,141,'P',NULL,'A','missense_variant',673,NULL,_binary '\0'),(2029,'R509Q','MUTATION','R509Q',509,509,'R',NULL,'Q','missense_variant',673,NULL,_binary '\0'),(2030,'P492S','MUTATION','P492S',492,492,'P',NULL,'S','missense_variant',673,NULL,_binary '\0'),(2031,'A400D','MUTATION','A400D',400,400,'A',NULL,'D','missense_variant',673,NULL,_binary '\0'),(2032,'S136fs','MUTATION','S136fs',136,136,'S',NULL,NULL,'frameshift_variant',673,NULL,_binary '\0'),(2033,'N581H','MUTATION','N581H',581,581,'N',NULL,'H','missense_variant',673,NULL,_binary '\0'),(2034,'T599_V600insEAT','MUTATION','T599_V600insEAT',600,599,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(2035,'T599_V600insETT','MUTATION','T599_V600insETT',600,599,NULL,NULL,NULL,'inframe_insertion',673,NULL,_binary '\0'),(2036,'K601_S602delinsNT','MUTATION','K601_S602delinsNT',602,601,NULL,NULL,NULL,'missense_variant',673,NULL,_binary '\0'),(16804,'Fusions','MUTATION','Fusions',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16805,'Oncogenic Mutations','MUTATION','Oncogenic Mutations',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16806,'FIG-ROS1 Fusion','MUTATION','FIG-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16807,'CD74-ROS1 Fusion','MUTATION','CD74-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16808,'LRIG3-ROS1 Fusion','MUTATION','LRIG3-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16809,'KDELR2-ROS1 Fusion','MUTATION','KDELR2-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16810,'SLC34A2-ROS1 Fusion','MUTATION','SLC34A2-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16811,'TPM3-ROS1 Fusion','MUTATION','TPM3-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16812,'SDC4-ROS1 Fusion','MUTATION','SDC4-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16813,'EZR-ROS1 Fusion','MUTATION','EZR-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16814,'CCDC6-ROS1 Fusion','MUTATION','CCDC6-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16815,'CEP85L-ROS1 Fusion','MUTATION','CEP85L-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16816,'TFG-ROS1 Fusion','MUTATION','TFG-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16817,'YWHAE-ROS1 Fusion','MUTATION','YWHAE-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16818,'LIMA1-ROS1 Fusion','MUTATION','LIMA1-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16819,'MSN-ROS1 Fusion','MUTATION','MSN-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16820,'G2032R','MUTATION','G2032R',2032,2032,'G',NULL,'R','missense_variant',6098,NULL,_binary '\0'),(16821,'L1951R','MUTATION','L1951R',1951,1951,'L',NULL,'R','missense_variant',6098,NULL,_binary '\0'),(16822,'D2033N','MUTATION','D2033N',2033,2033,'D',NULL,'N','missense_variant',6098,NULL,_binary '\0'),(16823,'C2060G','MUTATION','C2060G',2060,2060,'C',NULL,'G','missense_variant',6098,NULL,_binary '\0'),(16824,'E1935G','MUTATION','E1935G',1935,1935,'E',NULL,'G','missense_variant',6098,NULL,_binary '\0'),(16825,'G1971E','MUTATION','G1971E',1971,1971,'G',NULL,'E','missense_variant',6098,NULL,_binary '\0'),(16826,'L1947R','MUTATION','L1947R',1947,1947,'L',NULL,'R','missense_variant',6098,NULL,_binary '\0'),(16827,'V2098I','MUTATION','V2098I',2098,2098,'V',NULL,'I','missense_variant',6098,NULL,_binary '\0'),(16828,'G2101A','MUTATION','G2101A',2101,2101,'G',NULL,'A','missense_variant',6098,NULL,_binary '\0'),(16829,'S1986Y','MUTATION','S1986Y',1986,1986,'S',NULL,'Y','missense_variant',6098,NULL,_binary '\0'),(16830,'S1986F','MUTATION','S1986F',1986,1986,'S',NULL,'F','missense_variant',6098,NULL,_binary '\0'),(16831,'Amplification','MUTATION','Amplification',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16832,'GOPC-ROS1 Fusion','MUTATION','GOPC-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16833,'F2004V','MUTATION','F2004V',2004,2004,'F',NULL,'V','missense_variant',6098,NULL,_binary '\0'),(16834,'F2004C','MUTATION','F2004C',2004,2004,'F',NULL,'C','missense_variant',6098,NULL,_binary '\0'),(16835,'L2086F','MUTATION','L2086F',2086,2086,'L',NULL,'F','missense_variant',6098,NULL,_binary '\0'),(16836,'N1461S','MUTATION','N1461S',1461,1461,'N',NULL,'S','missense_variant',6098,NULL,_binary '\0'),(16837,'D1903G','MUTATION','D1903G',1903,1903,'D',NULL,'G','missense_variant',6098,NULL,_binary '\0'),(16838,'H113R','MUTATION','H113R',113,113,'H',NULL,'R','missense_variant',6098,NULL,_binary '\0'),(16839,'K1422R','MUTATION','K1422R',1422,1422,'K',NULL,'R','missense_variant',6098,NULL,_binary '\0'),(16840,'L1167*','MUTATION','L1167*',1167,1167,'L',NULL,'*','stop_gained',6098,NULL,_binary '\0'),(16841,'V447E','MUTATION','V447E',447,447,'V',NULL,'E','missense_variant',6098,NULL,_binary '\0'),(16842,'K35N','MUTATION','K35N',35,35,'K',NULL,'N','missense_variant',6098,NULL,_binary '\0'),(16843,'DCBLD1-ROS1 Fusion','MUTATION','DCBLD1-ROS1 Fusion',100000,-1,NULL,NULL,NULL,'NA',6098,NULL,_binary '\0'),(16844,'R1728S','MUTATION','R1728S',1728,1728,'R',NULL,'S','missense_variant',6098,NULL,_binary '\0'),(18246,'P322R','MUTATION','P322R',322,322,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18247,'H214R','MUTATION','H214R',214,214,'H',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18248,'R175H','MUTATION','R175H',175,175,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18249,'S121F','MUTATION','S121F',121,121,'S',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18250,'H214Q','MUTATION','H214Q',214,214,'H',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18251,'K291E','MUTATION','K291E',291,291,'K',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18252,'K292T','MUTATION','K292T',292,292,'K',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18253,'Q144R','MUTATION','Q144R',144,144,'Q',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18254,'I162M','MUTATION','I162M',162,162,'I',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18255,'R290G','MUTATION','R290G',290,290,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18256,'L344P','MUTATION','L344P',344,344,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18257,'R337C','MUTATION','R337C',337,337,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18258,'L330H','MUTATION','L330H',330,330,'L',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18259,'L330P','MUTATION','L330P',330,330,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18260,'L330R','MUTATION','L330R',330,330,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18261,'R337P','MUTATION','R337P',337,337,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18262,'R342P','MUTATION','R342P',342,342,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18263,'F341C','MUTATION','F341C',341,341,'F',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18264,'L344R','MUTATION','L344R',344,344,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18265,'A347T','MUTATION','A347T',347,347,'A',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18266,'R337H','MUTATION','R337H',337,337,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18267,'R337L','MUTATION','R337L',337,337,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18268,'F338I','MUTATION','F338I',338,338,'F',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18269,'L348S','MUTATION','L348S',348,348,'L',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18270,'P47S','MUTATION','P47S',47,47,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18271,'F134Y','MUTATION','F134Y',134,134,'F',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18272,'P152L','MUTATION','P152L',152,152,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18273,'R158H','MUTATION','R158H',158,158,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18274,'R158L','MUTATION','R158L',158,158,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18275,'Y163C','MUTATION','Y163C',163,163,'Y',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18276,'H193P','MUTATION','H193P',193,193,'H',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18277,'P219S','MUTATION','P219S',219,219,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18278,'C229R','MUTATION','C229R',229,229,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18279,'N235D','MUTATION','N235D',235,235,'N',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18280,'R248W','MUTATION','R248W',248,248,'R',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18281,'I254T','MUTATION','I254T',254,254,'I',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18282,'R267Q','MUTATION','R267Q',267,267,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18283,'R273H','MUTATION','R273H',273,273,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18284,'E285V','MUTATION','E285V',285,285,'E',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18285,'R175L','MUTATION','R175L',175,175,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18286,'R175C','MUTATION','R175C',175,175,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18287,'R248Q','MUTATION','R248Q',248,248,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18288,'R273C','MUTATION','R273C',273,273,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18289,'R273G','MUTATION','R273G',273,273,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18290,'R213Q','MUTATION','R213Q',213,213,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18291,'G245S','MUTATION','G245S',245,245,'G',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18292,'G245D','MUTATION','G245D',245,245,'G',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18293,'G245C','MUTATION','G245C',245,245,'G',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18294,'G245A','MUTATION','G245A',245,245,'G',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18295,'Y220C','MUTATION','Y220C',220,220,'Y',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18296,'Y220S','MUTATION','Y220S',220,220,'Y',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18297,'R282Q','MUTATION','R282Q',282,282,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18298,'H179R','MUTATION','H179R',179,179,'H',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18299,'H179Y','MUTATION','H179Y',179,179,'H',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18300,'H179L','MUTATION','H179L',179,179,'H',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18301,'H179Q','MUTATION','H179Q',179,179,'H',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18302,'H179N','MUTATION','H179N',179,179,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18303,'H193R','MUTATION','H193R',193,193,'H',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18304,'H193N','MUTATION','H193N',193,193,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18305,'C176F','MUTATION','C176F',176,176,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18306,'C176R','MUTATION','C176R',176,176,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18307,'I195T','MUTATION','I195T',195,195,'I',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18308,'G266E','MUTATION','G266E',266,266,'G',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18309,'R249S','MUTATION','R249S',249,249,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18310,'R249M','MUTATION','R249M',249,249,'R',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18311,'R249W','MUTATION','R249W',249,249,'R',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18312,'Y234C','MUTATION','Y234C',234,234,'Y',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18313,'Y234H','MUTATION','Y234H',234,234,'Y',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18314,'C238F','MUTATION','C238F',238,238,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18315,'C238S','MUTATION','C238S',238,238,'C',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18316,'V173E','MUTATION','V173E',173,173,'V',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18317,'R158C','MUTATION','R158C',158,158,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18318,'P151S','MUTATION','P151S',151,151,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18319,'P151H','MUTATION','P151H',151,151,'P',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18320,'P151A','MUTATION','P151A',151,151,'P',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18321,'Y236C','MUTATION','Y236C',236,236,'Y',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18322,'Y236D','MUTATION','Y236D',236,236,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18323,'Y236S','MUTATION','Y236S',236,236,'Y',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18324,'P278S','MUTATION','P278S',278,278,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18325,'P278L','MUTATION','P278L',278,278,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18326,'P278A','MUTATION','P278A',278,278,'P',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18327,'P278R','MUTATION','P278R',278,278,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18328,'C135Y','MUTATION','C135Y',135,135,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18329,'C135R','MUTATION','C135R',135,135,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18330,'C135S','MUTATION','C135S',135,135,'C',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18331,'C275S','MUTATION','C275S',275,275,'C',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18332,'M237K','MUTATION','M237K',237,237,'M',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18333,'S241F','MUTATION','S241F',241,241,'S',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18334,'S241T','MUTATION','S241T',241,241,'S',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18335,'V272M','MUTATION','V272M',272,272,'V',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18336,'V272L','MUTATION','V272L',272,272,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18337,'V157F','MUTATION','V157F',157,157,'V',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18338,'V157D','MUTATION','V157D',157,157,'V',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18339,'R280K','MUTATION','R280K',280,280,'R',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18340,'R280T','MUTATION','R280T',280,280,'R',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18341,'C242F','MUTATION','C242F',242,242,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18342,'N239S','MUTATION','N239S',239,239,'N',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18343,'S215C','MUTATION','S215C',215,215,'S',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18344,'C242S','MUTATION','C242S',242,242,'C',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18345,'C242Y','MUTATION','C242Y',242,242,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18346,'G244D','MUTATION','G244D',244,244,'G',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18347,'G244S','MUTATION','G244S',244,244,'G',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18348,'G244R','MUTATION','G244R',244,244,'G',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18349,'V274F','MUTATION','V274F',274,274,'V',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18350,'S121A','MUTATION','S121A',121,121,'S',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18351,'S121Y','MUTATION','S121Y',121,121,'S',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18352,'S121C','MUTATION','S121C',121,121,'S',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18353,'P153H','MUTATION','P153H',153,153,'P',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18354,'R156C','MUTATION','R156C',156,156,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18355,'F212Y','MUTATION','F212Y',212,212,'F',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18356,'E221Q','MUTATION','E221Q',221,221,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18357,'K291Q','MUTATION','K291Q',291,291,'K',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18358,'K292I','MUTATION','K292I',292,292,'K',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18359,'F328V','MUTATION','F328V',328,328,'F',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18360,'L348F','MUTATION','L348F',348,348,'L',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18361,'S241Y','MUTATION','S241Y',241,241,'S',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18362,'R248L','MUTATION','R248L',248,248,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18363,'G334R','MUTATION','G334R',334,334,'G',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18364,'R282W','MUTATION','R282W',282,282,'R',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18365,'P250L','MUTATION','P250L',250,250,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18366,'E258V','MUTATION','E258V',258,258,'E',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18367,'R110L','MUTATION','R110L',110,110,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18368,'R110P','MUTATION','R110P',110,110,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18369,'H168R','MUTATION','H168R',168,168,'H',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18370,'T123A','MUTATION','T123A',123,123,'T',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18371,'H168N','MUTATION','H168N',168,168,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18372,'H233N','MUTATION','H233N',233,233,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18373,'H297N','MUTATION','H297N',297,297,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18374,'H214N','MUTATION','H214N',214,214,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18375,'H115N','MUTATION','H115N',115,115,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18376,'A161T','MUTATION','A161T',161,161,'A',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18377,'G266R','MUTATION','G266R',266,266,'G',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18378,'A161S','MUTATION','A161S',161,161,'A',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18379,'V197E','MUTATION','V197E',197,197,'V',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18380,'M133K','MUTATION','M133K',133,133,'M',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18381,'I255F','MUTATION','I255F',255,255,'I',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18382,'V197L','MUTATION','V197L',197,197,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18383,'M133T','MUTATION','M133T',133,133,'M',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18384,'C141Y','MUTATION','C141Y',141,141,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18385,'V173L','MUTATION','V173L',173,173,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18386,'V173M','MUTATION','V173M',173,173,'V',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18387,'Y205C','MUTATION','Y205C',205,205,'Y',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18388,'R273L','MUTATION','R273L',273,273,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18389,'E286K','MUTATION','E286K',286,286,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18390,'I251S','MUTATION','I251S',251,251,'I',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18391,'E326L','MUTATION','E326L',326,326,'E',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18392,'R181C','MUTATION','R181C',181,181,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18393,'R202S','MUTATION','R202S',202,202,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18394,'T150I','MUTATION','T150I',150,150,'T',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18395,'S240R','MUTATION','S240R',240,240,'S',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18396,'G199R','MUTATION','G199R',199,199,'G',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18397,'Q144P','MUTATION','Q144P',144,144,'Q',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18398,'Y163N','MUTATION','Y163N',163,163,'Y',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18399,'H168Y','MUTATION','H168Y',168,168,'H',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18400,'R181H','MUTATION','R181H',181,181,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18401,'R181P','MUTATION','R181P',181,181,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18402,'E180K','MUTATION','E180K',180,180,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18403,'R181L','MUTATION','R181L',181,181,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18404,'K120R','MUTATION','K120R',120,120,'K',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18405,'K120M','MUTATION','K120M',120,120,'K',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18406,'K120E','MUTATION','K120E',120,120,'K',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18407,'R273K','MUTATION','R273K',273,273,'R',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18408,'K120N','MUTATION','K120N',120,120,'K',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18409,'C277Q','MUTATION','C277Q',277,277,'C',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18410,'C277R','MUTATION','C277R',277,277,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18411,'R248K','MUTATION','R248K',248,248,'R',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18412,'R283Q','MUTATION','R283Q',283,283,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18413,'H178D','MUTATION','H178D',178,178,'H',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18414,'D186A','MUTATION','D186A',186,186,'D',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18415,'C277W','MUTATION','C277W',277,277,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18416,'K120A','MUTATION','K120A',120,120,'K',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18417,'R280A','MUTATION','R280A',280,280,'R',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18418,'V143A','MUTATION','V143A',143,143,'V',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18419,'R290H','MUTATION','R290H',290,290,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18420,'R156H','MUTATION','R156H',156,156,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18421,'R267P','MUTATION','R267P',267,267,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18422,'Q331H','MUTATION','Q331H',331,331,'Q',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18423,'T329I','MUTATION','T329I',329,329,'T',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18424,'Q331R','MUTATION','Q331R',331,331,'Q',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18425,'G356A','MUTATION','G356A',356,356,'G',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18426,'P316L','MUTATION','P316L',316,316,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18427,'Oncogenic Mutations','MUTATION','Oncogenic Mutations',100000,-1,NULL,NULL,NULL,'NA',7157,NULL,_binary '\0'),(18428,'S215G','MUTATION','S215G',215,215,'S',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18429,'L145R','MUTATION','L145R',145,145,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18430,'Deletion','MUTATION','Deletion',100000,-1,NULL,NULL,NULL,'NA',7157,NULL,_binary '\0'),(18431,'Truncating Mutations','MUTATION','Truncating Mutations',100000,-1,NULL,NULL,NULL,'feature_truncation',7157,NULL,_binary '\0'),(18432,'E285K','MUTATION','E285K',285,285,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18433,'P191del','MUTATION','P191del',191,191,'P',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18434,'R196G','MUTATION','R196G',196,196,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18435,'R196L','MUTATION','R196L',196,196,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18436,'C124R','MUTATION','C124R',124,124,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18437,'C141W','MUTATION','C141W',141,141,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18438,'C176W','MUTATION','C176W',176,176,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18439,'G154V','MUTATION','G154V',154,154,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18440,'C277Y','MUTATION','C277Y',277,277,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18441,'Y236H','MUTATION','Y236H',236,236,'Y',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18442,'Y236N','MUTATION','Y236N',236,236,'Y',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18443,'H193Q','MUTATION','H193Q',193,193,'H',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18444,'H193D','MUTATION','H193D',193,193,'H',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18445,'H193Y','MUTATION','H193Y',193,193,'H',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18446,'H193L','MUTATION','H193L',193,193,'H',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18447,'Y163D','MUTATION','Y163D',163,163,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18448,'Y163H','MUTATION','Y163H',163,163,'Y',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18449,'V274A','MUTATION','V274A',274,274,'V',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18450,'V274D','MUTATION','V274D',274,274,'V',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18451,'C277F','MUTATION','C277F',277,277,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18452,'V274G','MUTATION','V274G',274,274,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18453,'V274L','MUTATION','V274L',274,274,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18454,'G244A','MUTATION','G244A',244,244,'G',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18455,'G244C','MUTATION','G244C',244,244,'G',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18456,'G244V','MUTATION','G244V',244,244,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18457,'G244F','MUTATION','G244F',244,244,'G',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18458,'V173A','MUTATION','V173A',173,173,'V',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18459,'V173G','MUTATION','V173G',173,173,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18460,'C277G','MUTATION','C277G',277,277,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18461,'Y205S','MUTATION','Y205S',205,205,'Y',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18462,'Y205D','MUTATION','Y205D',205,205,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18463,'Y205F','MUTATION','Y205F',205,205,'Y',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18464,'Y205H','MUTATION','Y205H',205,205,'Y',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18465,'Y205N','MUTATION','Y205N',205,205,'Y',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18466,'V143E','MUTATION','V143E',143,143,'V',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18467,'C242R','MUTATION','C242R',242,242,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18468,'C242W','MUTATION','C242W',242,242,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18469,'C242G','MUTATION','C242G',242,242,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18470,'V143G','MUTATION','V143G',143,143,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18471,'G266V','MUTATION','G266V',266,266,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18472,'V143M','MUTATION','V143M',143,143,'V',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18473,'G266K','MUTATION','G266K',266,266,'G',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18474,'R158P','MUTATION','R158P',158,158,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18475,'R158S','MUTATION','R158S',158,158,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18476,'R158F','MUTATION','R158F',158,158,'R',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18477,'R158G','MUTATION','R158G',158,158,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18478,'E286Q','MUTATION','E286Q',286,286,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18479,'E286A','MUTATION','E286A',286,286,'E',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18480,'H179P','MUTATION','H179P',179,179,'H',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18481,'H179D','MUTATION','H179D',179,179,'H',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18482,'H168P','MUTATION','H168P',168,168,'H',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18483,'H168L','MUTATION','H168L',168,168,'H',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18484,'C176S','MUTATION','C176S',176,176,'C',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18485,'C176G','MUTATION','C176G',176,176,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18486,'C176Y','MUTATION','C176Y',176,176,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18487,'C238R','MUTATION','C238R',238,238,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18488,'C238G','MUTATION','C238G',238,238,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18489,'C238W','MUTATION','C238W',238,238,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18490,'C238Y','MUTATION','C238Y',238,238,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18491,'E285Q','MUTATION','E285Q',285,285,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18492,'M237V','MUTATION','M237V',237,237,'M',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18493,'M237I','MUTATION','M237I',237,237,'M',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18494,'G154C','MUTATION','G154C',154,154,'G',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18495,'G154D','MUTATION','G154D',154,154,'G',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18496,'S241P','MUTATION','S241P',241,241,'S',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18497,'S241A','MUTATION','S241A',241,241,'S',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18498,'S241C','MUTATION','S241C',241,241,'S',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18499,'R282P','MUTATION','R282P',282,282,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18500,'R282G','MUTATION','R282G',282,282,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18501,'R282L','MUTATION','R282L',282,282,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18502,'P250R','MUTATION','P250R',250,250,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18503,'E286V','MUTATION','E286V',286,286,'E',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18504,'N239D','MUTATION','N239D',239,239,'N',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18505,'N239T','MUTATION','N239T',239,239,'N',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18506,'N239H','MUTATION','N239H',239,239,'N',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18507,'N239K','MUTATION','N239K',239,239,'N',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18508,'E286G','MUTATION','E286G',286,286,'E',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18509,'R280S','MUTATION','R280S',280,280,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18510,'R280G','MUTATION','R280G',280,280,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18511,'R280I','MUTATION','R280I',280,280,'R',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18512,'P278T','MUTATION','P278T',278,278,'P',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18513,'P278F','MUTATION','P278F',278,278,'P',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18514,'P278H','MUTATION','P278H',278,278,'P',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18515,'H178P','MUTATION','H178P',178,178,'H',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18516,'H178Q','MUTATION','H178Q',178,178,'H',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18517,'H178N','MUTATION','H178N',178,178,'H',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18518,'V197G','MUTATION','V197G',197,197,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18519,'V197M','MUTATION','V197M',197,197,'V',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18520,'M133R','MUTATION','M133R',133,133,'M',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18521,'M133I','MUTATION','M133I',133,133,'M',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18522,'P152Q','MUTATION','P152Q',152,152,'P',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18523,'P152R','MUTATION','P152R',152,152,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18524,'P152S','MUTATION','P152S',152,152,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18525,'P152T','MUTATION','P152T',152,152,'P',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18526,'P152K','MUTATION','P152K',152,152,'P',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18527,'C141R','MUTATION','C141R',141,141,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18528,'C141S','MUTATION','C141S',141,141,'C',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18529,'C141F','MUTATION','C141F',141,141,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18530,'C141G','MUTATION','C141G',141,141,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18531,'C275F','MUTATION','C275F',275,275,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18532,'C275R','MUTATION','C275R',275,275,'C',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18533,'C275W','MUTATION','C275W',275,275,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18534,'C275G','MUTATION','C275G',275,275,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18535,'C275Y','MUTATION','C275Y',275,275,'C',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18536,'I251F','MUTATION','I251F',251,251,'I',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18537,'I251M','MUTATION','I251M',251,251,'I',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18538,'I251N','MUTATION','I251N',251,251,'I',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18539,'E258A','MUTATION','E258A',258,258,'E',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18540,'E258Q','MUTATION','E258Q',258,258,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18541,'E258D','MUTATION','E258D',258,258,'E',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18542,'E258G','MUTATION','E258G',258,258,'E',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18543,'E258K','MUTATION','E258K',258,258,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18544,'I255S','MUTATION','I255S',255,255,'I',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18545,'I255T','MUTATION','I255T',255,255,'I',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18546,'I255N','MUTATION','I255N',255,255,'I',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18547,'I254S','MUTATION','I254S',254,254,'I',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18548,'I254V','MUTATION','I254V',254,254,'I',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18549,'I254F','MUTATION','I254F',254,254,'I',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18550,'I254L','MUTATION','I254L',254,254,'I',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18551,'I254M','MUTATION','I254M',254,254,'I',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18552,'I254N','MUTATION','I254N',254,254,'I',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18553,'V272G','MUTATION','V272G',272,272,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18554,'R337S','MUTATION','R337S',337,337,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18555,'V272A','MUTATION','V272A',272,272,'V',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18556,'V272E','MUTATION','V272E',272,272,'V',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18557,'V272K','MUTATION','V272K',272,272,'V',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18558,'R267W','MUTATION','R267W',267,267,'R',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18559,'R267L','MUTATION','R267L',267,267,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18560,'V157A','MUTATION','V157A',157,157,'V',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18561,'V157G','MUTATION','V157G',157,157,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18562,'V157L','MUTATION','V157L',157,157,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18563,'R110C','MUTATION','R110C',110,110,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18564,'R110H','MUTATION','R110H',110,110,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18565,'R249T','MUTATION','R249T',249,249,'R',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18566,'R249G','MUTATION','R249G',249,249,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18567,'R249I','MUTATION','R249I',249,249,'R',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18568,'R249K','MUTATION','R249K',249,249,'R',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18569,'R337G','MUTATION','R337G',337,337,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18570,'A161P','MUTATION','A161P',161,161,'A',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18571,'A161D','MUTATION','A161D',161,161,'A',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18572,'A161V','MUTATION','A161V',161,161,'A',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18573,'R267G','MUTATION','R267G',267,267,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18574,'Y220D','MUTATION','Y220D',220,220,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18575,'Y220H','MUTATION','Y220H',220,220,'Y',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18576,'Y220N','MUTATION','Y220N',220,220,'Y',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18577,'F134C','MUTATION','F134C',134,134,'F',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18578,'R273P','MUTATION','R273P',273,273,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18579,'R273S','MUTATION','R273S',273,273,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18580,'F134S','MUTATION','F134S',134,134,'F',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18581,'F134V','MUTATION','F134V',134,134,'F',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18582,'F134I','MUTATION','F134I',134,134,'F',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18583,'F134L','MUTATION','F134L',134,134,'F',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18584,'G245R','MUTATION','G245R',245,245,'G',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18585,'G245V','MUTATION','G245V',245,245,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18586,'G245N','MUTATION','G245N',245,245,'G',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18587,'S215R','MUTATION','S215R',215,215,'S',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18588,'S215T','MUTATION','S215T',215,215,'S',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18589,'S215I','MUTATION','S215I',215,215,'S',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18590,'S215N','MUTATION','S215N',215,215,'S',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18591,'Q144H','MUTATION','Q144H',144,144,'Q',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18592,'Q144L','MUTATION','Q144L',144,144,'Q',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18593,'Y234S','MUTATION','Y234S',234,234,'Y',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18594,'Y234D','MUTATION','Y234D',234,234,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18595,'Y234N','MUTATION','Y234N',234,234,'Y',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18596,'R248P','MUTATION','R248P',248,248,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18597,'R248G','MUTATION','R248G',248,248,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18598,'K132Q','MUTATION','K132Q',132,132,'K',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18599,'K132R','MUTATION','K132R',132,132,'K',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18600,'K132T','MUTATION','K132T',132,132,'K',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18601,'K132E','MUTATION','K132E',132,132,'K',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18602,'K132M','MUTATION','K132M',132,132,'K',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18603,'K132N','MUTATION','K132N',132,132,'K',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18604,'H214L','MUTATION','H214L',214,214,'H',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18605,'L130P','MUTATION','L130P',130,130,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18606,'L130R','MUTATION','L130R',130,130,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18607,'L130V','MUTATION','L130V',130,130,'L',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18608,'L130F','MUTATION','L130F',130,130,'L',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18609,'L130H','MUTATION','L130H',130,130,'L',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18610,'K164E','MUTATION','K164E',164,164,'K',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18611,'F109S','MUTATION','F109S',109,109,'F',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18612,'F109C','MUTATION','F109C',109,109,'F',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18613,'F109V','MUTATION','F109V',109,109,'F',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18614,'F109I','MUTATION','F109I',109,109,'F',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18615,'G105R','MUTATION','G105R',105,105,'G',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18616,'G105C','MUTATION','G105C',105,105,'G',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18617,'G105S','MUTATION','G105S',105,105,'G',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18618,'G105D','MUTATION','G105D',105,105,'G',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18619,'G105V','MUTATION','G105V',105,105,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18620,'E271Q','MUTATION','E271Q',271,271,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18621,'E271V','MUTATION','E271V',271,271,'E',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18622,'V172D','MUTATION','V172D',172,172,'V',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18623,'V172G','MUTATION','V172G',172,172,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18624,'T155I','MUTATION','T155I',155,155,'T',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18625,'L257P','MUTATION','L257P',257,257,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18626,'L257Q','MUTATION','L257Q',257,257,'L',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18627,'L257R','MUTATION','L257R',257,257,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18628,'L257V','MUTATION','L257V',257,257,'L',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18629,'N131I','MUTATION','N131I',131,131,'N',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18630,'N131K','MUTATION','N131K',131,131,'N',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18631,'L265P','MUTATION','L265P',265,265,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18632,'L265R','MUTATION','L265R',265,265,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18633,'L265M','MUTATION','L265M',265,265,'L',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18634,'G262V','MUTATION','G262V',262,262,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18635,'F113C','MUTATION','F113C',113,113,'F',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18636,'F113S','MUTATION','F113S',113,113,'F',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18637,'F113V','MUTATION','F113V',113,113,'F',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18638,'F113L','MUTATION','F113L',113,113,'F',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18639,'L111P','MUTATION','L111P',111,111,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18640,'L111Q','MUTATION','L111Q',111,111,'L',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18641,'L111R','MUTATION','L111R',111,111,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18642,'L111M','MUTATION','L111M',111,111,'L',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18643,'D259V','MUTATION','D259V',259,259,'D',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18644,'D259H','MUTATION','D259H',259,259,'D',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18645,'D259Y','MUTATION','D259Y',259,259,'D',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18646,'D259N','MUTATION','D259N',259,259,'D',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18647,'I232T','MUTATION','I232T',232,232,'I',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18648,'I232N','MUTATION','I232N',232,232,'I',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18649,'C135F','MUTATION','C135F',135,135,'C',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18650,'H214P','MUTATION','H214P',214,214,'H',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18651,'C135W','MUTATION','C135W',135,135,'C',NULL,'W','missense_variant',7157,NULL,_binary '\0'),(18652,'D281P','MUTATION','D281P',281,281,'D',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18653,'P151T','MUTATION','P151T',151,151,'P',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18654,'P151F','MUTATION','P151F',151,151,'P',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18655,'P151L','MUTATION','P151L',151,151,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18656,'D281A','MUTATION','D281A',281,281,'D',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18657,'R175G','MUTATION','R175G',175,175,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18658,'D281E','MUTATION','D281E',281,281,'D',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18659,'D281V','MUTATION','D281V',281,281,'D',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18660,'D281G','MUTATION','D281G',281,281,'D',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18661,'D281H','MUTATION','D281H',281,281,'D',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18662,'D281Y','MUTATION','D281Y',281,281,'D',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18663,'D281N','MUTATION','D281N',281,281,'D',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18664,'V216E','MUTATION','V216E',216,216,'V',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18665,'V216G','MUTATION','V216G',216,216,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18666,'V216L','MUTATION','V216L',216,216,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18667,'V216M','MUTATION','V216M',216,216,'V',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18668,'Y126C','MUTATION','Y126C',126,126,'Y',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18669,'Y126S','MUTATION','Y126S',126,126,'Y',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18670,'Y126D','MUTATION','Y126D',126,126,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18671,'Y126H','MUTATION','Y126H',126,126,'Y',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18672,'Y126N','MUTATION','Y126N',126,126,'Y',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18673,'A159P','MUTATION','A159P',159,159,'A',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18674,'A159D','MUTATION','A159D',159,159,'A',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18675,'A159V','MUTATION','A159V',159,159,'A',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18676,'L194P','MUTATION','L194P',194,194,'L',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18677,'L194R','MUTATION','L194R',194,194,'L',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18678,'L194F','MUTATION','L194F',194,194,'L',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18679,'L194H','MUTATION','L194H',194,194,'L',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18680,'F270S','MUTATION','F270S',270,270,'F',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18681,'F270C','MUTATION','F270C',270,270,'F',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18682,'F270V','MUTATION','F270V',270,270,'F',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18683,'F270I','MUTATION','F270I',270,270,'F',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18684,'F270L','MUTATION','F270L',270,270,'F',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18685,'M246R','MUTATION','M246R',246,246,'M',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18686,'M246T','MUTATION','M246T',246,246,'M',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18687,'M246V','MUTATION','M246V',246,246,'M',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18688,'M246I','MUTATION','M246I',246,246,'M',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18689,'M246K','MUTATION','M246K',246,246,'M',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18690,'M246L','MUTATION','M246L',246,246,'M',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18691,'S127P','MUTATION','S127P',127,127,'S',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18692,'S127C','MUTATION','S127C',127,127,'S',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18693,'S127T','MUTATION','S127T',127,127,'S',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18694,'S127F','MUTATION','S127F',127,127,'S',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18695,'S127Y','MUTATION','S127Y',127,127,'S',NULL,'Y','missense_variant',7157,NULL,_binary '\0'),(18696,'Q136P','MUTATION','Q136P',136,136,'Q',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18697,'Q136E','MUTATION','Q136E',136,136,'Q',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18698,'Q136H','MUTATION','Q136H',136,136,'Q',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18699,'Q136L','MUTATION','Q136L',136,136,'Q',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18700,'T125P','MUTATION','T125P',125,125,'T',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18701,'T125A','MUTATION','T125A',125,125,'T',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18702,'T125R','MUTATION','T125R',125,125,'T',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18703,'T125K','MUTATION','T125K',125,125,'T',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18704,'T125M','MUTATION','T125M',125,125,'T',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18705,'E271K','MUTATION','E271K',271,271,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18706,'V172F','MUTATION','V172F',172,172,'V',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18707,'T155P','MUTATION','T155P',155,155,'T',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18708,'T155N','MUTATION','T155N',155,155,'T',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18709,'N131S','MUTATION','N131S',131,131,'N',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18710,'F113P','MUTATION','F113P',113,113,'F',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18711,'D208E','MUTATION','D208E',208,208,'D',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18712,'D208V','MUTATION','D208V',208,208,'D',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18713,'D208G','MUTATION','D208G',208,208,'D',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18714,'D208N','MUTATION','D208N',208,208,'D',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18715,'N247I','MUTATION','N247I',247,247,'N',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18716,'Y107D','MUTATION','Y107D',107,107,'Y',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18717,'I232S','MUTATION','I232S',232,232,'I',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18718,'I232F','MUTATION','I232F',232,232,'I',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18719,'P190R','MUTATION','P190R',190,190,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18720,'P190S','MUTATION','P190S',190,190,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18721,'P190T','MUTATION','P190T',190,190,'P',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18722,'P190L','MUTATION','P190L',190,190,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18723,'V218E','MUTATION','V218E',218,218,'V',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18724,'V218G','MUTATION','V218G',218,218,'V',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18725,'V218L','MUTATION','V218L',218,218,'V',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18726,'V218M','MUTATION','V218M',218,218,'V',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18727,'N268_R273del','MUTATION','N268_R273del',273,268,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18728,'I195S','MUTATION','I195S',195,195,'I',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18729,'I195F','MUTATION','I195F',195,195,'I',NULL,'F','missense_variant',7157,NULL,_binary '\0'),(18730,'I195K','MUTATION','I195K',195,195,'I',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18731,'I195M','MUTATION','I195M',195,195,'I',NULL,'M','missense_variant',7157,NULL,_binary '\0'),(18732,'I195N','MUTATION','I195N',195,195,'I',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18733,'C135G','MUTATION','C135G',135,135,'C',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18734,'P151R','MUTATION','P151R',151,151,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18735,'G360V','MUTATION','G360V',360,360,'G',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18736,'V218dup','MUTATION','V218dup',218,218,'V',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18737,'102_292mis','MUTATION','DNA binding domain missense mutations',292,102,NULL,NULL,NULL,'missense_variant',7157,NULL,_binary '\0'),(18738,'102_292ins','MUTATION','DNA binding domain insertions',292,102,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18739,'102_292del','MUTATION','DNA binding domain deletions',292,102,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18740,'334_346del','MUTATION','Tetramerization domain deletions',346,334,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18741,'334_346ins','MUTATION','Tetramerization domain insertions',346,334,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18742,'V157I','MUTATION','V157I',157,157,'V',NULL,'I','missense_variant',7157,NULL,_binary '\0'),(18743,'A276V','MUTATION','A276V',276,276,'A',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18744,'R290C','MUTATION','R290C',290,290,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18745,'R156P','MUTATION','R156P',156,156,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18746,'R335L','MUTATION','R335L',335,335,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18747,'E336K','MUTATION','E336K',336,336,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18748,'E336Q','MUTATION','E336Q',336,336,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18749,'E336A','MUTATION','E336A',336,336,'E',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18750,'E336G','MUTATION','E336G',336,336,'E',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18751,'E336V','MUTATION','E336V',336,336,'E',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18752,'E336D','MUTATION','E336D',336,336,'E',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18753,'E343K','MUTATION','E343K',343,343,'E',NULL,'K','missense_variant',7157,NULL,_binary '\0'),(18754,'E343Q','MUTATION','E343Q',343,343,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18755,'E343A','MUTATION','E343A',343,343,'E',NULL,'A','missense_variant',7157,NULL,_binary '\0'),(18756,'E343G','MUTATION','E343G',343,343,'E',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18757,'E343V','MUTATION','E343V',343,343,'E',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18758,'E343D','MUTATION','E343D',343,343,'E',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18759,'A347P','MUTATION','A347P',347,347,'A',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18760,'A347S','MUTATION','A347S',347,347,'A',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18761,'A347D','MUTATION','A347D',347,347,'A',NULL,'D','missense_variant',7157,NULL,_binary '\0'),(18762,'A347G','MUTATION','A347G',347,347,'A',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18763,'A347V','MUTATION','A347V',347,347,'A',NULL,'V','missense_variant',7157,NULL,_binary '\0'),(18764,'R248R','MUTATION','R248R',248,248,'R',NULL,'R','synonymous_variant',7157,NULL,_binary '\0'),(18765,'R213L','MUTATION','R213L',213,213,'R',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18766,'R213G','MUTATION','R213G',213,213,'R',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18767,'R213P','MUTATION','R213P',213,213,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18768,'R282R','MUTATION','R282R',282,282,'R',NULL,'R','synonymous_variant',7157,NULL,_binary '\0'),(18769,'C176C','MUTATION','C176C',176,176,'C',NULL,'C','synonymous_variant',7157,NULL,_binary '\0'),(18770,'R196P','MUTATION','R196P',196,196,'R',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18771,'R196Q','MUTATION','R196Q',196,196,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18772,'S241S','MUTATION','S241S',241,241,'S',NULL,'S','synonymous_variant',7157,NULL,_binary '\0'),(18773,'V272V','MUTATION','V272V',272,272,'V',NULL,'V','synonymous_variant',7157,NULL,_binary '\0'),(18774,'V157V','MUTATION','V157V',157,157,'V',NULL,'V','synonymous_variant',7157,NULL,_binary '\0'),(18775,'C242C','MUTATION','C242C',242,242,'C',NULL,'C','synonymous_variant',7157,NULL,_binary '\0'),(18776,'A119P','MUTATION','A119P',119,119,'A',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18777,'A159G','MUTATION','A159G',159,159,'A',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18778,'A161G','MUTATION','A161G',161,161,'A',NULL,'G','missense_variant',7157,NULL,_binary '\0'),(18779,'R379C','MUTATION','R379C',379,379,'R',NULL,'C','missense_variant',7157,NULL,_binary '\0'),(18780,'R379S','MUTATION','R379S',379,379,'R',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18781,'R379H','MUTATION','R379H',379,379,'R',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18782,'Q104H','MUTATION','Q104H',104,104,'Q',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18783,'Q192H','MUTATION','Q192H',192,192,'Q',NULL,'H','missense_variant',7157,NULL,_binary '\0'),(18784,'Q192L','MUTATION','Q192L',192,192,'Q',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18785,'X331_splice','MUTATION','X331_splice',331,331,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18786,'X33_splice','MUTATION','X33_splice',33,33,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18787,'X307_splice','MUTATION','X307_splice',307,307,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18788,'X261_splice','MUTATION','X261_splice',261,261,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18789,'X126_splice','MUTATION','X126_splice',126,126,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18790,'X187_splice','MUTATION','X187_splice',187,187,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18791,'P98L','MUTATION','P98L',98,98,'P',NULL,'L','missense_variant',7157,NULL,_binary '\0'),(18792,'P390S','MUTATION','P390S',390,390,'P',NULL,'S','missense_variant',7157,NULL,_binary '\0'),(18793,'D7N','MUTATION','D7N',7,7,'D',NULL,'N','missense_variant',7157,NULL,_binary '\0'),(18794,'A86T','MUTATION','A86T',86,86,'A',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18795,'P58R','MUTATION','P58R',58,58,'P',NULL,'R','missense_variant',7157,NULL,_binary '\0'),(18796,'E298Q','MUTATION','E298Q',298,298,'E',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18797,'R306Q','MUTATION','R306Q',306,306,'R',NULL,'Q','missense_variant',7157,NULL,_binary '\0'),(18798,'A189P','MUTATION','A189P',189,189,'A',NULL,'P','missense_variant',7157,NULL,_binary '\0'),(18799,'Q144E','MUTATION','Q144E',144,144,'Q',NULL,'E','missense_variant',7157,NULL,_binary '\0'),(18800,'X225_splice','MUTATION','X225_splice',225,225,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18801,'X125_splice','MUTATION','X125_splice',125,125,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18802,'X224_splice','MUTATION','X224_splice',224,224,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18803,'X332_splice','MUTATION','X332_splice',332,332,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18804,'X32_splice','MUTATION','X32_splice',32,32,NULL,NULL,NULL,'splice_region_variant',7157,NULL,_binary '\0'),(18805,'G266T','MUTATION','G266T',266,266,'G',NULL,'T','missense_variant',7157,NULL,_binary '\0'),(18806,'D281_R282insP','MUTATION','D281_R282insP',282,281,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18807,'N263_N268del','MUTATION','N263_N268del',268,263,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18808,'G244_M246delinsV','MUTATION','G244_M246delinsV',246,244,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18809,'S240_C242del','MUTATION','S240_C242del',242,240,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18810,'F270del','MUTATION','F270del',270,270,'F',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18811,'C277dup','MUTATION','C277dup',277,277,'C',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18812,'L252del','MUTATION','L252del',252,252,'L',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18813,'I255dup','MUTATION','I255dup',255,255,'I',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18814,'S269_E271del','MUTATION','S269_E271del',271,269,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18815,'P250_L252del','MUTATION','P250_L252del',252,250,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18816,'C229_T230del','MUTATION','C229_T230del',230,229,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18817,'S269_E271delinsK','MUTATION','S269_E271delinsK',271,269,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18818,'Y234_M237del','MUTATION','Y234_M237del',237,234,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18819,'R282_R283del','MUTATION','R282_R283del',283,282,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18820,'L264_N268del','MUTATION','L264_N268del',268,264,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18821,'N235del','MUTATION','N235del',235,235,'N',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18822,'C238_N239delinsY','MUTATION','C238_N239delinsY',239,238,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18823,'C238_M243del','MUTATION','C238_M243del',243,238,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18824,'T231_I232insTDCTT','MUTATION','T231_I232insTDCTT',232,231,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18825,'V274dup','MUTATION','V274dup',274,274,'V',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18826,'E285_K292del','MUTATION','E285_K292del',292,285,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18827,'L252dup','MUTATION','L252dup',252,252,'L',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18828,'M243_I255del','MUTATION','M243_I255del',255,243,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18829,'S240_S241del','MUTATION','S240_S241del',241,240,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18830,'I254_T256del','MUTATION','I254_T256del',256,254,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18831,'S260_L264dup','MUTATION','S260_L264dup',264,260,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18832,'C275_R282delinsW','MUTATION','C275_R282delinsW',282,275,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18833,'T256_S260del','MUTATION','T256_S260del',260,256,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18834,'G245_P250delinsA','MUTATION','G245_P250delinsA',250,245,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18835,'N235_Y236dup','MUTATION','N235_Y236dup',236,235,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18836,'M237_N239del','MUTATION','M237_N239del',239,237,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18837,'T253_L257del','MUTATION','T253_L257del',257,253,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18838,'T256del','MUTATION','T256del',256,256,'T',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18839,'C229_H233del','MUTATION','C229_H233del',233,229,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18840,'N239dup','MUTATION','N239dup',239,239,'N',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18841,'G262del','MUTATION','G262del',262,262,'G',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18842,'C238_S240del','MUTATION','C238_S240del',240,238,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18843,'V274_G279del','MUTATION','V274_G279del',279,274,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18844,'Y236_M237delinsL','MUTATION','Y236_M237delinsL',237,236,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18845,'N239_S240del','MUTATION','N239_S240del',240,239,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18846,'A276_C277dup','MUTATION','A276_C277dup',277,276,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18847,'L252_I254del','MUTATION','L252_I254del',254,252,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18848,'T253dup','MUTATION','T253dup',253,253,'T',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18849,'V274_C275dup','MUTATION','V274_C275dup',275,274,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18850,'E286_E287del','MUTATION','E286_E287del',287,286,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18851,'A276_E285del','MUTATION','A276_E285del',285,276,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18852,'V272_R273insL','MUTATION','V272_R273insL',273,272,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18853,'S241dup','MUTATION','S241dup',241,241,'S',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18854,'P250_I251delinsL','MUTATION','P250_I251delinsL',251,250,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18855,'Y234del','MUTATION','Y234del',234,234,'Y',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18856,'L265del','MUTATION','L265del',265,265,'L',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18857,'I255del','MUTATION','I255del',255,255,'I',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18858,'N239_S241del','MUTATION','N239_S241del',241,239,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18859,'T231_I232del','MUTATION','T231_I232del',232,231,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18860,'D148_T155dup','MUTATION','D148_T155dup',155,148,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18861,'N131del','MUTATION','N131del',131,131,'N',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18862,'C141_P142del','MUTATION','C141_P142del',142,141,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18863,'P142dup','MUTATION','P142dup',142,142,'P',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18864,'D148del','MUTATION','D148del',148,148,'D',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18865,'K139_P142delinsT','MUTATION','K139_P142delinsT',142,139,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18866,'T150_R158del','MUTATION','T150_R158del',158,150,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18867,'V157_R158del','MUTATION','V157_R158del',158,157,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18868,'R158_A159dup','MUTATION','R158_A159dup',159,158,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18869,'A138_P142del','MUTATION','A138_P142del',142,138,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18870,'A161_I162del','MUTATION','A161_I162del',162,161,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18871,'A159del','MUTATION','A159del',159,159,'A',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18872,'D148_T150del','MUTATION','D148_T150del',150,148,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18873,'V147_D148dup','MUTATION','V147_D148dup',148,147,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18874,'L130del','MUTATION','L130del',130,130,'L',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18875,'Q136del','MUTATION','Q136del',136,136,'Q',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18876,'R158_A159del','MUTATION','R158_A159del',159,158,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18877,'A138_L145del','MUTATION','A138_L145del',145,138,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18878,'V143_P153del','MUTATION','V143_P153del',153,143,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18879,'T155_R158del','MUTATION','T155_R158del',158,155,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18880,'L130_A138dup','MUTATION','L130_A138dup',138,130,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18881,'P153_G154del','MUTATION','P153_G154del',154,153,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18882,'T155_R156del','MUTATION','T155_R156del',156,155,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18883,'R156del','MUTATION','R156del',156,156,'R',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18884,'Q136_C141del','MUTATION','Q136_C141del',141,136,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18885,'I162dup','MUTATION','I162dup',162,162,'I',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18886,'P128del','MUTATION','P128del',128,128,'P',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18887,'I162_Y163delinsN','MUTATION','I162_Y163delinsN',163,162,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18888,'C135_L137delinsW','MUTATION','C135_L137delinsW',137,135,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18889,'V157_R158dup','MUTATION','V157_R158dup',158,157,NULL,NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18890,'P128_L130del','MUTATION','P128_L130del',130,128,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18891,'G154_T155del','MUTATION','G154_T155del',155,154,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18892,'T155_A161del','MUTATION','T155_A161del',161,155,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18893,'P153del','MUTATION','P153del',153,153,'P',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18894,'H179del','MUTATION','H179del',179,179,'H',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18895,'H178_S183del','MUTATION','H178_S183del',183,178,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18896,'C176dup','MUTATION','C176dup',176,176,'C',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18897,'P177_C182del','MUTATION','P177_C182del',182,177,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18898,'C176_H179del','MUTATION','C176_H179del',179,176,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18899,'E180_D184del','MUTATION','E180_D184del',184,180,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18900,'P177_H178del','MUTATION','P177_H178del',178,177,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18901,'S183_G187delinsC','MUTATION','S183_G187delinsC',187,183,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18902,'H179_D184del','MUTATION','H179_D184del',184,179,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18903,'V173dup','MUTATION','V173dup',173,173,'V',NULL,NULL,'inframe_insertion',7157,NULL,_binary '\0'),(18904,'V173_R175del','MUTATION','V173_R175del',175,173,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18905,'E204_L206del','MUTATION','E204_L206del',206,204,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18906,'Q192del','MUTATION','Q192del',192,192,'Q',NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18907,'V197_N200del','MUTATION','V197_N200del',200,197,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18908,'G199_E204del','MUTATION','G199_E204del',204,199,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18909,'R202_L206del','MUTATION','R202_L206del',206,202,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18910,'I195_G199del','MUTATION','I195_G199del',199,195,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0'),(18911,'P191_L201del','MUTATION','P191_L201del',201,191,NULL,NULL,NULL,'inframe_deletion',7157,NULL,_binary '\0');
/*!40000 ALTER TABLE `alteration` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `alteration_reference_genome`
--

DROP TABLE IF EXISTS `alteration_reference_genome`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `alteration_reference_genome` (
  `alteration_id` int NOT NULL,
  `reference_genome` varchar(10) DEFAULT NULL,
  KEY `FKE8AFDC0BAAD5975` (`alteration_id`),
  CONSTRAINT `FKE8AFDC0BAAD5975` FOREIGN KEY (`alteration_id`) REFERENCES `alteration` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `alteration_reference_genome`
--

LOCK TABLES `alteration_reference_genome` WRITE;
/*!40000 ALTER TABLE `alteration_reference_genome` DISABLE KEYS */;
INSERT INTO `alteration_reference_genome` VALUES (1474,'GRCh37'),(1474,'GRCh38'),(1475,'GRCh37'),(1475,'GRCh38'),(1476,'GRCh37'),(1476,'GRCh38'),(1477,'GRCh37'),(1477,'GRCh38'),(1478,'GRCh37'),(1478,'GRCh38'),(1479,'GRCh37'),(1479,'GRCh38'),(1480,'GRCh37'),(1480,'GRCh38'),(1481,'GRCh37'),(1481,'GRCh38'),(1482,'GRCh37'),(1482,'GRCh38'),(1483,'GRCh37'),(1483,'GRCh38'),(1484,'GRCh37'),(1484,'GRCh38'),(1485,'GRCh37'),(1485,'GRCh38'),(1486,'GRCh37'),(1486,'GRCh38'),(1487,'GRCh37'),(1487,'GRCh38'),(1488,'GRCh37'),(1488,'GRCh38'),(1489,'GRCh37'),(1489,'GRCh38'),(1490,'GRCh37'),(1490,'GRCh38'),(1491,'GRCh37'),(1491,'GRCh38'),(1492,'GRCh37'),(1492,'GRCh38'),(1493,'GRCh37'),(1493,'GRCh38'),(1494,'GRCh37'),(1494,'GRCh38'),(1495,'GRCh37'),(1495,'GRCh38'),(1496,'GRCh37'),(1496,'GRCh38'),(1497,'GRCh37'),(1497,'GRCh38'),(1498,'GRCh37'),(1498,'GRCh38'),(1499,'GRCh37'),(1499,'GRCh38'),(1500,'GRCh37'),(1500,'GRCh38'),(1501,'GRCh37'),(1501,'GRCh38'),(1502,'GRCh37'),(1502,'GRCh38'),(1503,'GRCh37'),(1503,'GRCh38'),(1504,'GRCh37'),(1504,'GRCh38'),(1505,'GRCh37'),(1505,'GRCh38'),(1506,'GRCh37'),(1506,'GRCh38'),(1507,'GRCh37'),(1507,'GRCh38'),(1508,'GRCh37'),(1508,'GRCh38'),(1509,'GRCh37'),(1509,'GRCh38'),(1510,'GRCh37'),(1510,'GRCh38'),(1511,'GRCh37'),(1511,'GRCh38'),(1512,'GRCh37'),(1512,'GRCh38'),(1513,'GRCh37'),(1513,'GRCh38'),(1514,'GRCh37'),(1514,'GRCh38'),(1515,'GRCh37'),(1515,'GRCh38'),(1516,'GRCh37'),(1516,'GRCh38'),(1517,'GRCh37'),(1517,'GRCh38'),(1518,'GRCh37'),(1518,'GRCh38'),(1519,'GRCh37'),(1519,'GRCh38'),(1520,'GRCh37'),(1520,'GRCh38'),(1521,'GRCh37'),(1521,'GRCh38'),(1522,'GRCh37'),(1522,'GRCh38'),(1523,'GRCh37'),(1523,'GRCh38'),(1524,'GRCh37'),(1524,'GRCh38'),(1525,'GRCh37'),(1525,'GRCh38'),(1526,'GRCh37'),(1526,'GRCh38'),(1527,'GRCh37'),(1527,'GRCh38'),(1528,'GRCh37'),(1528,'GRCh38'),(1529,'GRCh37'),(1529,'GRCh38'),(1530,'GRCh37'),(1530,'GRCh38'),(1531,'GRCh37'),(1531,'GRCh38'),(1532,'GRCh37'),(1532,'GRCh38'),(1533,'GRCh37'),(1533,'GRCh38'),(1534,'GRCh37'),(1534,'GRCh38'),(1535,'GRCh37'),(1535,'GRCh38'),(1536,'GRCh37'),(1536,'GRCh38'),(1537,'GRCh37'),(1537,'GRCh38'),(1538,'GRCh37'),(1538,'GRCh38'),(1539,'GRCh37'),(1539,'GRCh38'),(1540,'GRCh37'),(1540,'GRCh38'),(1541,'GRCh37'),(1541,'GRCh38'),(1542,'GRCh37'),(1542,'GRCh38'),(1543,'GRCh37'),(1543,'GRCh38'),(1544,'GRCh37'),(1544,'GRCh38'),(1545,'GRCh37'),(1545,'GRCh38'),(1546,'GRCh37'),(1546,'GRCh38'),(1547,'GRCh37'),(1547,'GRCh38'),(1548,'GRCh37'),(1548,'GRCh38'),(1549,'GRCh37'),(1549,'GRCh38'),(1550,'GRCh37'),(1550,'GRCh38'),(1551,'GRCh37'),(1551,'GRCh38'),(1552,'GRCh37'),(1552,'GRCh38'),(1553,'GRCh37'),(1553,'GRCh38'),(1554,'GRCh37'),(1554,'GRCh38'),(1555,'GRCh37'),(1555,'GRCh38'),(1556,'GRCh37'),(1556,'GRCh38'),(1557,'GRCh37'),(1557,'GRCh38'),(1558,'GRCh37'),(1558,'GRCh38'),(1559,'GRCh37'),(1559,'GRCh38'),(1560,'GRCh37'),(1560,'GRCh38'),(1561,'GRCh37'),(1561,'GRCh38'),(1562,'GRCh37'),(1562,'GRCh38'),(1563,'GRCh37'),(1563,'GRCh38'),(1564,'GRCh37'),(1564,'GRCh38'),(1565,'GRCh37'),(1565,'GRCh38'),(1566,'GRCh37'),(1566,'GRCh38'),(1567,'GRCh37'),(1567,'GRCh38'),(1568,'GRCh37'),(1568,'GRCh38'),(1569,'GRCh37'),(1569,'GRCh38'),(1570,'GRCh37'),(1570,'GRCh38'),(1571,'GRCh37'),(1571,'GRCh38'),(1572,'GRCh37'),(1572,'GRCh38'),(1573,'GRCh37'),(1573,'GRCh38'),(1574,'GRCh37'),(1574,'GRCh38'),(1575,'GRCh37'),(1575,'GRCh38'),(1576,'GRCh37'),(1576,'GRCh38'),(1577,'GRCh37'),(1577,'GRCh38'),(1578,'GRCh37'),(1578,'GRCh38'),(1579,'GRCh37'),(1579,'GRCh38'),(1580,'GRCh37'),(1580,'GRCh38'),(1581,'GRCh37'),(1581,'GRCh38'),(1582,'GRCh37'),(1582,'GRCh38'),(1583,'GRCh37'),(1583,'GRCh38'),(1584,'GRCh37'),(1584,'GRCh38'),(1585,'GRCh37'),(1585,'GRCh38'),(1586,'GRCh37'),(1586,'GRCh38'),(1587,'GRCh37'),(1587,'GRCh38'),(1588,'GRCh37'),(1588,'GRCh38'),(1589,'GRCh37'),(1589,'GRCh38'),(1590,'GRCh37'),(1590,'GRCh38'),(1591,'GRCh37'),(1591,'GRCh38'),(1592,'GRCh37'),(1592,'GRCh38'),(1593,'GRCh37'),(1593,'GRCh38'),(1594,'GRCh37'),(1594,'GRCh38'),(1595,'GRCh37'),(1595,'GRCh38'),(1596,'GRCh37'),(1596,'GRCh38'),(1597,'GRCh37'),(1597,'GRCh38'),(1598,'GRCh37'),(1598,'GRCh38'),(1599,'GRCh37'),(1599,'GRCh38'),(1600,'GRCh37'),(1600,'GRCh38'),(1601,'GRCh37'),(1601,'GRCh38'),(1602,'GRCh37'),(1602,'GRCh38'),(1603,'GRCh37'),(1603,'GRCh38'),(1604,'GRCh37'),(1604,'GRCh38'),(1605,'GRCh37'),(1605,'GRCh38'),(1606,'GRCh37'),(1606,'GRCh38'),(1607,'GRCh37'),(1607,'GRCh38'),(1608,'GRCh37'),(1608,'GRCh38'),(1609,'GRCh37'),(1609,'GRCh38'),(1610,'GRCh37'),(1610,'GRCh38'),(1611,'GRCh37'),(1611,'GRCh38'),(1612,'GRCh37'),(1612,'GRCh38'),(1613,'GRCh37'),(1613,'GRCh38'),(1614,'GRCh37'),(1614,'GRCh38'),(1615,'GRCh37'),(1615,'GRCh38'),(1616,'GRCh37'),(1616,'GRCh38'),(1617,'GRCh37'),(1617,'GRCh38'),(1618,'GRCh37'),(1618,'GRCh38'),(1619,'GRCh37'),(1619,'GRCh38'),(1620,'GRCh37'),(1620,'GRCh38'),(1621,'GRCh37'),(1621,'GRCh38'),(1622,'GRCh37'),(1622,'GRCh38'),(1623,'GRCh37'),(1623,'GRCh38'),(1624,'GRCh37'),(1624,'GRCh38'),(1625,'GRCh37'),(1625,'GRCh38'),(1626,'GRCh37'),(1626,'GRCh38'),(1627,'GRCh37'),(1627,'GRCh38'),(1628,'GRCh37'),(1628,'GRCh38'),(1629,'GRCh37'),(1629,'GRCh38'),(1630,'GRCh37'),(1630,'GRCh38'),(1631,'GRCh37'),(1631,'GRCh38'),(1632,'GRCh37'),(1632,'GRCh38'),(1633,'GRCh37'),(1633,'GRCh38'),(1634,'GRCh37'),(1634,'GRCh38'),(1635,'GRCh37'),(1635,'GRCh38'),(1636,'GRCh37'),(1636,'GRCh38'),(1637,'GRCh37'),(1637,'GRCh38'),(1638,'GRCh37'),(1638,'GRCh38'),(1639,'GRCh37'),(1639,'GRCh38'),(1640,'GRCh37'),(1640,'GRCh38'),(1641,'GRCh37'),(1641,'GRCh38'),(1642,'GRCh37'),(1642,'GRCh38'),(1643,'GRCh37'),(1643,'GRCh38'),(1644,'GRCh37'),(1644,'GRCh38'),(1645,'GRCh37'),(1645,'GRCh38'),(1646,'GRCh37'),(1646,'GRCh38'),(1647,'GRCh37'),(1647,'GRCh38'),(1648,'GRCh37'),(1648,'GRCh38'),(1649,'GRCh37'),(1649,'GRCh38'),(1650,'GRCh37'),(1650,'GRCh38'),(1651,'GRCh37'),(1651,'GRCh38'),(1652,'GRCh37'),(1652,'GRCh38'),(1653,'GRCh37'),(1653,'GRCh38'),(1654,'GRCh37'),(1654,'GRCh38'),(1655,'GRCh37'),(1655,'GRCh38'),(1656,'GRCh37'),(1656,'GRCh38'),(1657,'GRCh37'),(1657,'GRCh38'),(1658,'GRCh37'),(1658,'GRCh38'),(1659,'GRCh37'),(1659,'GRCh38'),(1660,'GRCh37'),(1660,'GRCh38'),(1661,'GRCh37'),(1661,'GRCh38'),(1662,'GRCh37'),(1662,'GRCh38'),(1663,'GRCh37'),(1663,'GRCh38'),(1664,'GRCh37'),(1664,'GRCh38'),(1665,'GRCh37'),(1665,'GRCh38'),(1666,'GRCh37'),(1666,'GRCh38'),(1667,'GRCh37'),(1667,'GRCh38'),(1668,'GRCh37'),(1668,'GRCh38'),(1669,'GRCh37'),(1669,'GRCh38'),(1670,'GRCh37'),(1670,'GRCh38'),(1671,'GRCh37'),(1671,'GRCh38'),(1672,'GRCh37'),(1672,'GRCh38'),(1673,'GRCh37'),(1673,'GRCh38'),(1674,'GRCh37'),(1674,'GRCh38'),(1675,'GRCh37'),(1675,'GRCh38'),(1676,'GRCh37'),(1676,'GRCh38'),(1677,'GRCh37'),(1677,'GRCh38'),(1678,'GRCh37'),(1678,'GRCh38'),(1679,'GRCh37'),(1679,'GRCh38'),(1680,'GRCh37'),(1680,'GRCh38'),(1681,'GRCh37'),(1681,'GRCh38'),(1682,'GRCh37'),(1682,'GRCh38'),(1683,'GRCh37'),(1683,'GRCh38'),(1684,'GRCh37'),(1684,'GRCh38'),(1685,'GRCh37'),(1685,'GRCh38'),(1686,'GRCh37'),(1686,'GRCh38'),(1687,'GRCh37'),(1687,'GRCh38'),(1688,'GRCh37'),(1688,'GRCh38'),(1689,'GRCh37'),(1689,'GRCh38'),(1690,'GRCh37'),(1690,'GRCh38'),(1691,'GRCh37'),(1691,'GRCh38'),(1692,'GRCh37'),(1692,'GRCh38'),(1693,'GRCh37'),(1693,'GRCh38'),(1694,'GRCh37'),(1694,'GRCh38'),(1695,'GRCh37'),(1695,'GRCh38'),(1696,'GRCh37'),(1696,'GRCh38'),(1697,'GRCh37'),(1697,'GRCh38'),(1698,'GRCh37'),(1698,'GRCh38'),(1699,'GRCh37'),(1699,'GRCh38'),(1700,'GRCh37'),(1700,'GRCh38'),(1701,'GRCh37'),(1701,'GRCh38'),(1702,'GRCh37'),(1702,'GRCh38'),(1703,'GRCh37'),(1703,'GRCh38'),(1704,'GRCh37'),(1704,'GRCh38'),(1705,'GRCh37'),(1705,'GRCh38'),(1706,'GRCh37'),(1706,'GRCh38'),(1707,'GRCh37'),(1707,'GRCh38'),(1708,'GRCh37'),(1708,'GRCh38'),(1709,'GRCh37'),(1709,'GRCh38'),(1710,'GRCh37'),(1710,'GRCh38'),(1711,'GRCh37'),(1711,'GRCh38'),(1712,'GRCh37'),(1712,'GRCh38'),(1713,'GRCh37'),(1713,'GRCh38'),(1714,'GRCh37'),(1714,'GRCh38'),(1715,'GRCh37'),(1715,'GRCh38'),(1716,'GRCh37'),(1716,'GRCh38'),(1717,'GRCh37'),(1717,'GRCh38'),(1718,'GRCh37'),(1718,'GRCh38'),(1719,'GRCh37'),(1719,'GRCh38'),(1720,'GRCh37'),(1720,'GRCh38'),(1721,'GRCh37'),(1721,'GRCh38'),(1722,'GRCh37'),(1722,'GRCh38'),(1723,'GRCh37'),(1723,'GRCh38'),(1724,'GRCh37'),(1724,'GRCh38'),(1725,'GRCh37'),(1725,'GRCh38'),(1726,'GRCh37'),(1726,'GRCh38'),(1727,'GRCh37'),(1727,'GRCh38'),(1728,'GRCh37'),(1728,'GRCh38'),(1729,'GRCh37'),(1729,'GRCh38'),(1730,'GRCh37'),(1730,'GRCh38'),(1731,'GRCh37'),(1731,'GRCh38'),(1732,'GRCh37'),(1732,'GRCh38'),(1733,'GRCh37'),(1733,'GRCh38'),(1734,'GRCh37'),(1734,'GRCh38'),(1735,'GRCh37'),(1735,'GRCh38'),(1736,'GRCh37'),(1736,'GRCh38'),(1737,'GRCh37'),(1737,'GRCh38'),(1738,'GRCh37'),(1738,'GRCh38'),(1739,'GRCh37'),(1739,'GRCh38'),(1740,'GRCh37'),(1740,'GRCh38'),(1741,'GRCh37'),(1741,'GRCh38'),(1742,'GRCh37'),(1742,'GRCh38'),(1743,'GRCh37'),(1743,'GRCh38'),(1744,'GRCh37'),(1744,'GRCh38'),(1745,'GRCh37'),(1745,'GRCh38'),(1746,'GRCh37'),(1746,'GRCh38'),(1747,'GRCh37'),(1747,'GRCh38'),(1748,'GRCh37'),(1748,'GRCh38'),(1749,'GRCh37'),(1749,'GRCh38'),(1750,'GRCh37'),(1750,'GRCh38'),(1751,'GRCh37'),(1751,'GRCh38'),(1752,'GRCh37'),(1752,'GRCh38'),(1753,'GRCh37'),(1753,'GRCh38'),(1754,'GRCh37'),(1754,'GRCh38'),(1755,'GRCh37'),(1755,'GRCh38'),(1756,'GRCh37'),(1756,'GRCh38'),(1757,'GRCh37'),(1757,'GRCh38'),(1758,'GRCh37'),(1758,'GRCh38'),(1759,'GRCh37'),(1759,'GRCh38'),(1760,'GRCh37'),(1760,'GRCh38'),(1761,'GRCh37'),(1761,'GRCh38'),(1762,'GRCh37'),(1762,'GRCh38'),(1763,'GRCh37'),(1763,'GRCh38'),(1764,'GRCh37'),(1764,'GRCh38'),(1765,'GRCh37'),(1765,'GRCh38'),(1766,'GRCh37'),(1766,'GRCh38'),(1767,'GRCh37'),(1767,'GRCh38'),(1768,'GRCh37'),(1768,'GRCh38'),(1769,'GRCh37'),(1769,'GRCh38'),(1770,'GRCh37'),(1770,'GRCh38'),(1771,'GRCh37'),(1771,'GRCh38'),(1772,'GRCh37'),(1772,'GRCh38'),(1773,'GRCh37'),(1773,'GRCh38'),(1774,'GRCh37'),(1774,'GRCh38'),(1775,'GRCh37'),(1775,'GRCh38'),(1776,'GRCh37'),(1776,'GRCh38'),(1777,'GRCh37'),(1777,'GRCh38'),(1778,'GRCh37'),(1778,'GRCh38'),(1779,'GRCh37'),(1779,'GRCh38'),(1780,'GRCh37'),(1780,'GRCh38'),(1781,'GRCh37'),(1781,'GRCh38'),(1782,'GRCh37'),(1782,'GRCh38'),(1783,'GRCh37'),(1783,'GRCh38'),(1784,'GRCh37'),(1784,'GRCh38'),(1785,'GRCh37'),(1785,'GRCh38'),(1786,'GRCh37'),(1786,'GRCh38'),(1787,'GRCh37'),(1787,'GRCh38'),(1788,'GRCh37'),(1788,'GRCh38'),(1789,'GRCh37'),(1789,'GRCh38'),(1790,'GRCh37'),(1790,'GRCh38'),(1791,'GRCh37'),(1791,'GRCh38'),(1792,'GRCh37'),(1792,'GRCh38'),(1793,'GRCh37'),(1793,'GRCh38'),(1794,'GRCh37'),(1794,'GRCh38'),(1795,'GRCh37'),(1795,'GRCh38'),(1796,'GRCh37'),(1796,'GRCh38'),(1797,'GRCh37'),(1797,'GRCh38'),(1798,'GRCh37'),(1798,'GRCh38'),(1799,'GRCh37'),(1799,'GRCh38'),(1800,'GRCh37'),(1800,'GRCh38'),(1801,'GRCh37'),(1801,'GRCh38'),(1802,'GRCh37'),(1802,'GRCh38'),(1803,'GRCh37'),(1803,'GRCh38'),(1804,'GRCh37'),(1804,'GRCh38'),(1805,'GRCh37'),(1805,'GRCh38'),(1806,'GRCh37'),(1806,'GRCh38'),(1807,'GRCh37'),(1807,'GRCh38'),(1808,'GRCh37'),(1808,'GRCh38'),(1809,'GRCh37'),(1809,'GRCh38'),(1810,'GRCh37'),(1810,'GRCh38'),(1811,'GRCh37'),(1811,'GRCh38'),(1812,'GRCh37'),(1812,'GRCh38'),(1813,'GRCh37'),(1813,'GRCh38'),(1814,'GRCh37'),(1814,'GRCh38'),(1815,'GRCh37'),(1815,'GRCh38'),(1816,'GRCh37'),(1816,'GRCh38'),(1817,'GRCh37'),(1817,'GRCh38'),(1818,'GRCh37'),(1818,'GRCh38'),(1819,'GRCh37'),(1819,'GRCh38'),(1820,'GRCh37'),(1820,'GRCh38'),(1821,'GRCh37'),(1821,'GRCh38'),(1822,'GRCh37'),(1822,'GRCh38'),(1823,'GRCh37'),(1823,'GRCh38'),(1824,'GRCh37'),(1824,'GRCh38'),(1825,'GRCh37'),(1825,'GRCh38'),(1826,'GRCh37'),(1826,'GRCh38'),(1827,'GRCh37'),(1827,'GRCh38'),(1828,'GRCh37'),(1828,'GRCh38'),(1829,'GRCh37'),(1829,'GRCh38'),(1830,'GRCh37'),(1830,'GRCh38'),(1831,'GRCh37'),(1831,'GRCh38'),(1832,'GRCh37'),(1832,'GRCh38'),(1833,'GRCh37'),(1833,'GRCh38'),(1834,'GRCh37'),(1834,'GRCh38'),(1835,'GRCh37'),(1835,'GRCh38'),(1836,'GRCh37'),(1836,'GRCh38'),(1837,'GRCh37'),(1837,'GRCh38'),(1838,'GRCh37'),(1838,'GRCh38'),(1839,'GRCh37'),(1839,'GRCh38'),(1840,'GRCh37'),(1840,'GRCh38'),(1841,'GRCh37'),(1841,'GRCh38'),(1842,'GRCh37'),(1842,'GRCh38'),(1843,'GRCh37'),(1843,'GRCh38'),(1844,'GRCh37'),(1844,'GRCh38'),(1845,'GRCh37'),(1845,'GRCh38'),(1846,'GRCh37'),(1846,'GRCh38'),(1847,'GRCh37'),(1847,'GRCh38'),(1848,'GRCh37'),(1848,'GRCh38'),(1849,'GRCh37'),(1849,'GRCh38'),(1850,'GRCh37'),(1850,'GRCh38'),(1851,'GRCh37'),(1851,'GRCh38'),(1852,'GRCh37'),(1852,'GRCh38'),(1853,'GRCh37'),(1853,'GRCh38'),(1854,'GRCh37'),(1854,'GRCh38'),(1855,'GRCh37'),(1855,'GRCh38'),(1856,'GRCh37'),(1856,'GRCh38'),(1857,'GRCh37'),(1857,'GRCh38'),(1858,'GRCh37'),(1858,'GRCh38'),(1859,'GRCh37'),(1859,'GRCh38'),(1860,'GRCh37'),(1860,'GRCh38'),(1861,'GRCh37'),(1861,'GRCh38'),(1862,'GRCh37'),(1862,'GRCh38'),(1863,'GRCh37'),(1863,'GRCh38'),(1864,'GRCh37'),(1864,'GRCh38'),(1865,'GRCh37'),(1865,'GRCh38'),(1866,'GRCh37'),(1866,'GRCh38'),(1867,'GRCh37'),(1867,'GRCh38'),(1868,'GRCh37'),(1868,'GRCh38'),(1869,'GRCh37'),(1869,'GRCh38'),(1870,'GRCh37'),(1870,'GRCh38'),(1871,'GRCh37'),(1871,'GRCh38'),(1872,'GRCh37'),(1872,'GRCh38'),(1873,'GRCh37'),(1873,'GRCh38'),(1874,'GRCh37'),(1874,'GRCh38'),(1875,'GRCh37'),(1875,'GRCh38'),(1876,'GRCh37'),(1876,'GRCh38'),(1877,'GRCh37'),(1877,'GRCh38'),(1878,'GRCh37'),(1878,'GRCh38'),(1879,'GRCh37'),(1879,'GRCh38'),(1880,'GRCh37'),(1880,'GRCh38'),(1881,'GRCh37'),(1881,'GRCh38'),(1882,'GRCh37'),(1882,'GRCh38'),(1883,'GRCh37'),(1883,'GRCh38'),(1884,'GRCh37'),(1884,'GRCh38'),(1885,'GRCh37'),(1885,'GRCh38'),(1886,'GRCh37'),(1886,'GRCh38'),(1887,'GRCh37'),(1887,'GRCh38'),(1888,'GRCh37'),(1888,'GRCh38'),(1889,'GRCh37'),(1889,'GRCh38'),(1890,'GRCh37'),(1890,'GRCh38'),(1891,'GRCh37'),(1891,'GRCh38'),(1892,'GRCh37'),(1892,'GRCh38'),(1893,'GRCh37'),(1893,'GRCh38'),(1894,'GRCh37'),(1894,'GRCh38'),(1895,'GRCh37'),(1895,'GRCh38'),(1896,'GRCh37'),(1896,'GRCh38'),(1897,'GRCh37'),(1897,'GRCh38'),(1898,'GRCh37'),(1898,'GRCh38'),(1899,'GRCh37'),(1899,'GRCh38'),(1900,'GRCh37'),(1900,'GRCh38'),(1901,'GRCh37'),(1901,'GRCh38'),(1902,'GRCh37'),(1902,'GRCh38'),(1903,'GRCh37'),(1903,'GRCh38'),(1904,'GRCh37'),(1904,'GRCh38'),(1905,'GRCh37'),(1905,'GRCh38'),(1906,'GRCh37'),(1906,'GRCh38'),(1907,'GRCh37'),(1907,'GRCh38'),(1908,'GRCh37'),(1908,'GRCh38'),(1909,'GRCh37'),(1909,'GRCh38'),(1910,'GRCh37'),(1910,'GRCh38'),(1911,'GRCh37'),(1911,'GRCh38'),(1912,'GRCh37'),(1912,'GRCh38'),(1913,'GRCh37'),(1913,'GRCh38'),(1914,'GRCh37'),(1914,'GRCh38'),(1915,'GRCh37'),(1915,'GRCh38'),(1916,'GRCh37'),(1916,'GRCh38'),(1917,'GRCh37'),(1917,'GRCh38'),(1918,'GRCh37'),(1918,'GRCh38'),(1919,'GRCh37'),(1919,'GRCh38'),(1920,'GRCh37'),(1920,'GRCh38'),(1921,'GRCh37'),(1921,'GRCh38'),(1922,'GRCh37'),(1922,'GRCh38'),(1923,'GRCh37'),(1923,'GRCh38'),(1924,'GRCh37'),(1924,'GRCh38'),(1925,'GRCh37'),(1925,'GRCh38'),(1926,'GRCh37'),(1926,'GRCh38'),(1927,'GRCh37'),(1927,'GRCh38'),(1928,'GRCh37'),(1928,'GRCh38'),(1929,'GRCh37'),(1929,'GRCh38'),(1930,'GRCh37'),(1930,'GRCh38'),(1931,'GRCh37'),(1931,'GRCh38'),(1932,'GRCh37'),(1932,'GRCh38'),(1933,'GRCh37'),(1933,'GRCh38'),(1934,'GRCh37'),(1934,'GRCh38'),(1935,'GRCh37'),(1935,'GRCh38'),(1936,'GRCh37'),(1936,'GRCh38'),(1937,'GRCh37'),(1937,'GRCh38'),(1938,'GRCh37'),(1938,'GRCh38'),(1939,'GRCh37'),(1939,'GRCh38'),(1940,'GRCh37'),(1940,'GRCh38'),(1941,'GRCh37'),(1941,'GRCh38'),(1942,'GRCh37'),(1942,'GRCh38'),(1943,'GRCh37'),(1943,'GRCh38'),(1944,'GRCh37'),(1944,'GRCh38'),(1945,'GRCh37'),(1945,'GRCh38'),(1946,'GRCh37'),(1946,'GRCh38'),(1947,'GRCh37'),(1947,'GRCh38'),(1948,'GRCh37'),(1948,'GRCh38'),(1949,'GRCh37'),(1949,'GRCh38'),(1950,'GRCh37'),(1950,'GRCh38'),(1951,'GRCh37'),(1951,'GRCh38'),(1952,'GRCh37'),(1952,'GRCh38'),(1953,'GRCh37'),(1953,'GRCh38'),(1954,'GRCh37'),(1954,'GRCh38'),(1955,'GRCh37'),(1955,'GRCh38'),(1956,'GRCh37'),(1956,'GRCh38'),(1957,'GRCh37'),(1957,'GRCh38'),(1958,'GRCh37'),(1958,'GRCh38'),(1959,'GRCh37'),(1959,'GRCh38'),(1960,'GRCh37'),(1960,'GRCh38'),(1961,'GRCh37'),(1961,'GRCh38'),(1962,'GRCh37'),(1962,'GRCh38'),(1963,'GRCh37'),(1963,'GRCh38'),(1964,'GRCh37'),(1964,'GRCh38'),(1965,'GRCh37'),(1965,'GRCh38'),(1966,'GRCh37'),(1966,'GRCh38'),(1967,'GRCh37'),(1967,'GRCh38'),(1968,'GRCh37'),(1968,'GRCh38'),(1969,'GRCh37'),(1969,'GRCh38'),(1970,'GRCh37'),(1970,'GRCh38'),(1971,'GRCh37'),(1971,'GRCh38'),(1972,'GRCh37'),(1972,'GRCh38'),(1973,'GRCh37'),(1973,'GRCh38'),(1974,'GRCh37'),(1974,'GRCh38'),(1975,'GRCh37'),(1975,'GRCh38'),(1976,'GRCh37'),(1976,'GRCh38'),(1977,'GRCh37'),(1977,'GRCh38'),(1978,'GRCh37'),(1978,'GRCh38'),(1979,'GRCh37'),(1979,'GRCh38'),(1980,'GRCh37'),(1980,'GRCh38'),(1981,'GRCh37'),(1981,'GRCh38'),(1982,'GRCh37'),(1982,'GRCh38'),(1983,'GRCh37'),(1983,'GRCh38'),(1984,'GRCh37'),(1984,'GRCh38'),(1985,'GRCh37'),(1985,'GRCh38'),(1986,'GRCh37'),(1986,'GRCh38'),(1987,'GRCh37'),(1987,'GRCh38'),(1988,'GRCh37'),(1988,'GRCh38'),(1989,'GRCh37'),(1989,'GRCh38'),(1990,'GRCh37'),(1990,'GRCh38'),(1991,'GRCh37'),(1991,'GRCh38'),(1992,'GRCh37'),(1992,'GRCh38'),(1993,'GRCh37'),(1993,'GRCh38'),(1994,'GRCh37'),(1994,'GRCh38'),(1995,'GRCh37'),(1995,'GRCh38'),(1996,'GRCh37'),(1996,'GRCh38'),(1997,'GRCh37'),(1997,'GRCh38'),(1998,'GRCh37'),(1998,'GRCh38'),(1999,'GRCh37'),(1999,'GRCh38'),(2000,'GRCh37'),(2000,'GRCh38'),(2001,'GRCh37'),(2001,'GRCh38'),(2002,'GRCh37'),(2002,'GRCh38'),(2003,'GRCh37'),(2003,'GRCh38'),(2004,'GRCh37'),(2004,'GRCh38'),(2005,'GRCh37'),(2005,'GRCh38'),(2006,'GRCh37'),(2006,'GRCh38'),(2007,'GRCh37'),(2007,'GRCh38'),(2008,'GRCh37'),(2008,'GRCh38'),(2009,'GRCh37'),(2009,'GRCh38'),(2010,'GRCh37'),(2010,'GRCh38'),(2011,'GRCh37'),(2011,'GRCh38'),(2012,'GRCh37'),(2012,'GRCh38'),(2013,'GRCh37'),(2013,'GRCh38'),(2014,'GRCh37'),(2014,'GRCh38'),(2015,'GRCh37'),(2015,'GRCh38'),(2016,'GRCh37'),(2016,'GRCh38'),(2017,'GRCh37'),(2017,'GRCh38'),(2018,'GRCh37'),(2018,'GRCh38'),(2019,'GRCh37'),(2019,'GRCh38'),(2020,'GRCh37'),(2020,'GRCh38'),(2021,'GRCh37'),(2021,'GRCh38'),(2022,'GRCh37'),(2022,'GRCh38'),(2023,'GRCh37'),(2023,'GRCh38'),(2024,'GRCh37'),(2024,'GRCh38'),(2025,'GRCh37'),(2025,'GRCh38'),(2026,'GRCh37'),(2026,'GRCh38'),(2027,'GRCh37'),(2027,'GRCh38'),(2028,'GRCh37'),(2028,'GRCh38'),(2029,'GRCh37'),(2029,'GRCh38'),(2030,'GRCh37'),(2030,'GRCh38'),(2031,'GRCh37'),(2031,'GRCh38'),(2032,'GRCh37'),(2032,'GRCh38'),(2033,'GRCh37'),(2033,'GRCh38'),(2034,'GRCh37'),(2034,'GRCh38'),(2035,'GRCh37'),(2035,'GRCh38'),(2036,'GRCh37'),(2036,'GRCh38'),(16804,'GRCh37'),(16804,'GRCh38'),(16805,'GRCh37'),(16805,'GRCh38'),(16806,'GRCh37'),(16806,'GRCh38'),(16807,'GRCh37'),(16807,'GRCh38'),(16808,'GRCh37'),(16808,'GRCh38'),(16809,'GRCh37'),(16809,'GRCh38'),(16810,'GRCh37'),(16810,'GRCh38'),(16811,'GRCh37'),(16811,'GRCh38'),(16812,'GRCh37'),(16812,'GRCh38'),(16813,'GRCh37'),(16813,'GRCh38'),(16814,'GRCh37'),(16814,'GRCh38'),(16815,'GRCh37'),(16815,'GRCh38'),(16816,'GRCh37'),(16816,'GRCh38'),(16817,'GRCh37'),(16817,'GRCh38'),(16818,'GRCh37'),(16818,'GRCh38'),(16819,'GRCh37'),(16819,'GRCh38'),(16820,'GRCh37'),(16820,'GRCh38'),(16821,'GRCh37'),(16821,'GRCh38'),(16822,'GRCh37'),(16822,'GRCh38'),(16823,'GRCh37'),(16823,'GRCh38'),(16824,'GRCh37'),(16824,'GRCh38'),(16825,'GRCh37'),(16825,'GRCh38'),(16826,'GRCh37'),(16826,'GRCh38'),(16827,'GRCh37'),(16827,'GRCh38'),(16828,'GRCh37'),(16828,'GRCh38'),(16829,'GRCh37'),(16829,'GRCh38'),(16830,'GRCh37'),(16830,'GRCh38'),(16831,'GRCh37'),(16831,'GRCh38'),(16832,'GRCh37'),(16832,'GRCh38'),(16833,'GRCh37'),(16833,'GRCh38'),(16834,'GRCh37'),(16834,'GRCh38'),(16835,'GRCh37'),(16835,'GRCh38'),(16836,'GRCh37'),(16836,'GRCh38'),(16837,'GRCh37'),(16837,'GRCh38'),(16838,'GRCh37'),(16838,'GRCh38'),(16839,'GRCh37'),(16839,'GRCh38'),(16840,'GRCh37'),(16840,'GRCh38'),(16841,'GRCh37'),(16841,'GRCh38'),(16842,'GRCh37'),(16842,'GRCh38'),(16843,'GRCh37'),(16843,'GRCh38'),(16844,'GRCh37'),(16844,'GRCh38'),(18246,'GRCh37'),(18246,'GRCh38'),(18247,'GRCh37'),(18247,'GRCh38'),(18248,'GRCh37'),(18248,'GRCh38'),(18249,'GRCh37'),(18249,'GRCh38'),(18250,'GRCh37'),(18250,'GRCh38'),(18251,'GRCh37'),(18251,'GRCh38'),(18252,'GRCh37'),(18252,'GRCh38'),(18253,'GRCh37'),(18253,'GRCh38'),(18254,'GRCh37'),(18254,'GRCh38'),(18255,'GRCh37'),(18255,'GRCh38'),(18256,'GRCh37'),(18256,'GRCh38'),(18257,'GRCh37'),(18257,'GRCh38'),(18258,'GRCh37'),(18258,'GRCh38'),(18259,'GRCh37'),(18259,'GRCh38'),(18260,'GRCh37'),(18260,'GRCh38'),(18261,'GRCh37'),(18261,'GRCh38'),(18262,'GRCh37'),(18262,'GRCh38'),(18263,'GRCh37'),(18263,'GRCh38'),(18264,'GRCh37'),(18264,'GRCh38'),(18265,'GRCh37'),(18265,'GRCh38'),(18266,'GRCh37'),(18266,'GRCh38'),(18267,'GRCh37'),(18267,'GRCh38'),(18268,'GRCh37'),(18268,'GRCh38'),(18269,'GRCh37'),(18269,'GRCh38'),(18270,'GRCh37'),(18270,'GRCh38'),(18271,'GRCh37'),(18271,'GRCh38'),(18272,'GRCh37'),(18272,'GRCh38'),(18273,'GRCh37'),(18273,'GRCh38'),(18274,'GRCh37'),(18274,'GRCh38'),(18275,'GRCh37'),(18275,'GRCh38'),(18276,'GRCh37'),(18276,'GRCh38'),(18277,'GRCh37'),(18277,'GRCh38'),(18278,'GRCh37'),(18278,'GRCh38'),(18279,'GRCh37'),(18279,'GRCh38'),(18280,'GRCh37'),(18280,'GRCh38'),(18281,'GRCh37'),(18281,'GRCh38'),(18282,'GRCh37'),(18282,'GRCh38'),(18283,'GRCh37'),(18283,'GRCh38'),(18284,'GRCh37'),(18284,'GRCh38'),(18285,'GRCh37'),(18285,'GRCh38'),(18286,'GRCh37'),(18286,'GRCh38'),(18287,'GRCh37'),(18287,'GRCh38'),(18288,'GRCh37'),(18288,'GRCh38'),(18289,'GRCh37'),(18289,'GRCh38'),(18290,'GRCh37'),(18290,'GRCh38'),(18291,'GRCh37'),(18291,'GRCh38'),(18292,'GRCh37'),(18292,'GRCh38'),(18293,'GRCh37'),(18293,'GRCh38'),(18294,'GRCh37'),(18294,'GRCh38'),(18295,'GRCh37'),(18295,'GRCh38'),(18296,'GRCh37'),(18296,'GRCh38'),(18297,'GRCh37'),(18297,'GRCh38'),(18298,'GRCh37'),(18298,'GRCh38'),(18299,'GRCh37'),(18299,'GRCh38'),(18300,'GRCh37'),(18300,'GRCh38'),(18301,'GRCh37'),(18301,'GRCh38'),(18302,'GRCh37'),(18302,'GRCh38'),(18303,'GRCh37'),(18303,'GRCh38'),(18304,'GRCh37'),(18304,'GRCh38'),(18305,'GRCh37'),(18305,'GRCh38'),(18306,'GRCh37'),(18306,'GRCh38'),(18307,'GRCh37'),(18307,'GRCh38'),(18308,'GRCh37'),(18308,'GRCh38'),(18309,'GRCh37'),(18309,'GRCh38'),(18310,'GRCh37'),(18310,'GRCh38'),(18311,'GRCh37'),(18311,'GRCh38'),(18312,'GRCh37'),(18312,'GRCh38'),(18313,'GRCh37'),(18313,'GRCh38'),(18314,'GRCh37'),(18314,'GRCh38'),(18315,'GRCh37'),(18315,'GRCh38'),(18316,'GRCh37'),(18316,'GRCh38'),(18317,'GRCh37'),(18317,'GRCh38'),(18318,'GRCh37'),(18318,'GRCh38'),(18319,'GRCh37'),(18319,'GRCh38'),(18320,'GRCh37'),(18320,'GRCh38'),(18321,'GRCh37'),(18321,'GRCh38'),(18322,'GRCh37'),(18322,'GRCh38'),(18323,'GRCh37'),(18323,'GRCh38'),(18324,'GRCh37'),(18324,'GRCh38'),(18325,'GRCh37'),(18325,'GRCh38'),(18326,'GRCh37'),(18326,'GRCh38'),(18327,'GRCh37'),(18327,'GRCh38'),(18328,'GRCh37'),(18328,'GRCh38'),(18329,'GRCh37'),(18329,'GRCh38'),(18330,'GRCh37'),(18330,'GRCh38'),(18331,'GRCh37'),(18331,'GRCh38'),(18332,'GRCh37'),(18332,'GRCh38'),(18333,'GRCh37'),(18333,'GRCh38'),(18334,'GRCh37'),(18334,'GRCh38'),(18335,'GRCh37'),(18335,'GRCh38'),(18336,'GRCh37'),(18336,'GRCh38'),(18337,'GRCh37'),(18337,'GRCh38'),(18338,'GRCh37'),(18338,'GRCh38'),(18339,'GRCh37'),(18339,'GRCh38'),(18340,'GRCh37'),(18340,'GRCh38'),(18341,'GRCh37'),(18341,'GRCh38'),(18342,'GRCh37'),(18342,'GRCh38'),(18343,'GRCh37'),(18343,'GRCh38'),(18344,'GRCh37'),(18344,'GRCh38'),(18345,'GRCh37'),(18345,'GRCh38'),(18346,'GRCh37'),(18346,'GRCh38'),(18347,'GRCh37'),(18347,'GRCh38'),(18348,'GRCh37'),(18348,'GRCh38'),(18349,'GRCh37'),(18349,'GRCh38'),(18350,'GRCh37'),(18350,'GRCh38'),(18351,'GRCh37'),(18351,'GRCh38'),(18352,'GRCh37'),(18352,'GRCh38'),(18353,'GRCh37'),(18353,'GRCh38'),(18354,'GRCh37'),(18354,'GRCh38'),(18355,'GRCh37'),(18355,'GRCh38'),(18356,'GRCh37'),(18356,'GRCh38'),(18357,'GRCh37'),(18357,'GRCh38'),(18358,'GRCh37'),(18358,'GRCh38'),(18359,'GRCh37'),(18359,'GRCh38'),(18360,'GRCh37'),(18360,'GRCh38'),(18361,'GRCh37'),(18361,'GRCh38'),(18362,'GRCh37'),(18362,'GRCh38'),(18363,'GRCh37'),(18363,'GRCh38'),(18364,'GRCh37'),(18364,'GRCh38'),(18365,'GRCh37'),(18365,'GRCh38'),(18366,'GRCh37'),(18366,'GRCh38'),(18367,'GRCh37'),(18367,'GRCh38'),(18368,'GRCh37'),(18368,'GRCh38'),(18369,'GRCh37'),(18369,'GRCh38'),(18370,'GRCh37'),(18370,'GRCh38'),(18371,'GRCh37'),(18371,'GRCh38'),(18372,'GRCh37'),(18372,'GRCh38'),(18373,'GRCh37'),(18373,'GRCh38'),(18374,'GRCh37'),(18374,'GRCh38'),(18375,'GRCh37'),(18375,'GRCh38'),(18376,'GRCh37'),(18376,'GRCh38'),(18377,'GRCh37'),(18377,'GRCh38'),(18378,'GRCh37'),(18378,'GRCh38'),(18379,'GRCh37'),(18379,'GRCh38'),(18380,'GRCh37'),(18380,'GRCh38'),(18381,'GRCh37'),(18381,'GRCh38'),(18382,'GRCh37'),(18382,'GRCh38'),(18383,'GRCh37'),(18383,'GRCh38'),(18384,'GRCh37'),(18384,'GRCh38'),(18385,'GRCh37'),(18385,'GRCh38'),(18386,'GRCh37'),(18386,'GRCh38'),(18387,'GRCh37'),(18387,'GRCh38'),(18388,'GRCh37'),(18388,'GRCh38'),(18389,'GRCh37'),(18389,'GRCh38'),(18390,'GRCh37'),(18390,'GRCh38'),(18391,'GRCh37'),(18391,'GRCh38'),(18392,'GRCh37'),(18392,'GRCh38'),(18393,'GRCh37'),(18393,'GRCh38'),(18394,'GRCh37'),(18394,'GRCh38'),(18395,'GRCh37'),(18395,'GRCh38'),(18396,'GRCh37'),(18396,'GRCh38'),(18397,'GRCh37'),(18397,'GRCh38'),(18398,'GRCh37'),(18398,'GRCh38'),(18399,'GRCh37'),(18399,'GRCh38'),(18400,'GRCh37'),(18400,'GRCh38'),(18401,'GRCh37'),(18401,'GRCh38'),(18402,'GRCh37'),(18402,'GRCh38'),(18403,'GRCh37'),(18403,'GRCh38'),(18404,'GRCh37'),(18404,'GRCh38'),(18405,'GRCh37'),(18405,'GRCh38'),(18406,'GRCh37'),(18406,'GRCh38'),(18407,'GRCh37'),(18407,'GRCh38'),(18408,'GRCh37'),(18408,'GRCh38'),(18409,'GRCh37'),(18409,'GRCh38'),(18410,'GRCh37'),(18410,'GRCh38'),(18411,'GRCh37'),(18411,'GRCh38'),(18412,'GRCh37'),(18412,'GRCh38'),(18413,'GRCh37'),(18413,'GRCh38'),(18414,'GRCh37'),(18414,'GRCh38'),(18415,'GRCh37'),(18415,'GRCh38'),(18416,'GRCh37'),(18416,'GRCh38'),(18417,'GRCh37'),(18417,'GRCh38'),(18418,'GRCh37'),(18418,'GRCh38'),(18419,'GRCh37'),(18419,'GRCh38'),(18420,'GRCh37'),(18420,'GRCh38'),(18421,'GRCh37'),(18421,'GRCh38'),(18422,'GRCh37'),(18422,'GRCh38'),(18423,'GRCh37'),(18423,'GRCh38'),(18424,'GRCh37'),(18424,'GRCh38'),(18425,'GRCh37'),(18425,'GRCh38'),(18426,'GRCh37'),(18426,'GRCh38'),(18427,'GRCh37'),(18427,'GRCh38'),(18428,'GRCh37'),(18428,'GRCh38'),(18429,'GRCh37'),(18429,'GRCh38'),(18430,'GRCh37'),(18430,'GRCh38'),(18431,'GRCh37'),(18431,'GRCh38'),(18432,'GRCh37'),(18432,'GRCh38'),(18433,'GRCh37'),(18433,'GRCh38'),(18434,'GRCh37'),(18434,'GRCh38'),(18435,'GRCh37'),(18435,'GRCh38'),(18436,'GRCh37'),(18436,'GRCh38'),(18437,'GRCh37'),(18437,'GRCh38'),(18438,'GRCh37'),(18438,'GRCh38'),(18439,'GRCh37'),(18439,'GRCh38'),(18440,'GRCh37'),(18440,'GRCh38'),(18441,'GRCh37'),(18441,'GRCh38'),(18442,'GRCh37'),(18442,'GRCh38'),(18443,'GRCh37'),(18443,'GRCh38'),(18444,'GRCh37'),(18444,'GRCh38'),(18445,'GRCh37'),(18445,'GRCh38'),(18446,'GRCh37'),(18446,'GRCh38'),(18447,'GRCh37'),(18447,'GRCh38'),(18448,'GRCh37'),(18448,'GRCh38'),(18449,'GRCh37'),(18449,'GRCh38'),(18450,'GRCh37'),(18450,'GRCh38'),(18451,'GRCh37'),(18451,'GRCh38'),(18452,'GRCh37'),(18452,'GRCh38'),(18453,'GRCh37'),(18453,'GRCh38'),(18454,'GRCh37'),(18454,'GRCh38'),(18455,'GRCh37'),(18455,'GRCh38'),(18456,'GRCh37'),(18456,'GRCh38'),(18457,'GRCh37'),(18457,'GRCh38'),(18458,'GRCh37'),(18458,'GRCh38'),(18459,'GRCh37'),(18459,'GRCh38'),(18460,'GRCh37'),(18460,'GRCh38'),(18461,'GRCh37'),(18461,'GRCh38'),(18462,'GRCh37'),(18462,'GRCh38'),(18463,'GRCh37'),(18463,'GRCh38'),(18464,'GRCh37'),(18464,'GRCh38'),(18465,'GRCh37'),(18465,'GRCh38'),(18466,'GRCh37'),(18466,'GRCh38'),(18467,'GRCh37'),(18467,'GRCh38'),(18468,'GRCh37'),(18468,'GRCh38'),(18469,'GRCh37'),(18469,'GRCh38'),(18470,'GRCh37'),(18470,'GRCh38'),(18471,'GRCh37'),(18471,'GRCh38'),(18472,'GRCh37'),(18472,'GRCh38'),(18473,'GRCh37'),(18473,'GRCh38'),(18474,'GRCh37'),(18474,'GRCh38'),(18475,'GRCh37'),(18475,'GRCh38'),(18476,'GRCh37'),(18476,'GRCh38'),(18477,'GRCh37'),(18477,'GRCh38'),(18478,'GRCh37'),(18478,'GRCh38'),(18479,'GRCh37'),(18479,'GRCh38'),(18480,'GRCh37'),(18480,'GRCh38'),(18481,'GRCh37'),(18481,'GRCh38'),(18482,'GRCh37'),(18482,'GRCh38'),(18483,'GRCh37'),(18483,'GRCh38'),(18484,'GRCh37'),(18484,'GRCh38'),(18485,'GRCh37'),(18485,'GRCh38'),(18486,'GRCh37'),(18486,'GRCh38'),(18487,'GRCh37'),(18487,'GRCh38'),(18488,'GRCh37'),(18488,'GRCh38'),(18489,'GRCh37'),(18489,'GRCh38'),(18490,'GRCh37'),(18490,'GRCh38'),(18491,'GRCh37'),(18491,'GRCh38'),(18492,'GRCh37'),(18492,'GRCh38'),(18493,'GRCh37'),(18493,'GRCh38'),(18494,'GRCh37'),(18494,'GRCh38'),(18495,'GRCh37'),(18495,'GRCh38'),(18496,'GRCh37'),(18496,'GRCh38'),(18497,'GRCh37'),(18497,'GRCh38'),(18498,'GRCh37'),(18498,'GRCh38'),(18499,'GRCh37'),(18499,'GRCh38'),(18500,'GRCh37'),(18500,'GRCh38'),(18501,'GRCh37'),(18501,'GRCh38'),(18502,'GRCh37'),(18502,'GRCh38'),(18503,'GRCh37'),(18503,'GRCh38'),(18504,'GRCh37'),(18504,'GRCh38'),(18505,'GRCh37'),(18505,'GRCh38'),(18506,'GRCh37'),(18506,'GRCh38'),(18507,'GRCh37'),(18507,'GRCh38'),(18508,'GRCh37'),(18508,'GRCh38'),(18509,'GRCh37'),(18509,'GRCh38'),(18510,'GRCh37'),(18510,'GRCh38'),(18511,'GRCh37'),(18511,'GRCh38'),(18512,'GRCh37'),(18512,'GRCh38'),(18513,'GRCh37'),(18513,'GRCh38'),(18514,'GRCh37'),(18514,'GRCh38'),(18515,'GRCh37'),(18515,'GRCh38'),(18516,'GRCh37'),(18516,'GRCh38'),(18517,'GRCh37'),(18517,'GRCh38'),(18518,'GRCh37'),(18518,'GRCh38'),(18519,'GRCh37'),(18519,'GRCh38'),(18520,'GRCh37'),(18520,'GRCh38'),(18521,'GRCh37'),(18521,'GRCh38'),(18522,'GRCh37'),(18522,'GRCh38'),(18523,'GRCh37'),(18523,'GRCh38'),(18524,'GRCh37'),(18524,'GRCh38'),(18525,'GRCh37'),(18525,'GRCh38'),(18526,'GRCh37'),(18526,'GRCh38'),(18527,'GRCh37'),(18527,'GRCh38'),(18528,'GRCh37'),(18528,'GRCh38'),(18529,'GRCh37'),(18529,'GRCh38'),(18530,'GRCh37'),(18530,'GRCh38'),(18531,'GRCh37'),(18531,'GRCh38'),(18532,'GRCh37'),(18532,'GRCh38'),(18533,'GRCh37'),(18533,'GRCh38'),(18534,'GRCh37'),(18534,'GRCh38'),(18535,'GRCh37'),(18535,'GRCh38'),(18536,'GRCh37'),(18536,'GRCh38'),(18537,'GRCh37'),(18537,'GRCh38'),(18538,'GRCh37'),(18538,'GRCh38'),(18539,'GRCh37'),(18539,'GRCh38'),(18540,'GRCh37'),(18540,'GRCh38'),(18541,'GRCh37'),(18541,'GRCh38'),(18542,'GRCh37'),(18542,'GRCh38'),(18543,'GRCh37'),(18543,'GRCh38'),(18544,'GRCh37'),(18544,'GRCh38'),(18545,'GRCh37'),(18545,'GRCh38'),(18546,'GRCh37'),(18546,'GRCh38'),(18547,'GRCh37'),(18547,'GRCh38'),(18548,'GRCh37'),(18548,'GRCh38'),(18549,'GRCh37'),(18549,'GRCh38'),(18550,'GRCh37'),(18550,'GRCh38'),(18551,'GRCh37'),(18551,'GRCh38'),(18552,'GRCh37'),(18552,'GRCh38'),(18553,'GRCh37'),(18553,'GRCh38'),(18554,'GRCh37'),(18554,'GRCh38'),(18555,'GRCh37'),(18555,'GRCh38'),(18556,'GRCh37'),(18556,'GRCh38'),(18557,'GRCh37'),(18557,'GRCh38'),(18558,'GRCh37'),(18558,'GRCh38'),(18559,'GRCh37'),(18559,'GRCh38'),(18560,'GRCh37'),(18560,'GRCh38'),(18561,'GRCh37'),(18561,'GRCh38'),(18562,'GRCh37'),(18562,'GRCh38'),(18563,'GRCh37'),(18563,'GRCh38'),(18564,'GRCh37'),(18564,'GRCh38'),(18565,'GRCh37'),(18565,'GRCh38'),(18566,'GRCh37'),(18566,'GRCh38'),(18567,'GRCh37'),(18567,'GRCh38'),(18568,'GRCh37'),(18568,'GRCh38'),(18569,'GRCh37'),(18569,'GRCh38'),(18570,'GRCh37'),(18570,'GRCh38'),(18571,'GRCh37'),(18571,'GRCh38'),(18572,'GRCh37'),(18572,'GRCh38'),(18573,'GRCh37'),(18573,'GRCh38'),(18574,'GRCh37'),(18574,'GRCh38'),(18575,'GRCh37'),(18575,'GRCh38'),(18576,'GRCh37'),(18576,'GRCh38'),(18577,'GRCh37'),(18577,'GRCh38'),(18578,'GRCh37'),(18578,'GRCh38'),(18579,'GRCh37'),(18579,'GRCh38'),(18580,'GRCh37'),(18580,'GRCh38'),(18581,'GRCh37'),(18581,'GRCh38'),(18582,'GRCh37'),(18582,'GRCh38'),(18583,'GRCh37'),(18583,'GRCh38'),(18584,'GRCh37'),(18584,'GRCh38'),(18585,'GRCh37'),(18585,'GRCh38'),(18586,'GRCh37'),(18586,'GRCh38'),(18587,'GRCh37'),(18587,'GRCh38'),(18588,'GRCh37'),(18588,'GRCh38'),(18589,'GRCh37'),(18589,'GRCh38'),(18590,'GRCh37'),(18590,'GRCh38'),(18591,'GRCh37'),(18591,'GRCh38'),(18592,'GRCh37'),(18592,'GRCh38'),(18593,'GRCh37'),(18593,'GRCh38'),(18594,'GRCh37'),(18594,'GRCh38'),(18595,'GRCh37'),(18595,'GRCh38'),(18596,'GRCh37'),(18596,'GRCh38'),(18597,'GRCh37'),(18597,'GRCh38'),(18598,'GRCh37'),(18598,'GRCh38'),(18599,'GRCh37'),(18599,'GRCh38'),(18600,'GRCh37'),(18600,'GRCh38'),(18601,'GRCh37'),(18601,'GRCh38'),(18602,'GRCh37'),(18602,'GRCh38'),(18603,'GRCh37'),(18603,'GRCh38'),(18604,'GRCh37'),(18604,'GRCh38'),(18605,'GRCh37'),(18605,'GRCh38'),(18606,'GRCh37'),(18606,'GRCh38'),(18607,'GRCh37'),(18607,'GRCh38'),(18608,'GRCh37'),(18608,'GRCh38'),(18609,'GRCh37'),(18609,'GRCh38'),(18610,'GRCh37'),(18610,'GRCh38'),(18611,'GRCh37'),(18611,'GRCh38'),(18612,'GRCh37'),(18612,'GRCh38'),(18613,'GRCh37'),(18613,'GRCh38'),(18614,'GRCh37'),(18614,'GRCh38'),(18615,'GRCh37'),(18615,'GRCh38'),(18616,'GRCh37'),(18616,'GRCh38'),(18617,'GRCh37'),(18617,'GRCh38'),(18618,'GRCh37'),(18618,'GRCh38'),(18619,'GRCh37'),(18619,'GRCh38'),(18620,'GRCh37'),(18620,'GRCh38'),(18621,'GRCh37'),(18621,'GRCh38'),(18622,'GRCh37'),(18622,'GRCh38'),(18623,'GRCh37'),(18623,'GRCh38'),(18624,'GRCh37'),(18624,'GRCh38'),(18625,'GRCh37'),(18625,'GRCh38'),(18626,'GRCh37'),(18626,'GRCh38'),(18627,'GRCh37'),(18627,'GRCh38'),(18628,'GRCh37'),(18628,'GRCh38'),(18629,'GRCh37'),(18629,'GRCh38'),(18630,'GRCh37'),(18630,'GRCh38'),(18631,'GRCh37'),(18631,'GRCh38'),(18632,'GRCh37'),(18632,'GRCh38'),(18633,'GRCh37'),(18633,'GRCh38'),(18634,'GRCh37'),(18634,'GRCh38'),(18635,'GRCh37'),(18635,'GRCh38'),(18636,'GRCh37'),(18636,'GRCh38'),(18637,'GRCh37'),(18637,'GRCh38'),(18638,'GRCh37'),(18638,'GRCh38'),(18639,'GRCh37'),(18639,'GRCh38'),(18640,'GRCh37'),(18640,'GRCh38'),(18641,'GRCh37'),(18641,'GRCh38'),(18642,'GRCh37'),(18642,'GRCh38'),(18643,'GRCh37'),(18643,'GRCh38'),(18644,'GRCh37'),(18644,'GRCh38'),(18645,'GRCh37'),(18645,'GRCh38'),(18646,'GRCh37'),(18646,'GRCh38'),(18647,'GRCh37'),(18647,'GRCh38'),(18648,'GRCh37'),(18648,'GRCh38'),(18649,'GRCh37'),(18649,'GRCh38'),(18650,'GRCh37'),(18650,'GRCh38'),(18651,'GRCh37'),(18651,'GRCh38'),(18652,'GRCh37'),(18652,'GRCh38'),(18653,'GRCh37'),(18653,'GRCh38'),(18654,'GRCh37'),(18654,'GRCh38'),(18655,'GRCh37'),(18655,'GRCh38'),(18656,'GRCh37'),(18656,'GRCh38'),(18657,'GRCh37'),(18657,'GRCh38'),(18658,'GRCh37'),(18658,'GRCh38'),(18659,'GRCh37'),(18659,'GRCh38'),(18660,'GRCh37'),(18660,'GRCh38'),(18661,'GRCh37'),(18661,'GRCh38'),(18662,'GRCh37'),(18662,'GRCh38'),(18663,'GRCh37'),(18663,'GRCh38'),(18664,'GRCh37'),(18664,'GRCh38'),(18665,'GRCh37'),(18665,'GRCh38'),(18666,'GRCh37'),(18666,'GRCh38'),(18667,'GRCh37'),(18667,'GRCh38'),(18668,'GRCh37'),(18668,'GRCh38'),(18669,'GRCh37'),(18669,'GRCh38'),(18670,'GRCh37'),(18670,'GRCh38'),(18671,'GRCh37'),(18671,'GRCh38'),(18672,'GRCh37'),(18672,'GRCh38'),(18673,'GRCh37'),(18673,'GRCh38'),(18674,'GRCh37'),(18674,'GRCh38'),(18675,'GRCh37'),(18675,'GRCh38'),(18676,'GRCh37'),(18676,'GRCh38'),(18677,'GRCh37'),(18677,'GRCh38'),(18678,'GRCh37'),(18678,'GRCh38'),(18679,'GRCh37'),(18679,'GRCh38'),(18680,'GRCh37'),(18680,'GRCh38'),(18681,'GRCh37'),(18681,'GRCh38'),(18682,'GRCh37'),(18682,'GRCh38'),(18683,'GRCh37'),(18683,'GRCh38'),(18684,'GRCh37'),(18684,'GRCh38'),(18685,'GRCh37'),(18685,'GRCh38'),(18686,'GRCh37'),(18686,'GRCh38'),(18687,'GRCh37'),(18687,'GRCh38'),(18688,'GRCh37'),(18688,'GRCh38'),(18689,'GRCh37'),(18689,'GRCh38'),(18690,'GRCh37'),(18690,'GRCh38'),(18691,'GRCh37'),(18691,'GRCh38'),(18692,'GRCh37'),(18692,'GRCh38'),(18693,'GRCh37'),(18693,'GRCh38'),(18694,'GRCh37'),(18694,'GRCh38'),(18695,'GRCh37'),(18695,'GRCh38'),(18696,'GRCh37'),(18696,'GRCh38'),(18697,'GRCh37'),(18697,'GRCh38'),(18698,'GRCh37'),(18698,'GRCh38'),(18699,'GRCh37'),(18699,'GRCh38'),(18700,'GRCh37'),(18700,'GRCh38'),(18701,'GRCh37'),(18701,'GRCh38'),(18702,'GRCh37'),(18702,'GRCh38'),(18703,'GRCh37'),(18703,'GRCh38'),(18704,'GRCh37'),(18704,'GRCh38'),(18705,'GRCh37'),(18705,'GRCh38'),(18706,'GRCh37'),(18706,'GRCh38'),(18707,'GRCh37'),(18707,'GRCh38'),(18708,'GRCh37'),(18708,'GRCh38'),(18709,'GRCh37'),(18709,'GRCh38'),(18710,'GRCh37'),(18710,'GRCh38'),(18711,'GRCh37'),(18711,'GRCh38'),(18712,'GRCh37'),(18712,'GRCh38'),(18713,'GRCh37'),(18713,'GRCh38'),(18714,'GRCh37'),(18714,'GRCh38'),(18715,'GRCh37'),(18715,'GRCh38'),(18716,'GRCh37'),(18716,'GRCh38'),(18717,'GRCh37'),(18717,'GRCh38'),(18718,'GRCh37'),(18718,'GRCh38'),(18719,'GRCh37'),(18719,'GRCh38'),(18720,'GRCh37'),(18720,'GRCh38'),(18721,'GRCh37'),(18721,'GRCh38'),(18722,'GRCh37'),(18722,'GRCh38'),(18723,'GRCh37'),(18723,'GRCh38'),(18724,'GRCh37'),(18724,'GRCh38'),(18725,'GRCh37'),(18725,'GRCh38'),(18726,'GRCh37'),(18726,'GRCh38'),(18727,'GRCh37'),(18727,'GRCh38'),(18728,'GRCh37'),(18728,'GRCh38'),(18729,'GRCh37'),(18729,'GRCh38'),(18730,'GRCh37'),(18730,'GRCh38'),(18731,'GRCh37'),(18731,'GRCh38'),(18732,'GRCh37'),(18732,'GRCh38'),(18733,'GRCh37'),(18733,'GRCh38'),(18734,'GRCh37'),(18734,'GRCh38'),(18735,'GRCh37'),(18735,'GRCh38'),(18736,'GRCh37'),(18736,'GRCh38'),(18737,'GRCh37'),(18737,'GRCh38'),(18738,'GRCh37'),(18738,'GRCh38'),(18739,'GRCh37'),(18739,'GRCh38'),(18740,'GRCh37'),(18740,'GRCh38'),(18741,'GRCh37'),(18741,'GRCh38'),(18742,'GRCh37'),(18742,'GRCh38'),(18743,'GRCh37'),(18743,'GRCh38'),(18744,'GRCh37'),(18744,'GRCh38'),(18745,'GRCh37'),(18745,'GRCh38'),(18746,'GRCh37'),(18746,'GRCh38'),(18747,'GRCh37'),(18747,'GRCh38'),(18748,'GRCh37'),(18748,'GRCh38'),(18749,'GRCh37'),(18749,'GRCh38'),(18750,'GRCh37'),(18750,'GRCh38'),(18751,'GRCh37'),(18751,'GRCh38'),(18752,'GRCh37'),(18752,'GRCh38'),(18753,'GRCh37'),(18753,'GRCh38'),(18754,'GRCh37'),(18754,'GRCh38'),(18755,'GRCh37'),(18755,'GRCh38'),(18756,'GRCh37'),(18756,'GRCh38'),(18757,'GRCh37'),(18757,'GRCh38'),(18758,'GRCh37'),(18758,'GRCh38'),(18759,'GRCh37'),(18759,'GRCh38'),(18760,'GRCh37'),(18760,'GRCh38'),(18761,'GRCh37'),(18761,'GRCh38'),(18762,'GRCh37'),(18762,'GRCh38'),(18763,'GRCh37'),(18763,'GRCh38'),(18764,'GRCh37'),(18764,'GRCh38'),(18765,'GRCh37'),(18765,'GRCh38'),(18766,'GRCh37'),(18766,'GRCh38'),(18767,'GRCh37'),(18767,'GRCh38'),(18768,'GRCh37'),(18768,'GRCh38'),(18769,'GRCh37'),(18769,'GRCh38'),(18770,'GRCh37'),(18770,'GRCh38'),(18771,'GRCh37'),(18771,'GRCh38'),(18772,'GRCh37'),(18772,'GRCh38'),(18773,'GRCh37'),(18773,'GRCh38'),(18774,'GRCh37'),(18774,'GRCh38'),(18775,'GRCh37'),(18775,'GRCh38'),(18776,'GRCh37'),(18776,'GRCh38'),(18777,'GRCh37'),(18777,'GRCh38'),(18778,'GRCh37'),(18778,'GRCh38'),(18779,'GRCh37'),(18779,'GRCh38'),(18780,'GRCh37'),(18780,'GRCh38'),(18781,'GRCh37'),(18781,'GRCh38'),(18782,'GRCh37'),(18782,'GRCh38'),(18783,'GRCh37'),(18783,'GRCh38'),(18784,'GRCh37'),(18784,'GRCh38'),(18785,'GRCh37'),(18785,'GRCh38'),(18786,'GRCh37'),(18786,'GRCh38'),(18787,'GRCh37'),(18787,'GRCh38'),(18788,'GRCh37'),(18788,'GRCh38'),(18789,'GRCh37'),(18789,'GRCh38'),(18790,'GRCh37'),(18790,'GRCh38'),(18791,'GRCh37'),(18791,'GRCh38'),(18792,'GRCh37'),(18792,'GRCh38'),(18793,'GRCh37'),(18793,'GRCh38'),(18794,'GRCh37'),(18794,'GRCh38'),(18795,'GRCh37'),(18795,'GRCh38'),(18796,'GRCh37'),(18796,'GRCh38'),(18797,'GRCh37'),(18797,'GRCh38'),(18798,'GRCh37'),(18798,'GRCh38'),(18799,'GRCh37'),(18799,'GRCh38'),(18800,'GRCh37'),(18800,'GRCh38'),(18801,'GRCh37'),(18801,'GRCh38'),(18802,'GRCh37'),(18802,'GRCh38'),(18803,'GRCh37'),(18803,'GRCh38'),(18804,'GRCh37'),(18804,'GRCh38'),(18805,'GRCh37'),(18805,'GRCh38'),(18806,'GRCh37'),(18806,'GRCh38'),(18807,'GRCh37'),(18807,'GRCh38'),(18808,'GRCh37'),(18808,'GRCh38'),(18809,'GRCh37'),(18809,'GRCh38'),(18810,'GRCh37'),(18810,'GRCh38'),(18811,'GRCh37'),(18811,'GRCh38'),(18812,'GRCh37'),(18812,'GRCh38'),(18813,'GRCh37'),(18813,'GRCh38'),(18814,'GRCh37'),(18814,'GRCh38'),(18815,'GRCh37'),(18815,'GRCh38'),(18816,'GRCh37'),(18816,'GRCh38'),(18817,'GRCh37'),(18817,'GRCh38'),(18818,'GRCh37'),(18818,'GRCh38'),(18819,'GRCh37'),(18819,'GRCh38'),(18820,'GRCh37'),(18820,'GRCh38'),(18821,'GRCh37'),(18821,'GRCh38'),(18822,'GRCh37'),(18822,'GRCh38'),(18823,'GRCh37'),(18823,'GRCh38'),(18824,'GRCh37'),(18824,'GRCh38'),(18825,'GRCh37'),(18825,'GRCh38'),(18826,'GRCh37'),(18826,'GRCh38'),(18827,'GRCh37'),(18827,'GRCh38'),(18828,'GRCh37'),(18828,'GRCh38'),(18829,'GRCh37'),(18829,'GRCh38'),(18830,'GRCh37'),(18830,'GRCh38'),(18831,'GRCh37'),(18831,'GRCh38'),(18832,'GRCh37'),(18832,'GRCh38'),(18833,'GRCh37'),(18833,'GRCh38'),(18834,'GRCh37'),(18834,'GRCh38'),(18835,'GRCh37'),(18835,'GRCh38'),(18836,'GRCh37'),(18836,'GRCh38'),(18837,'GRCh37'),(18837,'GRCh38'),(18838,'GRCh37'),(18838,'GRCh38'),(18839,'GRCh37'),(18839,'GRCh38'),(18840,'GRCh37'),(18840,'GRCh38'),(18841,'GRCh37'),(18841,'GRCh38'),(18842,'GRCh37'),(18842,'GRCh38'),(18843,'GRCh37'),(18843,'GRCh38'),(18844,'GRCh37'),(18844,'GRCh38'),(18845,'GRCh37'),(18845,'GRCh38'),(18846,'GRCh37'),(18846,'GRCh38'),(18847,'GRCh37'),(18847,'GRCh38'),(18848,'GRCh37'),(18848,'GRCh38'),(18849,'GRCh37'),(18849,'GRCh38'),(18850,'GRCh37'),(18850,'GRCh38'),(18851,'GRCh37'),(18851,'GRCh38'),(18852,'GRCh37'),(18852,'GRCh38'),(18853,'GRCh37'),(18853,'GRCh38'),(18854,'GRCh37'),(18854,'GRCh38'),(18855,'GRCh37'),(18855,'GRCh38'),(18856,'GRCh37'),(18856,'GRCh38'),(18857,'GRCh37'),(18857,'GRCh38'),(18858,'GRCh37'),(18858,'GRCh38'),(18859,'GRCh37'),(18859,'GRCh38'),(18860,'GRCh37'),(18860,'GRCh38'),(18861,'GRCh37'),(18861,'GRCh38'),(18862,'GRCh37'),(18862,'GRCh38'),(18863,'GRCh37'),(18863,'GRCh38'),(18864,'GRCh37'),(18864,'GRCh38'),(18865,'GRCh37'),(18865,'GRCh38'),(18866,'GRCh37'),(18866,'GRCh38'),(18867,'GRCh37'),(18867,'GRCh38'),(18868,'GRCh37'),(18868,'GRCh38'),(18869,'GRCh37'),(18869,'GRCh38'),(18870,'GRCh37'),(18870,'GRCh38'),(18871,'GRCh37'),(18871,'GRCh38'),(18872,'GRCh37'),(18872,'GRCh38'),(18873,'GRCh37'),(18873,'GRCh38'),(18874,'GRCh37'),(18874,'GRCh38'),(18875,'GRCh37'),(18875,'GRCh38'),(18876,'GRCh37'),(18876,'GRCh38'),(18877,'GRCh37'),(18877,'GRCh38'),(18878,'GRCh37'),(18878,'GRCh38'),(18879,'GRCh37'),(18879,'GRCh38'),(18880,'GRCh37'),(18880,'GRCh38'),(18881,'GRCh37'),(18881,'GRCh38'),(18882,'GRCh37'),(18882,'GRCh38'),(18883,'GRCh37'),(18883,'GRCh38'),(18884,'GRCh37'),(18884,'GRCh38'),(18885,'GRCh37'),(18885,'GRCh38'),(18886,'GRCh37'),(18886,'GRCh38'),(18887,'GRCh37'),(18887,'GRCh38'),(18888,'GRCh37'),(18888,'GRCh38'),(18889,'GRCh37'),(18889,'GRCh38'),(18890,'GRCh37'),(18890,'GRCh38'),(18891,'GRCh37'),(18891,'GRCh38'),(18892,'GRCh37'),(18892,'GRCh38'),(18893,'GRCh37'),(18893,'GRCh38'),(18894,'GRCh37'),(18894,'GRCh38'),(18895,'GRCh37'),(18895,'GRCh38'),(18896,'GRCh37'),(18896,'GRCh38'),(18897,'GRCh37'),(18897,'GRCh38'),(18898,'GRCh37'),(18898,'GRCh38'),(18899,'GRCh37'),(18899,'GRCh38'),(18900,'GRCh37'),(18900,'GRCh38'),(18901,'GRCh37'),(18901,'GRCh38'),(18902,'GRCh37'),(18902,'GRCh38'),(18903,'GRCh37'),(18903,'GRCh38'),(18904,'GRCh37'),(18904,'GRCh38'),(18905,'GRCh37'),(18905,'GRCh38'),(18906,'GRCh37'),(18906,'GRCh38'),(18907,'GRCh37'),(18907,'GRCh38'),(18908,'GRCh37'),(18908,'GRCh38'),(18909,'GRCh37'),(18909,'GRCh38'),(18910,'GRCh37'),(18910,'GRCh38'),(18911,'GRCh37'),(18911,'GRCh38');
/*!40000 ALTER TABLE `alteration_reference_genome` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `article`
--

DROP TABLE IF EXISTS `article`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `article` (
  `id` int NOT NULL AUTO_INCREMENT,
  `abstract_content` varchar(255) DEFAULT NULL,
  `authors` varchar(255) DEFAULT NULL,
  `elocationId` varchar(255) DEFAULT NULL,
  `issue` varchar(255) DEFAULT NULL,
  `journal` varchar(255) DEFAULT NULL,
  `link` longtext,
  `pages` varchar(255) DEFAULT NULL,
  `pmid` varchar(255) DEFAULT NULL,
  `pub_date` varchar(255) DEFAULT NULL,
  `title` longtext,
  `uuid` varchar(40) DEFAULT NULL,
  `volume` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=16274 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `article`
--

LOCK TABLES `article` WRITE;
/*!40000 ALTER TABLE `article` DISABLE KEYS */;
INSERT INTO `article` VALUES (137,NULL,'Zehir A et al','doi: 10.1038/nm.4333','6','Nature medicine',NULL,'703-713','28481359','2017 Jun','Mutational landscape of metastatic cancer revealed from prospective clinical sequencing of 10,000 patients.',NULL,'23'),(197,NULL,'Ng PK et al','doi: 10.1016/j.ccell.2018.01.021','3','Cancer cell',NULL,'450-462.e10','29533785','2018 Mar 12','Systematic Functional Annotation of Somatic Mutations in Cancer.',NULL,'33'),(198,NULL,'Berger AH et al','doi: 10.1016/j.ccell.2016.06.022','2','Cancer cell',NULL,'214-228','27478040','2016 Aug 8','High-throughput Phenotyping of Lung Cancer Somatic Mutations.',NULL,'30'),(242,NULL,'Cancer Genome Atlas Research Network.','doi: 10.1038/nature13385','7511','Nature',NULL,'543-50','25079552','2014 Jul 31','Comprehensive molecular profiling of lung adenocarcinoma.',NULL,'511'),(246,NULL,'Friboulet L et al','doi: 10.1158/2159-8290.CD-13-0846','6','Cancer discovery',NULL,'662-673','24675041','2014 Jun','The ALK inhibitor ceritinib overcomes crizotinib resistance in non-small cell lung cancer.',NULL,'4'),(248,NULL,'Doebele RC et al','doi: 10.1158/1078-0432.CCR-11-2906','5','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'1472-82','22235099','2012 Mar 1','Mechanisms of resistance to crizotinib in patients with ALK gene rearranged non-small cell lung cancer.',NULL,'18'),(251,NULL,'Wu YL et al','doi: 10.1200/JCO.2017.75.5587','14','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'1405-1411','29596029','2018 May 10','Phase II Study of Crizotinib in East Asian Patients With ROS1-Positive Advanced Non-Small-Cell Lung Cancer.',NULL,'36'),(393,NULL,'Yaeger R et al','doi: 10.1016/j.ccell.2017.12.004','1','Cancer cell',NULL,'125-136.e3','29316426','2018 Jan 8','Clinical Sequencing Defines the Genomic Landscape of Metastatic Colorectal Cancer.',NULL,'33'),(451,NULL,'Wellbrock C et al','','11','Nature reviews. Molecular cell biology',NULL,'875-85','15520807','2004 Nov','The RAF proteins take centre stage.',NULL,'5'),(454,NULL,'Jacobsen E et al','doi: 10.1056/NEJMc1713676','24','The New England journal of medicine',NULL,'2398-2399','29236635','2017 Dec 14','Rosai-Dorfman Disease with Activating KRAS Mutation - Response to Cobimetinib.',NULL,'377'),(455,NULL,'Diamond EL et al','doi: 10.1038/s41586-019-1012-y','7749','Nature',NULL,'521-524','30867592','2019 Mar','Efficacy of MEK inhibition in patients with histiocytic neoplasms.',NULL,'567'),(508,NULL,'Hodis E et al','doi: 10.1016/j.cell.2012.06.024','2','Cell',NULL,'251-63','22817889','2012 Jul 20','A landscape of driver mutations in melanoma.',NULL,'150'),(577,NULL,'Devillier R et al','','10','Oncotarget',NULL,'8388-96','25860933','2015 Apr 10','Role of ASXL1 and TP53 mutations in the molecular classification and prognosis of acute myeloid leukemias with myelodysplasia-related changes.',NULL,'6'),(583,NULL,'Bejar R et al','doi: 10.1056/NEJMoa1013343','26','The New England journal of medicine',NULL,'2496-506','21714648','2011 Jun 30','Clinical effect of point mutations in myelodysplastic syndromes.',NULL,'364'),(585,NULL,'Haferlach T et al','doi: 10.1038/leu.2013.336','2','Leukemia',NULL,'241-7','24220272','2014 Feb','Landscape of genetic lesions in 944 patients with myelodysplastic syndromes.',NULL,'28'),(592,NULL,'Tefferi A et al','doi: 10.1182/bloodadvances.2016000216','1','Blood advances',NULL,'21-30','29296692','2016 Nov 29','Targeted deep sequencing in polycythemia vera and essential thrombocythemia.',NULL,'1'),(778,NULL,'Vaughan CA et al','doi: 10.1177/1947601912462719','7-8','Genes & cancer',NULL,'491-502','23264849','2012 Jul','Gain-of-Function Activity of Mutant p53 in Lung Cancer through Up-Regulation of Receptor Protein Tyrosine Kinase Axl.',NULL,'3'),(1084,NULL,'Terai K et al','','15','The EMBO journal',NULL,'3556-64','16858395','2006 Aug 9','The amino-terminal B-Raf-specific region mediates calcium-dependent homo- and hetero-dimerization of Raf.',NULL,'25'),(1085,NULL,'Yao Z et al','doi: 10.1016/j.ccell.2015.08.001','3','Cancer cell',NULL,'370-83','26343582','2015 Sep 14','BRAF Mutants Evade ERK-Dependent Feedback by Different Mechanisms that Determine Their Sensitivity to Pharmacologic Inhibition.',NULL,'28'),(1086,NULL,'Davies H et al','','6892','Nature',NULL,'949-54','12068308','2002 Jun 27','Mutations of the BRAF gene in human cancer.',NULL,'417'),(1087,NULL,'Poulikakos PI et al','doi: 10.1038/nature08902','7287','Nature',NULL,'427-30','20179705','2010 Mar 18','RAF inhibitors transactivate RAF dimers and ERK signalling in cells with wild-type BRAF.',NULL,'464'),(1088,NULL,'Lito P et al','doi: 10.1038/nm.3392','11','Nature medicine',NULL,'1401-9','24202393','2013 Nov','Tumor adaptation and resistance to RAF inhibitors.',NULL,'19'),(1089,NULL,'Ciampi R et al','','1','The Journal of clinical investigation',NULL,'94-101','15630448','2005 Jan','Oncogenic AKAP9-BRAF fusion is a novel mechanism of MAPK pathway activation in thyroid cancer.',NULL,'115'),(1090,NULL,'Cin H et al','doi: 10.1007/s00401-011-0817-z','6','Acta neuropathologica',NULL,'763-74','21424530','2011 Jun','Oncogenic FAM131B-BRAF fusion resulting from 7q34 deletion comprises an alternative mechanism of MAPK pathway activation in pilocytic astrocytoma.',NULL,'121'),(1091,NULL,'Hutchinson KE et al','doi: 10.1158/1078-0432.CCR-13-1746','24','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'6696-702','24345920','2013 Dec 15','BRAF fusions define a distinct molecular subset of melanomas with potential sensitivity to MEK inhibition.',NULL,'19'),(1092,NULL,'Palanisamy N et al','doi: 10.1038/nm.2166','7','Nature medicine',NULL,'793-8','20526349','2010 Jul','Rearrangements of the RAF kinase pathway in prostate cancer, gastric cancer and melanoma.',NULL,'16'),(1093,NULL,'Piotrowska Z et al','doi: 10.1158/2159-8290.CD-18-1022','12','Cancer discovery',NULL,'1529-1539','30257958','2018 Dec','Landscape of Acquired Resistance to Osimertinib in <i>EGFR</i>-Mutant NSCLC and Clinical Validation of Combined EGFR and RET Inhibition with Osimertinib and BLU-667 for Acquired <i>RET</i> Fusion.',NULL,'8'),(1094,NULL,'Jones DT et al','doi: 10.1158/0008-5472.CAN-08-2097','21','Cancer research',NULL,'8673-7','18974108','2008 Nov 1','Tandem duplication producing a novel oncogenic BRAF fusion gene defines the majority of pilocytic astrocytomas.',NULL,'68'),(1095,NULL,'Yao Z et al','doi: 10.1038/nature23291','7666','Nature',NULL,'234-238','28783719','2017 Aug 10','Tumours with class 3 BRAF mutants are sensitive to the inhibition of activated RAS.',NULL,'548'),(1096,NULL,'Ross JS et al','doi: 10.1002/ijc.29825','4','International journal of cancer',NULL,'881-90','26314551','2016 Feb 15','The distribution of BRAF gene fusions in solid tumors and response to targeted therapy.',NULL,'138'),(1097,NULL,'Grisham RN et al','doi: 10.1200/JCO.2015.62.4726','34','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'4099-105','26324360','2015 Dec 1','Extreme Outlier Analysis Identifies Occult Mitogen-Activated Protein Kinase Pathway Mutations in Patients With Low-Grade Serous Ovarian Cancer.',NULL,'33'),(1098,NULL,'Lee NV et al','doi: 10.1371/journal.pone.0039653','6','PloS one',NULL,'e39653','22745804','2012','A novel SND1-BRAF fusion confers resistance to c-Met inhibitor PF-04217903 in GTL16 cells through [corrected] MAPK activation.',NULL,'7'),(1099,NULL,'Oxnard GR et al','doi: 10.1001/jamaoncol.2018.2969','11','JAMA oncology',NULL,'1527-1534','30073261','2018 Nov 1','Assessment of Resistance Mechanisms and Clinical Implications in Patients With EGFR T790M-Positive Lung Cancer and Acquired Resistance to Osimertinib.',NULL,'4'),(1100,NULL,'Jang JS et al','doi: 10.1038/srep09755','','Scientific reports',NULL,'9755','25985019','2015 May 18','Common Oncogene Mutations and Novel SND1-BRAF Transcript Fusion in Lung Adenocarcinoma from Never Smokers.',NULL,'5'),(1101,NULL,'Menzies AM et al','doi: 10.1111/pcmr.12388','5','Pigment cell & melanoma research',NULL,'607-10','26072686','2015 Sep','Clinical activity of the MEK inhibitor trametinib in metastatic melanoma containing BRAF kinase fusion.',NULL,'28'),(1102,NULL,'Pratilas CA et al','doi: 10.1073/pnas.0900780106','11','Proceedings of the National Academy of Sciences of the United States of America',NULL,'4519-24','19251651','2009 Mar 17','(V600E)BRAF is associated with disabled feedback inhibition of RAF-MEK signaling and elevated transcriptional output of the pathway.',NULL,'106'),(1103,NULL,'Wan PT et al','','6','Cell',NULL,'855-67','15035987','2004 Mar 19','Mechanism of activation of the RAF-ERK signaling pathway by oncogenic mutations of B-RAF.',NULL,'116'),(1104,NULL,'Kaley T et al','doi: 10.1200/JCO.2018.78.9990','35','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'3477-3484','30351999','2018 Dec 10','BRAF Inhibition in <i>BRAF</i><sup>V600</sup>-Mutant Gliomas: Results From the VE-BASKET Study.',NULL,'36'),(1105,NULL,'Cancer Genome Atlas Network.','doi: 10.1016/j.cell.2015.05.044','7','Cell',NULL,'1681-96','26091043','2015 Jun 18','Genomic Classification of Cutaneous Melanoma.',NULL,'161'),(1106,NULL,'Cancer Genome Atlas Research Network.','doi: 10.1016/j.cell.2014.09.050','3','Cell',NULL,'676-90','25417114','2014 Oct 23','Integrated genomic characterization of papillary thyroid carcinoma.',NULL,'159'),(1107,NULL,'Cardarella S et al','doi: 10.1158/1078-0432.CCR-13-0657','16','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'4532-40','23833300','2013 Aug 15','Clinical, pathologic, and biologic features associated with BRAF mutations in non-small cell lung cancer.',NULL,'19'),(1108,NULL,'Long GV et al','doi: 10.1016/S1470-2045(12)70431-X','11','The Lancet. Oncology',NULL,'1087-95','23051966','2012 Nov','Dabrafenib in patients with Val600Glu or Val600Lys BRAF-mutant melanoma metastatic to the brain (BREAK-MB): a multicentre, open-label, phase 2 trial.',NULL,'13'),(1109,NULL,'Falchook GS et al','doi: 10.1016/S0140-6736(12)60398-5','9829','Lancet (London, England)',NULL,'1893-901','22608338','2012 May 19','Dabrafenib in patients with melanoma, untreated brain metastases, and other solid tumours: a phase 1 dose-escalation trial.',NULL,'379'),(1110,NULL,'Hauschild A et al','doi: 10.1016/S0140-6736(12)60868-X','9839','Lancet (London, England)',NULL,'358-65','22735384','2012 Jul 28','Dabrafenib in BRAF-mutated metastatic melanoma: a multicentre, open-label, phase 3 randomised controlled trial.',NULL,'380'),(1111,NULL,'McArthur GA et al','doi: 10.1016/S1470-2045(14)70012-9','3','The Lancet. Oncology',NULL,'323-32','24508103','2014 Mar','Safety and efficacy of vemurafenib in BRAF(V600E) and BRAF(V600K) mutation-positive melanoma (BRIM-3): extended follow-up of a phase 3, randomised, open-label study.',NULL,'15'),(1112,NULL,'Robert C et al','doi: 10.1056/NEJMoa1412690','1','The New England journal of medicine',NULL,'30-9','25399551','2015 Jan 1','Improved overall survival in melanoma with combined dabrafenib and trametinib.',NULL,'372'),(1113,NULL,'Chapman PB et al','doi: 10.1093/annonc/mdx339','10','Annals of oncology : official journal of the European Society for Medical Oncology',NULL,'2581-2587','28961848','2017 Oct 1','Vemurafenib in patients with BRAFV600 mutation-positive metastatic melanoma: final overall survival results of the randomized BRIM-3 study.',NULL,'28'),(1114,NULL,'Planchard D et al','doi: 10.1016/S1470-2045(17)30679-4','10','The Lancet. Oncology',NULL,'1307-1316','28919011','2017 Oct','Dabrafenib plus trametinib in patients with previously untreated BRAF<sup>V600E</sup>-mutant metastatic non-small-cell lung cancer: an open-label, phase 2 trial.',NULL,'18'),(1115,NULL,'Planchard D et al','doi: 10.1016/S1470-2045(16)30146-2','7','The Lancet. Oncology',NULL,'984-993','27283860','2016 Jul','Dabrafenib plus trametinib in patients with previously treated BRAF(V600E)-mutant metastatic non-small cell lung cancer: an open-label, multicentre phase 2 trial.',NULL,'17'),(1116,NULL,'Subbiah V et al','doi: 10.1200/JCO.2017.73.6785','1','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'7-13','29072975','2018 Jan 1','Dabrafenib and Trametinib Treatment in Patients With Locally Advanced or Metastatic BRAF V600-Mutant Anaplastic Thyroid Cancer.',NULL,'36'),(1117,NULL,'Kopetz S et al','doi: 10.1056/NEJMoa1908075','17','The New England journal of medicine',NULL,'1632-1643','31566309','2019 Oct 24','Encorafenib, Binimetinib, and Cetuximab in <i>BRAF</i> V600E-Mutated Colorectal Cancer.',NULL,'381'),(1118,NULL,'Corcoran RB et al','doi: 10.1158/2159-8290.CD-17-1226','4','Cancer discovery',NULL,'428-443','29431699','2018 Apr','Combined BRAF, EGFR, and MEK Inhibition in Patients with <i>BRAF</i><sup>V600E</sup>-Mutant Colorectal Cancer.',NULL,'8'),(1119,NULL,'Allen CE et al','doi: 10.1182/asheducation-2015.1.559','','Hematology. American Society of Hematology. Education Program',NULL,'559-64','26637772','2015','Biological and clinical significance of somatic mutations in Langerhans cell histiocytosis and related histiocytic neoplastic disorders.',NULL,'2015'),(1120,NULL,'Haroche J et al','doi: 10.1200/JCO.2014.57.1950','5','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'411-8','25422482','2015 Feb 10','Reproducible and sustained efficacy of targeted therapy with vemurafenib in patients with BRAF(V600E)-mutated Erdheim-Chester disease.',NULL,'33'),(1121,NULL,'Pettirossi V et al','doi: 10.1182/blood-2014-10-603100','8','Blood',NULL,'1207-16','25480661','2015 Feb 19','BRAF inhibitors reverse the unique molecular signature and phenotype of hairy cell leukemia and exert potent antileukemic activity.',NULL,'125'),(1122,NULL,'Turakhia S et al','doi: 10.1309/AJCP5WVXJ2KTLODO','1','American journal of clinical pathology',NULL,'87-93','26071465','2015 Jul','Immunohistochemistry for BRAF V600E in the Differential Diagnosis of Hairy Cell Leukemia vs Other Splenic B-Cell Lymphomas.',NULL,'144'),(1123,NULL,'Andrulis M et al','doi: 10.1097/PAS.0b013e3182549b50','12','The American journal of surgical pathology',NULL,'1796-800','22531170','2012 Dec','Application of a BRAF V600E mutation-specific antibody for the diagnosis of hairy cell leukemia.',NULL,'36'),(1124,NULL,'Tiacci E et al','doi: 10.1182/blood-2011-08-371179','1','Blood',NULL,'192-5','22028477','2012 Jan 5','Simple genetic diagnosis of hairy cell leukemia by sensitive detection of the BRAF-V600E mutation.',NULL,'119'),(1125,NULL,'Xi L et al','doi: 10.1182/blood-2011-09-379339','14','Blood',NULL,'3330-2','22210875','2012 Apr 5','Both variant and IGHV4-34-expressing hairy cell leukemia lack the BRAF V600E mutation.',NULL,'119'),(1126,NULL,'Arcaini L et al','doi: 10.1182/blood-2011-08-368209','1','Blood',NULL,'188-91','22072557','2012 Jan 5','The BRAF V600E mutation in hairy cell leukemia and other mature B-cell neoplasms.',NULL,'119'),(1127,NULL,'Boyd EM et al','doi: 10.1111/j.1365-2141.2011.08868.x','5','British journal of haematology',NULL,'609-12','21910720','2011 Dec','High resolution melting analysis for detection of BRAF exon 15 mutations in hairy cell leukaemia and other lymphoid malignancies.',NULL,'155'),(1128,NULL,'Shao H et al','doi: 10.1016/j.leukres.2012.11.021','4','Leukemia research',NULL,'401-409','23347903','2013 Apr','Distinguishing hairy cell leukemia variant from hairy cell leukemia: development and validation of diagnostic criteria.',NULL,'37'),(1129,NULL,'Tiacci E et al','doi: 10.1056/NEJMoa1014209','24','The New England journal of medicine',NULL,'2305-15','21663470','2011 Jun 16','BRAF mutations in hairy-cell leukemia.',NULL,'364'),(1130,NULL,'Wang XJ et al','','7','International journal of clinical and experimental pathology',NULL,'4323-8','25120816','2014','Immunohistochemical analysis using a BRAF V600E mutation specific antibody is highly sensitive and specific for the diagnosis of hairy cell leukemia.',NULL,'7'),(1131,NULL,'Tiacci E et al','doi: 10.1056/NEJMoa1506583','18','The New England journal of medicine',NULL,'1733-47','26352686','2015 Oct 29','Targeting Mutant BRAF in Relapsed or Refractory Hairy-Cell Leukemia.',NULL,'373'),(1132,NULL,'Badalian-Very G et al','doi: 10.1182/blood-2010-04-279083','11','Blood',NULL,'1919-23','20519626','2010 Sep 16','Recurrent BRAF mutations in Langerhans cell histiocytosis.',NULL,'116'),(1133,NULL,'Allen CE et al','doi: 10.1056/NEJMra1607548','9','The New England journal of medicine',NULL,'856-868','30157397','2018 Aug 30','Langerhans-Cell Histiocytosis.',NULL,'379'),(1134,NULL,'Bubolz AM et al','','12','Oncotarget',NULL,'4060-70','24938183','2014 Jun 30','Potential clinical implications of BRAF mutations in histiocytic proliferations.',NULL,'5'),(1135,NULL,'Touat M et al','doi: 10.1212/WNL.0000000000006171','11','Neurology',NULL,'523-525','30120137','2018 Sep 11','Vemurafenib and cobimetinib overcome resistance to vemurafenib in <i>BRAF</i>-mutant ganglioglioma.',NULL,'91'),(1136,NULL,'Hyman DM et al','doi: 10.1056/NEJMoa1502309','8','The New England journal of medicine',NULL,'726-36','26287849','2015 Aug 20','Vemurafenib in Multiple Nonmelanoma Cancers with BRAF V600 Mutations.',NULL,'373'),(1137,NULL,'Marks AM et al','doi: 10.1002/pbc.26969','5','Pediatric blood & cancer',NULL,'e26969','29380516','2018 May','Response to the BRAF/MEK inhibitors dabrafenib/trametinib in an adolescent with a BRAF V600E mutated anaplastic ganglioglioma intolerant to vemurafenib.',NULL,'65'),(1138,NULL,'Brown NF et al','doi: 10.2217/cns-2017-0006','4','CNS oncology',NULL,'291-296','28984141','2017 Oct','Dabrafenib and trametinib in BRAFV600E mutated glioma.',NULL,'6'),(1139,NULL,'Zhang J et al','doi: 10.1038/nature10725','7380','Nature',NULL,'157-63','22237106','2012 Jan 11','The genetic basis of early T-cell precursor acute lymphoblastic leukaemia.',NULL,'481'),(1141,NULL,'Bowyer SE et al','doi: 10.1097/CMR.0000000000000099','5','Melanoma research',NULL,'504-8','24933606','2014 Oct','Activity of trametinib in K601E and L597Q BRAF mutation-positive metastatic melanoma.',NULL,'24'),(1142,NULL,'Voskoboynik M et al','doi: 10.1111/pcmr.12450','2','Pigment cell & melanoma research',NULL,'222-8','26643848','2016 Mar','Clinicopathological characteristics associated with BRAF(K601E) and BRAF(L597) mutations in melanoma.',NULL,'29'),(1143,NULL,'Dahlman KB et al','','9','Cancer discovery',NULL,'791-7','22798288','2012 Sep','BRAF(L597) mutations in melanoma are associated with sensitivity to MEK inhibitors.',NULL,'2'),(1144,NULL,'Dankner M et al','doi: 10.1158/1078-0432.CCR-17-3384','24','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'6483-6494','29903896','2018 Dec 15','Dual MAPK Inhibition Is an Effective Therapeutic Strategy for a Subset of Class II BRAF Mutant Melanomas.',NULL,'24'),(1145,NULL,'Smalley KS et al','doi: 10.1038/onc.2008.362','1','Oncogene',NULL,'85-94','18794803','2009 Jan 8','CRAF inhibition induces apoptosis in melanoma cells with non-V600E BRAF mutations.',NULL,'28'),(1146,NULL,'Diamond EL et al','doi: 10.1001/jamaoncol.2017.5029','3','JAMA oncology',NULL,'384-388','29188284','2018 Mar 1','Vemurafenib for BRAF V600-Mutant Erdheim-Chester Disease and Langerhans Cell Histiocytosis: Analysis of Data From the Histology-Independent, Phase 2, Open-label VE-BASKET Study.',NULL,'4'),(1147,NULL,'Dummer R et al','doi: 10.1016/S1470-2045(18)30142-6','5','The Lancet. Oncology',NULL,'603-615','29573941','2018 May','Encorafenib plus binimetinib versus vemurafenib or encorafenib in patients with BRAF-mutant melanoma (COLUMBUS): a multicentre, open-label, randomised phase 3 trial.',NULL,'19'),(1148,NULL,'Flaherty KT et al','doi: 10.1056/NEJMoa1210093','18','The New England journal of medicine',NULL,'1694-703','23020132','2012 Nov 1','Combined BRAF and MEK inhibition in melanoma with BRAF V600 mutations.',NULL,'367'),(1149,NULL,'Amaria RN et al','doi: 10.1016/S1470-2045(18)30015-9','2','The Lancet. Oncology',NULL,'181-193','29361468','2018 Feb','Neoadjuvant plus adjuvant dabrafenib and trametinib versus standard of care in patients with high-risk, surgically resectable melanoma: a single-centre, open-label, randomised, phase 2 trial.',NULL,'19'),(1150,NULL,'Johnson DB et al','doi: 10.1200/JCO.2014.57.3535','33','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'3697-704','25287827','2014 Nov 20','Combined BRAF (Dabrafenib) and MEK inhibition (Trametinib) in patients with BRAFV600-mutant melanoma experiencing progression with single-agent BRAF inhibitor.',NULL,'32'),(1151,NULL,'Long GV et al','doi: 10.1056/NEJMoa1708539','19','The New England journal of medicine',NULL,'1813-1823','28891408','2017 Nov 9','Adjuvant Dabrafenib plus Trametinib in Stage III BRAF-Mutated Melanoma.',NULL,'377'),(1152,NULL,'Sullivan RJ et al','doi: 10.1038/s41591-019-0474-7','6','Nature medicine',NULL,'929-935','31171876','2019 Jun','Atezolizumab plus cobimetinib and vemurafenib in BRAF-mutated melanoma patients.',NULL,'25'),(1153,NULL,'Long GV et al','doi: 10.1200/JCO.2017.74.1025','7','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'667-673','28991513','2018 Mar 1','Long-Term Outcomes in Patients With BRAF V600-Mutant Metastatic Melanoma Who Received Dabrafenib Combined With Trametinib.',NULL,'36'),(1154,NULL,'Ascierto PA et al','doi: 10.1038/s41591-019-0448-9','6','Nature medicine',NULL,'941-946','31171878','2019 Jun','Dabrafenib, trametinib and pembrolizumab or placebo in BRAF-mutant melanoma.',NULL,'25'),(1155,NULL,'Ribas A et al','doi: 10.1038/s41591-019-0476-5','6','Nature medicine',NULL,'936-940','31171879','2019 Jun','Combined BRAF and MEK inhibition with PD-1 blockade immunotherapy in BRAF-mutant melanoma.',NULL,'25'),(1156,NULL,'Long GV et al','doi: 10.1056/NEJMoa1406037','20','The New England journal of medicine',NULL,'1877-88','25265492','2014 Nov 13','Combined BRAF and MEK inhibition versus BRAF inhibition alone in melanoma.',NULL,'371'),(1157,NULL,'Ascierto PA et al','doi: 10.1016/S1470-2045(16)30122-X','9','The Lancet. Oncology',NULL,'1248-60','27480103','2016 Sep','Cobimetinib combined with vemurafenib in advanced BRAF(V600)-mutant melanoma (coBRIM): updated efficacy results from a randomised, double-blind, phase 3 trial.',NULL,'17'),(1158,NULL,'Larkin J et al','doi: 10.1056/NEJMoa1408868','20','The New England journal of medicine',NULL,'1867-76','25265494','2014 Nov 13','Combined vemurafenib and cobimetinib in BRAF-mutated melanoma.',NULL,'371'),(1159,NULL,'Gutzmer R et al','doi: 10.1016/S0140-6736(20)30934-X','10240','Lancet (London, England)',NULL,'1835-1844','32534646','2020 Jun 13','Atezolizumab, vemurafenib, and cobimetinib as first-line treatment for unresectable advanced BRAF<sup>V600</sup> mutation-positive melanoma (IMspire150): primary analysis of the randomised, double-blind, placebo-controlled, phase 3 trial.',NULL,'395'),(1160,NULL,'Bucheit AD et al','doi: 10.1002/cncr.28306','21','Cancer',NULL,'3821-9','23922205','2013 Nov 1','Clinical characteristics and outcomes with specific BRAF and NRAS mutations in patients with metastatic melanoma.',NULL,'119'),(1161,NULL,'Menzies AM et al','doi: 10.1158/1078-0432.CCR-12-0052','12','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'3242-9','22535154','2012 Jun 15','Distinguishing clinicopathologic features of patients with V600E and V600K BRAF-mutant metastatic melanoma.',NULL,'18'),(1162,NULL,'Pires da Silva I et al','doi: 10.1158/1078-0432.CCR-18-1680','4','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'1272-1279','30630828','2019 Feb 15','Distinct Molecular Profiles and Immunotherapy Treatment Outcomes of V600E and V600K <i>BRAF</i>-Mutant Melanoma.',NULL,'25'),(1163,NULL,'Flaherty KT et al','doi: 10.1056/NEJMoa1203421','2','The New England journal of medicine',NULL,'107-14','22663011','2012 Jul 12','Improved survival with MEK inhibition in BRAF-mutated melanoma.',NULL,'367'),(1164,NULL,'Ribas A et al','doi: 10.1158/1078-0432.CCR-18-4180','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'46-53','31732523','2020 Jan 1','Extended 5-Year Follow-up Results of a Phase Ib Study (BRIM7) of Vemurafenib and Cobimetinib in <i>BRAF</i>-Mutant Melanoma.',NULL,'26'),(1165,NULL,'Heinzerling L et al','doi: 10.1038/bjc.2013.143','10','British journal of cancer',NULL,'2164-71','23579220','2013 May 28','Rare BRAF mutations in melanoma patients: implications for molecular testing in clinical practice.',NULL,'108'),(1166,NULL,'Colombino M et al','doi: 10.1200/JCO.2011.41.2452','20','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'2522-9','22614978','2012 Jul 10','BRAF/NRAS mutation frequencies among primary tumors and metastases in patients with melanoma.',NULL,'30'),(1167,NULL,'Champion KJ et al','doi: 10.1111/j.1399-0004.2010.01495.x','5','Clinical genetics',NULL,'468-74','20735442','2011 May','Germline mutation in BRAF codon 600 is compatible with human development: de novo p.V600G mutation identified in a patient with CFC syndrome.',NULL,'79'),(1168,NULL,'Solit DB et al','','7074','Nature',NULL,'358-62','16273091','2006 Jan 19','BRAF mutation predicts sensitivity to MEK inhibition.',NULL,'439'),(1169,NULL,'Lovly CM et al','doi: 10.1371/journal.pone.0035309','4','PloS one',NULL,'e35309','22536370','2012','Routine multiplex mutational profiling of melanomas enables enrollment in genotype-driven therapeutic trials.',NULL,'7'),(1171,NULL,'Parakh S et al','doi: 10.1111/jcpt.12229','1','Journal of clinical pharmacy and therapeutics',NULL,'121-3','25382067','2015 Feb','Response to MAPK pathway inhibitors in BRAF V600M-mutated metastatic melanoma.',NULL,'40'),(1172,NULL,'Zhang C et al','doi: 10.1038/nature14982','7574','Nature',NULL,'583-6','26466569','2015 Oct 22','RAF inhibitors that evade paradoxical MAPK pathway activation.',NULL,'526'),(1173,NULL,'Tutuka CSA et al','doi: 10.1186/s12943-017-0684-x','1','Molecular cancer',NULL,'112','28659148','2017 Jun 28','PLX8394, a new generation BRAF inhibitor, selectively inhibits BRAF in colonic adenocarcinoma cells and prevents paradoxical MAPK pathway activation.',NULL,'16'),(1174,NULL,'Yao Z et al','doi: 10.1038/s41591-018-0274-5','2','Nature medicine',NULL,'284-291','30559419','2019 Feb','RAF inhibitor PLX8394 selectively disrupts BRAF dimers and RAS-independent BRAF-mutant-driven signaling.',NULL,'25'),(1176,NULL,'Kim KB et al','doi: 10.1200/JCO.2012.43.5966','4','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'482-9','23248257','2013 Feb 1','Phase II study of the MEK1/MEK2 inhibitor Trametinib in patients with metastatic BRAF-mutant cutaneous melanoma previously treated with or without a BRAF inhibitor.',NULL,'31'),(1177,NULL,'Marconcini R et al','doi: 10.1186/s40164-017-0067-4','','Experimental hematology & oncology',NULL,'6','28344857','2017','Metastatic BRAF K601E-mutated melanoma reaches complete response to MEK inhibitor trametinib administered for over 36months.',NULL,'6'),(1178,NULL,'Sarkozy A et al','doi: 10.1002/humu.20955','4','Human mutation',NULL,'695-702','19206169','2009 Apr','Germline BRAF mutations in Noonan, LEOPARD, and cardiofaciocutaneous syndromes: molecular diversity and associated phenotypic spectrum.',NULL,'30'),(1179,NULL,'Wilson MA et al','doi: 10.1016/j.cancergen.2014.06.022','6','Cancer genetics',NULL,'272-5','25178945','2014 Jun','What you are missing could matter: a rare, complex BRAF mutation affecting codons 599, 600, and 601 uncovered by next generation sequencing.',NULL,'207'),(1180,NULL,'Greaves WO et al','doi: 10.1016/j.jmoldx.2012.10.002','2','The Journal of molecular diagnostics : JMD',NULL,'220-6','23273605','2013 Mar','Frequency and spectrum of BRAF mutations in a retrospective, single-institution study of 1112 cases of melanoma.',NULL,'15'),(1181,NULL,'Lod L et al','doi: 10.1038/bcj.2013.60','','Blood cancer journal',NULL,'e163','24270325','2013 Nov 22','Lack of BRAF V600E mutation in human myeloma cell lines established from myeloma patients with extramedullary disease.',NULL,'3'),(1182,NULL,'Heidorn SJ et al','doi: 10.1016/j.cell.2009.12.040','2','Cell',NULL,'209-21','20141835','2010 Jan 22','Kinase-dead BRAF and oncogenic RAS cooperate to drive tumor progression through CRAF.',NULL,'140'),(1183,NULL,'Kamata T et al','doi: 10.1158/0008-5472.CAN-10-0603','21','Cancer research',NULL,'8475-86','20978199','2010 Nov 1','BRAF inactivation drives aneuploidy by deregulating CRAF.',NULL,'70'),(1186,NULL,'Zheng G et al','doi: 10.1186/s12885-015-1811-y','','BMC cancer',NULL,'779','26498038','2015 Oct 24','Clinical detection and categorization of uncommon and concomitant mutations involving BRAF.',NULL,'15'),(1187,NULL,'Anastasaki C et al','doi: 10.1093/hmg/ddp186','14','Human molecular genetics',NULL,'2543-54','19376813','2009 Jul 15','Kinase-activating and kinase-impaired cardio-facio-cutaneous syndrome alleles have activity during zebrafish development and are sensitive to small molecule inhibitors.',NULL,'18'),(1188,NULL,'Rodriguez-Viciana P et al','','5765','Science (New York, N.Y.)',NULL,'1287-90','16439621','2006 Mar 3','Germline mutations in genes within the MAPK pathway cause cardio-facio-cutaneous syndrome.',NULL,'311'),(1189,NULL,'Ikenoue T et al','','23','Cancer research',NULL,'8132-7','14678966','2003 Dec 1','Functional analysis of mutations within the kinase activation segment of B-Raf in human colorectal tumors.',NULL,'63'),(1190,NULL,'Garnett MJ et al','','6','Molecular cell',NULL,'963-9','16364920','2005 Dec 22','Wild-type and mutant B-RAF activate C-RAF through distinct mechanisms involving heterodimerization.',NULL,'20'),(1191,NULL,'Holderfield M et al','doi: 10.1016/j.ccr.2013.03.033','5','Cancer cell',NULL,'594-602','23680146','2013 May 13','RAF inhibitors activate the MAPK pathway by relieving inhibitory autophosphorylation.',NULL,'23'),(1192,NULL,'Ikenoue T et al','','10','Cancer research',NULL,'3428-35','15150094','2004 May 15','Different effects of point mutations within the B-Raf glycine-rich loop in colorectal tumors on mitogen-activated protein/extracellular signal-regulated kinase kinase/extracellular signal-regulated kinase and nuclear factor kappaB pathway and cellular transformation.',NULL,'64'),(1193,NULL,'Houben R et al','doi: 10.1186/1477-3163-3-6','','Journal of carcinogenesis',NULL,'6','15046639','2004','Constitutive activation of the Ras-Raf signaling pathway in metastatic melanoma is associated with poor prognosis.',NULL,'3'),(1194,NULL,'Hollestelle A et al','','2','Molecular cancer research : MCR',NULL,'195-201','17314276','2007 Feb','Phosphatidylinositol-3-OH kinase or RAS pathway mutations in human breast cancer cell lines.',NULL,'5'),(1195,NULL,'Richtig G et al','','3','Case reports in oncology',NULL,'543-546','27790118','2016 Sep-Dec','Two Case Reports of Rare BRAF Mutations in Exon 11 and Exon 15 with Discussion of Potential Treatment Options.',NULL,'9'),(1196,NULL,'Zhang L et al','doi: 10.1002/ajh.23652','5','American journal of hematology',NULL,'499-504','24446311','2014 May','BRAF kinase domain mutations are present in a subset of chronic myelomonocytic leukemia with wild-type RAS.',NULL,'89'),(1197,NULL,'Sen B et al','doi: 10.1126/scitranslmed.3003513','136','Science translational medicine',NULL,'136ra70','22649091','2012 May 30','Kinase-impaired BRAF mutations in lung cancer confer sensitivity to dasatinib.',NULL,'4'),(1198,NULL,'Ohashi K et al','doi: 10.1073/pnas.1203530109','31','Proceedings of the National Academy of Sciences of the United States of America',NULL,'E2127-33','22773810','2012 Jul 31','Lung cancers with acquired resistance to EGFR inhibitors occasionally harbor BRAF gene mutations but lack mutations in KRAS, NRAS, or MEK1.',NULL,'109'),(1199,NULL,'Rring M et al','doi: 10.1038/emboj.2012.100','11','The EMBO journal',NULL,'2629-47','22510884','2012 May 30','Distinct requirement for an intact dimer interface in wild-type, V600E and kinase-dead B-Raf signalling.',NULL,'31'),(1200,NULL,'Damm F et al','doi: 10.1158/2159-8290.CD-14-0104','9','Cancer discovery',NULL,'1088-101','24920063','2014 Sep','Acquired initiating mutations in early hematopoietic cells of CLL patients.',NULL,'4'),(1201,NULL,'Gautschi O et al','doi: 10.1016/j.lungcan.2013.08.012','2','Lung cancer (Amsterdam, Netherlands)',NULL,'365-7','24035431','2013 Nov','Lung adenocarcinoma with BRAF G469L mutation refractory to vemurafenib.',NULL,'82'),(1202,NULL,'Chang MT et al','doi: 10.1038/nbt.3391','2','Nature biotechnology',NULL,'155-63','26619011','2016 Feb','Identifying recurrent mutations in cancer reveals widespread lineage diversity and mutational specificity.',NULL,'34'),(1203,NULL,'Hussain MR et al','doi: 10.1016/j.sjbs.2014.10.002','4','Saudi journal of biological sciences',NULL,'359-73','26150740','2015 Jul','BRAF gene: From human cancers to developmental syndromes.',NULL,'22'),(1204,NULL,'Libra M et al','','10','Cell cycle (Georgetown, Tex.)',NULL,'1382-4','16096377','2005 Oct','Analysis of BRAF mutation in primary and metastatic melanoma.',NULL,'4'),(1205,NULL,'Guo Y et al','doi: 10.1002/path.4176','1','The Journal of pathology',NULL,'17-27','23401075','2013 May','TSC1 involvement in bladder cancer: diverse effects and therapeutic implications.',NULL,'230'),(1206,NULL,'Casadei Gardini A et al','doi: 10.1186/s12885-016-2463-2','','BMC cancer',NULL,'429','27388325','2016 Jul 7','Efficacy of sorafenib in BRAF-mutated non-small-cell lung cancer (NSCLC) and no response in synchronous BRAF wild type-hepatocellular carcinoma: a case report.',NULL,'16'),(1207,NULL,'Luk PP et al','doi: 10.3978/j.issn.2218-6751.2014.08.08','2','Translational lung cancer research',NULL,'142-8','25870796','2015 Apr','BRAF mutations in non-small cell lung cancer.',NULL,'4'),(1208,NULL,'Marquette A et al','doi: 10.1038/nsmb.2022','5','Nature structural & molecular biology',NULL,'584-91','21478863','2011 May','ERK and PDE4 cooperate to induce RAF isoform switching in melanoma.',NULL,'18'),(1209,NULL,'Niihori T et al','','3','Nature genetics',NULL,'294-6','16474404','2006 Mar','Germline KRAS and BRAF mutations in cardio-facio-cutaneous syndrome.',NULL,'38'),(1210,NULL,'Razzaque MA et al','','8','Nature genetics',NULL,'1013-7','17603482','2007 Aug','Germline gain-of-function mutations in RAF1 cause Noonan syndrome.',NULL,'39'),(1211,NULL,'Brose MS et al','','23','Cancer research',NULL,'6997-7000','12460918','2002 Dec 1','BRAF and RAS mutations in human lung cancer and melanoma.',NULL,'62'),(1212,NULL,'Haling JR et al','doi: 10.1016/j.ccr.2014.07.007','3','Cancer cell',NULL,'402-413','25155755','2014 Sep 8','Structure of the BRAF-MEK complex reveals a kinase activity independent role for BRAF in MAPK signaling.',NULL,'26'),(1213,NULL,'Hu J et al','doi: 10.1128/MCB.00943-14','1','Molecular and cellular biology',NULL,'264-76','25348715','2015 Jan','Kinase regulation by hydrophobic spine assembly in cancer.',NULL,'35'),(1214,NULL,'Rebocho AP et al','doi: 10.1038/onc.2012.330','26','Oncogene',NULL,'3207-12','22926515','2013 Jun 27','ARAF acts as a scaffold to stabilize BRAF:CRAF heterodimers.',NULL,'32'),(1215,NULL,'Dela Cruz FS et al','doi: 10.1186/s13073-016-0366-0','1','Genome medicine',NULL,'116','27799065','2016 Oct 31','A case study of an integrative genomic and experimental therapeutic approach for rare tumors: identification of vulnerabilities in a pediatric poorly differentiated carcinoma.',NULL,'8'),(1216,NULL,'Sievert AJ et al','doi: 10.1073/pnas.1219232110','15','Proceedings of the National Academy of Sciences of the United States of America',NULL,'5957-62','23533272','2013 Apr 9','Paradoxical activation and RAF inhibitor resistance of BRAF protein kinase fusions characterizing pediatric astrocytomas.',NULL,'110'),(1218,NULL,'Milewska M et al','doi: 10.1371/journal.pone.0129859','6','PloS one',NULL,'e0129859','26065894','2015','Mitogen-Inducible Gene-6 Mediates Feedback Inhibition from Mutated BRAF towards the Epidermal Growth Factor Receptor and Thereby Limits Malignant Transformation.',NULL,'10'),(1219,NULL,'Hoogstraat M et al','doi: 10.1111/pcmr.12347','3','Pigment cell & melanoma research',NULL,'318-23','25515853','2015 May','Detailed imaging and genetic analysis reveal a secondary BRAF(L505H) resistance mutation and extensive intrapatient heterogeneity in metastatic BRAF mutant melanoma patients treated with vemurafenib.',NULL,'28'),(1220,NULL,'Wagenaar TR et al','doi: 10.1111/pcmr.12171','1','Pigment cell & melanoma research',NULL,'124-33','24112705','2014 Jan','Resistance to vemurafenib resulting from a novel mutation in the BRAFV600E kinase domain.',NULL,'27'),(1221,NULL,'Choi J et al','doi: 10.1111/pcmr.12197','2','Pigment cell & melanoma research',NULL,'253-62','24283590','2014 Mar','Identification of PLX4032-resistance mechanisms and implications for novel RAF inhibitors.',NULL,'27'),(1222,NULL,'Whittaker S et al','doi: 10.1126/scitranslmed.3000758','35','Science translational medicine',NULL,'35ra41','20538618','2010 Jun 9','Gatekeeper mutations mediate resistance to BRAF-targeted therapies.',NULL,'2'),(1223,NULL,'Daub H et al','','12','Nature reviews. Drug discovery',NULL,'1001-10','15573099','2004 Dec','Strategies to overcome resistance to targeted protein kinase inhibitors.',NULL,'3'),(1224,NULL,'Seth R et al','doi: 10.1136/gut.2008.159137','9','Gut',NULL,'1234-41','19474002','2009 Sep','Concomitant mutations and splice variants in KRAS and BRAF demonstrate complex perturbation of the Ras/Raf signalling pathway in advanced colorectal cancer.',NULL,'58'),(1225,NULL,'Hoeflich KP et al','doi: 10.1158/0008-5472.CAN-08-3563','7','Cancer research',NULL,'3042-51','19276360','2009 Apr 1','Antitumor efficacy of the novel RAF inhibitor GDC-0879 is predicted by BRAFV600E mutational status and sustained extracellular signal-regulated kinase/mitogen-activated protein kinase pathway suppression.',NULL,'69'),(1226,NULL,'Hutchinson KE et al','','26','Oncotarget',NULL,'22348-60','26084293','2015 Sep 8','ERBB activation modulates sensitivity to MEK1/2 inhibition in a subset of driver-negative melanoma.',NULL,'6'),(1227,NULL,'Stones CJ et al','doi: 10.3389/fgene.2013.00066','','Frontiers in genetics',NULL,'66','23658559','2013','Comparison of responses of human melanoma cell lines to MEK and BRAF inhibitors.',NULL,'4'),(1228,NULL,'Kordes M et al','doi: 10.1038/leu.2015.319','4','Leukemia',NULL,'937-46','26582644','2016 Apr','Cooperation of BRAF(F595L) and mutant HRAS in histiocytic sarcoma provides new insights into oncogenic BRAF signaling.',NULL,'30'),(1229,NULL,'Santarpia L et al','doi: 10.1016/j.humpath.2008.11.003','6','Human pathology',NULL,'827-33','19200582','2009 Jun','Detection and molecular characterization of a novel BRAF activated domain mutation in follicular variant of papillary thyroid carcinoma.',NULL,'40'),(1230,NULL,'Cho U et al','doi: 10.3346/jkms.2014.29.8.1054','8','Journal of Korean medical science',NULL,'1054-60','25120313','2014 Aug','Clinicopathological features of rare BRAF mutations in Korean thyroid cancer patients.',NULL,'29'),(1231,NULL,'Wan L et al','doi: 10.1158/2159-8290.CD-16-0647','4','Cancer discovery',NULL,'424-441','28174173','2017 Apr','The APC/C E3 Ligase Complex Activator FZR1 Restricts BRAF Oncogenic Function.',NULL,'7'),(1232,NULL,'Poulikakos PI et al','doi: 10.1038/nature10662','7377','Nature',NULL,'387-90','22113612','2011 Nov 23','RAF inhibitor resistance is mediated by dimerization of aberrantly spliced BRAF(V600E).',NULL,'480'),(1233,NULL,'Chen SH et al','doi: 10.1158/2159-8290.CD-15-0896','3','Cancer discovery',NULL,'300-15','26732095','2016 Mar','Oncogenic BRAF Deletions That Function as Homodimers and Are Sensitive to Inhibition by RAF Dimer Inhibitor LY3009120.',NULL,'6'),(1234,NULL,'Estep AL et al','','12','PloS one',NULL,'e1279','18060073','2007 Dec 5','Mutation analysis of BRAF, MEK1 and MEK2 in 15 ovarian cancer cell lines: implications for therapy.',NULL,'2'),(1235,NULL,'Jones DT et al','doi: 10.1038/ng.2682','8','Nature genetics',NULL,'927-32','23817572','2013 Aug','Recurrent somatic alterations of FGFR1 and NTRK2 in pilocytic astrocytoma.',NULL,'45'),(1236,NULL,'Trubini S et al','doi: 10.3892/mco.2018.1555','3','Molecular and clinical oncology',NULL,'460-462','29456854','2018 Mar','Novel BRAF mutation in melanoma: A case report.',NULL,'8'),(1237,NULL,'Jones DT et al','doi: 10.1038/onc.2009.73','20','Oncogene',NULL,'2119-23','19363522','2009 May 21','Oncogenic RAF1 rearrangement and a novel BRAF mutation as alternatives to KIAA1549:BRAF fusion in activating the MAPK pathway in pilocytic astrocytoma.',NULL,'28'),(1238,NULL,'Eisenhardt AE et al','doi: 10.1002/ijc.25893','9','International journal of cancer',NULL,'2297-303','21190184','2011 Nov 1','Functional characterization of a BRAF insertion mutant associated with pilocytic astrocytoma.',NULL,'129'),(1239,NULL,'Moretti S et al','','30','Oncogene',NULL,'4235-40','16501605','2006 Jul 13','Biochemical and molecular characterization of the novel BRAF(V599Ins) mutation detected in a classic papillary thyroid carcinoma.',NULL,'25'),(1240,NULL,'Matsuse M et al','doi: 10.1002/ijc.27709','3','International journal of cancer',NULL,'738-43','22752848','2013 Feb 1','Functional characterization of the novel BRAF complex mutation, BRAF(V600delinsYM) , identified in papillary thyroid carcinoma.',NULL,'132'),(1242,NULL,'Hou P et al','','3','Cell cycle (Georgetown, Tex.)',NULL,'377-9','17297294','2007 Feb 1','Functional characterization of the T1799-1801del and A1799-1816ins BRAF mutations in papillary thyroid cancer.',NULL,'6'),(1243,NULL,'Kulkarni A et al','doi: 10.1158/1078-0432.CCR-16-0758','18','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'5631-5638','28539463','2017 Sep 15','<i>BRAF</i> Fusion as a Novel Mechanism of Acquired Resistance to Vemurafenib in <i>BRAF</i><sup>V600E</sup> Mutant Melanoma.',NULL,'23'),(1244,NULL,'Sato K et al','doi: 10.1158/1078-0432.CCR-18-1574','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'378-389','30279230','2019 Jan 1','Fusion Kinases Identified by Genomic Analyses of Sporadic Microsatellite Instability-High Colorectal Cancers.',NULL,'25'),(1245,NULL,'Botton T et al','doi: 10.1111/pcmr.12148','6','Pigment cell & melanoma research',NULL,'845-51','23890088','2013 Nov','Recurrent BRAF kinase fusions in melanocytic tumors offer an opportunity for targeted therapy.',NULL,'26'),(1246,NULL,'Ricarte-Filho JC et al','doi: 10.1172/JCI69766','11','The Journal of clinical investigation',NULL,'4935-44','24135138','2013 Nov','Identification of kinase fusion oncogenes in post-Chernobyl radiation-induced thyroid cancers.',NULL,'123'),(1247,NULL,'Vojnic M et al','doi: 10.1016/j.jtho.2018.12.038','5','Journal of thoracic oncology : official publication of the International Association for the Study of Lung Cancer',NULL,'802-815','30831205','2019 May','Acquired BRAF Rearrangements Induce Secondary Resistance to EGFR therapy in EGFR-Mutated Lung Cancers.',NULL,'14'),(1248,NULL,'Roth JJ et al','doi: 10.1111/bpa.12167','2','Brain pathology (Zurich, Switzerland)',NULL,'182-92','25040262','2015 Mar','Chromosome band 7q34 deletions resulting in KIAA1549-BRAF and FAM131B-BRAF fusions in pediatric low-grade Gliomas.',NULL,'25'),(1249,NULL,'Tomi TT et al','doi: 10.1371/journal.pone.0175638','4','PloS one',NULL,'e0175638','28448514','2017','A new GTF2I-BRAF fusion mediating MAPK pathway activation in pilocytic astrocytoma.',NULL,'12'),(1250,NULL,'Fangusaro J et al','doi: 10.1016/S1470-2045(19)30277-3','7','The Lancet. Oncology',NULL,'1011-1022','31151904','2019 Jul','Selumetinib in paediatric patients with BRAF-aberrant or neurofibromatosis type 1-associated recurrent, refractory, or progressive low-grade glioma: a multicentre, phase 2 trial.',NULL,'20'),(1251,NULL,'Chmielecki J et al','doi: 10.1158/2159-8290.CD-14-0617','12','Cancer discovery',NULL,'1398-405','25266736','2014 Dec','Comprehensive genomic profiling of pancreatic acinar cell carcinomas identifies recurrent RAF fusions and frequent inactivation of DNA repair genes.',NULL,'4'),(1252,NULL,'Corcoran RB et al','doi: 10.1126/scisignal.2001148','149','Science signaling',NULL,'ra84','21098728','2010 Nov 23','BRAF gene amplification can promote acquired resistance to MEK inhibitors in cancer cells harboring the BRAF V600E mutation.',NULL,'3'),(1253,NULL,'Shi H et al','doi: 10.1038/ncomms1727','','Nature communications',NULL,'724','22395615','2012 Mar 6','Melanoma whole-exome sequencing identifies (V600E)B-RAF amplification-mediated acquired B-RAF inhibitor resistance.',NULL,'3'),(1254,NULL,'Tanami H et al','','54','Oncogene',NULL,'8796-804','15467732','2004 Nov 18','Involvement of overexpressed wild-type BRAF in the growth of malignant melanoma cell lines.',NULL,'23'),(1255,NULL,'Li M et al','doi: 10.17219/acem/37332','6','Advances in clinical and experimental medicine : official organ Wroclaw Medical University',NULL,'893-9','25618114','2014 Nov-Dec','BRAF overexpression induces rampant glioma proliferation independent of phospho-EGFR expression.',NULL,'23'),(1256,NULL,'Villanueva J et al','doi: 10.1016/j.celrep.2013.08.023','6','Cell reports',NULL,'1090-9','24055054','2013 Sep 26','Concurrent MEK2 mutation and BRAF amplification confer resistance to BRAF and MEK inhibitors in melanoma.',NULL,'4'),(1257,NULL,'Wong CW et al','','6','Journal of clinical pathology',NULL,'640-4','15917418','2005 Jun','BRAF and NRAS mutations are uncommon in melanomas arising in diverse internal organs.',NULL,'58'),(1258,NULL,'Sasaki H et al','','5','Experimental and therapeutic medicine',NULL,'771-775','22969966','2012 May','Braf and erbB2 mutations correlate with smoking status in lung cancer patients.',NULL,'3'),(1259,NULL,'Zheng D et al','doi: 10.1245/s10434-015-4640-y','','Annals of surgical oncology',NULL,'S1284-91','26202550','2015 Dec','Prevalence and Clinicopathological Characteristics of BRAF Mutations in Chinese Patients with Lung Adenocarcinoma.',NULL,'22 Suppl 3'),(1260,NULL,'Kang S et al','','3','Gynecologic oncology',NULL,'662-6','17360030','2007 Jun','Inverse correlation between RASSF1A hypermethylation, KRAS and BRAF mutations in cervical adenocarcinoma.',NULL,'105'),(1261,NULL,'Okines AF et al','doi: 10.1016/j.ejca.2013.02.007','9','European journal of cancer (Oxford, England : 1990)',NULL,'2116-25','23481512','2013 Jun','Biomarker analysis in oesophagogastric cancer: Results from the REAL3 and TransMAGIC trials.',NULL,'49'),(1262,NULL,'Carrera C et al','doi: 10.1001/jamadermatol.2014.4115','5','JAMA dermatology',NULL,'544-8','25651238','2015 May','Multiple BRAF Wild-Type Melanomas During Dabrafenib Treatment for Metastatic BRAF-Mutant Melanoma.',NULL,'151'),(1263,NULL,'Chiappetta C et al','doi: 10.1097/PAI.0000000000000071','3','Applied immunohistochemistry & molecular morphology : AIMM',NULL,'172-7','24710085','2015 Mar','BRAF and NRAS mutations are heterogeneous and not mutually exclusive in nodular melanoma.',NULL,'23'),(1264,NULL,'Shain AH et al','doi: 10.1056/NEJMoa1502583','20','The New England journal of medicine',NULL,'1926-36','26559571','2015 Nov 12','The Genetic Evolution of Melanoma from Precursor Lesions.',NULL,'373'),(1265,NULL,'Lionetti M et al','','27','Oncotarget',NULL,'24205-17','26090869','2015 Sep 15','Molecular spectrum of BRAF, NRAS and KRAS gene mutations in plasma cell dyscrasias: implication for MEK-ERK pathway activation.',NULL,'6'),(1266,NULL,'Thomas C et al','doi: 10.1097/PAT.0000000000000245','4','Pathology',NULL,'349-54','25938346','2015 Jun','BRAF mutation detection in hairy cell leukaemia from archival haematolymphoid specimens.',NULL,'47'),(1267,NULL,'Lu H et al','doi: 10.1158/0008-5472.CAN-16-2745','13','Cancer research',NULL,'3502-3512','28512244','2017 Jul 1','Engineering and Functional Characterization of Fusion Genes Identifies Novel Oncogenic Drivers of Cancer.',NULL,'77'),(1268,NULL,'Wang J et al','doi: 10.1158/2159-8290.CD-17-1263','9','Cancer discovery',NULL,'1130-1141','29880583','2018 Sep','A Secondary Mutation in <i>BRAF</i> Confers Resistance to RAF Inhibition in a <i>BRAF</i><sup>V600E</sup>-Mutant Brain Tumor.',NULL,'8'),(1269,NULL,'Kemper K et al','doi: 10.1016/j.celrep.2016.05.064','1','Cell reports',NULL,'263-277','27320919','2016 Jun 28','BRAF(V600E) Kinase Domain Duplication Identified in Therapy-Refractory Melanoma Patient-Derived Xenografts.',NULL,'16'),(1270,NULL,'Klempner SJ et al','doi: 10.1001/jamaoncol.2015.4437','2','JAMA oncology',NULL,'272-4','26562024','2016 Feb','Identification of BRAF Kinase Domain Duplications Across Multiple Tumor Types and Response to RAF Inhibitor Therapy.',NULL,'2'),(1801,NULL,'Cancer Genome Atlas Research Network.','doi: 10.1038/nature10166','7353','Nature',NULL,'609-15','21720365','2011 Jun 29','Integrated genomic analyses of ovarian carcinoma.',NULL,'474'),(2261,NULL,'Cancer Genome Atlas Network.','doi: 10.1038/nature11252','7407','Nature',NULL,'330-7','22810696','2012 Jul 18','Comprehensive molecular characterization of human colon and rectal cancer.',NULL,'487'),(2816,NULL,'Brennan CW et al','doi: 10.1016/j.cell.2013.09.034','2','Cell',NULL,'462-77','24120142','2013 Oct 10','The somatic genomic landscape of glioblastoma.',NULL,'155'),(3225,NULL,'Hainsworth JD et al','doi: 10.1200/JCO.2017.75.3780','6','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'536-542','29320312','2018 Feb 20','Targeted Therapy for Advanced Solid Tumors on the Basis of Molecular Profiles: Results From MyPathway, an Open-Label, Phase IIa Multiple Basket Study.',NULL,'36'),(3243,NULL,'Parsons DW et al','doi: 10.1126/science.1164382','5897','Science (New York, N.Y.)',NULL,'1807-12','18772396','2008 Sep 26','An integrated genomic analysis of human glioblastoma multiforme.',NULL,'321'),(3244,NULL,'Cancer Genome Atlas Research Network.','doi: 10.1038/nature07385','7216','Nature',NULL,'1061-8','18772890','2008 Oct 23','Comprehensive genomic characterization defines human glioblastoma genes and core pathways.',NULL,'455'),(3322,NULL,'Takeuchi K et al','doi: 10.1038/nm.2658','3','Nature medicine',NULL,'378-81','22327623','2012 Feb 12','RET, ROS1 and ALK fusions in lung cancer.',NULL,'18'),(5379,NULL,'Sullivan RJ et al','doi: 10.1158/2159-8290.CD-17-1119','2','Cancer discovery',NULL,'184-195','29247021','2018 Feb','First-in-Class ERK1/2 Inhibitor Ulixertinib (BVD-523) in Patients with MAPK Mutant Advanced Solid Tumors: Results of a Phase I Dose-Escalation and Expansion Study.',NULL,'8'),(5677,NULL,'Van Allen EM et al','doi: 10.1158/2159-8290.CD-13-0617','1','Cancer discovery',NULL,'94-109','24265153','2014 Jan','The genetic landscape of clinical resistance to RAF inhibition in metastatic melanoma.',NULL,'4'),(5682,NULL,'Gao J et al','doi: 10.1186/s13073-016-0393-x','1','Genome medicine',NULL,'4','28115009','2017 Jan 23','3D clusters of somatic mutations in cancer reveal numerous rare mutations as functional targets.',NULL,'9'),(6367,NULL,'Barretina J et al','doi: 10.1038/nature11003','7391','Nature',NULL,'603-7','22460905','2012 Mar 28','The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity.',NULL,'483'),(6660,NULL,'Ngan ES et al','doi: 10.1093/jnci/djn471','3','Journal of the National Cancer Institute',NULL,'162-75','19176457','2009 Feb 4','A germline mutation (A339V) in thyroid transcription factor-1 (TITF-1/NKX2.1) in patients with multinodular goiter and papillary thyroid carcinoma.',NULL,'101'),(6691,NULL,'Agrawal N et al','doi: 10.1126/science.1206923','6046','Science (New York, N.Y.)',NULL,'1154-7','21798897','2011 Aug 26','Exome sequencing of head and neck squamous cell carcinoma reveals inactivating mutations in NOTCH1.',NULL,'333'),(6865,NULL,'Drilon A et al','doi: 10.1158/2159-8290.CD-16-1237','4','Cancer discovery',NULL,'400-409','28183697','2017 Apr','Safety and Antitumor Activity of the Multitargeted Pan-TRK, ROS1, and ALK Inhibitor Entrectinib: Combined Results from Two Phase I Trials (ALKA-372-001 and STARTRK-1).',NULL,'7'),(6894,NULL,'Stransky N et al','doi: 10.1038/ncomms5846','','Nature communications',NULL,'4846','25204415','2014 Sep 10','The landscape of kinase fusions in cancer.',NULL,'5'),(7385,NULL,'Giannakis M et al','doi: 10.1016/j.celrep.2016.03.075','4','Cell reports',NULL,'857-865','27149842','2016 Apr 26','Genomic Correlates of Immune-Cell Infiltrates in Colorectal Carcinoma.',NULL,'15'),(7526,NULL,'Wang Z et al','doi: 10.1016/j.canlet.2013.08.032','1','Cancer letters',NULL,'36-42','23981578','2014 Jan 1','Two co-existing germline mutations P53 V157D and PMS2 R20Q promote tumorigenesis in a familial cancer syndrome.',NULL,'342'),(7755,NULL,'Bonilla X et al','doi: 10.1038/ng.3525','4','Nature genetics',NULL,'398-406','26950094','2016 Apr','Genomic analysis identifies new drivers and progression pathways in skin basal cell carcinoma.',NULL,'48'),(8073,NULL,'Giacomini CP et al','doi: 10.1371/journal.pgen.1003464','4','PLoS genetics',NULL,'e1003464','23637631','2013 Apr','Breakpoint analysis of transcriptional and genomic profiles uncovers novel gene fusions spanning multiple human cancer types.',NULL,'9'),(8158,NULL,'Imielinski M et al','doi: 10.1016/j.cell.2012.08.029','6','Cell',NULL,'1107-20','22980975','2012 Sep 14','Mapping the hallmarks of lung adenocarcinoma with massively parallel sequencing.',NULL,'150'),(8237,NULL,'Govindan R et al','doi: 10.1016/j.cell.2012.08.024','6','Cell',NULL,'1121-34','22980976','2012 Sep 14','Genomic landscape of non-small cell lung cancer in smokers and never-smokers.',NULL,'150'),(8403,NULL,'Giannakis M et al','doi: 10.1038/ng.3127','12','Nature genetics',NULL,'1264-6','25344691','2014 Dec','RNF43 is frequently mutated in colorectal and endometrial cancers.',NULL,'46'),(8431,NULL,'Rikova K et al','','6','Cell',NULL,'1190-203','18083107','2007 Dec 14','Global survey of phosphotyrosine signaling identifies oncogenic kinases in lung cancer.',NULL,'131'),(8432,NULL,'Uguen A et al','doi: 10.2217/fon-2016-0050','16','Future oncology (London, England)',NULL,'1911-28','27256160','2016 Aug','ROS1 fusions in cancer: a review.',NULL,'12'),(8433,NULL,'Lee J et al','doi: 10.1002/cncr.27967','9','Cancer',NULL,'1627-35','23400546','2013 May 1','Identification of ROS1 rearrangement in gastric adenocarcinoma.',NULL,'119'),(8434,NULL,'Birchmeier C et al','','24','Proceedings of the National Academy of Sciences of the United States of America',NULL,'9270-4','2827175','1987 Dec','Expression and rearrangement of the ROS1 gene in human glioblastoma cells.',NULL,'84'),(8435,NULL,'Gainor JF et al','doi: 10.1634/theoncologist.2013-0095','7','The oncologist',NULL,'865-75','23814043','2013','Novel targets in non-small cell lung cancer: ROS1 and RET fusions.',NULL,'18'),(8436,NULL,'Karayan-Tapon L et al','doi: 10.1016/j.ejca.2014.06.001','13','European journal of cancer (Oxford, England : 1990)',NULL,'2364-6','24999209','2014 Sep','Lack of GOPC-ROS1 (FIG-ROS1) rearrangement in adult human gliomas.',NULL,'50'),(8437,NULL,'Birch AH et al','doi: 10.1371/journal.pone.0028250','12','PloS one',NULL,'e28250','22163003','2011','Chromosome 3 anomalies investigated by genome wide SNP analysis of benign, low malignant potential and low grade ovarian serous tumours.',NULL,'6'),(8438,NULL,'Bergethon K et al','doi: 10.1200/JCO.2011.35.6345','8','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'863-70','22215748','2012 Mar 10','ROS1 rearrangements define a unique molecular class of lung cancers.',NULL,'30'),(8439,NULL,'Gu TL et al','doi: 10.1371/journal.pone.0015640','1','PloS one',NULL,'e15640','21253578','2011 Jan 6','Survey of tyrosine kinase signaling reveals ROS kinase fusions in human cholangiocarcinoma.',NULL,'6'),(8440,NULL,'Acquaviva J et al','doi: 10.1016/j.bbcan.2008.07.006','1','Biochimica et biophysica acta',NULL,'37-52','18778756','2009 Jan','The multifaceted roles of the receptor tyrosine kinase ROS in development and cancer.',NULL,'1795'),(8441,NULL,'Davies KD et al','doi: 10.1158/1078-0432.CCR-12-0550','17','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'4570-9','22919003','2012 Sep 1','Identifying and targeting ROS1 gene fusions in non-small cell lung cancer.',NULL,'18'),(8442,NULL,'Davies KD et al','doi: 10.1158/1078-0432.CCR-12-2851','15','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'4040-5','23719267','2013 Aug 1','Molecular pathways: ROS1 fusion proteins in cancer.',NULL,'19'),(8443,NULL,'Roskoski R Jr','doi: 10.1016/j.phrs.2017.04.022','','Pharmacological research',NULL,'202-212','28465216','2017 Jul','ROS1 protein-tyrosine kinase inhibitors in the treatment of ROS1 fusion protein-driven non-small cell lung cancers.',NULL,'121'),(8444,NULL,'Shaw AT et al','doi: 10.1016/S1470-2045(19)30655-2','12','The Lancet. Oncology',NULL,'1691-1701','31669155','2019 Dec','Lorlatinib in advanced ROS1-positive non-small-cell lung cancer: a multicentre, open-label, single-arm, phase 1-2 trial.',NULL,'20'),(8445,NULL,'Shaw AT et al','doi: 10.1056/NEJMoa1406766','21','The New England journal of medicine',NULL,'1963-71','25264305','2014 Nov 20','Crizotinib in ROS1-rearranged non-small-cell lung cancer.',NULL,'371'),(8446,NULL,'Shaw AT et al','doi: 10.1093/annonc/mdz131','7','Annals of oncology : official journal of the European Society for Medical Oncology',NULL,'1121-1126','30980071','2019 Jul 1','Crizotinib in ROS1-rearranged advanced non-small-cell lung cancer (NSCLC): updated results, including overall survival, from PROFILE 1001.',NULL,'30'),(8449,NULL,'Charest A et al','','1','Genes, chromosomes & cancer',NULL,'58-71','12661006','2003 May','Fusion of FIG to the receptor tyrosine kinase ROS in a glioblastoma with an interstitial del(6)(q21q21).',NULL,'37'),(8451,NULL,'Seo JS et al','doi: 10.1101/gr.145144.112','11','Genome research',NULL,'2109-19','22975805','2012 Nov','The transcriptional landscape and mutational profile of lung adenocarcinoma.',NULL,'22'),(8452,NULL,'Kohno T et al','doi: 10.3978/j.issn.2218-6751.2014.11.11','2','Translational lung cancer research',NULL,'156-64','25870798','2015 Apr','Beyond ALK-RET, ROS1 and other oncogene fusions in lung cancer.',NULL,'4'),(8453,NULL,'Davare MA et al','doi: 10.1158/1078-0432.CCR-18-1052','24','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'6471-6482','30171048','2018 Dec 15','Rare but Recurrent ROS1 Fusions Resulting From Chromosome 6q22 Microdeletions are Targetable Oncogenes in Glioma.',NULL,'24'),(8454,NULL,'Lovly CM et al','doi: 10.1158/2159-8290.CD-14-0377','8','Cancer discovery',NULL,'889-95','24875859','2014 Aug','Inflammatory myofibroblastic tumors harbor multiple potentially actionable kinase fusions.',NULL,'4'),(8455,NULL,'Katayama R et al','doi: 10.1158/1078-0432.CCR-14-1385','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'166-74','25351743','2015 Jan 1','Cabozantinib overcomes crizotinib resistance in ROS1 fusion-positive cancer.',NULL,'21'),(8457,NULL,'Awad MM et al','doi: 10.1056/NEJMoa1215530','25','The New England journal of medicine',NULL,'2395-401','23724914','2013 Jun 20','Acquired resistance to crizotinib from a mutation in CD74-ROS1.',NULL,'368'),(8458,NULL,'Gerlinger M et al','doi: 10.1056/NEJMc1309091','12','The New England journal of medicine',NULL,'1172-3','24047073','2013 Sep 19','Acquired resistance to crizotinib from a mutation in CD74-ROS1.',NULL,'369'),(8459,NULL,'Facchinetti F et al','','24','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'5983-5991','27401242','2016 Dec 15','Crizotinib-Resistant ROS1 Mutations Reveal a Predictive Kinase Inhibitor Sensitivity Model for ROS1- and ALK-Rearranged Lung Cancers.',NULL,'22'),(8460,NULL,'Song A et al','doi: 10.1158/1078-0432.CCR-14-1350','10','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'2379-87','25688157','2015 May 15','Molecular Changes Associated with Acquired Resistance to Crizotinib in ROS1-Rearranged Non-Small Cell Lung Cancer.',NULL,'21'),(8461,NULL,'Drilon A et al','doi: 10.1158/1078-0432.CCR-15-2013','10','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'2351-8','26673800','2016 May 15','A Novel Crizotinib-Resistant Solvent-Front Mutation Responsive to Cabozantinib Therapy in a Patient with ROS1-Rearranged Lung Cancer.',NULL,'22'),(8462,NULL,'Davare MA et al','doi: 10.1073/pnas.1319583110','48','Proceedings of the National Academy of Sciences of the United States of America',NULL,'19519-24','24218589','2013 Nov 26','Foretinib is a potent inhibitor of oncogenic ROS1 fusion proteins.',NULL,'110'),(8463,NULL,'Aljohani H et al','','24','Oncotarget',NULL,'20388-95','25978031','2015 Aug 21','ROS1 amplification mediates resistance to gefitinib in glioblastoma cells.',NULL,'6'),(9939,NULL,'Freed-Pastor WA et al','doi: 10.1101/gad.190678.112','12','Genes & development',NULL,'1268-86','22713868','2012 Jun 15','Mutant p53: one name, many proteins.',NULL,'26'),(9940,NULL,'Suzuki K et al','doi: 10.1155/2011/978312','','Journal of biomedicine & biotechnology',NULL,'978312','21765642','2011','Recent advances in p53 research and cancer treatment.',NULL,'2011'),(9941,NULL,'Vogelstein B et al','','6810','Nature',NULL,'307-10','11099028','2000 Nov 16','Surfing the p53 network.',NULL,'408'),(9942,NULL,'Kawaguchi T et al','','46','Oncogene',NULL,'6976-81','16007150','2005 Oct 20','The relationship among p53 oligomer formation, structure and transcriptional activity using a comprehensive missense mutation library.',NULL,'24'),(9943,NULL,'Jordan JJ et al','doi: 10.1158/1541-7786.MCR-09-0442','5','Molecular cancer research : MCR',NULL,'701-16','20407015','2010 May','Altered-function p53 missense mutations identified in breast cancers can have subtle effects on transactivation.',NULL,'8'),(9945,NULL,'Kakudo Y et al','','6','Cancer research',NULL,'2108-14','15781620','2005 Mar 15','Lack of correlation between p53-dependent transcriptional activity and the ability to induce apoptosis among 179 mutant p53s.',NULL,'65'),(9946,NULL,'Bullock AN et al','','10','Oncogene',NULL,'1245-56','10713666','2000 Mar 2','Quantitative analysis of residual folding and DNA binding in mutant p53 core domain: definition of mutant states for rescue in cancer therapy.',NULL,'19'),(9947,NULL,'Loizou E et al','doi: 10.1158/2159-8290.CD-18-1391','7','Cancer discovery',NULL,'962-979','31068365','2019 Jul','A Gain-of-Function p53-Mutant Oncogene Promotes Cell Fate Plasticity and Myeloid Leukemia through the Pluripotency Factor FOXH1.',NULL,'9'),(9948,NULL,'Wasserman JD et al','doi: 10.1200/JCO.2013.52.6863','6','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'602-9','25584008','2015 Feb 20','Prevalence and functional consequence of TP53 mutations in pediatric adrenocortical carcinoma: a children\'s oncology group study.',NULL,'33'),(9949,NULL,'Xu J et al','doi: 10.1038/nchembio.546','5','Nature chemical biology',NULL,'285-95','21445056','2011 May','Gain of function of mutant p53 by coaggregation with multiple tumor suppressors.',NULL,'7'),(9950,NULL,'Cho Y et al','','5170','Science (New York, N.Y.)',NULL,'346-55','8023157','1994 Jul 15','Crystal structure of a p53 tumor suppressor-DNA complex: understanding tumorigenic mutations.',NULL,'265'),(9951,NULL,'Bullock AN et al','','1','Nature reviews. Cancer',NULL,'68-76','11900253','2001 Oct','Rescuing the function of mutant p53.',NULL,'1'),(9952,NULL,'Imagawa T et al','doi: 10.1016/j.ab.2009.01.030','2','Analytical biochemistry',NULL,'249-56','19454241','2009 Apr 15','Evaluation of transcriptional activity of p53 in individual living mammalian cells.',NULL,'387'),(9953,NULL,'Itahana Y et al','doi: 10.1074/jbc.M805696200','8','The Journal of biological chemistry',NULL,'5158-64','19106109','2009 Feb 20','p53 Oligomerization is essential for its C-terminal lysine acetylation.',NULL,'284'),(9954,NULL,'Kamada R et al','doi: 10.1074/jbc.M110.174698','1','The Journal of biological chemistry',NULL,'252-8','20978130','2011 Jan 7','Cancer-associated p53 tetramerization domain mutants: quantitative analysis reveals a low threshold for tumor suppressor inactivation.',NULL,'286'),(9955,NULL,'DiGiammarino EL et al','','1','Nature structural biology',NULL,'12-6','11753428','2002 Jan','A novel mechanism of tumorigenesis involving pH-dependent destabilization of a mutant p53 tetramer.',NULL,'9'),(9956,NULL,'Petitjean A et al','','6','Human mutation',NULL,'622-9','17311302','2007 Jun','Impact of mutant p53 functional properties on TP53 mutation patterns and tumor phenotype: lessons from recent developments in the IARC TP53 database.',NULL,'28'),(9957,NULL,'Ribeiro RC et al','','16','Proceedings of the National Academy of Sciences of the United States of America',NULL,'9330-5','11481490','2001 Jul 31','An inherited p53 mutation that contributes in a tissue-specific manner to pediatric adrenal cortical carcinoma.',NULL,'98'),(9958,NULL,'Smith PD et al','','15','Oncogene',NULL,'2451-9','10229196','1999 Apr 15','Novel p53 mutants selected in BRCA-associated tumours which dissociate transformation suppression from other wild-type p53 functions.',NULL,'18'),(9959,NULL,'Baroni TE et al','','14','Proceedings of the National Academy of Sciences of the United States of America',NULL,'4930-5','15037740','2004 Apr 6','A global suppressor motif for p53 cancer mutants.',NULL,'101'),(9960,NULL,'Dearth LR et al','','2','Carcinogenesis',NULL,'289-98','16861262','2007 Feb','Inactive full-length p53 mutants lacking dominant wild-type p53 inhibition highlight loss of heterozygosity as an important aspect of p53 status in human cancers.',NULL,'28'),(9961,NULL,'Scian MJ et al','','25','Oncogene',NULL,'4430-43','15077194','2004 May 27','Tumor-derived p53 mutants induce oncogenesis by transactivating growth-promoting genes.',NULL,'23'),(9962,NULL,'Berge EO et al','','3','Biochimica et biophysica acta',NULL,'2790-7','23246812','2013 Mar','Functional characterisation of p53 mutants identified in breast cancers with suboptimal responses to anthracyclines or mitomycin.',NULL,'1830'),(9963,NULL,'Khoueiry R et al','doi: 10.1136/jmg.2008.057943','9','Journal of medical genetics',NULL,'583-8','18762571','2008 Sep','Dynamic CpG methylation of the KCNQ1OT1 gene during maturation of human oocytes.',NULL,'45'),(9964,NULL,'Quesnel S et al','','27','Oncogene',NULL,'3970-8','10435620','1999 Jul 8','p53 compound heterozygosity in a severely affected child with Li-Fraumeni syndrome.',NULL,'18'),(9965,NULL,'Tan BS et al','doi: 10.1038/cddis.2015.191','','Cell death & disease',NULL,'e1826','26181206','2015 Jul 16','Mutant p53-R273H mediates cancer cell survival and anoikis resistance through AKT-dependent suppression of BCL2-modifying factor (BMF).',NULL,'6'),(9966,NULL,'Xiao G et al','doi: 10.1158/0008-5472.CAN-19-1036','3','Cancer research',NULL,'394-405','31776133','2020 Feb 1','Gain-of-Function Mutant p53 R273H Interacts with Replicating DNA and PARP1 in Breast Cancer.',NULL,'80'),(9967,NULL,'Russell-Swetek A et al','doi: 10.1136/jmg.2008.059568','9','Journal of medical genetics',NULL,'603-6','18762572','2008 Sep','Identification of a novel TP53 germline mutation E285V in a rare case of paediatric adrenocortical carcinoma and choroid plexus carcinoma.',NULL,'45'),(9969,NULL,'Xu-Monette ZY et al','doi: 10.1182/blood-2012-05-433334','19','Blood',NULL,'3986-96','22955915','2012 Nov 8','Mutational profile and prognostic significance of TP53 in diffuse large B-cell lymphoma patients treated with R-CHOP: report from an International DLBCL Rituximab-CHOP Consortium Program Study.',NULL,'120'),(9970,NULL,'Mizuarai S et al','','12','Cancer research',NULL,'6319-26','16778209','2006 Jun 15','Mutant p53 induces the GEF-H1 oncogene, a guanine nucleotide exchange factor-H1 for RhoA, resulting in accelerated cell proliferation in tumor cells.',NULL,'66'),(9971,NULL,'Follis AV et al','doi: 10.1038/nsmb.2829','6','Nature structural & molecular biology',NULL,'535-43','24814347','2014 Jun','The DNA-binding domain mediates both nuclear and cytosolic functions of p53.',NULL,'21'),(9972,NULL,'Eldar A et al','doi: 10.1093/nar/gkt630','18','Nucleic acids research',NULL,'8748-59','23863845','2013 Oct','Structural studies of p53 inactivation by DNA-contact mutations and its rescue by suppressor mutations via alternative protein-DNA interactions.',NULL,'41'),(9973,NULL,'Odell AF et al','doi: 10.1074/jbc.M112.443168','23','The Journal of biological chemistry',NULL,'16704-14','23612969','2013 Jun 7','A novel p53 mutant found in iatrogenic urothelial cancers is dysfunctional and can be rescued by a second-site global suppressor mutation.',NULL,'288'),(9974,NULL,'Li J et al','doi: 10.1002/humu.22528','5','Human mutation',NULL,'575-84','24677579','2014 May','Mutants TP53 p.R273H and p.R273C but not p.R273G enhance cancer cell malignancy.',NULL,'35'),(9975,NULL,'Pan Y et al','','27','Oncogene',NULL,'3095-100','10871862','2000 Jun 22','Identification of a tumor-derived p53 mutant with novel transactivating selectivity.',NULL,'19'),(9976,NULL,'','','3','Hematology/oncology clinics of North America',NULL,'507-709','2358418','1990 Jun','Bone marrow transplantation.',NULL,'4'),(9977,NULL,'Pintus SS et al','doi: 10.1080/07391102.2012.691364','1','Journal of biomolecular structure & dynamics',NULL,'78-86','22803791','2013','The substitutions G245C and G245D in the Zn(2+)-binding pocket of the p53 protein result in differences of conformational flexibility of the DNA-binding domain.',NULL,'31'),(9978,NULL,'Zhang CZ et al','doi: 10.4161/cc.11.2.18758','2','Cell cycle (Georgetown, Tex.)',NULL,'322-34','22214764','2012 Jan 15','Interaction between ZBP-89 and p53 mutants and its contribution to effects of HDACi on hepatocellular carcinoma.',NULL,'11'),(9979,NULL,'Chuang HC et al','doi: 10.1371/journal.pone.0104821','8','PloS one',NULL,'e104821','25119136','2014','The p53-reactivating small molecule RITA induces senescence in head and neck cancer cells.',NULL,'9'),(9980,NULL,'Perez RE et al','doi: 10.1002/jcp.22285','2','Journal of cellular physiology',NULL,'394-405','20589832','2010 Nov','Restoration of DNA-binding and growth-suppressive activity of mutant forms of p53 via a PCAF-mediated acetylation pathway.',NULL,'225'),(9981,NULL,'Rauf SM et al','doi: 10.1007/s10930-012-9458-x','1','The protein journal',NULL,'68-74','23315175','2013 Jan','Effect of Y220C mutation on p53 and its rescue mechanism: a computer chemistry approach.',NULL,'32'),(9982,NULL,'Liu X et al','doi: 10.1093/nar/gkt305','12','Nucleic acids research',NULL,'6034-44','23630318','2013 Jul','Small molecule induced reactivation of mutant p53 in cancer cells.',NULL,'41'),(9983,NULL,'Capponcelli S et al','','2','Human mutation',NULL,'94-103','15977174','2005 Aug','Evaluation of the molecular mechanisms involved in the gain of function of a Li-Fraumeni TP53 mutation.',NULL,'26'),(9984,NULL,'Tu C et al','doi: 10.1107/S0907444908003338','Pt 5','Acta crystallographica. Section D, Biological crystallography',NULL,'471-7','18453682','2008 May','Impact of low-frequency hotspot mutation R282Q on the structure of p53 DNA-binding domain as revealed by crystallography at 1.54 angstroms resolution.',NULL,'64'),(9985,NULL,'Solomon H et al','doi: 10.1242/jcs.099663','Pt 13','Journal of cell science',NULL,'3144-52','22427690','2012 Jul 1','Various p53 mutant proteins differently regulate the Ras circuit to induce a cancer-related gene signature.',NULL,'125'),(9986,NULL,'Ashur-Fabian O et al','','5','Cell cycle (Georgetown, Tex.)',NULL,'589-94','17361096','2007 Mar 1','Apaf1 in chronic myelogenous leukemia (CML) progression: reduced Apaf1 expression is correlated with a H179R p53 mutation during clinical blast crisis.',NULL,'6'),(9987,NULL,'Yang D et al','','1-2','Molecular and cellular biochemistry',NULL,'219-26','17530187','2007 Oct','The over-expression of p53 H179Y residue mutation causes the increase of cyclin A1 and Cdk4 expression in HELF cells.',NULL,'304'),(9988,NULL,'Yeudall WA et al','doi: 10.1007/978-1-62703-236-0_11','','Methods in molecular biology (Clifton, N.J.)',NULL,'135-46','23150443','2013','Mutant p53 in cell adhesion and motility.',NULL,'962'),(9989,NULL,'Carson C et al','doi: 10.1111/j.1755-148X.2012.01010.x','4','Pigment cell & melanoma research',NULL,'514-26','22540896','2012 Jul','A prognostic signature of defective p53-dependent G1 checkpoint function in melanoma cell lines.',NULL,'25'),(9990,NULL,'Ahn J et al','','10','Cell cycle (Georgetown, Tex.)',NULL,'1603-15','19462533','2009 May 15','Dissection of the sequence-specific DNA binding and exonuclease  activities reveals a superactive yet apoptotically impaired mutant p53  protein.',NULL,'8'),(9991,NULL,'Di Como CJ et al','','19','Oncogene',NULL,'2527-39','9627118','1998 May 14','Human tumor-derived p53 proteins exhibit binding site selectivity and temperature sensitivity for transactivation in a yeast-based assay.',NULL,'16'),(9992,NULL,'Skinner HD et al','doi: 10.1158/1078-0432.CCR-11-2260','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'290-300','22090360','2012 Jan 1','TP53 disruptive mutations lead to head and neck cancer treatment failure through inhibition of radiation-induced senescence.',NULL,'18'),(9993,NULL,'Brachmann RK et al','','9','Proceedings of the National Academy of Sciences of the United States of America',NULL,'4091-5','8633021','1996 Apr 30','Dominant-negative p53 mutations selected in yeast hit cancer hot spots.',NULL,'93'),(9994,NULL,'Friedler A et al','','26','The Journal of biological chemistry',NULL,'24108-12','12700230','2003 Jun 27','Kinetic instability of p53 core domain mutants: implications for rescue by small molecules.',NULL,'278'),(9995,NULL,'Grochova D et al','','9','Oncogene',NULL,'1243-52','17724467','2008 Feb 21','Analysis of transactivation capability and conformation of p53 temperature-dependent mutants and their reactivation by amifostine in yeast.',NULL,'27'),(9996,NULL,'Friedler A et al','','1','Journal of molecular biology',NULL,'187-96','14741214','2004 Feb 6','Structural distortion of p53 by the mutation R249S and its rescue by a designed peptide: implications for \"mutant conformation\".',NULL,'336'),(9997,NULL,'Suad O et al','doi: 10.1016/j.jmb.2008.10.063','1','Journal of molecular biology',NULL,'249-65','18996393','2009 Jan 9','Structural basis of restoring sequence-specific DNA binding and transactivation to mutant p53 by suppressor mutations.',NULL,'385'),(9998,NULL,'Golubovskaya VM et al','','5','Molecular carcinogenesis',NULL,'373-82','17999388','2008 May','p53 regulates FAK expression in human tumor cells.',NULL,'47'),(9999,NULL,'Soragni A et al','doi: 10.1016/j.ccell.2015.12.002','1','Cancer cell',NULL,'90-103','26748848','2016 Jan 11','A Designed Inhibitor of p53 Aggregation Rescues p53 Tumor Suppression in Ovarian Carcinomas.',NULL,'29'),(10000,NULL,'Jagosova J et al','doi: 10.3892/ijo.2012.1520','3','International journal of oncology',NULL,'1157-63','22710932','2012 Sep','Transactivation and reactivation capabilities of temperature-dependent             p53 mutants in yeast and human cells.',NULL,'41'),(10001,NULL,'Stoner CS et al','doi: 10.1021/bi900757q','38','Biochemistry',NULL,'9156-69','19681600','2009 Sep 29','Effect of thioredoxin deletion and p53 cysteine replacement on human p53 activity in wild-type and thioredoxin reductase null yeast.',NULL,'48'),(10002,NULL,'Gorgoulis VG et al','','3','British journal of cancer',NULL,'374-84','9472631','1998','Effects of p53 mutants derived from lung carcinomas on the p53-responsive element (p53RE) of the MDM2 gene.',NULL,'77'),(10003,NULL,'Cardellino U et al','','6','Mutagenesis',NULL,'417-23','17947339','2007 Nov','Transcriptional properties of feline p53 and its tumour-associated mutants: a yeast-based approach.',NULL,'22'),(10005,NULL,'Jordan JJ et al','doi: 10.1073/pnas.1205971109','36','Proceedings of the National Academy of Sciences of the United States of America',NULL,'14387-92','22908277','2012 Sep 4','Low-level p53 expression changes transactivation rules and reveals superactivating sequences.',NULL,'109'),(10006,NULL,'Okada M et al','','3A','Anticancer research',NULL,'2023-8','16827139','2006 May-Jun','P53 mutants suppress ZBP-89 function.',NULL,'26'),(10007,NULL,'Campomenosi P et al','','27','Oncogene',NULL,'3573-9','11429705','2001 Jun 14','p53 mutants can often transactivate promoters containing a p21 but not Bax or PIG3 responsive elements.',NULL,'20'),(10008,NULL,'Vaughan CA et al','doi: 10.1016/j.bbrc.2012.09.029','1','Biochemical and biophysical research communications',NULL,'6-10','22989750','2012 Nov 9','Allele specific gain-of-function activity of p53 mutants in lung cancer cells.',NULL,'428'),(10009,NULL,'Wang B et al','doi: 10.1038/cdd.2013.132','4','Cell death and differentiation',NULL,'521-32','24076587','2014 Apr','Mapping the p53 transcriptome universe using p53 natural polymorphs.',NULL,'21'),(10010,NULL,'Galy O et al','doi: 10.1155/2011/697162','','Hepatitis research and treatment',NULL,'697162','21760996','2011','Mutations in TP53 and CTNNB1 in Relation to Hepatitis B and C Infections in Hepatocellular Carcinomas from Thailand.',NULL,'2011'),(10011,NULL,'Dong P et al','doi: 10.1038/onc.2012.334','27','Oncogene',NULL,'3286-95','22847613','2013 Jul 4','Mutant p53 gain-of-function induces epithelial-mesenchymal transition through modulation of the miR-130b-ZEB1 axis.',NULL,'32'),(10012,NULL,'Aranda M et al','','5','Oncology reports',NULL,'1213-7','17914575','2007 Nov','Loss of TP53-DNA interaction induced by p.C135R in lung cancer.',NULL,'18'),(10013,NULL,'Schaefer KN et al','doi: 10.1021/bi501424v','3','Biochemistry',NULL,'932-41','25584637','2015 Jan 27','Oxidation of p53 through DNA charge transport involves a network of disulfides within the DNA-binding domain.',NULL,'54'),(10014,NULL,'Slovackova J et al','','11','Cell cycle (Georgetown, Tex.)',NULL,'2141-8','20505364','2010 Jun 1','Transactivation by temperature-dependent p53 mutants in yeast and human cells.',NULL,'9'),(10015,NULL,'Nazeer FI et al','doi: 10.1038/onc.2011.29','27','Oncogene',NULL,'3073-83','21383700','2011 Jul 7','p53 inhibits mRNA 3\' processing through its interaction with the CstF/BARD1 complex.',NULL,'30'),(10016,NULL,'Monti P et al','','11','Oncogene',NULL,'1641-8','11896595','2002 Mar 7','Tumour p53 mutations exhibit promoter selective dominance over wild type p53.',NULL,'21'),(10017,NULL,'North S et al','','3','Molecular carcinogenesis',NULL,'181-8','11870884','2002 Mar','Restoration of wild-type conformation and activity of a temperature-sensitive mutant of p53 (p53(V272M)) by the cytoprotective aminothiol WR1065 in the esophageal cancer cell line TE-1.',NULL,'33'),(10018,NULL,'Calhoun S et al','doi: 10.1021/bi200192j','23','Biochemistry',NULL,'5345-53','21561095','2011 Jun 14','Structural effects of the L145Q, V157F, and R282W cancer-associated mutations in the p53 DNA-binding core domain.',NULL,'50'),(10019,NULL,'Boudreau HE et al','doi: 10.1038/bjc.2014.165','10','British journal of cancer',NULL,'2569-82','24714748','2014 May 13','Wild-type and mutant p53 differentially regulate NADPH oxidase 4 in TGF--mediated migration of human lung and breast epithelial cells.',NULL,'110'),(10020,NULL,'Lin C et al','doi: 10.1016/j.neulet.2012.09.022','1','Neuroscience letters',NULL,'60-5','22999923','2012 Oct 31','R280T mutation of p53 gene promotes proliferation of human glioma cells through GSK-3/PTEN pathway.',NULL,'529'),(10021,NULL,'Joerger AC et al','','15','Oncogene',NULL,'2226-42','17401432','2007 Apr 2','Structure-function-rescue: the diverse nature of common p53 cancer mutants.',NULL,'26'),(10022,NULL,'Jia S et al','doi: 10.7150/ijbs.4176','5','International journal of biological sciences',NULL,'596-605','22553460','2012','The gain of function of p53 mutant p53S in promoting tumorigenesis by cross-talking with H-RasV12.',NULL,'8'),(10023,NULL,'Monti P et al','','34','Oncogene',NULL,'5252-60','12917626','2003 Aug 14','Characterization of the p53 mutants ability to inhibit p73 beta transactivation using a yeast-based functional assay.',NULL,'22'),(10024,NULL,'Huqun et al','','8','Cancer science',NULL,'718-24','12901798','2003 Aug','A naturally occurring p73 mutation in a p73-p53 double-mutant lung cancer cell line encodes p73 alpha protein with a dominant-negative function.',NULL,'94'),(10025,NULL,'Hu H et al','doi: 10.1038/srep20221','','Scientific reports',NULL,'20221','26818906','2016 Jan 28','Genetic and functional analysis of a Li Fraumeni syndrome family in China.',NULL,'6'),(10026,NULL,'Portis T et al','','5','Journal of virology',NULL,'2185-93','11160722','2001 Mar','Analysis of p53 inactivation in a human T-cell leukemia virus type 1 Tax transgenic mouse model.',NULL,'75'),(10027,NULL,'Gurova KV et al','','11','Cancer research',NULL,'2905-12','12782597','2003 Jun 1','Cooperation of two mutant p53 alleles contributes to Fas resistance of prostate carcinoma cells.',NULL,'63'),(10028,NULL,'Zupnick A et al','','29','The Journal of biological chemistry',NULL,'20464-73','16687402','2006 Jul 21','Mutational analysis of the p53 core domain L1 loop.',NULL,'281'),(10029,NULL,'Pekova S et al','doi: 10.1016/j.leukres.2010.12.016','7','Leukemia research',NULL,'889-98','21232794','2011 Jul','A comprehensive study of TP53 mutations in chronic lymphocytic leukemia: Analysis of 1287 diagnostic and 1148 follow-up CLL samples.',NULL,'35'),(10030,NULL,'Read ML et al','doi: 10.1002/mc.22254','1','Molecular carcinogenesis',NULL,'15-26','25408419','2016 Jan','The proto-oncogene PBF binds p53 and is associated with prognostic features in colorectal cancer.',NULL,'55'),(10031,NULL,'Ikeda J et al','','1','International journal of cancer',NULL,'35-43','11668476','2001 Oct 1','Restoration of endogenous wild-type p53 activity in a glioblastoma cell line with intrinsic temperature-sensitive p53 induces growth arrest but not apoptosis.',NULL,'94'),(10032,NULL,'Herbert BS et al','','6','Oncotarget',NULL,'405-22','21311097','2010 Oct','A molecular signature of normal breast epithelial and stromal cells from Li-Fraumeni syndrome mutation carriers.',NULL,'1'),(10033,NULL,'Schlereth K et al','doi: 10.1016/j.molcel.2010.02.037','3','Molecular cell',NULL,'356-68','20471942','2010 May 14','DNA binding cooperativity of p53 modulates the decision between cell-cycle arrest and apoptosis.',NULL,'38'),(10034,NULL,'Sykes SM et al','','6','Molecular cell',NULL,'841-51','17189187','2006 Dec 28','Acetylation of the p53 DNA-binding domain regulates apoptosis induction.',NULL,'24'),(10035,NULL,'Thukral SK et al','','9','Molecular and cellular biology',NULL,'5196-202','7651437','1995 Sep','Discrimination of DNA binding sites by mutant p53 proteins.',NULL,'15'),(10036,NULL,'Pietsch EC et al','doi: 10.1074/jbc.M710539200','30','The Journal of biological chemistry',NULL,'21294-304','18524770','2008 Jul 25','Oligomerization of BAK by p53 utilizes conserved residues of the p53 DNA binding domain.',NULL,'283'),(10037,NULL,'Dang CV et al','','30','The Journal of biological chemistry',NULL,'18019-23','2553699','1989 Oct 25','Nuclear and nucleolar targeting sequences of c-erb-A, c-myb, N-myc, p53, HSP70, and HIV tat proteins.',NULL,'264'),(10038,NULL,'May P et al','','53','Oncogene',NULL,'7621-36','10618702','1999 Dec 13','Twenty years of p53 research: structural and functional aspects of the p53 protein.',NULL,'18'),(10039,NULL,'Verdijk RM et al','doi: 10.1097/NEN.0b013e3181c55d55','1','Journal of neuropathology and experimental neurology',NULL,'16-26','20010306','2010 Jan','TP53 mutation analysis of malignant peripheral nerve sheath tumors.',NULL,'69'),(10040,NULL,'Ohgami RS et al','doi: 10.1038/modpathol.2014.160','5','Modern pathology : an official journal of the United States and Canadian Academy of Pathology, Inc',NULL,'706-14','25412851','2015 May','Next-generation sequencing of acute myeloid leukemia identifies the significance of TP53, U2AF1, ASXL1, and TET2 mutations.',NULL,'28'),(10041,NULL,'Rcker FG et al','doi: 10.1182/blood-2011-08-375758','9','Blood',NULL,'2114-21','22186996','2012 Mar 1','TP53 alterations in acute myeloid leukemia with complex karyotype correlate with specific copy number alterations, monosomal karyotype, and dismal outcome.',NULL,'119'),(10042,NULL,'Bowen D et al','doi: 10.1038/leu.2008.173','1','Leukemia',NULL,'203-6','18596741','2009 Jan','TP53 gene mutation is frequent in patients with acute myeloid leukemia and complex karyotype, and is associated with very poor prognosis.',NULL,'23'),(10043,NULL,'Cleven AH et al','doi: 10.1038/modpathol.2014.153','4','Modern pathology : an official journal of the United States and Canadian Academy of Pathology, Inc',NULL,'552-63','25412846','2015 Apr','High p53 protein expression in therapy-related myeloid neoplasms is associated with adverse karyotype and poor outcome.',NULL,'28'),(10044,NULL,'Ok CY et al','doi: 10.1186/s13045-015-0139-z','','Journal of hematology & oncology',NULL,'45','25952993','2015 May 8','TP53 mutation characteristics in therapy-related myelodysplastic syndromes and acute myeloid leukemia is similar to de novo diseases.',NULL,'8'),(10045,NULL,'Stilgenbauer S et al','doi: 10.1182/blood-2014-01-546150','21','Blood',NULL,'3247-54','24652989','2014 May 22','Gene mutations and treatment outcome in chronic lymphocytic leukemia: results from the CLL8 trial.',NULL,'123'),(10046,NULL,'Rossi D et al','doi: 10.1182/blood-2012-09-458265','8','Blood',NULL,'1403-12','23243274','2013 Feb 21','Integrated mutational and cytogenetic analysis identifies new prognostic subgroups in chronic lymphocytic leukemia.',NULL,'121'),(10047,NULL,'Chigrinova E et al','doi: 10.1182/blood-2013-03-489518','15','Blood',NULL,'2673-82','24004666','2013 Oct 10','Two main genetic pathways lead to the transformation of chronic lymphocytic leukemia to Richter syndrome.',NULL,'122'),(10048,NULL,'Zenz T et al','doi: 10.1200/JCO.2009.27.8762','29','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'4473-9','20697090','2010 Oct 10','TP53 mutation and survival in chronic lymphocytic leukemia.',NULL,'28'),(10049,NULL,'Rossi D et al','doi: 10.1158/1078-0432.CCR-08-1630','3','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'995-1004','19188171','2009 Feb 1','The prognostic value of TP53 mutations in chronic lymphocytic leukemia is independent of Del17p13: implications for overall survival and chemorefractoriness.',NULL,'15'),(10050,NULL,'Bejar R et al','doi: 10.1200/JCO.2013.52.3381','25','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'2691-8','25092778','2014 Sep 1','Somatic mutations predict poor outcome in patients with myelodysplastic syndrome after hematopoietic stem-cell transplantation.',NULL,'32'),(10051,NULL,'Milosevic JD et al','doi: 10.1002/ajh.23309','11','American journal of hematology',NULL,'1010-6','22887079','2012 Nov','Clinical significance of genetic aberrations in secondary acute myeloid leukemia.',NULL,'87'),(10052,NULL,'Lundberg P et al','doi: 10.1182/blood-2013-11-537167','14','Blood',NULL,'2220-8','24478400','2014 Apr 3','Clonal evolution and clinical correlates of somatic mutations in myeloproliferative neoplasms.',NULL,'123'),(10053,NULL,'Rampal R et al','doi: 10.1073/pnas.1407792111','50','Proceedings of the National Academy of Sciences of the United States of America',NULL,'E5401-10','25516983','2014 Dec 16','Genomic and functional analysis of leukemic transformation of myeloproliferative neoplasms.',NULL,'111'),(10054,NULL,'Raza S et al','doi: 10.1002/ajh.22216','2','American journal of hematology',NULL,'204-6','22052707','2012 Feb','TP53 mutations and polymorphisms in primary myelofibrosis.',NULL,'87'),(10055,NULL,'Nordstrm L et al','doi: 10.1111/bjh.12854','1','British journal of haematology',NULL,'98-108','24684350','2014 Jul','SOX11 and TP53 add prognostic information to MIPI in a homogenously treated cohort of mantle cell lymphoma--a Nordic Lymphoma Group study.',NULL,'166'),(10056,NULL,'Greiner TC et al','','10','Blood',NULL,'4302-10','8639789','1996 May 15','p53 mutations in mantle cell lymphoma are associated with variant cytology and predict a poor prognosis.',NULL,'87'),(10057,NULL,'Louie DC et al','','8','Blood',NULL,'2892-9','7579380','1995 Oct 15','p53 overexpression as a marker of poor prognosis in mantle cell lymphomas with t(11;14)(q13;q32).',NULL,'86'),(10058,NULL,'Delfau-Larue MH et al','doi: 10.1182/blood-2015-02-628792','5','Blood',NULL,'604-11','26022239','2015 Jul 30','High-dose cytarabine does not overcome the adverse prognostic value of CDKN2A and TP53 deletions in mantle cell lymphoma.',NULL,'126'),(10059,NULL,'Eskelund CW et al','doi: 10.1182/blood-2017-04-779736','17','Blood',NULL,'1903-1910','28819011','2017 Oct 26','<i>TP53</i> mutations identify younger mantle cell lymphoma patients who do not benefit from intensive chemoimmunotherapy.',NULL,'130'),(10060,NULL,'Slovackova J et al','doi: 10.4149/neo_2012_077','6','Neoplasma',NULL,'606-12','22862161','2012','Roscovitine-induced apoptosis of H1299 cells depends on functional status of p53.',NULL,'59'),(10061,NULL,'Lindenbergh-van der Plas M et al','doi: 10.1158/1078-0432.CCR-11-0183','11','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'3733-41','21467160','2011 Jun 1','Prognostic significance of truncating TP53 mutations in head and neck squamous cell carcinoma.',NULL,'17'),(10062,NULL,'Shirole NH et al','pii: e17929. doi: 10.7554/eLife.17929','','eLife',NULL,'','27759562','2016 Oct 19','<i>TP53</i> exon-6 truncating mutations produce separation of function isoforms with pro-tumorigenic functions.',NULL,'5'),(10063,NULL,'Wang Y et al','doi: 10.1016/j.ccr.2009.04.001','6','Cancer cell',NULL,'514-26','19477430','2009 Jun 2','Expression of mutant p53 proteins implicates a lineage relationship between neural stem cells and malignant astrocytic glioma in a murine model.',NULL,'15'),(10064,NULL,'Bougeard G et al','','6','Oncogene',NULL,'840-6','12584563','2003 Feb 13','Screening for TP53 rearrangements in families with the Li-Fraumeni syndrome reveals a complete deletion of the TP53 gene.',NULL,'22'),(10065,NULL,'Liu Y et al','doi: 10.1038/nature17157','7595','Nature',NULL,'471-475','26982726','2016 Mar 24','Deletions linked to TP53 loss drive cancer through p53-independent mechanisms.',NULL,'531'),(10066,NULL,'Drach J et al','','3','Blood',NULL,'802-9','9680348','1998 Aug 1','Presence of a p53 gene deletion in patients with multiple myeloma predicts for short survival after conventional-dose chemotherapy.',NULL,'92'),(10067,NULL,'Avet-Loiseau H et al','','8','Blood',NULL,'3489-95','17209057','2007 Apr 15','Genetic abnormalities and survival in multiple myeloma: the experience of the Intergroupe Francophone du Mylome.',NULL,'109'),(10068,NULL,'Xiong W et al','doi: 10.1182/blood-2007-10-119123','10','Blood',NULL,'4235-46','18337559','2008 Nov 15','An analysis of the clinical and biologic significance of TP53 loss and the identification of potential novel transcriptional targets of TP53 in multiple myeloma.',NULL,'112'),(10070,NULL,'Holstege H et al','doi: 10.1158/0008-5472.CAN-08-3426','8','Cancer research',NULL,'3625-33','19336573','2009 Apr 15','High incidence of protein-truncating TP53 mutations in BRCA1-related breast cancer.',NULL,'69'),(10071,NULL,'Barzon L et al','','2','European journal of endocrinology',NULL,'207-12','11454518','2001 Aug','Molecular analysis of CDKN1C and TP53 in sporadic adrenal tumors.',NULL,'145'),(10072,NULL,'Shiraishi K et al','','1','The Journal of biological chemistry',NULL,'348-55','14559903','2004 Jan 2','Isolation of temperature-sensitive p53 mutations from a comprehensive missense mutation library.',NULL,'279'),(10073,NULL,'Higashimoto Y et al','','6','Biochemistry',NULL,'1608-19','16460008','2006 Feb 14','Unfolding, aggregation, and amyloid formation by the tetramerization domain from mutant p53 associated with lung cancer.',NULL,'45'),(10074,NULL,'Tada M et al','','3','Molecular carcinogenesis',NULL,'171-6','9115587','1997 Mar','Reappraisal of p53 mutations in human malignant astrocytic neoplasms by p53 functional assay: comparison with conventional structural analyses.',NULL,'18'),(10075,NULL,'Malcikova J et al','doi: 10.1182/blood-2009-07-234708','26','Blood',NULL,'5307-14','19850740','2009 Dec 17','Monoallelic and biallelic inactivation of TP53 gene in chronic lymphocytic leukemia: selection, impact on survival, and response to DNA damage.',NULL,'114'),(10076,NULL,'Hattori H et al','doi: 10.1158/1535-7163.MCT-10-0781','4','Molecular cancer therapeutics',NULL,'670-8','21289082','2011 Apr','Context dependence of checkpoint kinase 1 as a therapeutic target for pancreatic cancers deficient in the BRCA2 tumor suppressor.',NULL,'10'),(10701,NULL,'Klein O et al','doi: 10.1016/j.ejca.2012.11.004','5','European journal of cancer (Oxford, England : 1990)',NULL,'1073-9','23237741','2013 Mar','BRAF inhibitor activity in V600R metastatic melanoma.',NULL,'49'),(10807,NULL,'Subbiah V et al','doi: 10.1016/S1470-2045(20)30321-1','9','The Lancet. Oncology',NULL,'1234-1243','32818466','2020 Sep','Dabrafenib plus trametinib in patients with BRAF<sup>V600E</sup>-mutated biliary tract cancer (ROAR): a phase 2, open-label, single-arm, multicentre basket trial.',NULL,'21'),(10808,NULL,'Carter J et al','doi: 10.1309/AJCP85ATMJOZOUDJ','4','American journal of clinical pathology',NULL,'620-8','26386083','2015 Oct','Non-p.V600E BRAF Mutations Are Common Using a More Sensitive and Broad Detection Tool.',NULL,'144'),(10809,NULL,'Noeparast A et al','doi: 10.18632/oncotarget.11635','36','Oncotarget',NULL,'60094-60108','28947956','2017 Sep 1','Non-V600 BRAF mutations recurrently found in lung cancer predict sensitivity to the combination of Trametinib and Dabrafenib.',NULL,'8'),(10810,NULL,'Acquaviva G et al','pii: E430. doi: 10.3390/cancers12020430','2','Cancers',NULL,'','32059434','2020 Feb 12','<i>BRAF</i> Exon 15 Mutations in Papillary Carcinoma and Adjacent Thyroid Parenchyma: A Search for the Early Molecular Events Associated with Tumor Development.',NULL,'12'),(10811,NULL,'Klempner SJ et al','doi: 10.1158/2159-8290.CD-15-1192','6','Cancer discovery',NULL,'594-600','27048246','2016 Jun','BRAFV600E Mutations in High-Grade Colorectal Neuroendocrine Tumors May Predict Responsiveness to BRAF-MEK Combination Therapy.',NULL,'6'),(10812,NULL,'Foster SA et al','doi: 10.1016/j.ccell.2016.02.010','4','Cancer cell',NULL,'477-493','26996308','2016 Apr 11','Activation Mechanism of Oncogenic Deletion Mutations in BRAF, EGFR, and HER2.',NULL,'29'),(10813,NULL,'Wrzeszczynski KO et al','pii: a004424. doi: 10.1101/mcs.a004424','6','Cold Spring Harbor molecular case studies',NULL,'','31519698','2019 Dec','Identification of targetable BRAF N486_P490 variant by whole-genome sequencing leading to dabrafenib-induced remission of a <i>BRAF</i>-mutant pancreatic adenocarcinoma.',NULL,'5'),(10814,NULL,'Yaeger R et al','doi: 10.1158/1078-0432.CCR-19-2004','23','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'7089-7097','31515458','2019 Dec 1','Response to Anti-EGFR Therapy in Patients with BRAF non-V600-Mutant Metastatic Colorectal Cancer.',NULL,'25'),(10874,NULL,'Boyault S et al','','1','Hepatology (Baltimore, Md.)',NULL,'42-52','17187432','2007 Jan','Transcriptome classification of HCC is related to gene alterations and to new therapeutic targets.',NULL,'45'),(10893,NULL,'Laurent-Puig P et al','','7','Gastroenterology',NULL,'1763-73','11375957','2001 Jun','Genetic alterations associated with hepatocellular carcinomas define distinct pathways of hepatocarcinogenesis.',NULL,'120'),(10956,NULL,'Wardell CP et al','doi: 10.1016/j.jhep.2018.01.009','5','Journal of hepatology',NULL,'959-969','29360550','2018 May','Genomic characterization of biliary tract cancers identifies driver genes and predisposing mutations.',NULL,'68'),(11059,NULL,'Persky NS et al','doi: 10.1038/s41594-019-0358-z','1','Nature structural & molecular biology',NULL,'92-104','31925410','2020 Jan','Defining the landscape of ATP-competitive inhibitor resistance residues in protein kinases.',NULL,'27'),(11190,NULL,'Davare MA et al','doi: 10.1073/pnas.1515281112','39','Proceedings of the National Academy of Sciences of the United States of America',NULL,'E5381-90','26372962','2015 Sep 29','Structural insight into selectivity and resistance profiles of ROS1 tyrosine kinase inhibitors.',NULL,'112'),(11204,NULL,'Rogiers A et al','doi: 10.1097/CMR.0000000000000376','5','Melanoma research',NULL,'507-510','28800030','2017 Oct','Response to targeted therapy in two patients with metastatic melanoma carrying rare BRAF exon 15 mutations: A598_T599insV and V600_K601delinsE.',NULL,'27'),(11227,NULL,'Kumamoto T et al','doi: 10.1007/s12185-018-2553-9','2','International journal of hematology',NULL,'228-232','30361829','2019 Feb','A case of recurrent histiocytic sarcoma with MAP2K1 pathogenic variant treated with the MEK inhibitor trametinib.',NULL,'109'),(11228,NULL,'Messinger YH et al','doi: 10.1002/pbc.28712','12','Pediatric blood & cancer',NULL,'e28712','32991018','2020 Dec','Langerhans cell histiocytosis with BRAF p.N486_P490del or MAP2K1 p.K57_G61del treated by the MEK inhibitor trametinib.',NULL,'67'),(11230,NULL,'Qin C et al','doi: 10.3389/fonc.2020.522816','','Frontiers in oncology',NULL,'522816','33117675','2020','Multidisciplinary Therapy Managed Recurrent Glioblastoma in a BRAF-V600E Mutant Pregnant Female: A Case Report and Review of the Literature.',NULL,'10'),(11231,NULL,'Robinson GW et al','doi: 10.1186/1471-2407-14-258','','BMC cancer',NULL,'258','24725538','2014 Apr 12','Complete clinical regression of a BRAF V600E-mutant pediatric glioblastoma multiforme after BRAF inhibitor therapy.',NULL,'14'),(11232,NULL,'Shi L et al','doi: 10.1080/15384047.2018.1529115','4','Cancer biology & therapy',NULL,'431-434','30462564','2019','Successful treatment of a BRAF V600E-mutant extracranial metastatic anaplastic oligoastrocytoma with vemurafenib and everolimus.',NULL,'20'),(11233,NULL,'Ceccon G et al','pii: E1090. doi: 10.3390/ijms19041090','4','International journal of molecular sciences',NULL,'','29621181','2018 Apr 5','Dabrafenib Treatment in a Patient with an Epithelioid Glioblastoma and BRAF V600E Mutation.',NULL,'19'),(11234,NULL,'Lin Z et al','doi: 10.1097/CAD.0000000000001130','','Anti-cancer drugs',NULL,'','34232949','2021 Jul 4','Effective treatment of a BRAF V600E-mutant epithelioid glioblastoma patient by vemurafenib: a case report.',NULL,''),(11235,NULL,'Burger MC et al','doi: 10.3892/or.2017.6013','6','Oncology reports',NULL,'3291-3296','29039591','2017 Dec','Dabrafenib in patients with recurrent, BRAF V600E mutated malignant glioma and leptomeningeal disease.',NULL,'38'),(11236,NULL,'Lassaletta A et al','doi: 10.1002/pbc.26086','11','Pediatric blood & cancer',NULL,'2038-41','27398937','2016 Nov','Profound clinical and radiological response to BRAF inhibition in a 2-month-old diencephalic child with hypothalamic/chiasmatic glioma.',NULL,'63'),(11237,NULL,'Aguilera D et al','doi: 10.1002/pbc.25787','3','Pediatric blood & cancer',NULL,'541-3','26579623','2016 Mar','Successful Retreatment of a Child with a Refractory Brainstem Ganglioglioma with Vemurafenib.',NULL,'63'),(11238,NULL,'Lee EQ et al','doi: 10.1200/JCO.2013.51.1766','10','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'e87-9','25092772','2016 Apr 1','Successful Treatment of a Progressive BRAF V600E-Mutated Anaplastic Pleomorphic Xanthoastrocytoma With Vemurafenib Monotherapy.',NULL,'34'),(11239,NULL,'Philippe L et al','doi: 10.1007/s11060-019-03280-2','1','Journal of neuro-oncology',NULL,'135-141','31502039','2019 Oct','Efficacy of Dabrafenib for three children with brainstem BRAF<sup>V600E</sup> positive ganglioglioma.',NULL,'145'),(11240,NULL,'del Bufalo F et al','doi: 10.1186/s12967-014-0356-1','','Journal of translational medicine',NULL,'356','25524464','2014 Dec 19','Response of recurrent BRAFV600E mutated ganglioglioma to Vemurafenib as single agent.',NULL,'12'),(11241,NULL,'Yau WH et al','doi: 10.1111/jcpt.13112','5','Journal of clinical pharmacy and therapeutics',NULL,'1172-1174','31985841','2020 Oct','Combination of BRAF and MEK inhibition in BRAF V600E mutant low-grade ganglioglioma.',NULL,'45'),(11242,NULL,'Skrypek M et al','doi: 10.1002/pbc.25084','11','Pediatric blood & cancer',NULL,'2099-100','24821190','2014 Nov','Pilomyxoid astrocytoma treated successfully with vemurafenib.',NULL,'61'),(11243,NULL,'Branco B et al','doi: 10.1111/ejh.13303','4','European journal of haematology',NULL,'444-448','31376203','2019 Oct','Targeted therapy of BRAF V600E-mutant histiocytic sarcoma: A case report and review of the literature.',NULL,'103'),(11244,NULL,'Mastropolo R et al','doi: 10.1182/bloodadvances.2019000093','12','Blood advances',NULL,'1848-1853','31213430','2019 Jun 25','<i>BRAF</i>-V600E-mutated Rosai-Dorfman-Destombes disease and Langerhans cell histiocytosis with response to BRAF inhibitor.',NULL,'3'),(11245,NULL,'Idbaih A et al','doi: 10.1212/WNL.0000000000000880','16','Neurology',NULL,'1478-80','25209580','2014 Oct 14','Dramatic response of a BRAF V600E-mutated primary CNS histiocytic sarcoma to vemurafenib.',NULL,'83'),(11246,NULL,'Chui MH et al','pii: PO.21.00055. doi: 10.1200/PO.21.00055','','JCO precision oncology',NULL,'','34568720','2021','Spectrum of <i>BRAF</i> Mutations and Gene Rearrangements in Ovarian Serous Carcinoma.',NULL,'5'),(11376,NULL,'Yen I et al','doi: 10.1038/s41586-021-03515-1','7863','Nature',NULL,'418-423','33953400','2021 Jun','ARAF mutations confer resistance to the RAF inhibitor belvarafenib in melanoma.',NULL,'594'),(11454,NULL,'Powers J et al','doi: 10.1158/0008-5472.CAN-20-1390','17','Cancer research',NULL,'3732-3744','32675277','2020 Sep 1','A Rare <i>TP53</i> Mutation Predominant in Ashkenazi Jews Confers Risk of Multiple Cancers.',NULL,'80'),(11461,NULL,'Lim SM et al','doi: 10.1200/JCO.2016.71.3701','23','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'2613-2618','28520527','2017 Aug 10','Open-Label, Multicenter, Phase II Study of Ceritinib in Patients With Non-Small-Cell Lung Cancer Harboring ROS1 Rearrangement.',NULL,'35'),(11480,NULL,'Salama AKS et al','doi: 10.1200/JCO.20.00762','33','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'3895-3904','32758030','2020 Nov 20','Dabrafenib and Trametinib in Patients With Tumors With <i>BRAF<sup>V600E</sup></i> Mutations: Results of the NCI-MATCH Trial Subprotocol H.',NULL,'38'),(11498,NULL,'Wen PY et al','doi: 10.1016/S1470-2045(21)00578-7','1','The Lancet. Oncology',NULL,'53-64','34838156','2022 Jan','Dabrafenib plus trametinib in patients with BRAF<sup>V600E</sup>-mutant low-grade and high-grade glioma (ROAR): a multicentre, open-label, single-arm, phase 2, basket trial.',NULL,'23'),(11500,NULL,'Sheikine Y et al','pii: PO.17.00172. doi: 10.1200/PO.17.00172','','JCO precision oncology',NULL,'','32913992','2018','<i>BRAF</i> in Lung Cancers: Analysis of Patient Cases Reveals Recurrent <i>BRAF</i> Mutations, Fusions, Kinase Duplications, and Concurrent Alterations.',NULL,'2'),(11512,NULL,'Soong R et al','','16','European journal of cancer (Oxford, England : 1990)',NULL,'2053-60','11044641','2000 Oct','Prognostic significance of TP53 gene mutation in 995 cases of colorectal carcinoma. Influence of tumour site, stage, adjuvant chemotherapy and type of mutation.',NULL,'36'),(11513,NULL,'de Anta JM et al','','24','Oncogene',NULL,'2951-8','9416838','1997 Dec 11','TP53 mutational pattern in Spanish and Polish non-small cell lung cancer patients: null mutations are associated with poor prognosis.',NULL,'15'),(11514,NULL,'Munch-Petersen HD et al','doi: 10.1186/s40478-016-0307-6','','Acta neuropathologica communications',NULL,'40','27101868','2016 Apr 22','TP53 hotspot mutations are predictive of survival in primary central nervous system lymphoma patients treated with combination chemotherapy.',NULL,'4'),(11548,NULL,'Lomax ME et al','','5','Oncogene',NULL,'643-9','9704930','1998 Aug 6','Characterization of p53 oligomerization domain mutations isolated from Li-Fraumeni and Li-Fraumeni like family members.',NULL,'17'),(11560,NULL,'Fu S et al','doi: 10.1093/annonc/mdv066','5','Annals of oncology : official journal of the European Society for Medical Oncology',NULL,'1012-1018','25669829','2015 May','Phase I study of pazopanib and vorinostat: a therapeutic approach for inhibiting mutant p53-mediated angiogenesis and facilitating mutant p53 degradation.',NULL,'26'),(11561,NULL,'Barta JA et al','doi: 10.1093/carcin/bgz087','1','Carcinogenesis',NULL,'67-77','31067569','2020 Mar 13','The lung-enriched p53 mutants V157F and R158L/P regulate a gain of function transcriptome in lung cancer.',NULL,'41'),(11623,NULL,'Sano D et al','doi: 10.1158/1078-0432.CCR-11-0046','21','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'6658-70','21903770','2011 Nov 1','Disruptive TP53 mutation is associated with aggressive disease characteristics in an orthotopic murine model of oral tongue cancer.',NULL,'17'),(11872,NULL,'Buzek J et al','','11','Nucleic acids research',NULL,'2340-8','12034820','2002 Jun 1','Redox state of tumor suppressor p53 regulates its sequence-specific DNA binding in DNA-damaged cells by cysteine 277.',NULL,'30'),(11873,NULL,'Barnas C et al','','1','International journal of cancer',NULL,'79-87','9096669','1997 Mar 28','Inactivation of the p53 protein in cell lines derived from human esophageal cancers.',NULL,'71'),(11874,NULL,'Dummer R et al','doi: 10.1200/JCO.21.02659','','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'JCO2102659','35862871','2022 Jul 21','COLUMBUS 5-Year Update: A Randomized, Open-Label, Phase III Trial of Encorafenib Plus Binimetinib Versus Vemurafenib or Encorafenib in Patients With BRAF V600-Mutant Melanoma.',NULL,''),(11910,NULL,'Hayward NK et al','doi: 10.1038/nature22071','7653','Nature',NULL,'175-180','28467829','2017 May 11','Whole-genome landscapes of major melanoma subtypes.',NULL,'545'),(11918,NULL,'Tang J et al','doi: 10.1038/s41388-021-01949-5','35','Oncogene',NULL,'5416-5426','34282274','2021 Sep','Cancer cells escape p53\'s tumor suppression through ablation of ZDHHC1-mediated p53 palmitoylation.',NULL,'40'),(11919,NULL,'Maslah N et al','doi: 10.1182/bloodadvances.2021005867','9','Blood advances',NULL,'2813-2823','35030630','2022 May 10','Single-cell analysis reveals selection of TP53-mutated clones after MDM2 inhibition.',NULL,'6'),(11920,NULL,'Pospsilov S et al','','5','Molecular cancer research : MCR',NULL,'296-304','15192123','2004 May','Constitutive and DNA damage inducible activation of pig3 and MDM2 genes by tumor-derived p53 mutant C277Y.',NULL,'2'),(11921,NULL,'Bouaoun L et al','doi: 10.1002/humu.23035','9','Human mutation',NULL,'865-76','27328919','2016 Sep','TP53 Variations in Human Cancers: New Lessons from the IARC TP53 Database and Genomics Data.',NULL,'37'),(11922,NULL,'Kato S et al','','14','Proceedings of the National Academy of Sciences of the United States of America',NULL,'8424-9','12826609','2003 Jul 8','Understanding the function-structure and function-mutation relationships of p53 tumor suppressor protein by high-resolution missense mutation analysis.',NULL,'100'),(11923,NULL,'Tang Z et al','doi: 10.3390/cancers12071903','7','Cancers',NULL,'','32674491','2020 Jul 14','Inconsistent Intersample ALK FISH Results in Patients with Lung Cancer: Analysis of Potential Causes.',NULL,'12'),(11924,NULL,'Kuo SJ et al','doi: 10.1016/j.cancergencyto.2010.05.013','2','Cancer genetics and cytogenetics',NULL,'94-101','20682393','2010 Sep','Number of somatic mutations in the mitochondrial D-loop region indicates poor prognosis in breast cancer, independent of TP53 mutation.',NULL,'201'),(11925,NULL,'Szasz AM et al','doi: 10.1007/s00428-014-1662-y','2','Virchows Archiv : an international journal of pathology',NULL,'237-41','25432630','2015 Feb','Basaloid carcinoma of the pancreas--clinicopathological presentation and oncogenetic snapshot of a rare entity.',NULL,'466'),(11926,NULL,'Ueno H et al','doi: 10.1182/bloodadvances.2019001307','20','Blood advances',NULL,'5165-5173','33095873','2020 Oct 27','Landscape of driver mutations and their clinical impacts in pediatric B-cell precursor acute lymphoblastic leukemia.',NULL,'4'),(11927,NULL,'Giacomelli AO et al','doi: 10.1038/s41588-018-0204-y','10','Nature genetics',NULL,'1381-1387','30224644','2018 Oct','Mutational processes shape the landscape of TP53 mutations in human cancer.',NULL,'50'),(11928,NULL,'James CD et al','','3','International journal of oncology',NULL,'547-53','10427138','1999 Sep','Tumor suppressor gene alterations in malignant gliomas: histopathological associations and prognostic evaluation.',NULL,'15'),(11929,NULL,'Heide I et al','','8','European journal of cancer (Oxford, England : 1990)',NULL,'1314-22','9301461','1997 Jul','The status of p53 in the metastatic progression of colorectal cancer.',NULL,'33'),(11930,NULL,'Kotler E et al','doi: 10.1016/j.molcel.2018.06.012','1','Molecular cell',NULL,'178-190.e8','29979965','2018 Jul 5','A Systematic p53 Mutation Library Links Differential Functional Impact to Cancer Mutation Pattern and Evolutionary Conservation.',NULL,'71'),(11931,NULL,'Blaszyk H et al','','10','Oncogene',NULL,'2159-66','8950983','1996 Nov 21','Molecular epidemiology of breast cancers in northern and southern Japan: the frequency, clustering, and patterns of p53 gene mutations differ among these two low-risk populations.',NULL,'13'),(11932,NULL,'Russo A et al','','11','Cancer epidemiology, biomarkers & prevention : a publication of the American Association for Cancer Research, cosponsored by the American Society of Preventive Oncology',NULL,'1322-31','12433709','2002 Nov','p53 mutations in L3-loop zinc-binding domain, DNA-ploidy, and S phase fraction are independent prognostic indicators in colorectal cancer: a prospective study with a five-year follow-up.',NULL,'11'),(11933,NULL,'Toguchida J et al','','22','Cancer research',NULL,'6194-9','1423262','1992 Nov 15','Mutation spectrum of the p53 gene in bone and soft tissue sarcomas.',NULL,'52'),(11934,NULL,'Kobayashi S et al','','6','American journal of surgery',NULL,'497-502','10414702','1999 Jun','The p53 gene mutation is of prognostic value in esophageal squamous cell carcinoma patients in unified stages of curability.',NULL,'177'),(11935,NULL,'Dang RK et al','','3','Molecular pathology : MP',NULL,'177-81','12032228','2002 Jun','Limitations of the use of single base changes in the p53 gene to detect minimal residual disease of breast cancer.',NULL,'55'),(11936,NULL,'Kerbauy FR et al','','10','Leukemia & lymphoma',NULL,'2071-8','15370252','2004 Oct','Detection and possible prognostic relevance of p53 gene mutations in diffuse large B-cell lymphoma. An analysis of 51 cases and review of the literature.',NULL,'45'),(11937,NULL,'Bharaj BS et al','','7','Clinical chemistry',NULL,'1397-403','9665415','1998 Jul','Rapid sequencing of the p53 gene with a new automated DNA sequencer.',NULL,'44'),(11938,NULL,'Wada I et al','','6 Suppl','Radiation research',NULL,'S125-7','10564952','1999 Dec','High rate of small TP53 mutations and infrequent loss of heterozygosity in malignant liver tumors associated with thorotrast: implications for alpha-particle carcinogenesis.',NULL,'152'),(11939,NULL,'Smith JS et al','','16','Journal of the National Cancer Institute',NULL,'1246-56','11504770','2001 Aug 15','PTEN mutation, EGFR amplification, and outcome in patients with anaplastic astrocytoma and glioblastoma multiforme.',NULL,'93'),(11940,NULL,'Yamazaki Y et al','','2','Oral oncology',NULL,'163-9','12509970','2003 Feb','Specific p53 mutations predict poor prognosis in oral squamous cell carcinoma.',NULL,'39'),(11941,NULL,'Chan WY et al','','3','Diagnostic molecular pathology : the American journal of surgical pathology, part B',NULL,'153-60','11552717','2001 Sep','Epstein-Barr virus-associated gastric lymphomas are distinct from mucosa-associated lymphoid tissue-type lymphomas: genetic abnormalities of p53 gene.',NULL,'10'),(11942,NULL,'Gunduz M et al','','3','Cancer science',NULL,'531-8','18081876','2008 Mar','Downregulation of ING3 mRNA expression predicts poor prognosis in head and neck cancer.',NULL,'99'),(11943,NULL,'Davies AJ et al','','2','British journal of haematology',NULL,'286-93','17278262','2007 Jan','Transformation of follicular lymphoma to diffuse large B-cell lymphoma proceeds by distinct oncogenic mechanisms.',NULL,'136'),(11944,NULL,'Boersma BJ et al','','13','Journal of the National Cancer Institute',NULL,'911-9','16818855','2006 Jul 5','Association of breast cancer outcome with status of p53 and MDM2 SNP309.',NULL,'98'),(11945,NULL,'Cuny M et al','','4','Cancer research',NULL,'1077-83','10706127','2000 Feb 15','Relating genotype and phenotype in breast cancer: an analysis of the prognostic significance of amplification at eight different genes or loci and of p53 mutations.',NULL,'60'),(11946,NULL,'Ahmed AA et al','doi: 10.1002/path.2696','1','The Journal of pathology',NULL,'49-56','20229506','2010 May','Driver mutations in TP53 are ubiquitous in high grade serous carcinoma of the ovary.',NULL,'221'),(11947,NULL,'Barnabas N et al','','2','Cancer',NULL,'285-93','11180073','2001 Jan 15','Significance of p53 mutations in patients with chronic lymphocytic leukemia: a sequential study of 30 patients.',NULL,'91'),(11948,NULL,'Snchez-Pernaute A et al','','5','Oncology reports',NULL,'1129-33','9683822','1998 Sep-Oct','Prognostic significance of p53 gene mutations in squamous cell carcinoma of the lung.',NULL,'5'),(11949,NULL,'Cottu PH et al','','12','Oncogene',NULL,'2727-30','9000147','1996 Dec 19','Inverse correlation between RER+ status and p53 mutation in colorectal cancer cell lines.',NULL,'13'),(11950,NULL,'Sheu JC et al','','21','Cancer research',NULL,'6098-100','1327523','1992 Nov 1','Mutation of p53 gene in hepatocellular carcinoma in Taiwan.',NULL,'52'),(11951,NULL,'Volpe G et al','','1','Genes, chromosomes & cancer',NULL,'21-30','9162193','1996 May','Molecular heterogeneity of B-lineage diffuse large cell lymphoma.',NULL,'16'),(11952,NULL,'Munirajan AK et al','','3','International journal of cancer',NULL,'297-300','8621246','1996 May 3','p53 gene mutations in oral carcinomas from India.',NULL,'66'),(11953,NULL,'Andersen TI et al','','3','British journal of cancer',NULL,'540-8','8102535','1993 Sep','Prognostic significance of TP53 alterations in breast carcinoma.',NULL,'68'),(11954,NULL,'Oda T et al','','22','Cancer research',NULL,'6358-64','1330291','1992 Nov 15','p53 gene mutation spectrum in hepatocellular carcinoma.',NULL,'52'),(11955,NULL,'Peng H et al','','2','The American journal of pathology',NULL,'643-8','8579126','1996 Feb','Replication error phenotype and p53 gene mutation in lymphomas of mucosa-associated lymphoid tissue.',NULL,'148'),(11956,NULL,'Slingerland JM et al','','7','Blood',NULL,'1500-7','2009369','1991 Apr 1','Mutation of the p53 gene in human acute myelogenous leukemia.',NULL,'77'),(11957,NULL,'Erill N et al','','4','Diagnostic molecular pathology : the American journal of surgical pathology, part B',NULL,'217-23','15538112','2004 Dec','Genetic and immunophenotype analyses of TP53 in bladder cancer: TP53 alterations are associated with tumor progression.',NULL,'13'),(11958,NULL,'Safran H et al','','6','Cancer',NULL,'1203-10','8826941','1996 Sep 15','p53 mutations do not predict response to paclitaxel/radiation for nonsmall cell lung carcinoma.',NULL,'78'),(11959,NULL,'Gelfi C et al','','15','Electrophoresis',NULL,'2921-7','9504831','1997 Dec','Detection of p53 point mutations by double-gradient, denaturing gradient gel electrophoresis.',NULL,'18'),(11960,NULL,'Baldini L et al','','1','Blood',NULL,'270-8','8018922','1994 Jul 1','Frequent p53 gene involvement in splenic B-cell leukemia/lymphomas of possible marginal zone origin.',NULL,'84'),(11961,NULL,'Bukholm IR et al','','2','Breast cancer research and treatment',NULL,'199-206','12908823','2003 Jul','Overexpression of cyclin A overrides the effect of p53 alterations in breast cancer patients with long follow-up time.',NULL,'80'),(11962,NULL,'Huang CC et al','','7','Environmental and molecular mutagenesis',NULL,'527-32','16721749','2006 Aug','Different p53 mutation patterns in colorectal tumors from smokers and nonsmokers.',NULL,'47'),(11963,NULL,'Kang YK et al','','1','Virchows Archiv : an international journal of pathology',NULL,'27-32','9463584','1998 Jan','p53 mutation and overexpression in hepatocellular carcinoma and dysplastic nodules in the liver.',NULL,'432'),(11964,NULL,'Ohgaki H et al','','19','Cancer research',NULL,'6892-9','15466178','2004 Oct 1','Genetic pathways to glioblastoma: a population-based study.',NULL,'64'),(11965,NULL,'Fallows S et al','','1','The Journal of pathology',NULL,'68-75','11329143','2001 May','P53 mutation does not affect prognosis in ovarian epithelial malignancies.',NULL,'194'),(11966,NULL,'Pavlova S et al','','1','International journal of oncology',NULL,'121-31','12792784','2003 Jul','High frequency of temperature-sensitive mutations of p53 tumor suppressor in acute myeloid leukemia revealed by functional assay in yeast.',NULL,'23'),(11967,NULL,'Ichikawa A et al','','8','The New England journal of medicine',NULL,'529-34','9262496','1997 Aug 21','Mutations of the p53 gene as a prognostic factor in aggressive B-cell lymphoma.',NULL,'337'),(11968,NULL,'Shi XB et al','','7','BJU international',NULL,'996-1002','15541116','2004 Nov','A modified yeast assay used on archival samples of localized prostate cancer tissue improves the detection of p53 abnormalities and increases their predictive value.',NULL,'94'),(11969,NULL,'Okada T et al','','3','FEBS letters',NULL,'583-90','14675778','2003 Dec 18','Gene expression profile linked to p53 status in hepatitis C virus-related hepatocellular carcinoma.',NULL,'555'),(11970,NULL,'Cao W et al','','4','Cancer',NULL,'834-44','15305417','2004 Aug 15','Mutational spectra of p53 in geographically localized esophageal squamous cell carcinoma groups in China.',NULL,'101'),(11971,NULL,'Hernndez S et al','','15','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'5444-50','16061860','2005 Aug 1','FGFR3 and Tp53 mutations in T1G3 transitional bladder carcinomas: independent distribution and lack of association with prognosis.',NULL,'11'),(11972,NULL,'Ito T et al','','6','Oncology reports',NULL,'1937-42','14534722','2003 Nov-Dec','Clinical significance in molecular detection of p53 mutation in serum of patients with colorectal carcinoma.',NULL,'10'),(11973,NULL,'Esteller M et al','','12','Cancer research',NULL,'4689-92','11406538','2001 Jun 15','Promoter hypermethylation of the DNA repair gene O(6)-methylguanine-DNA methyltransferase is associated with the presence of G:C to A:T transition mutations in p53 in human colorectal tumorigenesis.',NULL,'61'),(11974,NULL,'Wang Y et al','','3','British journal of cancer',NULL,'678-85','14760384','2004 Feb 9','TP53 mutations in early-stage ovarian carcinoma, relation to long-term survival.',NULL,'90'),(11975,NULL,'Ott K et al','','6','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'2307-15','12796400','2003 Jun','Chromosomal instability rather than p53 mutation is associated with response to neoadjuvant cisplatin-based chemotherapy in gastric carcinoma.',NULL,'9'),(11989,NULL,'Torregrossa L et al','','11','The Journal of clinical endocrinology and metabolism',NULL,'4413-4420','27571181','2016 Nov','Papillary Thyroid Carcinoma With Rare Exon 15 BRAF Mutation Has Indolent Behavior: A Single-Institution Experience.',NULL,'101'),(12018,NULL,'Elamin YY et al','doi: 10.1016/j.ccell.2022.06.006','7','Cancer cell',NULL,'754-767.e6','35820397','2022 Jul 11','Poziotinib for EGFR exon 20-mutant NSCLC: Clinical efficacy, resistance mechanisms, and impact of insertion location on drug sensitivity.',NULL,'40'),(12029,NULL,'Lacy SE et al','doi: 10.1182/blood.2019003535','20','Blood',NULL,'1759-1771','32187361','2020 May 14','Targeted sequencing in DLBCL, molecular subtypes, and outcomes: a Haematological Malignancy Research Network report.',NULL,'135'),(12042,NULL,'Tessoulin B et al','doi: 10.1186/s13045-018-0679-0','1','Journal of hematology & oncology',NULL,'137','30545397','2018 Dec 13','Whole-exon sequencing of human myeloma cell lines shows mutations related to myeloma patients at relapse with major hits in the DNA regulation and repair pathways.',NULL,'11'),(12096,NULL,'Sawada G et al','doi: 10.1053/j.gastro.2016.01.035','5','Gastroenterology',NULL,'1171-1182','26873401','2016 May','Genomic Landscape of Esophageal Squamous Cell Carcinoma ina Japanese Population.',NULL,'150'),(12110,NULL,'Berns EM et al','','7','British journal of cancer',NULL,'1130-6','9569050','1998 Apr','Mutations in residues of TP53 that directly contact DNA predict poor outcome in human primary breast cancer.',NULL,'77'),(12111,NULL,'Matozaki T et al','','1','Biochemical and biophysical research communications',NULL,'215-23','1370612','1992 Jan 15','Missense mutations and a deletion of the p53 gene in human gastric cancer.',NULL,'182'),(12112,NULL,'Birindelli S et al','','6','Laboratory investigation; a journal of technical methods and pathology',NULL,'833-44','11406645','2001 Jun','Rb and TP53 pathway alterations in sporadic and NF1-related malignant peripheral nerve sheath tumors.',NULL,'81'),(12113,NULL,'Brresen-Dale AL et al','','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'203-10','9516972','1998 Jan','TP53 and long-term prognosis in colorectal cancer: mutations in the L3 zinc-binding domain predict poor survival.',NULL,'4'),(12114,NULL,'Hsieh LL et al','','1-2','Cancer letters',NULL,'107-13','8620428','1996 Feb 27','p53 mutations in gastric cancers from Taiwan.',NULL,'100'),(12115,NULL,'Brachman DG et al','','17','Cancer research',NULL,'4832-6','1324797','1992 Sep 1','Occurrence of p53 gene deletions and human papilloma virus infection in human head and neck cancer.',NULL,'52'),(12116,NULL,'Tai YC et al','','11','Pathology international',NULL,'811-8','15533223','2004 Nov','Higher frequency of p53 gene mutations in diffuse large B-cell lymphoma with MALT component.',NULL,'54'),(12117,NULL,'Olivier M et al','','4','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'1157-67','16489069','2006 Feb 15','The clinical value of somatic TP53 gene mutations in 1,794 patients with breast cancer.',NULL,'12'),(12118,NULL,'Le Calvez F et al','','12','Cancer research',NULL,'5076-83','15958551','2005 Jun 15','TP53 and KRAS mutation load and types in lung cancers in relation to tobacco smoke: distinct patterns in never, former, and current smokers.',NULL,'65'),(12119,NULL,'Chrisanthar R et al','doi: 10.1371/journal.pone.0003062','8','PloS one',NULL,'e3062','18725978','2008 Aug 26','CHEK2 mutations affecting kinase activity together with mutations in TP53 indicate a functional pathway associated with resistance to epirubicin in primary breast cancer.',NULL,'3'),(12120,NULL,'Oda S et al','','5','Nucleic acids research',NULL,'1628-36','15778432','2005','Two modes of microsatellite instability in human cancer: differential connection of defective DNA mismatch repair to dinucleotide repeat instability.',NULL,'33'),(12121,NULL,'Akkiprik M et al','','1','Journal of gastrointestinal and liver diseases : JGLD',NULL,'11-7','17410283','2007 Mar','Clinical significance of p53, K-ras and DCC gene alterations in the stage I-II colorectal cancers.',NULL,'16'),(12122,NULL,'Moriya M et al','doi: 10.1002/ijc.26077','6','International journal of cancer',NULL,'1532-6','21413016','2011 Sep 15','TP53 Mutational signature for aristolochic acid: an environmental carcinogen.',NULL,'129'),(12123,NULL,'Yamazaki Y et al','','8','Head & neck',NULL,'676-81','15957194','2005 Aug','Clinical value of genetically diagnosed lymph node micrometastasis for patients with oral squamous cell carcinoma.',NULL,'27'),(12124,NULL,'Fujita M et al','','12','Japanese journal of cancer research : Gann',NULL,'1247-56','7852189','1994 Dec','Alteration of the p53 tumor suppressor gene occurs independently of K-ras activation and more frequently in serous adenocarcinomas than in other common epithelial tumors of the human ovary.',NULL,'85'),(12125,NULL,'Van Emburgh BO et al','','2','Molecular carcinogenesis',NULL,'88-99','17683074','2008 Feb','Polymorphisms in drug metabolism genes, smoking, and p53 mutations in breast cancer.',NULL,'47'),(12126,NULL,'el-Mahdani N et al','','4','British journal of cancer',NULL,'528-36','9052405','1997','Overexpression of p53 mRNA in colorectal cancer and its relationship to p53 gene mutation.',NULL,'75'),(12127,NULL,'Guinee DG Jr et al','','5','Carcinogenesis',NULL,'993-1002','7767998','1995 May','Gender comparisons in human lung cancer: analysis of p53 mutations, anti-p53 serum antibodies and C-erbB-2 expression.',NULL,'16'),(12128,NULL,'Zhou S et al','','18','Proceedings of the National Academy of Sciences of the United States of America',NULL,'7540-5','17456604','2007 May 1','Frequency and phenotypic implications of mitochondrial DNA mutations in human squamous cell cancers of the head and neck.',NULL,'104'),(12129,NULL,'Yan L et al','','7','Gastroenterology',NULL,'2124-30','15940643','2005 Jun','Molecular analysis to detect pancreatic ductal adenocarcinoma in high-risk groups.',NULL,'128'),(12130,NULL,'Chevillard S et al','','12 Pt 1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'2471-8','9815649','1997 Dec','Biological and clinical significance of concurrent p53 gene alterations, MDR1 gene expression, and S-phase fraction analyses in breast cancer patients treated with primary chemotherapy or radiotherapy.',NULL,'3'),(12131,NULL,'Wang Y et al','','1','Human mutation',NULL,'21-34','15221786','2004 Jul','Effect of the codon 72 polymorphism (c.215G>C, p.Arg72Pro) in combination with somatic sequence variants in the TP53 gene on survival in patients with advanced ovarian carcinoma.',NULL,'24'),(12132,NULL,'Benhattar J et al','','3','International journal of cancer',NULL,'190-2','8682586','1996 Jun 21','p53 mutations as a possible predictor of response to chemotherapy in metastatic colorectal carcinomas.',NULL,'69'),(12133,NULL,'Jnsson G et al','','32','Oncogene',NULL,'4738-48','17260012','2007 Jul 12','Genomic profiling of malignant melanoma using tiling-resolution arrayCGH.',NULL,'26'),(12134,NULL,'Hayashi Y et al','','1','Journal of neurosurgery',NULL,'138-42','11453385','2001 Jul','Malignant transformation of a gangliocytoma/ganglioglioma into a glioblastoma multiforme: a molecular genetic analysis. Case report.',NULL,'95'),(12135,NULL,'Umekita Y et al','','8','Japanese journal of cancer research : Gann',NULL,'825-30','7928628','1994 Aug','Nuclear accumulation of p53 protein correlates with mutations in the p53 gene on archival paraffin-embedded tissues of human breast cancer.',NULL,'85'),(12136,NULL,'Fernndez-Cuesta L et al','','3','Breast cancer research : BCR',NULL,'R70','22551440','2012 May 2','Prognostic and predictive value of TP53 mutations in node-positive breast cancer patients treated with anthracycline- or anthracycline/taxane-based adjuvant therapy: results from the BIG 02-98 phase III trial.',NULL,'14'),(12137,NULL,'Arango D et al','','3','Gastroenterology',NULL,'874-84','16143127','2005 Sep','Gene-expression profiling predicts recurrence in Dukes\' C colorectal cancer.',NULL,'129'),(12138,NULL,'Paquette RL et al','','4','Cancer',NULL,'1272-80','8393371','1993 Aug 15','Mutations of p53 and human papillomavirus infection in cervical carcinoma.',NULL,'72'),(12139,NULL,'Mechanic LE et al','','3','Carcinogenesis',NULL,'597-604','15564288','2005 Mar','Polymorphisms in XPD and TP53 and mutation in human lung cancer.',NULL,'26'),(12140,NULL,'Harms PW et al','doi: 10.1158/0008-5472.CAN-15-0702','18','Cancer research',NULL,'3720-3727','26238782','2015 Sep 15','The Distinctive Mutational Spectra of Polyomavirus-Negative Merkel Cell Carcinoma.',NULL,'75'),(12141,NULL,'Oscier DG et al','','4','Blood',NULL,'1177-84','12149195','2002 Aug 15','Multivariate analysis of prognostic factors in CLL: clinical stage, IGVH gene mutational status, and loss or mutation of the p53 gene are independent prognostic factors.',NULL,'100'),(12142,NULL,'Thangnipon W et al','','1-2','Cancer letters',NULL,'195-201','10454262','1999 Jul 1','Distinct pattern of PCR-SSCP analysis of p53 mutations in human astrocytomas.',NULL,'141'),(12143,NULL,'Hsieh LL et al','','9','Carcinogenesis',NULL,'1497-503','11532872','2001 Sep','Characteristics of mutations in the p53 gene in oral squamous cell carcinoma associated with betel quid chewing and cigarette smoking in Taiwanese.',NULL,'22'),(12144,NULL,'Akahane T et al','','3','International journal of gynecological pathology : official journal of the International Society of Gynecological Pathologists',NULL,'345-51','17581423','2007 Jul','The role of p53 mutation in the carcinomas arising from endometriosis.',NULL,'26'),(12145,NULL,'Lam KY et al','','2','International journal of cancer',NULL,'212-9','9133458','1997 Apr 22','Prevalence and predictive value of p53 mutation in patients with oesophageal squamous cell carcinomas: a prospective clinico-pathological study and survival analysis of 70 patients.',NULL,'74'),(12146,NULL,'Ko JL et al','','9','DNA repair',NULL,'755-62','12509279','2002 Sep 4','A novel p53 mutant retained functional activity in lung carcinomas.',NULL,'1'),(12147,NULL,'Soong R et al','','6','International journal of cancer',NULL,'642-7','9421363','1997 Dec 19','Detection of p53 gene mutation by rapid PCR-SSCP and its association with poor survival in breast cancer.',NULL,'74'),(12148,NULL,'Thongsuksai P et al','','1','Cancer letters',NULL,'1-7','14580680','2003 Nov 10','p53 mutations in betel-associated oral cancer from Thailand.',NULL,'201'),(12149,NULL,'Takagi Y et al','','22','Cancer research',NULL,'5354-7','7585600','1995 Nov 15','Distinct mutational spectrum of the p53 gene in lung cancers from Chinese women in Hong Kong.',NULL,'55'),(12150,NULL,'Balz V et al','','6','Cancer research',NULL,'1188-91','12649174','2003 Mar 15','Is the p53 inactivation frequency in squamous cell carcinomas of the head and neck underestimated? Analysis of p53 exons 2-11 and human papillomavirus 16/18 E6 transcripts in 123 unselected tumor specimens.',NULL,'63'),(12151,NULL,'Xu Y et al','doi: 10.1002/ijc.23454','12','International journal of cancer',NULL,'2761-6','18348141','2008 Jun 15','Effect of p53 codon 72 genotype on breast cancer survival depends on p53 gene status.',NULL,'122'),(12152,NULL,'Sundaresan V et al','','10','Oncogene',NULL,'1989-97','1408139','1992 Oct','p53 and chromosome 3 abnormalities, characteristic of malignant lung tumours, are detectable in preinvasive lesions of the bronchus.',NULL,'7'),(12153,NULL,'Konishi M et al','','8','Japanese journal of cancer research : Gann',NULL,'893-9','8407553','1993 Aug','Genetic changes and histopathological grades in human hepatocellular carcinomas.',NULL,'84'),(12154,NULL,'Kovalev S et al','','6','Human pathology',NULL,'613-9','9635683','1998 Jun','Loss of p53 function in uterine papillary serous carcinoma.',NULL,'29'),(12174,NULL,'Andersson M et al','','7','Cancer epidemiology, biomarkers & prevention : a publication of the American Association for Cancer Research, cosponsored by the American Society of Preventive Oncology',NULL,'765-70','8672994','1995 Oct-Nov','Mutations in the tumor suppressor gene p53 in human liver cancer induced by alpha-particles.',NULL,'4'),(12175,NULL,'Cogliatti SB et al','','4','The Journal of pathology',NULL,'470-8','11113864','2000 Dec','Significantly different bcl-2 expression profiles in gastric and non-gastric primary extranodal high-grade B-cell lymphomas.',NULL,'192'),(12176,NULL,'Ibrahim SO et al','','4','International journal of cancer',NULL,'527-34','10225439','1999 May 17','Mutations of the p53 gene in oral squamous-cell carcinomas from Sudanese dippers of nitrosamine-rich toombak and non-snuff-dippers from the Sudan and Scandinavia.',NULL,'81'),(12177,NULL,'Onda M et al','','1','Fukushima journal of medical science',NULL,'29-39','9431782','1997 Jun','K-ras and p53 gene mutations in colorectal cancer in 57 Japanese patients.',NULL,'43'),(12178,NULL,'Gokgoz N et al','','8','Cancer',NULL,'2181-9','11596036','2001 Oct 15','Comparison of p53 mutations in patients with localized osteosarcoma and metastatic osteosarcoma.',NULL,'92'),(12179,NULL,'Khan ZA et al','','11','Journal of cancer research and clinical oncology',NULL,'675-80','11710597','2001 Nov','P53 mutation and response to hepatic arterial floxuridine in patients with colorectal liver metastases.',NULL,'127'),(12180,NULL,'Conde E et al','','3 Pt 1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'710-7','16467080','2006 Feb 1','Molecular context of the EGFR mutations: evidence for the activation of mTOR/S6K signaling.',NULL,'12'),(12181,NULL,'Bertorelle R et al','','2','Clinical molecular pathology',NULL,'M85-90','16696056','1996 Apr','p53 gene alterations and protein accumulation in colorectal cancer.',NULL,'49'),(12182,NULL,'O\'Malley FP et al','','9','Human pathology',NULL,'955-63','8816892','1996 Sep','The predictive power of semiquantitative immunohistochemical assessment of p53 and c-erb B-2 in lymph node-negative breast cancer.',NULL,'27'),(12183,NULL,'Deissler H et al','','6','Oncology reports',NULL,'1281-6','15138567','2004 Jun','Spectrum of p53 mutations in biopsies from breast cancer patients selected for preoperative chemotherapy analysed by the functional yeast assay to predict therapeutic response.',NULL,'11'),(12184,NULL,'Tang JL et al','','5B','Anticancer research',NULL,'3757-61','9854490','1998 Sep-Oct','P53 mutation in advanced stage of primary myelodysplastic syndrome.',NULL,'18'),(12185,NULL,'Wilson WH et al','','2','Blood',NULL,'601-9','9002964','1997 Jan 15','Relationship of p53, bcl-2, and tumor proliferation to clinical drug resistance in non-Hodgkin\'s lymphomas.',NULL,'89'),(12186,NULL,'Parmar J et al','','3','Oncology research',NULL,'149-55','11216673','2000','Radiation-induced cell cycle delays and p53 status of early passage melanoma cell lines.',NULL,'12'),(12187,NULL,'Takayama H et al','','9','Japanese journal of cancer research : Gann',NULL,'941-7','11011123','2000 Sep','p53 mutations in prostatic intraepithelial neoplasia and concurrent carcinoma: analysis of laser capture microdissected specimens from non-transition and transition zones.',NULL,'91'),(12188,NULL,'Ralhan R et al','','6','International journal of cancer',NULL,'791-5','10709097','2000 Mar 15','Circulating p53 antibodies, p53 gene mutational profile and product accumulation in esophageal squamous-cell carcinoma in India.',NULL,'85'),(12189,NULL,'Huang FY et al','doi: 10.1186/s12885-019-6002-9','1','BMC cancer',NULL,'789','31395065','2019 Aug 8','Targeted genomic profiling identifies frequent deleterious mutations in FAT4 and TP53 genes in HBV-associated hepatocellular carcinoma.',NULL,'19'),(12190,NULL,'Dillon DA et al','','5','Acta cytologica',NULL,'841-7','12365217','2002 Sep-Oct','p53 mutations as tumor markers in fine needle aspirates of palpable breast masses.',NULL,'46'),(12191,NULL,'Lee LN et al','','6 Pt 1','American journal of respiratory and critical care medicine',NULL,'1667-71','7952630','1994 Dec','Exon 8 mutation of p53 gene associated with nodal metastasis in non-small-cell lung cancer.',NULL,'150'),(12192,NULL,'Hartmann A et al','','8','British journal of cancer',NULL,'896-901','8611423','1996 Apr','High frequency of p53 gene mutations in primary breast cancers in Japanese women, a low-incidence population.',NULL,'73'),(12193,NULL,'Goan YG et al','','11','Cancer science',NULL,'758-65','16271069','2005 Nov','Risk of p53 gene mutation in esophageal squamous cell carcinoma and habit of betel quid chewing in Taiwanese.',NULL,'96'),(12194,NULL,'Gumerlock PH et al','','1','Journal of the National Cancer Institute',NULL,'66-71','8978408','1997 Jan 1','p53 abnormalities in primary prostate cancer: single-strand conformation polymorphism analysis of complementary DNA in comparison with genomic DNA. The Cooperative Prostate Network.',NULL,'89'),(12195,NULL,'Shibagaki I et al','','7','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'769-73','9816044','1995 Jul','p53 mutation, murine double minute 2 amplification, and human papillomavirus infection are frequently involved but not associated with each other in esophageal squamous cell carcinoma.',NULL,'1'),(12196,NULL,'Kihara C et al','','2','Japanese journal of cancer research : Gann',NULL,'190-8','10761706','2000 Feb','Mutations in zinc-binding domains of p53 as a prognostic marker of esophageal-cancer patients.',NULL,'91'),(12197,NULL,'Chung KY et al','','7','Cancer research',NULL,'1676-83','8453641','1993 Apr 1','Discordant p53 gene mutations in primary head and neck cancers and corresponding second primary cancers of the upper aerodigestive tract.',NULL,'53'),(12198,NULL,'Shaw P et al','','11','Oncogene',NULL,'2121-8','1945416','1991 Nov','Occurrence of Ki-ras and p53 mutations in primary colorectal tumors.',NULL,'6'),(12199,NULL,'Gad S et al','','2','British journal of cancer',NULL,'336-40','17133269','2007 Jan 29','Mutations in BHD and TP53 genes, but not in HNF1beta gene, in a large series of sporadic chromophobe renal cell carcinoma.',NULL,'96'),(12200,NULL,'Kishimoto Y et al','','17','Cancer research',NULL,'4799-804','1324794','1992 Sep 1','Aberrations of the p53 tumor suppressor gene in human non-small cell carcinomas of the lung.',NULL,'52'),(12201,NULL,'Ouerhani S et al','doi: 10.3109/07357900902849707','10','Cancer investigation',NULL,'998-1007','19909015','2009 Dec','Combined analysis of smoking, TP53, and FGFR3 mutations in Tunisian patients with invasive and superficial high-grade bladder tumors.',NULL,'27'),(12202,NULL,'Rosenstein BS et al','','5','Photochemistry and photobiology',NULL,'798-806','10568172','1999 Nov','p53 mutations in basal cell carcinomas arising in routine users of sunscreens.',NULL,'70'),(12203,NULL,'Trbusek M et al','','6','Leukemia',NULL,'1159-61','16572201','2006 Jun','Inactivation of p53 and deletion of ATM in B-CLL patients in relation to IgVH mutation status and previous treatment.',NULL,'20'),(12204,NULL,'Shivapurkar N et al','','1','Cancer letters',NULL,'39-46','9097977','1997 May 1','K-ras and p53 mutations in aberrant crypt foci and colonic tumors from colon cancer patients.',NULL,'115'),(12205,NULL,'Holmila R et al','doi: 10.1016/j.mrfmmm.2009.12.003','1-2','Mutation research',NULL,'9-14','20025891','2010 Apr 1','Profile of TP53 gene mutations in sinonasal cancer.',NULL,'686'),(12206,NULL,'Hsu H et al','','6','International journal of oncology',NULL,'1341-7','21567059','1994 Jun','Mutations of p53 gene in hepatocellular-carcinoma (hcc) correlate with tumor progression and patient prognosis - a study of 138 patients with unifocal hcc.',NULL,'4'),(12207,NULL,'Huang C et al','','19','Oncogene',NULL,'2469-77','9627113','1998 May 14','Mutations in exon 7 and 8 of p53 as poor prognostic factors in patients with non-small cell lung cancer.',NULL,'16'),(12208,NULL,'Hara Y et al','','1','Cancer letters',NULL,'109-16','12637159','2003 Mar 20','ING1 and p53 tumor suppressor gene alterations in adenocarcinomas of the esophagogastric junction.',NULL,'192'),(12209,NULL,'Forslund A et al','','3','International journal of oncology',NULL,'501-6','11494027','2001 Sep','Serum anti-p53 in relation to mutations across the entire translated p53 gene in colorectal carcinomas.',NULL,'19'),(12210,NULL,'Hayashi M et al','','5','Pathology international',NULL,'270-6','12713560','2003 May','Microsatellite instability in esophageal squamous cell carcinoma is not associated with hMLH1 promoter hypermethylation.',NULL,'53'),(12211,NULL,'Schmidt MC et al','','4','Journal of neuropathology and experimental neurology',NULL,'321-8','11939587','2002 Apr','Impact of genotype and morphology on the prognosis of glioblastoma.',NULL,'61'),(12212,NULL,'Reifenberger J et al','','1','The British journal of dermatology',NULL,'43-51','15656799','2005 Jan','Somatic mutations in the PTCH, SMOH, SUFUH and TP53 genes in sporadic basal cell carcinomas.',NULL,'152'),(12213,NULL,'van Oijen MG et al','','4','Cancer',NULL,'884-93','10679659','2000 Feb 15','The origins of multiple squamous cell carcinomas in the aerodigestive tract.',NULL,'88'),(12214,NULL,'Knig EA et al','','4','Leukemia',NULL,'706-11','10764158','2000 Apr','p53 mutations in hairy cell leukemia.',NULL,'14'),(12215,NULL,'Lu ML et al','','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'171-9','11801555','2002 Jan','Impact of alterations affecting the p53 pathway in bladder cancer on clinical outcome, assessed by conventional and array-based methods.',NULL,'8'),(12216,NULL,'Bazan V et al','','2','Journal of cellular physiology',NULL,'503-9','15316933','2005 Feb','Laser pressure catapulting (LPC): optimization LPC-system and genotyping of colorectal carcinomas.',NULL,'202'),(12217,NULL,'Oyasu R et al','','2','Modern pathology : an official journal of the United States and Canadian Academy of Pathology, Inc',NULL,'170-6','7777479','1995 Feb','p53 gene mutations in human urothelial carcinomas: analysis by immunohistochemistry and single-strand conformation polymorphism.',NULL,'8'),(12218,NULL,'Muzeau F et al','','5','Gastroenterologie clinique et biologique',NULL,'430-7','8761140','1996','[Profile of p53 mutations and abnormal expression of P53 protein in 2 forms of esophageal cancer].',NULL,'20'),(12219,NULL,'Wen WH et al','','1','International journal of gynecological pathology : official journal of the International Society of Gynecological Pathologists',NULL,'29-41','9891239','1999 Jan','p53 mutations and expression in ovarian cancers: correlation with overall survival.',NULL,'18'),(12220,NULL,'Huang J et al','','13','Zhonghua yi xue za zhi',NULL,'1130-3','12921629','2003 Jul 10','[p53 mutations in sporadic colorectal cancers with microsatellite instability].',NULL,'83'),(12221,NULL,'Wong N et al','','10','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'4000-9','11051249','2000 Oct','Genomic aberrations in human hepatocellular carcinomas of differing etiologies.',NULL,'6'),(12222,NULL,'Smith G et al','','14','Proceedings of the National Academy of Sciences of the United States of America',NULL,'9433-8','12093899','2002 Jul 9','Mutations in APC, Kirsten-ras, and p53--alternative genetic pathways to colorectal cancer.',NULL,'99'),(12223,NULL,'Habuchi T et al','','16','Cancer research',NULL,'3795-9','8339293','1993 Aug 15','Influence of cigarette smoking and schistosomiasis on p53 gene mutation in urothelial cancer.',NULL,'53'),(12224,NULL,'Chen YJ et al','','13','Cancer research',NULL,'6473-6','16818615','2006 Jul 1','Association of mutant TP53 with alternative lengthening of telomeres and favorable prognosis in glioma.',NULL,'66'),(12225,NULL,'Ding X et al','','1','Japanese journal of infectious diseases',NULL,'12-8','12711820','2003 Feb','Geographic characterization of hepatitis virus infections, genotyping of hepatitis B virus, and p53 mutation in hepatocellular carcinoma analyzed by in situ detection of viral genomes from carcinoma tissues: comparison among six different countries.',NULL,'56'),(12226,NULL,'Redston MS et al','','11','Cancer research',NULL,'3025-33','8187092','1994 Jun 1','p53 mutations in pancreatic carcinoma and evidence of common involvement of homocopolymer tracts in DNA microdeletions.',NULL,'54'),(12227,NULL,'Forslund A et al','','2','International journal of oncology',NULL,'409-15','12118339','2002 Aug','P53 mutations in colorectal cancer assessed in both genomic DNA and cDNA as compared to the presence of p53 LOH.',NULL,'21'),(12228,NULL,'van Oijen MG et al','','8','Journal of oral pathology & medicine : official publication of the International Association of Oral Pathologists and the American Academy of Oral Pathology',NULL,'367-75','9736425','1998 Sep','Expression of p21 (Waf1/Cip1) in head and neck cancer in relation to proliferation, differentiation, p53 status and cyclin D1 expression.',NULL,'27'),(12229,NULL,'Akasaka T et al','','9','Japanese journal of cancer research : Gann',NULL,'930-7','8878455','1996 Sep','p53 mutation in B-cell lymphoid neoplasms with reference to oncogene rearrangements associated with chromosomal translocations.',NULL,'87'),(12230,NULL,'Kaneko H et al','','5-6','Leukemia & lymphoma',NULL,'449-55','9172810','1996 May','Lack of CD54 expression and mutation of p53 gene relate to the prognosis of childhood Burkitt\'s lymphoma.',NULL,'21'),(12231,NULL,'Rou G et al','doi: 10.1158/1078-0432.CCR-08-0388','21','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'6907-15','18980985','2008 Nov 1','Bendamustine is effective in p53-deficient B-cell neoplasms and requires oxidative stress and caspase-independent signaling.',NULL,'14'),(12232,NULL,'Nakanishi H et al','','6','Laboratory investigation; a journal of technical methods and pathology',NULL,'727-33','9645763','1998 Jun','Mutation of the p53 gene in postradiation sarcoma.',NULL,'78'),(12233,NULL,'Gu QH et al','','3','Zhonghua yi xue yi chuan xue za zhi = Zhonghua yixue yichuanxue zazhi = Chinese journal of medical genetics',NULL,'319-21','17557246','2007 Jun','[Study on the relationship between the polymorphism of p53 gene intron 7 and non-small cell lung cancer (NSCLC) and p53 mutation in NSCLC tissues].',NULL,'24'),(12234,NULL,'Young KH et al','','13','Blood',NULL,'4396-405','17881637','2007 Dec 15','Mutations in the DNA-binding codons of TP53, which are associated with decreased expression of TRAILreceptor-2, predict for poor survival in diffuse large B-cell lymphoma.',NULL,'110'),(12235,NULL,'Ruggeri B et al','','8','Oncogene',NULL,'1503-11','1630814','1992 Aug','Human pancreatic carcinomas and cell lines reveal frequent and multiple alterations in the p53 and Rb-1 tumor-suppressor genes.',NULL,'7'),(12236,NULL,'Harano H et al','','4','Nihon Hinyokika Gakkai zasshi. The japanese journal of urology',NULL,'487-95','10355250','1999 Apr','[p53 tumor suppressor gene mutation and prognosis in 105 cases of bladder cancer--the relationship between mutation of the p53 gene with clinicopathological features and smoking].',NULL,'90'),(12237,NULL,'Bertorelle R et al','','4','The American journal of surgical pathology',NULL,'463-71','7694948','1995 Apr','Association of p53 gene and protein alterations with metastases in colorectal cancer.',NULL,'19'),(12238,NULL,'Marrogi AJ et al','','8','Journal of cutaneous pathology',NULL,'369-78','10551408','1999 Sep','p53 tumor suppressor gene mutations in transformed cutaneous T-cell lymphoma: a study of 12 cases.',NULL,'26'),(12239,NULL,'Pauls E et al','','3','Journal of leukocyte biology',NULL,'659-67','16844765','2006 Sep','Inhibition of HIV-1 replication by RNA interference of p53 expression.',NULL,'80'),(12240,NULL,'Schroeder JC et al','','21','Cancer research',NULL,'7530-8','14612556','2003 Nov 1','p53 mutations in bladder cancer: evidence for exogenous versus endogenous risk factors.',NULL,'63'),(12241,NULL,'Servomaa K et al','','1','Molecular pathology : MP',NULL,'24-30','10884918','2000 Feb','p53 and K-ras gene mutations in carcinoma of the rectum among Finnish women.',NULL,'53'),(12383,NULL,'Jong YJ et al','','1','Cancer genetics and cytogenetics',NULL,'55-65','14697642','2004 Jan 1','Chromosomal comparative genomic hybridization abnormalities in early- and late-onset human breast cancers: correlation with disease progression and TP53 mutations.',NULL,'148'),(12384,NULL,'Lens D et al','','6','Blood',NULL,'2015-23','9058723','1997 Mar 15','p53 abnormalities in B-cell prolymphocytic leukemia.',NULL,'89'),(12385,NULL,'Noffsinger AE et al','','5','Histopathology',NULL,'482-92','11737306','2001 Nov','A unique basal pattern of p53 expression in ulcerative colitis is associated with mutation in the p53 gene.',NULL,'39'),(12386,NULL,'Aas T et al','','7','Nature medicine',NULL,'811-4','8673929','1996 Jul','Specific P53 mutations are associated with de novo resistance to doxorubicin in breast cancer patients.',NULL,'2'),(12387,NULL,'Kuwabara S et al','','4','Japanese journal of cancer research : Gann',NULL,'405-10','9617346','1998 Apr','Heterogeneity of p53 mutational status in esophageal squamous cell carcinoma.',NULL,'89'),(12388,NULL,'Uchida T et al','','4','The Journal of urology',NULL,'1097-104','7869472','1995 Apr','p53 mutations and prognosis in bladder tumors.',NULL,'153'),(12389,NULL,'Klumb CE et al','','2','European journal of haematology',NULL,'81-90','12890146','2003 Aug','DNA sequence profile of TP53 gene mutations in childhood B-cell non-Hodgkin\'s lymphomas: prognostic implications.',NULL,'71'),(12390,NULL,'Moll UM et al','','12','Human pathology',NULL,'1293-301','8522300','1995 Dec','p53 mediated tumor cell response to chemotherapeutic DNA damage: a preliminary study in matched pairs of breast cancer biopsies.',NULL,'26'),(12391,NULL,'Ren ZP et al','','10','Journal of neuropathology and experimental neurology',NULL,'944-54','17917588','2007 Oct','Molecular genetic analysis of p53 intratumoral heterogeneity in human astrocytic brain tumors.',NULL,'66'),(12392,NULL,'Sommer SS et al','','4','Journal of the National Cancer Institute',NULL,'246-52','1734086','1992 Feb 19','Pattern of p53 gene mutations in breast cancers of women of the midwestern United States.',NULL,'84'),(12393,NULL,'Brockmller J et al','','6','Pharmacogenetics',NULL,'535-45','9014203','1996 Dec','Polymorphic enzymes of xenobiotic metabolism as modulators of acquired P53 mutations in bladder cancer.',NULL,'6'),(12394,NULL,'Elledge RM et al','','1-2','Breast cancer research and treatment',NULL,'95-102','8260732','1993','William L. McGuire Memorial Symposium. The role and prognostic significance of p53 gene alterations in breast cancer.',NULL,'27'),(12395,NULL,'Kyritsis AP et al','','1','Molecular carcinogenesis',NULL,'1-4','8561860','1996 Jan','Correlation of p53 immunoreactivity and sequencing in patients with glioma.',NULL,'15'),(12396,NULL,'Lipton L et al','','22','Cancer research',NULL,'7595-9','14633673','2003 Nov 15','Carcinogenesis in MYH-associated polyposis follows a distinct genetic pathway.',NULL,'63'),(12397,NULL,'Taylor JA et al','','8889','Lancet (London, England)',NULL,'86-7','7903781','1994 Jan 8','p53 mutation hotspot in radon-associated lung cancer.',NULL,'343'),(12398,NULL,'Brash DE et al','','22','Proceedings of the National Academy of Sciences of the United States of America',NULL,'10124-8','1946433','1991 Nov 15','A role for sunlight in skin cancer: UV-induced p53 mutations in squamous cell carcinoma.',NULL,'88'),(12399,NULL,'Meyer-Puttlitz B et al','','3','The American journal of pathology',NULL,'853-7','9284834','1997 Sep','Molecular genetic analysis of giant cell glioblastomas.',NULL,'151'),(12400,NULL,'Dansonka-Mieszkowska A et al','','Pt 5','Annals of human genetics',NULL,'594-604','16907706','2006 Sep','Geographical variations in TP53 mutational spectrum in ovarian carcinomas.',NULL,'70'),(12401,NULL,'Sanchez-Cespedes M et al','','43','Oncogene',NULL,'5843-9','10557071','1999 Oct 21','Inactivation of the INK4A/ARF locus frequently coexists with TP53 mutations in non-small cell lung cancer.',NULL,'18'),(12402,NULL,'Sanchez-Cespedes M et al','','9','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'2450-4','10499618','1999 Sep','Molecular detection of neoplastic cells in lymph nodes of metastatic colorectal cancer patients predicts recurrence.',NULL,'5'),(12403,NULL,'Gentile M et al','','8','European journal of cancer (Oxford, England : 1990)',NULL,'1202-7','10615230','1999 Aug','p53 and survival in early onset breast cancer: analysis of gene mutations, loss of heterozygosity and protein accumulation.',NULL,'35'),(12404,NULL,'Lanza C et al','','1','Genes, chromosomes & cancer',NULL,'48-53','8824725','1996 Jan','Distribution of TP53 mutations among acute leukemias with MLL rearrangements.',NULL,'15'),(12405,NULL,'Mazars R et al','','9','Oncogene',NULL,'1685-90','1923532','1991 Sep','p53 mutations in ovarian cancer: a late event?',NULL,'6'),(12406,NULL,'Blau O et al','','8','Leukemia research',NULL,'721-9','9379679','1997 Aug','Exon 5 mutations in the p53 gene in relapsed childhood acute lymphoblastic leukemia.',NULL,'21'),(12407,NULL,'Radig K et al','','11','Human pathology',NULL,'1310-6','9824113','1998 Nov','p53 gene mutations in osteosarcomas of low-grade malignancy.',NULL,'29'),(12408,NULL,'De Re V et al','','5','International journal of cancer',NULL,'662-7','8314342','1994 Mar 1','p53 protein over-expression and p53 gene abnormalities in HIV-1-related non-Hodgkin\'s lymphomas.',NULL,'56'),(12409,NULL,'Suzuki H et al','','5','The Prostate',NULL,'318-24','8610059','1996 May','Detection of human papillomavirus DNA and p53 gene mutations in human prostate cancer.',NULL,'28'),(12410,NULL,'Kleivi K et al','','','Molecular cancer',NULL,'2','17201907','2007 Jan 3','Gene expression profiles of primary colorectal carcinomas, liver metastases, and carcinomatoses.',NULL,'6'),(12411,NULL,'Simmons ML et al','','3','Cancer research',NULL,'1122-8','11221842','2001 Feb 1','Analysis of complex relationships between age, p53, epidermal growth factor receptor, and survival in glioblastoma patients.',NULL,'61'),(12412,NULL,'Bamford S et al','','2','British journal of cancer',NULL,'355-8','15188009','2004 Jul 19','The COSMIC (Catalogue of Somatic Mutations in Cancer) database and website.',NULL,'91'),(12413,NULL,'Park YK et al','','3','Oncology reports',NULL,'533-7','11295075','2001 May-Jun','P53 mutations in Ewing\'s sarcoma.',NULL,'8'),(12414,NULL,'Shaminie J et al','','1','Pathology',NULL,'39-44','15875732','2005 Feb','p53 alterations in sequential biopsies of Asian follicular lymphoma: a study of immunohistochemical staining pattern and gene mutations by PCR-SSCP in paraffin-embedded tissues.',NULL,'37'),(12415,NULL,'Tannapfel A et al','','5','The Journal of pathology',NULL,'624-31','12210082','2002 Aug','Genetic and epigenetic alterations of the INK4a-ARF pathway in cholangiocarcinoma.',NULL,'197'),(12416,NULL,'Galic V et al','','3','Genes, chromosomes & cancer',NULL,'239-47','17171684','2007 Mar','Common polymorphisms in TP53 and MDM2 and the relationship to TP53 mutations and clinical outcomes in women with ovarian and peritoneal carcinomas.',NULL,'46'),(12417,NULL,'Alsner J et al','','10','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'3923-31','11051239','2000 Oct','Heterogeneity in the clinical phenotype of TP53 mutations in breast cancer patients.',NULL,'6'),(12418,NULL,'Agell L et al','doi: 10.1038/modpathol.2008.145','12','Modern pathology : an official journal of the United States and Canadian Academy of Pathology, Inc',NULL,'1470-8','19020536','2008 Dec','KLF6 and TP53 mutations are a rare event in prostate cancer: distinguishing between Taq polymerase artifacts and true mutations.',NULL,'21'),(12419,NULL,'Rhei E et al','','15','Cancer research',NULL,'3193-6','9699640','1998 Aug 1','Molecular genetic characterization of BRCA1- and BRCA2-linked hereditary ovarian cancers.',NULL,'58'),(12420,NULL,'Lam V et al','','10','Leukemia research',NULL,'871-80','10573131','1999 Oct','p53 gene status and chemosensitivity of childhood acute lymphoblastic leukemia cells to adriamycin.',NULL,'23'),(12421,NULL,'McDaniel AS et al','doi: 10.1158/0008-5472.CAN-15-1004','24','Cancer research',NULL,'5219-27','26670561','2015 Dec 15','Genomic Profiling of Penile Squamous Cell Carcinoma Reveals New Opportunities for Targeted Therapy.',NULL,'75'),(12422,NULL,'Halatsch ME et al','','3','Journal of neurosurgery',NULL,'523-33','15035290','2004 Mar','Inverse correlation of epidermal growth factor receptor messenger RNA induction and suppression of anchorage-independent growth by OSI-774, an epidermal growth factor receptor tyrosine kinase inhibitor, in glioblastoma multiforme cell lines.',NULL,'100'),(12423,NULL,'Conway K et al','','7','Cancer research',NULL,'1987-95','11929815','2002 Apr 1','Prevalence and spectrum of p53 mutations associated with smoking in breast cancer.',NULL,'62'),(12424,NULL,'Rose SL et al','','11','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'4139-44','14519637','2003 Sep 15','The impact of p53 protein core domain structural alteration on ovarian cancer survival.',NULL,'9'),(12425,NULL,'Ramchurren N et al','','3','International journal of cancer',NULL,'237-44','7628866','1995 Jul 28','Molecular events underlying schistosomiasis-related bladder cancer.',NULL,'62'),(12426,NULL,'Ng IO et al','','3','Oncogene',NULL,'985-90','8108145','1994 Mar','p53 gene mutation spectrum in hepatocellular carcinomas in Hong Kong Chinese.',NULL,'9'),(12427,NULL,'Anelli A et al','','9','Cancer',NULL,'2233-8','7712430','1995 May 1','Mutations of the p53 gene in male breast cancer.',NULL,'75'),(12428,NULL,'Ichikawa A et al','','10','Blood',NULL,'2701-7','1375111','1992 May 15','Mutations of p53 gene and their relation to disease progression in B-cell lymphoma.',NULL,'79'),(12429,NULL,'Koga T et al','','4','International journal of cancer',NULL,'232-9','11400116','2001 Jul 20','Heterogeneous distribution of P53 immunoreactivity in human lung adenocarcinoma correlates with MDM2 protein expression, rather than with P53 gene mutation.',NULL,'95'),(12430,NULL,'Das A et al','','2','Journal of neuro-oncology',NULL,'117-25','12635658','2002 Nov','Glioblastoma multiforme in an Asian population: evidence for a distinct genetic pathway.',NULL,'60'),(12431,NULL,'Skilling JS et al','','1','Oncogene',NULL,'117-23','8700537','1996 Jul 4','An abundance of p53 null mutations in ovarian carcinoma.',NULL,'13'),(12432,NULL,'Bernardini S et al','','4','The Journal of urology',NULL,'1496-501','10492244','1999 Oct','Immunohistochemical detection of p53 protein overexpression versus gene sequencing in urinary bladder carcinomas.',NULL,'162'),(12433,NULL,'Rosty C et al','','3','International journal of cancer',NULL,'162-7','11307149','2001 May 20','Determination of microsatellite instability, p53 and K-RAS mutations in hepatic metastases from patients with colorectal cancer: relationship with response to 5-fluorouracil and survival.',NULL,'95'),(12434,NULL,'Chen FM et al','','1','Cancer letters',NULL,'59-67','15050734','2004 Apr 15','High frequency of G/C transversion on p53 gene alterations in breast cancers from Taiwan.',NULL,'207'),(12435,NULL,'Miller CW et al','','7','Cancer research',NULL,'1695-8','1312896','1992 Apr 1','p53 mutations in human lung tumors.',NULL,'52'),(12436,NULL,'Miyaki M et al','','43','Oncogene',NULL,'6689-93','12242668','2002 Sep 26','High incidence of protein-truncating mutations of the p53 gene in liver metastases of colorectal carcinomas.',NULL,'21'),(12437,NULL,'Lindgren D et al','','18','Oncogene',NULL,'2685-96','16532037','2006 Apr 27','Molecular characterization of early-stage bladder carcinomas by expression profiles, FGFR3 mutation status, and loss of 9q.',NULL,'25'),(12438,NULL,'Ooe A et al','','3','Breast cancer research and treatment',NULL,'305-15','16821082','2007 Mar','Possible involvement of CCT5, RGS3, and YKT6 genes up-regulated in p53-mutated tumors in resistance to docetaxel in human breast cancers.',NULL,'101'),(12439,NULL,'Kashiwazaki H et al','','22','Oncogene',NULL,'2667-74','9400993','1997 Nov 27','High frequency of p53 mutations in human oral epithelial dysplasia and primary squamous cell carcinoma detected by yeast functional assay.',NULL,'15'),(12440,NULL,'Kim CJ et al','','1','European journal of cancer (Oxford, England : 1990)',NULL,'136-41','14687797','2004 Jan','Genetic analysis of the LKB1/STK11 gene in hepatocellular carcinomas.',NULL,'40'),(12441,NULL,'Jou YS et al','','9','Cancer research',NULL,'3030-6','15126338','2004 May 1','Clustering of minimal deleted regions reveals distinct genetic pathways of human hepatocellular carcinoma.',NULL,'64'),(12442,NULL,'Xinarianos G et al','','3','International journal of cancer',NULL,'248-52','12209975','2002 Sep 20','p53 status correlates with the differential expression of the DNA mismatch repair protein MSH2 in non-small cell lung carcinoma.',NULL,'101'),(12443,NULL,'Serizawa RR et al','doi: 10.1002/ijc.25651','1','International journal of cancer',NULL,'78-87','20824703','2011 Jul 1','Integrated genetic and epigenetic analysis of bladder cancer reveals an additive diagnostic value of FGFR3 mutations and hypermethylation events.',NULL,'129'),(12444,NULL,'Bellido M et al','','9','Haematologica',NULL,'908-17','12217802','2002 Sep','Bcl-6 p53 mutations in lymphomas carrying the bcl-2/Jh rearrangement.',NULL,'87'),(12445,NULL,'Lpez-Guerrero JA et al','','6','Laboratory investigation; a journal of technical methods and pathology',NULL,'803-14','11406642','2001 Jun','Molecular analysis of the 9p21 locus and p53 genes in Ewing family tumors.',NULL,'81'),(12446,NULL,'Dahiya R et al','','2','British journal of cancer',NULL,'264-8','8688333','1996 Jul','P53 tumour-suppressor gene mutations are mainly localised on exon 7 in human primary and metastatic prostate cancer.',NULL,'74'),(12447,NULL,'Hu N et al','','4','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'883-91','11309337','2001 Apr','Frequent inactivation of the TP53 gene in esophageal squamous cell carcinoma from a high-risk population in China.',NULL,'7'),(12448,NULL,'Russell SJ et al','','6','Cancer',NULL,'1339-42','7882284','1995 Mar 15','p53 mutations, O6-alkylguanine DNA alkyltransferase activity, and sensitivity to procarbazine in human brain tumors.',NULL,'75'),(12449,NULL,'Hongyo T et al','','2','Oncology reports',NULL,'265-71','15643509','2005 Feb','p53, K-ras, c-kit and beta-catenin gene mutations in sinonasal NK/T-cell lymphoma in Korea and Japan.',NULL,'13'),(12450,NULL,'Andriani F et al','','1','International journal of cancer',NULL,'91-6','14618621','2004 Jan 1','Detecting lung cancer in plasma with the use of multiple genetic markers.',NULL,'108'),(12451,NULL,'Meinhold-Heerlein I et al','','2','Oncology',NULL,'176-88','11244334','2001','Evaluation of methods to detect p53 mutations in ovarian cancer.',NULL,'60'),(12452,NULL,'Yeargin J et al','','5','The Journal of clinical investigation',NULL,'2111-7','8486778','1993 May','P53 mutation in acute T cell lymphoblastic leukemia is of somatic origin and is stable during establishment of T cell acute lymphoblastic leukemia cell lines.',NULL,'91'),(12453,NULL,'Szymaska K et al','doi: 10.1016/j.canlet.2009.11.024','1','Cancer letters',NULL,'92-8','20137853','2010 Jul 1','TP53, EGFR, and KRAS mutations in relation to VHL inactivation and lifestyle risk factors in renal-cell carcinoma from central and eastern Europe.',NULL,'293'),(12454,NULL,'Hamelin R et al','','3','International journal of cancer',NULL,'336-40','8168993','1994 May 1','p53 mutations in human tumors with chimeric EWS/FLI-1 genes.',NULL,'57'),(12455,NULL,'Pinyol M et al','','6','The American journal of pathology',NULL,'1987-96','10854221','2000 Jun','INK4a/ARF locus alterations in human non-Hodgkin\'s lymphomas mainly occur in tumors with wild-type p53 gene.',NULL,'156'),(12456,NULL,'Chilosi M et al','','10','Blood',NULL,'4012-20','8916968','1996 Nov 15','p21/WAF1 cyclin-kinase inhibitor expression in non-Hodgkin\'s lymphomas: a potential marker of p53 tumor-suppressor gene function.',NULL,'88'),(12457,NULL,'Ueno Y et al','','2','Cancer letters',NULL,'289-300','16459017','2006 Sep 28','Prognostic significance of p53 mutation in suboptimally resected advanced ovarian carcinoma treated with the combination chemotherapy of paclitaxel and carboplatin.',NULL,'241'),(12458,NULL,'Puhalla H et al','','2C','Anticancer research',NULL,'1201-6','15154647','2004 Mar-Apr','p53 analysis in gallbladder cancer: comparison of gene analysis versus immunohistochemistry.',NULL,'24'),(12459,NULL,'Kastrinakis WV et al','','4','Oncogene',NULL,'647-52','7651727','1995 Aug 17','Increased incidence of p53 mutations is associated with hepatic metastasis in colorectal neoplastic progression.',NULL,'11'),(12460,NULL,'Tomizawa Y et al','','4','Oncogene',NULL,'1007-14','10023676','1999 Jan 28','Correlation between the status of the p53 gene and survival in patients with stage I non-small cell lung carcinoma.',NULL,'18'),(12461,NULL,'Nishimori H et al','','1','Journal of neuro-oncology',NULL,'17-22','10896202','2000','Analysis of the p300/CBP-Associated Factor (PCAF) gene in astrocytic tumors.',NULL,'46'),(12462,NULL,'Peller S et al','','3','Molecular carcinogenesis',NULL,'166-72','7619219','1995 Jul','p53 mutations in matched primary and metastatic human tumors.',NULL,'13'),(12463,NULL,'Saxena A et al','','23','Cancer research',NULL,'6716-21','1358438','1992 Dec 1','Evidence for the involvement of a potential second tumor suppressor gene on chromosome 17 distinct from p53 in malignant astrocytomas.',NULL,'52'),(12464,NULL,'van Meyel DJ et al','','1','Annals of neurology',NULL,'120-2','8285583','1994 Jan','Absence of hereditary mutations in exons 5 through 9 of the p53 gene and exon 24 of the neurofibromin gene in families with glioma.',NULL,'35'),(12465,NULL,'Du M et al','','12','Blood',NULL,'4587-93','8541549','1995 Dec 15','The accumulation of p53 abnormalities is associated with progression of mucosa-associated lymphoid tissue lymphoma.',NULL,'86'),(12466,NULL,'Buzin CH et al','','1','Nutrition and cancer',NULL,'72-7','11588905','2001','Low frequency of p53 gene mutations in breast cancers of Japanese-American women.',NULL,'39'),(12467,NULL,'Ding Q et al','','11','Zhonghua wai ke za zhi [Chinese journal of surgery]',NULL,'684-6','8731915','1995 Nov','[The study of p53 gene mutation in human bladder cancer].',NULL,'33'),(12468,NULL,'Niklinska W et al','','2','European journal of cancer prevention : the official journal of the European Cancer Prevention Organisation (ECP)',NULL,'81-7','10830574','2000 Apr','p53 gene mutation and protein expression in operable non-small cell lung cancer in Poland.',NULL,'9'),(12469,NULL,'Hanada K et al','','7','European journal of cancer (Oxford, England : 1990)',NULL,'1136-40','9376195','1997 Jun','TP53 mutations in stage I gallbladder carcinoma with special attention to growth patterns.',NULL,'33'),(12470,NULL,'Cunningham J et al','','7','Cancer research',NULL,'1974-80','1551126','1992 Apr 1','Expression of p53 and 17p allelic loss in colorectal carcinoma.',NULL,'52'),(12471,NULL,'Norberg T et al','','4','International journal of cancer',NULL,'376-83','9699530','1998 Aug 21','Comparison between p53 protein measurements using the luminometric immunoassay and immunohistochemistry with detection of p53 gene mutations using cDNA sequencing in human breast tumors.',NULL,'79'),(12472,NULL,'Nam CW et al','','22','World journal of gastroenterology',NULL,'3526-33','18567082','2008 Jun 14','Fragile histidine triad gene alterations are not essential for hepatocellular carcinoma development in South Korea.',NULL,'14'),(12473,NULL,'Zhang LF et al','','12','Carcinogenesis',NULL,'2949-51','8001261','1994 Dec','p53 mutations in larynx cancer.',NULL,'15'),(12474,NULL,'Yamamoto S et al','','2','Japanese journal of cancer research : Gann',NULL,'181-9','10761705','2000 Feb','p53 status in multiple human urothelial cancers: assessment for clonality by the yeast p53 functional assay in combination with p53 immunohistochemistry.',NULL,'91'),(12475,NULL,'Niimi H et al','','4','Leukemia',NULL,'635-44','16467864','2006 Apr','Hyperactivation of the RAS signaling pathway in myelodysplastic syndrome with AML1/RUNX1 point mutations.',NULL,'20'),(12476,NULL,'Kang JH et al','','4','Laboratory investigation; a journal of technical methods and pathology',NULL,'573-9','11304577','2001 Apr','Methylation in the p53 promoter is a supplementary route to breast carcinogenesis: correlation between CpG methylation in the p53 promoter and the mutation of the p53 gene in the progression from ductal carcinoma in situ to invasive ductal carcinoma.',NULL,'81'),(12477,NULL,'Giglia G et al','','19','Cancer research',NULL,'4402-9','9766670','1998 Oct 1','p53 mutations in skin and internal tumors of xeroderma pigmentosum patients belonging to the complementation group C.',NULL,'58'),(12478,NULL,'Suspitsin EN et al','doi: 10.1016/j.cancergencyto.2008.03.005','2','Cancer genetics and cytogenetics',NULL,'119-21','18617062','2008 Jul 15','TP53 mutations in synchronous and metachronous bilateral breast carcinomas.',NULL,'184'),(12479,NULL,'Bakkar AA et al','','23','Cancer research',NULL,'8108-12','14678961','2003 Dec 1','FGFR3 and TP53 gene mutations define two distinct pathways in urothelial cell carcinoma of the bladder.',NULL,'63'),(12480,NULL,'Ichimura K et al','','2','Cancer research',NULL,'417-24','10667596','2000 Jan 15','Deregulation of the p14ARF/MDM2/p53 pathway is a prerequisite for human astrocytic gliomas with G1-S transition control gene abnormalities.',NULL,'60'),(12481,NULL,'Marchetti A et al','','3','The Journal of pathology',NULL,'240-6','9614374','1998 Mar','FHIT and p53 gene abnormalities in bronchioloalveolar carcinomas. Correlations with clinicopathological data and K-ras mutations.',NULL,'184'),(12482,NULL,'Inukai T et al','doi: 10.1002/ajh.21738','7','American journal of hematology',NULL,'535-7','20575032','2010 Jul','A specific linkage between the incidence of TP53 mutations and type of chromosomal translocations in B-precursor acute lymphoblastic leukemia cell lines.',NULL,'85'),(12483,NULL,'Duddy PM et al','','3','The Journal of molecular diagnostics : JMD',NULL,'139-44','11229518','2000 Aug','Improving the detection of p53 mutations in breast cancer by use of the FASAY, a functional assay.',NULL,'2'),(12484,NULL,'Mori S et al','','8','Cancer',NULL,'1673-82','15073856','2004 Apr 15','p53 apoptotic pathway molecules are frequently and simultaneously altered in nonsmall cell lung carcinoma.',NULL,'100'),(12485,NULL,'Lianidou ES et al','','1B','Anticancer research',NULL,'749-56','10216487','1999 Jan-Feb','Immunofluorometric assay of p53 protein versus sequencing of p53 exons 5 to 9 for the detection of p53 abnormalities in ovarian carcinoma.',NULL,'19'),(12486,NULL,'Jones S et al','doi: 10.1126/science.1164368','5897','Science (New York, N.Y.)',NULL,'1801-6','18772397','2008 Sep 26','Core signaling pathways in human pancreatic cancers revealed by global genomic analyses.',NULL,'321'),(12487,NULL,'Tang R et al','','6','International journal of cancer',NULL,'863-8','11275993','2001 Mar 15','Mutations of p53 gene in human colorectal cancer: distinct frameshifts among populations.',NULL,'91'),(12488,NULL,'Kanashiro CA et al','','26','Proceedings of the National Academy of Sciences of the United States of America',NULL,'15836-41','14660794','2003 Dec 23','Inhibition of mutant p53 expression and growth of DMS-153 small cell lung carcinoma by antagonists of growth hormone-releasing hormone and bombesin.',NULL,'100'),(12489,NULL,'Tsuda H et al','','4','Japanese journal of cancer research : Gann',NULL,'394-401','8099903','1993 Apr','p53 mutations and c-erbB-2 amplification in intraductal and invasive breast carcinomas of high histologic grade.',NULL,'84'),(12490,NULL,'Caleffi M et al','','8','Cancer',NULL,'2147-56','8156519','1994 Apr 15','p53 gene mutations and steroid receptor status in breast cancer. Clinicopathologic correlations and prognostic assessment.',NULL,'73'),(12491,NULL,'Fogelfeld L et al','','8','The Journal of clinical endocrinology and metabolism',NULL,'3039-44','8768871','1996 Aug','p53 gene mutations in radiation-induced thyroid cancer.',NULL,'81'),(12492,NULL,'Salani R et al','','3','International journal of gynecological cancer : official journal of the International Gynecological Cancer Society',NULL,'487-91','17692090','2008 May-Jun','Assessment of TP53 mutation using purified tissue samples of ovarian serous carcinomas reveals a higher mutation rate than previously reported and does not correlate with drug resistance.',NULL,'18'),(12493,NULL,'Amikura T et al','','2','Gynecologic oncology',NULL,'365-71','16337994','2006 Feb','Mutational analysis of TP53 and p21 in familial and sporadic ovarian cancer in Japan.',NULL,'100'),(12494,NULL,'Robert V et al','','4','Carcinogenesis',NULL,'563-5','10753186','2000 Apr','High frequency in esophageal cancers of p53 alterations inactivating the regulation of genes involved in cell cycle and apoptosis.',NULL,'21'),(12495,NULL,'Mazars R et al','','14','Cancer research',NULL,'3918-23','1617667','1992 Jul 15','p53 mutations occur in aggressive breast cancer.',NULL,'52'),(12496,NULL,'Schorge JO et al','','5','Cancer research',NULL,'1361-4','10728699','2000 Mar 1','BRCA1-related papillary serous carcinoma of the peritoneum has a unique molecular pathogenesis.',NULL,'60'),(12497,NULL,'Rand A et al','','2','Cancer letters',NULL,'183-91','8556707','1996 Jan 2','p53 functional loss in a colon cancer cell line with two missense mutations (218leu and 248trp) on separate alleles.',NULL,'98'),(12498,NULL,'Platz A et al','','6','British journal of cancer',NULL,'936-41','8826861','1996 Sep','Genes involved in cell cycle G1 checkpoint control are frequently mutated in human melanoma metastases.',NULL,'74'),(12499,NULL,'Shiraishi S et al','','2','Cancer',NULL,'249-57','12124823','2002 Jul 15','Influence of p53 mutations on prognosis of patients with glioblastoma.',NULL,'95'),(12500,NULL,'Kortm KM et al','doi: 10.1111/bjh.13171','4','British journal of haematology',NULL,'507-10','25302557','2015 Feb','Targeted sequencing using a 47 gene multiple myeloma mutation panel (M(3) P) in -17p high risk disease.',NULL,'168'),(12501,NULL,'Powell B et al','','2','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'443-51','10690522','2000 Feb','Prognostic significance of mutations to different structural and functional regions of the p53 gene in breast cancer.',NULL,'6'),(12556,NULL,'Takahashi Y et al','','1','Oncology',NULL,'54-60','12457032','2003','Clinical application of oligonucleotide probe array for full-length gene sequencing of TP53 in colon cancer.',NULL,'64'),(12628,NULL,'Abou Alaiwi S et al','doi: 10.1158/2326-6066.CIR-19-0866','8','Cancer immunology research',NULL,'1075-1084','32321774','2020 Aug','Mammalian SWI/SNF Complex Genomic Alterations and Immune Checkpoint Blockade in Solid Tumors.',NULL,'8'),(12653,NULL,'Hou JY et al','doi: 10.1002/cncr.30473','8','Cancer',NULL,'1333-1344','28026870','2017 Apr 15','Vulvar and vaginal melanoma: A unique subclass of mucosal melanoma based on a comprehensive molecular analysis of 51 cases compared with 2253 cases of nongynecologic melanoma.',NULL,'123'),(12654,NULL,'Ollila TA et al','doi: 10.1182/blood.2020007236','8','Blood',NULL,'1120-1124','32877514','2021 Feb 25','Genomic subtypes may predict the risk of central nervous system recurrence in diffuse large B-cell lymphoma.',NULL,'137'),(12655,NULL,'Kim N et al','doi: 10.3390/ijms22073783','7','International journal of molecular sciences',NULL,'','33917428','2021 Apr 6','Novel and Potent Small Molecules against Melanoma Harboring BRAF Class I/II/III Mutants for Overcoming Drug Resistance.',NULL,'22'),(12656,NULL,'Silva R et al','doi: 10.1016/j.crwh.2022.e00395','','Case reports in women\'s health',NULL,'e00395','35198414','2022 Apr','Investigating a clinically actionable BRAF mutation for monitoring low-grade serous ovarian cancer: A case report.',NULL,'34'),(12657,NULL,'Li S et al','doi: 10.1097/CAD.0000000000001367','9','Anti-cancer drugs',NULL,'963-965','36136993','2022 Oct 1','Response to osimertinib plus trametinib in a heavily treated epidermal growth factor receptor (EGFR)-positive NSCLC harboring a rare, acquired rapidly accelerated fibrosarcoma B-type (BRAF) p.D594N mutation: a case report.',NULL,'33'),(12658,NULL,'Owsley J et al','doi: 10.1177/1535370220959657','1','Experimental biology and medicine (Maywood, N.J.)',NULL,'31-39','33019809','2021 Jan','Prevalence of class I-III BRAF mutations among 114,662 cancer patients in a large genomic database.',NULL,'246'),(12659,NULL,'Cheasley D et al','doi: 10.1038/s41467-019-11862-x','1','Nature communications',NULL,'3935','31477716','2019 Sep 2','The molecular origin and taxonomy of mucinous ovarian carcinoma.',NULL,'10'),(12660,NULL,'Dagogo-Jack I et al','doi: 10.1158/1078-0432.CCR-18-2062','1','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'158-165','30224342','2019 Jan 1','Impact of BRAF Mutation Class on Disease Characteristics and Clinical Outcomes in BRAF-mutant Lung Cancer.',NULL,'25'),(12661,NULL,'Johnson DB et al','doi: 10.1158/1078-0432.CCR-19-3443','8','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'1812-1819','31924734','2020 Apr 15','Trametinib Activity in Patients with Solid Tumors and Lymphomas Harboring BRAF Non-V600 Mutations or Fusions: Results from NCI-MATCH (EAY131).',NULL,'26'),(12662,NULL,'Midthun L et al','doi: 10.21037/jgo.2019.01.10','3','Journal of gastrointestinal oncology',NULL,'577-581','31183211','2019 Jun','Concomitant KRAS and BRAF mutations in colorectal cancer.',NULL,'10'),(12664,NULL,'Bernocchi O et al','doi: 10.3390/ph14020159','2','Pharmaceuticals (Basel, Switzerland)',NULL,'','33669326','2021 Feb 16','Tumor Type Agnostic Therapy Carrying BRAF Mutation: Case Reports and Review of Literature.',NULL,'14'),(12665,NULL,'Mazieres J et al','doi: 10.1016/j.annonc.2019.10.022','2','Annals of oncology : official journal of the European Society for Medical Oncology',NULL,'289-294','31959346','2020 Feb','Vemurafenib in non-small-cell lung cancer patients with BRAF(V600) and BRAF(nonV600) mutations.',NULL,'31'),(12666,NULL,'Karoulia Z et al','doi: 10.1016/j.ccell.2016.06.024','3','Cancer cell',NULL,'485-498','27523909','2016 Sep 12','An Integrated Model of RAF Inhibitor Action Predicts Inhibitor Activity against Oncogenic BRAF Signaling.',NULL,'30'),(12668,NULL,'Villaruz LC et al','doi: 10.1002/cncr.29042','3','Cancer',NULL,'448-56','25273224','2015 Feb 1','Clinicopathologic features and outcomes of patients with lung adenocarcinomas harboring BRAF mutations in the Lung Cancer Mutation Consortium.',NULL,'121'),(12669,NULL,'Si L et al','doi: 10.1016/j.ejca.2011.06.056','1','European journal of cancer (Oxford, England : 1990)',NULL,'94-100','21788131','2012 Jan','Prevalence of BRAF V600E mutation in Chinese melanoma patients: large scale analysis of BRAF and NRAS mutations in a 432-case cohort.',NULL,'48'),(12670,NULL,'Kinno T et al','doi: 10.1093/annonc/mdt495','1','Annals of oncology : official journal of the European Society for Medical Oncology',NULL,'138-42','24297085','2014 Jan','Clinicopathological features of nonsmall cell lung carcinomas with BRAF mutations.',NULL,'25'),(12671,NULL,'Gupta A et al','doi: 10.1038/s41416-019-0673-5','4','British journal of cancer',NULL,'506-516','31839677','2020 Feb','Dual-specificity protein phosphatase DUSP4 regulates response to MEK inhibition in BRAF wild-type melanoma.',NULL,'122'),(12672,NULL,'Mu Y et al','doi: 10.3389/fonc.2020.00603','','Frontiers in oncology',NULL,'603','32411601','2020','Clinical Characteristics and Treatment Outcomes of 65 Patients With BRAF-Mutated Non-small Cell Lung Cancer.',NULL,'10'),(12673,NULL,'Nichols RJ et al','doi: 10.1038/s41556-018-0169-1','9','Nature cell biology',NULL,'1064-1073','30104724','2018 Sep','RAS nucleotide cycling underlies the SHP2 phosphatase dependence of mutant BRAF-, NF1- and RAS-driven cancers.',NULL,'20'),(12674,NULL,'Otani H et al','doi: 10.1371/journal.pone.0129838','6','PloS one',NULL,'e0129838','26090892','2015','TAE226, a Bis-Anilino Pyrimidine Compound, Inhibits the EGFR-Mutant Kinase Including T790M Mutant to Show Anti-Tumor Effect on EGFR-Mutant Non-Small Cell Lung Cancer Cells.',NULL,'10'),(12675,NULL,'Negrao MV et al','doi: 10.1016/j.jtho.2020.05.021','10','Journal of thoracic oncology : official publication of the International Association for the Study of Lung Cancer',NULL,'1611-1623','32540409','2020 Oct','Molecular Landscape of BRAF-Mutant NSCLC Reveals an Association Between Clonality and Driver Mutations and Identifies Targetable Non-V600 Driver Mutations.',NULL,'15'),(12676,NULL,'Wen C et al','doi: 10.1364/BOE.4.002835','12','Biomedical optics express',NULL,'2835-45','24409384','2013','Single-molecule force measurement via optical tweezers reveals different kinetic features of two BRaf mutants responsible for cardio-facial-cutaneous (CFC) syndrome.',NULL,'4'),(12677,NULL,'Zhao Y et al','doi: 10.1001/jamanetworkopen.2020.35479','1','JAMA network open',NULL,'e2035479','33507258','2021 Jan 4','Assessment of RAS Dependency for BRAF Alterations Using Cancer Genomic Databases.',NULL,'4'),(12678,NULL,'Peng SB et al','doi: 10.1016/j.ccell.2015.08.002','3','Cancer cell',NULL,'384-98','26343583','2015 Sep 14','Inhibition of RAF Isoforms and Active Dimers by LY3009120 Leads to Anti-tumor Activities in RAS or BRAF Mutant Cancers.',NULL,'28'),(12679,NULL,'Tong L et al','doi: 10.7150/thno.34070','19','Theranostics',NULL,'5532-5541','31534501','2019','Tumor-derived DNA from pleural effusion supernatant as a promising alternative to tumor tissue in genomic profiling of advanced lung cancer.',NULL,'9'),(12680,NULL,'Sadashima S et al','doi: 10.1111/neup.12693','6','Neuropathology : official journal of the Japanese Society of Neuropathology',NULL,'646-650','32996219','2020 Dec','A juvenile case of epilepsy-associated, isocitrate dehydrogenase wild-type/histone 3 wild-type diffuse glioma with a rare BRAF (A598T) mutation.',NULL,'40'),(12681,NULL,'Chao WR et al','doi: 10.1016/j.tjog.2021.09.019','6','Taiwanese journal of obstetrics & gynecology',NULL,'1072-1077','34794740','2021 Nov','High frequency of BRAF mutations in primary mucinous ovarian carcinoma of Taiwanese patients.',NULL,'60'),(12682,NULL,'Kiniwa Y et al','doi: 10.1186/s12885-021-08016-y','1','BMC cancer',NULL,'287','33731038','2021 Mar 17','Usefulness of monitoring circulating tumor cells as a therapeutic biomarker in melanoma with BRAF mutation.',NULL,'21'),(12683,NULL,'Jain P et al','doi: 10.18632/oncotarget.20949','49','Oncotarget',NULL,'84697-84713','29156677','2017 Oct 17','Overcoming resistance to single-agent therapy for oncogenic BRAF gene fusions via combinatorial targeting of MAPK and PI3K/mTOR signaling pathways.',NULL,'8'),(12684,NULL,'Turner JA et al','doi: 10.1038/s41388-018-0514-7','8','Oncogene',NULL,'1296-1308','30254212','2019 Feb','BRAF fusions identified in melanomas have variable treatment responses and phenotypes.',NULL,'38'),(12753,NULL,'Su PL et al','doi: 10.1016/j.jtocrr.2021.100202','8','JTO clinical and research reports',NULL,'100202','34590045','2021 Aug','Durable Response to Combined Dabrafenib and Trametinib in a Patient With BRAF K601E Mutation-Positive Lung Adenocarcinoma: A Case Report.',NULL,'2'),(12754,NULL,'Caterino M et al','doi: 10.3390/medicina58050666','5','Medicina (Kaunas, Lithuania)',NULL,'','35630083','2022 May 17','Sorafenib in Metastatic Papillary Thyroid Carcinoma with BRAF K601E Mutation on Liquid Biopsy: A Case Report and Literature Review.',NULL,'58'),(12755,NULL,'Munck JM et al','doi: 10.1158/1535-7163.MCT-20-0909','10','Molecular cancer therapeutics',NULL,'1757-1768','34330842','2021 Oct','ASTX029, a Novel Dual-mechanism ERK Inhibitor, Modulates Both the Phosphorylation and Catalytic Activity of ERK.',NULL,'20'),(12756,NULL,'Gray-Schopfer VC et al','','1','Cancer research',NULL,'122-9','17210691','2007 Jan 1','Tumor necrosis factor-alpha blocks apoptosis in melanoma cells when BRAF signaling is inhibited.',NULL,'67'),(12757,NULL,'Girotti MR et al','doi: 10.1016/j.ccell.2014.11.006','1','Cancer cell',NULL,'85-96','25500121','2015 Jan 12','Paradox-breaking RAF inhibitors that also target SRC are effective in drug-resistant BRAF mutant melanoma.',NULL,'27'),(12758,NULL,'Drobysheva A et al','doi: 10.6004/jnccn.2017.0139','8','Journal of the National Comprehensive Cancer Network : JNCCN',NULL,'978-982','28784858','2017 Aug','Targeted MAPK Pathway Inhibitors in Patients With Disseminated Pilocytic Astrocytomas.',NULL,'15'),(12759,NULL,'Comito F et al','doi: 10.1097/CMR.0000000000000854','6','Melanoma research',NULL,'477-484','36039514','2022 Dec 1','Clinical characteristics and treatment outcomes of non-V600 E/K BRAF mutant melanoma patients: a single-institution experience.',NULL,'32'),(12760,NULL,'Casadevall D et al','doi: 10.1186/s13256-016-0953-0','1','Journal of medical case reports',NULL,'158','27255157','2016 Jun 2','Dabrafenib in an elderly patient with metastatic melanoma and BRAF V600R mutation: a case report.',NULL,'10'),(12761,NULL,'Bahadoran P et al','doi: 10.1200/JCO.2012.46.1061','19','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'e324-6','23715574','2013 Jul 1','Major clinical response to a BRAF inhibitor in a patient with a BRAF L597R-mutated melanoma.',NULL,'31'),(12762,NULL,'Dai X et al','doi: 10.1016/j.celrep.2021.110250','3','Cell reports',NULL,'110250','35045286','2022 Jan 18','Acetylation-dependent regulation of BRAF oncogenic function.',NULL,'38'),(12763,NULL,'Lehtomaki KI et al','doi: 10.21873/anticanres.13791','11','Anticancer research',NULL,'5867-5877','31704811','2019 Nov','Clonal Evolution of MEK/MAPK Pathway Activating Mutations in a Metastatic Colorectal Cancer Case.',NULL,'39'),(12764,NULL,'Nebhan CA et al','doi: 10.1002/onco.13795','9','The oncologist',NULL,'731-e1498','33861486','2021 Sep','Efficacy and Safety of Trametinib in Non-V600 BRAF Mutant Melanoma: A Phase II Study.',NULL,'26'),(12765,NULL,'Shimada Y et al','doi: 10.1016/j.jss.2018.06.020','','The Journal of surgical research',NULL,'72-81','30463788','2018 Dec','Clinical Significance of BRAF Non-V600E Mutations in Colorectal Cancer: A Retrospective Study of Two Institutions.',NULL,'232'),(12766,NULL,'Awada G et al','doi: 10.1097/CCO.0000000000000817','2','Current opinion in oncology',NULL,'115-122','35050937','2022 Mar 1','Melanoma with genetic alterations beyond the BRAFV600 mutation: management and new insights.',NULL,'34'),(12767,NULL,'Osumi H et al','doi: 10.1038/s41598-020-74530-x','1','Scientific reports',NULL,'17455','33060766','2020 Oct 15','Evaluation of the RAS signaling network in response to MEK inhibition using organoids derived from a familial adenomatous polyposis patient.',NULL,'10'),(12768,NULL,'Johnson FM et al','doi: 10.1200/JCO.2010.30.5474','30','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'4609-15','20855820','2010 Oct 20','Phase II study of dasatinib in patients with advanced non-small-cell lung cancer.',NULL,'28'),(12769,NULL,'Yap J et al','doi: 10.1126/sciadv.abg0390','24','Science advances',NULL,'','34108213','2021 Jun','The stability of R-spine defines RAF inhibitor resistance: A comprehensive analysis of oncogenic BRAF mutants with in-frame insertion of C-4 loop.',NULL,'7'),(12770,NULL,'Das TK et al','doi: 10.1016/j.isci.2021.102306','4','iScience',NULL,'102306','33855281','2021 Apr 23','Drosophila RASopathy models identify disease subtype differences and biomarkers of drug efficacy.',NULL,'24'),(12771,NULL,'Remagen W et al','','3','Skeletal radiology',NULL,'251-3','3010472','1986','Case report 359: gigantic benign fibrous histiocytoma (nonossifying fibroma).',NULL,'15'),(12772,NULL,'Dietlein F et al','doi: 10.1016/j.cell.2015.05.053','1','Cell',NULL,'146-59','26140595','2015 Jul 2','A Synergistic Interaction between Chk1- and MK2 Inhibitors in KRAS-Mutant Cancer.',NULL,'162'),(12773,NULL,'Gautschi O et al','doi: 10.1097/JTO.0000000000000625','10','Journal of thoracic oncology : official publication of the International Association for the Study of Lung Cancer',NULL,'1451-7','26200454','2015 Oct','Targeted Therapy for Patients with BRAF-Mutant Lung Cancer: Results from the European EURAF Cohort.',NULL,'10'),(12774,NULL,'miech M et al','doi: 10.3390/genes11111342','11','Genes',NULL,'','33198372','2020 Nov 12','Emerging BRAF Mutations in Cancer Progression and Their Possible Effects on Transcriptional Networks.',NULL,'11'),(12776,NULL,'Gallo S et al','doi: 10.1097/CMR.0000000000000417','2','Melanoma research',NULL,'143-146','29232305','2018 Apr','BRAF-inhibitors can exert control of disease in BRAF T599I mutated melanoma: a case report.',NULL,'28'),(12777,NULL,'Zhang S et al','doi: 10.3389/fphar.2022.1019217','','Frontiers in pharmacology',NULL,'1019217','36686670','2022','BRAF L485-P490 deletion mutant metastatic melanoma sensitive to BRAF and MEK inhibition: A case report and literature review.',NULL,'13'),(12778,NULL,'Shin JE et al','doi: 10.3389/fonc.2022.976450','','Frontiers in oncology',NULL,'976450','36505826','2022','Efficacy of dabrafenib/trametinib in pancreatic ductal adenocarcinoma with BRAF NVTAP deletion: A case report.',NULL,'12'),(12779,NULL,'Pham NA et al','doi: 10.1038/s41598-021-90049-1','1','Scientific reports',NULL,'10619','34011980','2021 May 19','Patient-derived tumor xenograft and organoid models established from resected pancreatic, duodenal and biliary cancers.',NULL,'11'),(12780,NULL,'Oberl M et al','doi: 10.1200/PO.21.00417','','JCO precision oncology',NULL,'e2100417','35319964','2022 Mar','Response to BRAF and MEK Inhibitors in BRAF Thr599dup-Mutated Melanoma.',NULL,'6'),(12781,NULL,'Allen A et al','doi: 10.1371/journal.pone.0217399','6','PloS one',NULL,'e0217399','31158244','2019','Rare BRAF mutations in pancreatic neuroendocrine tumors may predict response to RAF and MEK inhibition.',NULL,'14'),(12782,NULL,'McNulty SN et al','doi: 10.1038/s41598-021-99278-w','1','Scientific reports',NULL,'19999','34625582','2021 Oct 8','BRAF mutations may identify a clinically distinct subset of glioblastoma.',NULL,'11'),(12783,NULL,'Li J et al','doi: 10.1158/0008-5472.CAN-17-0938','2','Cancer research',NULL,'338-347','28972077','2018 Jan 15','Multiregional Sequencing Reveals Genomic Alterations and Clonal Dynamics in Primary Malignant Melanoma of the Esophagus.',NULL,'78'),(12799,NULL,'Dutton-Regester K et al','doi: 10.1111/pcmr.12153','6','Pigment cell & melanoma research',NULL,'852-60','23890154','2013 Nov','Melanomas of unknown primary have a mutation profile consistent with cutaneous sun-exposed melanoma.',NULL,'26'),(12855,NULL,'Garcia A et al','doi: 10.1002/cncy.22648','1','Cancer cytopathology',NULL,'50-57','36200799','2023 Jan','Assessment of BRAF V600E (VE1) immunochemistry for the detection of BRAF V600E mutation in non-small cell lung carcinoma cytology specimens.',NULL,'131'),(12856,NULL,'Lei L et al','doi: 10.21037/tcr-20-480','10','Translational cancer research',NULL,'6039-6049','35117215','2020 Oct','Association between BRAF mutant classification and the efficacy of pemetrexed-based chemotherapy in Chinese advanced non-small cell lung cancer patients: a multicenter retrospective study.',NULL,'9'),(12858,NULL,'Lin Q et al','doi: 10.1186/s12967-019-2036-7','1','Journal of translational medicine',NULL,'298','31470866','2019 Aug 30','The association between BRAF mutation class and clinical features in BRAF-mutant Chinese non-small cell lung cancer patients.',NULL,'17'),(12859,NULL,'Melin A et al','doi: 10.1016/j.ejca.2019.08.029','','European journal of cancer (Oxford, England : 1990)',NULL,'109-112','31569065','2019 Nov','BRAF exon 11 mutant melanoma and sensitivity to BRAF/MEK inhibition: Two case reports.',NULL,'121'),(12860,NULL,'Lin YF et al','doi: 10.3390/ph15020136','2','Pharmaceuticals (Basel, Switzerland)',NULL,'','35215249','2022 Jan 24','Predicting Anticancer Drug Resistance Mediated by Mutations.',NULL,'15'),(12940,NULL,'Degenhardt Y et al','doi: 10.1158/1535-7163.MCT-10-0095','7','Molecular cancer therapeutics',NULL,'2079-89','20571075','2010 Jul','Sensitivity of cancer cells to Plk1 inhibitor GSK461364A is associated with loss of p53 function and chromosome instability.',NULL,'9'),(12942,NULL,'Abida W et al','doi: 10.1200/PO.17.00029','','JCO precision oncology',NULL,'','28825054','2017 Jul','Prospective Genomic Profiling of Prostate Cancer Across Disease States Reveals Germline and Somatic Alterations That May Affect Clinical Decision Making.',NULL,'2017'),(12943,NULL,'Murciano-Goroff YR et al','doi: 10.1038/s41416-021-01679-1','6','British journal of cancer',NULL,'889-898','34963703','2022 Apr','Immune biomarkers and response to checkpoint inhibition of BRAF(V600) and BRAF non-V600 altered lung cancers.',NULL,'126'),(12944,NULL,'Rasool M et al','doi: 10.1016/j.sjbs.2021.06.048','10','Saudi journal of biological sciences',NULL,'5906-5912','34588906','2021 Oct','Mutational spectrum of BRAF gene in colorectal cancer patients in Saudi Arabia.',NULL,'28'),(12949,NULL,'Xin HY et al','doi: 10.1001/jamanetworkopen.2023.1476','3','JAMA network open',NULL,'e231476','36867406','2023 Mar 1','Association of BRAF Variants With Disease Characteristics, Prognosis, and Targeted Therapy Response in Intrahepatic Cholangiocarcinoma.',NULL,'6'),(12950,NULL,'Mao C et al','doi: 10.1371/journal.pone.0036653','5','PloS one',NULL,'e36653','22586484','2012','KRAS, BRAF and PIK3CA mutations and the loss of PTEN expression in Chinese patients with colorectal cancer.',NULL,'7'),(12951,NULL,'Kiel C et al','doi: 10.7554/eLife.12814','','eLife',NULL,'e12814','26744778','2016 Jan 8','The yin-yang of kinase activation and unfolding explains the peculiarity of Val600 in the activation segment of BRAF.',NULL,'5'),(12952,NULL,'Cho NY et al','','8','International journal of cancer',NULL,'1858-62','16721785','2006 Oct 15','BRAF and KRAS mutations in prostatic adenocarcinoma.',NULL,'119'),(12953,NULL,'Reyes R et al','doi: 10.1016/j.cllc.2019.02.022','3','Clinical lung cancer',NULL,'e219-e223','30926357','2019 May','Clinical Benefit From BRAF/MEK Inhibition in a Double Non-V600E BRAF Mutant Lung Adenocarcinoma: A Case Report.',NULL,'20'),(12954,NULL,'Kim HS et al','doi: 10.1038/onc.2016.486','23','Oncogene',NULL,'3334-3345','28092667','2017 Jun 8','Oncogenic BRAF fusions in mucosal melanomas activate the MAPK pathway and are sensitive to MEK/PI3K inhibition or MEK/CDK4/6 inhibition.',NULL,'36'),(12955,NULL,'Feng K et al','doi: 10.1186/s13578-022-00950-z','1','Cell & bioscience',NULL,'211','36585710','2022 Dec 30','SPOP inhibits BRAF-dependent tumorigenesis through promoting non-degradative ubiquitination of BRAF.',NULL,'12'),(12959,NULL,'Sase H et al','doi: 10.1158/1535-7163.MCT-17-1022','10','Molecular cancer therapeutics',NULL,'2217-2225','30045926','2018 Oct','Acquired JHDM1D-BRAF Fusion Confers Resistance to FGFR Inhibition in FGFR2-Amplified Gastric Cancer.',NULL,'17'),(12960,NULL,'Terrell EM et al','doi: 10.1016/j.molcel.2019.09.004','6','Molecular cell',NULL,'872-884.e5','31606273','2019 Dec 19','Distinct Binding Preferences between Ras and Raf Family Members and the Impact on Oncogenic Ras Signaling.',NULL,'76'),(12961,NULL,'Ashktorab H et al','doi: 10.18632/oncotarget.13977','5','Oncotarget',NULL,'7852-7866','28002797','2017 Jan 31','Targeted exome sequencing reveals distinct pathogenic variants in Iranians with colorectal cancer.',NULL,'8'),(12962,NULL,'Nomura M et al','doi: 10.1007/s00401-017-1771-1','6','Acta neuropathologica',NULL,'941-956','28852847','2017 Dec','Distinct molecular profile of diffuse cerebellar gliomas.',NULL,'134'),(12963,NULL,'Hudson AM et al','doi: 10.1126/scisignal.aan6776','526','Science signaling',NULL,'','29666306','2018 Apr 17','Truncation- and motif-based pan-cancer analysis reveals tumor-suppressing kinases.',NULL,'11'),(12964,NULL,'Quaio CR et al','doi: 10.6061/clinics/2013(08)03','8','Clinics (Sao Paulo, Brazil)',NULL,'1079-83','24037001','2013','Tegumentary manifestations of Noonan and Noonan-related syndromes.',NULL,'68'),(13042,NULL,'Shi XB et al','','1','The Prostate',NULL,'59-72','11920959','2002 Apr 1','Complex functions of mutant p53 alleles from human prostate cancer.',NULL,'51'),(13117,NULL,'Subbiah V et al','doi: 10.1038/s41591-023-02321-8','5','Nature medicine',NULL,'1103-1112','37059834','2023 May','Dabrafenib plus trametinib in BRAFV600E-mutated rare cancers: the phase 2 ROAR trial.',NULL,'29'),(13983,NULL,'Chung T et al','doi: 10.5009/gnl210174','4','Gut and liver',NULL,'613-624','34810298','2022 Jul 15','Genetic, Clinicopathological, and Radiological Features of Intrahepatic Cholangiocarcinoma with Ductal Plate Malformation Pattern.',NULL,'16'),(13984,NULL,'Id Said B et al','doi: 10.1002/humu.23025','9','Human mutation',NULL,'889-92','27297285','2016 Sep','Super-Transactivation TP53 Variant in the Germline of a Family with Li-Fraumeni Syndrome.',NULL,'37'),(13985,NULL,'Doffe F et al','doi: 10.1038/s41418-020-00672-0','5','Cell death and differentiation',NULL,'1477-1492','33257846','2021 May','Identification and functional characterization of new missense SNPs in the coding region of the TP53 gene.',NULL,'28'),(14001,NULL,'Sun H et al','doi: 10.1371/journal.pcbi.1003729','7','PLoS computational biology',NULL,'e1003729','25033171','2014 Jul','P-loop conformation governed crizotinib resistance in G2032R-mutated ROS1 tyrosine kinase: clues from free energy landscape.',NULL,'10'),(14002,NULL,'Gou W et al','doi: 10.1016/j.canlet.2018.02.032','','Cancer letters',NULL,'19-28','29477381','2018 May 28','CD74-ROS1 G2032R mutation transcriptionally up-regulates Twist1 in non-small cell lung cancer cells leading to increased migration, invasion, and resistance to crizotinib.',NULL,'422'),(14086,NULL,'Riely GJ et al','doi: 10.1200/JCO.23.00774','21','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'3700-3711','37270692','2023 Jul 20','Phase II, Open-Label Study of Encorafenib Plus Binimetinib in Patients With BRAF(V600)-Mutant Metastatic Non-Small-Cell Lung Cancer.',NULL,'41'),(14275,NULL,'O\'Neil SR et al','doi: 10.1200/PO.23.00174','','JCO precision oncology',NULL,'e2300174','37487149','2023 Jul','Exceptional Response to Crizotinib With Subsequent Response to Cabozantinib in Metastatic, ROS1-GOPC Fusion-Mutated Breast Cancer.',NULL,'7'),(14710,NULL,'Janku F et al','doi: 10.1158/1535-7163.MCT-18-1244','6','Molecular cancer therapeutics',NULL,'1149-1157','31015311','2019 Jun','Molecular Profiling of Tumor Tissue and Plasma Cell-Free DNA from Patients with Non-Langerhans Cell Histiocytosis.',NULL,'18'),(14711,NULL,'Dankner M et al','doi: 10.1200/PO.22.00107','','JCO precision oncology',NULL,'e2200107','35977349','2022 Aug','Clinical Activity of Mitogen-Activated Protein Kinase-Targeted Therapies in Patients With Non-V600 BRAF-Mutant Tumors.',NULL,'6'),(14822,NULL,'Kilburn LB et al','doi: 10.1038/s41591-023-02668-y','1','Nature medicine',NULL,'207-217','37978284','2024 Jan','The type II RAF inhibitor tovorafenib in relapsed/refractory pediatric low-grade glioma: the phase 2 FIREFLY-1 trial.',NULL,'30'),(15040,NULL,'Wang H et al','doi: 10.1038/s41392-023-01347-1','1','Signal transduction and targeted therapy',NULL,'92','36859359','2023 Mar 1','Targeting p53 pathways: mechanisms, structures, and advances in therapy.',NULL,'8'),(15041,NULL,'Aubrey BJ et al','doi: 10.1101/cshperspect.a026062','5','Cold Spring Harbor perspectives in medicine',NULL,'','27141080','2016 May 2','Tumor-Suppressor Functions of the TP53 Pathway.',NULL,'6'),(15042,NULL,'Alaseem AM','doi: 10.1016/j.jsps.2023.101790','10','Saudi pharmaceutical journal : SPJ : the official publication of the Saudi Pharmaceutical Society',NULL,'101790','37818252','2023 Oct','Advancements in MDM2 inhibition: Clinical and pre-clinical investigations of combination therapeutic regimens.',NULL,'31'),(15043,NULL,'Abuetabh Y et al','doi: 10.1038/s12276-022-00863-4','10','Experimental & molecular medicine',NULL,'1658-1669','36207426','2022 Oct','DNA damage response revisited: the p53 family and its regulators provide endless cancer therapy opportunities.',NULL,'54'),(15044,NULL,'Zerdoumi Y et al','doi: 10.1093/hmg/ddx106','14','Human molecular genetics',NULL,'2591-2602','28369373','2017 Jul 15','Germline TP53 mutations result into a constitutive defect of p53 DNA binding and transcriptional response to DNA damage.',NULL,'26'),(15075,NULL,'Bouffet E et al','doi: 10.1056/NEJMoa2303815','12','The New England journal of medicine',NULL,'1108-1120','37733309','2023 Sep 21','Dabrafenib plus Trametinib in Pediatric Glioma with BRAF V600 Mutations.',NULL,'389'),(15214,NULL,'Zhou G et al','doi: 10.1016/j.molcel.2014.04.024','6','Molecular cell',NULL,'960-974','24857548','2014 Jun 19','Gain-of-function mutant p53 promotes cell growth and cancer cell metabolism via inhibition of AMPK activation.',NULL,'54'),(15215,NULL,'Xu J et al','doi: 10.1038/cddis.2014.75','3','Cell death & disease',NULL,'e1108','24603336','2014 Mar 6','Unequal prognostic potentials of p53 gain-of-function mutations in human cancers associate with drug-metabolizing activity.',NULL,'5'),(15216,NULL,'Feng R et al','doi: 10.1016/j.canlet.2023.216154','','Cancer letters',NULL,'216154','37030635','2023 May 28','Mutant p53 activates hnRNPA2B1-AGAP1-mediated exosome formation to promote esophageal squamous cell carcinoma progression.',NULL,'562'),(15217,NULL,'Nguyen SA et al','doi: 10.1158/1078-0432.CCR-13-1856','18','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'4894-903','25078279','2014 Sep 15','Novel MSH6 mutations in treatment-nave glioblastoma and anaplastic oligodendroglioma contribute to temozolomide resistance independently of MGMT promoter methylation.',NULL,'20'),(15218,NULL,'Neilsen PM et al','doi: 10.1038/onc.2012.305','24','Oncogene',NULL,'2992-3000','22797073','2013 Jun 13','Mutant p53 drives invasion in breast tumors through up-regulation of miR-155.',NULL,'32'),(15219,NULL,'Boettcher S et al','doi: 10.1126/science.aax3649','6453','Science (New York, N.Y.)',NULL,'599-604','31395785','2019 Aug 9','A dominant-negative effect drives selection of TP53 missense mutations in myeloid malignancies.',NULL,'365'),(15220,NULL,'Chen Z et al','doi: 10.1016/j.jamcollsurg.2011.02.024','6','Journal of the American College of Surgeons',NULL,'1008-1017.e1','21458303','2011 Jun','Molecular diagnosis of response to neoadjuvant chemoradiation therapy in patients with locally advanced rectal cancer.',NULL,'212'),(15255,'Bouffet et al. Abstract# LGG-49, Neuro-Oncology 2020.',NULL,NULL,NULL,NULL,'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7715318',NULL,NULL,NULL,NULL,NULL,NULL),(15256,'Bouffet et al. Abstract# LBA2002, ASCO 2022.',NULL,NULL,NULL,NULL,'https://ascopubs.org/doi/abs/10.1200/JCO.2022.40.17_suppl.LBA2002',NULL,NULL,NULL,NULL,NULL,NULL),(15257,'Janku et al. Abstract# B176, AACR-NCI-EORTC 2017.',NULL,NULL,NULL,NULL,'http://mct.aacrjournals.org/content/17/1_Supplement/B176',NULL,NULL,NULL,NULL,NULL,NULL),(15258,'Fuente et al. Abstract# 3006, J. Clin. Oncol. 2023, Vol 41, Suppl 16.',NULL,NULL,NULL,NULL,'https://ascopubs.org/doi/10.1200/JCO.2023.41.16_suppl.3006',NULL,NULL,NULL,NULL,NULL,NULL),(15259,'Wolf et al. Abstract# 1387p, Annals of Oncology 2020.',NULL,NULL,NULL,NULL,'https://www.annalsofoncology.org/article/S0923-7534%2820%2941697-7/fulltext',NULL,NULL,NULL,NULL,NULL,NULL),(15260,'Citarella et al. Precis Cancer Med 2022.',NULL,NULL,NULL,NULL,'https://pcm.amegroups.com/article/view/6792/html#B19',NULL,NULL,NULL,NULL,NULL,NULL),(15261,'Severson et al. Abstract #40P, Annals of Oncology 2022.',NULL,NULL,NULL,NULL,'https://www.annalsofoncology.org/article/S0923-7534%2822%2900058-8/fulltext',NULL,NULL,NULL,NULL,NULL,NULL),(15262,'Rose et al. Abstract# 531P, Annals of Oncology 2021.',NULL,NULL,NULL,NULL,'https://www.annalsofoncology.org/article/S0923-7534%2821%2903282-8/fulltext',NULL,NULL,NULL,NULL,NULL,NULL),(15449,'Drilon et al. Abstract# CT060, AACR 2017.',NULL,NULL,NULL,NULL,'https://cancerres.aacrjournals.org/content/77/13_Supplement/CT060)(Abstract',NULL,NULL,NULL,NULL,NULL,NULL),(15450,'# 64P, ESMO 2019.',NULL,NULL,NULL,NULL,'https://academic.oup.com/annonc/article/30/Supplement_9/mdz420.003/5638397',NULL,NULL,NULL,NULL,NULL,NULL),(15451,'Cho et al. Abstract# OA03.06, J. of Thoracic Oncol. Nov. 2023.',NULL,NULL,NULL,NULL,'https://www.jto.org/article/S1556-0864%2823%2900837-7/fulltext',NULL,NULL,NULL,NULL,NULL,NULL),(15452,'Hegde et al. JCO PO, 2018.',NULL,NULL,NULL,NULL,'https://ascopubs.org/doi/full/10.1200/PO.18.00267',NULL,NULL,NULL,NULL,NULL,NULL),(15453,'Dimou et al. JCO PO, 2019.',NULL,NULL,NULL,NULL,'https://ascopubs.org/doi/full/10.1200/PO.19.00013',NULL,NULL,NULL,NULL,NULL,NULL),(15454,'Doebele et al. Abstract# LBA-28, ESMO 2019.',NULL,NULL,NULL,NULL,'https://www.sciencedirect.com/science/article/pii/S0923753419603777',NULL,NULL,NULL,NULL,NULL,NULL),(15465,'Dumble et al. Abstract# LB006, AACR 2021.',NULL,NULL,NULL,NULL,'https://aacrjournals.org/cancerres/article/81/13_Supplement/LB006/669897/Abstract-LB006-PC14586-The-first-orally',NULL,NULL,NULL,NULL,NULL,NULL),(15466,'Hofste op Bruinink, D. et al. Abstract# Blood 128:3271, 2017.',NULL,NULL,NULL,NULL,'http://www.bloodjournal.org/content/128/22/3271?sso-checked=true',NULL,NULL,NULL,NULL,NULL,NULL),(15521,NULL,'Tanaka R et al','doi: 10.1200/PO.24.00055','','JCO precision oncology',NULL,'e2400055','38781546','2024 May','Dabrafenib for Pilocytic Astrocytoma With BRAF V599ins.',NULL,'8'),(15676,NULL,'Katayama R et al','doi: 10.1038/s41467-019-11496-z','1','Nature communications',NULL,'3604','31399568','2019 Aug 9','The new-generation selective ROS1/NTRK inhibitor DS-6051b overcomes crizotinib resistant ROS1-G2032R mutation in preclinical models.',NULL,'10'),(15677,NULL,'Li W et al','doi: 10.1200/JCO.24.00731','22','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'2660-2670','38822758','2024 Aug 1','Efficacy and Safety of Taletrectinib in Chinese Patients With ROS1+ Non-Small Cell Lung Cancer: The Phase II TRUST-I Study.',NULL,'42'),(15679,NULL,'Xia ZJ et al','doi: 10.1038/s41401-020-00513-3','6','Acta pharmacologica Sinica',NULL,'998-1004','32918045','2021 Jun','SAF-189s, a potent new-generation ROS1 inhibitor, is active against crizotinib-resistant ROS1 mutant-driven tumors.',NULL,'42'),(15680,NULL,'Lin JJ et al','doi: 10.1158/1078-0432.CCR-21-0032','10','Clinical cancer research : an official journal of the American Association for Cancer Research',NULL,'2899-2909','33685866','2021 May 15','Spectrum of Mechanisms of Resistance to Crizotinib and Lorlatinib in ROS1 Fusion-Positive Lung Cancer.',NULL,'27'),(15681,NULL,'Gainor JF et al','doi: 10.1200/PO.17.00063','','JCO precision oncology',NULL,'','29333528','2017','Patterns of Metastatic Spread and Mechanisms of Resistance to Crizotinib in ROS1-Positive Non-Small-Cell Lung Cancer.',NULL,'2017'),(15682,NULL,'Jri B et al','doi: 10.3390/curroncol29090520','9','Current oncology (Toronto, Ont.)',NULL,'6628-6634','36135089','2022 Sep 16','Acquired G2032R Resistance Mutation in ROS1 to Lorlatinib Therapy Detected with Liquid Biopsy.',NULL,'29'),(15683,NULL,'Weber U et al','doi: 10.1016/j.jtocrr.2024.100673','7','JTO clinical and research reports',NULL,'100673','39091594','2024 Jul','L2086F Mutant ROS1-Rearranged NSCLC Resistant to Repotrectinib Responds to Cabozantinib: A Case Report.',NULL,'5'),(15684,NULL,'Thawani R et al','doi: 10.1038/s41698-024-00663-1','1','NPJ precision oncology',NULL,'175','39117775','2024 Aug 8','TKI type switching overcomes ROS1 L2086F in ROS1 fusion-positive cancers.',NULL,'8'),(15685,NULL,'Sakamoto M et al','doi: 10.1038/s41698-023-00381-0','1','NPJ precision oncology',NULL,'56','37291202','2023 Jun 8','Exceptional response to lorlatinib and cabozantinib in ROS1-rearranged NSCLC with acquired F2004V and L2086F resistance.',NULL,'7'),(15851,NULL,'Kopetz S et al','doi: 10.1038/s41591-024-03443-3','','Nature medicine',NULL,'','39863775','2025 Jan 25','Encorafenib, cetuximab and chemotherapy in BRAF-mutant colorectal cancer: a randomized phase 3 trial.',NULL,''),(15924,NULL,'Awada G et al','doi: 10.1182/bloodadvances.2018021782','16','Blood advances',NULL,'2156-2158','30154124','2018 Aug 28','Long-term disease control of Langerhans cell histiocytosis using combined BRAF and MEK inhibition.',NULL,'2'),(15925,NULL,'Hazim AZ et al','doi: 10.1002/onco.13541','12','The oncologist',NULL,'1001-1004','32985015','2020 Dec','Efficacy of BRAF-Inhibitor Therapy in BRAF(V600E) -Mutated Adult Langerhans Cell Histiocytosis.',NULL,'25'),(16000,NULL,'Kim HS et al','doi: 10.1038/s41587-023-01949-8','8','Nature biotechnology',NULL,'1254-1262','37697151','2024 Aug','Direct measurement of engineered cancer mutations and their transcriptional phenotypes in single cells.',NULL,'42'),(16001,'Dumbrava et al. Abstract# 3003, ASCO 2022.',NULL,NULL,NULL,NULL,'https://ascopubs.org/doi/abs/10.1200/JCO.2022.40.16_suppl.3003',NULL,NULL,NULL,NULL,NULL,NULL),(16002,'Schram et al. Abstract# LB_A25, AACR 2023.',NULL,NULL,NULL,NULL,'https://aacrjournals.org/mct/article/22/12_Supplement/LB_A25/730549/Abstract-LB-A25-Updated-Phase-1-results-from-the',NULL,NULL,NULL,NULL,NULL,NULL),(16003,'Schram et al. Abstract# LBA_26, SGO 2024.',NULL,NULL,NULL,NULL,'https://www.gynecologiconcology-online.net/article/S0090-8258(24)00453-0/fulltext',NULL,NULL,NULL,NULL,NULL,NULL),(16004,NULL,'Osakabe M et al','','6','Pathology international',NULL,'322-7','17539962','2007 Jun','Characteristics of vulvar squamous cell carcinoma in Japanese women.',NULL,'57'),(16005,NULL,'Lim B et al','','26','Oncotarget',NULL,'22179-90','26109429','2015 Sep 8','Genome-wide mutation profiles of colorectal tumors and associated liver metastases at the exome and transcriptome levels.',NULL,'6'),(16240,NULL,'Prol M et al','doi: 10.1200/JCO-25-00275','16','Journal of clinical oncology : official journal of the American Society of Clinical Oncology',NULL,'1920-1929','40179330','2025 Jun','Taletrectinib in ROS1+ Non-Small Cell Lung Cancer: TRUST.',NULL,'43'),(16273,NULL,'Fortuno C et al','doi: 10.1002/humu.23739','6','Human mutation',NULL,'788-800','30840781','2019 Jun','A quantitative model to predict pathogenicity of missense variants in the TP53 gene.',NULL,'40');
/*!40000 ALTER TABLE `article` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `cancer_type`
--

DROP TABLE IF EXISTS `cancer_type`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `cancer_type` (
  `id` int NOT NULL AUTO_INCREMENT,
  `code` varchar(255) NOT NULL,
  `color` varchar(255) NOT NULL,
  `level` int NOT NULL,
  `main_type` varchar(255) NOT NULL,
  `subtype` varchar(255) NOT NULL,
  `tissue` varchar(255) NOT NULL,
  `tumor_form` varchar(255) DEFAULT NULL,
  `parent` int DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `FKAC902999B4A9E9F8` (`parent`),
  CONSTRAINT `FKAC902999B4A9E9F8` FOREIGN KEY (`parent`) REFERENCES `cancer_type` (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=989 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `cancer_type`
--

LOCK TABLES `cancer_type` WRITE;
/*!40000 ALTER TABLE `cancer_type` DISABLE KEYS */;
INSERT INTO `cancer_type` VALUES (1,'MMB','Gray',3,'Embryonal Tumor','Medullomyoblastoma','CNS/Brain','SOLID',347),(2,'GCB','LimeGreen',6,'Mature B-Cell Neoplasms','Germinal Center B-Cell Type','Lymphoid','LIQUID',490),(3,'SBLU','LimeGreen',5,'Mature B-Cell Neoplasms','Splenic B-Cell Lymphoma/Leukemia, Unclassifiable','Lymphoid','LIQUID',530),(4,'OHNCA','DarkRed',2,'Head and Neck Cancer','Head and Neck Carcinoma, Other','Head and Neck','SOLID',303),(5,'PAOS','White',3,'Bone Cancer','Parosteal Osteosarcoma','Bone','SOLID',631),(6,'TMDS','LightSalmon',5,'Leukemia','Therapy-Related Myelodysplastic Syndrome','Myeloid','LIQUID',418),(7,'ARMS','LightYellow',3,'Soft Tissue Sarcoma','Alveolar Rhabdomyosarcoma','Soft Tissue','SOLID',211),(8,'SCST','LightBlue',2,'Sex Cord Stromal Tumor','Sex Cord Stromal Tumor','Ovary/Fallopian Tube','SOLID',557),(9,'ITLPDGI','LimeGreen',5,'Mature T and NK Neoplasms','Indolent T-Cell Lymphoproliferative Disorder of the GI Tract','Lymphoid','LIQUID',243),(10,'MBC','HotPink',2,'Breast Cancer','Metaplastic Breast Cancer','Breast','SOLID',799),(11,'AWDNET','SaddleBrown',3,'Gastrointestinal Neuroendocrine Tumor','Well-Differentiated Neuroendocrine Tumor of the Appendix','Bowel','SOLID',399),(12,'AMLCBFBMYH11','LightSalmon',5,'Leukemia','AML with inv(16)(p13.1q22) or t(16;16)(p13.1;q22);CBFB-MYH11','Myeloid','LIQUID',198),(13,'ROCY','Orange',4,'Renal Cell Carcinoma','Renal Oncocytoma','Kidney','SOLID',380),(14,'LAM','Gainsboro',2,'Non-Small Cell Lung Cancer','Pulmonary Lymphangiomyomatosis','Lung','SOLID',475),(15,'SBL','DarkRed',2,'Sialoblastoma','Sialoblastoma','Head and Neck','SOLID',303),(16,'AWM','LightSalmon',5,'Leukemia','AML without Maturation','Myeloid','LIQUID',601),(17,'AMLMRC','LightSalmon',4,'Leukemia','AML with Myelodysplasia-Related Changes','Myeloid','LIQUID',527),(18,'CABC','Teal',2,'Cervical Cancer','Cervical Adenoid Basal Carcinoma','Cervix','SOLID',80),(19,'PCGP','Gray',3,'Sellar Tumor','Craniopharyngioma, Papillary Type','CNS/Brain','SOLID',26),(20,'SCCO','LightBlue',3,'Ovarian Cancer','Small Cell Carcinoma of the Ovary','Ovary/Fallopian Tube','SOLID',168),(21,'LGT','Green',2,'Lacrimal Gland Tumor','Lacrimal Gland Tumor','Eye','SOLID',280),(22,'GBM','Gray',4,'Glioma','Glioblastoma Multiforme','CNS/Brain','SOLID',666),(23,'BL','LimeGreen',5,'Mature B-Cell Neoplasms','Burkitt Lymphoma','Lymphoid','LIQUID',530),(24,'MFS','LightYellow',2,'Soft Tissue Sarcoma','Myxofibrosarcoma','Soft Tissue','SOLID',683),(25,'AN','Black',2,'Skin Cancer, Non-Melanoma','Atypical Nevus','Skin','SOLID',543),(26,'SELT','Gray',2,'Sellar Tumor','Sellar Tumor','CNS/Brain','SOLID',644),(27,'AMOL','LightSalmon',5,'Leukemia','Acute Monoblastic/Monocytic Leukemia','Myeloid','LIQUID',601),(28,'PERL','LightSalmon',5,'Leukemia','Pure Erythroid Leukemia','Myeloid','LIQUID',601),(29,'AMLCEBPA','LightSalmon',5,'Leukemia','AML with Biallelic Mutations of CEBPA','Myeloid','LIQUID',198),(30,'UASC','PeachPuff',3,'Endometrial Cancer','Uterine Adenosquamous Carcinoma','Uterus','SOLID',687),(31,'LNM','LimeGreen',2,'Lymphatic Cancer, NOS','Lymphoid Neoplasm','Lymphoid','LIQUID',258),(32,'PXA','Gray',3,'Glioma','Pleomorphic Xanthoastrocytoma','CNS/Brain','SOLID',54),(33,'PMBL','LimeGreen',5,'Mature B-Cell Neoplasms','Primary Mediastinal (Thymic) Large B-Cell Lymphoma','Lymphoid','LIQUID',530),(34,'BIALCL','LimeGreen',6,'Mature T and NK Neoplasms','Breast Implant-Associated Anaplastic Large-Cell Lymphoma','Lymphoid','LIQUID',731),(35,'SRAP','SaddleBrown',3,'Appendiceal Cancer','Signet Ring Cell Type of the Appendix','Bowel','SOLID',566),(36,'MIXED','Black',3,'Cancer of Unknown Primary','Mixed Cancer Types','Other','SOLID',712),(37,'LXSC','DarkRed',3,'Head and Neck Cancer','Larynx Squamous Cell Carcinoma','Head and Neck','SOLID',44),(38,'ADRENAL_GLAND','Purple',1,'Adrenal Gland Cancer, NOS','Adrenal Gland','Adrenal Gland','SOLID',491),(39,'WM','LimeGreen',6,'Mature B-Cell Neoplasms','Waldenstrom Macroglobulinemia','Lymphoid','LIQUID',534),(40,'RHM','Gray',3,'CNS Cancer','Rhabdoid Meningioma','CNS/Brain','SOLID',750),(41,'WDLS','LightYellow',3,'Soft Tissue Sarcoma','Well-Differentiated Liposarcoma','Soft Tissue','SOLID',313),(42,'PEMESO','Green',2,'Mesothelioma','Peritoneal Mesothelioma','Peritoneum','SOLID',129),(43,'CNL','LightSalmon',4,'Myeloproliferative Neoplasms','Chronic Neutrophilic Leukemia','Myeloid','LIQUID',185),(44,'HNSC','DarkRed',2,'Head and Neck Cancer','Head and Neck Squamous Cell Carcinoma','Head and Neck','SOLID',303),(45,'BLLRGA','LimeGreen',4,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with Recurrent Genetic Abnormalities','Lymphoid','LIQUID',95),(46,'DESM','Black',3,'Melanoma','Desmoplastic Melanoma','Skin','SOLID',453),(47,'OEC','LightBlue',3,'Germ Cell Tumor','Embryonal Carcinoma','Ovary/Fallopian Tube','SOLID',84),(48,'SCLC','Gainsboro',3,'Small Cell Lung Cancer','Small Cell Lung Cancer','Lung','SOLID',532),(49,'EMPSGC','Black',2,'Skin Cancer, Non-Melanoma','Endocrine Mucin Producing Sweat Gland Carcinoma','Skin','SOLID',543),(50,'ACBC','HotPink',3,'Breast Cancer','Adenoid Cystic Breast Cancer','Breast','SOLID',338),(51,'OPHSC','DarkRed',3,'Head and Neck Cancer','Oropharynx Squamous Cell Carcinoma','Head and Neck','SOLID',44),(52,'CERMS','Teal',2,'Cervical Cancer','Cervical Rhabdomyosarcoma','Cervix','SOLID',80),(53,'GCEMU','Teal',4,'Cervical Cancer','Gastric Type Mucinous Carcinoma','Cervix','SOLID',830),(54,'ENCG','Gray',2,'Glioma','Encapsulated Glioma','CNS/Brain','SOLID',644),(55,'PTLD','LimeGreen',3,'Posttransplant Lymphoproliferative Disorders','Posttransplant Lymphoproliferative Disorders','Lymphoid','LIQUID',31),(56,'SUBE','Gray',3,'CNS Cancer','Subependymoma','CNS/Brain','SOLID',832),(57,'SCGBC','Green',3,'Hepatobiliary Cancer','Small Cell Gallbladder Carcinoma','Biliary Tract','SOLID',553),(58,'FIOS','White',3,'Bone Cancer','Fibroblastic Osteosarcoma','Bone','SOLID',631),(59,'HGSOC','LightBlue',4,'Ovarian Cancer','High-Grade Serous Ovarian Cancer','Ovary/Fallopian Tube','SOLID',547),(60,'THHC','Teal',2,'Thyroid Cancer','Hurthle Cell Thyroid Cancer','Thyroid','SOLID',648),(61,'LUNE','Gainsboro',3,'Non-Small Cell Lung Cancer','Large Cell Neuroendocrine Carcinoma','Lung','SOLID',532),(62,'MGUSIGG','LimeGreen',6,'Mature B-Cell Neoplasms','IgG','Lymphoid','LIQUID',392),(63,'UPDC','PeachPuff',3,'Endometrial Cancer','Poorly Differentiated Carcinoma of the Uterus','Uterus','SOLID',687),(64,'OGBL','LightBlue',3,'Sex Cord Stromal Tumor','Gonadoblastoma','Ovary/Fallopian Tube','SOLID',8),(65,'CCPRC','Orange',4,'Renal Cell Carcinoma','Clear Cell Papillary Renal Cell Carcinoma','Kidney','SOLID',380),(66,'PSC','Purple',3,'Pancreatic Cancer','Serous Cystadenoma of the Pancreas','Pancreas','SOLID',464),(67,'RSCC','Orange',4,'Renal Cell Carcinoma','Renal Small Cell Carcinoma','Kidney','SOLID',380),(68,'BCCA','Gray',3,'Germ Cell Tumor','Choriocarcinoma','CNS/Brain','SOLID',238),(69,'SRCBC','Yellow',2,'Bladder Cancer','Plasmacytoid/Signet Ring Cell Bladder Carcinoma','Bladder/Urinary Tract','SOLID',108),(70,'PEL','LimeGreen',5,'Mature B-Cell Neoplasms','Primary Effusion Lymphoma','Lymphoid','LIQUID',530),(71,'CCS','LightYellow',2,'Soft Tissue Sarcoma','Clear Cell Sarcoma','Soft Tissue','SOLID',683),(72,'UPA','Yellow',2,'Bladder Cancer','Urothelial Papilloma','Bladder/Urinary Tract','SOLID',108),(73,'MFH','LightYellow',2,'Soft Tissue Sarcoma','Undifferentiated Pleomorphic Sarcoma/Malignant Fibrous Histiocytoma/High-Grade Spindle Cell Sarcoma','Soft Tissue','SOLID',683),(74,'FIBS','LightYellow',2,'Soft Tissue Sarcoma','Fibrosarcoma','Soft Tissue','SOLID',683),(75,'COM','HotPink',4,'Breast Cancer','Carcinoma with Osseous Metaplasia','Breast','SOLID',776),(76,'BMGCT','Gray',3,'Germ Cell Tumor','Mixed Germ Cell Tumor','CNS/Brain','SOLID',238),(77,'PVMF','LightSalmon',5,'Myeloproliferative Neoplasms','Polycythaemia Vera Myelofibrosis','Myeloid','LIQUID',806),(78,'SCHW','Gray',3,'Nerve Sheath Tumor','Schwannoma','Peripheral Nervous System','SOLID',242),(79,'BTBEOV','LightBlue',4,'Ovarian Cancer','Brenner Tumor, Benign','Ovary/Fallopian Tube','SOLID',602),(80,'CERVIX','Teal',1,'Cervical Cancer, NOS','Cervix','Cervix','SOLID',491),(81,'SIC','SaddleBrown',2,'Small Bowel Cancer','Small Intestinal Carcinoma','Bowel','SOLID',289),(82,'NSCHL','LimeGreen',5,'Hodgkin Lymphoma','Nodular Sclerosis Classical Hodgkin Lymphoma','Lymphoid','LIQUID',500),(83,'NSCLC','Gainsboro',2,'Non-Small Cell Lung Cancer','Non-Small Cell Lung Cancer','Lung','SOLID',475),(84,'OGCT','LightBlue',2,'Germ Cell Tumor','Ovarian Germ Cell Tumor','Ovary/Fallopian Tube','SOLID',557),(85,'CEVG','Teal',3,'Cervical Cancer','Villoglandular Carcinoma','Cervix','SOLID',159),(86,'MAC','Black',2,'Skin Cancer, Non-Melanoma','Microcystic Adnexal Carcinoma','Skin','SOLID',543),(87,'VDYS','Purple',3,'Germ Cell Tumor','Dysgerminoma','Vulva/Vagina','SOLID',383),(88,'PHCH','Green',3,'Hepatobiliary Cancer','Perihilar Cholangiocarcinoma','Biliary Tract','SOLID',283),(89,'BPLL','LimeGreen',5,'Mature B-Cell Neoplasms','B-Cell Prolymphocytic Leukemia','Lymphoid','LIQUID',530),(90,'GMN','Gray',3,'Germ Cell Tumor','Germinoma','CNS/Brain','SOLID',238),(91,'GCTSTM','Red',3,'Germ Cell Tumor','Germ Cell Tumor with Somatic-Type Malignancy','Testis','SOLID',169),(92,'BCAC','DarkRed',3,'Salivary Gland Cancer','Basal Cell Adenocarcinoma','Head and Neck','SOLID',522),(93,'OM','Green',2,'Melanoma','Ocular Melanoma','Eye','SOLID',280),(94,'BPSCC','Blue',3,'Penile Cancer','Basaloid Penile Squamous Cell Carcinoma','Penis','SOLID',481),(95,'BLL','LimeGreen',3,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma','Lymphoid','LIQUID',31),(96,'PCNSMT','Gray',2,'Primary CNS Melanocytic Tumors','Primary CNS Melanocytic Tumors','CNS/Brain','SOLID',644),(97,'MLNFGFR1','LightSalmon',4,'Leukemia','Myeloid/Lymphoid Neoplasms with FGFR1 Rearrangement','Myeloid','LIQUID',693),(98,'OSMBT','LightBlue',3,'Ovarian Cancer','Ovarian Seromucinous Borderline Tumor','Ovary/Fallopian Tube','SOLID',168),(99,'MDSID5Q','LightSalmon',4,'Myelodysplastic Syndromes','MDS with Isolated Del(5q)','Myeloid','LIQUID',355),(100,'PHPTLD','LimeGreen',4,'Posttransplant Lymphoproliferative Disorders','Plasmacytic Hyperplasia PTLD','Lymphoid','LIQUID',55),(101,'ASPS','LightYellow',2,'Soft Tissue Sarcoma','Alveolar Soft Part Sarcoma','Soft Tissue','SOLID',683),(102,'NETNOS','Black',3,'Cancer of Unknown Primary','Neuroendocrine Tumor, NOS','Other','SOLID',712),(103,'AMPULLA_OF_VATER','Purple',1,'Ampullary Carcinoma, NOS','Ampulla of Vater','Ampulla of Vater','SOLID',491),(104,'MPALBCRABL1','LightSalmon',4,'Leukemia','Mixed Phenotype Acute Leukemia with t(9;22)(q34.1;q11.2); BCR-ABL1','Myeloid','LIQUID',734),(105,'MCBCL','LimeGreen',5,'Mature B-Cell Neoplasms','Monoclonal B-Cell Lymphocytosis','Lymphoid','LIQUID',530),(106,'ALKLBCL','LimeGreen',5,'Mature B-Cell Neoplasms','ALK Positive Large B-Cell Lymphoma','Lymphoid','LIQUID',530),(107,'EATL','LimeGreen',5,'Mature T and NK Neoplasms','Enteropathy-Associated T-Cell Lymphoma','Lymphoid','LIQUID',243),(108,'BLADDER','Yellow',1,'Bladder/Urinary Tract Cancer, NOS','Bladder/Urinary Tract','Bladder/Urinary Tract','SOLID',491),(109,'TCCA','Red',3,'Germ Cell Tumor','Choriocarcinoma','Testis','SOLID',169),(110,'MMBL','Gray',3,'Embryonal Tumor','Melanotic Medulloblastoma','CNS/Brain','SOLID',347),(111,'SWDNET','LightSkyBlue',3,'Gastrointestinal Neuroendocrine Tumor','Well-Differentiated Neuroendocrine Tumors of the Stomach','Esophagus/Stomach','SOLID',331),(112,'MUCC','DarkRed',3,'Salivary Gland Cancer','Mucoepidermoid Carcinoma','Head and Neck','SOLID',522),(113,'USTUMP','PeachPuff',4,'Uterine Sarcoma','Uterine Smooth Muscle Tumor of Uncertain Malignant Potential','Uterus','SOLID',754),(114,'USARC','PeachPuff',2,'Uterine Sarcoma','Uterine Sarcoma/Mesenchymal','Uterus','SOLID',629),(115,'OCS','LightBlue',3,'Ovarian Cancer','Ovarian Carcinosarcoma/Malignant Mixed Mesodermal Tumor','Ovary/Fallopian Tube','SOLID',168),(116,'CMPT','Gainsboro',3,'Non-Small Cell Lung Cancer','Ciliated Muconodular Papillary Tumor of the Lung','Lung','SOLID',83),(117,'DSRCT','LightYellow',2,'Soft Tissue Sarcoma','Desmoplastic Small-Round-Cell Tumor','Soft Tissue','SOLID',683),(118,'ALUCA','Gainsboro',3,'Non-Small Cell Lung Cancer','Atypical Lung Carcinoid','Lung','SOLID',532),(119,'SPTCL','LimeGreen',5,'Mature T and NK Neoplasms','Subcutaneous Panniculitis-Like T-Cell Lymphoma','Lymphoid','LIQUID',243),(120,'SPC','HotPink',3,'Breast Cancer','Solid Papillary Carcinoma of the Breast','Breast','SOLID',338),(121,'EBOV','LightBlue',3,'Ovarian Cancer','Endometrioid Borderlin Ovarian Tumor','Ovary/Fallopian Tube','SOLID',168),(122,'THPD','Teal',2,'Thyroid Cancer','Poorly Differentiated Thyroid Cancer','Thyroid','SOLID',648),(123,'SGO','DarkRed',3,'Salivary Gland Cancer','Salivary Gland Oncocytoma','Head and Neck','SOLID',522),(124,'MBGN','LightSalmon',2,'Blood Cancer, NOS','Myeloid Benign','Myeloid','LIQUID',387),(125,'MRC','Orange',4,'Renal Cell Carcinoma','Renal Medullary Carcinoma','Kidney','SOLID',380),(126,'CSCHW','Gray',4,'Nerve Sheath Tumor','Cellular Schwannoma','Peripheral Nervous System','SOLID',78),(127,'EVN','Gray',3,'Miscellaneous Neuroepithelial Tumor','Extraventricular Neurocytoma','CNS/Brain','SOLID',420),(128,'USC','PeachPuff',3,'Endometrial Cancer','Uterine Serous Carcinoma/Uterine Papillary Serous Carcinoma','Uterus','SOLID',687),(129,'PERITONEUM','Green',1,'Peritoneal Cancer, NOS','Peritoneum','Peritoneum','SOLID',491),(130,'NECNOS','Black',3,'Cancer of Unknown Primary','Neuroendocrine Carcinoma, NOS','Other','SOLID',712),(131,'MIDDA','LimeGreen',6,'Mature B-Cell Neoplasms','Amyloidosis','Lymphoid','LIQUID',828),(132,'WDTC','Teal',2,'Thyroid Cancer','Well-Differentiated Thyroid Cancer','Thyroid','SOLID',648),(133,'AHCD','LimeGreen',5,'Mature B-Cell Neoplasms','Alpha Heavy-Chain Disease','Lymphoid','LIQUID',530),(134,'CSCC','Black',2,'Skin Cancer, Non-Melanoma','Cutaneous Squamous Cell Carcinoma','Skin','SOLID',543),(135,'BYST','Gray',3,'Germ Cell Tumor','Yolk Sac Tumor','CNS/Brain','SOLID',238),(136,'OSGCT','Purple',3,'Pancreatic Cancer','Osteoclastic Giant Cell Tumor','Pancreas','SOLID',552),(137,'BLLHYPER','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with Hyperdiploidy','Lymphoid','LIQUID',45),(138,'SBOV','LightBlue',3,'Ovarian Cancer','Serous Borderline Ovarian Tumor','Ovary/Fallopian Tube','SOLID',168),(139,'EMBCA','Red',3,'Germ Cell Tumor','Embryonal Carcinoma','Testis','SOLID',169),(140,'ADPA','Black',2,'Skin Cancer, Non-Melanoma','Aggressive Digital Papillary Adenocarcinoma','Skin','SOLID',543),(141,'LGSOC','LightBlue',4,'Ovarian Cancer','Low-Grade Serous Ovarian Cancer','Ovary/Fallopian Tube','SOLID',547),(142,'HGESS','PeachPuff',4,'Uterine Sarcoma','High-Grade Endometrial Stromal Sarcoma','Uterus','SOLID',637),(143,'EBVDLBCLNOS','LimeGreen',5,'Mature B-Cell Neoplasms','EBV Positive DLBCL, NOS','Lymphoid','LIQUID',530),(144,'SEBA','Black',2,'Skin Cancer, Non-Melanoma','Sebaceous Carcinoma','Skin','SOLID',543),(145,'BLLIAMP21','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with iAMP21','Lymphoid','LIQUID',45),(146,'HNNE','DarkRed',3,'Head and Neck Cancer','Head and Neck Neuroendocrine Carcinoma','Head and Neck','SOLID',4),(147,'MCHS','White',3,'Bone Cancer','Mesenchymal Chondrosarcoma','Bone','SOLID',742),(148,'NPC','DarkRed',2,'Head and Neck Cancer','Nasopharyngeal Carcinoma','Head and Neck','SOLID',303),(149,'APMF','LightSalmon',5,'Leukemia','Acute Panmyelosis with Myelofibrosis','Myeloid','LIQUID',601),(150,'LGESS','PeachPuff',4,'Uterine Sarcoma','Low-Grade Endometrial Stromal Sarcoma','Uterus','SOLID',637),(151,'PLSMESO','Blue',3,'Mesothelioma','Pleural Mesothelioma, Sarcomatoid Type','Pleura','SOLID',379),(152,'HCL-V','LimeGreen',6,'Mature B-Cell Neoplasms','Hairy Cell Leukemia-Variant','Lymphoid','LIQUID',3),(153,'PPCT','Black',2,'Skin Cancer, Non-Melanoma','Proliferating Pilar Cystic Tumor','Skin','SOLID',543),(154,'OMGCT','LightBlue',3,'Germ Cell Tumor','Mixed Germ Cell Tumor','Ovary/Fallopian Tube','SOLID',84),(155,'OPE','LightBlue',3,'Germ Cell Tumor','Polyembryoma','Ovary/Fallopian Tube','SOLID',84),(156,'TPLL','LimeGreen',5,'Mature T and NK Neoplasms','T-Cell Prolymphocytic Leukemia','Lymphoid','LIQUID',243),(157,'TAC','SaddleBrown',2,'Tubular Adenoma of the Colon','Tubular Adenoma of the Colon','Bowel','SOLID',289),(158,'UA','Yellow',3,'Bladder Cancer','Urachal Adenocarcinoma','Bladder/Urinary Tract','SOLID',323),(159,'CEAD','Teal',2,'Cervical Cancer','Cervical Adenocarcinoma','Cervix','SOLID',80),(160,'ALCLALKN','LimeGreen',6,'Mature T and NK Neoplasms','Anaplastic Large-Cell Lymphoma ALK Negative','Lymphoid','LIQUID',731),(161,'HEMA','LightYellow',2,'Soft Tissue Sarcoma','Hemangioma','Soft Tissue','SOLID',683),(162,'CMCD','LightSalmon',4,'Mastocytosis','Cutaneous Mastocytosis','Myeloid','LIQUID',322),(163,'PLLS','LightYellow',3,'Soft Tissue Sarcoma','Pleomorphic Liposarcoma','Soft Tissue','SOLID',313),(164,'DDLS','LightYellow',3,'Soft Tissue Sarcoma','Dedifferentiated Liposarcoma','Soft Tissue','SOLID',313),(165,'LUMEC','Gainsboro',4,'Non-Small Cell Lung Cancer','Mucoepidermoid Carcinoma of the Lung','Lung','SOLID',287),(166,'MDLC','HotPink',3,'Breast Cancer','Breast Mixed Ductal and Lobular Carcinoma','Breast','SOLID',338),(167,'ISMCL','LimeGreen',6,'Mature B-Cell Neoplasms','In Situ Mantle Cell Neoplasia','Lymphoid','LIQUID',586),(168,'OVT','LightBlue',2,'Ovarian Cancer','Ovarian Epithelial Tumor','Ovary/Fallopian Tube','SOLID',557),(169,'NSGCT','Red',2,'Germ Cell Tumor','Non-Seminomatous Germ Cell Tumor','Testis','SOLID',623),(170,'OYST','LightBlue',3,'Germ Cell Tumor','Yolk Sac Tumor','Ovary/Fallopian Tube','SOLID',84),(171,'SPN','Purple',2,'Pancreatic Cancer','Solid Pseudopapillary Neoplasm of the Pancreas','Pancreas','SOLID',525),(172,'MDSMPNU','LightSalmon',4,'Myelodysplastic/Myeloproliferative Neoplasms','MDS/MPN, Unclassifiable','Myeloid','LIQUID',285),(173,'UMNC','PeachPuff',3,'Endometrial Cancer','Uterine Mesonephric Carcinoma','Uterus','SOLID',687),(174,'MNM','LightSalmon',2,'Blood Cancer, NOS','Myeloid Neoplasm','Myeloid','LIQUID',387),(175,'SDCA','DarkRed',3,'Salivary Gland Cancer','Salivary Duct Carcinoma','Head and Neck','SOLID',522),(176,'AGA','SaddleBrown',2,'Anal Cancer','Anal Gland Adenocarcinoma','Bowel','SOLID',289),(177,'PTFL','LimeGreen',5,'Mature B-Cell Neoplasms','Pediatric-Type Follicular Lymphoma','Lymphoid','LIQUID',530),(178,'BRSRCC','HotPink',3,'Breast Cancer','Breast Carcinoma with Signet Ring','Breast','SOLID',338),(179,'TYST','Red',3,'Germ Cell Tumor','Yolk Sac Tumor','Testis','SOLID',169),(180,'PAAD','Purple',2,'Pancreatic Cancer','Pancreatic Adenocarcinoma','Pancreas','SOLID',525),(181,'MGCT','Red',3,'Germ Cell Tumor','Mixed Germ Cell Tumor','Testis','SOLID',169),(182,'CECC','Teal',3,'Cervical Cancer','Cervical Clear Cell Carcinoma','Cervix','SOLID',159),(183,'LMS','LightYellow',2,'Soft Tissue Sarcoma','Leiomyosarcoma','Soft Tissue','SOLID',683),(184,'AMLBCRABL1','LightSalmon',5,'Leukemia','AML with BCR-ABL1','Myeloid','LIQUID',198),(185,'MPN','LightSalmon',3,'Myeloproliferative Neoplasms','Myeloproliferative Neoplasms','Myeloid','LIQUID',174),(186,'SGAD','Black',2,'Skin Cancer, Non-Melanoma','Sweat Gland Adenocarcinoma','Skin','SOLID',543),(187,'SCBC','Yellow',2,'Bladder Cancer','Small Cell Bladder Cancer','Bladder/Urinary Tract','SOLID',108),(188,'DIG','Gray',3,'Miscellaneous Neuroepithelial Tumor','Desmoplastic Infantile Ganglioglioma','CNS/Brain','SOLID',420),(189,'DMBL','Gray',3,'Embryonal Tumor','Desmoplastic/Nodular Medulloblastoma','CNS/Brain','SOLID',347),(190,'MRT','Orange',2,'Rhabdoid Cancer','Rhabdoid Cancer','Kidney','SOLID',431),(191,'IMTB','Yellow',2,'Bladder Cancer','Inflammatory Myofibroblastic Bladder Tumor','Bladder/Urinary Tract','SOLID',108),(192,'MELC','Gray',3,'Melanocytoma','Melanocytoma','CNS/Brain','SOLID',96),(193,'BLLETV6RUNX1','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with t(12;21)(p13.2;q22.1); ETV6-RUNX1','Lymphoid','LIQUID',45),(194,'LDCHL','LimeGreen',5,'Hodgkin Lymphoma','Lymphocyte-Depleted Classical Hodgkin Lymphoma','Lymphoid','LIQUID',500),(195,'APXA','Gray',3,'Glioma','Anaplastic Pleomorphic Xanthoastrocytoma','CNS/Brain','SOLID',54),(196,'CCHM','HotPink',4,'Breast Cancer','Carcinoma with Chondroid Metaplasia','Breast','SOLID',776),(197,'PINC','Gray',3,'Pineal Tumor','Pineocytoma','CNS/Brain','SOLID',266),(198,'AMLRGA','LightSalmon',4,'Leukemia','AML with Recurrent Genetic Abnormalities','Myeloid','LIQUID',527),(199,'MT','Gray',3,'Miscellaneous Brain Tumor','Malignant Tumor','CNS/Brain','SOLID',587),(200,'MCSL','LightSalmon',4,'Mastocytosis','Mast Cell Sarcoma','Myeloid','LIQUID',322),(201,'BMGT','Gray',3,'Germ Cell Tumor','Malignant Teratoma','CNS/Brain','SOLID',238),(202,'SPB','LimeGreen',5,'Mature B-Cell Neoplasms','Solitary Plasmacytoma of Bone','Lymphoid','LIQUID',530),(203,'CCM','Gray',3,'CNS Cancer','Clear cell Meningioma','CNS/Brain','SOLID',750),(204,'SM','LightSalmon',4,'Mastocytosis','Systemic Mastocytosis','Myeloid','LIQUID',322),(205,'LECLC','Gainsboro',4,'Non-Small Cell Lung Cancer','Lymphoepithelioma-like Carcinoma of the Lung','Lung','SOLID',342),(206,'BCLU','LimeGreen',5,'Mature B-Cell Neoplasms','B-Cell Lymphoma, Unclassifiable, with Features Intermediate between DLBCL and Classical Hodgkin lymphoma','Lymphoid','LIQUID',530),(207,'BLLHYPO','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with Hypodiploidy','Lymphoid','LIQUID',45),(208,'CUPNOS','Black',3,'Cancer of Unknown Primary','Cancer of Unknown Primary, NOS','Other','SOLID',712),(209,'LIMNET','MediumSeaGreen',2,'Hepatobiliary Cancer','Malignant Nonepithelial Tumor of the Liver','Liver','SOLID',567),(210,'IFS','LightYellow',2,'Infantile Fibrosarcoma','Infantile Fibrosarcoma','Soft Tissue','SOLID',683),(211,'RMS','LightYellow',2,'Soft Tissue Sarcoma','Rhabdomyosarcoma','Soft Tissue','SOLID',683),(212,'OCNOS','LightBlue',3,'Ovarian Cancer','Ovarian Choriocarcinoma, NOS','Ovary/Fallopian Tube','SOLID',861),(213,'THPA','Teal',3,'Thyroid Cancer','Papillary Thyroid Cancer','Thyroid','SOLID',132),(214,'CHRCC','Orange',4,'Renal Cell Carcinoma','Chromophobe Renal Cell Carcinoma','Kidney','SOLID',380),(215,'PNS','Gray',1,'Peripheral Nervous System Cancer, NOS','Peripheral Nervous System','Peripheral Nervous System','SOLID',491),(216,'ANM','Gray',3,'CNS Cancer','Anaplastic Meningioma','CNS/Brain','SOLID',750),(217,'PLBL','LimeGreen',5,'Mature B-Cell Neoplasms','Plasmablastic Lymphoma','Lymphoid','LIQUID',530),(218,'LAIS','Gainsboro',2,'Non-Small Cell Lung Cancer','Lung Adenocarcinoma In Situ','Lung','SOLID',475),(219,'UEC','PeachPuff',3,'Endometrial Cancer','Uterine Endometrioid Carcinoma','Uterus','SOLID',687),(220,'SFTCNS','Gray',3,'CNS Cancer','Solitary Fibrous Tumor of the Central Nervous System','CNS/Brain','SOLID',750),(221,'ENKL','LimeGreen',5,'Mature T and NK Neoplasms','Extranodal NK-/T-Cell Lymphoma, Nasal Type','Lymphoid','LIQUID',243),(222,'VMA','Purple',2,'Vulvar Carcinoma','Mucinous Adenocarcinoma of the Vulva/Vagina','Vulva/Vagina','SOLID',853),(223,'ASTR','Gray',3,'Glioma','Astrocytoma','CNS/Brain','SOLID',619),(224,'CMML','LightSalmon',4,'Myelodysplastic/Myeloproliferative Neoplasms','Chronic Myelomonocytic Leukemia','Myeloid','LIQUID',285),(225,'VA','Purple',2,'Vaginal Cancer','Vaginal Adenocarcinoma','Vulva/Vagina','SOLID',853),(226,'GNBL','Gray',2,'Peripheral Nervous System','Ganglioneuroblastoma','Peripheral Nervous System','SOLID',215),(227,'MDSMD','LightSalmon',4,'Myelodysplastic Syndromes','MDS with Multilineage Dysplasia','Myeloid','LIQUID',355),(228,'MIDDO','LimeGreen',6,'Mature B-Cell Neoplasms','Monoclonal Immunoglobulin Deposition Diseases, Other','Lymphoid','LIQUID',828),(229,'MOV','LightBlue',3,'Ovarian Cancer','Mucinous Ovarian Cancer','Ovary/Fallopian Tube','SOLID',168),(230,'TNET','Purple',2,'Thymic Tumor','Thymic Neuroendocrine Tumor','Thymus','SOLID',686),(231,'GNC','Gray',3,'Glioma','Gangliocytoma','CNS/Brain','SOLID',54),(232,'APE','Gray',3,'CNS Cancer','Anaplastic Ependymoma','CNS/Brain','SOLID',832),(233,'CCHDM','White',3,'Bone Cancer','Conventional Type Chordoma','Bone','SOLID',450),(234,'BTBOV','LightBlue',4,'Ovarian Cancer','Brenner Tumor, Borderline','Ovary/Fallopian Tube','SOLID',602),(235,'GSARC','Gray',4,'Glioma','Gliosarcoma','CNS/Brain','SOLID',666),(236,'PTCL','LimeGreen',5,'Mature T and NK Neoplasms','Peripheral T-Cell lymphoma, NOS','Lymphoid','LIQUID',243),(237,'CTAAP','SaddleBrown',3,'Appendiceal Cancer','Colonic Type Adenocarcinoma of the Appendix','Bowel','SOLID',566),(238,'BGCT','Gray',2,'Germ Cell Tumor','Germ Cell Tumor, Brain','CNS/Brain','SOLID',644),(239,'RBL','Green',2,'Retinoblastoma','Retinoblastoma','Eye','SOLID',280),(240,'SPCC','Gainsboro',3,'Non-Small Cell Lung Cancer','Spindle Cell Carcinoma of the Lung','Lung','SOLID',83),(241,'ANGL','Gray',3,'Miscellaneous Neuroepithelial Tumor','Angiocentric Glioma','CNS/Brain','SOLID',420),(242,'NST','Gray',2,'Nerve Sheath Tumor','Nerve Sheath Tumor','Peripheral Nervous System','SOLID',215),(243,'MTNN','LimeGreen',4,'Mature T and NK Neoplasms','Mature T and NK Neoplasms','Lymphoid','LIQUID',576),(244,'CCRCC','Orange',3,'Renal Cell Carcinoma','Renal Clear Cell Carcinoma','Kidney','SOLID',850),(245,'UUC','PeachPuff',3,'Endometrial Cancer','Uterine Undifferentiated Carcinoma','Uterus','SOLID',687),(246,'AMLMLLT3KMT2A','LightSalmon',5,'Leukemia','AML with t(9;11)(p21.3;q23.3);MLLT3-KMT2A','Myeloid','LIQUID',198),(247,'EGCT','Black',2,'Germ Cell Tumor','Extra Gonadal Germ Cell Tumor','Other','SOLID',716),(248,'BMT','Gray',3,'Germ Cell Tumor','Mature Teratoma','CNS/Brain','SOLID',238),(249,'UCCA','PeachPuff',3,'Gestational Trophoblastic Disease','Choriocarcinoma','Uterus','SOLID',417),(250,'SCUP','Black',3,'Cancer of Unknown Primary','Small Cell Carcinoma of Unknown Primary','Other','SOLID',712),(251,'TAM','LightSalmon',5,'Leukemia','Transient Abnormal Myelopoiesis','Myeloid','LIQUID',844),(252,'LATL','LimeGreen',2,'Lymphatic Cancer, NOS','Lymphoid Atypical','Lymphoid','LIQUID',258),(253,'AFH','LightYellow',2,'Angiomatoid Fibrous Histiocytoma','Angiomatoid Fibrous Histiocytoma','Soft Tissue','SOLID',683),(254,'IAMPCA','Purple',3,'Ampullary Cancer','Intestinal Ampullary Carcinoma','Ampulla of Vater','SOLID',732),(255,'MLADS','LightSalmon',5,'Leukemia','Myeloid Leukemia Associated with Down Syndrome','Myeloid','LIQUID',844),(256,'USTAD','LightSkyBlue',3,'Esophagogastric Cancer','Undifferentiated Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',361),(257,'CESE','Teal',3,'Cervical Cancer','Cervical Serous Carcinoma','Cervix','SOLID',159),(258,'LYMPH','LimeGreen',1,'Lymphatic Cancer, NOS','Lymphoid','Lymphoid','LIQUID',491),(259,'LGGNOS','Gray',3,'Glioma','Low-Grade Glioma, NOS','CNS/Brain','SOLID',54),(260,'APLPMLRARA','LightSalmon',5,'Leukemia','APL with PML-RARA','Myeloid','LIQUID',198),(261,'MLYM','Gray',3,'Miscellaneous Brain Tumor','Malignant Lymphoma','CNS/Brain','SOLID',587),(262,'SRCC','Orange',4,'Renal Cell Carcinoma','Sarcomatoid Renal Cell Carcinoma','Kidney','SOLID',380),(263,'BRAME','HotPink',2,'Breast Cancer','Adenomyoepithelioma of the Breast','Breast','SOLID',799),(264,'AITL','LimeGreen',5,'Mature T and NK Neoplasms','Angioimmunoblastic T-Cell Lymphoma','Lymphoid','LIQUID',243),(265,'PAAC','Purple',2,'Pancreatic Cancer','Acinar Cell Carcinoma of the Pancreas','Pancreas','SOLID',525),(266,'PINT','Gray',2,'Pineal Tumor','Pineal Tumor','CNS/Brain','SOLID',644),(267,'MDSRS','LightSalmon',4,'Myelodysplastic Syndromes','MDS with Ring Sideroblasts','Myeloid','LIQUID',355),(268,'HS','LightSalmon',4,'Histiocytosis','Histiocytic Sarcoma','Myeloid','LIQUID',352),(269,'CHLPTLD','LimeGreen',4,'Posttransplant Lymphoproliferative Disorders','Classical Hodgkin Lymphoma PTLD','Lymphoid','LIQUID',55),(270,'SRCCR','SaddleBrown',3,'Colorectal Cancer','Signet Ring Cell Adenocarcinoma of the Colon and Rectum','Bowel','SOLID',382),(271,'OIMT','LightBlue',3,'Germ Cell Tumor','Immature Teratoma','Ovary/Fallopian Tube','SOLID',84),(272,'VGCE','Teal',2,'Cervical Cancer','Villoglandular Adenocarcinoma of the Cervix','Cervix','SOLID',80),(273,'MPALKMT2A','LightSalmon',4,'Leukemia','Mixed Phenotype Acute Leukemia with t(v;11q23.3); KMT2A Rearranged','Myeloid','LIQUID',734),(274,'UELMS','PeachPuff',4,'Uterine Sarcoma','Uterine Epithelioid Leiomyosarcoma','Uterus','SOLID',754),(275,'PRSC','Cyan',2,'Prostate Cancer','Prostate Squamous Cell Carcinoma','Prostate','SOLID',374),(276,'HCL','LimeGreen',5,'Mature B-Cell Neoplasms','Hairy Cell Leukemia','Lymphoid','LIQUID',530),(277,'CESC','Teal',2,'Cervical Cancer','Cervical Squamous Cell Carcinoma','Cervix','SOLID',80),(278,'SYNS','LightYellow',2,'Soft Tissue Sarcoma','Synovial Sarcoma','Soft Tissue','SOLID',683),(279,'ERMS','LightYellow',3,'Soft Tissue Sarcoma','Embryonal Rhabdomyosarcoma','Soft Tissue','SOLID',211),(280,'EYE','Green',1,'Eye Cancer, NOS','Eye','Eye','SOLID',491),(281,'ASM','LightSalmon',5,'Mastocytosis','Aggressive Systemic Mastocytosis','Myeloid','LIQUID',204),(282,'UM','Green',3,'Melanoma','Uveal Melanoma','Eye','SOLID',93),(283,'CHOL','Green',2,'Hepatobiliary Cancer','Cholangiocarcinoma','Biliary Tract','SOLID',393),(284,'ESST','LightYellow',2,'Soft Tissue Sarcoma','Ewing Sarcoma of Soft Tissue','Soft Tissue','SOLID',683),(285,'MDS/MPN','LightSalmon',3,'Myelodysplastic/Myeloproliferative Neoplasms','Myelodysplastic/Myeloproliferative Neoplasms','Myeloid','LIQUID',174),(286,'FL','LimeGreen',5,'Mature B-Cell Neoplasms','Follicular Lymphoma','Lymphoid','LIQUID',530),(287,'SGTTL','Gainsboro',3,'Non-Small Cell Lung Cancer','Salivary Gland-Type Tumor of the Lung','Lung','SOLID',83),(288,'PTH','DarkRed',2,'Parathyroid Cancer','Parathyroid Cancer','Head and Neck','SOLID',303),(289,'BOWEL','SaddleBrown',1,'Bowel Cancer, NOS','Bowel','Bowel','SOLID',491),(290,'WT','Orange',2,'Wilms Tumor','Wilms\' Tumor','Kidney','SOLID',431),(291,'TET','Purple',2,'Thymic Tumor','Thymic Epithelial Tumor','Thymus','SOLID',686),(292,'IUP','Yellow',2,'Bladder Cancer','Inverted Urothelial Papilloma','Bladder/Urinary Tract','SOLID',108),(293,'TLL','LimeGreen',3,'T-Lymphoblastic Leukemia/Lymphoma','T-Lymphoblastic Leukemia/Lymphoma','Lymphoid','LIQUID',31),(294,'SS','LimeGreen',5,'Mature T and NK Neoplasms','Sezary Syndrome','Lymphoid','LIQUID',243),(295,'HGNEC','SaddleBrown',3,'Gastrointestinal Neuroendocrine Tumor','High-Grade Neuroendocrine Carcinoma of the Colon and Rectum','Bowel','SOLID',399),(296,'AODG','Gray',3,'Glioma','Anaplastic Oligodendroglioma','CNS/Brain','SOLID',619),(297,'PORO','Black',2,'Skin Cancer, Non-Melanoma','Poroma/Acrospiroma','Skin','SOLID',543),(298,'HTAT','Teal',2,'Thyroid Cancer','Hyalinizing Trabecular Adenoma of the Thyroid','Thyroid','SOLID',648),(299,'LDD','Gray',3,'Miscellaneous Neuroepithelial Tumor','Dysplastic Gangliocytoma of the Cerebellum/Lhermitte-Duclos Disease','CNS/Brain','SOLID',420),(300,'HGSFT','LightBlue',3,'Ovarian Cancer','High-Grade Serous Fallopian Tube Cancer','Ovary/Fallopian Tube','SOLID',861),(301,'NMCHN','DarkRed',3,'Head and Neck Cancer','NUT Midline Carcinoma of the Head and Neck','Head and Neck','SOLID',4),(302,'UMLMS','PeachPuff',4,'Uterine Sarcoma','Uterine Myxoid Leiomyosarcoma','Uterus','SOLID',754),(303,'HEAD_NECK','DarkRed',1,'Head and Neck Cancer, NOS','Head and Neck','Head and Neck','SOLID',491),(304,'UDMN','Black',3,'Cancer of Unknown Primary','Undifferentiated Malignant Neoplasm','Other','SOLID',712),(305,'ACRM','Black',3,'Melanoma','Acral Melanoma','Skin','SOLID',453),(306,'MLNPDGFRA','LightSalmon',4,'Leukemia','Myeloid/Lymphoid Neoplasms with PDGFRA Rearrangement','Myeloid','LIQUID',693),(307,'PHC','Purple',2,'Pheochromocytoma','Pheochromocytoma','Adrenal Gland','SOLID',38),(308,'ATLL','LimeGreen',5,'Mature T and NK Neoplasms','Adult T-Cell Leukemia/Lymphoma','Lymphoid','LIQUID',243),(309,'THRLBCL','LimeGreen',5,'Mature B-Cell Neoplasms','T-Cell/Histiocyte-Rich Large B-Cell Lymphoma','Lymphoid','LIQUID',530),(310,'PTAD','Gray',3,'Sellar Tumor','Pituitary Adenoma','CNS/Brain','SOLID',26),(311,'ESMM','LightSkyBlue',2,'Melanoma','Mucosal Melanoma of the Esophagus','Esophagus/Stomach','SOLID',541),(312,'MNG','Gray',3,'CNS Cancer','Meningioma','CNS/Brain','SOLID',750),(313,'LIPO','LightYellow',2,'Soft Tissue Sarcoma','Liposarcoma','Soft Tissue','SOLID',683),(314,'LBLIRF4','LimeGreen',5,'Mature B-Cell Neoplasms','Large B-Cell Lymphoma with IRF4 Rearrangement','Lymphoid','LIQUID',530),(315,'DCIS','HotPink',2,'Breast Cancer','Breast Ductal Carcinoma In Situ','Breast','SOLID',799),(316,'TT','Red',3,'Germ Cell Tumor','Teratoma','Testis','SOLID',169),(317,'PCSMTPLD','LimeGreen',5,'Mature T and NK Neoplasms','Primary Cutaneous CD4 Positive Small/Medium T-Cell Lymphoproliferative Disorder','Lymphoid','LIQUID',243),(318,'AMML','LightSalmon',5,'Leukemia','Acute Myelomonocytic Leukemia','Myeloid','LIQUID',601),(319,'ESCA','LightSkyBlue',3,'Esophagogastric Cancer','Esophageal Adenocarcinoma','Esophagus/Stomach','SOLID',361),(320,'MCC','Black',2,'Skin Cancer, Non-Melanoma','Merkel Cell Carcinoma','Skin','SOLID',543),(321,'URCC','Orange',4,'Renal Cell Carcinoma','Unclassified Renal Cell Carcinoma','Kidney','SOLID',380),(322,'MCD','LightSalmon',3,'Mastocytosis','Mastocytosis','Myeloid','LIQUID',174),(323,'URCA','Yellow',2,'Bladder Cancer','Urachal Carcinoma','Bladder/Urinary Tract','SOLID',108),(324,'OMT','LightBlue',3,'Germ Cell Tumor','Mature Teratoma','Ovary/Fallopian Tube','SOLID',84),(325,'LYG','LimeGreen',5,'Mature B-Cell Neoplasms','Lymphomatoid Granulomatosis','Lymphoid','LIQUID',530),(326,'SPDAC','LightSkyBlue',4,'Esophagogastric Cancer','Poorly Differentiated Carcinoma of the Stomach','Esophagus/Stomach','SOLID',256),(327,'NUTCL','Gainsboro',3,'Non-Small Cell Lung Cancer','NUT Carcinoma of the Lung','Lung','SOLID',83),(328,'SEF','LightYellow',3,'Soft Tissue Sarcoma','Sclerosing Epithelioid Fibrosarcoma','Soft Tissue','SOLID',74),(329,'AECA','Black',2,'Skin Cancer, Non-Melanoma','Sweat Gland Carcinoma/Apocrine Eccrine Carcinoma','Skin','SOLID',543),(330,'ETMF','LightSalmon',5,'Myeloproliferative Neoplasms','Essential Thrombocythemia Myelofibrosis','Myeloid','LIQUID',695),(331,'GINETES','LightSkyBlue',2,'Esophagogastric Cancer','Gastrointestinal Neuroendocrine Tumors of the Esophagus/Stomach','Esophagus/Stomach','SOLID',541),(332,'OAST','Gray',3,'Glioma','Oligoastrocytoma','CNS/Brain','SOLID',619),(333,'DES','LightYellow',2,'Soft Tissue Sarcoma','Desmoid/Aggressive Fibromatosis','Soft Tissue','SOLID',683),(334,'PAASC','Purple',2,'Pancreatic Cancer','Adenosquamous Carcinoma of the Pancreas','Pancreas','SOLID',525),(335,'SNSC','DarkRed',3,'Head and Neck Cancer','Sinonasal Squamous Cell Carcinoma','Head and Neck','SOLID',44),(336,'ALCLALKP','LimeGreen',6,'Mature T and NK Neoplasms','Anaplastic Large-Cell Lymphoma ALK Positive','Lymphoid','LIQUID',731),(337,'MDSRSSLD','LightSalmon',5,'Myelodysplastic Syndromes','MDS with Ring Sideroblasts and Single Lineage Dysplasia','Myeloid','LIQUID',267),(338,'BRCA','HotPink',2,'Breast Cancer','Invasive Breast Carcinoma','Breast','SOLID',799),(339,'IHCH','Green',3,'Hepatobiliary Cancer','Intrahepatic Cholangiocarcinoma','Biliary Tract','SOLID',283),(340,'ODGC','DarkRed',3,'Head and Neck Cancer','Odontogenic Carcinoma','Head and Neck','SOLID',4),(341,'EBVMCU','LimeGreen',5,'Mature B-Cell Neoplasms','EBV Positive Mucocutaneous Ulcer','Lymphoid','LIQUID',530),(342,'LCLC','Gainsboro',3,'Non-Small Cell Lung Cancer','Large Cell Lung Carcinoma','Lung','SOLID',83),(343,'STAS','LightSkyBlue',3,'Esophagogastric Cancer','Adenosquamous Carcinoma of the Stomach','Esophagus/Stomach','SOLID',361),(344,'COAD','SaddleBrown',3,'Colorectal Cancer','Colon Adenocarcinoma','Bowel','SOLID',382),(345,'SCT','LightBlue',3,'Sex Cord Stromal Tumor','Steroid Cell Tumor, NOS','Ovary/Fallopian Tube','SOLID',8),(346,'EHAE','LightYellow',2,'Soft Tissue Sarcoma','Epithelioid Hemangioendothelioma','Soft Tissue','SOLID',683),(347,'EMBT','Gray',2,'Embryonal Tumor','Embryonal Tumor','CNS/Brain','SOLID',644),(348,'READ','SaddleBrown',3,'Colorectal Cancer','Rectal Adenocarcinoma','Bowel','SOLID',382),(349,'CNC','Gray',3,'Miscellaneous Neuroepithelial Tumor','Central Neurocytoma','CNS/Brain','SOLID',420),(350,'UCS','PeachPuff',3,'Endometrial Cancer','Uterine Carcinosarcoma/Uterine Malignant Mixed Mullerian Tumor','Uterus','SOLID',687),(351,'BLLBCRABL1','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with t(9;22)(q34.1;q11.2);BCR-ABL1','Lymphoid','LIQUID',45),(352,'HDCN','LightSalmon',3,'Histiocytosis','Histiocytic and Dendritic Cell Neoplasms','Myeloid','LIQUID',174),(353,'MTSCC','Orange',4,'Renal Cell Carcinoma','Renal Mucinous Tubular Spindle Cell Carcinoma','Kidney','SOLID',380),(354,'MP','PeachPuff',3,'Gestational Trophoblastic Disease','Molar Pregnancy','Uterus','SOLID',417),(355,'MDS','LightSalmon',3,'Myelodysplastic Syndromes','Myelodysplastic Syndromes','Myeloid','LIQUID',174),(356,'VPE','Purple',3,'Germ Cell Tumor','Polyembryoma','Vulva/Vagina','SOLID',383),(357,'FHPTLD','LimeGreen',4,'Posttransplant Lymphoproliferative Disorders','Florid Follicular Hyperplasia PTLD','Lymphoid','LIQUID',55),(358,'URMM','Yellow',2,'Melanoma','Mucosal Melanoma of the Urethra','Bladder/Urinary Tract','SOLID',108),(359,'BLL11Q','LimeGreen',5,'Mature B-Cell Neoplasms','Burkitt-Like Lymphoma with 11q Aberration','Lymphoid','LIQUID',530),(360,'SNUC','DarkRed',3,'Head and Neck Cancer','Sinonasal Undifferentiated Carcinoma','Head and Neck','SOLID',4),(361,'EGC','LightSkyBlue',2,'Esophagogastric Cancer','Esophagogastric Adenocarcinoma','Esophagus/Stomach','SOLID',541),(362,'MDSEB','LightSalmon',4,'Myelodysplastic Syndromes','MDS with Excess Blasts','Myeloid','LIQUID',355),(363,'MYXO','LightYellow',2,'Soft Tissue Sarcoma','Myxoma','Soft Tissue','SOLID',683),(364,'DF','Black',2,'Skin Cancer, Non-Melanoma','Dermatofibroma','Skin','SOLID',543),(365,'CCLC','Gainsboro',4,'Non-Small Cell Lung Cancer','Clear Cell Carcinoma of the Lung','Lung','SOLID',342),(366,'NBL','Gray',2,'Peripheral Nervous System','Neuroblastoma','Peripheral Nervous System','SOLID',215),(367,'RDD','LightSalmon',4,'Histiocytosis','Rosai-Dorfman Disease','Myeloid','LIQUID',352),(368,'DDCHDM','White',3,'Bone Cancer','Dedifferentiated Chordoma','Bone','SOLID',450),(369,'IDCT','LightSalmon',4,'Histiocytosis','Indeterminate Dendritic Cell Tumor','Myeloid','LIQUID',352),(370,'LUAD','Gainsboro',3,'Non-Small Cell Lung Cancer','Lung Adenocarcinoma','Lung','SOLID',83),(371,'SLCT','LightBlue',3,'Sex Cord Stromal Tumor','Sertoli-Leydig Cell Tumor','Ovary/Fallopian Tube','SOLID',8),(372,'INTS','LightYellow',2,'Soft Tissue Sarcoma','Intimal Sarcoma','Soft Tissue','SOLID',683),(373,'UAD','Yellow',3,'Bladder Cancer','Urethral Adenocarcinoma','Bladder/Urinary Tract','SOLID',647),(374,'PROSTATE','Cyan',1,'Prostate Cancer, NOS','Prostate','Prostate','SOLID',491),(375,'FTCL','LimeGreen',5,'Mature T and NK Neoplasms','Follicular T-Cell Lymphoma','Lymphoid','LIQUID',243),(376,'HCC','MediumSeaGreen',2,'Hepatobiliary Cancer','Hepatocellular Carcinoma','Liver','SOLID',567),(377,'ACN','Black',3,'Cancer of Unknown Primary','Acinar Cell Carcinoma, NOS','Other','SOLID',712),(378,'MDSEB2','LightSalmon',5,'Myelodysplastic Syndromes','MDS with excess blasts-2','Myeloid','LIQUID',362),(379,'PLMESO','Blue',2,'Mesothelioma','Pleural Mesothelioma','Pleura','SOLID',707),(380,'NCCRCC','Orange',3,'Renal Cell Carcinoma','Renal Non-Clear Cell Carcinoma','Kidney','SOLID',850),(381,'LBGN','LimeGreen',2,'Lymphatic Cancer, NOS','Lymphoid Benign','Lymphoid','LIQUID',258),(382,'COADREAD','SaddleBrown',2,'Colorectal Cancer','Colorectal Adenocarcinoma','Bowel','SOLID',289),(383,'VGCT','Purple',2,'Germ Cell Tumor','Germ Cell Tumor of the Vulva','Vulva/Vagina','SOLID',853),(384,'MPE','Gray',3,'CNS Cancer','Myxopapillary Ependymoma','CNS/Brain','SOLID',832),(385,'CLNC','Gray',3,'Miscellaneous Neuroepithelial Tumor','Cerebellar Liponeurocytoma','CNS/Brain','SOLID',420),(386,'PGNG','Gray',2,'Miscellaneous Neuroepithelial Tumor','Paraganglioma','Soft Tissue','SOLID',683),(387,'MYELOID','LightSalmon',1,'Blood Cancer, NOS','Myeloid','Myeloid','LIQUID',491),(388,'EMALT','LimeGreen',6,'Mature B-Cell Neoplasms','Extranodal Marginal Zone Lymphoma of Mucosa-Associated Lymphoid Tissue (MALT lymphoma)','Lymphoid','LIQUID',643),(389,'BONE','White',1,'Bone Cancer, NOS','Bone','Bone','SOLID',491),(390,'AMLRBM15MKL1','LightSalmon',5,'Leukemia','AML (megakaryoblastic) with t(1;22)(p13.3;q13.3);RBM15-MKL1','Myeloid','LIQUID',198),(391,'IPMN','Purple',3,'Pancreatic Cancer','Intraductal Papillary Mucinous Neoplasm','Pancreas','SOLID',464),(392,'MGUS','LimeGreen',5,'Mature B-Cell Neoplasms','Monoclonal Gammopathy of Undetermined Significance','Lymphoid','LIQUID',530),(393,'BILIARY_TRACT','Green',1,'Biliary Tract Cancer, NOS','Biliary Tract','Biliary Tract','SOLID',491),(394,'EMPD','Black',2,'Skin Cancer, Non-Melanoma','Extramammary Paget Disease','Skin','SOLID',543),(395,'CHOS','White',3,'Bone Cancer','Chondroblastic Osteosarcoma','Bone','SOLID',631),(396,'RLCLC','Gainsboro',4,'Non-Small Cell Lung Cancer','Large Cell Lung Carcinoma With Rhabdoid Phenotype','Lung','SOLID',342),(397,'OSACA','DarkRed',3,'Salivary Gland Cancer','Salivary Carcinoma, Other','Head and Neck','SOLID',522),(398,'PANET','Purple',2,'Pancreatic Cancer','Pancreatic Neuroendocrine Tumor','Pancreas','SOLID',525),(399,'GINET','SaddleBrown',2,'Gastrointestinal Neuroendocrine Tumor','Gastrointestinal Neuroendocrine Tumors','Bowel','SOLID',289),(400,'SFT','LightYellow',2,'Soft Tissue Sarcoma','Solitary Fibrous Tumor/Hemangiopericytoma','Soft Tissue','SOLID',683),(401,'EPM','Gray',3,'Glioma','Ependymoma','CNS/Brain','SOLID',832),(402,'HMBL','Gray',3,'Miscellaneous Brain Tumor','Hemangioblastoma','CNS/Brain','SOLID',587),(403,'HGSOS','White',3,'Bone Cancer','High-Grade Surface Osteosarcoma','Bone','SOLID',631),(404,'TLYM','Red',2,'Non-Hodgkin Lymphoma','Testicular Lymphoma','Testis','SOLID',623),(405,'PNET','Gray',3,'Embryonal Tumor','Primitive Neuroectodermal Tumor','CNS/Brain','SOLID',347),(406,'TMESO','Red',2,'Mesothelioma','Testicular Mesothelioma','Testis','SOLID',623),(407,'PAST','Gray',3,'Glioma','Pilocytic Astrocytoma','CNS/Brain','SOLID',54),(408,'DFSP','Black',2,'Skin Cancer, Non-Melanoma','Dermatofibrosarcoma Protuberans','Skin','SOLID',543),(409,'THAP','Teal',2,'Thyroid Cancer','Anaplastic Thyroid Cancer','Thyroid','SOLID',648),(410,'SKLMM','Black',3,'Melanoma','Lentigo Maligna Melanoma','Skin','SOLID',453),(411,'TGCT','LightYellow',2,'Soft Tissue Sarcoma','Tenosynovial Giant Cell Tumor Diffuse Type','Soft Tissue','SOLID',683),(412,'PTCY','Gray',3,'Sellar Tumor','Pituicytoma','CNS/Brain','SOLID',26),(413,'ASCT','DarkRed',3,'Head and Neck Cancer','Adenosquamous Carcinoma of the Tongue','Head and Neck','SOLID',4),(414,'BNNOS','HotPink',2,'Breast Cancer','Breast Neoplasm, NOS','Breast','SOLID',799),(415,'MACR','SaddleBrown',3,'Colorectal Cancer','Mucinous Adenocarcinoma of the Colon and Rectum','Bowel','SOLID',382),(416,'IBC','HotPink',2,'Breast Cancer','Inflammatory Breast Cancer','Breast','SOLID',799),(417,'GTD','PeachPuff',2,'Gestational Trophoblastic Disease','Gestational Trophoblastic Disease','Uterus','SOLID',629),(418,'TMN','LightSalmon',4,'Leukemia','Therapy-Related Myeloid Neoplasms','Myeloid','LIQUID',527),(419,'PCT','Black',2,'Skin Cancer, Non-Melanoma','Porphyria Cutania Tarda','Skin','SOLID',543),(420,'MNET','Gray',2,'Miscellaneous Neuroepithelial Tumor','Miscellaneous Neuroepithelial Tumor','CNS/Brain','SOLID',644),(421,'MEITL','LimeGreen',5,'Mature T and NK Neoplasms','Monomorphic Epitheliotropic Intestinal T-Cell Lymphoma','Lymphoid','LIQUID',243),(422,'PRAD','Cyan',2,'Prostate Cancer','Prostate Adenocarcinoma','Prostate','SOLID',374),(423,'MRTL','MediumSeaGreen',2,'Malignant Rhabdoid Tumor of the Liver','Malignant Rhabdoid Tumor of the Liver','Liver','SOLID',567),(424,'NKCLL','LimeGreen',4,'T-Lymphoblastic Leukemia/Lymphoma','Natural Killer (NK) Cell Lymphoblastic Leukemia/Lymphoma','Lymphoid','LIQUID',293),(425,'OSOS','White',3,'Bone Cancer','Osteoblastic Osteosarcoma','Bone','SOLID',631),(426,'AIS','Black',2,'Adenocarcinoma In Situ','Adenocarcinoma In Situ','Other','SOLID',716),(427,'AASTR','Gray',3,'Glioma','Anaplastic Astrocytoma','CNS/Brain','SOLID',619),(428,'VIMT','Purple',3,'Germ Cell Tumor','Immature Teratoma','Vulva/Vagina','SOLID',383),(429,'AMLDEKNUP214','LightSalmon',5,'Leukemia','AML with t(6;9)(p23;q34.1);DEK-NUP214','Myeloid','LIQUID',198),(430,'MGUSIGA','LimeGreen',6,'Mature B-Cell Neoplasms','IgA','Lymphoid','LIQUID',392),(431,'KIDNEY','Orange',1,'Kidney Cancer, NOS','Kidney','Kidney','SOLID',491),(432,'MDEP','Gray',3,'Embryonal Tumor','Medulloepithelioma','CNS/Brain','SOLID',347),(433,'SSM','LightSalmon',5,'Mastocytosis','Smoldering Systemic Mastocytosis','Myeloid','LIQUID',204),(434,'CMML2','LightSalmon',5,'Myelodysplastic/Myeloproliferative Neoplasms','Chronic Myelomonocytic Leukemia-2','Myeloid','LIQUID',224),(435,'HNSCUP','DarkRed',3,'Head and Neck Cancer','Head and Neck Squamous Cell Carcinoma of Unknown Primary','Head and Neck','SOLID',44),(436,'IMS','LightYellow',2,'Myofibromatosis','Myofibromatosis','Soft Tissue','SOLID',683),(437,'SMN','LightSkyBlue',2,'Esophagogastric Cancer','Smooth Muscle Neoplasm, NOS','Esophagus/Stomach','SOLID',541),(438,'PRSCC','Cyan',2,'Prostate Cancer','Prostate Small Cell Carcinoma','Prostate','SOLID',374),(439,'DSTAD','LightSkyBlue',4,'Esophagogastric Cancer','Diffuse Type Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',620),(440,'SECOS','White',3,'Bone Cancer','Secondary Osteosarcoma','Bone','SOLID',631),(441,'NMZL','LimeGreen',6,'Mature B-Cell Neoplasms','Nodal Marginal Zone Lymphoma','Lymphoid','LIQUID',643),(442,'PT','HotPink',3,'Breast Sarcoma','Phyllodes Tumor of the Breast','Breast','SOLID',655),(443,'MSCHW','Gray',4,'Nerve Sheath Tumor','Melanotic Schwannoma','Peripheral Nervous System','SOLID',78),(444,'ACPG','Gray',3,'Sellar Tumor','Craniopharyngioma, Adamantinomatous Type','CNS/Brain','SOLID',26),(445,'SCCE','Teal',2,'Cervical Cancer','Small Cell Carcinoma of the Cervix','Cervix','SOLID',80),(446,'NPTLTFH','LimeGreen',5,'Mature T and NK Neoplasms','Nodal Peripheral T-Cell Lymphoma with TFH Phenotype','Lymphoid','LIQUID',243),(447,'VMM','Purple',2,'Melanoma','Mucosal Melanoma of the Vulva/Vagina','Vulva/Vagina','SOLID',853),(448,'MGST','LightYellow',2,'Malignant Glomus Tumor','Malignant Glomus Tumor','Soft Tissue','SOLID',683),(449,'BTMOV','LightBlue',4,'Ovarian Cancer','Brenner Tumor, Malignant','Ovary/Fallopian Tube','SOLID',602),(450,'CHDM','White',2,'Bone Cancer','Chordoma','Bone','SOLID',389),(451,'ACPP','Gray',3,'Choroid Plexus Tumor','Atypical Choroid Plexus Papilloma','CNS/Brain','SOLID',746),(452,'PSTAD','LightSkyBlue',4,'Esophagogastric Cancer','Papillary Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',620),(453,'MEL','Black',2,'Melanoma','Melanoma','Skin','SOLID',543),(454,'CM','Green',3,'Melanoma','Conjunctival Melanoma','Eye','SOLID',93),(455,'HGNEE','LightSkyBlue',3,'Gastrointestinal Neuroendocrine Tumors of the Esophagus/Stomach','High-Grade Neuroendocrine Carcinoma of the Esophagus','Esophagus/Stomach','SOLID',331),(456,'MAMPCA','Purple',3,'Ampullary Cancer','Mixed Ampullary Carcinoma','Ampulla of Vater','SOLID',732),(457,'PRCC','Orange',4,'Renal Cell Carcinoma','Papillary Renal Cell Carcinoma','Kidney','SOLID',380),(458,'BLPT','HotPink',4,'Breast Sarcoma','Borderline Phyllodes Tumor of the Breast','Breast','SOLID',442),(459,'LUPC','Gainsboro',3,'Non-Small Cell Lung Cancer','Pleomorphic Carcinoma of the Lung','Lung','SOLID',83),(460,'VMGCT','Purple',3,'Germ Cell Tumor','Mixed Germ Cell Tumor','Vulva/Vagina','SOLID',383),(461,'GCLC','Gainsboro',4,'Non-Small Cell Lung Cancer','Giant Cell Carcinoma of the Lung','Lung','SOLID',342),(462,'OFMT','LightYellow',3,'Soft Tissue Sarcoma','Ossifying Fibromyxoid Tumor','Soft Tissue','SOLID',363),(463,'BPDCN','LightSalmon',3,'Blastic Plasmacytoid Dendritic Cell Neoplasm','Blastic Plasmacytoid Dendritic Cell Neoplasm','Myeloid','LIQUID',174),(464,'PACT','Purple',2,'Pancreatic Cancer','Cystic Tumor of the Pancreas','Pancreas','SOLID',525),(465,'EHCH','Green',3,'Hepatobiliary Cancer','Extrahepatic Cholangiocarcinoma','Biliary Tract','SOLID',283),(466,'LUSC','Gainsboro',3,'Non-Small Cell Lung Cancer','Lung Squamous Cell Carcinoma','Lung','SOLID',83),(467,'FLC','MediumSeaGreen',2,'Hepatobiliary Cancer','Fibrolamellar Carcinoma','Liver','SOLID',567),(468,'TAML','LightSalmon',5,'Leukemia','Therapy-Related Acute Myeloid Leukemia','Myeloid','LIQUID',418),(469,'CMC','SaddleBrown',2,'Colorectal Cancer','Medullary Carcinoma of the Colon','Bowel','SOLID',289),(470,'APTAD','Gray',3,'Sellar Tumor','Atypical Pituitary Adenoma','CNS/Brain','SOLID',26),(471,'ATM','Gray',3,'CNS Cancer','Atypical Meningioma','CNS/Brain','SOLID',750),(472,'MYCF','LimeGreen',5,'Mature T and NK Neoplasms','Mycosis Fungoides','Lymphoid','LIQUID',243),(473,'MASCC','HotPink',4,'Breast Cancer','Metaplastic Adenocarcinoma with Spindle Cell Differentiation','Breast','SOLID',839),(474,'CHOM','Gray',3,'CNS Cancer','Chordoid Meningioma','CNS/Brain','SOLID',750),(475,'LUNG','Gainsboro',1,'Lung Cancer, NOS','Lung','Lung','SOLID',491),(476,'ALT','LightYellow',2,'Soft Tissue Sarcoma','Atypical Lipomatous Tumor','Soft Tissue','SOLID',683),(477,'VPDC','Purple',2,'Vaginal Cancer','Poorly Differentiated Vaginal Carcinoma','Vulva/Vagina','SOLID',853),(478,'AFX','Black',2,'Skin Cancer, Non-Melanoma','Atypical Fibroxanthoma','Skin','SOLID',543),(479,'MBOV','LightBlue',3,'Ovarian Cancer','Mucinous Borderline Ovarian Tumor','Ovary/Fallopian Tube','SOLID',168),(480,'PMA','Gray',3,'Glioma','Pilomyxoid Astrocytoma','CNS/Brain','SOLID',54),(481,'PSCC','Blue',2,'Penile Cancer','Penile Squamous Cell Carcinoma','Penis','SOLID',737),(482,'CEAIS','Teal',2,'Cervical Cancer','Cervical Adenocarcinoma In Situ','Cervix','SOLID',80),(483,'ODYS','LightBlue',3,'Germ Cell Tumor','Dysgerminoma','Ovary/Fallopian Tube','SOLID',84),(484,'FHRCC','Orange',4,'Renal Cell Carcinoma','FH-Deficient Renal Cell Carcinoma','Kidney','SOLID',380),(485,'PD','HotPink',3,'Breast Cancer','Paget Disease of the Nipple','Breast','SOLID',315),(486,'FRCT','LightSalmon',4,'Histiocytosis','Fibroblastic Reticular Cell Tumor','Myeloid','LIQUID',352),(487,'AUL','LightSalmon',4,'Leukemia','Acute Undifferentiated Leukemia','Myeloid','LIQUID',734),(488,'TLGL','LimeGreen',5,'Mature T and NK Neoplasms','T-Cell Large Granular Lymphocytic Leukemia','Lymphoid','LIQUID',243),(489,'ARMM','SaddleBrown',2,'Melanoma','Anorectal Mucosal Melanoma','Bowel','SOLID',289),(490,'DLBCLNOS','LimeGreen',5,'Mature B-Cell Neoplasms','Diffuse Large B-Cell Lymphoma, NOS','Lymphoid','LIQUID',530),(491,'TISSUE','',0,'','Tissue','','MIXED',NULL),(492,'PCLPD','LimeGreen',5,'Mature T and NK Neoplasms','Primary Cutaneous CD30 Positive T-Cell Lymphoproliferative Disorders','Lymphoid','LIQUID',243),(493,'PMFOFS','LightSalmon',5,'Myeloproliferative Neoplasms','Primary Myelofibrosis,Overt Fibrotic Stage','Myeloid','LIQUID',632),(494,'PCNSM','LightSkyBlue',3,'Melanoma','Primary CNS Melanoma','CNS/Brain','SOLID',96),(495,'THME','Teal',2,'Thyroid Cancer','Medullary Thyroid Cancer','Thyroid','SOLID',648),(496,'IDC','HotPink',3,'Breast Cancer','Breast Invasive Ductal Carcinoma','Breast','SOLID',338),(497,'TSTAD','LightSkyBlue',4,'Esophagogastric Cancer','Tubular Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',620),(498,'NSCLCPD','Gainsboro',3,'Non-Small Cell Lung Cancer','Poorly Differentiated Non-Small Cell Lung Cancer','Lung','SOLID',83),(499,'UNEC','PeachPuff',3,'Endometrial Cancer','Uterine Neuroendocrine Carcinoma','Uterus','SOLID',687),(500,'CHL','LimeGreen',4,'Hodgkin Lymphoma','Classical Hodgkin Lymphoma','Lymphoid','LIQUID',727),(501,'OSMAD','LightBlue',3,'Ovarian Cancer','Ovarian Seromucinous Adenoma','Ovary/Fallopian Tube','SOLID',168),(502,'BLCA','Yellow',2,'Bladder Cancer','Bladder Urothelial Carcinoma','Bladder/Urinary Tract','SOLID',108),(503,'CSNOS','HotPink',3,'Breast Cancer','Breast Invasive Carcinosarcoma, NOS','Breast','SOLID',338),(504,'PPTLD','LimeGreen',4,'Posttransplant Lymphoproliferative Disorders','Polymorphic PTLD','Lymphoid','LIQUID',55),(505,'USCC','Yellow',3,'Bladder Cancer','Urethral Squamous Cell Carcinoma','Bladder/Urinary Tract','SOLID',647),(506,'CELI','Teal',2,'Cervical Cancer','Cervical Leiomyosarcoma','Cervix','SOLID',80),(507,'EMCHS','White',3,'Bone Cancer','Extraskeletal Myxoid Chondrosarcoma','Bone','SOLID',742),(508,'HHV8DLBCL','LimeGreen',5,'Mature B-Cell Neoplasms','HHV8 Positive DLBCL, NOS','Lymphoid','LIQUID',530),(509,'ONBL','Gray',3,'Embryonal Tumor','Olfactory Neuroblastoma','CNS/Brain','SOLID',347),(510,'HGONEC','LightBlue',3,'Ovarian Cancer','High-Grade Neuroendocrine Carcinoma of the Ovary','Ovary/Fallopian Tube','SOLID',861),(511,'SPZM','Black',3,'Melanoma','Spitzoid Melanoma','Skin','SOLID',453),(512,'IMTL','Gainsboro',2,'Non-Small Cell Lung Cancer','Inflammatory Myofibroblastic Lung Tumor','Lung','SOLID',475),(513,'BLLIL3IGH','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with t(5;14)(q31.1;q32.3) IL3-IGH','Lymphoid','LIQUID',45),(514,'PBS','HotPink',2,'Breast Sarcoma','Breast Sarcoma','Breast','SOLID',799),(515,'UMC','PeachPuff',3,'Endometrial Cancer','Uterine Mucinous Carcinoma','Uterus','SOLID',687),(516,'VMT','Purple',3,'Germ Cell Tumor','Mature Teratoma','Vulva/Vagina','SOLID',383),(517,'SMMCL','LightSalmon',5,'Mastocytosis','Mast Cell Leukemia','Myeloid','LIQUID',204),(518,'MPC','LightYellow',2,'Soft Tissue Sarcoma','Myopericytoma','Soft Tissue','SOLID',683),(519,'ASTB','Gray',3,'Miscellaneous Neuroepithelial Tumor','Astroblastoma','CNS/Brain','SOLID',420),(520,'ICEMU','Teal',4,'Cervical Cancer','Intestinal Type Mucinous Carcinoma','Cervix','SOLID',830),(521,'DDCHS','White',3,'Bone Cancer','Dedifferentiated Chondrosarcoma','Bone','SOLID',742),(522,'SACA','DarkRed',2,'Salivary Gland Cancer','Salivary Carcinoma','Head and Neck','SOLID',303),(523,'PADA','DarkRed',3,'Salivary Gland Cancer','Pleomorphic Adenoma','Head and Neck','SOLID',522),(524,'PLRMS','LightYellow',3,'Soft Tissue Sarcoma','Pleomorphic Rhabdomyosarcoma','Soft Tissue','SOLID',211),(525,'PANCREAS','Purple',1,'Pancreatic Cancer, NOS','Pancreas','Pancreas','SOLID',491),(526,'HNMUCM','DarkRed',2,'Melanoma','Head and Neck Mucosal Melanoma','Head and Neck','SOLID',303),(527,'AML','LightSalmon',3,'Leukemia','Acute Myeloid Leukemia','Myeloid','LIQUID',174),(528,'RWDNET','SaddleBrown',3,'Gastrointestinal Neuroendocrine Tumor','Well-Differentiated Neuroendocrine Tumor of the Rectum','Bowel','SOLID',399),(529,'CCOC','DarkRed',4,'Head and Neck Cancer','Clear Cell Odontogenic Carcinoma','Head and Neck','SOLID',340),(530,'MBN','LimeGreen',4,'Mature B-Cell Neoplasms','Mature B-Cell Neoplasms','Lymphoid','LIQUID',576),(531,'HGBCLMYCBCL2','LimeGreen',5,'Mature B-Cell Neoplasms','High-Grade B-Cell Lymphoma, with MYC and BCL2 and/or BCL6 Rearrangements','Lymphoid','LIQUID',530),(532,'LNET','Gainsboro',2,'Non-Small Cell Lung Cancer','Lung Neuroendocrine Tumor','Lung','SOLID',475),(533,'CLLSLL','LimeGreen',5,'Mature B-Cell Neoplasms','Chronic Lymphocytic Leukemia/Small Lymphocytic Lymphoma','Lymphoid','LIQUID',530),(534,'LPL','LimeGreen',5,'Mature B-Cell Neoplasms','Lymphoplasmacytic Lymphoma','Lymphoid','LIQUID',530),(535,'LUACC','Gainsboro',4,'Non-Small Cell Lung Cancer','Adenoid Cystic Carcinoma of the Lung','Lung','SOLID',287),(536,'MAAP','SaddleBrown',3,'Appendiceal Cancer','Mucinous Adenocarcinoma of the Appendix','Bowel','SOLID',566),(537,'AMKL','LightSalmon',5,'Leukemia','Acute Megakaryoblastic Leukemia','Myeloid','LIQUID',601),(538,'ABC','LimeGreen',6,'Mature B-Cell Neoplasms','Activated B-cell Type','Lymphoid','LIQUID',490),(539,'PMHE','LightYellow',2,'Soft Tissue Sarcoma','Pseudomyogenic Hemangioendothelioma','Soft Tissue','SOLID',683),(540,'PRNET','Gray',3,'Miscellaneous Brain Tumor','Primary Neuroepithelial Tumor','CNS/Brain','SOLID',587),(541,'STOMACH','LightSkyBlue',1,'Esophageal/Stomach Cancer, NOS','Esophagus/Stomach','Esophagus/Stomach','SOLID',491),(542,'BA','HotPink',3,'Breast Sarcoma','Breast Angiosarcoma','Breast','SOLID',514),(543,'SKIN','Black',1,'Skin Cancer, NOS','Skin','Skin','SOLID',491),(544,'HGNET','Gray',3,'Miscellaneous Brain Tumor','High-Grade Neuroepithelial Tumor','CNS/Brain','SOLID',587),(545,'CELNOS','LightSalmon',4,'Myeloproliferative Neoplasms','Chronic Eosinophilic Leukemia, NOS','Myeloid','LIQUID',185),(546,'ECAD','Teal',3,'Cervical Cancer','Endocervical Adenocarcinoma','Cervix','SOLID',159),(547,'SOC','LightBlue',3,'Ovarian Cancer','Serous Ovarian Cancer','Ovary/Fallopian Tube','SOLID',168),(548,'BLSC','Yellow',2,'Bladder Cancer','Bladder Squamous Cell Carcinoma','Bladder/Urinary Tract','SOLID',108),(549,'MBL','Gray',3,'Embryonal Tumor','Medulloblastoma','CNS/Brain','SOLID',347),(550,'CHBL','White',2,'Bone Cancer','Chondroblastoma','Bone','SOLID',389),(551,'ES','White',2,'Bone Cancer','Ewing Sarcoma','Bone','SOLID',389),(552,'UCP','Purple',2,'Pancreatic Cancer','Undifferentiated Carcinoma of the Pancreas','Pancreas','SOLID',525),(553,'GBC','Green',2,'Hepatobiliary Cancer','Gallbladder Cancer','Biliary Tract','SOLID',393),(554,'JXG','LightSalmon',4,'Histiocytosis','Disseminated Juvenile Xanthogranuloma','Myeloid','LIQUID',352),(555,'DIA','Gray',3,'Miscellaneous Neuroepithelial Tumor','Desmoplastic Infantile Astrocytoma','CNS/Brain','SOLID',420),(556,'PBL','Gray',3,'Pineal Tumor','Pineoblastoma','CNS/Brain','SOLID',266),(557,'OVARY','LightBlue',1,'Ovarian/Fallopian Tube Cancer, NOS','Ovary/Fallopian Tube','Ovary/Fallopian Tube','SOLID',491),(558,'PTCA','Gray',3,'Sellar Tumor','Pituitary Carcinoma','CNS/Brain','SOLID',26),(559,'SNA','DarkRed',3,'Head and Neck Cancer','Sinonasal Adenocarcinoma','Head and Neck','SOLID',4),(560,'CML','LightSalmon',4,'Myeloproliferative Neoplasms','Chronic Myelogenous Leukemia','Myeloid','LIQUID',185),(561,'UDDC','PeachPuff',3,'Endometrial Cancer','Uterine Dedifferentiated Carcinoma','Uterus','SOLID',687),(562,'TSCST','Red',2,'Sex Cord Stromal Tumor','Sex Cord Stromal Tumor','Testis','SOLID',623),(563,'HDCS','LightYellow',3,'Soft Tissue Sarcoma','Histiocytic Dendritic Cell Sarcoma','Soft Tissue','SOLID',837),(564,'MASC','HotPink',4,'Breast Cancer','Metaplastic Adenosquamous Carcinoma','Breast','SOLID',839),(565,'AOAST','Gray',3,'Glioma','Anaplastic Oligoastrocytoma','CNS/Brain','SOLID',619),(566,'APAD','SaddleBrown',2,'Appendiceal Cancer','Appendiceal Adenocarcinoma','Bowel','SOLID',289),(567,'LIVER','MediumSeaGreen',1,'Liver Cancer, NOS','Liver','Liver','SOLID',491),(568,'MCCE','Teal',2,'Cervical Cancer','Mixed Cervical Carcinoma','Cervix','SOLID',80),(569,'EMYOCA','DarkRed',3,'Head and Neck Cancer','Epithelial-Myoepithelial Carcinoma','Head and Neck','SOLID',4),(570,'LUAS','Gainsboro',3,'Non-Small Cell Lung Cancer','Lung Adenosquamous Carcinoma','Lung','SOLID',83),(571,'CEMN','Teal',3,'Cervical Cancer','Mesonephric Carcinoma','Cervix','SOLID',159),(572,'HNMASC','DarkRed',3,'Salivary Gland Cancer','Mammary Analogue Secretory Carcinoma of Salivary Gland Origin','Head and Neck','SOLID',522),(573,'BLCLC','Gainsboro',4,'Non-Small Cell Lung Cancer','Basaloid Large Cell Carcinoma of the Lung','Lung','SOLID',342),(574,'BIMT','Gray',3,'Germ Cell Tumor','Immature Teratoma','CNS/Brain','SOLID',238),(575,'MPTLD','LimeGreen',4,'Posttransplant Lymphoproliferative Disorders','Monomorphic PTLD (B- and T-/NK-cell types)','Lymphoid','LIQUID',55),(576,'NHL','LimeGreen',3,'Non-Hodgkin Lymphoma','Non-Hodgkin Lymphoma','Lymphoid','LIQUID',31),(577,'MNGLP','LightSalmon',3,'Myeloid Neoplasms with Germ Line Predisposition','Myeloid Neoplasms with Germ Line Predisposition','Myeloid','LIQUID',174),(578,'SMZL','LimeGreen',6,'Mature B-Cell Neoplasms','Splenic Marginal Zone Lymphoma','Lymphoid','LIQUID',643),(579,'CDRCC','Orange',4,'Renal Cell Carcinoma','Collecting Duct Renal Cell Carcinoma','Kidney','SOLID',380),(580,'LGCOS','White',3,'Bone Cancer','Low-Grade Central Osteosarcoma','Bone','SOLID',631),(581,'CHM','PeachPuff',4,'Gestational Trophoblastic Disease','Complete Hydatidiform Mole','Uterus','SOLID',354),(582,'RCSNOS','LightYellow',2,'Soft Tissue Sarcoma','Round Cell Sarcoma, NOS','Soft Tissue','SOLID',683),(583,'ISTAD','LightSkyBlue',4,'Esophagogastric Cancer','Intestinal Type Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',620),(584,'GHCD','LimeGreen',5,'Mature B-Cell Neoplasms','Gamma Heavy-Chain Disease','Lymphoid','LIQUID',530),(585,'SCEMU','Teal',4,'Cervical Cancer','Signet Ring Mucinous Carcinoma','Cervix','SOLID',830),(586,'MCL','LimeGreen',5,'Mature B-Cell Neoplasms','Mantle Cell Lymphoma','Lymphoid','LIQUID',530),(587,'MBT','Gray',2,'Miscellaneous Brain Tumor','Miscellaneous Brain Tumor','CNS/Brain','SOLID',644),(588,'PCLBCLLT','LimeGreen',5,'Mature B-Cell Neoplasms','Primary Cutaneous DLBCL, Leg Type','Lymphoid','LIQUID',530),(589,'CMML0','LightSalmon',5,'Myelodysplastic/Myeloproliferative Neoplasms','Chronic Myelomonocytic Leukemia-0','Myeloid','LIQUID',224),(590,'AMLGATA2MECOM','LightSalmon',5,'Leukemia','AML with inv(3)(q21.3q26.2) or t(3;3)(q21.3;q26.2); GATA2, MECOM','Myeloid','LIQUID',198),(591,'UTUC','Yellow',2,'Bladder Cancer','Upper Tract Urothelial Carcinoma','Bladder/Urinary Tract','SOLID',108),(592,'GBAD','Green',3,'Hepatobiliary Cancer','Gallbladder Adenocarcinoma, NOS','Biliary Tract','SOLID',553),(593,'STSC','LightSkyBlue',3,'Esophagogastric Cancer','Small Cell Carcinoma of the Stomach','Esophagus/Stomach','SOLID',361),(594,'SCB','Yellow',2,'Bladder Cancer','Sarcomatoid Carcinoma of the Urinary Bladder','Bladder/Urinary Tract','SOLID',108),(595,'UMEC','PeachPuff',3,'Endometrial Cancer','Uterine Mixed Endometrial Carcinoma','Uterus','SOLID',687),(596,'SCCRCC','Orange',4,'Renal Cell Carcinoma','Renal Clear Cell Carcinoma with Sarcomatoid Features','Kidney','SOLID',244),(597,'CMLBCRABL1','LightSalmon',5,'Myeloproliferative Neoplasms','Chronic Myeloid Leukemia, BCR-ABL1+','Myeloid','LIQUID',560),(598,'EPIS','LightYellow',2,'Soft Tissue Sarcoma','Epithelioid Sarcoma','Soft Tissue','SOLID',683),(599,'SCOAH','Gray',3,'Sellar Tumor','Spindle Cell Oncocytoma of the Adenohypophysis','CNS/Brain','SOLID',26),(600,'MCHSCNS','Gray',3,'Miscellaneous Brain Tumor','Mesenchymal Chondrosarcoma of the CNS','CNS/Brain','SOLID',587),(601,'AMLNOS','LightSalmon',4,'Leukemia','AML, NOS','Myeloid','LIQUID',527),(602,'BTOV','LightBlue',3,'Ovarian Cancer','Brenner Tumor','Ovary/Fallopian Tube','SOLID',168),(603,'ETANTR','Gray',3,'Embryonal Tumor','Embryonal Tumor with Abundant Neuropil and True Rosettes','CNS/Brain','SOLID',347),(604,'BPT','HotPink',4,'Breast Sarcoma','Benign Phyllodes Tumor of the Breast','Breast','SOLID',442),(605,'SAAD','DarkRed',3,'Salivary Gland Cancer','Salivary Adenocarcinoma','Head and Neck','SOLID',522),(606,'MDSSLD','LightSalmon',4,'Myelodysplastic Syndromes','MDS with Single Lineage Dysplasia','Myeloid','LIQUID',355),(607,'FDCS','LightYellow',4,'Soft Tissue Sarcoma','Follicular Dendritic Cell Sarcoma','Myeloid','LIQUID',352),(608,'RGNT','Gray',3,'Miscellaneous Neuroepithelial Tumor','Rosette-forming Glioneuronal Tumor of the Fourth Ventricle','CNS/Brain','SOLID',420),(609,'WPSCC','Blue',3,'Penile Cancer','Warty Penile Squamous Cell Carcinoma','Penis','SOLID',481),(610,'ACA','Purple',2,'Adrenocortical Adenoma','Adrenocortical Adenoma','Adrenal Gland','SOLID',38),(611,'ECD','LightSalmon',4,'Histiocytosis','Erdheim-Chester Disease','Myeloid','LIQUID',352),(612,'PAC','DarkRed',3,'Salivary Gland Cancer','Polymorphous Adenocarcinoma','Head and Neck','SOLID',522),(613,'SEBVTLC','LimeGreen',5,'Mature T and NK Neoplasms','Systemic EBV Positive T-Cell Lymphoma of Childhood','Lymphoid','LIQUID',243),(614,'UAS','PeachPuff',3,'Uterine Sarcoma','Uterine Adenosarcoma','Uterus','SOLID',114),(615,'LGFMS','LightYellow',2,'Soft Tissue Sarcoma','Low-Grade Fibromyxoid Sarcoma','Soft Tissue','SOLID',683),(616,'PB','Purple',2,'Pancreatic Cancer','Pancreatoblastoma','Pancreas','SOLID',525),(617,'SEM','Red',2,'Germ Cell Tumor','Seminoma','Testis','SOLID',623),(618,'HSTCL','LimeGreen',5,'Mature T and NK Neoplasms','Hepatosplenic T-cell Lymphoma','Lymphoid','LIQUID',243),(619,'DIFG','Gray',2,'Glioma','Diffuse Glioma','CNS/Brain','SOLID',644),(620,'STAD','LightSkyBlue',3,'Esophagogastric Cancer','Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',361),(621,'PCALCL','LimeGreen',6,'Mature T and NK Neoplasms','Primary Cutaneous Anaplastic Large Cell Lymphoma','Lymphoid','LIQUID',492),(622,'DTE','Black',2,'Skin Cancer, Non-Melanoma','Desmoplastic Trichoepithelioma','Skin','SOLID',543),(623,'TESTIS','Red',1,'Testicular Cancer, NOS','Testis','Testis','SOLID',491),(624,'CAEXPA','DarkRed',3,'Salivary Gland Cancer','Carcinoma ex Pleomorphic Adenoma','Head and Neck','SOLID',522),(625,'POCA','Black',2,'Skin Cancer, Non-Melanoma','Porocarcinoma/Spiroadenocarcinoma','Skin','SOLID',543),(626,'UESL','MediumSeaGreen',2,'Undifferentiated Embryonal Sarcoma of the Liver','Undifferentiated Embryonal Sarcoma of the Liver','Liver','SOLID',567),(627,'LCIS','HotPink',2,'Breast Cancer','Breast Lobular Carcinoma In Situ','Breast','SOLID',799),(628,'ANSC','SaddleBrown',2,'Anal Cancer','Anal Squamous Cell Carcinoma','Bowel','SOLID',289),(629,'UTERUS','PeachPuff',1,'Uterine Cancer, NOS','Uterus','Uterus','SOLID',491),(630,'SCLG','Green',3,'Lacrimal Gland Tumor','Squamous Cell Carcinoma of the Lacrimal Gland','Eye','SOLID',21),(631,'OS','White',2,'Bone Cancer','Osteosarcoma','Bone','SOLID',389),(632,'PMF','LightSalmon',4,'Myeloproliferative Neoplasms','Primary Myelofibrosis','Myeloid','LIQUID',185),(633,'MPALTNOS','LightSalmon',4,'Leukemia','Mixed Phenotype Acute Leukemia, T/Myeloid, NOS','Myeloid','LIQUID',734),(634,'SARCL','Gainsboro',2,'Non-Small Cell Lung Cancer','Sarcomatoid Carcinoma of the Lung','Lung','SOLID',475),(635,'ODG','Gray',3,'Glioma','Oligodendroglioma','CNS/Brain','SOLID',619),(636,'EP','LimeGreen',5,'Mature B-Cell Neoplasms','Extraosseous Plasmacytoma','Lymphoid','LIQUID',530),(637,'ESS','PeachPuff',3,'Uterine Sarcoma','Endometrial Stromal Sarcoma','Uterus','SOLID',114),(638,'NFIB','Gray',3,'Nerve Sheath Tumor','Neurofibroma','Peripheral Nervous System','SOLID',242),(639,'MRLS','LightYellow',3,'Soft Tissue Sarcoma','Myxoid/Round-Cell Liposarcoma','Soft Tissue','SOLID',313),(640,'GCTB','White',2,'Bone Cancer','Giant Cell Tumor of Bone','Bone','SOLID',389),(641,'GS','LightYellow',2,'Soft Tissue Sarcoma','Glomangiosarcoma','Soft Tissue','SOLID',683),(642,'GIST','LightYellow',2,'Gastrointestinal Stromal Tumor','Gastrointestinal Stromal Tumor','Soft Tissue','SOLID',683),(643,'MZL','LimeGreen',5,'Mature B-Cell Neoplasms','Marginal Zone Lymphoma','Lymphoid','LIQUID',530),(644,'BRAIN','Gray',1,'CNS/Brain Cancer, NOS','CNS/Brain','CNS/Brain','SOLID',491),(645,'PCM','LimeGreen',5,'Mature B-Cell Neoplasms','Plasma Cell Myeloma','Lymphoid','LIQUID',530),(646,'SPIR','Black',2,'Skin Cancer, Non-Melanoma','Spiroma/Spiradenoma','Skin','SOLID',543),(647,'UCA','Yellow',2,'Bladder Cancer','Urethral Cancer','Bladder/Urinary Tract','SOLID',108),(648,'THYROID','Teal',1,'Thyroid Cancer, NOS','Thyroid','Thyroid','SOLID',491),(649,'MPT','HotPink',4,'Breast Sarcoma','Malignant Phyllodes Tumor of the Breast','Breast','SOLID',442),(650,'MLNPDGFRB','LightSalmon',4,'Leukemia','Myeloid/Lymphoid Neoplasms with PDGFRB Rearrangement','Myeloid','LIQUID',693),(651,'PSTT','PeachPuff',3,'Gestational Trophoblastic Disease','Placental Site Trophoblastic Tumor','Uterus','SOLID',417),(652,'BRCNOS','HotPink',3,'Breast Cancer','Breast Invasive Carcinoma, NOS','Breast','SOLID',338),(653,'PPB','Gainsboro',2,'Non-Small Cell Lung Cancer','Pleuropulmonary Blastoma','Lung','SOLID',475),(654,'OSMCA','LightBlue',3,'Ovarian Cancer','Ovarian Seromucinous Carcinoma','Ovary/Fallopian Tube','SOLID',168),(655,'BFN','HotPink',2,'Breast Sarcoma','Breast Fibroepithelial Neoplasms','Breast','SOLID',799),(656,'LCS','LightSalmon',4,'Histiocytosis','Langerhans Cell Sarcoma','Myeloid','LIQUID',352),(657,'SCOS','White',3,'Bone Cancer','Small Cell Osteosarcoma','Bone','SOLID',631),(658,'SMAHN','LightSalmon',5,'Mastocytosis','Systemic Mastocytosis with an Associated Hematological Neoplasm','Myeloid','LIQUID',204),(659,'ANKL','LimeGreen',5,'Mature T and NK Neoplasms','Aggressive NK-Cell Leukemia','Lymphoid','LIQUID',243),(660,'AA','LightYellow',2,'Soft Tissue Sarcoma','Aggressive Angiomyxoma','Soft Tissue','SOLID',683),(661,'GBASC','Green',3,'Hepatobiliary Cancer','Adenosquamous Carcinoma of the Gallbladder','Biliary Tract','SOLID',553),(662,'LRCHL','LimeGreen',5,'Hodgkin Lymphoma','Lymphocyte-Rich Classical Hodgkin Lymphoma','Lymphoid','LIQUID',500),(663,'ATRT','Gray',3,'Embryonal Tumor','Atypical Teratoid/Rhabdoid Tumor','CNS/Brain','SOLID',347),(664,'CLPDNK','LimeGreen',5,'Mature T and NK Neoplasms','Chronic Lymphoproliferative Disorder of NK Cells','Lymphoid','LIQUID',243),(665,'PRNE','Cyan',2,'Prostate Cancer','Prostate Neuroendocrine Carcinoma','Prostate','SOLID',374),(666,'GB','Gray',3,'Glioma','Glioblastoma','CNS/Brain','SOLID',619),(667,'LM','LightYellow',2,'Soft Tissue Cancer','Leiomyoma','Soft Tissue','SOLID',683),(668,'FT','LightBlue',3,'Sex Cord Stromal Tumor','Fibrothecoma','Ovary/Fallopian Tube','SOLID',8),(669,'GNG','Gray',3,'Glioma','Ganglioglioma','CNS/Brain','SOLID',54),(670,'DFL','LimeGreen',6,'Mature B-Cell Neoplasms','Duodenal-Type Follicular Lymphoma','Lymphoid','LIQUID',286),(671,'AMLMD','LightSalmon',5,'Leukemia','AML with Minimal Differentiation','Myeloid','LIQUID',601),(672,'MPALBNOS','LightSalmon',4,'Leukemia','Mixed Phenotype Acute Leukemia, B/Myeloid, NOS','Myeloid','LIQUID',734),(673,'SARCNOS','LightYellow',2,'Soft Tissue Sarcoma','Sarcoma, NOS','Soft Tissue','SOLID',683),(674,'FA','HotPink',3,'Breast Sarcoma','Fibroadenoma','Breast','SOLID',655),(675,'MHCD','LimeGreen',5,'Mature B-Cell Neoplasms','Mu Heavy-Chain Disease','Lymphoid','LIQUID',530),(676,'BLAD','Yellow',2,'Bladder Cancer','Bladder Adenocarcinoma','Bladder/Urinary Tract','SOLID',108),(677,'EOV','LightBlue',3,'Ovarian Cancer','Endometrioid Ovarian Cancer','Ovary/Fallopian Tube','SOLID',168),(678,'STMYEC','LightYellow',2,'Soft Tissue Sarcoma','Soft Tissue Myoepithelial Carcinoma','Soft Tissue','SOLID',683),(679,'AGNG','Gray',3,'Glioma','Anaplastic Ganglioglioma','CNS/Brain','SOLID',54),(680,'RCYC','LightSalmon',4,'Myelodysplastic Syndromes','Refractory Cytopenia of Childhood','Myeloid','LIQUID',355),(681,'LYP','LimeGreen',6,'Mature T and NK Neoplasms','Lymphomatoid Papulosis','Lymphoid','LIQUID',492),(682,'MSCC','HotPink',4,'Breast Cancer','Metaplastic Squamous Cell Carcinoma','Breast','SOLID',839),(683,'SOFT_TISSUE','LightYellow',1,'Soft Tissue Cancer, NOS','Soft Tissue','Soft Tissue','SOLID',491),(684,'PTHC','DarkRed',3,'Parathyroid Cancer','Parathyroid Carcinoma','Head and Neck','SOLID',288),(685,'CEAS','Teal',2,'Cervical Cancer','Cervical Adenosquamous Carcinoma','Cervix','SOLID',80),(686,'THYMUS','Purple',1,'Thymic Cancer, NOS','Thymus','Thymus','SOLID',491),(687,'UCEC','PeachPuff',2,'Endometrial Cancer','Endometrial Carcinoma','Uterus','SOLID',629),(688,'BLLBCRABL1L','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma, BCR-ABL1 Like','Lymphoid','LIQUID',45),(689,'CSCLC','Gainsboro',2,'Small Cell Lung Cancer','Combined Small Cell Lung Carcinoma','Lung','SOLID',475),(690,'PAMPCA','Purple',3,'Ampullary Cancer','Pancreatobiliary Ampullary Carcinoma','Ampulla of Vater','SOLID',732),(691,'MLNPCM1JAK2','LightSalmon',4,'Leukemia','Myeloid/Lymphoid Neoplasms with PCM1-JAK2','Myeloid','LIQUID',693),(692,'JMML','LightSalmon',4,'Myelodysplastic/Myeloproliferative Neoplasms','Juvenile Myelomonocytic Leukemia','Myeloid','LIQUID',285),(693,'MLNER','LightSalmon',3,'Leukemia','Myeloid/Lymphoid Neoplasms with Eosinophilia and Rearrangement of  PDGFRA/PDGFRB or FGFR1 or with PCM1-JAK2','Myeloid','LIQUID',174),(694,'PCGDTCL','LimeGreen',5,'Mature T and NK Neoplasms','Primary Cutaneous Gamma Delta T-Cell Lymphoma','Lymphoid','LIQUID',243),(695,'ET','LightSalmon',4,'Myeloproliferative Neoplasms','Essential Thrombocythemia','Myeloid','LIQUID',185),(696,'PHM','PeachPuff',4,'Gestational Trophoblastic Disease','Partial Hydatidiform Mole','Uterus','SOLID',354),(697,'CPC','Gray',3,'Choroid Plexus Tumor','Choroid Plexus Carcinoma','CNS/Brain','SOLID',746),(698,'SKCM','Black',3,'Melanoma','Cutaneous Melanoma','Skin','SOLID',453),(699,'ACC','Purple',2,'Adrenocortical Carcinoma','Adrenocortical Carcinoma','Adrenal Gland','SOLID',38),(700,'ADNOS','Black',3,'Cancer of Unknown Primary','Adenocarcinoma, NOS','Other','SOLID',712),(701,'TRCC','Orange',4,'Renal Cell Carcinoma','Translocation-Associated Renal Cell Carcinoma','Kidney','SOLID',380),(702,'SCCNOS','Black',3,'Cancer of Unknown Primary','Squamous Cell Carcinoma, NOS','Other','SOLID',712),(703,'SCSRMS','LightYellow',3,'Soft Tissue Sarcoma','Spindle Cell/Sclerosing Rhabdomyosarcoma','Soft Tissue','SOLID',211),(704,'CACC','Teal',2,'Cervical Cancer','Cervical Adenoid Cystic Carcinoma','Cervix','SOLID',80),(705,'IHM','PeachPuff',4,'Gestational Trophoblastic Disease','Invasive Hydatidiform Mole','Uterus','SOLID',354),(706,'HGBCL','LimeGreen',5,'Mature B-Cell Neoplasms','High-Grade B-Cell Lymphoma, NOS','Lymphoid','LIQUID',530),(707,'PLEURA','Blue',1,'Pleural Cancer, NOS','Pleura','Pleura','SOLID',491),(708,'VOEC','Purple',3,'Germ Cell Tumor','Embryonal Carcinoma','Vulva/Vagina','SOLID',383),(709,'MDSU','LightSalmon',4,'Myelodysplastic Syndromes','MDS, Unclassifiable','Myeloid','LIQUID',355),(710,'PPTID','Gray',3,'Pineal Tumor','Pineal Parenchymal Tumor of Intermediate Differentiation','CNS/Brain','SOLID',266),(711,'CCOV','LightBlue',3,'Ovarian Cancer','Clear Cell Ovarian Cancer','Ovary/Fallopian Tube','SOLID',168),(712,'CUP','Black',2,'Cancer of Unknown Primary','Cancer of Unknown Primary','Other','SOLID',716),(713,'GNOS','Gray',3,'Glioma','Glioma, NOS','CNS/Brain','SOLID',619),(714,'RAML','Orange',4,'Renal Cell Carcinoma','Renal Angiomyolipoma','Kidney','SOLID',380),(715,'HVLL','LimeGreen',5,'Mature T and NK Neoplasms','Hydroa Vacciniforme Like Lymphoproliferative Disorder','Lymphoid','LIQUID',243),(716,'OTHER','Black',1,'Other Cancer, NOS','Other','Other','SOLID',491),(717,'OCSC','DarkRed',3,'Head and Neck Cancer','Oral Cavity Squamous Cell Carcinoma','Head and Neck','SOLID',44),(718,'SDRPL','LimeGreen',6,'Mature B-Cell Neoplasms','Splenic Diffuse Red Pulp Small B-Cell Lymphoma','Lymphoid','LIQUID',3),(719,'ANGS','LightYellow',2,'Soft Tissue Sarcoma','Angiosarcoma','Soft Tissue','SOLID',683),(720,'GEJ','LightSkyBlue',3,'Esophagogastric Cancer','Adenocarcinoma of the Gastroesophageal Junction','Esophagus/Stomach','SOLID',361),(721,'CPP','Gray',3,'Choroid Plexus Tumor','Choroid Plexus Papilloma','CNS/Brain','SOLID',746),(722,'IVBCL','LimeGreen',5,'Mature B-Cell Neoplasms','Intravascular Large B-Cell Lymphoma','Lymphoid','LIQUID',530),(723,'NLPHL','LimeGreen',4,'Hodgkin Lymphoma','Nodular Lymphocyte-Predominant Hodgkin Lymphoma','Lymphoid','LIQUID',727),(724,'THYC','Purple',3,'Thymic Tumor','Thymic Carcinoma','Thymus','SOLID',291),(725,'ILC','HotPink',3,'Breast Cancer','Breast Invasive Lobular Carcinoma','Breast','SOLID',338),(726,'PCNSL','LimeGreen',5,'Mature B-Cell Neoplasms','Primary DLBCL of the central nervous system','Lymphoid','LIQUID',530),(727,'HL','LimeGreen',3,'Hodgkin Lymphoma','Hodgkin Lymphoma','Lymphoid','LIQUID',31),(728,'LAMN','SaddleBrown',2,'Appendiceal Cancer','Low-grade Appendiceal Mucinous Neoplasm','Bowel','SOLID',289),(729,'PBT','Gray',3,'Miscellaneous Brain Tumor','Primary Brain Tumor','CNS/Brain','SOLID',587),(730,'ULM','PeachPuff',4,'Uterine Sarcoma','Uterine Leiomyoma','Uterus','SOLID',754),(731,'ALCL','LimeGreen',5,'Mature T and NK Neoplasms','Anaplastic Large Cell Lymphoma','Lymphoid','LIQUID',243),(732,'AMPCA','Purple',2,'Ampullary Cancer','Ampullary Carcinoma','Ampulla of Vater','SOLID',103),(733,'RNET','Orange',2,'Renal Neuroendocrine Tumor','Renal Neuroendocrine Tumor','Kidney','SOLID',431),(734,'ALAL','LightSalmon',3,'Leukemia','Acute Leukemias of Ambiguous Lineage','Myeloid','LIQUID',174),(735,'DNT','Gray',3,'Glioma','Dysembryoplastic Neuroepithelial Tumor','CNS/Brain','SOLID',54),(736,'GRCT','LightBlue',3,'Sex Cord Stromal Tumor','Granulosa Cell Tumor','Ovary/Fallopian Tube','SOLID',8),(737,'PENIS','Blue',1,'Penile Cancer, NOS','Penis','Penis','SOLID',491),(738,'ACYC','DarkRed',3,'Salivary Gland Cancer','Adenoid Cystic Carcinoma','Head and Neck','SOLID',522),(739,'VPSCC','Blue',3,'Penile Cancer','Verrucous Penile Squamous Cell Carcinoma','Penis','SOLID',481),(740,'MCCHL','LimeGreen',5,'Hodgkin Lymphoma','Mixed Cellularity Classical Hodgkin Lymphoma','Lymphoid','LIQUID',500),(741,'CEEN','Teal',3,'Cervical Cancer','Cervical Endometrioid Carcinoma','Cervix','SOLID',159),(742,'CHS','White',2,'Bone Cancer','Chondrosarcoma','Bone','SOLID',389),(743,'BLLTCF3PBX1','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with t(1;19)(q23;p13.3);TCF3-PBX1','Lymphoid','LIQUID',45),(744,'BEC','Gray',3,'Germ Cell Tumor','Embryonal Carcinoma','CNS/Brain','SOLID',238),(745,'HGGNOS','Gray',3,'Glioma','High-Grade Glioma, NOS','CNS/Brain','SOLID',619),(746,'CPT','Gray',2,'Choroid Plexus Tumor','Choroid Plexus Tumor','CNS/Brain','SOLID',644),(747,'HPHSC','DarkRed',3,'Head and Neck Cancer','Hypopharynx Squamous Cell Carcinoma','Head and Neck','SOLID',44),(748,'OAT','Teal',2,'Thyroid Cancer','Oncocytic Adenoma of the Thyroid','Thyroid','SOLID',648),(749,'LIHB','MediumSeaGreen',2,'Hepatobiliary Cancer','Hepatoblastoma','Liver','SOLID',567),(750,'MNGT','Gray',2,'CNS Cancer','Meningothelial Tumor','CNS/Brain','SOLID',644),(751,'IMPTLD','LimeGreen',4,'Posttransplant Lymphoproliferative Disorders','Infectious Mononucleosis PTLD','Lymphoid','LIQUID',55),(752,'CCBOV','LightBlue',3,'Ovarian Cancer','Clear Cell Borderline Ovarian Tumor','Ovary/Fallopian Tube','SOLID',168),(753,'DIPG','Gray',3,'Glioma','Diffuse Intrinsic Pontine Glioma','CNS/Brain','SOLID',619),(754,'USMT','PeachPuff',3,'Uterine Sarcoma','Uterine Smooth Muscle Tumor','Uterus','SOLID',114),(755,'PMFPES','LightSalmon',5,'Myeloproliferative Neoplasms','Primary Myelofibrosis, Prefibrotic/Early Stage','Myeloid','LIQUID',632),(756,'ULMS','PeachPuff',4,'Uterine Sarcoma','Uterine Leiomyosarcoma','Uterus','SOLID',754),(757,'GRC','LightSkyBlue',3,'Esophagogastric Cancer','Gastric Remnant Adenocarcinoma','Esophagus/Stomach','SOLID',361),(758,'OUSARC','PeachPuff',3,'Uterine Sarcoma','Uterine Sarcoma, Other','Uterus','SOLID',114),(759,'DLBCLCI','LimeGreen',5,'Mature B-Cell Neoplasms','DLBCL Associated with Chronic Inflammation','Lymphoid','LIQUID',530),(760,'CENE','Teal',2,'Cervical Cancer','Cervical Neuroendocrine Tumor','Cervix','SOLID',80),(761,'LUCA','Gainsboro',3,'Non-Small Cell Lung Cancer','Lung Carcinoid','Lung','SOLID',532),(762,'TEOS','White',3,'Bone Cancer','Telangiectatic Osteosarcoma','Bone','SOLID',631),(763,'MBEN','Gray',3,'Embryonal Tumor','Medulloblastoma with Extensive Nodularity','CNS/Brain','SOLID',347),(764,'THYM','Purple',3,'Thymic Tumor','Thymoma','Thymus','SOLID',291),(765,'ACLG','Green',3,'Lacrimal Gland Tumor','Adenoid Cystic Carcinoma of the Lacrimal Gland','Eye','SOLID',21),(766,'THFO','Teal',3,'Thyroid Cancer','Follicular Thyroid Cancer','Thyroid','SOLID',132),(767,'EPDCA','LightSkyBlue',2,'Esophagogastric Cancer','Esophageal Poorly Differentiated Carcinoma','Esophagus/Stomach','SOLID',541),(768,'ADMA','White',2,'Bone Cancer','Adamantinoma','Bone','SOLID',389),(769,'PECOMA','LightYellow',2,'Soft Tissue Sarcoma','Perivascular Epithelioid Cell Tumor','Soft Tissue','SOLID',683),(770,'MDSEB1','LightSalmon',5,'Myelodysplastic Syndromes','MDS with excess blasts-1','Myeloid','LIQUID',362),(771,'UUS','PeachPuff',3,'Uterine Sarcoma','Undifferentiated Uterine Sarcoma','Uterus','SOLID',114),(772,'BLLNOS','LimeGreen',4,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma, NOS','Lymphoid','LIQUID',95),(773,'ACCC','DarkRed',3,'Salivary Gland Cancer','Acinic Cell Carcinoma','Head and Neck','SOLID',522),(774,'SKCN','Black',3,'Melanoma','Congenital Nevus','Skin','SOLID',453),(775,'BLLKMT2A','LimeGreen',5,'B-Lymphoblastic Leukemia/Lymphoma','B-Lymphoblastic Leukemia/Lymphoma with t(v;11q23.3);KMT2A Rearranged','Lymphoid','LIQUID',45),(776,'MMBC','HotPink',3,'Breast Cancer','Mixed Type Metaplastic Breast Cancer','Breast','SOLID',10),(777,'GN','Gray',2,'Peripheral Nervous System','Ganglioneuroma','Peripheral Nervous System','SOLID',215),(778,'SCGBM','Gray',4,'Glioma','Small Cell Glioblastoma','CNS/Brain','SOLID',666),(779,'PCAECTCL','LimeGreen',5,'Mature T and NK Neoplasms','Primary Cutaneous CD8 Positive Aggressive Epidermotropic Cytotoxic T-Cell Lymphoma','Lymphoid','LIQUID',243),(780,'PGNT','Gray',3,'Miscellaneous Neuroepithelial Tumor','Papillary Glioneuronal Tumor','CNS/Brain','SOLID',420),(781,'ESCC','LightSkyBlue',2,'Esophagogastric Cancer','Esophageal Squamous Cell Carcinoma','Esophagus/Stomach','SOLID',541),(782,'SBMOV','LightBlue',3,'Ovarian Cancer','Serous Borderline Ovarian Tumor, Micropapillary','Ovary/Fallopian Tube','SOLID',168),(783,'IMMC','HotPink',3,'Breast Cancer','Breast Invasive Mixed Mucinous Carcinoma','Breast','SOLID',338),(784,'MS','LightSalmon',4,'Leukemia','Myeloid Sarcoma','Myeloid','LIQUID',527),(785,'RAS','LightYellow',2,'Soft Tissue Sarcoma','Radiation-Associated Sarcoma','Soft Tissue','SOLID',683),(786,'SBWDNET','SaddleBrown',3,'Gastrointestinal Neuroendocrine Tumor','Small Bowel Well-Differentiated Neuroendocrine Tumor','Bowel','SOLID',399),(787,'MF','LightYellow',2,'Soft Tissue Sarcoma','Myofibroma','Soft Tissue','SOLID',683),(788,'LIAS','MediumSeaGreen',2,'Hepatobiliary Cancer','Liver Angiosarcoma','Liver','SOLID',567),(789,'AMLNPM1','LightSalmon',5,'Leukemia','AML with Mutated NPM1','Myeloid','LIQUID',198),(790,'UCCC','PeachPuff',3,'Endometrial Cancer','Uterine Clear Cell Carcinoma','Uterus','SOLID',687),(791,'AMLRUNX1RUNX1T1','LightSalmon',5,'Leukemia','AML with t(8;21)(q22;q22.1);RUNX1-RUNX1T1','Myeloid','LIQUID',198),(792,'VYST','Purple',3,'Germ Cell Tumor','Yolk Sac Tumor','Vulva/Vagina','SOLID',383),(793,'HPCCNS','Gray',3,'CNS Cancer','Hemangiopericytoma of the Central Nervous System','CNS/Brain','SOLID',750),(794,'OUTT','PeachPuff',2,'Endometrial Cancer','Other Uterine Tumor','Uterus','SOLID',629),(795,'IDCS','LightYellow',4,'Soft Tissue Sarcoma','Interdigitating Dendritic Cell Sarcoma','Myeloid','LIQUID',352),(796,'LGNET','Gray',3,'Miscellaneous Brain Tumor','Low-Grade Neuroepithelial Tumor','CNS/Brain','SOLID',587),(797,'PLEMESO','Blue',3,'Mesothelioma','Pleural Mesothelioma, Epithelioid Type','Pleura','SOLID',379),(798,'HCCIHCH','MediumSeaGreen',2,'Hepatobiliary Cancer','Hepatocellular Carcinoma plus Intrahepatic Cholangiocarcinoma','Liver','SOLID',567),(799,'BREAST','HotPink',1,'Breast Cancer, NOS','Breast','Breast','SOLID',491),(800,'LIAD','MediumSeaGreen',2,'Hepatobiliary Cancer','Hepatocellular Adenoma','Liver','SOLID',567),(801,'MDSMPNRST','LightSalmon',4,'Myelodysplastic/Myeloproliferative Neoplasms','MDS/MPN with Ring Sideroblasts and Thrombocytosis','Myeloid','LIQUID',285),(802,'ETT','PeachPuff',3,'Gestational Trophoblastic Disease','Epithelioid Trophoblastic Tumor','Uterus','SOLID',417),(803,'IMT','LightYellow',2,'Soft Tissue Sarcoma','Inflammatory Myofibroblastic Tumor','Soft Tissue','SOLID',683),(804,'PDC','Black',3,'Cancer of Unknown Primary','Poorly Differentiated Carcinoma, NOS','Other','SOLID',712),(805,'AMLRUNX1','LightSalmon',5,'Leukemia','AML with Mutated RUNX1','Myeloid','LIQUID',198),(806,'PV','LightSalmon',4,'Myeloproliferative Neoplasms','Polycythemia Vera','Myeloid','LIQUID',185),(807,'SBC','SaddleBrown',2,'Small Bowel Cancer','Small Bowel Cancer','Bowel','SOLID',289),(808,'CAIS','SaddleBrown',3,'Colorectal Cancer','Colon Adenocarcinoma In Situ','Bowel','SOLID',382),(809,'MXOV','LightBlue',3,'Ovarian Cancer','Mixed Ovarian Carcinoma','Ovary/Fallopian Tube','SOLID',168),(810,'MUP','Black',3,'Melanoma','Melanoma of Unknown Primary','Skin','SOLID',453),(811,'JSCB','HotPink',2,'Breast Cancer','Juvenile Secretory Carcinoma of the Breast','Breast','SOLID',799),(812,'UCU','Yellow',3,'Bladder Cancer','Urethral Urothelial Carcinoma','Bladder/Urinary Tract','SOLID',647),(813,'HGNES','LightSkyBlue',3,'Gastrointestinal Neuroendocrine Tumors of the Esophagus/Stomach','High-Grade Neuroendocrine Carcinoma of the Stomach','Esophagus/Stomach','SOLID',331),(814,'ISM','LightSalmon',5,'Mastocytosis','Indolent Systemic Mastocytosis','Myeloid','LIQUID',204),(815,'MATPL','LightSalmon',2,'Blood Cancer, NOS','Myeloid Atypical','Myeloid','LIQUID',387),(816,'MPNST','Gray',3,'Nerve Sheath Tumor','Malignant Peripheral Nerve Sheath Tumor','Peripheral Nervous System','SOLID',242),(817,'PTPR','Gray',3,'Pineal Tumor','Papillary Tumor of the Pineal Region','CNS/Brain','SOLID',266),(818,'PCATCL','LimeGreen',5,'Mature T and NK Neoplasms','Primary Cutaneous Acral CD8 Positive T-Cell Lymphoma','Lymphoid','LIQUID',243),(819,'PCFCL','LimeGreen',5,'Mature B-Cell Neoplasms','Primary Cutaneous Follicle Center Lymphoma','Lymphoid','LIQUID',530),(820,'DA','SaddleBrown',3,'Small Bowel Cancer','Duodenal Adenocarcinoma','Bowel','SOLID',807),(821,'CMML1','LightSalmon',5,'Myelodysplastic/Myeloproliferative Neoplasms','Chronic Myelomonocytic Leukemia-1','Myeloid','LIQUID',224),(822,'ABL','LightSalmon',5,'Leukemia','Acute Basophilic Leukemia','Myeloid','LIQUID',601),(823,'MYCHS','White',3,'Bone Cancer','Myxoid Chondrosarcoma','Bone','SOLID',742),(824,'CCSK','Orange',2,'Clear Cell Sarcoma of Kidney','Clear Cell Sarcoma of Kidney','Kidney','SOLID',431),(825,'MCS','HotPink',4,'Breast Cancer','Metaplastic Carcinosarcoma','Breast','SOLID',776),(826,'PLBMESO','Blue',3,'Mesothelioma','Pleural Mesothelioma, Biphasic Type','Pleura','SOLID',379),(827,'MYEC','DarkRed',3,'Salivary Gland Cancer','Myoepithelial Carcinoma','Head and Neck','SOLID',522),(828,'MIDD','LimeGreen',5,'Mature B-Cell Neoplasms','Monoclonal Immunoglobulin Deposition Diseases','Lymphoid','LIQUID',530),(829,'CEGCC','Teal',2,'Cervical Cancer','Glassy Cell Carcinoma of the Cervix','Cervix','SOLID',80),(830,'CEMU','Teal',3,'Cervical Cancer','Mucinous Carcinoma','Cervix','SOLID',159),(831,'MPNU','LightSalmon',4,'Myeloproliferative Neoplasms','Myeloproliferative Neoplasms, Unclassifiable','Myeloid','LIQUID',185),(832,'EPMT','Gray',2,'CNS Cancer','Ependymomal Tumor','CNS/Brain','SOLID',644),(833,'GCT','Gray',3,'Sellar Tumor','Granular Cell Tumor','CNS/Brain','SOLID',26),(834,'PEOS','White',3,'Bone Cancer','Periosteal Osteosarcoma','Bone','SOLID',631),(835,'ACML','LightSalmon',4,'Myelodysplastic/Myeloproliferative Neoplasms','Atypical Chronic Myeloid Leukemia, BCR-ABL1-','Myeloid','LIQUID',285),(836,'SKAC','Black',2,'Skin Cancer, Non-Melanoma','Skin Adnexal Carcinoma','Skin','SOLID',543),(837,'DCS','LightYellow',2,'Soft Tissue Sarcoma','Dendritic Cell Sarcoma','Soft Tissue','SOLID',683),(838,'CCE','Gray',3,'CNS Cancer','Clear Cell Ependymoma','CNS/Brain','SOLID',832),(839,'EMBC','HotPink',3,'Breast Cancer','Epithelial Type Metaplastic Breast Cancer','Breast','SOLID',10),(840,'MDSRSMD','LightSalmon',5,'Myelodysplastic Syndromes','MDS with Ring Sideroblasts and Multilineage Dysplasia','Myeloid','LIQUID',267),(841,'BCC','Black',2,'Skin Cancer, Non-Melanoma','Basal Cell Carcinoma','Skin','SOLID',543),(842,'ISFN','LimeGreen',6,'Mature B-Cell Neoplasms','In Situ Follicular Neoplasia','Lymphoid','LIQUID',286),(843,'UPECOMA','PeachPuff',3,'Uterine Sarcoma','Uterine Perivascular Epithelioid Cell Tumor','Uterus','SOLID',114),(844,'MPRDS','LightSalmon',4,'Leukemia','Myeloid Proliferations Related to Down Syndrome','Myeloid','LIQUID',527),(845,'SCRMS','LightYellow',3,'Soft Tissue Sarcoma','Spindle Cell Rhabdomyosarcoma','Soft Tissue','SOLID',211),(846,'MCN','Purple',3,'Pancreatic Cancer','Mucinous Cystic Neoplasm','Pancreas','SOLID',464),(847,'GCCAP','SaddleBrown',3,'Appendiceal Cancer','Goblet Cell Carcinoid of the Appendix','Bowel','SOLID',566),(848,'AMBL','Gray',3,'Embryonal Tumor','Large Cell/Anaplastic Medulloblastoma','CNS/Brain','SOLID',347),(849,'CHGL','Gray',3,'Miscellaneous Neuroepithelial Tumor','Chordoid Glioma of the Third Ventricle','CNS/Brain','SOLID',420),(850,'RCC','Orange',2,'Renal Cell Carcinoma','Renal Cell Carcinoma','Kidney','SOLID',431),(851,'MGUSIGM','LimeGreen',6,'Mature B-Cell Neoplasms','IgM','Lymphoid','LIQUID',392),(852,'BRCANOS','HotPink',3,'Breast Cancer','Breast Invasive Cancer, NOS','Breast','SOLID',338),(853,'VULVA','Purple',1,'Vulvar/Vaginal Cancer, NOS','Vulva/Vagina','Vulva/Vagina','SOLID',491),(854,'SSRCC','LightSkyBlue',4,'Esophagogastric Cancer','Signet Ring Cell Carcinoma of the Stomach','Esophagus/Stomach','SOLID',256),(855,'DASTR','Gray',3,'Glioma','Diffuse Astrocytoma','CNS/Brain','SOLID',619),(856,'MSTAD','LightSkyBlue',4,'Esophagogastric Cancer','Mucinous Stomach Adenocarcinoma','Esophagus/Stomach','SOLID',620),(857,'ETPLL','LimeGreen',4,'T-Lymphoblastic Leukemia/Lymphoma','Early T-Cell Precursor Lymphoblastic Leukemia','Lymphoid','LIQUID',293),(858,'PTES','LightYellow',3,'Soft Tissue Sarcoma','Proximal-Type Epithelioid Sarcoma','Soft Tissue','SOLID',598),(859,'PPM','Gray',3,'CNS Cancer','Papillary Meningioma','CNS/Brain','SOLID',750),(860,'VSC','Purple',2,'Vaginal Cancer','Squamous Cell Carcinoma of the Vulva/Vagina','Vulva/Vagina','SOLID',853),(861,'OOVC','LightBlue',2,'Ovarian Cancer','Ovarian Cancer, Other','Ovary/Fallopian Tube','SOLID',557),(862,'LCH','LightSalmon',4,'Histiocytosis','Langerhans Cell Histiocytosis','Myeloid','LIQUID',352),(863,'PSEC','Green',2,'Peritoneal Cancer, NOS','Peritoneal Serous Carcinoma','Peritoneum','SOLID',129),(864,'AM','LightSalmon',5,'Leukemia','AML with Maturation','Myeloid','LIQUID',601),(865,'','Gray',0,'Embryonal Tumor','','CNS/Brain','SOLID',NULL),(866,'','LimeGreen',0,'Mature B-Cell Neoplasms','','Lymphoid','LIQUID',NULL),(867,'','DarkRed',0,'Head and Neck Cancer','','Head and Neck','SOLID',NULL),(868,'','White',0,'Bone Cancer','','Bone','SOLID',NULL),(869,'','LightSalmon',0,'Leukemia','','Myeloid','LIQUID',NULL),(870,'','LightYellow',0,'Soft Tissue Sarcoma','','MIXED','MIXED',NULL),(871,'','MIXED',0,'Sex Cord Stromal Tumor','','MIXED','SOLID',NULL),(872,'','LimeGreen',0,'Mature T and NK Neoplasms','','Lymphoid','LIQUID',NULL),(873,'','HotPink',0,'Breast Cancer','','Breast','SOLID',NULL),(874,'','MIXED',0,'Gastrointestinal Neuroendocrine Tumor','','MIXED','SOLID',NULL),(875,'','Orange',0,'Renal Cell Carcinoma','','Kidney','SOLID',NULL),(876,'','Gainsboro',0,'Non-Small Cell Lung Cancer','','Lung','SOLID',NULL),(877,'','DarkRed',0,'Sialoblastoma','','Head and Neck','SOLID',NULL),(878,'','Teal',0,'Cervical Cancer','','Cervix','SOLID',NULL),(879,'','Gray',0,'Sellar Tumor','','CNS/Brain','SOLID',NULL),(880,'','LightBlue',0,'Ovarian Cancer','','Ovary/Fallopian Tube','SOLID',NULL),(881,'','Green',0,'Lacrimal Gland Tumor','','Eye','SOLID',NULL),(882,'','Gray',0,'Glioma','','CNS/Brain','SOLID',NULL),(883,'','Black',0,'Skin Cancer, Non-Melanoma','','Skin','SOLID',NULL),(884,'','PeachPuff',0,'Endometrial Cancer','','Uterus','SOLID',NULL),(885,'','LimeGreen',0,'Lymphatic Cancer, NOS','','Lymphoid','LIQUID',NULL),(886,'','SaddleBrown',0,'Appendiceal Cancer','','Bowel','SOLID',NULL),(887,'','Black',0,'Cancer of Unknown Primary','','Other','SOLID',NULL),(888,'','Purple',0,'Adrenal Gland Cancer, NOS','','Adrenal Gland','SOLID',NULL),(889,'','Gray',0,'CNS Cancer','','CNS/Brain','SOLID',NULL),(890,'','MIXED',0,'Mesothelioma','','MIXED','SOLID',NULL),(891,'','LightSalmon',0,'Myeloproliferative Neoplasms','','Myeloid','LIQUID',NULL),(892,'','LimeGreen',0,'B-Lymphoblastic Leukemia/Lymphoma','','Lymphoid','LIQUID',NULL),(893,'','MIXED',0,'Melanoma','','MIXED','SOLID',NULL),(894,'','MIXED',0,'Germ Cell Tumor','','MIXED','SOLID',NULL),(895,'','Gainsboro',0,'Small Cell Lung Cancer','','Lung','SOLID',NULL),(896,'','LimeGreen',0,'Posttransplant Lymphoproliferative Disorders','','Lymphoid','LIQUID',NULL),(897,'','MIXED',0,'Hepatobiliary Cancer','','MIXED','SOLID',NULL),(898,'','Teal',0,'Thyroid Cancer','','Thyroid','SOLID',NULL),(899,'','Purple',0,'Pancreatic Cancer','','Pancreas','SOLID',NULL),(900,'','Yellow',0,'Bladder Cancer','','Bladder/Urinary Tract','SOLID',NULL),(901,'','Gray',0,'Nerve Sheath Tumor','','Peripheral Nervous System','SOLID',NULL),(902,'','Teal',0,'Cervical Cancer, NOS','','Cervix','SOLID',NULL),(903,'','SaddleBrown',0,'Small Bowel Cancer','','Bowel','SOLID',NULL),(904,'','LimeGreen',0,'Hodgkin Lymphoma','','Lymphoid','LIQUID',NULL),(905,'','DarkRed',0,'Salivary Gland Cancer','','Head and Neck','SOLID',NULL),(906,'','Blue',0,'Penile Cancer','','Penis','SOLID',NULL),(907,'','Gray',0,'Primary CNS Melanocytic Tumors','','CNS/Brain','SOLID',NULL),(908,'','LightSalmon',0,'Myelodysplastic Syndromes','','Myeloid','LIQUID',NULL),(909,'','Purple',0,'Ampullary Carcinoma, NOS','','Ampulla of Vater','SOLID',NULL),(910,'','Yellow',0,'Bladder/Urinary Tract Cancer, NOS','','Bladder/Urinary Tract','SOLID',NULL),(911,'','PeachPuff',0,'Uterine Sarcoma','','Uterus','SOLID',NULL),(912,'','LightSalmon',0,'Blood Cancer, NOS','','Myeloid','LIQUID',NULL),(913,'','Gray',0,'Miscellaneous Neuroepithelial Tumor','','MIXED','SOLID',NULL),(914,'','Green',0,'Peritoneal Cancer, NOS','','Peritoneum','SOLID',NULL),(915,'','SaddleBrown',0,'Tubular Adenoma of the Colon','','Bowel','SOLID',NULL),(916,'','LightSalmon',0,'Mastocytosis','','Myeloid','LIQUID',NULL),(917,'','LightSalmon',0,'Myelodysplastic/Myeloproliferative Neoplasms','','Myeloid','LIQUID',NULL),(918,'','SaddleBrown',0,'Anal Cancer','','Bowel','SOLID',NULL),(919,'','Orange',0,'Rhabdoid Cancer','','Kidney','SOLID',NULL),(920,'','Gray',0,'Melanocytoma','','CNS/Brain','SOLID',NULL),(921,'','Gray',0,'Pineal Tumor','','CNS/Brain','SOLID',NULL),(922,'','Gray',0,'Miscellaneous Brain Tumor','','CNS/Brain','SOLID',NULL),(923,'','LightYellow',0,'Infantile Fibrosarcoma','','Soft Tissue','SOLID',NULL),(924,'','Gray',0,'Peripheral Nervous System Cancer, NOS','','Peripheral Nervous System','SOLID',NULL),(925,'','Purple',0,'Vulvar Carcinoma','','Vulva/Vagina','SOLID',NULL),(926,'','Purple',0,'Vaginal Cancer','','Vulva/Vagina','SOLID',NULL),(927,'','Gray',0,'Peripheral Nervous System','','Peripheral Nervous System','SOLID',NULL),(928,'','Purple',0,'Thymic Tumor','','Thymus','SOLID',NULL),(929,'','Green',0,'Retinoblastoma','','Eye','SOLID',NULL),(930,'','PeachPuff',0,'Gestational Trophoblastic Disease','','Uterus','SOLID',NULL),(931,'','LightYellow',0,'Angiomatoid Fibrous Histiocytoma','','Soft Tissue','SOLID',NULL),(932,'','Purple',0,'Ampullary Cancer','','Ampulla of Vater','SOLID',NULL),(933,'','LightSkyBlue',0,'Esophagogastric Cancer','','Esophagus/Stomach','SOLID',NULL),(934,'','LightSalmon',0,'Histiocytosis','','Myeloid','LIQUID',NULL),(935,'','SaddleBrown',0,'Colorectal Cancer','','Bowel','SOLID',NULL),(936,'','Cyan',0,'Prostate Cancer','','Prostate','SOLID',NULL),(937,'','Green',0,'Eye Cancer, NOS','','Eye','SOLID',NULL),(938,'','DarkRed',0,'Parathyroid Cancer','','Head and Neck','SOLID',NULL),(939,'','SaddleBrown',0,'Bowel Cancer, NOS','','Bowel','SOLID',NULL),(940,'','Orange',0,'Wilms Tumor','','Kidney','SOLID',NULL),(941,'','LimeGreen',0,'T-Lymphoblastic Leukemia/Lymphoma','','Lymphoid','LIQUID',NULL),(942,'','DarkRed',0,'Head and Neck Cancer, NOS','','Head and Neck','SOLID',NULL),(943,'','Purple',0,'Pheochromocytoma','','Adrenal Gland','SOLID',NULL),(944,'','Cyan',0,'Prostate Cancer, NOS','','Prostate','SOLID',NULL),(945,'','White',0,'Bone Cancer, NOS','','Bone','SOLID',NULL),(946,'','Green',0,'Biliary Tract Cancer, NOS','','Biliary Tract','SOLID',NULL),(947,'','MIXED',0,'Non-Hodgkin Lymphoma','','MIXED','MIXED',NULL),(948,'','MediumSeaGreen',0,'Malignant Rhabdoid Tumor of the Liver','','Liver','SOLID',NULL),(949,'','Black',0,'Adenocarcinoma In Situ','','Other','SOLID',NULL),(950,'','Orange',0,'Kidney Cancer, NOS','','Kidney','SOLID',NULL),(951,'','LightYellow',0,'Myofibromatosis','','Soft Tissue','SOLID',NULL),(952,'','HotPink',0,'Breast Sarcoma','','Breast','SOLID',NULL),(953,'','LightYellow',0,'Malignant Glomus Tumor','','Soft Tissue','SOLID',NULL),(954,'','Gray',0,'Choroid Plexus Tumor','','CNS/Brain','SOLID',NULL),(955,'','LightSkyBlue',0,'Gastrointestinal Neuroendocrine Tumors of the Esophagus/Stomach','','Esophagus/Stomach','SOLID',NULL),(956,'','LightSalmon',0,'Blastic Plasmacytoid Dendritic Cell Neoplasm','','Myeloid','LIQUID',NULL),(957,'','Gainsboro',0,'Lung Cancer, NOS','','Lung','SOLID',NULL),(958,'','Purple',0,'Pancreatic Cancer, NOS','','Pancreas','SOLID',NULL),(959,'','LightSkyBlue',0,'Esophageal/Stomach Cancer, NOS','','Esophagus/Stomach','SOLID',NULL),(960,'','Black',0,'Skin Cancer, NOS','','Skin','SOLID',NULL),(961,'','LightBlue',0,'Ovarian/Fallopian Tube Cancer, NOS','','Ovary/Fallopian Tube','SOLID',NULL),(962,'','MediumSeaGreen',0,'Liver Cancer, NOS','','Liver','SOLID',NULL),(963,'','LightSalmon',0,'Myeloid Neoplasms with Germ Line Predisposition','','Myeloid','LIQUID',NULL),(964,'','Purple',0,'Adrenocortical Adenoma','','Adrenal Gland','SOLID',NULL),(965,'','Red',0,'Testicular Cancer, NOS','','Testis','SOLID',NULL),(966,'','MediumSeaGreen',0,'Undifferentiated Embryonal Sarcoma of the Liver','','Liver','SOLID',NULL),(967,'','PeachPuff',0,'Uterine Cancer, NOS','','Uterus','SOLID',NULL),(968,'','LightYellow',0,'Gastrointestinal Stromal Tumor','','Soft Tissue','SOLID',NULL),(969,'','Gray',0,'CNS/Brain Cancer, NOS','','CNS/Brain','SOLID',NULL),(970,'','Teal',0,'Thyroid Cancer, NOS','','Thyroid','SOLID',NULL),(971,'','LightYellow',0,'Soft Tissue Cancer','','Soft Tissue','SOLID',NULL),(972,'','LightYellow',0,'Soft Tissue Cancer, NOS','','Soft Tissue','SOLID',NULL),(973,'','Purple',0,'Thymic Cancer, NOS','','Thymus','SOLID',NULL),(974,'','Purple',0,'Adrenocortical Carcinoma','','Adrenal Gland','SOLID',NULL),(975,'','Blue',0,'Pleural Cancer, NOS','','Pleura','SOLID',NULL),(976,'','Black',0,'Other Cancer, NOS','','Other','SOLID',NULL),(977,'','Orange',0,'Renal Neuroendocrine Tumor','','Kidney','SOLID',NULL),(978,'','Blue',0,'Penile Cancer, NOS','','Penis','SOLID',NULL),(979,'','HotPink',0,'Breast Cancer, NOS','','Breast','SOLID',NULL),(980,'','Orange',0,'Clear Cell Sarcoma of Kidney','','Kidney','SOLID',NULL),(981,'','Purple',0,'Vulvar/Vaginal Cancer, NOS','','Vulva/Vagina','SOLID',NULL),(982,'','',-1,'All Tumors','','','MIXED',NULL),(983,'','',-1,'All Liquid Tumors','','','LIQUID',NULL),(984,'','',-1,'All Solid Tumors','','','SOLID',NULL),(985,'','',-1,'Germline Disposition','','','MIXED',NULL),(986,'','',-1,'Other Tumor Types','','','MIXED',NULL),(987,'','',-1,'Other Solid Tumor Types','','','SOLID',NULL),(988,'','',-1,'Other Liquid Tumor Types','','','LIQUID',NULL);
/*!40000 ALTER TABLE `cancer_type` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `cancer_type_child`
--

DROP TABLE IF EXISTS `cancer_type_child`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `cancer_type_child` (
  `cancer_type_id` int NOT NULL,
  `cancer_type_child_id` int NOT NULL,
  PRIMARY KEY (`cancer_type_id`,`cancer_type_child_id`),
  KEY `FK25672AF690C0BC52` (`cancer_type_child_id`),
  KEY `FK25672AF64C5916CF` (`cancer_type_id`),
  CONSTRAINT `FK25672AF64C5916CF` FOREIGN KEY (`cancer_type_id`) REFERENCES `cancer_type` (`id`),
  CONSTRAINT `FK25672AF690C0BC52` FOREIGN KEY (`cancer_type_child_id`) REFERENCES `cancer_type` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `cancer_type_child`
--

LOCK TABLES `cancer_type_child` WRITE;
/*!40000 ALTER TABLE `cancer_type_child` DISABLE KEYS */;
INSERT INTO `cancer_type_child` VALUES (347,1),(490,2),(530,3),(303,4),(631,5),(418,6),(211,7),(557,8),(243,9),(799,10),(399,11),(198,12),(380,13),(475,14),(303,15),(601,16),(527,17),(80,18),(26,19),(168,20),(280,21),(666,22),(530,23),(683,24),(543,25),(644,26),(601,27),(601,28),(198,29),(687,30),(258,31),(54,32),(530,33),(731,34),(566,35),(712,36),(44,37),(491,38),(534,39),(750,40),(313,41),(129,42),(185,43),(303,44),(95,45),(453,46),(84,47),(532,48),(543,49),(338,50),(44,51),(80,52),(830,53),(644,54),(31,55),(832,56),(553,57),(631,58),(547,59),(648,60),(532,61),(392,62),(687,63),(8,64),(380,65),(464,66),(380,67),(238,68),(108,69),(530,70),(683,71),(108,72),(683,73),(683,74),(776,75),(238,76),(806,77),(242,78),(602,79),(491,80),(289,81),(500,82),(475,83),(557,84),(159,85),(543,86),(383,87),(283,88),(530,89),(238,90),(169,91),(522,92),(280,93),(481,94),(31,95),(644,96),(693,97),(168,98),(355,99),(55,100),(683,101),(712,102),(491,103),(734,104),(530,105),(530,106),(243,107),(491,108),(169,109),(347,110),(331,111),(522,112),(754,113),(629,114),(168,115),(83,116),(683,117),(532,118),(243,119),(338,120),(168,121),(648,122),(522,123),(387,124),(380,125),(78,126),(420,127),(687,128),(491,129),(712,130),(828,131),(648,132),(530,133),(543,134),(238,135),(552,136),(45,137),(168,138),(169,139),(543,140),(547,141),(637,142),(530,143),(543,144),(45,145),(4,146),(742,147),(303,148),(601,149),(637,150),(379,151),(3,152),(543,153),(84,154),(84,155),(243,156),(289,157),(323,158),(80,159),(731,160),(683,161),(322,162),(313,163),(313,164),(287,165),(338,166),(586,167),(557,168),(623,169),(84,170),(525,171),(285,172),(687,173),(387,174),(522,175),(289,176),(530,177),(338,178),(169,179),(525,180),(169,181),(159,182),(683,183),(198,184),(174,185),(543,186),(108,187),(420,188),(347,189),(431,190),(108,191),(96,192),(45,193),(500,194),(54,195),(776,196),(266,197),(527,198),(587,199),(322,200),(238,201),(530,202),(750,203),(322,204),(342,205),(530,206),(45,207),(712,208),(567,209),(683,210),(683,211),(861,212),(132,213),(380,214),(491,215),(750,216),(530,217),(475,218),(687,219),(750,220),(243,221),(853,222),(619,223),(285,224),(853,225),(215,226),(355,227),(828,228),(168,229),(686,230),(54,231),(832,232),(450,233),(602,234),(666,235),(243,236),(566,237),(644,238),(280,239),(83,240),(420,241),(215,242),(576,243),(850,244),(687,245),(198,246),(716,247),(238,248),(417,249),(712,250),(844,251),(258,252),(683,253),(732,254),(844,255),(361,256),(159,257),(491,258),(54,259),(198,260),(587,261),(380,262),(799,263),(243,264),(525,265),(644,266),(355,267),(352,268),(55,269),(382,270),(84,271),(80,272),(734,273),(754,274),(374,275),(530,276),(80,277),(683,278),(211,279),(491,280),(204,281),(93,282),(393,283),(683,284),(174,285),(530,286),(83,287),(303,288),(491,289),(431,290),(686,291),(108,292),(31,293),(243,294),(399,295),(619,296),(543,297),(648,298),(420,299),(861,300),(4,301),(754,302),(491,303),(712,304),(453,305),(693,306),(38,307),(243,308),(530,309),(26,310),(541,311),(750,312),(683,313),(530,314),(799,315),(169,316),(243,317),(601,318),(361,319),(543,320),(380,321),(174,322),(108,323),(84,324),(530,325),(256,326),(83,327),(74,328),(543,329),(695,330),(541,331),(619,332),(683,333),(525,334),(44,335),(731,336),(267,337),(799,338),(283,339),(4,340),(530,341),(83,342),(361,343),(382,344),(8,345),(683,346),(644,347),(382,348),(420,349),(687,350),(45,351),(174,352),(380,353),(417,354),(174,355),(383,356),(55,357),(108,358),(530,359),(4,360),(541,361),(355,362),(683,363),(543,364),(342,365),(215,366),(352,367),(450,368),(352,369),(83,370),(8,371),(683,372),(647,373),(491,374),(243,375),(567,376),(712,377),(362,378),(707,379),(850,380),(258,381),(289,382),(853,383),(832,384),(420,385),(683,386),(491,387),(643,388),(491,389),(198,390),(464,391),(530,392),(491,393),(543,394),(631,395),(342,396),(522,397),(525,398),(289,399),(683,400),(832,401),(587,402),(631,403),(623,404),(347,405),(623,406),(54,407),(543,408),(648,409),(453,410),(683,411),(26,412),(4,413),(799,414),(382,415),(799,416),(629,417),(527,418),(543,419),(644,420),(243,421),(374,422),(567,423),(293,424),(631,425),(716,426),(619,427),(383,428),(198,429),(392,430),(491,431),(347,432),(204,433),(224,434),(44,435),(683,436),(541,437),(374,438),(620,439),(631,440),(643,441),(655,442),(78,443),(26,444),(80,445),(243,446),(853,447),(683,448),(602,449),(389,450),(746,451),(620,452),(543,453),(93,454),(331,455),(732,456),(380,457),(442,458),(83,459),(383,460),(342,461),(363,462),(174,463),(525,464),(283,465),(83,466),(567,467),(418,468),(289,469),(26,470),(750,471),(243,472),(839,473),(750,474),(491,475),(683,476),(853,477),(543,478),(168,479),(54,480),(737,481),(80,482),(84,483),(380,484),(315,485),(352,486),(734,487),(243,488),(289,489),(530,490),(243,492),(632,493),(96,494),(648,495),(338,496),(620,497),(83,498),(687,499),(727,500),(168,501),(108,502),(338,503),(55,504),(647,505),(80,506),(742,507),(530,508),(347,509),(861,510),(453,511),(475,512),(45,513),(799,514),(687,515),(383,516),(204,517),(683,518),(420,519),(830,520),(742,521),(303,522),(522,523),(211,524),(491,525),(303,526),(174,527),(399,528),(340,529),(576,530),(530,531),(475,532),(530,533),(530,534),(287,535),(566,536),(601,537),(490,538),(683,539),(587,540),(491,541),(514,542),(491,543),(587,544),(185,545),(159,546),(168,547),(108,548),(347,549),(389,550),(389,551),(525,552),(393,553),(352,554),(420,555),(266,556),(491,557),(26,558),(4,559),(185,560),(687,561),(623,562),(837,563),(839,564),(619,565),(289,566),(491,567),(80,568),(4,569),(83,570),(159,571),(522,572),(342,573),(238,574),(55,575),(31,576),(174,577),(643,578),(380,579),(631,580),(354,581),(683,582),(620,583),(530,584),(830,585),(530,586),(644,587),(530,588),(224,589),(198,590),(108,591),(553,592),(361,593),(108,594),(687,595),(244,596),(560,597),(683,598),(26,599),(587,600),(527,601),(168,602),(347,603),(442,604),(522,605),(355,606),(352,607),(420,608),(481,609),(38,610),(352,611),(522,612),(243,613),(114,614),(683,615),(525,616),(623,617),(243,618),(644,619),(361,620),(492,621),(543,622),(491,623),(522,624),(543,625),(567,626),(799,627),(289,628),(491,629),(21,630),(389,631),(185,632),(734,633),(475,634),(619,635),(530,636),(114,637),(242,638),(313,639),(389,640),(683,641),(683,642),(530,643),(491,644),(530,645),(543,646),(108,647),(491,648),(442,649),(693,650),(417,651),(338,652),(475,653),(168,654),(799,655),(352,656),(631,657),(204,658),(243,659),(683,660),(553,661),(500,662),(347,663),(243,664),(374,665),(619,666),(683,667),(8,668),(54,669),(286,670),(601,671),(734,672),(683,673),(655,674),(530,675),(108,676),(168,677),(683,678),(54,679),(355,680),(492,681),(839,682),(491,683),(288,684),(80,685),(491,686),(629,687),(45,688),(475,689),(732,690),(693,691),(285,692),(174,693),(243,694),(185,695),(354,696),(746,697),(453,698),(38,699),(712,700),(380,701),(712,702),(211,703),(80,704),(354,705),(530,706),(491,707),(383,708),(355,709),(266,710),(168,711),(716,712),(619,713),(380,714),(243,715),(491,716),(44,717),(3,718),(683,719),(361,720),(746,721),(530,722),(727,723),(291,724),(338,725),(530,726),(31,727),(289,728),(587,729),(754,730),(243,731),(103,732),(431,733),(174,734),(54,735),(8,736),(491,737),(522,738),(481,739),(500,740),(159,741),(389,742),(45,743),(238,744),(619,745),(644,746),(44,747),(648,748),(567,749),(644,750),(55,751),(168,752),(619,753),(114,754),(632,755),(754,756),(361,757),(114,758),(530,759),(80,760),(532,761),(631,762),(347,763),(291,764),(21,765),(132,766),(541,767),(389,768),(683,769),(362,770),(114,771),(95,772),(522,773),(453,774),(45,775),(10,776),(215,777),(666,778),(243,779),(420,780),(541,781),(168,782),(338,783),(527,784),(683,785),(399,786),(683,787),(567,788),(198,789),(687,790),(198,791),(383,792),(750,793),(629,794),(352,795),(587,796),(379,797),(567,798),(491,799),(567,800),(285,801),(417,802),(683,803),(712,804),(198,805),(185,806),(289,807),(382,808),(168,809),(453,810),(799,811),(647,812),(331,813),(204,814),(387,815),(242,816),(266,817),(243,818),(530,819),(807,820),(224,821),(601,822),(742,823),(431,824),(776,825),(379,826),(522,827),(530,828),(80,829),(159,830),(185,831),(644,832),(26,833),(631,834),(285,835),(543,836),(683,837),(832,838),(10,839),(267,840),(543,841),(286,842),(114,843),(527,844),(211,845),(464,846),(566,847),(347,848),(420,849),(431,850),(392,851),(338,852),(491,853),(256,854),(619,855),(620,856),(293,857),(598,858),(750,859),(853,860),(557,861),(352,862),(129,863),(601,864);
/*!40000 ALTER TABLE `cancer_type_child` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `drug`
--

DROP TABLE IF EXISTS `drug`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `drug` (
  `id` int NOT NULL AUTO_INCREMENT,
  `description` longtext,
  `drug_name` longtext NOT NULL,
  `ncit_code` varchar(20) DEFAULT NULL,
  `type` varchar(20) DEFAULT NULL,
  `uuid` varchar(40) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=157 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `drug`
--

LOCK TABLES `drug` WRITE;
/*!40000 ALTER TABLE `drug` DISABLE KEYS */;
INSERT INTO `drug` VALUES (9,NULL,'Crizotinib','C74061','DRUG','f05d12ab-6204-4a1c-803f-f3a7f6e30af2'),(10,NULL,'Ceritinib','C115112','DRUG','5e9aa568-30cf-48ef-b9ce-18851af2f622'),(13,NULL,'Lorlatinib','C113655','DRUG','ed76aa01-059c-4b22-b391-94ec86320122'),(17,NULL,'Cobimetinib','C68923','DRUG','eb357145-3b18-4aca-b75a-5e18dd2bf4f9'),(18,NULL,'Trametinib','C77908','DRUG','fb2bb01c-c0ec-4641-abf7-87f486075022'),(24,NULL,'Dabrafenib','C82386','DRUG','939cd40b-b515-499d-b099-fd29027c0d17'),(25,NULL,'Vemurafenib','C64768','DRUG','4e91da20-6cf0-4e07-995f-7f7db4c7c077'),(26,NULL,'Encorafenib','C98283','DRUG','001e534f-3e63-432f-90a6-d1af1759e4e2'),(27,NULL,'Binimetinib','C84865','DRUG','feb9f4a3-e374-4c75-8a3b-0f1fbcdbf677'),(28,NULL,'Cetuximab','C1723','DRUG','5fce3074-e420-4c36-9603-2423daf20118'),(29,NULL,'Panitumumab','C1857','DRUG','d7b1d12a-e942-4801-bb64-916c9bdfaaf3'),(30,NULL,'FOLFOX Regimen','C11197','DRUG','92952193-f6fb-4da4-83a6-784211b67623'),(31,NULL,'Selumetinib','C66939','DRUG','3f154145-525b-4865-8baf-12b374697b2a'),(32,NULL,'Atezolizumab','C106250','DRUG','9a54c73a-15d8-41a8-8ea0-730f874831da'),(33,NULL,'Tovorafenib','C106254','DRUG','6a182f62-9354-4684-9c8a-127efe81cf00'),(34,NULL,'Plixorafenib','C113330','DRUG','44faeeae-ad6d-4370-8c99-97955f9e0feb'),(119,NULL,'Cabozantinib','C52200','DRUG','6acebe88-de89-4ec1-bc38-74cb63861d23'),(130,NULL,'Entrectinib','C114984','DRUG','2371f7f5-6407-4a4b-b951-a0f786355455'),(131,NULL,'Repotrectinib','C133821','DRUG','d28edb2a-6f51-4dd0-b1ec-aca045f50f9f'),(150,NULL,'Taletrectinib','C118948','DRUG','fec72a32-427e-4ee6-91b4-6faf4a7880e2'),(155,NULL,'Rezatapopt','C175961','DRUG','89814a16-3ae4-4cf0-8d69-34c03964345f');
/*!40000 ALTER TABLE `drug` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `drug_family`
--

DROP TABLE IF EXISTS `drug_family`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `drug_family` (
  `drug_id` int NOT NULL,
  `drug_family_id` int NOT NULL,
  PRIMARY KEY (`drug_id`,`drug_family_id`),
  KEY `FKD5F254E34E48DF2` (`drug_family_id`),
  KEY `FKD5F254E344975E15` (`drug_id`),
  CONSTRAINT `FKD5F254E344975E15` FOREIGN KEY (`drug_id`) REFERENCES `drug` (`id`),
  CONSTRAINT `FKD5F254E34E48DF2` FOREIGN KEY (`drug_family_id`) REFERENCES `drug` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `drug_family`
--

LOCK TABLES `drug_family` WRITE;
/*!40000 ALTER TABLE `drug_family` DISABLE KEYS */;
/*!40000 ALTER TABLE `drug_family` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `drug_synonym`
--

DROP TABLE IF EXISTS `drug_synonym`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `drug_synonym` (
  `drug_id` int NOT NULL,
  `synonym` longtext,
  KEY `FKC10E725C44975E15` (`drug_id`),
  CONSTRAINT `FKC10E725C44975E15` FOREIGN KEY (`drug_id`) REFERENCES `drug` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `drug_synonym`
--

LOCK TABLES `drug_synonym` WRITE;
/*!40000 ALTER TABLE `drug_synonym` DISABLE KEYS */;
INSERT INTO `drug_synonym` VALUES (9,'MET Tyrosine Kinase Inhibitor PF-02341066'),(9,'Xalkori'),(9,'Crizotinib'),(9,'2-Pyridinamine, 3-((1R)-1-(2,6-Dichloro-3-Fluorophenyl)Ethoxy)-5-(1-(4-Piperidinyl)-1H-Pyrazol-4-yl)-'),(9,'PF-02341066'),(9,'CRIZOTINIB'),(9,'PF-2341066'),(9,'MET tyrosine kinase inhibitor PF-02341066'),(9,'(R)-3-(1-(2,6-Dichloro-3-Fluorophenyl)Ethoxy)-5-(1-(Piperidin-4-Yl)-1h-Pyrazol-4-Yl)Pyridin-2-Amine'),(10,'LDK378'),(10,'2,4-Pyrimidinediamine, 5-chloro-N4-(2-((1-methylethyl)sulfonyl)phenyl)-N2-(5-methyl-2-(1-methylethoxy)-4-(4-piperidinyl)phenyl)-'),(10,'LDK 378'),(10,'CERITINIB'),(10,'Zykadia'),(10,'Ceritinib'),(13,'Lorlatinib'),(13,'Lorbrena'),(13,'2H-4,8-Methenopyrazolo(4,3-H)(2,5,11)benzoxadiazacyclotetradecine-3-carbonitrile, 7-amino-12-fluoro-10,15,16,17-tetrahydro-2,10,16-trimethyl-15-oxo-, (10R)-'),(13,'LORLATINIB'),(13,'7-Amino-12-fluoro-2,10,16-trimethyl-15-oxo-10,15,16,17-tetrahydro-2H-8,4-(metheno)pyrazolo(4,3-h)(2,5,11)benzoxadiazacyclotetradecine-3-carbonitrile'),(13,'PF-06463922'),(17,'Cotellic'),(17,'Cobimetinib'),(17,'XL518'),(17,'COBIMETINIB'),(17,'GDC-0973'),(17,'MEK Inhibitor GDC-0973'),(18,'TRAMETINIB'),(18,'JTP-74057'),(18,'MEK Inhibitor GSK1120212'),(18,'N-(3-{3-cyclopropyl-5-[(2-fluoro-4-iodophenyl)amino]-6,8-dimethyl-2,4,7-trioxo-3,4,6,7-tetrahydropyrido[4,3-d]pyrimidin-1(2H)-yl}phenyl)acetamide'),(18,'GSK1120212'),(18,'Trametinib'),(24,'DABRAFENIB'),(24,'GSK-2118436'),(24,'GSK-2118436A'),(24,'BRAF Inhibitor GSK2118436'),(24,'Dabrafenib'),(24,'GSK2118436'),(24,'Benzenesulfonamide, N-(3-(5-(2-amino-4-pyrimidinyl)-2-(1,1-dimethylethyl)-4-thiazolyl)-2-fluorophenyl)-2,6-difluoro-'),(25,'Vemurafenib'),(25,'BRAF(V600E) Kinase Inhibitor RO5185426'),(25,'RO 5185426'),(25,'PLX4032'),(25,'PLX-4032'),(25,'1-propanesulfonamide, n-(3-((5-(4-chlorophenyl)-1h-pyrrolo(2,3-b)pyridin-3-yl)carbonyl)-2,4-difluorophenyl)-'),(25,'RG 7204'),(25,'Zelboraf'),(25,'BRAF (V600E) kinase inhibitor RO5185426'),(25,'RG7204'),(25,'VEMURAFENIB'),(26,'LGX-818'),(26,'Encorafenib'),(26,'LGX818'),(26,'LGX 818'),(26,'ENCORAFENIB'),(26,'Braftovi'),(27,'ARRY-438162'),(27,'Mektovi'),(27,'Binimetinib'),(27,'ARRY-162'),(27,'MEK162'),(27,'BINIMETINIB'),(28,'CETUXIMAB'),(28,'Cetuximab Biosimilar CMAB009'),(28,'IMC-C225'),(28,'Chimeric Anti-EGFR Monoclonal Antibody'),(28,'Cetuximab Biosimilar KL 140'),(28,'Cetuximab Biosimilar CDP-1'),(28,'cetuximab'),(28,'Chimeric Monoclonal Antibody C225'),(28,'Cetuximab'),(28,'Chimeric MoAb C225'),(28,'Erbitux'),(29,'ABX-EGF, Clone E7.6.3'),(29,'PANITUMUMAB'),(29,'Vectibix'),(29,'Panitumumab'),(29,'MoAb ABX-EGF'),(29,'ABX-EGF'),(29,'ABX-EGF Monoclonal Antibody'),(29,'Monoclonal Antibody ABX-EGF'),(29,'panitumumab'),(30,'FOLFOX regimen'),(30,'Fluorouracil/Leucovorin Calcium/Oxaliplatin'),(30,'FOLFOX Regimen'),(30,'CF/5-FU/L-OHP'),(30,'FOLFOX'),(31,'Selumetinib'),(31,'AZD6244'),(31,'SELUMETINIB'),(31,'ARRY-142886'),(31,'MEK Inhibitor AZD6244'),(31,'MEK inhibitor AZD6244'),(32,'MPDL 3280A'),(32,'MPDL328OA'),(32,'MPDL 328OA'),(32,'Tecentriq'),(32,'Immunoglobulin G1, Anti-(human CD Antigen cd274) (Human Monoclonal MPDL3280A Heavy Chain), Disulfide with Human Monoclonal MPDL3280A Kappa-chain, Dimer'),(32,'ATEZOLIZUMAB'),(32,'MPDL-3280A'),(32,'RG7446'),(32,'MPDL3280A'),(32,'RO5541267'),(32,'Atezolizumab'),(33,'TAK-580'),(33,'TAK580'),(33,'MLN-2480'),(33,'BIIB-024'),(33,'MLN2480'),(33,'pan-RAF Kinase Inhibitor TAK-580'),(33,'(R)-2-(1-(6-amino-5-chloropyrimidine-4-carboxamido)ethyl)-N-(5-chloro-4-(trifluoromethyl)pyridin-2-yl)thiazole-5-carboxamide'),(34,'PLX8394'),(34,'BRAF Inhibitor PLX8394'),(119,'CABOZANTINIB'),(119,'Cabozantinib'),(119,'1,1-Cyclopropanedicarboxamide, N\'-[4-[(6,7-dimethoxy-4-quinolinyl)oxy]phenyl]-N-(4- fluorophenyl)-'),(119,'N-{4-[(6,7-dimethoxyquinolin-4-yl)oxy]phenyl}-N\'-(4-fluorophenyl)cyclopropane-1,1- dicarboxamide'),(130,'ENTRECTINIB'),(130,'RXDX-101'),(130,'N-(5-(3,5-Difluorobenzyl)-1H-indazol-3-yl)-4-(4-methylpiperazin-1yl)-2-(tetrahydro-2H-pyran-4-ylamino)benzamide'),(130,'RXDX101'),(130,'RXDX 101'),(130,'Entrectinib'),(130,'Rozlytrek'),(131,'REPOTRECTINIB'),(131,'Repotrectinib'),(131,'Multikinase Inhibitor TPX-0005'),(131,'Multi-kinase Inhibitor TPX-0005'),(131,'TPX-0005'),(131,'ALK/ROS1/NTRK/SRC/FAK Multikinase Inhibitor TPX-0005'),(150,'DS-6051b'),(150,'TALETRECTINIB'),(150,'AB 106'),(150,'AB106'),(150,'Taletrectinib'),(150,'AB-106'),(155,'PC-14586'),(155,'PC 14586'),(155,'p53 Y220C Mutant Reactivator PC14586'),(155,'Mutant p53 Activator PC14586'),(155,'PC14586');
/*!40000 ALTER TABLE `drug_synonym` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `evidence`
--

DROP TABLE IF EXISTS `evidence`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `evidence` (
  `id` int NOT NULL AUTO_INCREMENT,
  `additional_info` longtext,
  `description` longtext,
  `evidence_type` varchar(255) DEFAULT NULL,
  `fda_level` varchar(255) DEFAULT NULL,
  `for_germline` bit(1) DEFAULT NULL,
  `known_effect` varchar(255) DEFAULT NULL,
  `last_edit` datetime DEFAULT NULL,
  `last_review` datetime DEFAULT NULL,
  `level_of_evidence` varchar(255) DEFAULT NULL,
  `liquid_propagation_level` varchar(255) DEFAULT NULL,
  `solid_propagation_level` varchar(255) DEFAULT NULL,
  `uuid` varchar(40) DEFAULT NULL,
  `entrez_gene_id` int DEFAULT NULL,
  `name` longtext,
  PRIMARY KEY (`id`),
  KEY `FK16D39E57C602BC92` (`entrez_gene_id`),
  CONSTRAINT `FK16D39E57C602BC92` FOREIGN KEY (`entrez_gene_id`) REFERENCES `gene` (`entrez_gene_id`)
) ENGINE=InnoDB AUTO_INCREMENT=30680 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `evidence`
--

LOCK TABLES `evidence` WRITE;
/*!40000 ALTER TABLE `evidence` DISABLE KEYS */;
INSERT INTO `evidence` VALUES (2445,NULL,'BRAF, an intracellular kinase, is frequently mutated in melanoma, thyroid and lung cancers among others.','GENE_SUMMARY',NULL,_binary '\0',NULL,'2023-06-22 11:21:36',NULL,NULL,NULL,NULL,'fb828c84-3a02-4f45-8721-01c20b7794d3',673,NULL),(2446,NULL,'BRAF is a serine/threonine kinase that plays a key role in the regulation of the mitogen-activated protein kinase (MAPK) cascade (PMID: 15520807), which under physiologic conditions regulates the expression of genes involved in cellular functions, including proliferation (PMID: 24202393). Genetic alterations in BRAF are found in a large percentage of melanomas, thyroid cancers and histiocytic neoplasms as well as a small fraction of lung and colorectal cancers. The most common BRAF point mutation is V600E, which deregulates the protein\'s kinase activity leading to constitutive BRAF activation, as BRAF V600E can signal as a monomer independently of RAS or upstream activation (PMID: 20179705). Other BRAF mutations have been found that affect the protein\'s propensity to dimerize (PMID: 16858395, 26343582, 12068308). The product of these alterations is a BRAF kinase that can activate MAPK signaling in an unregulated manner and, in some instances, is directly responsible for cancer growth (PMID: 15520807). Inhibitors of mutant BRAF, including vemurafenib and dabrafenib, are FDA-approved for the treatment of late-stage or unresectable melanoma.','GENE_BACKGROUND',NULL,_binary '\0',NULL,'2019-03-05 19:00:04',NULL,NULL,NULL,NULL,'56032b8d-22d9-4e78-97f4-6b3337c21d38',673,NULL),(2447,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2023-11-01 17:35:53',NULL,NULL,NULL,NULL,'1de14103-e502-40c7-a6a6-587f8fc53135',673,NULL),(2448,NULL,'The class I activating exon 15 BRAF V600E mutation is located in the kinase domain of the BRAF protein and is highly recurrent in melanoma, lung and thyroid cancer, among others (PMID: 28783719, 26091043, 25079552, 23833300, 25417114, 28783719, 12068308). This mutation has been comprehensively biologically characterized and has been shown to activate the downstream MAPK pathway independent of RAS (PMID: 15035987, 12068308, 19251651, 26343582), to render BRAF constitutively activated in monomeric form (PMID: 20179705), and to retain sensitivity to RAF monomer inhibitors such as vemurafenib and dabrafenib (PMID:26343582, 28783719, 20179705, 30351999).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-11-01 17:35:33',NULL,NULL,NULL,NULL,'a53f5aa8-a4bb-4e65-8e30-93cf7045029e',673,NULL),(2449,NULL,'The RAF-targeted inhibitors encorafenib, dabrafenib and vemurafenib alone or in combination with the MEK-targeted inhibitors binimetinib, trametinib and cobimetinib, respectively, are FDA-approved for the treatment of patients with BRAF V600E/K mutant melanoma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 14:14:06',NULL,NULL,NULL,NULL,'814e645a-9256-493d-b2c7-6771d4328876',673,NULL),(2450,NULL,'Dabrafenib is an orally bioavailable RAF inhibitor that is FDA-approved for use in patients with BRAF V600E- and V600K-mutant metastatic melanoma. FDA approval is based on the randomized Phase III trial in which dabrafenib (150 mg orally twice daily) was compared with dacarbazine (1000 mg/m2 intravenously every three weeks) in 250 patients with BRAF V600E-mutated metastatic melanoma. Dabrafenib was associated with improved progression-free survival (median 5.1 months vs. 2.7 months with dacarbazine; hazard ratio = 0.30, p<0.0001) (PMID: 22735384). Dabrafenib may also be effective against brain metastases, as demonstrated by a Phase II trial in which approximately 40% of previously untreated and 30% of previously treated patients experienced an overall intracranial response and a separate trial in which nine out of ten patients had a reduction in the size of their brain metastases (PMID: 23051966, 22608338).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:40:21',NULL,'LEVEL_1','NO','LEVEL_3B','86887f5d-62df-429b-9968-7b73c31a9504',673,NULL),(2451,NULL,'Vemurafenib is an orally available kinse inhibitor of V600-mutant BRAF that is FDA-approved for treatment of patients with unresectable or metastatic melanoma with the BRAF V600E mutation. Vemurafenib has been shown to have nearly equivalent activity against melanomas with BRAF V600E and V600K mutations (PMID: 24508103). In a randomized Phase III trial comparing vemurafenib (960 mg orally twice daily) with dacarbazine (1000 mg/m2 i.v. every 3 weeks) for treatment-naive, metastatic, BRAF V600E-mutant melanoma, vemurafenib was associated with better overall survival (median survival 13.6 months vs. 9.7 months; hazard ratio 0.70, p=.0008) and longer median progression-free survival (6.9 months vs. 1.6 months) (PMID: 24508103). Final overall survival data from the BRIM-3 study showed that the survival advantage of vemurafenib over dacarbazine persisted through the 4-year landmark, with survival rates for vemurafenib and dacarbazine at the 4-year landmark being 17.0% and 15.6%, respectively (PMID: 28961848). However, a trial evaluating clinical outcomes in patients with melanoma treated with either combination therapy of dabrafenib and trametinib compared to those treated with vemurafenib monotherapy demonstrated improved survival outcomes in the combination-therapy group compared to the vemurafenib group (PMID: 25399551).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:40:30',NULL,'LEVEL_1','NO','LEVEL_3B','09c8f359-ae07-4746-a2b1-2cc3d948b09f',673,NULL),(2452,NULL,'The RAF-targeted inhibitors dabrafenib and encorafenib in combination with the MEK1/2-targeted inhibitors trametinib and binimetinib, respectively, are FDA-approved for the treatment of patients with BRAF V600E mutant non-small cell lung cancer.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2023-10-18 16:15:16',NULL,NULL,NULL,NULL,'dc939314-9786-4d02-942c-004c3444c33d',673,NULL),(2453,NULL,'Dabrafenib, an orally bioavailable RAF inhibitor, and trametinib, an orally bioavailable MEK1/2 inhibitor, are FDA-approved alone or in combination for the treatment of patients with metastatic non-small cell lung cancer (NSCLC) with BRAF V600E mutation. FDA approval was based on a Phase II, multicenter, non-randomized, open-label study of dabrafenib in combination with trametinib in patients with chemotherapy-pretreated, metastatic, stage IV BRAF(V600E)-mutant non-small cell lung cancer, in which 36 of 57 patients (63%) achieved overall tumor response, with two patients experiencing durable complete responses and 34 patients having durable partial responses (PMID: 27283860). In the third arm of that study, which included patients with untreated, metastatic, stage IV BRAF(V600E)-mutant non-small cell lung cancer, 23 of 26 patients (64%, 95% CI 46-79) had an overall response, with two patients (6%) having durable complete response and 21 (58%) having partial response (PMID: 28919011).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:40:54',NULL,'LEVEL_1','NO','LEVEL_3B','c63b34cb-c1fa-4784-c031-98566ad048bb',673,NULL),(2454,NULL,'Encorafenib, a small molecule inhibitor of RAF kinases, and binimetinib, a small molecule inhibitor of MEK1/2 kinases, are FDA-approved in combination for the treatment of patients with BRAF V600E mutant non-small cell lung cancer (NSCLC). FDA approval was based on the Phase II, open-label, multicenter, single-arm PHAROS study of encorafenib in combination with binimetinib in treatment-nave (n=59) and previously treated (n=39) patients with BRAF V600E-mutant metastatic NSCLC (PMID: 37270692). In the Phase II PHAROS study, the objective response rate and duration of response for treatment-nave patients were 72% (95% CI=62, 85) and 44 months (95% CI=23.1, NE), with 15% experiencing a complete response and 59% experiencing a partial response, compared to 46% (95% CI=30, 63) and 18 months (95% CI=7.4, NE) in previously treated patients, with 10% experiencing a complete response and 36% experiencing a partial response (PMID: 37270692).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-10-18 13:04:43',NULL,'LEVEL_1','NO','LEVEL_3B','f9d7a2ff-3395-49ac-b86f-a141d0337167',673,NULL),(2455,NULL,'The RAF-targeted inhibitor dabrafenib in combination with the MEK1/2-targeted inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E mutant anaplastic thyroid cancer.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 14:15:08',NULL,NULL,NULL,NULL,'a3898773-037a-4e4b-8a58-41d07b189222',673,NULL),(2456,NULL,'Dabrafenib, an orally bioavailable RAF inhibitor, and trametinib, an orally bioavailable MEK1/2 inhibitor, are FDA-approved alone or in combination for the treatment of patients with locally advanced or metastatic anaplastic thyroid cancer (ATC) with BRAF V600E mutation and with no satisfactory locoregional treatment options. FDA approval was based on results of the Phase II study of dabrafenib combined with trametinib in patients with BRAF V600E-positive ATC in which the overall response rate in sixteen evaluable patients was 69% (11/16; 95% CI= 41%-89%)(PMID: 29072975).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:36:13',NULL,'LEVEL_1','NO','LEVEL_3B','87b2f23f-8d71-4664-b259-c2384202937b',673,NULL),(2457,NULL,'The RAF-targeted inhibitor encorafenib in combination with the anti-EGFR antibody cetuximab is FDA-approved for patients with previously treated BRAF V600E mutant colorectal cancer. The combination of the RAF-inhibitor encorafenib with the anti-EGFR antibody cetuximab and chemotherapy regimen FOLFOX is FDA-approved for patients with untreated metastatic BRAF V600E mutant colorectal cancer.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2025-02-05 16:57:49',NULL,NULL,NULL,NULL,'cfcc19a6-df02-4230-85fb-992ed74d464a',673,NULL),(2458,NULL,'Encorafenib, a small molecule RAF-targeted inhibitor, and cetuximab, an anti-EGFR antibody, are FDA-approved in combination for the treatment of adult patients with metastatic CRC with a BRAF V600E mutation, as detected by an FDA-approved test, after prior therapy. BRAF V600E mutations for treatment with encorafenib plus cetuximab were detected by the FoundationOne Liquid CDx, the MI Cancer Seek assay or the therascreen BRAF V600E RGQ PCR Kit. FDA approval was based on results of the Phase III BEACON (NCT02928224) study of encorafenib plus cetuximab (n=220) versus triplet treatment (including a MEK1/2 inhibitor) versus chemotherapy (n=221) in 665 patients with BRAF V600E-mutant colorectal cancer. \n\nIn the Phase III BEACON (NCT02928224) trial, the overall response rate (complete or partial response) was 20% (95% CI=13-29) in the doublet arm versus 2% (95% CI=<1%-7%) in the chemotherapy arm (n=221), with median overall survival of 8.4 months in the doublet arm (95% CI= 7.5-11.0) and 5.4 months in the chemotherapy arm (95% CI= 4.8-6.6) (PMID: 31566309).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2025-02-05 16:30:54',NULL,'LEVEL_1','NO','LEVEL_3B','5d3556f9-89b7-4be9-9fa8-69fee10f032b',673,NULL),(2459,NULL,'Encorafenib, a small molecule inhibitor of RAF kinase, and panitumumab, an antibody that targets EGFR, are NCCN-compendium listed in combination as level 2A therapy for patients with BRAF V600E-positive colorectal cancer. In the Phase III BEACON study of encorafenib + cetuximab, another EGFR antibody, versus triplet treatment (including a MEK1/2 inhibitor) versus chemotherapy in 665 patients with BRAF V600E-mutant colorectal cancer, the overall response rate (complete or partial response) was 20% (95% CI= 13-29) in the doublet arm (n=220) versus 2% (95% CI= <1%-7%) in the chemotherapy arm (n=221), with median overall survival of 8.4 months in the doublet arm (95% CI= 7.5-11.0) and 5.4 months in the control arm (95% CI= 4.8-6.6) (PMID: 31566309). Panitumumab has also been used in doublet and triplet combination therapy, with a response rate in one study of 26% (PMID: 29431699).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:37:14',NULL,'LEVEL_2','NO','LEVEL_3B','d9b444f3-76b2-4f91-a919-8b8b4d853846',673,NULL),(2460,NULL,'Encorafenib, a small molecule RAF-targeted inhibitor, and cetuximab, an anti-EGFR antibody, are FDA-approved in combination with FOLFOX6 for the treatment of patients with metastatic colorectal cancer (mCRC) with a BRAF V600E mutation, as detected by an FDA-approved test. BRAF V600E mutations were detected by the therascreen BRAF V600E RGQ PCR Kit. FDA approval was based on the results of the Phase III BREAKWATER (NCT04607421) trial of encorafenib plus cetuximab versus encorafenib plus cetuximab with FOLFOX6 versus standard-of-care in 479 patients with BRAF V600E-mutant mCRC.\n\nIn the Phase III BREAKWATER (NCT04607421) trial, the encorafenib plus cetuximab with FOLFOX6 cohort (n=110) demonstrated an overall response rate (ORR) of 60.9% (95% CI=51.6-69.5), with a 2.7% (n=3) complete response (CR) rate, 58.2% (n=64) partial response (PR) rate and 28.2% (n=31) stable disease (SD) rate, and a median duration of response (DOR) of 13.9 months (95% CI=8.5-NE) (PMID: 39863775). The standard-of-care cohort (n=110) demonstrated an ORR of 40.0% (95% CI=31.3-49.3) (odds ratio=2.443 [95% CI=1.403-4.253; 99.8% CI=1.019-5.855]; p=0.0008), with a 1.8% (n=2) CR rate, 38.2% (n=42) PR rate and 30.9% (n=34) SD rate, and a median DOR of 11.1 months (95% CI=6.7-12.7) (PMID: 39863775).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2025-02-06 09:45:42',NULL,'LEVEL_1','NO','LEVEL_3B','ece8443f-a430-410b-8ea1-8c6766046dea',673,NULL),(2461,NULL,'This assertion is supported by (PMID: 25422482, 26637772).','DIAGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Dx3',NULL,NULL,'9bf86162-5aab-4df2-bd5d-de3313c2e96f',673,NULL),(2462,NULL,'The presence of a BRAF V600E mutation may assist in the diagnosis of [[tumor type]].','DIAGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'d8ad6e9b-738e-47d1-a796-f3f27485813e',673,NULL),(2463,NULL,'The RAF-targeted inhibitor vemurafenib is NCCN-compendium listed for the treatment of patients with relapsed or refractory BRAF V600E-mutant hairy cell leukemia.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 12:28:23',NULL,NULL,NULL,NULL,'fe5e6dec-78e9-438d-9a22-762fcdc91ac2',673,NULL),(2464,NULL,'This assertion is supported by (PMID: 21663470, 25480661, 22072557, 22028477, 21910720, 22531170, 22210875, 23347903, 25120816, 26071465).','DIAGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Dx2',NULL,NULL,'565e1375-b008-47ab-9b14-5807616072fb',673,NULL),(2465,NULL,'The presence of a BRAF V600E mutation supports a diagnosis of [[tumor type]].','DIAGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2025-04-16 17:05:19',NULL,NULL,NULL,NULL,'82098d7c-b495-46db-b841-ee584037f871',673,NULL),(2466,NULL,'Vemurafenib is an orally available kinse inhibitor of V600-mutant BRAF that is FDA-approved for treatment of patients with various BRAF V600-harboring tumors, including melanoma and Edheim-Chester Disease (ECD). Vemurafenib is listed as a 2A therapeutic in the NCCN guidelines for Hairy Cell Leukemia (HCL) based on data from two Phase II trials. In one study of vemurafenib in patients with BRAF V600E-mutant HCL (n=24), the overall response rate was 100%, with 42% of patients (10 of 24) having complete responses, with a response duration of 11.7 months and one year progression free and overall survival rates of 73% and 91%, repectively (PMID: 26352686). In another study of vemurafenib in patients with BRAF V600E-mutant HCL (n=26), the overall response rate was 96%, with 35% of patients (9 of 26) having complete responses (PMID: 26352686).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:40:00',NULL,'LEVEL_2','NO','LEVEL_3B','4e56b02d-394e-4397-9465-a3cc2c5026cd',673,NULL),(2467,NULL,'This assertion is supported by (PMID: 24938183, 20519626, 30157397, 26637772).','DIAGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Dx2',NULL,NULL,'97cde25b-b793-4cf6-b4e2-b787c0a398e9',673,NULL),(2468,NULL,'The presence of a BRAF V600E mutation supports a diagnosis of [[tumor type]].','DIAGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2025-04-16 17:05:19',NULL,NULL,NULL,NULL,'086643d5-bb51-4e6d-815a-d39d2549d792',673,NULL),(2469,NULL,'The RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E-mutant solid tumors.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-30 14:15:21',NULL,NULL,NULL,NULL,'114b09bd-f15e-4f71-a3dd-048c746b20c3',673,NULL),(2470,NULL,'The RAF inhibitor vemurafenib in combination with the MEK inhibitor cobimetinib is FDA-approved for patients with melanoma, non-small cell lung cancer, and anaplastic thyroid cancer, and is NCCN-compendium listed as recommended treatment of central nervous system cancers such as pilocytic astrocytoma, pleomorphic xanthoastrocytoma (PXA) and ganglioglioma. In a case series of two patients with BRAF V600E-mutant PXA, combination treatment of dabrafenib with trametinib led to partial responses by RANO criteria in both patients (PMID: 28984141). In a case report of a sixteen-year-old female with BRAF V600E-mutant anaplastic ganglioglioma, dabrafenib and trametinib in combination led to a significant response that was maintained at least six months after treatment initiation (PMID: 29380516). In a separate case report of a 28-year-old man with BRAF V600E-mutant anaplastic ganglioglioma, combination treatment of vemurafenib and cobimetinib led to a complete response after three months, with no evidence of recurrence after sixteen months of treatment (PMID: 30120137).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 16:41:32',NULL,'LEVEL_2','NO','LEVEL_3B','7f76d72e-ec85-4f42-91fe-b247dbe0e0e9',673,NULL),(2471,NULL,'The RAF-targeted inhibitor vemurafenib is FDA-approved for the treatment of patients with BRAF V600-mutant Erdheim-Chester Disease (ECD) and there is promising clinical data in patients with non-ECD BRAF V600E mutant histiocytic neoplasms treated with vemurafenib or dabrafenib as monotherapy.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-09-30 18:58:45',NULL,NULL,NULL,NULL,'92d5ed0f-6425-4b2b-9a75-b7f673840cee',673,NULL),(2472,NULL,'While the RAF-inhibitor dabrafenib in combination with the MEK1/2-targeted inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E mutant solid tumors, the clinical utility of dabrafenib in combination with trametinib in patients with [[variant]] has yet to be defined.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-31 11:29:47',NULL,NULL,NULL,NULL,'215bd1c9-120d-4fc1-b906-5f317772b820',673,NULL),(2473,NULL,'The RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E-mutant solid tumors.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-30 14:33:01',NULL,NULL,NULL,NULL,'06d6c080-3109-4164-9558-557d0f1d319f',673,NULL),(2474,NULL,'Selumetinib is a small molecule tyrosine kinase inhibitor of MEK1/2. In stratum one of the phase II PBTC study of selumetinib in 25 patients with pilocytic astrocytoma harboring a KIAA1549BRAF fusion or BRAF V600E mutation, seven of eighteen patients with a KIAA1549BRAF fusion had a partial response to treatment (PMID: 31151904). Additionally, the two year progression-free survival rate for the BRAF study population (n=25) was 70% (95% CI = 4785) (PMID: 31151904).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:41:11',NULL,'LEVEL_2','NO','LEVEL_3B','9ae514fa-cea6-4a01-9f6e-ae9ca1a43049',673,NULL),(2475,NULL,'The RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E-mutant solid tumors.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-30 14:02:21',NULL,NULL,NULL,NULL,'d9d2b475-f103-4960-a827-3a032f107402',673,NULL),(2476,NULL,'The RAF inhibitor vemurafenib in combination with the MEK inhibitor cobimetinib is FDA-approved for patients with BRAF V600E-positive melanoma, non-small cell lung cancer, and anaplastic thyroid cancer, and is NCCN-listed for patients with low-grade glioma, anaplastic glioma, and glioblastoma. Patients with various glioma subtypes harboring BRAF V600E mutations have shown responses to RAF and MEK inhibition. Three out of five patients with anaplastic astrocytoma had a response (one partial response, two stable disease) to vemurafenib monotherapy (PMID: 30351999). One patient with high-grade glioma (NOS) had stable disease in response to vemurafenib monotherapy (PMID:30351999). Four out of eight patients with glioma (NOS) had a response (one partial response, three stable disease, two progressive disease, two not evaluable) to vemurafenib monotherapy (PMID: 26287849). One patient with anaplastic oligoastrocytoma had stable disease in response to vemurafenib monotherapy (PMID: 30462564). Four of seven patients with glioblastoma multiforme had a response (one complete response, three stable disease, one progressive disease, two not evaluable) to vemurafenib monotherapy (PMID: 30351999, 24725538). Four patients with glioblastoma (including epithelioid glioblastoma) had a response (one near-complete response, three stable disease) to either dabrafenib monotherapy (n=2) or vemurafenib monotherapy (n=2) (PMID: 29621181, 29039591, 33117675, 34232949).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 16:39:14',NULL,'LEVEL_2','NO','LEVEL_3B','7b50f8a8-5662-4ad0-b63a-f2a0cf2c9fee',673,NULL),(2477,NULL,'The RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E-mutant solid tumors.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-30 14:27:05',NULL,NULL,NULL,NULL,'5be59610-1305-4ace-a94c-30ef26d7523a',673,NULL),(2478,NULL,'The RAF inhibitor vemurafenib in combination with the MEK inhibitor cobimetinib is FDA-approved for patients with melanoma, non-small cell lung cancer, and anaplastic thyroid cancer, and is NCCN-listed for patients with low-grade glioma, anaplastic glioma, and glioblastoma. Patients with various glioma subtypes harboring BRAF V600E mutations have shown responses to RAF and MEK inhibition. Six patients with ganglioglioma had a partial response to either vemurafenib monotherapy (n=2), vemurafenib + cobimetinib (n=1), or dabrafenib monotherapy (n=3)(PMID: 25524464, 31985841, 26579623, 31502039). Twelve out of thirteen patients with pleomorphic xanthoastrocytoma had a response (one complete response, one near-complete response, six partial response, four stable disease) to either dabrafenib + trametinib (n=2) or vemurafenib monotherapy (n=11) (PMID: 28984141, 26287849, 30351999). Three patients with anaplastic pleomorphic xanthoastrocytoma had a response (one complete response, one near-complete response, one partial response) to either vemurafenib monotherapy (n=1) or dabrafenib monotherapy (n=2)(PMID: 25092772, 29039591). One patient with low-grade glioma, NOS had a partial response to dabrafenib monotherapy (PMID: 27398937). Two patients with pilocytic astrocytoma had a response (one partial response, one stable disease) to vemurafenib monotherapy (PMID: 30351999). Three of four patients with anaplastic ganglioglioma had a response (one partial response, one \"significant response\", one stable disease, one not evaluable) to either dabrafenib + trametinib (n=1) or vemurafenib monotherapy (n=3)(PMID: 29380516, 30351999). One patient with pilomyxoid astrocytoma had a partial response to vemurafenib monotherapy (PMID: 24821190).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 16:39:40',NULL,'LEVEL_2','NO','LEVEL_3B','8a864f31-163a-4868-bf67-6b1fc449a2c6',673,NULL),(2479,NULL,'The RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E-mutant solid tumors.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-30 14:00:21',NULL,NULL,NULL,NULL,'26d2dac0-be4d-408b-9440-14131797c627',673,NULL),(2480,NULL,'Dabrafenib, an orally bioavailable RAF inhibitor, and trametinib, an orally bioavailable MEK1/2 inhibitor, are FDA-approved in combination for the treatment of patients with solid tumors other than colorectal harboring BRAF V600E mutation. FDA approval was based on data from 131 adult patients with solid tumors treated with dabrafenib and trametinib in the BRF117019 and NCI-MATCH trials and 36 pediatric patients treated with dabrafenib and trametinib in the CTMT212X2101 study. Of the 131 adult patients treated with dabrafenib and trametinib, the overall response rate was 41% (54/131; 95% CI = 33-50) and of the 36 pediatric patients treated with dabrafenib and trametinib (low-grade glioma, n=34; high-grade glioma, n=2), the overall response rate was 25% (95% CI = 12-24) (PMID: 32818466, 34838156, 32758030)(Abstract: Bouffet et al. Abstract# LGG-49, Neuro-Oncology 2020. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7715318/). Approval was supported by results in COMBI-d, COMBI-v and BRF113928 studies in melanoma and lung cancer (PMID: 23020132, 25399551, 27283860). In the Phase II ROAR basket trial of dabrafenib + trametinib in patients with BRAF V600E-mutated rare cancers (biliary tract cancer, n=43), the overall response rate was 53%, the median progression-free survival was nine months (95% CI = 5.5, 9.4) and the median overall survival was 13.5 months (95% CI = 10.4, 17.6) (PMID: 37059834). Of the 43 patients with biliary tract cancer, 53% of patients (23/43) had a partial response and 37% (16/43) had stable disease (PMID: 37059834). Additionally, in subprotocol H of the NCI-MATCH basket trial of dabrafenib + trametinib in patients with BRAF V600E-mutant solid tumors (n=4 patients with intrahepatic cholangiocarcinoma), the overall response rate was 100%, with 4/4 patients achieving a partial response (PMID: 32758030).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-05-23 18:30:56',NULL,'LEVEL_1','NO','LEVEL_3B','6d286407-4acf-4feb-9649-472fbb1b95fe',673,NULL),(2481,NULL,'The RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E-mutant solid tumors.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2022-08-30 14:36:04',NULL,NULL,NULL,NULL,'a1317432-a427-476c-86f1-84c15b50d5c5',673,NULL),(2482,NULL,'Dabrafenib, an orally bioavailable RAF inhibitor, and trametinib, an orally bioavailable MEK1/2 inhibitor, are FDA-approved in combination for the treatment of patients with solid tumors other than colorectal harboring BRAF V600E mutation. FDA approval was based on data from 131 adult patients with solid tumors treated with dabrafenib and trametinib in the BRF117019 and NCI-MATCH trials and 36 pediatric patients treated with dabrafenib and trametinib in the CTMT212X2101 study. Of the 131 adult patients treated with dabrafenib and trametinib, the overall response rate was 41% (54/131; 95% CI = 33-50) and of the 36 pediatric patients treated with dabrafenib and trametinib (low-grade glioma, n=34; high-grade glioma, n=2), the overall response rate was 25% (95% CI = 12-24) (PMID: 32818466, 34838156, 32758030)(Abstract: Bouffet et al. Abstract# LGG-49, Neuro-Oncology 2020. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7715318/). In the randomized, Phase II study of dabrafenib and trametinib in 110 patients with BRAF V600mutant pediatric low-grade glioma (dabrafenib + trametinib treatment, n=37; carboplatin + vincristine treatment, n=37), the overall response rate was 47% (95% CI= 35%-59%) with dabrafenib and trametinib and 11% (95% CI= 3%-25%) with carboplatin and vincristine, and the progression-free survival was 20.1 months (95% CI= 12.8 mo-not estimable) with dabrafenib and trametinib and 7.4 months (95% CI= 3.6-11.8 mo) with carboplatin and vincristine (Abstract: Bouffet et al. Abstract# LBA2002, ASCO 2022. https://ascopubs.org/doi/abs/10.1200/JCO.2022.40.17_suppl.LBA2002). FDA approval was supported by results in COMBI-d, COMBI-v and BRF113928 studies in melanoma and lung cancer (PMID: 23020132, 25399551, 27283860).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:35:41',NULL,'LEVEL_1','NO','LEVEL_3B','6d039136-d681-4915-a342-16e424f5e42c',673,NULL),(2483,NULL,'The pan-RAF kinase inhibitor tovorafenib is FDA-approved for the treatment of patients 6 months of age and older with BRAF V600-mutant pediatric low-grade glioma. Additionally, the RAF inhibitor dabrafenib in combination with the MEK1/2 inhibitor trametinib is FDA-approved for the treatment of patients 1 year of age and older with BRAF V600E-mutant pediatric low-grade glioma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-06-21 09:55:51',NULL,NULL,NULL,NULL,'65384f19-67f6-4187-a4e9-b0e67c0d2d44',673,NULL),(2484,NULL,'Dabrafenib, a small molecule RAF inhibitor, and trametinib, a small molecule MEK1/2 inhibitor, are FDA-approved in combination for the treatment of pediatric patients 1 year of age and older with low-grade glioma (LGG) with a BRAF V600E mutation who require systemic therapy. FDA approval is based on the results of the Phase II Study CDRB436G2201 (NCT02684058) of dabrafenib plus trametinib versus carboplatin plus vincristine in 110 patients with BRAF V600-mutant LGG, as assessed by local or central laboratory tests. In the Phase II Study CDRB436G2201 (NCT02684058), the dabrafenib plus trametinib cohort (n=73 [n=70, V600E; n=3, other]) demonstrated an overall response rate (ORR) of 47% (95% CI=35-59), with a 3% (n=2) complete response (CR) rate, a 44% (n=32) partial response (PR) rate and a 41% (n=30) stable disease (SD) rate, and a median duration of response (DOR) of 20.3 months (95% CI=12.0-NE) while the chemotherapy cohort (n=37 [n=35, V600E; n=1, other; n=1, nonmutant]) demonstrated an ORR of 11% (95% CI=3-25) (OR=7.19 [95% CI=2.30-22.40]; RR=4.31 [95% CI=1.70-11.20]; p<0.001), with a 3% (n=1) CR rate, an 8% (n=3) PR rate and a 41% (n=15) SD rate, and a median DOR that was not estimable (95% CI=6.6-NE) (PMID: 37733309).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2024-06-21 09:49:51',NULL,'LEVEL_1','NO','LEVEL_3B','8d8631fe-755f-4d9a-a505-578963eb0f4d',673,NULL),(2485,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'f48ceb40-9fac-4681-90d7-163ed51dd6e2',673,NULL),(2486,NULL,'The class II BRAF L597S mutation is located in the kinase domain of the protein (PMID: 33019809). This mutation has been found in melanoma, among other cancers (PMID: 26643848). In vitro studies have demonstrated that this mutation is likely activating as measured by increased downstream signaling compared to wildtype (PMID: 22798288). Drug efficacy studies in cell lines demonstrated that the BRAF L597S mutation might be sensitive to the RAF inhibitor vemurafenib and the MEK inhibitor trametinib (PMID: 22798288). The BRAF L597S mutation has also been identified in patients with positive responses to trametinib, the allosteric MEK inhibitor TAK-733 and combined dabrafenib and trametinib treatment (PMID: 24933606, 22798288, 29903896).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-08 15:57:49',NULL,NULL,NULL,NULL,'bd08e589-0a0b-4ff9-b514-fe3fe1d84f30',673,NULL),(2487,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2018-08-17 14:01:34',NULL,NULL,NULL,NULL,'68016989-6e6d-4221-98d3-238110b00ac6',673,NULL),(2488,NULL,'The class II BRAF K601E mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719, 26343582). This mutation is recurrent in melanoma (PMID: 22798288, 26643848). Biological characterization of BRAF K601E mutations demonstrated that they activate the downstream MAPK pathway independently of RAS and render BRAF active as a dimer with CRAF and itself (PMID: 22798288, 20179705, 26343582). Preclinical studies with patient-derived xenograft models of colorectal cancer expressing BRAF K601E demonstrated sensitivity to treatment with PLX8394 with or without cetuximab as measured by reduced tumor growth, and resistance to treatment with cetuximab and vemurafenib as measured by no reduction in tumor growth upon treatment (PMID: 30559419, 31515458). A patient with metastatic BRAF K601E-mutated melanoma achieved a complete response to the MEK inhibitor trametinib (PMID: 28344857). In a case report, a patient with lung adenocarcinoma harboring BRAF K601E was treated with a combination therapy of dabrafenib plus trametinib and demonstrated partial response with a progression-free survival of nine months (PMID: 34590045). In another case report, a patient with papillary thyroid carcinoma harboring BRAF K601E was treated with a combination therapy of sorafenib plus denosumab and demonstrated a complete response (PMID: 35630083). In a Phase II trial, a patient with melanoma harboring BRAF K601E was treated with trametinib and demonstrated partial response with a progression-free survival of 32 weeks (PMID: 23248257). In another Phase II trial, six patients with advanced solid tumors harboring BRAF K601E were treated with vemurafenib and did not clinically respond to treatment (PMID: 29320312). In a Phase I trial, two patients with melanoma harboring BRAF K601E were treated with dabrafenib and did not clinically respond to treatment (PMID: 22608338).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:29:35',NULL,NULL,NULL,NULL,'9b64a01a-fe4a-4764-bb83-50a7dddc2093',673,NULL),(2489,NULL,'The RAF-targeted inhibitor vemurafenib is FDA-approved for the treatment of patients with BRAF V600-mutant Erdheim-Chester disease.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 12:26:19',NULL,NULL,NULL,NULL,'39484c36-282c-4392-909d-cd9ab20724df',673,NULL),(2490,NULL,'Vemurafenib is an orally available, small molecule RAF-targeted inhibitor that is FDA-approved for the treatment of patients with Erdheim-Chester disease (ECD) with BRAF V600 mutation. FDA approval was based on the results of the Phase II VE-BASKET (NCT01524978) trial of vemurafenib in 26 patients with BRAF V600-mutant non-melanoma cancers (n=22, ECD; n=4, Langerhans cell histiocytosis [LCH]).\nIn the Phase II VE-BASKET (NCT01524978) trial, the ECD cohort demonstrated an overall response rate of 54.5% (12/22) (95% CI=32.2-75.6), with one patient (4.5%) achieving complete response and eleven patients (50%) achieving partial response, and the median progression-free survival and overall survival were not reached (PMID: 29188284, 26287849).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2025-03-13 16:26:24',NULL,'LEVEL_1','NO','NO','729d93d7-aa18-4bf0-9c84-13b9ecce25fb',673,NULL),(2491,NULL,'The RAF-targeted inhibitors vemurafenib and dabrafenib are NCCN-listed as monotherapy for the treatment of BRAF V600E-mutant Langerhans Cell Histiocytosis and vemurafenib is FDA-approved for the treatment of BRAF V600-mutant Erdheim Chester Disease.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-10-01 11:54:06',NULL,NULL,NULL,NULL,'b1acc1d7-c638-4217-a6e4-986b99f66f0a',673,NULL),(2492,NULL,'Vemurafenib is an orally available, small molecule RAF-targeted inhibitor that is FDA-approved for the treatment of patients with Erdheim-Chester disease (ECD) with BRAF V600 mutation. Vemurafenib and dabrafenib are listed as monotherapies in the NCCN Histiocytic Neoplasms Guidelines (V3.2024) under the section \"Principles of Systemic Therapy\" as recommended treatment regimens for patients with Langerhans cell histiocytosis (LCH) harboring BRAF V600E mutations. NCCN recommendation was based on the results of the Phase II VE-BASKET (NCT01524978) and case reports.\nIn the Phase II VE-BASKET (NCT01524978) trial, all four patients of the LCH cohort achieved partial response (PMID: 29188284). In the case series evaluating the use of BRAF inhibitors in six adult patients with BRAF V600Emutant LCH, patients treated with vemurafenib monotherapy (n=3) achieved one complete response (CR), one partial response (PR) and one stable disease (SD) response while patients treated with dabrafenib monotherapy (n=3) achieved one CR and two PRs (PMID: 32985015). In a separate case report, a patient with BRAF V600E-mutant LCH was treated with dabrafenib plus trametinib and demonstrated a sustained metabolic response (PMID: 30154124).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2025-03-13 16:44:27',NULL,'LEVEL_2','NO','NO','a0a7271e-c46e-43b8-8f85-7e1418192d6c',673,NULL),(2493,NULL,'The FDA-approved RAF-inhibitors dabrafenib and vemurafenib have minimal activity as single agents in BRAF V600-mutant colorectal cancer (CRC). The triplet therapy combination of encorafenib + binimetinib + cetuximab, a RAF-inhibitor, a MEK-inhibitor and an anti-EGFR antibody respectively, is NCCN-compendium listed and has received FDA-breakthrough therapy designation for the treatment of patients with BRAF V600E mutant colorectal cancer.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 14:11:38',NULL,NULL,NULL,NULL,'8d714d43-7ccc-416f-99b8-4f49cc046590',673,NULL),(2494,NULL,'The RAF-targeted inhibitors encorafenib, dabrafenib and vemurafenib, alone or in combination with the MEK1/2-targeted inhibitors binimetinib, trametinib and cobimetinib respectively, are FDA-approved for the treatment of patients with BRAF V600E/K mutant melanoma and NCCN-compendium listed for the treatment of patients with BRAF V600-mutant melanoma. Additionally, the anti-PD-L1 antibody atezolizumab in combination with cobimetinib + vemurafenib is FDA-approved for the treatment of patients with unresectable or advanced BRAF V600-mutant melanoma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2020-08-24 16:58:52',NULL,NULL,NULL,NULL,'49c8da54-77d7-4480-af56-bd58072c54c0',673,NULL),(2495,NULL,'The combination of vemurafenib, an inhibitor of V600-mutant BRAF, and cobimetinib, an inhibitor of MEK1/2, with atezolizumab, an immunotherapeutic PD-L1 antibody, is FDA-approved for patients with BRAF V600 mutation-positive unresectable or metastatic melanoma. FDA approval was based on the results of the Phase III double-blind, randomized, placebo-controlled IMspire150 trial of Atezolizumab + Cobimetinib + Vemurafenib versus Placebo + Cobimetinib + Vemurafenib in 514 patients with BRAF V600-mutant melanoma in which the median progression-free survival was 15.1 mos (95% CI=11.4,18.4) in the triplet arm versus 10.6 mos (95% CI=9.3,12.7) in the doublet + placebo arm (HR=0.78; 95% CI= 0.63, 0.97; p=0.0249) (PMID: 32534646).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 14:56:33',NULL,'LEVEL_1','NO','LEVEL_3B','2e0a9edd-9248-4da3-b5f8-acdfe19039d6',673,NULL),(2496,NULL,'While the RAF-targeted inhibitor dabrafenib in combination with the MEK1/2-targeted inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E mutant non-small cell lung cancer (NSCLC), this combination is not specifically approved for the treatment of patients with BRAF [[mutation]] mutant NSCLC.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-12-18 13:32:08',NULL,NULL,NULL,NULL,'593f072b-2d62-4436-c06f-1d01f05298a8',673,NULL),(2497,NULL,'While the anti-PD-L1 antibody atezolizumab is FDA-approved in combination with the MEK1/2-targeted inhibitor cobimetinib and the RAF-targeted inhibitor vemurafenib for the treatment of patients with BRAF V600-mutant melanoma, and the pan-RAF kinase inhibitor tovorafenib is FDA-approved for the treatment of patients with BRAF V600-mutant pediatric low-grade glioma, the clinical utility of RAF- and MEK-targeted inhibitors (alone or in combination) in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-04-30 12:19:23',NULL,NULL,NULL,NULL,'822ab686-6d4b-4605-8133-9387af0a4fbc',673,NULL),(2498,NULL,'The RAF-targeted inhibitor vemurafenib is FDA-approved for the treatment of patients with BRAF V600-mutant Erdheim-Chester Disease (ECD) and there is promising clinical data in patients with non-ECD BRAF V600-mutant histiocytic neoplasms treated with vemurafenib or dabrafenib as monotherapy.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-10-05 11:03:34',NULL,NULL,NULL,NULL,'86fce41b-4b83-4bff-9d45-d96ea3767cde',673,NULL),(2499,NULL,'Vemurafenib is an orally available, small molecule RAF-targeted inhibitor that is FDA-approved for the treatment of patients with Erdheim-Chester disease (ECD) with BRAF V600 mutation. Vemurafenib and dabrafenib are listed as monotherapies in the NCCN Histiocytic Neoplasms Guidelines (V3.2024) under the section \"Principles of Systemic Therapy\" as recommended treatment regimens for patients with ECD and Langerhans cell histiocytosis (LCH) harboring BRAF V600E mutations. There is promising clinical data to support the use of vemurafenib and dabrafenib as monotherapies in patients with non-ECD and non-LCH BRAF V600-mutant histiocytic neoplasms.\nIn a case study, a patient with heterozygous BRAF V600E-mutant histiocytic sarcoma was treated with vemurafenib and maintained both clinical and radiologic tumor responses for three months until disease progression (PMID: 25209580). In a second case study, a patient with BRAF V600E-mutant histiocytic sarcoma was treated with vemurafenib plus chemotherapy, antibiotics and steroids and remained in complete remission eighteen months after treatment initiation (PMID: 31376203). In a third case study, a pediatric patient with BRAF V600E-mutant mixed systemic Rosai-Dorfman-Destombes disease and LCH reported sustained clinical and radiographic responses after treatment with dabrafenib prior to disease relapse at thirteen months (PMID: 31213430).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2025-03-13 16:05:44',NULL,'LEVEL_3A','NO','NO','ccfa69b9-f284-427a-84b5-a4ffcc5606ca',673,NULL),(2500,NULL,'The pan-RAF kinase inhibitor tovorafenib is FDA-approved for the treatment of patients 6 months of age and older with BRAF V600-mutant pediatric low-grade glioma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-04-26 14:19:32',NULL,NULL,NULL,NULL,'e3ed11df-cbf0-485f-9065-af982e6282ce',673,NULL),(2501,NULL,'Tovorafenib is an orally available, pan-RAF small molecule inhibitor that is FDA-approved for the treatment of pediatric patients six months of age and older with relapsed or refractory pediatric low-grade glioma (LGG) harboring a BRAF fusion or rearrangement, or BRAF V600 mutation. FDA approval was based on the results of the Phase II FIREFLY-1 (NCT04775485) trial of tovorafenib in 76 patients (median age=8 years old [range=2-21]) with relapsed or refractory pediatric LGG harboring an activating BRAF alteration based on local laboratory testing. In the Phase II FIREFLY-1 (NCT04775485) trial, the overall RAPNO-LGG cohort demonstrated an objective response rate (ORR) of 51% (95% CI=40-63), with a 37% (n=28) partial response rate and 14% (n=11) minor response rate, a median duration of response (DOR) of 13.8 months (95% CI=11.3-NE) in 39 patients and a median progression-free survival of 13.8 months (95% CI=8.3-16.9) (PMID: 37978284). The BRAF V600E mutation subcohort (n=12) demonstrated an ORR of 50% (95% CI=21-79) and a median DOR that was not evaluable (95% CI=8.4-NE) (PMID: 37978284).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2024-04-26 14:17:11',NULL,'LEVEL_1','NO','LEVEL_3B','365e9c4e-bc1b-490c-98f7-f122d9504f66',673,NULL),(2502,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'fa5603f0-2ef7-43cf-aeb2-daf1793b84ee',673,NULL),(2503,NULL,'The class I (PMID: 28783719) activating exon 15 BRAF V600K missense mutation is located in the kinase domain of the BRAF protein. BRAF V600 alterations (predominantly V600E) are recurrent in lung (PMID: 25079552, 23833300) and thyroid (PMID: 25417114) cancers, among others. In metastatic melanoma, frequency of V600K mutation is second only to V600E, the most common BRAF alteration (PMID: 26091043). Analysis of clinicopathologic features and outcome in patients with metastatic melanoma harboring activating BRAF mutations revealed that, compared to V600E, V600K mutation was significantly associated with male sex, older age (median age, head/neck primary tumor location, shorter interval to stage IV disease, and shorter overall survival from the time of diagnosis of stage IV disease (PMID: 23922205, 22535154). Comprehensive biological characterization of BRAF V600K mutation (PMID: 15035987, 26343582) has demonstrated that this mutation activates the downstream MAPK pathway independently of RAS (PMID: 19251651, 26343582), renders BRAF constitutively activated in monomeric form (PMID: 20179705) and retains sensitivity to RAF monomer inhibitors such as vemurafenib (PMID: 26343582, 28783719, 20179705). Preliminary studies suggest that BRAF V600K mutant tumors may not respond to combined BRAF and MEK inhibition as well as BRAF V600E tumors due to a decreased reliance on the MAPK pathway, though further studies are needed to validate this finding (PMID: 30630828).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2019-03-05 19:00:20',NULL,NULL,NULL,NULL,'7f32379e-0513-4cd5-aa97-bf5ebcb9f2fe',673,NULL),(2504,NULL,'The RAF-targeted inhibitors encorafenib, dabrafenib and vemurafenib alone or in combination with the MEK-targeted inhibitors binimetinib, trametinib and cobimetinib, respectively, are FDA-approved for the treatment of patients with BRAF V600E/K mutant melanoma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 14:16:21',NULL,NULL,NULL,NULL,'603dab24-2e01-4179-a0ac-cff1fe8c81ca',673,NULL),(2505,NULL,'While the RAF-targeted inhibitor dabrafenib in combination with the MEK1/2-targeted inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E mutant non-small cell lung cancer (NSCLC), this combination is not specifically approved for the treatment of patients with BRAF V600K mutant NSCLC. However, laboratory data demonstrate that cells expressing BRAF V600K respond similarly to dabrafenib + trametinib as cells with BRAF V600E, and patients with BRAF V600K mutant melanoma derived similar clinical benefit to dabrafenib + trametinib as those with BRAF V600E.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-12-17 17:00:05',NULL,NULL,NULL,NULL,'33bea22d-cd06-4a27-95d1-23125badb1f6',673,NULL),(2506,NULL,'While the MEK1/2-targeted inhibitor trametinib, alone or in combination with the RAF-targeted inhibitor dabrafenib, and the MEK-targeted inhibitor cobimetinib in combination with the RAF-targeted inhibitor vemurafenib are FDA-approved for the treatment of patients with BRAF V600K mutant melanoma, their clinical utility in patients with [[variant]] has yet to be defined.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 14:17:06',NULL,NULL,NULL,NULL,'5719cd2c-bf51-496f-9d5f-287583907e50',673,NULL),(2507,NULL,'Dabrafenib, an orally bioavailable RAF inhibitor, and trametinib, an orally bioavailable MEK1/2 inhibitor, are FDA-approved alone or in combination for the treatment of patients with metastatic melanoma harboring a V600E or V600K BRAF mutation. FDA approval of dabrafenib in combination with trametinib was based on results from an open-label Phase III study of combination therapy versus dabrafenib monotherapy in 247 patients with metastatic melanoma who were naive to treatment with BRAF inhibitors. Combined dabrafenib and trametinib, administered in full monotherapy doses, improved the response rate in patients with BRAF V600-mutant metastatic melanoma versus dabrafenib monotherapy (67% vs.51%; p<0.002). However, median progression-free survival improved by only 2 weeks (9.3 months vs 8.8 months; HR = 0.75) compared with dabrafenib monotherapy (PMID: 23020132). Combination therapy is associated with less cutaneous toxicity than monotherapy, but systemic toxicity may be increased (PMID: 25287827, 25399551). Follow-up trials have demonstrated that all clinical measures inclusive of overall and median progression-free survival as well as objective response rates, median duration of response and number of patients with complete response favored patients treated with combination dabrafenib and trametinib, administered in full monotherapy doses, compared to either dabrafenib or vemurafenib monotherapy, including patients who previously progressed on BRAF inhibitor monotherapy (PMID: 25287827, 25399551, 25265492). Additionally, patients with melanoma treated with dabrafenib and trametinib in both the neoadjuvant and adjuvant settings have improved survival over patients given standard of care (PMID: 29361468, 28991513, 28891408). Promising clinical data has also suggested that addition of an immunotherapy agent to combination RAF and MEK inhibitor treatment may improve rate of overall response and duration of response in melanoma patients (PMID: 31171876, 31171879, 31171878).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:53:07',NULL,'LEVEL_1','NO','LEVEL_3B','bc28ec2e-60ae-4a5d-9401-524e357b5bed',673,NULL),(2508,NULL,'Trametinib is an oral small molecule inhibitor of MEK1/2 that is FDA-approved alone or with dabrafenib for the treatment of patients with metastatic melanoma harboring a V600E or V600K BRAF mutation. In an open-label, randomized Phase III trial, patients with BRAF V600E/K-mutated unresectable, metastatic melanoma received oral trametinib (2 mg once daily) or an intravenous regimen of either dacarbazine (1000 mg/m2) or paclitaxel (175 mg/m2) every three weeks. Trametinib demonstrated improved progression-free survival (HR for disease progression or death = 0.45) and six-month overall survival (81% vs. 67%; death HR = 0.54; p=0.01) (PMID: 22663011). However, like other MEK inhibitors, the benefit of trametinib is limited by adverse reactions, most notably grade three or four rash and diarrhea (PMID: 22663011). Trametinib is not typically used as monotherapy for patients with BRAF V600K melanoma given its lower response rate compared to BRAF inhibitors and combined BRAF and MEK inhibitors. Patients previously treated with a RAF inhibitor appear to be less likely than untreated patients to respond to trametinib treatment (PMID: 22663011), and FDA guidelines state that trametinib as a monotherapy is not indicated for these patients. Dabrafenib and trametinib are FDA-approved as a combination therapy, which has superior clinical outcomes compared to dabrafenib or trametinib monotherapy (PMID: 25399551, 25265492). Additionally, patients with melanoma treated with dabrafenib and trametinib in both the neoadjuvant and adjuvant settings had improved survival over patients given standard of care (PMID: 29361468).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:52:56',NULL,'LEVEL_1','NO','LEVEL_3B','873eaae0-4bf1-4d6d-a312-f8ff9770f393',673,NULL),(2509,NULL,'The RAF inhibitor vemurafenib in combination with the MEK inhibitor cobimetinib is FDA-approved for the treatment of patients with BRAF V600-mutant metastatic or unresectable locally advanced melanoma. FDA approval was based on results from the randomized Phase III coBRIM trial of combined vemurafenib and cobimetinib versus vemurafenib and placebo in 495 patients with metastatic BRAF V600-mutant melanoma that demonstrated superior clinical benefit measures in the combination versus the control arm. Specifically, median progression-free survival was 9.9 months in the combination arm versus 6.2 months in the control arm (HR = 0.51), with a complete response rate of 10% versus 4%, respectively, and interim nine-month overall survival of 81% versus 73%, respectively (PMID: 25265494). Follow-up analysis of the coBRIM trial showed two-year overall survival was 48.3% in the combination group versus 38.0% in the monotherapy group (PMID: 27480103), and five-year followup of the BRIM7 study showed a median overall survival of 31.8 months and a five-year survival rate of 39.2% (PMID: 31732523).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:41:59',NULL,'LEVEL_1','NO','LEVEL_3B','19caa107-e92a-4246-9d01-6d707b9ecd4e',673,NULL),(2510,NULL,'The combination of encorafenib, an inhibitor of V600E- or V600K-mutant BRAF, and binimetinib, an inhibitor of MEK1/2, is FDA-approved in combination for patients with unresectable or metastatic melanoma with a BRAF V600E or V600K mutation. FDA approval was based on the results of the Phase III COLUMBUS trial of combined encorafenib plus binimetinib versus single-agent vemurafenib in 577 patients with BRAF V600E- or V600K-mutant metastatic melanoma in which the median progression-free survival was 14.9 months (95% CI = 11.0-18.5) in the encorafenib plus binimetinib group versus 7.3 months (95% CI = 5.6-8.2) in the single agent vemurafenib group (HR= 0.54, 95% CI = 0.41-0.71; p<00001) (PMID: 29573941). In the five-year update of the Phase III COLUMBUS trial, the progression-free survival and overall survival were 23% and 35% respectively in the encorafenib plus binimetinib group (n=192) versus 10% and 21% respectively in the single agent vemurafenib group (n=191), and the median duration of response and disease control rate were 18.6 months and 92.2% in the encorafenib plus binimetinib group versus 12.3 months and 81.2% in the single-agent vemurafenib group (PMID: 35862871).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 16:42:13',NULL,'LEVEL_1','NO','LEVEL_3B','b76c8407-558c-49e9-ab3c-90a181df0044',673,NULL),(2511,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'dc31c7a1-bb00-4f06-975e-5b2fc013f800',673,NULL),(2512,NULL,'The class I activating exon 15 BRAF V600D mutation is located in the kinase domain of the protein (PMID: 28783719). This mutation is recurrent in melanoma (PMID: 26091043, 23579220, 22614978). Comprehensive biological characterization of the BRAF V600D mutation has demonstrated that this mutation activates the downstream MAPK pathway independently of RAS and renders BRAF constitutively activated in monomeric form (PMID: 15035987, 12068308, 19251651, 26343582, 20179705). Preclinical studies with melanoma cells expressing BRAF V600D demonstrate sensitivity to treatment with CCT196969, CCT241161, AZ628, TAK-632, PLX7904, vemurafenib, dabrafenib, ASTX029 and cobimetinib plus HM95573 as measured by inhibition of MEK and ERK phosphorylation and cellular proliferation (PMID: 25500121, 17210691, 27523909, 33953400, 34330842). In a case report, a patient with pilocytic astrocytoma harboring BRAF V600D was treated with single-agent dabrafenib then subsequently treated with combination therapy of dabrafenib plus trametinib and demonstrated stable disease (PMID: 28784858). In a Phase II trial, a patient with non-small cell lung cancer harboring BRAF V600D was treated with vemurafenib and demonstrated partial response with a progression-free survival of 3.8 months (PMID: 31959346).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:32:10',NULL,NULL,NULL,NULL,'2d9f29be-c010-4fd7-8bf1-724be975d211',673,NULL),(2513,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'67171333-e67d-4efb-b109-1ec059584108',673,NULL),(2514,NULL,'The class I activating exon 15 BRAF V600R missense mutation is located in the kinase domain of the BRAF protein (PMID: 28783719). BRAF V600R is the third most common mutation occurring in 5-7% of patients with BRAF-mutant melanoma (PMID: 22536370, 23237741, 22614978). Comprehensive biological characterization of the BRAF V600R mutation has demonstrated that this mutation activates the downstream MAPK pathway independently of RAS and renders BRAF constitutively activated in monomeric form (PMID: 15035987, 28783719, 26343582, 20179705). Preclinical studies with melanoma cells expressing BRAF V600R demonstrate sensitivity to treatment with AZ628, TAK-632, PLX7904, vemurafenib and dabrafenib as measured by inhibition of cellular proliferation (PMID: 27523909). In a case report, a patient with melanoma harboring BRAF V600R was treated with a combination therapy of dabrafenib plus trametinib and demonstrated partial response with a progression-free survival of 52.2 months (PMID: 36039514). In another case report, a patient with melanoma harboring BRAF V600R was treated with dabrafenib and demonstrated partial response with lesion reduction and progression-free survival of seven months (PMID: 27255157).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:33:40',NULL,NULL,NULL,NULL,'12d0ecbc-50f2-4179-b883-236acf79ec53',673,NULL),(2515,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'fe0fcc32-3e63-485c-9c5c-e71282f366da',673,NULL),(2516,NULL,'The class I (PMID: 28783719) activating exon 15 BRAF V600G mutation is located in the kinase domain of the protein. This mutation has been found in cardio-facio-cutaneous syndrome (CFC), lung cancer and esophagogastric cancer (PMID: 26287849). Substitutions at position V600 of BRAF, including the common BRAF V600E mutation, are known to be activating and oncogenic (PMID: 12068308, 16273091, 15035987). Expression of this mutation in 293T cells demonstrated that this mutation is activating as measured by increased downstream ERK activation and increased transcription of downstream MAPK pathway transcriptional targets (e.g. ELK) compared to wildtype BRAF (PMID: 20735442). A patient with non-small cell lung cancer (NSCLC) harboring the BRAF V600G mutation had stable disease in response to treatment with vemutafenib, while a patient with sarcoma and a patient with esophagogastric cancer both harboring the BRAF V600G mutation had progressive disease in response to vemurafenib (PMID: 26287849).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-09-27 11:03:34',NULL,NULL,NULL,NULL,'aa1b3f05-546a-4bd5-a85e-9f826728ce4d',673,NULL),(2517,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2021-12-28 10:31:16',NULL,NULL,NULL,NULL,'1e543634-e853-4b7c-b61e-76537ae06c5c',673,NULL),(2518,NULL,'The BRAF V600M mutation is located in the BRAF kinase domain. This mutation has been found in melanoma (PMID: 25382067, 22536370). In vitro studies have demonstrated that this mutation, similar to other class 1 BRAF V600 mutations, activates the downstream MAPK pathway independent of RAS, as measured by increased levels of phosphorylated MEK and decreased RAS-GTP in V600E mutant cells compared to wildtype (PMID: 26343582, 28783719). The BRAF V600M mutation renders BRAF constitutively activated in monomeric form and is sensitive to inhibition by vemurafenib (PMID: 28783719, 25382067, 26343582). However, It is unknown whether the BRAF V600M mutation actually arises in cancer as noted by several studies (PMID: 22536370, 23579220, 25382067) or previous sequencing studies miscalled a V600R dinucleotide change (chromosome 7: position 140453136 AC > CT) as a single nucleotide change (chromosome 7: position 140453137 C >T).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2021-12-28 10:31:15',NULL,NULL,NULL,NULL,'838d4e27-f794-42e0-8f8e-d6b4955c1b4e',673,NULL),(2519,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'af609f8b-ee1e-4ff5-a363-2327464330ab',673,NULL),(2520,NULL,'The class II BRAF L597R mutation is located in the kinase domain of the protein (PMID: 33019809). This mutation has been found in melanoma, among other cancers (PMID: 26643848). In vitro studies have demonstrated that this mutation is likely activating as measured by increased downstream signaling compared to wildtype (PMID: 22798288). Drug efficacy studies in cell lines demonstrated that the BRAF L597R mutation might be sensitive to the RAF inhibitor vemurafenib and the MEK inhibitor trametinib (PMID: 22798288). The BRAF L597R mutation has also been identified in patients with positive responses to trametinib and the allosteric MEK inhibitor TAK-733 (PMID: 24933606). In a case report, a patient with melanoma harboring BRAF L597R was treated with vemurafenib and demonstrated partial response with 30% growth inhibition of skin and lung nodules (PMID: 23715574). In another case report, a patient with lung adenocarcinoma harboring BRAF L597R was treated with a combination therapy of dabrafenib and trametinib and demonstrated partial response with a progression-free survival of twelve months (PMID: 32540409). In another case report, a patient with colorectal cancer harboring BRAF L597R was treated with a combination therapy of panitumumab plus FOLFIRI and demonstrated stable disease with a progression-free survival of 16.5 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 18:35:08',NULL,NULL,NULL,NULL,'57225720-7b20-42b8-aed1-561e9d1a6922',673,NULL),(2521,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'d138d84d-4bcf-4d28-a586-c09e8fcd6c80',673,NULL),(2522,NULL,'The class II BRAF L597V mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719, 26343582). This mutation is recurrent in melanoma (PMID: 12068308, 15035987, 26643848). Biological characterization of BRAF L597V mutation demonstrated that it activates the downstream MAPK pathway independent of RAS and renders BRAF active as a dimer with CRAF and itself (PMID: 12068308, 15035987, 22798288, 26343582, 28783719, 20179705). Preclinical studies with patient-derived lung adenocarcinoma cells expressing BRAF L597V demonstrated sensitivity to treatment with dabrafenib plus trametinib, trametinib, BGB-283 and LXH 254, as measured by cellular proliferation inhibition, and resistance to treatment with dabrafenib, encorafenib, vemurafenib, TAK-580, LY3214996, MK-8353, GDC-0994 and BVD-523, as measured by sustained cellular proliferation upon drug treatment (PMID: 32540409). In a Phase II trial, a patient with endometrial adenocarcinoma harboring BRAF L597V was treated with trametinib and demonstrated stable disease with a progression-free survival of 7.9 months (PMID: 31924734).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:36:20',NULL,NULL,NULL,NULL,'a9e6b8b3-d1fb-4685-a5ec-78811f7bb602',673,NULL),(2523,NULL,'There is promising clinical data in patients with BRAF L597-mutant melanoma treated with the FDA-approved MEK-targeted inhibitor trametinib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 12:14:20',NULL,NULL,NULL,NULL,'30a2e81e-265e-4103-ab0b-30bc2e3da638',673,NULL),(2524,NULL,'Trametinib is a small-molecule inhibitor of the MEK1/2 kinases that is FDA-approved in combination with a RAF inhibitor for use in BRAF V600E-positive metastatic melanoma. \nIn a Phase II (NCT02296112) trial of trametinib in patients with non-BRAF V600-mutant melanoma, a patient harboring BRAF L597Q demonstrated partial response with a tumor reduction of 38% and an ongoing progression-free survival of 8.3 months (PMID: 33861486). \nIn vitro studies of cell lines engineered to express BRAF L597R or L597S mutations demonstrated sensitivity to MEK inhibition as measured by decreased activation of downstream effector proteins upon treatment with trametinib (PMID: 22798288, 29903896).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2025-03-12 15:13:20',NULL,'LEVEL_3A','NO','LEVEL_3B','af5ea563-3d74-49e1-b931-ebb6f3fdeb43',673,NULL),(2525,NULL,'While there is promising clinical data supporting the use of the FDA-approved MEK-inhibitor trametinib in patients with BRAF L597-mutant melanoma, its clinical utility in patients with [[variant]] is unknown. BRAF class II mutations including [[gene]] [[mutation]] are insensitive to the RAF inhibitors vemurafenib and dabrafenib. Laboratory data suggest that cancer cells with oncogenic, BRAF class II mutations may be sensitive to novel BRAF inhibitors such as plixorafenib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-06-10 17:54:52',NULL,NULL,NULL,NULL,'9b3811aa-85b0-40d0-90fc-f5c724b37f06',673,NULL),(2526,NULL,'Plixorafenib is an orally available, second-generation, small-molecule BRAF inhibitor that inhibits mutated, oncogenic BRAF without causing paradoxical MAPK pathway activation as is seen with first-generation BRAF inhibitors (PMID: 28659148). In a phase I clinical trial for patients with refractory or relapsed solid tumors with or without BRAF mutation, there were two (28%) partial responses of seven evaluable BRAF-mutant patients treated with plixorafenib and cobicistat, a CYP3A4 inhibitor (Abstract: Janku et al. Abstract# B176, AACR-NCI-EORTC 2017. http://mct.aacrjournals.org/content/17/1_Supplement/B176). Both responses were in patients with BRAF V600 mutations. Preclinical studies show that plixorafenib specifically inhibits BRAF-containing dimers and inhibits proliferation and pathway activation in BRAF-mutated colorectal cancer cell lines (PMID: 26466569, 28659148, 30559419, 28783719).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2024-06-10 17:54:43',NULL,'LEVEL_4','NO','LEVEL_4','1fd2d2b7-b69a-4e58-b205-81ef99c8c2e9',673,NULL),(2527,NULL,'There is promising clinical data in patients with BRAF K601E mutant melanoma treated with the FDA-approved MEK-targeted inhibitor trametinib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 12:08:16',NULL,NULL,NULL,NULL,'f54dc0ea-805f-407b-acc0-878409be447b',673,NULL),(2528,NULL,'Trametinib is a small molecule inhibitor of the MEK1/2 kinases that is FDA-approved in combination with a RAF inhibitor for use in BRAF V600E-positive metastatic melanoma. Anecdotal evidence of clinical activity of trametinib in BRAF K601-mutant melanoma includes a patient who experienced a complete response to trametinib over 36 months (PMID: 28344857), and four patients in various other studies who achieved a partial response or stable disease upon treatment with trametinib (PMID: 23248257, 24933606). In vitro studies of cell lines engineered to express BRAF K601E demonstrated sensitivity to MEK inhibition as measured by decreased activation of downstream effector proteins upon treatment with trametinib (PMID: 22798288, 29903896).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 14:52:16',NULL,'LEVEL_3A','NO','LEVEL_3B','b17106e4-9dd3-4157-90d2-0e791c15278b',673,NULL),(2529,NULL,'While there is promising clinical data supporting the use of the MEK-targeted inhibitor trametinib in patients with BRAF K601E mutant melanoma, its clinical utility in patients with [[variant]] is unknown. BRAF class II mutations including [[gene]] [[mutation]] are insensitive to the RAF inhibitors vemurafenib and dabrafenib. Laboratory data suggest that cancer cells with oncogenic, BRAF mutations may be sensitive to novel BRAF inhibitors such as plixorafenib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-06-10 17:53:55',NULL,NULL,NULL,NULL,'e6703560-987d-4350-a9aa-931bdaea463b',673,NULL),(2530,NULL,'Plixorafenib is an orally available, second-generation, small-molecule BRAF inhibitor that inhibits mutated, oncogenic BRAF without causing paradoxical MAPK pathway activation as is seen with first-generation BRAF inhibitors (PMID: 28659148). In a phase I clinical trial for patients with refractory or relapsed solid tumors with or without BRAF mutation, there were two (28%) partial responses of seven evaluable BRAF-mutant patients treated with plixorafenib and cobicistat, a CYP3A4 inhibitor (Abstract: Janku et al. Abstract# B176, AACR-NCI-EORTC 2017. http://mct.aacrjournals.org/content/17/1_Supplement/B176). Both responses were in patients with BRAF V600 mutations. In a phase I/IIa trial of plixorafenib for patients with BRAF-altered, advanced solid or CNS tumors (n=110; n=61, patients with class 1 mutations; n=19, patients with class 2 mutations [excluding fusions]; n=17, patients with BRAF fusions), partial response was achieved in 39% (n=9/23) of patients with MAPKi-naive, V600-mutant tumors, with a median duration of response (DoR) of 32 months, and in 18% (n=3/17) of patients with MAPKi treated, V600-mutant tumors (n=6, patients with V600-mutant glioma; n=3, patients with V600-mutant ovarian cancer; n=1, patients with V600-mutant small bowel cancer; n=1, patients with V600-mutant papillary thyroid cancer; n=1, patients with V600-mutant anaplastic thyroid cancer) (Abstract: Fuente et al. Abstract# 3006, J. Clin. Oncol. 2023, Vol 41, Suppl 16. https://ascopubs.org/doi/10.1200/JCO.2023.41.16_suppl.3006). Preclinical studies show that plixorafenib specifically inhibits BRAF-containing dimers and inhibits proliferation and pathway activation in BRAF-mutated colorectal cancer cell lines (PMID: 26466569, 28659148, 30559419, 28783719).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2024-06-10 17:53:43',NULL,'LEVEL_4','NO','LEVEL_4','690ed068-5612-438a-ac07-948a18228375',673,NULL),(2531,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'b3965dfe-f320-4ad5-8f89-7d2f351908f0',673,NULL),(2532,NULL,'The class II BRAF K601Q mutation is located in the kinase domain in exon 15 of the protein (PMID: 33019809). This mutation has been found in patients with cardio-facio-cutaneous syndrome (PMID: 19206169). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by increased downstream pathway activation compared to wildtype BRAF, though to a lesser extent than BRAF V600E (PMID: 19206169). Preclinical studies with melanoma cells expressing BRAF K601Q demonstrated resistance to treatment with vemurafenib as measured by sustained MEK and ERK phosphorylation following treatment (PMID: 35045286).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-08 15:07:15',NULL,NULL,NULL,NULL,'bf3ce630-ff39-4267-a35a-5ea53feee665',673,NULL),(2533,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:37:28',NULL,NULL,NULL,NULL,'a4f10719-c5cb-4074-b753-acbe2a67dd47',673,NULL),(2534,NULL,'The class II BRAF L597Q mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719, 26343582). This mutation is recurrent in melanoma (PMID: 12068308, 15035987, 26643848). Biological characterization of BRAF L597Q demonstrated that it activates the downstream MAPK pathway independent of RAS and renders BRAF active as a dimer with CRAF and itself (PMID: 12068308, 15035987, 22798288, 26343582, 28783719). Preclinical studies with patient-derived colon adenocarcinoma cells expressing BRAF L597Q demonstrated sensitivity to treatment with trametinib, selumetinib, AZD8931 and trametinib plus cetuximab as measured by inhibition of tumor growth (PMID: 31704811). In a Phase I trial, a patient with lung cancer harboring BRAF L597Q was treated with BVD-523 and demonstrated partial response (PMID: 29247021). In a Phase II trial, a patient with melanoma harboring BRAF L597Q was treated with trametinib and demonstrated partial response with a 38% tumor reduction and ongoing progression-free survival of 8.3 months (PMID: 33861486). In another Phase II trial, a patient with an advanced solid tumor harboring BRAF L597Q was treated with vemurafenib and did not clinically respond to treatment (PMID: 29320312).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:40:03',NULL,NULL,NULL,NULL,'1f81f2af-cf83-4764-9fe2-b5f4de735c42',673,NULL),(2535,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:35:42',NULL,NULL,NULL,NULL,'9cc4015a-8efe-4920-8520-02ef0003980b',673,NULL),(2536,NULL,'The class II BRAF K601N mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719, 26343582). This mutation has been identified in melanoma and multiple myeloma (PMID: 12068308, 15035987, 25178945, 23273605, 24270325). Biological characterization of BRAF K601N demonstrated that it activates the downstream MAPK pathway independently of RAS and renders BRAF active as a dimer with CRAF and itself (PMID: 26343582, 20179705). Preclinical studies with NIH3T3 and SK-MEL-239 cells expressing BRAF K601N demonstrated sensitivity to treatment with LY3009120 and PLX8394 as measured by reduced proliferation (PMID: 26732095, 30559419). In a Phase Ib trial, a patient with non-small cell lung cancer harboring BRAF K601N was treated with combination therapy of LTT462 and LXH 254 and demonstrated an unconfirmed partial response (Abstract: Wolf et al. Abstract# 1387p, Annals of Oncology 2020. https://www.annalsofoncology.org/article/S0923-7534%2820%2941697-7/fulltext). In a Phase II trial, two patients with non-small cell lung cancer harboring BRAF K601N were treated vemurafenib and did not clinically respond to treatment (PMID: 31959346).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:41:35',NULL,NULL,NULL,NULL,'99bbe560-98df-4ece-97ad-2ccf9fa0b7d8',673,NULL),(2537,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:46:14',NULL,NULL,NULL,NULL,'f856d787-ee6c-4ece-b0b5-4bf4c20d1dbb',673,NULL),(2538,NULL,'The class III BRAF D594A mutation is located in the kinase domain in exon 15 of the protein (PMID: 20141835, 28783719). This mutation has been recurrently found in melanoma, lung and colorectal cancers (PMID: 20141835, 28481359). Biological characterization of the BRAF D594A mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 20141835, 28783719). Thus, when BRAF D594A is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719, 20141835). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:49:17',NULL,NULL,NULL,NULL,'d8ab8fe1-4049-418a-8beb-1bcf5c7c0adb',673,NULL),(2539,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'01ba7555-1f48-4cef-99a7-5bb8f0439f47',673,NULL),(2540,NULL,'The class III BRAF D594E mutation is located in the kinase domain in exon 15 of the protein. This mutation has been found in non-small cell lung cancer and diffuse large B-cell lymphoma (PMID: 28947956, 32877514). In vivo studies with conditional knock-in mice expressing BRAF D594 demonstrate that the mutation activates the MAPK pathway through CRAF (PMID: 20978199). Expression of BRAF D594E in HEK293 cells demonstrated that it is activating as measured by increased downstream ERK signaling compared to wildtype BRAF (PMID: 28947956). In vitro studies have suggested that this mutation is resistant to dabrafenib, as measured by increased MAPK pathway activation upon drug treatment, but sensitive to trametinib, as measured by decreased MAPK pathway activation upon drug treatment (PMID: 28947956). As BRAF D594E is a kinase-dead mutation, it is suggested to be insensitive to treatment with vemurafenib (PMID: 28026870).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:49:30',NULL,NULL,NULL,NULL,'eac9874d-79d9-454a-81b6-87c68e93307f',673,NULL),(2541,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:45:27',NULL,NULL,NULL,NULL,'5ea90d79-44f3-49e0-bcab-decd35590115',673,NULL),(2542,NULL,'The class III BRAF D594G mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719). This mutation has been found in colorectal cancer and melanomas, among others (PMID: 26498038, 28481359). Despite having hypoactive kinase activity, in vitro studies have demonstrated that this mutation is likely activating as measured by increased CRAF dimerization and downstream signaling compared to wildtype (PMID: 20141835, 26498038, 28783719). Preclinical studies with WM3670 cells expressing BRAF D594G demonstrated resistance to treatment with vemurafenib and PLX8394, as measured by persistent MEK, ERK and AKT activity, and sensitivity to treatment with SIJ1777, as measured by AKT and MAPK inhibition (PMID: 33917428). In a case report, a patient with low-grade serous ovarian cancer harboring BRAF D594G was treated with trametinib and demonstrated minimal clinical response (PMID: 35198414).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:50:46',NULL,NULL,NULL,NULL,'2f4e594e-c475-49b7-a06a-dd2b1c5d4189',673,NULL),(2543,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:44:26',NULL,NULL,NULL,NULL,'e8e0818f-cedb-4eae-a9c1-e61cab753b74',673,NULL),(2544,NULL,'The class III BRAF D594N mutation occurs in the kinase domain in exon 15 of the protein (PMID: 28783719). This mutation has been found in non-small cell lung cancer and melanoma (PMID: 22798288, 23833300, 26498038, 28783719, 28947956). Biological characterization of the BRAF D594N mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 28783719, 26498038). Thus, when BRAF D594N is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719). In a case report, a patient with EGFR-positive, MET-amplified non-small cell lung cancer acquired BRAF D594N as a secondary resistance mutation in response to combination treatment with osimertinib plus crizotinib (PMID: 36136993). The patient demonstrated clinical response to combination treatment with osimertinib plus trametinib (PMID: 36136993).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 19:33:13',NULL,NULL,NULL,NULL,'b0ec34c2-01be-4bd8-96db-5aea6eb0706c',673,NULL),(2545,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'d98714d2-a2af-4af6-9722-63683c1ef6b4',673,NULL),(2546,NULL,'The class III BRAF D594V mutation is located in the kinase domain in exon 15 of the protein (PMID: 33019809). This mutation has been found in colorectal cancer and cardio-facio-cutaneous (CFC) syndrome (PMID: 16439621, 14678966). Expression of this mutation in D04 or COS7 cells demonstrated that, despite being kinase-dead, the mutation activates the MAPK pathway through dimerization with CRAF as measured by downstream MEK and ERK phosphorylation (PMID: 15035987, 14678966). When expressed in zebrafish embryos, the BRAF D594V mutation causes the same aberrant elongation phenotype as BRAF V600E, suggesting that it is likely activating in vivo (PMID: 19376813).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 19:34:23',NULL,NULL,NULL,NULL,'50b2db09-21b0-4bd8-859e-e6432d3a7483',673,NULL),(2547,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'e9ee68af-6814-4e3d-afcc-699156f2d920',673,NULL),(2548,NULL,'The class III BRAF D594Y mutation is located in the kinase domain in exon 15 of the protein (PMID: 30224342). This mutation has been identified in lung adenocarcinoma and ovarian cancer (PMID: 28481359, 31477716). Analysis of patients with BRAF-altered lung cancers demonstrated that BRAF D594Y has impaired kinase function (PMID: 32913992). As a BRAF class III mutation, it is suggested that BRAF D594Y is sensitive to MEK inhibition by targeted inhibitors such as trametinib and insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-06 19:35:00',NULL,NULL,NULL,NULL,'4f361386-9f20-415b-ad55-2b6e73a3bc5a',673,NULL),(2549,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:39:07',NULL,NULL,NULL,NULL,'c8e8b1f4-06ab-4749-b420-7a5600a95e1e',673,NULL),(2550,NULL,'The class II BRAF G464E mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719, 26343582). This mutation has been reported in colorectal cancer (PMID: 15150094). Biological characterization of the BRAF G464E mutation demonstrated it activates the downstream MAPK pathway independent of RAS and renders BRAF active as a dimer with CRAF (PMID: 15035987, 23680146, 15150094, 26343582, 20179705, 16364920). Preclinical studies with cancer cell lines expressing BRAF G464E demonstrate resistance to vemurafenib as measured by prolonged ERK signaling following treatment (PMID: 26343582, 28783719, 26343582). A patient with spindle cell carcinoma harboring BRAF G464E treated with trametinib resulted in stable disease with a progression-free survival of 6.9 months (PMID: 31924734).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:56:56',NULL,NULL,NULL,NULL,'40f0164f-978b-4b99-87bc-392ce3f14d64',673,NULL),(2551,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'fdb9c569-972c-4267-b851-c566fe8e0d01',673,NULL),(2552,NULL,'The class II BRAF G464R mutation is located in the kinase domain in exon 11 of the protein (PMID: 33019809). This mutation has been found in melanoma (PMID: 15046639). Expression of this mutation in NIH-3T3 cells demonstrated that this mutation is activating as measured by increased downstream pathway activation compared to wildtype BRAF, though to a lesser extent than BRAF V600E (PMID: 15046639). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF G464R demonstrate that the mutation is activating as measured by increased cell viability and proliferation compared to wildtype (PMID: 29533785). In a case report for a patient with metastatic colorectal cancer harboring BRAF G464R and KRAS G12S, the patient was treated with bevacizumab plus FOLFIRI and then later treated with bevacizumab plus FOLFOXIRI and demonstrated recurrent disease in the sigmoid colon following both treatment regimens (PMID: 31183211).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:57:26',NULL,NULL,NULL,NULL,'6ac26115-a639-4eee-aaac-dd0eba053dd6',673,NULL),(2553,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'5bab3643-9d50-43bf-be7b-307a2e48c9c4',673,NULL),(2554,NULL,'The class II BRAF G464V mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719, 26343582). This mutation has been reported in melanoma, colorectal cancer and breast cancer (PMID: 15035987, 15150094, 17314276). Biological characterization of the BRAF G464V mutation demonstrated it activates the downstream MAPK pathway independent of RAS and renders BRAF active as a dimer with CRAF (PMID:  15035987, 12068308, 15150094, 26343582, 20179705). Preclinical studies with KRAS-mutant HEK293 cells expressing BRAF G464V demonstrated sensitivity to treatment with RAF inhibitor Cmpd A as measured by reduced phosphorylated ERK levels (PMID: 23680146). A patient with lung adenocarcinoma harboring BRAF G464V treated with trametinib did not respond to treatment and demonstrated progressive disease (PMID: 31924734). Two patients with advanced solid tumors harboring BRAF G464V were treated with vemurafenib and did not have an objective response to treatment (PMID: 29320312).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 19:40:08',NULL,NULL,NULL,NULL,'532ebf53-588d-495d-92a3-fd13cb96b34b',673,NULL),(2555,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:43:08',NULL,NULL,NULL,NULL,'6ebe87c5-27e7-40a0-baa2-6ed29b00648c',673,NULL),(2556,NULL,'The class III BRAF G466A mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719, 15035987, 28783719). This mutation has been identified in melanoma and lung adenocarcinoma (PMID: 12068308, 28481359). Biological characterization of the BRAF G466A mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 15035987, 28783719). Thus, when BRAF G466A is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). A patient with non-small cell lung cancer harboring BRAF G466A treated with vemurafenib did not clinically respond to treatment and demonstrated progressive disease (PMID: 31959346). Patients with advanced solid tumors harboring BRAF G466A were treated with trametinib and demonstrated either progressive or stable disease (PMID: 31924734). In a case report, a patient with metastatic Bellini duct carcinoma harboring BRAF G466A treated with trametinib plus dabrafenib demonstrated 11 months of stable disease following treatment (PMID: 33669326). In a Phase Ib trial, a patient with metastatic non-small cell lung cancer harboring BRAF G466A was treated with combination therapy of the ERK1/2 inhibitor LTT462 and the BRAF/CRAF inhibitor LXH 254 and demonstrated stable disease with a 25% tumor reduction following treatment (Abstract: Wolf et al. Abstract# 1387p, Annals of Oncology 2020. https://www.annalsofoncology.org/article/S0923-7534%2820%2941697-7/fulltext).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 19:42:36',NULL,NULL,NULL,NULL,'855e100e-1702-43f9-ba27-bbb53498df41',673,NULL),(2557,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:42:51',NULL,NULL,NULL,NULL,'3aaa447d-1aaf-41b9-b0ee-d17185ce9016',673,NULL),(2558,NULL,'The class III BRAF G466E mutation occurs in the kinase domain in exon 11 of the protein (PMID: 15035987, 28783719). This mutation has been reported in melanoma, lung cancer, colorectal cancer and chronic myelomonocytic leukemia (PMID: 28481359, 15046639, 27790118, 24446311). Biological characterization of the BRAF G466E mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 15035987, 15046639, 28783719). Thus, when BRAF G466E is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). Preclinical studies with melanoma cell lines expressing BRAF G466E demonstrated sensitivity to treatment with AZ628 and TAK-632 as measured by reduced cell proliferation, and insensitivity to treatment with PLX7904, vemurafenib and dabrafenib, as measured by prolonged cell viability (PMID: 27523909). In a clinical study, two patients with metastatic colorectal cancer harboring BRAF G466E were treated with combination therapy of panitumumab and irinotecan and demonstrated partial response with a progression-free survival of 8.3 months (PMID: 31515458). In a case report, a patient with melanoma harboring BRAF G466E was treated with ipilimumab and demonstrated progressive disease (PMID: 27790118).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 19:43:32',NULL,NULL,NULL,NULL,'09553532-91a2-47fb-9468-ff505476c94e',673,NULL),(2559,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'903e2063-c368-4075-aab6-7d3f314f610a',673,NULL),(2560,NULL,'The class III BRAF G466R mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719). This mutation has been identified in lung adenocarcinoma and melanoma (PMID: 24297085, 21788131). Expression of this mutation in NIH3T3 cells demonstrated that this mutation is likely activating as measured by increased downstream ERK activation (PMID: 15046639). While this mutation has kinase activity in vitro less than wildtype BRAF, it likely activates downstream ERK through dimerization with CRAF (PMID: 15035987, 16364920). A patient with stage IV lung adenocarcinoma harboring BRAF G466R was treated with selumetinib and demonstrated an overall survival of 9.1 months (PMID: 25273224). In a case report, a patient with non-small cell lung cancer harboring BRAF G466R was treated with a combination therapy of dabrafenib plus trametinib and did not clinically respond to treatment (Abstract: Citarella et al. Precis Cancer Med 2022. https://pcm.amegroups.com/article/view/6792/html#B19). Another patient with non-small cell lung cancer harboring BRAF G466R was treated with a combination therapy of nivolumab plus anlotinib and demonstrated progressive disease with a progression-free survival of two months (PMID: 32411601). In a Phase II trial, a patient with melanoma harboring BRAF G466R and NRAS Q61H was treated with a combination therapy of docetaxel plus selumetinib and demonstrated stable disease with a progression-free survival of seven months (PMID: 31839677).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-06 19:44:53',NULL,NULL,NULL,NULL,'5414eff5-1ecb-4030-adf2-da44ef4b3778',673,NULL),(2561,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:39:53',NULL,NULL,NULL,NULL,'b6105994-7ba4-42af-b463-1b0eb367d81a',673,NULL),(2562,NULL,'The class III BRAF G466V missense mutation is located in the kinase domain in exon 11 of the protein (PMID: 15035987, 28783719). This mutation has been reported in melanoma, lung cancer, colorectal cancer and cardio-facio-cutaneous syndrome (PMID: 28481359, 15046639, 26498038, 16439621, 22649091). Biological characterization of the BRAF G466V mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 15035987, 26498038, 22649091, 16364920, 24202393, 28783719). Thus, when BRAF G466V is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). Preclinical studies with lung cancer cell lines expressing BRAF G466V demonstrated sensitivity to treatment with TAK-632, TAE226 and RMC-4550 as measured by inhibition of cell proliferation and reduced ERK phosphorylation following treatment (PMID: 26090892, 27523909, 30104724). Other preclinical studies with patient-derived xenograft colorectal cancer models expressing BRAF G466V demonstrated resistance to treatment with PLX8394 as measured by continued ERK phosphorylation and inability to suppress tumor growth, and sensitivity to treatment with cetuximab as measured by reduced tumor growth and inhibition of ERK phosphorylation (PMID: 30559419). In a Phase II trial, three patients with non-small cell lung cancer harboring BRAF G466V were treated with vemurafenib and did not clinically respond to treatment (PMID: 31959346). In another Phase II trial, a patient with lung adenocarcinoma harboring BRAF G466V was treated with trametinib and demonstrated progressive disease (PMID: 31924734). In a case report, a patient with metastatic colorectal cancer harboring BRAF G466V was treated with a combination therapy of irinotecan plus panitumumab and demonstrated a partial response with a progression-free survival of 6.1 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-07 17:57:38',NULL,NULL,NULL,NULL,'4bd4a259-ff20-48b2-aa59-8d1202991f65',673,NULL),(2563,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'acc42ebe-bf76-4b5c-9c2c-59865e0f39f6',673,NULL),(2564,NULL,'The class II BRAF G469A mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719, 26343582). This mutation has been reported in melanoma, non-small cell lung cancer and cardio-facio-cutaneous syndrome (PMID: 12068308, 23833300, 16439621). Biological characterization of the BRAF G469A mutation demonstrated that it activates the downstream MAPK pathway independently of RAS and renders BRAF active as a dimer with CRAF and itself (PMID: 12068308, 23833300, 26343582, 20179705, 22510884). Ectopic expression of BRAF G469A in drug-sensitive EGFR-mutant cells conferred resistance to EGFR TKIs that was overcome by addition of a MEK inhibitor (PMID: 22773810). Preclinical studies with patient-derived lung adenocarcinoma cells expressing BRAF G469A demonstrated sensitivity to treatment with LXH 254, lifirafenib and dabrafenib plus trametinib, as measured by inhibition of cell growth, and resistance to treatment with LY3214996, MK-8353, ravoxertinib, ulixertinib, TAK-580, vemurafenib and dabrafenib, as measured by persistent cell growth (PMID: 32540409). In a clinical study, patients with metastatic colorectal cancer harboring BRAF G469A were treated with cetuximab plus FOLFIRI, cetuximab plus irinotecan or panitumumab plus irinotecan and each treatment group demonstrated stable disease with a progression-free survival of 4.4 months, 3.5 months and 4.0 months, respectively (PMID: 31515458). In a Phase I trial, a patient with head and neck cancer and a patient with small bowel cancer, both harboring BRAF G469A, were treated with ulixertinib and demonstrated partial response (PMID: 29247021). In a Phase II trial, a patient with lung adenocarcinoma harboring BRAF G469A was treated with trametinib and demonstrated stable disease with a progression-free survival of 20.4 months (PMID: 31924734). In another Phase II trial, two patients with advanced solid tumors harboring BRAF G469A were treated with vemurafenib and did not clinically respond to treatment (PMID: 29320312).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-07 17:58:34',NULL,NULL,NULL,NULL,'ab9c04c7-415e-41da-93cd-82cbb6ce102c',673,NULL),(2565,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:43:47',NULL,NULL,NULL,NULL,'1ec9dffe-c9c5-4aa0-8439-5aca0e35730f',673,NULL),(2566,NULL,'The class III BRAF G469E mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719). This mutation has been reported in melanoma, lung cancer, colorectal cancer and cardio-facio-cutaneous syndrome (PMID: 26498038, 18794803, 16439621). Biological characterization of the BRAF G469E mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 15035987, 28783719, 15150094, 22510884). Thus, in melanoma, the G469E variant, similar to other class III BRAF mutants, would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung or colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). Preclinical studies with melanoma cells expressing BRAF G469E demonstrated sensitivity to treatment with sorafenib, as measured by inhibition of growth and induced apoptosis, and resistance to treatment with U0126, as measured by persistent cell growth (PMID: 18794803). In a Phase II trial, a patient with breast ductal carcinoma harboring BRAF G469E was treated with trametinib and demonstrated partial response with a 50% tumor size reduction (PMID: 31924734).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:43:01',NULL,NULL,NULL,NULL,'2bb6b849-bcd7-4f63-b423-6d2f5260e49c',673,NULL),(2567,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'10eb50de-c687-463b-811a-6271aaef3431',673,NULL),(2568,NULL,'The BRAF G469L mutation is located in the protein\'s kinase domain in exon 11. While this mutation has not been functionally characterized, other substitutions at this position (e.g., G469R) have been shown to be activating and oncogenic (PMID: 24920063 ). This mutation was found in a patient with non-small cell lung cancer who failed to respond to treatment with vemurafenib (PMID: 24035431).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'29e7843e-2973-4cc5-9288-3a3f17757604',673,NULL),(2569,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:38:36',NULL,NULL,NULL,NULL,'f8af2c28-f6db-41e3-94d8-b1a27435331c',673,NULL),(2570,NULL,'The class II BRAF G469R mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719, 26343582). This mutation has been reported in non-small cell lung cancer, chronic lymphocytic leukemia, melanoma and a range of other cancer types (PMID: 26498038, 24920063, 16096377, 26150740, 26619011). Biological characterization of the BRAF G469R mutation demonstrated that it activates the downstream MAPK pathway independent of RAS renders BRAF active as a dimer with CRAF and itself (PMID: 26343582, 28783719, 20179705). While sensitive to MEK inhibition by targeted inhibitors such as trametinib, this mutation is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 26343582, 28783719, 18794803, 26343582, 28783719). In a Phase II trial, a patient with melanoma harboring BRAF G469R and KIT G12V was treated with trametinib and demonstrated stable disease with a progression-free survival of 22.3 months (PMID: 33861486).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-08 14:30:52',NULL,NULL,NULL,NULL,'75cc7602-ac02-434e-89b6-5551765ba3a5',673,NULL),(2571,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:38:08',NULL,NULL,NULL,NULL,'b8f3cd76-e1a5-4bd1-96c4-7a5241b5fe8b',673,NULL),(2572,NULL,'The class II BRAF G469V mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719, 26343582). This mutation has been reported in non-small cell lung cancer and bladder cancer (PMID: 25870796, 27388325, 23401075). Biological characterization of the BRAF G469V mutation demonstrated that it activates the downstream MAPK pathway independently of RAS and renders BRAF active as a dimer with CRAF and itself (PMID: 26343582, 28783719, 20179705). Preclinical studies with SKBR3 cells expressing BRAF G469V demonstrated resistance to treatment with vemurafenib as measured by persistent MEK and ERK activation following treatment (PMID: 26343582). In a case report, two patients with metastatic colorectal cancer harboring BRAF G469V were treated with either panitumumab plus FOLFOX or cetuximab plus irinotecan and both treatment options demonstrated partial response with a progression-free survival of 18.9 months and 10.9 months, respectively (PMID: 31515458). In another case report, a patient with lung adenocarcinoma harboring BRAF G469V was treated with combination therapy of trametinib and dabrafenib and progressed after nine weeks of treatment (PMID: 32540409). In another case report, a patient with hepatocellular carcinoma and non-small cell lung cancer harboring BRAF G469V was treated with sorafenib and demonstrated a complete response in the lung and stability of the hepatocellular carcinoma (PMID: 27388325).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:48:58',NULL,NULL,NULL,NULL,'94fe5461-1eb2-432a-9b78-a2e7d2e26cb7',673,NULL),(2573,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'31844270-7a3e-4505-88f9-d75959551599',673,NULL),(2574,NULL,'The BRAF G469del mutation is located in the kinase domain in exon 11 of the protein. This mutation has been found in non-small cell lung cancer (PMID: 23833300). Expression of this mutation in NIH-3T3 cells demonstrated that though in vitro kinase activity and in vivo downstream MEK and ERK activation were decreased compared to wildtype BRAF, colony formation was mildly increased compared to wildtype BRAF (PMID: 23833300).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'687e219a-c2f3-4559-b626-101529bf2bf0',673,NULL),(2575,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'6877f3c4-e48a-42d4-bed6-0d55a2a9c10d',673,NULL),(2576,NULL,'The BRAF S151A mutation is located proximal to the RAS-binding domain in exon 3 of the protein. Expression of this mutation in NHEM cells demonstrated that it is likely activating, as measured by increased binding of BRAF to NRAS (PMID: 21478863). As the S151 residue of BRAF is a suspected site of negative-regulatory phosphorylation of BRAF by ERK, mutation of this serine to alanine would be predicted to result in increased BRAF activity (PMID: 21478863).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'b67cc890-136d-4b8f-978d-643d6acd35f4',673,NULL),(2577,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'3bb6e5b5-4599-40e7-930e-de0be6fa52f4',673,NULL),(2578,NULL,'The BRAF Q201H mutation is located in the RAS-binding domain in exon 4 of the protein. This mutation has been found in an ovarian cancer cell line (PMID: 18060073). While this mutation has not been functionally validated, SIFT prediction software predicted that this mutation would likely be neutral (PMID: 18060073).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'9f372272-5687-477b-9727-e2ea8ae52574',673,NULL),(2579,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'d86aa3d7-afc8-4512-b4ad-03e1fcbec9fe',673,NULL),(2580,NULL,'The BRAF T241P mutation is located in the cysteine-rich domain in exon 6 of the protein. This mutation has been found in cardio-facio-cutaneous syndrome (CFC) (PMID: 19206169). Expression of this mutation in NIH-3T3 cells demonstrated that it is likely activating as measured by moderately increased activation of MEK and ERK and slightly increased colony formation compared to wildtype BRAF (PMID: 19206169).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2019-07-12 18:32:31',NULL,NULL,NULL,NULL,'274d4c28-1e34-4982-872d-414ea59a5b73',673,NULL),(2581,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'de63a301-2e58-4312-abe2-cae003e0d59c',673,NULL),(2582,NULL,'The BRAF A246P mutation is located in the cysteine-rich domain in exon 6 of the protein. This mutation is found in cardio-facio-cutaneous (CFC) syndrome (PMID: 16474404). Expression of BRAF A246P in NIH-3T3 cells engineered to concurrently express an ERK-dependent luciferase construct demonstrated that this mutation is activating, as measured by increased luciferase activity compared to wildtype BRAF (PMID: 16474404). Single-molecule binding assays with BRAF A246P also demonstrate that the mutation is activating as measured by increased RAS binding affinity compared to wildtype (PMID: 24409384). Preclinical studies with zebrafish embryos expressing BRAF A246P demonstrated sensitivity to treatment with MEK inhibitors CI-1040 and PD0325901 as measured by the prevention of CFC development (PMID: 19376813).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:43:09',NULL,NULL,NULL,NULL,'f0a7fcca-369a-4d7d-b125-ac14c41ddc57',673,NULL),(2583,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'d53d983d-004b-4658-896b-f9d5fd19e085',673,NULL),(2584,NULL,'The class II BRAF Q257R mutation is located in the cysteine-rich domain in exon 6 of the protein (PMID: 33019809). This mutation has been found in cardio-facio-cutaneous syndrome (PMID: 16474404, 16439621). Expression of this mutation in NIH-3T3 cells demonstrated that this mutation is activating as measured by increased transcription of downstream MAPK pathway transcriptional targets (e.g., ELK) compared to wildtype BRAF (PMID: 16474404). When expressed in zebrafish embryos, the Q257R mutation causes the same aberrant elongation phenotype as BRAF V600E, suggesting that it is likely activating in vivo (PMID: 19376813).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-08 16:03:33',NULL,NULL,NULL,NULL,'eef8ecb6-23d2-442e-927d-d5350813178d',673,NULL),(2585,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'bcd54ffa-98df-4028-9e75-d5858ab5aed2',673,NULL),(2586,NULL,'The BRAF E275K mutation is located in the cysteine-rich domain in exon 6 of the protein. This mutation has been found in cardio-facio-cutaneous syndrome (PMID: 19206169). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by increased colony formation despite decreased MEK and ERK phosphorylation compared to wildtype BRAF (PMID: 19206169).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'0142dec9-1c86-41f8-a39e-a2c12c36d6dd',673,NULL),(2587,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:39:26',NULL,NULL,NULL,NULL,'33f20b93-a5aa-46ba-83b1-26871c2016bf',673,NULL),(2588,NULL,'The class III BRAF D287H mutation occurs in the cysteine-rich domain in exon 6 of the protein (PMID: 28783719). Biological characterization of the BRAF D287H mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 26343582, 28783719). In melanoma, the D287H variant, similar to other class III BRAF mutants, would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung or colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:46:22',NULL,NULL,NULL,NULL,'05840d43-e046-488f-bcd9-42517eee4a78',673,NULL),(2589,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'0cb22fc2-ff96-4abc-adb1-82fcee800dd6',673,NULL),(2590,NULL,'The BRAF I326V mutation is located in the flexible hinge domain of the BRAF protein. This mutation has been found as a germline mutation in Noonan Syndrome (PMID: 17603482). In vitro studies have demonstrated that this mutation is likely neutral as measured by downstream signaling compared to wildtype (PMID: 17603482). In a clinical study, a patient with colorectal cancer harboring BRAF I326V was treated with panitumumab and irinotecan plus S-1 and demonstrated progressive disease (PMID: 30463788).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-08 14:41:23',NULL,NULL,NULL,NULL,'c81b4261-e7a5-49db-80fc-e85e54ab4055',673,NULL),(2591,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:39:39',NULL,NULL,NULL,NULL,'7269fee4-78fd-4b9f-9589-35ff6398cb79',673,NULL),(2592,NULL,'The class III hypoactivating exon 6 BRAF V459L missense mutation occurs in the cysteine-rich domain of the protein (PMID: 28783719). This mutation has been reported in lung cancer and melanoma (PMID: 12460918). BRAF V459L signals as a dimer with CRAF, is RAS-dependent in its activation of the MAPK pathway, and, in tumors, would typically co-occur with another driver mechanism that amplifies RAS signaling (PMID: 28783719). Thus, like other class III BRAF mutants, in melanoma, the V459L variant would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung and colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib.','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-01-27 14:54:03',NULL,NULL,NULL,NULL,'74cb3e73-70a1-43c1-bb48-50a77184ce71',673,NULL),(2593,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'60619485-6bf6-456b-99bd-bf3ca8dd3286',673,NULL),(2594,NULL,'The BRAF R462E mutation is located in the kinase domain of the BRAF protein. In vitro studies have demonstrated that the BRAF L597S mutation is likely activating as measured by CRAF activity and downstream signaling compared to wildtype (PMID: 25155755).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2020-09-20 17:22:08',NULL,NULL,NULL,NULL,'612a0f55-afdf-4931-abbc-16550552dd4c',673,NULL),(2595,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'36c50b89-b6e6-4761-ade8-349acdcd73f1',673,NULL),(2596,NULL,'The class II BRAF R462I mutation is located in the kinase domain in exon 11 of the protein (PMID: 35050937). This mutation has been found in colorectal cancer (PMID: 15150094). Expression of this mutation in NIH-3T3 or IEC-6 cells demonstrated that this mutation is neutral as measured by kinase activity, foci-formation and cell proliferation comparable to wildtype BRAF (PMID: 15150094, 15035987). Additionally, expression of this mutation with HRAS G12V failed to activate the MAPK pathway either as a monomer or a dimer with CRAF, suggesting it is not activating (PMID: 15150094).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-08 16:03:51',NULL,NULL,NULL,NULL,'89566b58-387b-4ade-bec1-847857ef762a',673,NULL),(2597,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'26295e28-2d03-4e12-b4c1-b70e00c7166c',673,NULL),(2598,NULL,'The class II BRAF I463S mutation is located in the kinase domain in exon 11 of the protein (PMID: 33019809). This mutation has been found in colorectal cancer (PMID: 15150094). Expression of this mutation in NIH-3T3 or IEC-6 cells demonstrated that it is likely neutral, as measured by kinase activity, foci formation, and cell proliferation comparable to wildtype BRAF (PMID: 15150094, 15035987). Additionally, expression of BRAF I463S with HRAS G12V does not lead to increased MAPK pathway activation either alone as a monomer or through dimerization with CRAF (PMID: 15150094).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-08 14:42:05',NULL,NULL,NULL,NULL,'d5c4b6b2-1741-441c-ac40-b223d4060c9b',673,NULL),(2599,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'94d0f8da-898a-4f3a-9674-d2e8d7a4dec1',673,NULL),(2600,NULL,'The class III BRAF S467A mutation is located in the kinase domain in exon 11 of the protein (PMID: 35050937). This mutation has been found in cardio-facio-cutaneous syndrome (PMID: 16439621). Expression of this mutation in 293T cells demonstrated that this mutation is activating as measured by increased activation of downstream MAPK effectors compared to wildtype BRAF (PMID: 16439621). When expressed in zebrafish embryos, the BRAF S467A mutation causes the same aberrant elongation phenotype as BRAF V600E, suggesting that it is likely activating in vivo (PMID: 19376813).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 14:35:21',NULL,NULL,NULL,NULL,'b1f6b30a-c28b-4f4b-b0fd-04e76e8b5d9d',673,NULL),(2601,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'391497af-3c28-4451-8b55-b4af8b0278c5',673,NULL),(2602,NULL,'The BRAF F468C mutation is located in the kinase domain in exon 11 of the protein. This mutation has been found in colorectal cancer (PMID: 15150094). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by increased in vitro kinase activity, increased proliferation and increased foci formation compared to wildtype BRAF (PMID: 15150094).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'9665e855-859a-42c2-b65e-fceb34f4b3ab',673,NULL),(2603,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'9152f85c-59b4-4978-b42a-315ea2528a46',673,NULL),(2604,NULL,'The class II BRAF V471F mutation is located in the kinase domain in exon 11 of the protein (PMID: 33019809). Expression of this mutation in 293T cells demonstrated that this mutation is activating as measured by increased downstream pathway activation and moderate transforming activity compared to wildtype BRAF (PMID: 25348715). Though this mutation is kinase-dead, it activates the MAPK pathway by dimerizing with CRAF (PMID: 25348715).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:51:02',NULL,NULL,NULL,NULL,'a97a1859-463d-44b0-a455-7708ac87643d',673,NULL),(2605,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'f1f455b0-d3f4-4086-aee9-7af13bf507f0',673,NULL),(2606,NULL,'The BRAF Y472C mutation is located in the kinase domain in exon 11 of the protein. This mutation is found in non-small cell lung cancer (PMID: 22649091). Expression of this mutation in COS7, H661, or H226 cells demonstrated that it is activating as measured by increased downstream MEK and ERK activation compared to wildtype BRAF, despite having impaired kinase activity (PMID: 22649091). Similar to other kinase-inactivating BRAF mutations, the Y472C mutant promotes downstream oncogenic signaling (via MEK/ERK pathway) by transactivating CRAF, and therefore is considered to be likely oncogenic in the clinical setting (PMID: 22649091). In a Phase II trial, a patient with non-small cell lung cancer harboring BRAF Y472C was treated with dasatinib and demonstrated a complete response (PMID: 20855820).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:51:36',NULL,NULL,NULL,NULL,'38abf378-98d7-47ca-b0b7-d277dfcc627f',673,NULL),(2607,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'290eff4d-b13d-419b-8e92-693f1f479f96',673,NULL),(2608,NULL,'The BRAF G478C mutation is located in the kinase domain in exon 12 of the protein. This mutation was engineered to mimic the ARAF G331C mutation (PMID: 22926515). Expression of this mutation in COS-7 cells demonstrated that this mutation is inactivating, as shown by abolished kinase activity compared to wildtype BRAF (PMID: 22926515).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'52557f96-60a6-4b41-9fba-74de217198da',673,NULL),(2609,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'7dfeeb0c-1c68-417c-9267-78d3ccd709eb',673,NULL),(2610,NULL,'The class III BRAF K483E mutation occurs in the tyrosine kinase domain of the BRAF protein (PMID: 33019809). This mutation has been identified in a patient diagnosed with pediatric poorly differentiated carcinoma (PMID: 27799065). Overexpression of this variant in 293T cells resulted in activation of MEK1/2, however, the activation was comparable to overexpression of wildtype BRAF (PMID: 27799065). Activation likely occurs via allosteric activation of BRAF or CRAF because of the lack of BRAF K483E kinase activity (PMID: 27799065). Preclinical studies with patient-derived colorectal cancer organoids expressing BRAF K483E demonstrated sensitivity to treatment with binimetinib as measured by decreased cell viability, and resistance to treatment with cetuximab as measured by no change in cell viability (PMID: 33060766).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 18:52:08',NULL,NULL,NULL,NULL,'0be98f82-fc8b-47a0-9ccc-7413b607a840',673,NULL),(2611,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'e3a61279-8c09-41f0-83aa-2903fc667917',673,NULL),(2612,NULL,'The class III BRAF K483M mutation is located in the kinase domain of the BRAF protein (PMID: 15035987). In vitro studies have demonstrated that this mutation is likely activating as measured by CRAF activation and downstream signaling compared to wildtype (PMID: 15035987, 23533272). Preclinical studies with D04 cells expressing BRAF K483M demonstrated sensitivity to treatment with sorafenib as measured by inhibition of MEK activation (PMID: 20141835).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-08 14:42:57',NULL,NULL,NULL,NULL,'a979f929-319b-4dad-917f-ba9414e5d0f6',673,NULL),(2613,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'7d10414f-305b-4724-9b43-f6544d241b44',673,NULL),(2614,NULL,'The class II BRAF L485F mutation is located in the kinase domain in exon 11 of the protein (PMID: 31515458). This mutation has been found in cardio-facio-cutaneous syndrome (PMID: 16474404, 16439621). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by significantly increased transcription of downstream MAPK pathway transcriptional targets (e.g., ELK) compared to wildtype BRAF (PMID: 16474404). Preclinical studies with 293T cells expressing BRAF L485F demonstrated sensitivity to treatment with LY3009120, as measured by inhibition of ERK signaling, and resistance to treatment with dabrafenib, PLX8394 and vemurafenib, as measured by persistent ERK signaling (PMID: 34108213). In a clinical study, a patient with metastatic colorectal cancer harboring BRAF L485F was treated with a combination therapy of cetuximab plus irinotecan and demonstrated stable disease with a progression-free survival of 6.3 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-08 15:23:51',NULL,NULL,NULL,NULL,'839326e3-e318-490f-aefb-5d1cc6446c31',673,NULL),(2615,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'cb00df9d-198b-4dad-9bf7-65978cb68d7f',673,NULL),(2616,NULL,'The class II BRAF K499E mutation is located in the kinase domain in exon 12 of the protein (PMID: 33019809). This mutation has been found in cardio-facio-cutaneous syndrome (PMID: 16474404, 16439621). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by significantly increased transcription of downstream MAPK pathway transcriptional targets (e.g., ELK) compared to wildtype BRAF (PMID: 16474404).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-08 14:43:44',NULL,NULL,NULL,NULL,'d46a95e1-2983-436e-9368-5485cf138059',673,NULL),(2617,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'ef03b869-b19a-406d-805f-6dc48d4c0c48',673,NULL),(2618,NULL,'The BRAF E501G mutation is located in the kinase domain in exon 12 of the protein. This mutation has been found in cardio-facio-cutaneous (CFC) syndrome (PMID: 16439621). Expression of BRAF E501G in NIH-3T3 cells demonstrated that this mutation is slightly loss of function and therefore likely not oncogenic, as measured by MAPK pathway activation in the form of downstream ELK transcription comparable to or less than wildtype BRAF (PMID: 26065894, 16474404, 16439621).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2019-02-03 19:52:37',NULL,NULL,NULL,NULL,'2662435c-a398-48e3-ad80-840da9137e2b',673,NULL),(2619,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'139692b6-39a3-4cc0-a9f7-1879cd870b0f',673,NULL),(2620,NULL,'The BRAF E501K mutation is located in the kinase domain of the BRAF protein. This mutation has been found as a germline mutation in Noonan syndrome and cardio-facio-cutaneous syndrome (PMID: 17603482, 16474404). In vitro studies have demonstrated that this mutation might be inactivating as measured by decreased BRAF kinase activity in a cell line with a second BRAF mutation compared to controls (PMID: 17603482). However, another in vitro study found increased downstream pathway output compared to wildtype (PMID: 16474404). BRAF E501K is suggested to be RAS-dependent due to enrichment of RAS regulatory genes in tumors harboring the mutation in a pan-cancer genomic data analysis (PMID: 33507258).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-07 18:05:18',NULL,NULL,NULL,NULL,'ec6aaa98-f05b-4a8d-bb3c-5e1b59380208',673,NULL),(2621,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'dfc4709e-b7be-4ea2-9784-a5cfd49b7b6d',673,NULL),(2622,NULL,'The class II BRAF L505H mutation is located in the kinase domain in exon 12 of the protein (PMID: 33019809). This mutation has been found in prostate cancer (PMID: 24283590). Expression of this mutation in 293T cells or RWPE-1 prostate cells demonstrated that it is activating as measured by increased downstream activation of MEK and ERK as well as increased colony formation in soft agar compared to wildtype BRAF (PMID: 24283590). In vitro and in vivo studies of BRAF L505H suggest that this mutation is resistant to vemurafenib as measured by the occurrence of the L505H mutation in a vemurafenib resistance screen, sustained MAPK pathway signaling upon treatment with vemurafenib in cells expressing the mutation and sustained tumor growth in a xenograft model of melanoma cells harboring this mutation (PMID: 24283590, 24112705). In one study, one patient with metastatic melanoma harboring BRAF L505H had progressive disease in response to vemurafenib treatment (PMID: 25515853). In vitro studies, however, suggest that BRAF L505H is sensitive to selumetinib as measured by decreased MAPK pathway signaling upon drug treatment (PMID: 24283590).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 18:55:32',NULL,NULL,NULL,NULL,'07f22e94-4359-4e3b-a747-1a846bbf7294',673,NULL),(2623,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'b1152f1c-13e3-4fdb-b675-1c0f378d1014',673,NULL),(2624,NULL,'The BRAF R509H mutation is located in the kinase domain in exon 13 of the protein. This mutation lies along the dimer interface of the protein. Expression of this mutation in 293H cells demonstrated that it is inactivating, as shown by decreased protein dimerization and decreased downstream cell signaling (PMID: 20179705).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-12-13 12:00:41',NULL,NULL,NULL,NULL,'5ee9e9a1-ed4b-4e83-ab9d-9c2e3969ceea',673,NULL),(2625,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'4fcd2047-f412-4b97-880d-cf0da88a2e43',673,NULL),(2626,NULL,'The BRAF T529I mutation is located in the kinase domain in exon 13 of the protein. This alteration is considered a gatekeeper mutation as mutations at the analogous residue in other kinases are known to cause resistance to drugs that bind in the ATP-binding pocket (PMID: 15573099). Expression of this mutation in COS7 cells demonstrated it is likely neutral as measured by kinase activity comparable to wildtype BRAF (PMID: 20538618). However, expression with BRAF V600E in COS7 cells demonstrated that this mutation confers resistance to RAF inhibitors such as vemurafenib, but maintains sensitivity to MEK inhibitors (PMID: 20538618).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'cd172ba8-907b-446b-8fa5-999d9842636d',673,NULL),(2627,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'7f29f128-5d8b-4338-bc2c-de5e0d021224',673,NULL),(2628,NULL,'The BRAF T529M mutation is located in the kinase domain in exon 13 of the protein. This alteration is considered a gatekeeper mutation as mutations at the analogous residue in other kinases are known to cause resistance to drugs that bind in the ATP-binding pocket (PMID: 15573099). Expression of this mutation in COS7 cells demonstrated it is likely neutral as measured by kinase activity comparable to wildtype BRAF (PMID: 20538618). However, expression with BRAF V600E in COS7 cells demonstrated that this mutation confers resistance to RAF inhibitors such as vemurafenib, but maintains sensitivity to MEK inhibitors (PMID: 20538618).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'ba37133a-b86e-4e2f-9c4a-3a20bba75a12',673,NULL),(2629,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'78859ec7-08e2-4cde-be35-dc1a8251ebe3',673,NULL),(2630,NULL,'The BRAF T529N mutation is located in the kinase domain in exon 13 of the protein. This alteration is considered a gatekeeper mutation as mutations at the analogous residue in other kinases are known to cause resistance to drugs that bind in the ATP-binding pocket (PMID: 15573099). Expression of this mutation in COS7 cells demonstrated it is likely neutral as measured by kinase activity comparable to wildtype BRAF (PMID: 20538618). However, expression with BRAF V600E in COS7 cells demonstrated that this mutation confers resistance to RAF inhibitors such as vemurafenib, but maintains sensitivity to MEK inhibitors (PMID: 20538618).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'700df84a-fd4b-4548-bc55-645b5151c41c',673,NULL),(2631,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-09 14:31:47',NULL,NULL,NULL,NULL,'26a09bcd-4109-4a76-b3ee-47b09f771091',673,NULL),(2632,NULL,'The BRAF W531C mutation is located in the cysteine-rich domain in exon 6 of the protein. This mutation has been found in patients with cardio-facio-cutaneous syndrome (PMID: 19206169). There is conflicting evidence on the biological and oncogenic effect of this mutation. Expression of this mutation in NIH-3T3 cells demonstrated that this mutation is likely neutral, as measured by downstream pathway activation comparable to wildtype BRAF (PMID: 19206169). However, in vitro studies with Ba/F3 and MCF10A cells and in vivo studies with drosophila expressing BRAF W531C demonstrate that the mutation is activating as measured by increased MEK and ERK signaling and cellular proliferation (PMID: 29533785, 33855281).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-09 18:56:08',NULL,NULL,NULL,NULL,'5ba45d26-0662-48e4-aca8-af8d9790b5fb',673,NULL),(2633,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-14 16:45:52',NULL,NULL,NULL,NULL,'90e25372-22a3-4004-a2f9-84f2187fbcfa',673,NULL),(2634,NULL,'The class III BRAF N581D mutation is located in the kinase domain in exon 15 of the protein and is a statistically significant hotspot (PMID: 3010472). This mutation is found in cardio-facio-cutaneous syndrome (PMID: 16474404). Expression of this mutation in NIH-3T3 cells demonstrated that it is likely neutral as measured by transcription of downstream MAPK pathway target genes (e.g., ELK) comparable to wildtype BRAF (PMID: 16474404). Preclinical studies with patient-derived xenograft models of non-small cell lung cancer expressing BRAF N581D demonstrated sensitivity to treatment with RMC-4550 as measured by tumor growth inhibition (PMID: 30104724).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-09 18:58:53',NULL,NULL,NULL,NULL,'102f2a9d-2991-464a-984f-13df8d99e510',673,NULL),(2635,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:43:59',NULL,NULL,NULL,NULL,'d51426b1-b27b-42be-b84e-20cd1d12953d',673,NULL),(2636,NULL,'The class III BRAF N581S mutation is located in the kinase domain in exon 15 of the protein (PMID:28783719). This mutation has been reported in lung cancer, colorectal cancer and melanoma (PMID: 26498038). Biological characterization of the BRAF N581S mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 15035987, 28783719). Thus, in melanoma, the N581S variant, similar to other class III BRAF mutants, would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung or colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719).  Preclinical studies with patient-derived lung adenocarcinoma cells expressing BRAF N581S demonstrated sensitivity to treatment with a combination therapy of PF-477736 plus PF3644022 as measured by increased apoptosis (PMID: 26140595). In a Phase II trial, two patients with lung adenocarcinoma and gynecologic cancer harboring BRAF N581S were treated with trametinib and demonstrated progressive disease and stable disease, respectively (PMID: 31924734). In a case report, a patient with metastatic colorectal cancer harboring BRAF N581S was treated with a combination therapy of cetuximab plus FOLFIRI and demonstrated partial response with a progression-free survival of sixteen months (PMID: 31515458). In another Phase II trial, two patients with advanced solid tumors harboring the mutation were treated with vemurafenib and did not clinically respond to treatment (PMID: 29320312).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:00:45',NULL,NULL,NULL,NULL,'db1a6f73-d838-41b4-9004-b2b5a0684b32',673,NULL),(2637,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-08 16:00:51',NULL,NULL,NULL,NULL,'c4f7d30f-8184-4ffb-bf52-768f6c3cb365',673,NULL),(2638,NULL,'The class III BRAF N581Y mutation is located in the kinase domain of BRAF in exon 13 (PMID: 30224342). This noncanonical BRAF mutation has been identified in a screen of colorectal cell lines and in a patient with melanoma (PMID: 19474002, 26084293). In vitro studies with Ba/F3 cells expressing BRAF N581Y demonstrate that the mutation is activating as measured by increased cell viability compared to wildtype (PMID: 32540409). The HT-55 colon cancer cell line harboring this mutation was insensitive to treatment with the GDC-0879 RAF kinase inhibitor (PMID: 19276360).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-08 16:00:52',NULL,NULL,NULL,NULL,'96dc7292-f9e2-4308-ac25-83ebc2eae671',673,NULL),(2639,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2020-10-04 23:37:23',NULL,NULL,NULL,NULL,'1c855bc9-a6bf-48d2-b7f9-3f26397c63a3',673,NULL),(2640,NULL,'The BRAF L584F mutation is located in the kinase domain of the BRAF protein. This mutation has been found in melanoma, as well as other cancers (PMID: 23658559, 26386083). In vitro studies have demonstrated that this mutation may be activating as measured by sensitivity to RAF inhibition compared to wildtype (PMID: 23658559).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2020-10-04 23:36:43',NULL,NULL,NULL,NULL,'31484336-b5ef-4566-9139-0889007f5784',673,NULL),(2641,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'770191ac-6644-4f4e-92fa-52bed11b6a88',673,NULL),(2642,NULL,'The class II BRAF E586K mutation is located in the kinase domain in exon 15 of the protein (PMID: 33019809). This mutation has been identified in lung adenocarcinoma (PMID: 31534501). Expression of this mutation in COS cells demonstrated that it is activating as measured by increased in vitro kinase activity compared to wildtype BRAF (PMID: 15035987). Preclinical studies with HEK293 cells expressing BRAF E586K demonstrated sensitivity to treatment with pan-RAF inhibitor LY3009120 as measured by inhibited phospho-MEK and -ERK activities (PMID: 26343583).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-07 18:08:03',NULL,NULL,NULL,NULL,'2a7480c9-f5a2-4f5d-ba10-1b5d56dd2c20',673,NULL),(2643,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:46:57',NULL,NULL,NULL,NULL,'4f7f3cec-7f72-4795-8bd8-0ac1ad166b47',673,NULL),(2644,NULL,'The class III BRAF F595L mutation is located in the kinase domain in exon 11 of the protein (PMID: 28783719). This mutation has been reported in histiocytic sarcoma, colorectal cancer and cardio-facio-cutaneous syndrome (PMID: 26582644, 15150094, 26150740). Biological characterization of the BRAF F595L mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 28783719, 15035987, 15150094). Thus, when BRAF F595L is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719, 20141835). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719, 26582644). A patient with metastatic colorectal cancer harboring BRAF F595L was treated with a combination therapy of FOLFOX plus panitumumab and experienced partial response with a progression-free survival of 10.2 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-07 18:09:17',NULL,NULL,NULL,NULL,'5fb6a290-f77f-4f7e-aae1-ae8e7c6eb37a',673,NULL),(2645,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'2f902a6a-5e44-4837-8e09-cc39f20fce39',673,NULL),(2646,NULL,'The class III BRAF G596C mutation is located in the kinase domain of the BRAF protein (PMID: 28947956). This mutation has been found in non-small cell lung cancers, among others (PMID: 28947956). In vitro experiments have demonstrated that this mutation is activating as measured by increased downstream signaling compared to wildtype controls (PMID: 28947956). The BRAF G596C mutation confers resistance to the RAF inhibitor dabrafenib, but retains sensitivity to MEK inhibitor trametinib, as measured by MAPK pathway activation (PMID: 28947956).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-08 14:33:31',NULL,NULL,NULL,NULL,'7c6857fc-1772-420f-b26a-2e122ca12071',673,NULL),(2647,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:47:37',NULL,NULL,NULL,NULL,'0f1e6399-024b-4bd4-9d5d-175175fce18b',673,NULL),(2648,NULL,'The class III BRAF G596R mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719). This mutation has been reported in melanoma, lung cancer and colorectal cancer (PMID: 14678966, 26498038). Biological characterization of the BRAF G596R mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 15035987, 28783719, 16364920). Thus, in melanoma, the G596R variant, similar to other class III BRAF mutants, would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung or colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). Preclinical studies with colorectal cancer cells expressing BRAF G596R demonstrated sensitivity to treatment with PLX8394, RMC-4550 and trametinib plus dabrafenib as measured by inhibition of cellular proliferation, and resistance to treatment with vemurafenib as measured by persistent ERK and MEK activation (PMID: 30559419, 30104724,  28947956, 28783719). In a Phase II trial, a patient with non-small cell lung cancer was treated with vemurafenib and did not clinically respond to treatment (PMID: 31959346).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:02:14',NULL,NULL,NULL,NULL,'4c2cce08-23cc-49a9-bb8c-3244d90f7ff2',673,NULL),(2649,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'c84458f5-3f34-4136-8ab3-239bb9996d4e',673,NULL),(2650,NULL,'The class III BRAF G596V mutation is located in the kinase domain in exon 15 of the protein (PMID: 33198372). This mutation has been found in cardio-facio-cutaneous syndrome (PMID: 16439621). Expression of this mutant in HEK293T cells demonstrated it has impaired kinase activity and is not able to induce downstream MEK and ERK activation in this context (PMID: 16439621). However, when expressed in zebrafish embryos, the G596V mutation causes the same aberrant elongation phenotype as BRAF V600E, suggesting that it might function in a manner similar to BRAF activating mutants during development in vivo (PMID: 19376813). In a Phase II trial, a patient with non-small cell lung cancer harboring BRAF G596V was treated with vemurafenib and demonstrated a partial response (PMID: 26200454).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 19:04:17',NULL,NULL,NULL,NULL,'86f2dcf1-34f4-426b-83d7-6336964f61fa',673,NULL),(2651,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'4fc6ccc4-22d3-4bfb-a6cd-8eead0f36523',673,NULL),(2652,NULL,'The BRAF A598T mutation is located in the kinase domain of the BRAF protein. This mutation has been identified in glioma (PMID: 32996219). In vitro experiments have demonstrated that is mutation is inactivating as measured by reduced BRAF kinase activity compared to wildtype controls (PMID: 22926515, 32059434).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2023-03-06 15:44:49',NULL,NULL,NULL,NULL,'b0e7125c-946a-4890-b86a-9cb16ddc21bd',673,NULL),(2653,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'094c1215-bf19-4821-8044-9d3e991acd7c',673,NULL),(2654,NULL,'The class II BRAF A598V mutation is located in the kinase domain in exon 15 of the protein (PMID: 34794740). This mutation has been found in papillary thyroid carcinoma (PMID: 19200582). In vitro studies of BRAF A598V in patient samples have demonstrated that this mutation is activating as measured by increased phosphorylated MEK and ERK levels compared to wildtype BRAF (PMID: 19200582). Preclinical studies with patient-derived circulating tumor cells expressing BRAF A598V demonstrate resistance to combination treatment with BRAF/MEK inhibitors dabrafenib and trametinib as measured by sustained detection following treatment (PMID: 33731038).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:44:59',NULL,NULL,NULL,NULL,'4e3b29fc-7373-4ca1-a7d9-0ec1b1595842',673,NULL),(2655,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'700e3725-521c-4138-a19f-10d02ba99db4',673,NULL),(2656,NULL,'The class II BRAF T599I mutation is located in the kinase domain in exon 15 of the protein (Abstract: Severson et al. Abstract #40P, Annals of Oncology 2022. https://www.annalsofoncology.org/article/S0923-7534%2822%2900058-8/fulltext). Expression of this mutation in COS cells demonstrated that it is activating as measured by increased kinase activity and downstream ERK activation compared to wildtype BRAF (PMID: 15035987). In a case report, a patient with metastatic melanoma harboring BRAF T599I was treated with vemurafenib and demonstrated a clinical response to treatment (PMID: 29232305).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:04:49',NULL,NULL,NULL,NULL,'25bbc115-da6c-4110-c0c1-d51486913e97',673,NULL),(2657,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'1e49df5e-858a-4e65-a507-1d50e08e2682',673,NULL),(2658,NULL,'The class II BRAF T599R mutation is located in the kinase domain in exon 15 of the protein (PMID: 33019809). This mutation has been found in cardio-facio-cutaneous syndrome and thyroid carcinoma (PMID: 19206169, 25120313). Expression of BRAF T599R in NIH3T3 cells demonstrated that it is activating as measured by increased colony formation and downstream MAPK pathway activation compared to wildtype BRAF (PMID: 19206169).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 14:33:57',NULL,NULL,NULL,NULL,'762acae2-d84d-4f7e-9f1b-18c23afc75d4',673,NULL),(2659,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'03ccaae2-8f53-4fa9-94f0-e1c098c93431',673,NULL),(2660,NULL,'The BRAF R671Q mutation is located in the kinase domain of the BRAF protein. This mutation has been found in colorectal cancers, among others (PMID: 27048246). In vitro studies have demonstrated that this mutation is likely activating as measured by increased downstream signaling compared to wildtype (PMID: 28174173).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2020-10-05 00:06:38',NULL,NULL,NULL,NULL,'c66a2b0d-3f33-4f91-85f3-4c0cac591ad8',673,NULL),(2661,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'4aaf8250-0888-4fb7-87f8-fd8a3648d50c',673,NULL),(2662,NULL,'The BRAF A728V mutation is located in the C-terminal end of exon 18 of the protein. This mutation has been identified in papillary thyroid carcinoma (PMID: 19176457).  Expression of this mutation in COS cells demonstrated that it is activating as measured by increased phosphorylated ERK levels and increased kinase activity compared to wildtype BRAF (PMID: 15035987).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:45:16',NULL,NULL,NULL,NULL,'84adeb92-4025-4ce5-be2a-d0b900086110',673,NULL),(2663,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:13:39',NULL,NULL,NULL,NULL,'c02b208b-d750-4898-957c-a9b175327513',673,NULL),(2664,NULL,'The class II BRAF L485_P490del mutation is located in the kinase domain in exon 11 of the protein (PMID: 36686670). Expression of this mutation in HEK293 or NIH-3T3 cells demonstrated that it is activating as measured by kinase activity and colony formation in soft agar comparable to BRAF V600E (PMID: 26732095). However, a second study in human cell lines demonstrated that this mutation has kinase activity comparable to wildtype BRAF (PMID: 26996308). This mutation is dependent on BRAF homodimer formation, as BRAF L485_P490del fails to activate downstream MAPK pathway effectors as a monomer (PMID: 26732095). Preclinical studies with NIH3T3 cells expressing BRAF L485_P490del demonstrated sensitivity to treatment with LY3009120, as measured by inhibition of MEK and ERK phosphorylation, and resistance to treatment with vemurafenib, as measured by sustained MEK and ERK signaling and proliferation (PMID: 26732095). In a case report, a patient with metastatic melanoma harboring BRAF L485_P490del and amplified PD-L1 expression was treated with a combination therapy of trametinib and dabrafenib and demonstrated partial response with a progression-free survival of 18 months (PMID: 36686670).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 19:05:21',NULL,NULL,NULL,NULL,'fcc850b0-b6f4-4c5e-9f3d-b1776da2b284',673,NULL),(2665,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-07 16:49:59',NULL,NULL,NULL,NULL,'9a46e5bd-4749-4733-afde-592265f17157',673,NULL),(2666,NULL,'The BRAF L485_Q494del mutation is located in the kinase domain in exon 11 of the protein. While this variant has not been functionally validated, another deletion in this amino acid range (L485_P490del) has been shown to be gain-of-function and oncogenic (PMID: 26732095). Experiments in cell lines and mouse models demonstrate that BRAF alpha-C helix indel mutants, such as L485_Q494del, are insensitive to vemurafenib and maintain activity of the MAPK pathway upon drug treatment (PMID: 26732095)','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-07 16:50:01',NULL,NULL,NULL,NULL,'9b3aa240-60ed-4132-8cb3-b6122d96703a',673,NULL),(2667,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'c3a491ee-e144-4017-9f6f-d2e809d40ccc',673,NULL),(2668,NULL,'The BRAF L485_P490delinsF mutation is located in the kinase domain in exon 11 of the protein. While this variant has not been functionally validated, another deletion in this amino acid range (L485_P490del) has been shown to be gain-of-function and oncogenic (PMID: 26732095). Experiments in cell lines and mouse models demonstrate that BRAF alpha-C helix indel mutants, such as L485_P490delinsF, are insensitive to vemurafenib and maintain activity of the MAPK pathway upon drug treatment (PMID: 26732095)','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-12-18 15:35:54',NULL,NULL,NULL,NULL,'2c5fe2dc-3d3c-4dd0-9342-4487716516bb',673,NULL),(2669,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-07 16:51:39',NULL,NULL,NULL,NULL,'46020abb-e172-4a09-90ee-6cedc0a7e4e7',673,NULL),(2670,NULL,'The class II BRAF L485_P490delinsY mutation is located in the kinase domain in exon 11 of the protein (PMID: 33019809). This mutation has been identified in non-small lung cancer (PMID: 12068308). While this variant has not been functionally validated, preclinical studies with non-small lung cancer cell lines and mouse models expressing BRAF L485_P490delinsY demonstrated sensitivity to treatment with LY3009120 and trametinib as measured by inhibition of cellular proliferation, and resistance to vemurafenib as measured by sustained MEK and ERK pathway activation (PMID: 26732095).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-09 19:07:06',NULL,NULL,NULL,NULL,'43ac21b6-f89c-4d26-9582-a4afa9412a21',673,NULL),(2671,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 22:45:05',NULL,NULL,NULL,NULL,'93be4a13-1158-4135-8c3b-062eaea0d56c',673,NULL),(2672,NULL,'The class II BRAF N486_P490del mutation is located in the kinase domain in exon 11 of the protein (PMID: 33019809). This mutation has been found in an ovarian cancer cell line and in patients with pancreatic cancer (PMID: 18060073, 26996308, 31519698). Expression of this mutation in a human cell line demonstrated that it is activating via a CRAF- and dimer-independent mechanism as measured by increased kinase activity and downstream signaling compared to wildtype (PMID: 26996308). Preclinical studies with patient-derived xenograft models of pancreatic ductal adenocarcinoma expressing BRAF N486_P490del demonstrated sensitivity to treatment with trametinib and LY3009120 alone or in combination as measured by reduced tumor growth (PMID: 34011980). A patient with pancreatic cancer harboring this mutation had a partial response to treatment with the BRAF inhibitor dabrafenib (PMID: 31519698). In a case report, two patients with Langerhans cell histiocytosis harboring BRAF N486_P490del were treated with trametinib and one patient demonstrated a complete response with a progression-free survival of over 18 months while the second patient demonstrated stable disease with a progression-free survival of over twelve months (PMID: 32991018). In another case report, a patient with metastatic pancreatic ductal adenocarcinoma harboring BRAF N486_P490del was treated with a combination therapy of dabrafenib plus trametinib and demonstrated partial response (PMID: 36505826).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 19:09:02',NULL,NULL,NULL,NULL,'5211e083-6d72-4505-a5fe-29da60c439a0',673,NULL),(2673,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-09 14:32:56',NULL,NULL,NULL,NULL,'d4da6583-0755-4b4c-a0a9-88cdd924ae26',673,NULL),(2674,NULL,'The BRAF V487_P492delinsA mutation is located in the kinase domain in exon 11 of the protein. In vitro studies with BxPC-3 cells expressing BRAF V487_P492delinsA demonstrate that the mutation is activating as measured by increased pathway activation and cellular proliferation compared to wildtype (PMID: 26732095). Experiments in cell lines and mouse models demonstrate that BRAF alpha-C helix indel mutants, such as V487_P492delinsA, are insensitive to vemurafenib and maintain activity of the MAPK pathway upon drug treatment (PMID: 26732095).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 14:32:57',NULL,NULL,NULL,NULL,'ab4e90a9-8fdf-480f-9078-717b7eae8560',673,NULL),(2675,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'a26d68bf-e048-4eb0-86c6-f03db77c89b7',673,NULL),(2676,NULL,'The class II BRAF T488_P492del mutation is located in the kinase domain in exon 11 of the protein (Abstract: Severson et al. Abstract #40P, Annals of Oncology 2022. https://www.annalsofoncology.org/article/S0923-7534%2822%2900058-8/fulltext). This mutation has been found in pancreatic cancer (PMID: 26996308). Expression of this mutation in a human cell line demonstrated that it is activating as measured by increased kinase activity and downstream signaling compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 14:34:39',NULL,NULL,NULL,NULL,'a73f9de4-3f0a-4a19-a926-8527631caf46',673,NULL),(2677,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'3b3cd9ed-bf98-4a48-8686-4462e5daffda',673,NULL),(2678,NULL,'The class II BRAF P490_Q494del mutation is located in the kinase domain in exon 11 of the protein (Abstract: Severson et al. Abstract #40P, Annals of Oncology 2022. https://www.annalsofoncology.org/article/S0923-7534%2822%2900058-8/fulltext). This mutation has been found in pancreatic cancer (PMID: 26996308). Expression of this mutation in a human cell line demonstrated that it is activating as measured by increased kinase activity and downstream signaling compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-08 16:03:07',NULL,NULL,NULL,NULL,'bdfa68e2-fbeb-4cd0-be3a-5f99c698a832',673,NULL),(2679,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'2af03236-4075-4338-a940-501eff3ca3a6',673,NULL),(2680,NULL,'The BRAF R506_K507insVLR mutation (also referred to as BRAF V504_R506dup) is located in the kinase domain in exon 13 of the protein. This mutation has been found in pilocytic astrocytoma (PMID: 23817572). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by increased downstream pathway activation compared to wildtype BRAF (PMID: 23817572). Preclinical studies with HEK293T cells expressing BRAF R506_K507insVLR demonstrated sensitivity to treatment with trametinib as measured by inhibition of ERK signaling, and resistance to treatment with PLX8394, dabrafenib, vemurafenib, LY3009120 and sorafenib as measured by sustained ERK signaling (PMID: 34108213).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:11:19',NULL,NULL,NULL,NULL,'e33ccee2-e00e-4d90-ab7c-b8dd00dc6146',673,NULL),(2681,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'cea46944-4ddb-4879-85e5-4f5e9096b01b',673,NULL),(2682,NULL,'The BRAF T599dup mutation is located in the kinase domain in exon 15 of the protein. This mutation has been found in melanoma and pilocytic astrocytoma (PMID: 19363522, 21190184, 29456854, 38781546). Expression of this mutation in HEK293, NIH-3T3 or MCF-10A cells demonstrated that this mutation is activating as measured by kinase activity, anchorage-independent growth and changes in contact inhibition comparable to BRAF V600E (PMID: 19363522, 21190184, 23833300). In a case report, four patients with metastatic melanoma harboring BRAF T599dup were treated with a combination therapy of dabrafenib plus trametinib and demonstrated partial response (PMID: 35319964). Another patient with metastatic melanoma harboring BRAF T599dup was treated with a combination therapy of binimetinib plus encorafenib and demonstrated a partial response and a progression-free survival of at least nine months (PMID: 35319964). Two other patients with metastatic melanoma harboring BRAF T599dup were treated with vemurafenib and demonstrated progressive disease (PMID: 35319964). In a case report, a pediatric patient with pilocytic astrocytoma harboring BRAF T599ins was successfully treated with dabrafenib (PMID: 38781546). In another case report, a patient with metastatic colorectal cancer was treated with a combination therapy of panitumumab plus FOLFOX and demonstrated progressive disease (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2024-08-21 16:49:34',NULL,NULL,NULL,NULL,'aa40d7f6-5e58-4764-a5d0-ee4acadfd3ad',673,NULL),(2683,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'9da9cbef-6692-46d3-91d5-4c2f5c0259d5',673,NULL),(2684,NULL,'The BRAF T599insTT mutation is located in the kinase domain in exon 15 of the protein. This mutation has been found in pilocytic astrocytoma (PMID: 21190184). Expression of this mutation in HEK293 cells or MCF-10A cells demonstrated that this mutation is activating as measured by kinase activity and changes in contact inhibition comparable to the BRAF V600E mutation (PMID: 21190184).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'9d37b580-d00b-4bb8-ae07-6b3c35e024ed',673,NULL),(2685,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'0aee4a9e-61b8-4030-9b4e-dac8f285ac2a',673,NULL),(2686,NULL,'The BRAF T599_V600insV mutation is located in the kinase domain in exon 15 of the protein. This mutation has been found in papillary thyroid carcinoma (PMID: 16501605). Expression of BRAF T599_V600insV in HEK293 or NIH-3T3 cells demonstrated that it is activating and transforming as measured by increased kinase activity, increased downstream pathway activation, and increased foci formation compared to wildtype BRAF (PMID: 16501605).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'666cbc41-40e1-4e09-8c5d-c452e60b3d37',673,NULL),(2687,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'e28500fd-4a6d-423f-b6d3-7c78170548bd',673,NULL),(2688,NULL,'The BRAF V600delinsYM mutation is located in the kinase domain in exon 15 of the protein. This mutation has been found in papillary thyroid carcinoma (PMID: 22752848). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by increased kinase activity and increased cell proliferation compared to wildtype BRAF (PMID: 22752848).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'228dffe7-0f4b-4554-bb4f-088824fab2ae',673,NULL),(2689,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'037bd74e-dbb8-43bf-ba1a-d3d48ba1d100',673,NULL),(2690,NULL,'The BRAF V600_K601insDFGLAT alteration is located in the kinase domain in exon 15 of the protein. This mutation has been found in papillary thyroid cancer (PMID: 17297294). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating as measured by increased downstream pathway activation and colony formation compared to wildtype BRAF (PMID: 17297294).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2020-11-17 20:51:17',NULL,NULL,NULL,NULL,'e5797c7f-7268-48b6-8ead-783fcab2d999',673,NULL),(2691,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'ab18a9a6-b6c2-421c-b74d-08ab02c44d09',673,NULL),(2692,NULL,'BRAF fusions generally arise from chromosomal translocations that fuse the N-terminal end of a partner gene with the C-terminal end of BRAF (exons 9-18, containing the kinase domain), such that the fusion protein excludes the BRAF CR1 regulatory domain (PMID:15630448), thereby resulting in a constitutively active BRAF kinase. These class II hyperactivating BRAF fusions have been found in melanoma, prostate cancer, gastric cancer, and multiple other cancers (PMID: 28783719, 26343582, 24345920, 20526349, 25985019, 26324360, 18974108). Biological characterization of diverse BRAF fusion proteins demonstrate that they activate the downstream MAPK pathway independent of RAS (PMID: 24345920, 21424530, 22745804, 21424530, 18974108, 26343582), render BRAF active as a homo- or heterodimer dimer with CRAF (PMID: 26343582), and, while sensitive to MEK inhibition by targeted inhibitors such as trametinib (PMID: 24345920, 28783719, 26343582, 26314551), are insensitive to RAF monomer inhibitors such as vemurafenib and dabrafenib (PMID: 26343582, 28783719). BRAF fusions have been found across multiple studies in post-treatment samples of patients with EGFR-mutant lung cancer who progressed on osimertinib (PMID: 30257958, 30073261).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-06-05 11:14:09',NULL,NULL,NULL,NULL,'896baaf7-1159-4038-9434-8a5539537581',673,NULL),(2693,NULL,'There is promising clinical data from a case study in a patient with BRAF fusion-positive ovarian cancer treated with the FDA-approved MEK1/2-targeted inhibitor selumetinib in combination with paclitaxel, and in patients with BRAF fusion-positive melanoma treated with the FDA-approved MEK1/2-targeted inhibitor trametinib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-26 15:43:50',NULL,NULL,NULL,NULL,'2a6d18bb-0a61-4615-a43d-1818b7dd46ea',673,NULL),(2694,NULL,'Cobimetinib and trametinib are small molecule inhibitors of the MEK1/2 kinases that are FDA-approved in combination with a RAF inhibitor for use in BRAF V600E-positive metastatic melanoma. In a study of patients with low-grade serous ovarian cancer, a patient with a CUL1-BRAF fusion who progressed on carboplatin and paclitaxel had a sustained complete response to treatment with paclitaxel and a MEK1/2 inhibitor (selumetinib) (PMID: 26324360).  Additionally, in vitro studies have demonstrated that BRAF fusions are sensitive to MEK inhibition as measured decreased downstream MAPK pathway signaling in BRAF fusion-containing cells upon treatment with trametinib (PMID: 24345920).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 14:50:08',NULL,'LEVEL_3A','NO','LEVEL_3B','e569bfd4-15f5-421d-9d4e-31dfb7a82903',673,NULL),(2695,NULL,'There is promising clinical data in patients with BRAF fusion-positive melanoma treated with the FDA-approved MEK1/2-targeted inhibitor trametinib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-26 15:44:02',NULL,NULL,NULL,NULL,'4d13aa6b-fb7c-45e4-8c43-fe1223fdd53f',673,NULL),(2696,NULL,'Cobimetinib and trametinib are orally available, small-molecule inhibitors of the MEK1/2 kinases that are FDA-approved in combination with a RAF inhibitor for use in BRAF V600E-positive metastatic melanoma. There is promising clinical data in patients with BRAF fusion-positive melanoma treated with cobimetinib and trametinib monotherapies.\nTwo studies from independent groups have shown that three patients with BRAF-fusion positive metastatic melanoma experienced clinical benefit, such as improvement of ECOG status, tumor shrinkage, or stable disease upon treatment with the MEK inhibitor trametinib (PMID: 26314551, 26072686). In the Phase II VICC MEL 1457 (NCT02296112) trial of trametinib in patients with BRAF non-V600 mutant metastatic melanoma (n=2, patient with BRAF fusion), one patient harboring a BRAF-AGAP3 fusion achieved a partial response with an on-going progression-free survival of 1.8 months and one patient harboring a BRAF-MZT1 fusion demonstrated progressive disease (PMID: 33861486).\nIn vitro studies have demonstrated that BRAF fusions are sensitive to MEK inhibition as measured by decreased downstream MAPK pathway signaling in BRAF fusion-containing cells upon treatment with trametinib (PMID: 24345920).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2025-03-12 15:07:32',NULL,'LEVEL_3A','NO','LEVEL_3B','cec96519-6479-4b9e-9d89-42e3b2a2bc94',673,NULL),(2697,NULL,'While the pan-RAF kinase inhibitor tovorafenib is FDA-approved for the treatment of patients 6 months of age and older with pediatric low-grade glioma harboring a BRAF fusion, and there is promising clinical data demonstrating the clinical utility of MEK1/2-targeted inhibitors such as FDA-approved trametinib in patients with BRAF fusion-positive melanoma and ovarian cancer, the clinical utility of these inhibitors in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-04-30 12:10:54',NULL,NULL,NULL,NULL,'67f8217f-063d-454f-c0a1-9ad1481afc59',673,NULL),(2698,NULL,'The MEK inhibitor selumetinib is NCCN-compendium listed as a treatment option for patients with BRAF fusion-positive pilocytic astrocytoma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-08-26 18:34:29',NULL,NULL,NULL,NULL,'5306d2e7-1e09-4cfe-ae94-152e3244c7f6',673,NULL),(2699,NULL,'Selumetinib is a small molecule tyrosine kinase inhibitor of MEK1/2. In stratum one of the phase II PBTC study of selumetinib in 25 patients with pilocytic astrocytoma harboring a KIAA1549BRAF fusion or BRAF V600E mutation, seven of eighteen patients with a KIAA1549BRAF fusion had a partial response to treatment (PMID: 31151904). Additionally, the two year progression-free survival rate for the BRAF study population (n=25) was 70% (95% CI = 4785) (PMID: 31151904).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 14:50:30',NULL,'LEVEL_2','NO','LEVEL_3B','a05f8f2e-5d26-4639-9c10-019b8183fc33',673,NULL),(2700,NULL,'The pan-RAF kinase inhibitor tovorafenib is FDA-approved for the treatment of patients 6 months of age and older with pediatric low-grade glioma harboring a BRAF fusion or rearrangement.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-04-26 14:21:48',NULL,NULL,NULL,NULL,'6b87637f-0271-4c56-8c1b-5a6ccaca7780',673,NULL),(2701,NULL,'Tovorafenib is an orally available, pan-RAF small molecule inhibitor that is FDA-approved for the treatment of pediatric patients six months of age and older with relapsed or refractory pediatric low-grade glioma (LGG) harboring a BRAF fusion or rearrangement, or BRAF V600 mutation. FDA approval was based on the results of the Phase II FIREFLY-1 (NCT04775485) trial of tovorafenib in 76 patients (median age=8 years old [range=2-21]) with relapsed or refractory pediatric LGG harboring an activating BRAF alteration based on local laboratory testing. In the Phase II FIREFLY-1 (NCT04775485) trial, the overall RAPNO-LGG cohort demonstrated an objective response rate (ORR) of 51% (95% CI=40-63), with a 37% (n=28) partial response rate and 14% (n=11) minor response rate, a median duration of response (DOR) of 13.8 months (95% CI=11.3-NE) in 39 patients and a median progression-free survival of 13.8 months (95% CI=8.3-16.9) (PMID: 37978284). The BRAF fusion subcohort (n=64 [KIAA1549-BRAF fusion, n=56; other BRAF duplication or rearrangement by FISH, n=8]) demonstrated an ORR of 52% (95% CI=39-64) and a median DOR of 13.8 months (95% CI=11.3-NE) (PMID: 37978284).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2024-04-26 14:21:56',NULL,'LEVEL_1','NO','LEVEL_3B','8bd8579f-8d4f-42ef-9271-8e3149c2490f',673,NULL),(2702,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2019-07-29 23:12:54',NULL,NULL,NULL,NULL,'bb7f48fd-2962-44c1-a76a-08465a8aa7cf',673,NULL),(2703,NULL,'The AGAP3-BRAF fusion protein results from an inversion on chromosome 7, joining exons 1-9 of AGAP3 with a 3\' segment of the BRAF gene. The AGAP3-BRAF fusion protein contains the full BRAF kinase domain, but lacks the regulatory RAS-binding domain and the cysteine-rich (CR1) domain, which are responsible for BRAF binding to RAS and auto-inhibition. This fusion protein has been found in melanoma, colorectal cancer and ovarian serous carcinoma (PMID: 26314551, 28539463, 30279230, 34568720).  Expression of this fusion in NIH3T3 cells demonstrated that it is activating as measured by increased MAPK pathway activity and cellular growth compared to wildtype BRAF (PMID: 30279230). Expression of this fusion in cell lines demonstrated that it is sensitive to trametinib but not vemurafenib as measured by in vitro colony formation and cell proliferation (PMID: 28539463). A patient with BRAF V600E-mutated metastatic melanoma, who experienced disease relapse after an initial response to the BRAF inhibitor Vemurafenib, was subsequently treated with dabrafenib plus trametinib (RAF inhibitor plus MEK inhibitor), and had a partial response for a period of two months (PMID: 28539463).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-10-12 15:51:05',NULL,NULL,NULL,NULL,'9f208268-d2d3-415b-820f-472944a8850e',673,NULL),(2704,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'e535c945-69cf-4a98-8ec3-6161c1706862',673,NULL),(2705,NULL,'The AGK-BRAF fusion protein results from an inversion of chromosome 7, which fuses the first two exons of AGK with exons 8-18 of BRAF. The AGK-BRAF fusion protein contains the full BRAF kinase domain, but lacks the regulatory RAS-binding domain and the cysteine-rich (CR1) domain, which are responsible for BRAF binding to RAS and auto-inhibition (PMID: 23890088, 24135138). This fusion has been found in melanoma, glioma, lung cancer, ovarian serous carcinoma and pediatric post-Chernobyl thyroid cancer (PMID: 23890088, 24135138, 26314551, 34568720). Expression of AGK-BRAF in mouse fibroblasts demonstrated that this fusion is activating and transforming as measured by increased growth and colony formation in soft agar compared to GFP control (PMID: 24135138). In vitro studies of AGK-BRAF in melanoma cell lines suggest that this fusion is sensitive to sorafenib but not vemurafenib as measured by cell viability upon drug treatment (PMID: 23890088). AGK-BRAF fusion was observed in 3 of 174 EGFR-mutant lung adenocarcinoma with primary resistance to EGFR tyrosine kinase inhibitors (TKIs).  Functional studies with the induction of AGK-BRAF fusion in H1975 ( EGFR L858R + T790M), PC9 (EGFR ex19del) and HCC827 (EGFR ex19del) cell lines demonstrated that the fusion conferred resistance to growth inhibition by EGFR TKIs (PMID: 30831205).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2021-10-12 16:18:20',NULL,NULL,NULL,NULL,'ec17e9b0-111a-46a9-a462-3608dcfa3eac',673,NULL),(2706,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'60c9eb9a-dff8-4f07-a2de-5e0df3641d31',673,NULL),(2707,NULL,'The AKAP9-BRAF fusion protein results from an inversion of chromosome 7 that fuses exons 1-8 of AKAP9 and exons 9-18 of BRAF. This fusion contains the BRAF kinase domain, but excludes the BRAF CR1 domain responsible for RAS binding and auto-inhibition (PMID:15630448). Expression of AKAP9-BRAF in COS7 cells demonstrated that it is activating as measured by increased kinase activity compared to wildtype BRAF, and expression of AKAP9-BRAF in NIH-3T3 cells demonstrated that it is transforming as measured by increased foci formation compared to wildtype BRAF (PMID:15630448).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'db923d23-b9a7-4aec-9c62-f8360acb48a7',673,NULL),(2708,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'1c201a83-1800-4ce4-a7d9-82e7ff1a3fd0',673,NULL),(2709,NULL,'The CUL1-BRAF fusion protein results from an in-frame translocation of chromosome 7 that fuses exons 1-7 of CUL1 with exons 9-18 of BRAF (PMID: 26324360). The CUL1-BRAF fusion protein contains the BRAF kinase domain, but lacks the BRAF CR1 domain that is responsible for RAS-binding and auto-inhibition. This fusion has been found in low-grade serous ovarian cancer (PMID: 26324360). While this fusion has not been functionally validated, other fusions that contain the BRAF kinase domain an exclude the CR1 domain have been shown to be activating and oncogenic (PMID: 22745804, 25985019, 15630448). A patient with low-grade serious ovarian cancer harboring the CUL1-BRAF fusion achieved a sustained complete response to combination selumetinib and paclitaxel (PMID: 26324360). In a study of patients with low-grade serous ovarian cancer, a patient with a CUL1-BRAF fusion who progressed on carboplatin and paclitaxel had a sustained complete response to treatment with combined selumetinib and paclitaxel (PMID: 26324360).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-08-18 17:09:00',NULL,NULL,NULL,NULL,'4a726bda-f66f-4715-b83d-a3cd788ba09d',673,NULL),(2710,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'25243c5f-b3b1-4811-82e3-123c7c50193a',673,NULL),(2711,NULL,'The FAM131B-BRAF fusion is the result of a 2.5 Mb deletion on chromosome 7 which fuses the first two exons of the FAM131B gene with exons 9-18 of the BRAF gene (PMID: 21424530). This fusion contains the BRAF kinase domain, but excludes the BRAF CR1 domain responsible for RAS binding and auto-inhibition (PMID:15630448). The FAM131B-BRAF fusion has been found in pilocytic astrocytomas and pediatric low-grade gliomas (PMID: 21424530, 25040262). Expression of this mutation in NIH-3T3 cells demonstrated that it is activating, as measured by increased downstream MEK phosphorylation and decreased contact inhibition compared to wildtype BRAF (PMID: 21424530). Preclinical studies with mice and NIH3T3 cells expressing FAM131B-BRAF demonstrate sensitivity to treatment with trametinib as measured by suppressed tumor growth and anchorage-independent growth; however, acquired resistance was observed following prolonged treatment (PMID: 29156677). FAM131B-BRAF is suggested to acquire resistance against MEK inhibitors selumetinib, binimetinib and trametinib through enhancing PI3K/AKT/mTOR pathway signaling to overcome MEK inhibition (PMID: 29156677). Combination treatment of mTOR inhibitor everolimus with trametinib in mice expressing FAM131B-BRAF prolonged tumor suppression and delayed acquired resistance (PMID: 29156677).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-07 18:12:22',NULL,NULL,NULL,NULL,'81a3f713-19b7-4236-bdad-ec27d8558e27',673,NULL),(2712,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-11-29 19:47:36',NULL,NULL,NULL,NULL,'71d38212-4aaa-400f-a4e5-e1b22e33a12c',673,NULL),(2713,NULL,'The GTF2I-BRAF fusion protein results from a rearrangement on chromosome 7, joining exons 1-19 of GTF21 to intron 9 BRAF, thus retaining the full kinase domain of BRAF. The GTF2I-BRAF fusion has been identified in melanoma and pilocytic astrocytoma (PMID: 26314551, 28448514). When expressed in a human embryonic kidney cell line, the fusion protein elevated basal levels of phosphorylated ERK compared to the wildtype BRAF (PMID: 28448514). The fusion was detected in a patient with pilocytic astrocytoma and immunohistochemistry confirmed elevated phospho-ERK in tumor tissue from this patient (PMID: 28448514).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 19:13:34',NULL,NULL,NULL,NULL,'6ef06341-1ca9-4f6c-8a4c-293b89cb6f0d',673,NULL),(2714,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'55d693ef-39c8-4ddb-9623-8a864eea593b',673,NULL),(2715,NULL,'The KIAA1549-BRAF fusion protein results from an internal tandem duplication event on chromosome 7 that fuses exons 1-16 of KIAA1549 and exons 9-18 of BRAF. This fusion contains the BRAF kinase domain, but excludes the BRAF CR1 domain responsible for RAS binding and auto-inhibition (PMID: 18974108, 24345920). This fusion has been found in pilocytic astrocytoma (PMID: 18974108, 24345920). Expression of this fusion in NIH-3T3 cells demonstrated that it is activating as measured by constitutive kinase activity and increased anchorage independent growth in soft agar compared to wildtype BRAF (PMID: 18974108). KIAA1549-BRAF fusion protein functions as a dimer and is therefore not sensitive to RAF inhibitors such as vemurafenib in vitro (PMID: 23533272). Preclinical evidence suggests that these oncoproteins may be targeted by second-generation RAF inhibitors, which are reported to inhibit RAF dimers rather than induce their transactivation of RAF (as is the case with older inhibitors such as vemurafenib) (PMID: 23533272). In vitro studies have suggested that pediatric astrocytoma cell lines harboring the KIAA1549-BRAF fusion are resistant to treatment with first generation RAF inhibitors (PMID: 23533272, 26072686). In a Phase II clinical trial of selumetinib in 25 patients with WHO grade I pilocytic astrocytoma harboring either the KIAA1549-BRAF fusion or the BRAF V600E mutation, nine patients (36% [95% CI:18-57%]) had a sustained partial response (PMID: 31151904).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2019-10-27 21:12:36',NULL,NULL,NULL,NULL,'f98b6b73-bf97-4869-aec2-1234627962f3',673,NULL),(2716,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'89c2cc4a-b480-4509-983b-a785ba959878',673,NULL),(2717,NULL,'The MKRN1-BRAF fusion results from an inversion of chromosome 7 that fuses exons 1-4 of MKRN1 and exons 11-18 of BRAF (PMID: 23817572). This fusion contains the BRAF kinase domain but excludes the BRAF cysteine-rich (CR1) domain responsible for RAS binding and auto-inhibition (PMID:15630448). This mutation has been found in pediatric pilocytic astrocytoma, colorectal cancer, head and neck cancer, and ovarian serous cancer (PMID: 23817572, 26314551, 26324360, 34568720). While this fusion has not been functionally validated, other fusions that contain exons 11-18 of BRAF (e.g., AKAP9-BRAF, KIAA1549-BRAF) have been shown to be activating and oncogenic (PMID: 15630448, 18974108, 23817572). The MKRN1-BRAF fusion was detected in a patient with low-grade serous ovarian cancer who achieved a sustained complete response to combination selumetinib and paclitaxel (PMID: 26324360, 34568720).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-10-12 16:18:54',NULL,NULL,NULL,NULL,'621199d9-da07-472a-b38b-2674f36450af',673,NULL),(2718,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'5d30335b-e959-4424-9f22-37e9577b519e',673,NULL),(2719,NULL,'The PAPSS1-BRAF fusion protein results from a fusion of chromosomes 4 and 7 that fuses exons 1-5 of PAPSS1 and exons 9-18 of BRAF (PMID: 24345920). This fusion contains the BRAF kinase domain, but excludes the BRAF CR1 domain responsible for RAS binding and auto-inhibition (PMID:15630448). This fusion has been found in melanoma (PMID: 24345920). Expression of this fusion in 293H cells demonstrated that it is activating as measured by activation of downstream MEK and ERK that is comparable to BRAF V600E (PMID: 24345920). In vitro experiments have suggested that the PAPSS1-BRAF fusion is resistant to vemurafenib but sensitive to trametinib as measured by changes in MEK and ERK phosphorylation upon drug treatment (PMID: 24345920).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'aac8fa91-a07d-4eb8-b3be-b9f2132cb9b4',673,NULL),(2720,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic',NULL,NULL,NULL,NULL,NULL,'0e288881-f144-4f60-b565-d607cb2c22d2',673,NULL),(2721,NULL,'The PPFIBP2-BRAF fusion joins the N-terminus of PPFIBP2 with the C-terminus of BRAF resulting in a fusion protein containing the BRAF kinase domain but lacking the BRAF CR1 regulatory domain. While this fusion has not been functionally validated, other BRAF fusions lacking the BRAF CR1 regulatory domain (e.g. KIAA 1549-BRAF fusion) have been shown to be activating and oncogenic. The PPFIBP2-BRAF fusion was found in a patient with metastatic melanoma who responded to MEK inhibition with trametinib (PMID: 26072686).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2018-02-28 20:53:06',NULL,NULL,NULL,NULL,'935876ae-3fd7-430d-b5bd-f9f34f8f540b',673,NULL),(2722,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'dfdf1602-37b6-40f4-a110-c0c62594f6e3',673,NULL),(2723,NULL,'The SND1-BRAF fusion protein results from a rearrangement of chromosome 7 and fuses exons 1-16 of SND1 with exons 9-18 of BRAF. The SND1-BRAF fusion protein contains the BRAF kinase domain, but lacks the BRAF CR1 domain that is responsible for RAS-binding and auto-inhibition (PMID: 24345920). Expression of SND1-BRAF in a lung adenocarcinoma cell line, H1299, or a gastric cancer cell line, GTL16, demonstrated that this fusion is activating as measured by increased activation of downstream MEK and ERK signaling proteins compared to wildtype BRAF (PMID: 22745804, 25985019). Expression of SND1-BRAF in Ba/F3 cells demonstrated that this fusion is transforming as measured by IL-3-independent growth (PMID: 25266736). In vitro studies suggest that this fusion is sensitive to MEK inhibition, as measured by decreased cell viability upon treatment with trametinib (PMID: 25266736).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'8674d546-7ec4-40ba-a4b1-0dbb0e112823',673,NULL),(2724,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'b5a0f9dd-a623-480e-a8ce-fc163330249e',673,NULL),(2725,NULL,'The TRIM24-BRAF fusion protein results from an inversion of chromosome 7 that fuses exons 1-9 of TRIM24 and exons 9-18 of BRAF (PMID: 24345920). This fusion contains the BRAF kinase domain, but excludes the BRAF CR1 domain responsible for RAS binding and auto-inhibition (PMID:15630448). This fusion has been found in melanoma (PMID: 24345920). Expression of this fusion in 293H cells demonstrated that it is activating as measured by activation of downstream MEK and ERK that is comparable to BRAF V600E (PMID: 24345920). In vitro experiments have suggested that the TRIM24-BRAF fusion is resistant to vemurafenib but sensitive to trametinib as measured by changes in MEK and ERK phosphorylation upon drug treatment (PMID: 24345920).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2019-07-20 11:35:05',NULL,NULL,NULL,NULL,'2ae1fb52-59a2-4a65-8fea-3fe33187dca6',673,NULL),(2726,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic',NULL,NULL,NULL,NULL,NULL,'90c7bd08-6966-4b0b-aaf0-acf1e1625bdd',673,NULL),(2727,NULL,'The ZKSCAN1-BRAF fusion joins the N-terminus of ZKSCAN1 with the C-terminus of BRAF, resulting in a fusion protein containing the BRAF kinase domain but lacking the BRAF CR1 regulatory domain. While this fusion has not been functionally validated, other BRAF fusions lacking the BRAF CR1 regulatory domain (e.g. KIAA 1549-BRAF fusion) have been shown to be activating and oncogenic. The ZKSCAN1-BRAF fusion was found in a patient with spitzoid melanoma who responded to the MEK inhibitor trametinib (PMID: 26314551).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2018-03-05 16:20:25',NULL,NULL,NULL,NULL,'465766c4-ccec-4657-9331-3b277c8d36eb',673,NULL),(2728,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'ee6a98fc-190a-40d0-8b1e-6aa8726a3a2b',673,NULL),(2729,NULL,'BRAF amplification is a rare event in cancer and has been found to occur at very low frequency in melanoma and glioma (cBioPortal, MSKCC, April 2017). It has been demonstrated that amplification of wildtype BRAF in melanoma cells leads to overexpression of the protein and constitutive activation of MAPK pathway, demonstrated as a BRAF-dependent increase in ERK phosphorylation, as well as an increase in cell proliferation (PMID: 15467732). BRAF overexpression has been associated with increased proliferation in gliomas (PMID: 25618114). Amplification of BRAF V600E allele has been described as a mechanism of resistance to MEK/RAF inhibitors in melanoma via MAPK pathway hyperactivation (PMID: 22395615, 24055054, 21098728).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2019-07-20 11:26:14',NULL,NULL,NULL,NULL,'7c4682f3-b122-4160-bb8b-fd1585a3b73e',673,NULL),(2730,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:46:25',NULL,NULL,NULL,NULL,'fd6a7bec-aedd-46f3-8c87-1a3ed90ed1f5',673,NULL),(2731,NULL,'The class III BRAF D594H missense mutation occurs in the kinase domain in exon 15 of the protein (PMID: 28783719). This mutation has been reported in melanoma, lung and colorectal cancers (PMID: 28481359). Biological characterization of the BRAF D594H mutation has demonstrated that it is RAS dependent in its activation of the MAPK pathway, signals as a dimer with CRAF, and typically co-occurs with another driver mechanism that amplifies RAS signaling (PMID: 28783719). Thus, when BRAF D594H is observed in melanoma, it typically co-occurs with an activating RAS mutation or NF1 inactivation, whereas when observed in non-small cell lung or colorectal cancers, it is associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719, 20141835). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-06 15:51:18',NULL,NULL,NULL,NULL,'be6acfec-a41a-46ca-b9ce-e79a9b7ee84a',673,NULL),(2732,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:44:11',NULL,NULL,NULL,NULL,'08e3242b-d663-42c0-bdfe-978c4e27c4e7',673,NULL),(2733,NULL,'The class III BRAF N581I mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719). This mutation has been reported in lung cancer and vulval malignant melanoma (PMID: 22969966, 26202550, 15917418). Biochemically, it has been demonstrated that the BRAF N581I mutant signals as a dimer with CRAF, is RAS-dependent in its activation of the MAPK pathway, and, in tumors, would typically co-occur with another driver mechanism that amplifies RAS signaling (PMID: 28783719). Thus, in melanoma, the N581I variant, similar to other class III BRAF mutants, would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung or colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719). In a case report, a patient with metastatic colorectal cancer harboring BRAF N581I was treated with a combination therapy of cetuximab plus irinotecan and demonstrated stable disease with a progression-free survival of 10.3 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:14:01',NULL,NULL,NULL,NULL,'0cc9d835-eac3-44d0-b116-4aaca9cf6dac',673,NULL),(2734,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:47:22',NULL,NULL,NULL,NULL,'ae1e8ae2-ae37-432a-b750-d53fd88d0c79',673,NULL),(2735,NULL,'The class III BRAF G596D mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719). This mutation has been reported in cervical adenocarcinoma and esophagogastric cancer (PMID: 17360030, 23481512). Biochemically, it has been demonstrated that the BRAF G596D mutant signals as a dimer with CRAF, is RAS-dependent in its activation of the MAPK pathway, and, in tumors, would typically co-occur with another driver mechanism that amplifies RAS signaling (PMID: 28783719). Thus, like other class III BRAF mutants, in melanoma, the G596D variant would be expected to co-occur with activating RAS mutation or NF1 inactivation, whereas in non-small cell lung and colorectal cancers, this mutation would most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719). Preclinical studies with SKBR3 cells expressing BRAF G596D demonstrate resistance to treatment with dabrafenib as measured by sustained ERK phosphorylation following treatment (PMID: 31158244).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:15:15',NULL,NULL,NULL,NULL,'49953b55-2bdb-486f-b7e4-89fe7668145b',673,NULL),(2736,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic',NULL,NULL,NULL,NULL,NULL,'b43cee2e-c580-4a59-b413-2a9b73025243',673,NULL),(2737,NULL,'The BRAF P367R mutation occurs in exon 8 of the gene. A study evaluating the effects of rare somatic mutations in lung cancers showed that cells transfected with BRAF P367R mutation display a mild increase in ERK phophorylation compared to wildtype (PMID: 27478040).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function',NULL,NULL,NULL,NULL,NULL,'31cdd5bf-1988-4e5c-9e4e-763b9e2f48cb',673,NULL),(2738,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic',NULL,NULL,NULL,NULL,NULL,'c12b63c0-7113-4bff-bb65-5012665295d1',673,NULL),(2739,NULL,'The BRAF H574Q mutation is located in exon 14 of the kinase domain of the protein. This mutation has been identified in glioma and melanoma (PMID: 34625582, 28972077). A study evaluating the effects of rare somatic mutations in lung cancers showed that cells transfected with BRAF H574Q mutation display a robust increase in ERK phosphorylation, comparable to BRAF V600E mutation (PMID: 27478040).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-09 19:15:46',NULL,NULL,NULL,NULL,'6e4e140e-18bf-423f-94d3-f0747d28683c',673,NULL),(2740,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral',NULL,NULL,NULL,NULL,NULL,'81aec819-7b4c-4046-9d85-fcf61e35c5d5',673,NULL),(2741,NULL,'The BRAF W450L mutation is located in the kinase domain of the BRAF protein. This mutation has been found in lung cancers, among others (PMID: 27478040). In vitro studies have demonstrated that this mutation is likely neutral as measured by downstream signaling compared to wildtype (PMID: 27478040).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2020-09-20 17:34:17',NULL,NULL,NULL,NULL,'a4bddc9f-5015-46ac-bb44-29ac9c0ba4ab',673,NULL),(2742,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:43:30',NULL,NULL,NULL,NULL,'1e49a5d0-e272-4aae-9808-df8d11f19caf',673,NULL),(2743,NULL,'The class III BRAF S467L mutation occurs in the kinase domain in exon 11 of the protein (PMID: 28783719). This mutation has been reported in melanoma (PMID: 24710085, 25651238, 26498038, 26559571). Biochemically, it has been demonstrated that the BRAF S467L mutant signals as a dimer with CRAF, is RAS-dependent in its activation of the MAPK pathway, and, in tumors, would typically co-occur with another driver mechanism that amplifies RAS signaling (PMID: 28783719). Therefore, class III BRAF mutants like the BRAF S467L variant would be generally expected to co-occur with activating RAS mutations or NF1 inactivation in melanomas, whereas in non-small cell lung or colorectal cancers, this mutation will most likely be associated with receptor tyrosine kinase (RTK) hyperactivation at the protein level (PMID: 28783719).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:16:16',NULL,NULL,NULL,NULL,'2cff07a9-00cc-46ec-b107-d2f5af04f8ba',673,NULL),(2744,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2018-06-19 11:36:00',NULL,NULL,NULL,NULL,'e284c089-14e5-44d4-ad50-ab260d2d8f45',673,NULL),(2745,NULL,'The class II BRAF K601T mutation is located in the kinase domain in exon 15 of the protein (PMID: 28783719, 26343582). This mutation has been reported in hairy cell leukemia and multiple myeloma (PMID: 25938346, 26090869). Biochemical characterization of BRAF K601T has demonstrated that it activates the downstream MAPK pathway independent of RAS and renders BRAF active as a dimer with CRAF and itself (PMID: 26343582). While this mutation is sensitive to MEK inhibition by targeted inhibitors such as trametinib, it is insensitive to RAF monomer inhibitors such as vemurafenib (PMID: 28783719, 26343582, 22798288).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2023-03-09 19:16:52',NULL,NULL,NULL,NULL,'e34774d6-313e-4da7-824b-87c3d126c302',673,NULL),(2746,NULL,'BRAF class II mutations including [[gene]] [[mutation]] are insensitive to the RAF-targeted inhibitors vemurafenib and dabrafenib. Laboratory data suggest that cancer cells with oncogenic, non-V600 BRAF mutations may be sensitive to novel BRAF inhibitors such as plixorafenib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-06-10 17:52:41',NULL,NULL,NULL,NULL,'522e6413-dfd4-4099-8da9-1f79b192e6e4',673,NULL),(2747,NULL,'Plixorafenib is an orally available, second-generation, small-molecule BRAF inhibitor that inhibits mutated, oncogenic BRAF without causing paradoxical MAPK pathway activation as is seen with first-generation BRAF inhibitors (PMID: 28659148). In a phase I clinical trial for patients with refractory or relapsed solid tumors with or without BRAF mutation, there were two (28%) partial responses of seven evaluable BRAF-mutant patients treated with plixorafenib and cobicistat, a CYP3A4 inhibitor (Abstract: Janku et al. Abstract# B176, AACR-NCI-EORTC 2017. http://mct.aacrjournals.org/content/17/1_Supplement/B176). Both responses were in patients with BRAF V600 mutations. In a phase I/IIa trial of plixorafenib for patients with BRAF-altered, advanced solid or CNS tumors (n=110; n=61, patients with class 1 mutations; n=19, patients with class 2 mutations [excluding fusions]; n=17, patients with BRAF fusions), partial response was achieved in 39% (n=9/23) of patients with MAPKi-naive, V600-mutant tumors, with a median duration of response (DoR) of 32 months, and in 18% (n=3/17) of patients with MAPKi treated, V600-mutant tumors (n=6, patients with V600-mutant glioma; n=3, patients with V600-mutant ovarian cancer; n=1, patients with V600-mutant small bowel cancer; n=1, patients with V600-mutant papillary thyroid cancer; n=1, patients with V600-mutant anaplastic thyroid cancer) (Abstract: Fuente et al. Abstract# 3006, J. Clin. Oncol. 2023, Vol 41, Suppl 16. https://ascopubs.org/doi/10.1200/JCO.2023.41.16_suppl.3006). \nPreclinical studies show that plixorafenib specifically inhibits BRAF-containing dimers and inhibits proliferation and pathway activation in BRAF-mutated colorectal cancer cell lines (PMID: 26466569, 28659148, 30559419, 28783719).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2024-06-10 17:52:28',NULL,'LEVEL_4','NO','LEVEL_4','96034660-351c-4fce-9f82-cdfe98f7a0d2',673,NULL),(2748,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2018-10-17 15:58:17',NULL,NULL,NULL,NULL,'94ff2e7c-e9c1-490b-8f9f-0de4c1b633d1',673,NULL),(2749,NULL,'The class III BRAF F247L mutation is located in the RAS binding and kinase autoinhibitory domain of the BRAF protein (PMID: 31515458). This mutation has been found in colorectal cancers, among others (PMID: 31515458). In vitro studies have demonstrated that this mutation is activating as measured by increased downstream signaling compared to wildtype (PMID: 28512244, 29533785). A patient with metastatic colorectal cancer harboring BRAF F247L was treated with a combination therapy of FOLFIRI plus cetuximab and demonstrated a complete response with a progression-free survival of 12.6 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-07 18:16:10',NULL,NULL,NULL,NULL,'6da5e4e4-ed2a-4c99-83cd-555ceb42bd76',673,NULL),(2750,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2018-10-19 14:02:07',NULL,NULL,NULL,NULL,'06ea9f39-5cdd-4c53-9d3b-fd0740e47125',673,NULL),(2751,NULL,'The BRAF L514V mutation is located in the ATP binding pocket of the kinase domain of the protein. This mutation was found in a patient with BRAF V600E-mutant anaplastic ganglioglioma who progressed on treatment with dabrafenib (PMID: 29880583). Co-expression of this mutation in human cells with BRAF V600E in cis demonstrated that it is activating, promotes dimerization, and confers resistance to type I C-OUT/DFG-IN BRAF inhibitors such as vemurafenib and dabrafenib (PMID: 29880583). The BRAF L514V mutation was shown to be sensitive to inhibition with RAF dimer inhibitors, such as BGB3245 (PMID: 29880583).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2018-10-19 14:44:23',NULL,NULL,NULL,NULL,'fa20c208-4419-4084-96ea-938a2dcd9353',673,NULL),(2752,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2019-02-07 22:56:32',NULL,NULL,NULL,NULL,'d76c1ca3-0fc0-4275-83e3-e01280b4e5f5',673,NULL),(2753,NULL,'The PCBP2-BRAF fusion results from the fusion of the 3\' end of the PCBP2 gene with the 5\' end of the BRAF gene. While this alteration has not been functionally characterized per se, many other BRAF fusions have been shown to be oncogenic suggesting that the PCBP2-BRAF fusion is likely oncogenic (PMID: 24345920, 21424530, 22745804, 21424530, 18974108, 26343582). In a study of patients with non-small cell lung cancer who progressed on osimertinib, one patient harbored the PCBP2-BRAF fusion (PMID: 30257958). Knockdown of BRAF in a cell line created from this patient led to a reduction of cell growth and resensitization to osimertinib (PMID: 30257958). While these cells were insensitive to the BRAF inhibitors dabrafenib and LXH245, they were sensitive to the MEK inhibitor trametinib (PMID: 30257958).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2019-07-20 12:11:32',NULL,NULL,NULL,NULL,'fc022e3e-d9df-4908-b6c7-e4881d450db2',673,NULL),(2754,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2019-03-15 01:28:23',NULL,NULL,NULL,NULL,'373a848a-67c3-465e-a52e-c2d81b49eaea',673,NULL),(2755,NULL,'The BRAF N486_T491delinsK mutation is located in the kinase domain of the protein. This mutation has been found in histiocytic neoplasms (PMID: 30867592). Expression of this mutation in Ba/F3 cells demonstrates that it is activating as measured by increased downstream ERK signaling, cytokine-independent growth, and MEK1/2-inhibitor sensitivity in vitro compared to wildtype BRAF (PMID: 30867592). A patient with histiocytic disease harboring the BRAF N486_T491delinsK mutation had a complete response to treatment with the MEK1/2 inhibitor cobimetinib (PMID: 30867592).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2019-03-24 21:45:12',NULL,NULL,NULL,NULL,'a4caf016-33fd-4eb9-b967-6b18c944b350',673,NULL),(2756,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2019-07-29 16:14:51',NULL,NULL,NULL,NULL,'d8c54d72-0695-4cf9-aefa-e77540c5ffa0',673,NULL),(2757,NULL,'Duplication of the BRAF kinase domain (BRAF-KDD) occurs from a breakpoint in intron 9 and results in a duplication of exons 10-18 to generate a protein that is approximately 140 kDa with a second kinase domain and an alternative 3\' untranslated region. This alteration has been found in lung cancer, melanoma and others (PMID: 32913992, 27320919, 26562024). In vitro studies with HEK293T cells expressing BRAF-KDD demonstrate that the mutation is activating as measured by increased downstream MAPK pathway signaling compared to control (PMID: 27320919). Analysis from sequencing data of clinical samples across multiple tumor types harboring BRAF-KDD also demonstrates that the mutation is activating as measured by increased BRAF mRNA and CCND1 expression, demonstrating increased activation of MAPK signaling (PMID: 26562024). The oncogenic function of BRAF-KDD is predicted to be dimerization-dependent based on preclinical studies of BRAF-KDD in BRAF V600E-positive cells demonstrating sensitivity to pan-RAF dimerization inhibitor LY3009120 (PMID: 27320919). Knockdown of BRAF-KDD in BRAF V600E-positive cells expressing the kinase domain duplication demonstrated restored sensitivity to BRAF inhibitors such as dabrafenib (PMID: 27320919).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-05-22 15:38:21',NULL,NULL,NULL,NULL,'00b243ca-25a4-46b0-a189-7e9c80a3af07',673,NULL),(2758,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2019-07-29 23:17:49',NULL,NULL,NULL,NULL,'b2bfc7d1-dca7-424c-aea1-f4d4f269a868',673,NULL),(2759,NULL,'The ARMC10-BRAF fusion protein combines the N-terminus of ARMC10 with the C-terminus of BRAF, preserving the kinase domain of BRAF. This fusion protein is found in patients with sporadic microsatellite instability-high colorectal cancers (PMID: 30279230). Expression of this fusion protein in NIH3T3 cells demonstrated that it was activating as shown by increased colony formation and pathway activation compared to expression of BRAF alone (PMID: 30279230). Preclinical studies with patient-derived xenograft models expressing ARMC10-BRAF demonstrated sensitivity to treatment with MEK inhibitor trametinib and ERK inhibitor SCH772984 as measured by tumor regression and decreased phosphorylated ERK following treatment (PMID: 30254212).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-06 15:45:43',NULL,NULL,NULL,NULL,'ee026fa7-42df-40e7-bcce-6fc7f23610f4',673,NULL),(2760,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2021-01-11 23:01:24',NULL,NULL,NULL,NULL,'7fb3f203-c29f-4970-9fce-a49bbc9aaba2',673,NULL),(2761,NULL,'The BRAF N486del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely neutral as measured by kinase activity comparable to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2021-01-11 23:04:17',NULL,NULL,NULL,NULL,'8988fbbf-4d5e-4827-8715-7b7ab6fec169',673,NULL),(2762,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:07:09',NULL,NULL,NULL,NULL,'b7c09079-ec39-4eaf-a48a-29699f0561bc',673,NULL),(2763,NULL,'The BRAF N486_V487del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely activating as measured by modestly increased kinase activity compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-01-11 23:07:41',NULL,NULL,NULL,NULL,'efd9fd75-4a60-44c8-afaa-488fdcd9d630',673,NULL),(2764,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:10:06',NULL,NULL,NULL,NULL,'33d8da06-3401-4fe6-ab4e-f2937c13438a',673,NULL),(2765,NULL,'The BRAF N486_T488del mutation is located in the kinase domain in exon 11 of the protein. This mutation has been identified in non-small cell lung cancer (PMID: 36200799). Expression of this mutation in a human cell line demonstrated that it is likely activating as measured by increased kinase activity compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-08 15:59:58',NULL,NULL,NULL,NULL,'8754dd0b-a399-4490-a974-b60c3ff0d914',673,NULL),(2766,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:11:31',NULL,NULL,NULL,NULL,'65cb1cc4-f06b-4618-af23-3e11834cb4ca',673,NULL),(2767,NULL,'The BRAF N486_A489del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely activating as measured by modestly increased kinase activity compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-01-11 23:11:33',NULL,NULL,NULL,NULL,'09be3075-692e-4804-bdef-15c34c209716',673,NULL),(2768,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:26:45',NULL,NULL,NULL,NULL,'f7bc56df-3d31-4f59-9949-cd88d5c8a63e',673,NULL),(2769,NULL,'The BRAF N486_T491del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely activating as measured by modestly increased kinase activity compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-01-11 23:27:22',NULL,NULL,NULL,NULL,'605adff6-663f-455d-9b22-a60cade42cfc',673,NULL),(2770,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2021-01-11 23:29:15',NULL,NULL,NULL,NULL,'6e00085e-9da2-492a-a8c6-a635ebebfb65',673,NULL),(2771,NULL,'The BRAF P490del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely neutral as measured by kinase activity comparable to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2021-01-11 23:29:39',NULL,NULL,NULL,NULL,'beace36e-656a-43af-9044-faf7cf48039c',673,NULL),(2772,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:42:10',NULL,NULL,NULL,NULL,'14e19c47-eb20-45a7-bf36-a4acc714db68',673,NULL),(2773,NULL,'The BRAF A489_P490del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely activating as measured by modestly increased kinase activity compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-01-11 23:42:11',NULL,NULL,NULL,NULL,'3a2c8706-35f2-4056-a822-6cb6135aa7f8',673,NULL),(2774,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2021-01-11 23:44:51',NULL,NULL,NULL,NULL,'6a5b6c22-8a6b-47af-b330-771367813431',673,NULL),(2775,NULL,'The BRAF T488_P490del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely neutral as measured by kinase activity comparable to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2021-01-11 23:45:05',NULL,NULL,NULL,NULL,'93d252f6-11a4-4b50-9716-2d6603567f22',673,NULL),(2776,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2021-01-11 23:48:23',NULL,NULL,NULL,NULL,'b6c74337-4d02-4a34-b5eb-018219d39c2f',673,NULL),(2777,NULL,'The BRAF V487_P490del mutation is located in the kinase domain in exon 11 of the protein. Expression of this mutation in a human cell line demonstrated that it is likely activating as measured by increased kinase activity compared to wildtype (PMID: 26996308).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2021-01-11 23:48:24',NULL,NULL,NULL,NULL,'8fba81d3-6bfd-46b2-b3c8-be2174cf5cef',673,NULL),(2778,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2021-05-05 21:00:58',NULL,NULL,NULL,NULL,'8e1b288e-93a8-45fa-8fad-90b6fda31633',673,NULL),(2779,NULL,'The BRAF A598_T599insV mutation is located in the kinase domain of the protein. While this mutation has not been functionally characterized, a patient with melanoma harboring this mutation had a partial response to treatment with the combination of dabrafenib and trametinib (PMID: 28800030).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2021-05-05 21:03:10',NULL,NULL,NULL,NULL,'5cc55276-326f-45e3-974a-fc29b92930fa',673,NULL),(2780,NULL,'While the RAF-inhibitor dabrafenib in combination with the MEK-inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E mutant non-small cell lung cancer, their clinical utility in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2017-11-27 13:23:13',NULL,NULL,NULL,NULL,'ec7c8254-cf2c-461a-810e-e8f8e452de0e',673,NULL),(2781,NULL,'While the RAF-inhibitors dabrafenib and vemurafenib, alone and in combination with the MEK-inhibitors trametinib and cobimetinib respectively, are FDA-approved for the treatment of patients with BRAF V600E and V600K mutant melanoma, their clinical utility in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2018-08-22 15:30:00',NULL,NULL,NULL,NULL,'68ce6de9-5658-4535-b52d-a0b574d3c573',673,NULL),(2782,NULL,'While the RAF-inhibitor dabrafenib in combination with the MEK-inhibitor trametinib is FDA-approved for the treatment of patients with BRAF V600E mutant melanoma, non-small cell lung cancer, and anaplastic thyroid cancer, and the RAF-inhibitor vemurafenib is FDA-approved as monotherapy in BRAF V600-mutant Erdheim-Chester disease, their clinical utility in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2019-08-22 12:21:22',NULL,NULL,NULL,NULL,'e849d451-64e6-4dc1-ac77-0071b21a22f8',673,NULL),(2783,NULL,'This assertion is supported by (PMID: 22237106).','DIAGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Dx3',NULL,NULL,'46b031e3-4ae9-418b-8b17-3a6acfd197d8',673,NULL),(2784,NULL,'The presence of a BRAF [[mutation]] [[[mutation]]] may assist in the diagnosis of [[tumor type]].','DIAGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'614bf7fb-7926-49db-9fb0-6dd00fc3f603',673,NULL),(2785,NULL,'There is promising clinical data in patients with histiocytic neoplasms harboring MAPK pathway alterations treated with the MEK-targeted inhibitors cobimetinib or trametinib as monotherapy.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-10-20 10:48:08',NULL,NULL,NULL,NULL,'c96628e0-742b-477e-8eff-c79aa725a02d',673,NULL),(2786,NULL,'Trametinib and cobimetinib are MEK1/2 kinase inhibitors that are FDA-approved for patients with unresectable or metastatic melanoma with BRAF V600E or V600K mutations and both are NCCN-listed for patients with histiocytic neoplasms harboring MAPK pathway mutations. In a case study of one eighteen-year-old patient with refractory histiocytic sarcoma harboring a pathogenic variant of MAP2K1 (F53L), the patient was treated with trametinib and showed an excellent response prior to treatment interruption due to severe adverse events (PMID: 30361829).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 14:54:24',NULL,'LEVEL_3A','NO','NO','658a788e-390c-41c6-8081-0376b6121bb0',673,NULL),(2787,NULL,'The MEK-targeted inhibitors cobimetinib and trametinib are NCCN-listed as monotherapy for the treatment of patients with Erdheim-Chester Disease with MAPK pathway alterations, including [[gene]][[mutation]].','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-10-20 10:51:43',NULL,NULL,NULL,NULL,'e95d9545-8b3b-480e-9222-bddc0869dba2',673,NULL),(2788,NULL,'Trametinib and cobimetinib are MEK1/2 kinase inhibitors that are FDA-approved for patients with unresectable or metastatic melanoma with BRAF V600E or V600K mutations and both are NCCN-listed for patients with Erdheim-Chester Disease (ECD) harboring MAPK pathway mutations. In an open-label phase II trial (NCT02649972) of cobimetinib in eighteen adult patients with histiocytoses (n = twelve patients with ECD) of any mutational status, the overall response rate was 89% (90% CI: 73100), 100% of responses were ongoing at one year, and 94% of patients remained progression-free (PMID: 30867592).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 14:54:05',NULL,'LEVEL_2','NO','NO','f85c5ffd-a7dc-4769-acdc-012ff3b7ac5d',673,NULL),(2789,NULL,'The MEK-targeted inhibitors cobimetinib and trametinib are NCCN-listed as monotherapy for the treatment of patients with Langerhans Cell Histiocytosis with MAPK pathway alterations, including [[gene]][[mutation]].','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-10-20 10:53:50',NULL,NULL,NULL,NULL,'7e539aa7-e805-4ac4-827e-a7a552199d1f',673,NULL),(2790,NULL,'Trametinib and cobimetinib are MEK1/2 kinase inhibitors that are FDA-approved for patients with unresectable or metastatic melanoma with BRAF V600E or V600K mutations and both are NCCN-listed for patients with Langerhans Cell Histiocytosis (LCH) harboring MAPK pathway mutations. In an open-label phase II trial (NCT02649972) of cobimetinib in eighteen adult patients with histiocytoses (n = two patients with LCH) of any mutational status, the overall response rate was 89% (90% CI: 73100), 100% of responses were ongoing at one year, and 94% of patients remained progression-free (PMID: 30867592). In a study of three patients with LCH who were treated with trametinib, two patients harbored BRAF N486_P490del mutations, of which one had no reactivation and the other had a partial response, and the third patient with a MAP2K1 K57_G61del mutation had a complete response with no active disease after 22 months (PMID: 32991018).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 14:54:40',NULL,'LEVEL_2','NO','NO','a7ce9682-0287-4744-895c-69df4bb7792b',673,NULL),(2791,NULL,'The MEK-targeted inhibitors cobimetinib and trametinib are NCCN-listed as monotherapy for the treatment of patients with Rosai-Dorfman Disease with MAPK pathway alterations, including [[gene]][[mutation]].','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-10-20 10:55:24',NULL,NULL,NULL,NULL,'67779f76-43de-4ac6-8b31-27064c3ab215',673,NULL),(2792,NULL,'Trametinib and cobimetinib are MEK1/2 kinase inhibitors that are FDA-approved for patients with unresectable or metastatic melanoma with BRAF V600E or V600K mutations and are NCCN-listed for patients with Rosai-Dorfman Disease. One adult patient with Rosai-Dorfman Disease with an activating KRAS G12R mutation was treated with cobimetinib, and a follow-up CT scan two months after treatment showed a substantial response (PMID: 29236635).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2023-02-15 14:55:00',NULL,'LEVEL_2','NO','NO','6680f049-0c87-41fb-a234-09c1a2361419',673,NULL),(2793,NULL,'The RAF-targeted inhibitors encorafenib, dabrafenib and vemurafenib, alone or in combination with the MEK1/2-targeted inhibitors binimetinib, trametinib and cobimetinib respectively, are FDA-approved for the treatment of patients with BRAF V600E/K mutant melanoma and NCCN-compendium listed for the treatment of patients with BRAF V600-mutant melanoma. Additionally, the anti-PD-L1 antibody atezolizumab in combination with cobimetinib + vemurafenib is FDA-approved for the treatment of patients with unresectable or advanced BRAF V600-mutant melanoma.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2021-11-18 16:48:24',NULL,NULL,NULL,NULL,'6cbb7fd3-76a2-4d64-8332-bdb68e972b03',673,NULL),(2794,NULL,'The combination of encorafenib, an inhibitor of V600E- or V600K-mutant BRAF, and binimetinib, an inhibitor of MEK1/2, is FDA-approved in combination for patients with unresectable or metastatic melanoma with a BRAF V600E or V600K mutation. FDA approval was based on the results of the Phase III COLUMBUS trial of combined encorafenib plus binimetinib versus single agent vemurafenib in 577 patients with BRAF V600E- or V600K-mutant metastatic melanoma in which the median progression-free survival was 14.9 months (95% CI: 11.0-18.5) in the encorafenib plus binimetinib group versus 7.3 months (95% CI: 5.6-8.2) in the single agent vemurafenib group (HR= 0.54, 95% CI: 0.41-0.71; p<00001) (PMID: 29573941).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 14:57:31',NULL,'LEVEL_2','NO','LEVEL_3B','009d1441-ac80-4048-ae01-371ac1591c33',673,NULL),(2795,NULL,'Dabrafenib, an orally bioavailable RAF inhibitor, and trametinib, an orally bioavailable MEK1/2 inhibitor, are FDA-approved alone or in combination for the treatment of patients with metastatic melanoma harboring a V600E or V600K BRAF mutation. FDA approval of dabrafenib in combination with trametinib was based on results from an open-label Phase III study of combination therapy versus dabrafenib monotherapy in 247 patients with metastatic melanoma who were naive to treatment with BRAF inhibitors. Combined dabrafenib and trametinib, administered in full monotherapy doses, improved the response rate in patients with BRAF V600-mutant metastatic melanoma versus dabrafenib monotherapy (67% vs.51%; p<0.002). However, median progression-free survival improved by only two weeks (9.3 months vs 8.8 months; HR = 0.75) compared with dabrafenib monotherapy (PMID: 23020132). Combination therapy is associated with less cutaneous toxicity than monotherapy, but systemic toxicity may be increased (PMID: 25287827, 25399551). Follow-up trials have demonstrated that all clinical measures inclusive of overall and median progression-free survival as well as objective response rates, median duration of response and number of patients with complete response favored patients treated with combination dabrafenib and trametinib, administered in full monotherapy doses, compared to either dabrafenib or vemurafenib monotherapy, including patients who previously progressed on BRAF inhibitor monotherapy (PMID: 25287827, 25399551, 25265492). Additionally, patients with melanoma treated with dabrafenib and trametinib in both the neoadjuvant and adjuvant settings have improved survival over patients given standard of care (PMID: 29361468, 28991513, 28891408). Promising clinical data has also suggested that addition of an immunotherapy agent to combination RAF and MEK inhibitor treatment may improve rate of overall response and duration of response in melanoma patients (PMID: 31171876, 31171879, 31171878).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 14:57:46',NULL,'LEVEL_2','NO','LEVEL_3B','55f1261c-2f2c-4537-aa81-92adcb798585',673,NULL),(2796,NULL,'The combination of vemurafenib, an orally available kinse inhibitor of V600-mutant BRAF, and cobimetinib, an orally available kinase inhibitor of MEK1/2, is FDA-approved for the treatment of patients with BRAF V600-mutant metastatic or unresectable locally advanced melanoma. FDA approval was based on results from the randomized Phase III coBRIM trial of combined vemurafenib and cobimetinib versus vemurafenib and placebo in 495 patients with metastatic BRAF V600-mutant melanoma that demonstrated superior clinical benefit measures in the combination versus the control arm. Specifically, median progression-free survival was 9.9 months in the combination arm versus 6.2 months in the control arm (HR = 0.51), with a complete response rate of 10% versus 4%, respectively, and interim nine-month overall survival of 81% versus 73%, respectively (PMID: 25265494). Follow-up analysis of the coBRIM trial showed two-year overall survival was 48.3% in the combination group versus 38.0% in the monotherapy group (PMID: 27480103).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-15 14:58:00',NULL,'LEVEL_2','NO','LEVEL_3B','f9aaeb93-ca4e-49cc-8424-792f49ce9006',673,NULL),(2797,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-24 14:08:36',NULL,NULL,NULL,NULL,'2d9f2a61-bb25-4bb2-ba0d-6ebe58fd7f6c',673,NULL),(2798,NULL,'The p61BRAF(V600E) mutation is the product of alternative splicing of the BRAF mRNA transcript, which contains the V600E mutation but eliminates exons 4-8 of the gene (PMID: 22113612). This mutation encodes a truncated, 16 kDa BRAF protein that has the V600E mutation but lacks the RAS binding domain and constitutively dimerizes (PMID: 22113612). This mutation has been found in melanoma (PMID: 22113612). In concordance with its ability to constitutively dimerize, in vitro studies of this mutation suggest that it is resistant to vemurafenib, as measured by persistent MAPK pathway signaling and cell proliferation upon drug treatment (PMID: 22113612). In one study, six out of nineteen patients with melanomas that were resistant to vemurafenib harbored the p61BRAF(V600E) mutation (PMID: 22113612). In a second study, BRAF splice variants were found in several patient-derived xenografts that were generated from patients who progressed on BRAF inhibitors (PMID: 27320919).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2022-05-16 15:33:17',NULL,NULL,NULL,NULL,'540b58a0-be50-41d4-9241-ce4d8b597398',673,NULL),(2799,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-20 16:12:18',NULL,NULL,NULL,NULL,'753b067a-2371-47dd-aff5-722c63c023e0',673,NULL),(2800,NULL,'The BRAF I592_A598dup mutation is located in the kinase domain in exon 15 of the protein. This mutation has been identified in papillary thyroid carcinoma, inguinal adenopathy and primary melanoma, and is a statistically significant hotspot (PMID: 27571181, 28800030 ). This mutation has not been functionally characterized in vitro. BRAF exon 15 mutations are sensitive to treatment with combination MAPK-directed targeted therapy based on the clinical benefit of two patients with rare BRAF exon 15 insertion mutations involving the A598 residue, BRAF A598_T599insV and V600_K601delinsE (PMID: 28800030).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2022-12-20 16:12:22',NULL,NULL,NULL,NULL,'e094703b-28dd-45a1-a8c6-8f3f018fa030',673,NULL),(2801,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2024-03-12 10:30:48',NULL,NULL,NULL,NULL,'62fd5ccd-0491-4148-af9e-48bbc95dfd01',673,NULL),(2802,NULL,'The BRAF L485W mutation is located in the kinase domain of the protein (PMID: 35117215). This mutation has been identified in bladder adenocarcinoma and is a statistically significant hotspot (PMID: 28481359). The oncogenic and biologic effect of this mutation has conflicting evidence. An in vitro study with Ba/F3 and MCF10A cells expressing BRAF L485W demonstrates the mutation is inactivating as measured by decreased cellular viability in one cell line compared to wildtype (PMID: 29533785). In contrast, BRAF L485W has been cited as an activating class II BRAF mutation and is suggested to be an atypical activating mutation due to sensitivity to MAPK targeting inhibitors (PMID: 35977349, 35117215, 31015311). Clinical studies suggest that BRAF L485W is sensitive to MAPK pathway inhibition as measured by a patient with gallbladder cancer harboring the mutation demonstrating sensitivity to the ERK1/2 kinase inhibitor ulixertinib as measured by a year-long complete response to treatment (PMID: 29247021).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2024-03-12 16:29:08',NULL,NULL,NULL,NULL,'8322f1de-86af-4e6f-8a96-b05f246d1b16',673,NULL),(2803,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:54:38',NULL,NULL,NULL,NULL,'888eef19-2da3-4810-94b0-5f2e48293460',673,NULL),(2804,NULL,'The class II BRAF G469S mutation is located in the kinase domain in exon 11 of the protein (PMID: 31470866). This mutation has been identified in malignant melanoma and is a statistically significant hotspot (PMID: 16096377). In vitro studies with HEK293T cells expressing BRAF G469S demonstrate that the mutation is activating as measured by increased phosphorylated MEK and ERK levels compared to wildtype (PMID: 28947956). In a case report, a patient with metastatic melanoma harboring BRAF G469S was treated with a combination therapy of trametinib plus dabrafenib and demonstrated stable disease with a progression-free survival of three months (PMID: 31569065). In a Phase II trial, a patient with melanoma harboring BRAF G469S was treated with combination therapy of binimetinib and encorafenib and demonstrated an unconfirmed partial response (Abstract: Rose et al. Abstract# 531P, Annals of Oncology 2021. https://www.annalsofoncology.org/article/S0923-7534%2821%2903282-8/fulltext).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-15 14:47:02',NULL,NULL,NULL,NULL,'926f93dc-c93f-42cb-b07d-fef65da7985a',673,NULL),(2805,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-01 16:59:25',NULL,NULL,NULL,NULL,'3c8deca4-ab8f-4724-8503-d513c1d9080b',673,NULL),(2806,NULL,'The class III BRAF N581T mutation is located in the kinase domain of the protein (PMID: 31515458). This mutation has been identified in diffuse large B-cell lymphoma and is a statistically significant hotspot (PMID: 32187361). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF N581T demonstrate that the mutation is likely neutral as measured by cell viability and proliferation comparable to wildtype (PMID: 29533785). In a case report, a patient with metastatic colorectal cancer harboring BRAF N581T was treated with a combination therapy of cetuximab plus FOLFOX and demonstrated partial response with a progression-free survival of 1.4 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-08 16:02:42',NULL,NULL,NULL,NULL,'7d9ab560-2699-413a-9758-d8faab353cd4',673,NULL),(2807,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:39:47',NULL,NULL,NULL,NULL,'19aa6810-e3ad-4768-bcd1-eb3e988abba1',673,NULL),(2808,NULL,'The class III BRAF G596S mutation is located in the kinase domain of the protein and is a statistically significant hotspot (PMID: 33198372). In vitro studies with Ba/F3 cells expressing BRAF G596S demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785). In silico analysis suggests that BRAF G596S is resistant to vemurafenib (PMID: 35215249).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-08 14:36:27',NULL,NULL,NULL,NULL,'776007fd-a78b-4c5d-8aa2-4d557760ffac',673,NULL),(2809,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-28 09:48:49',NULL,NULL,NULL,NULL,'3cc907ca-954a-4ed8-89d2-4b5207ed1077',673,NULL),(2810,NULL,'The BRAF G606W mutation is located in the kinase domain of the protein. This mutation has been identified in lung adenocarcinoma (PMID: 28481359). In vitro studies with MCF10A cells expressing BRAF G606W demonstrate that the mutation is inactivating as measured by decreased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-03-28 09:49:02',NULL,NULL,NULL,NULL,'9eefa994-e657-43db-93fb-f9ce51268979',673,NULL),(2811,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-28 09:49:24',NULL,NULL,NULL,NULL,'5e9d52ed-5c92-45e2-a5d2-f09983848227',673,NULL),(2812,NULL,'The BRAF A762V mutation is located in an unknown functional domain of the protein. This mutation has been identified in colorectal cancer (PMID: 29316426). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF A762V demonstrate that the mutation is likely neutral as measured by transformation and proliferation comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-28 09:49:31',NULL,NULL,NULL,NULL,'79f3270f-b928-4965-b727-bdd74b0490eb',673,NULL),(2813,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Resistance','2023-03-28 09:49:59',NULL,NULL,NULL,NULL,'1229dd70-6fb5-4bbe-9d3e-a48b364433c8',673,NULL),(2814,NULL,'The BRAF R682W mutation is located in the kinase domain of the protein. This mutation has been identified in lung adenocarcinoma (PMID: 22980975). Preclinical studies with PC9 cells expressing BRAF R682W and EGFR exon 19 in-frame deletion mutations demonstrated that the mutation is resistant to treatment with erlotinib as measured by sustained cellular viability following treatment (PMID: 27478040).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-28 09:50:12',NULL,NULL,NULL,NULL,'b4690794-43f1-4d4e-a7f1-4a916b592fe0',673,NULL),(2815,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-28 09:50:40',NULL,NULL,NULL,NULL,'38e74d85-b24a-4938-acd7-b1c857b16a9c',673,NULL),(2816,NULL,'The BRAF P367S mutation is located in an unknown functional domain of the protein. This mutation has been identified in melanoma (PMID: 24265153). In vitro studies with Ba/F3 cells expressing BRAF P367S demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-28 09:50:46',NULL,NULL,NULL,NULL,'91da86fe-2f02-4d1d-a26a-84d8f52f049b',673,NULL),(2817,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-28 09:51:16',NULL,NULL,NULL,NULL,'23c14370-2a01-46a1-9120-a04fa88e507f',673,NULL),(2818,NULL,'The BRAF A33T mutation is located in an unknown functional domain of the protein. This mutation has been identified in colorectal cancer (PMID: 28481359). In vitro studies with mouse embryonic fibroblasts expressing BRAF A33T demonstrate that the mutation is likely neutral as measured by ERK activation comparable to wildtype (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-28 09:51:21',NULL,NULL,NULL,NULL,'704334e4-f538-4143-a116-dbce48de5258',673,NULL),(2819,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-28 09:51:46',NULL,NULL,NULL,NULL,'ea70fe95-5c10-480b-9fde-1b4c0408f6db',673,NULL),(2820,NULL,'The BRAF R271H mutation is located in the zinc-finger domain of the protein. This mutation has been identified in prostate cancer (PMID: 28825054). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF R271H demonstrate that the mutation is likely neutral as measured by cellular proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-28 09:51:49',NULL,NULL,NULL,NULL,'7f946eea-0e70-4ba0-8599-1c4599a1d9f5',673,NULL),(2821,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-28 09:52:14',NULL,NULL,NULL,NULL,'b93338b9-4378-4b67-8331-97afe3c46559',673,NULL),(2822,NULL,'The BRAF H725Y mutation is located in an unknown functional domain of the protein. This mutation has been identified in merkel cell carcinoma (PMID: 28481359). In vitro studies with MCF10A cells expressing BRAF H725Y demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-28 09:52:16',NULL,NULL,NULL,NULL,'ea3019e5-f0bf-40b0-b577-7d467486d2b8',673,NULL),(2823,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-28 09:52:31',NULL,NULL,NULL,NULL,'5156f3f4-6113-4df7-a5a4-c3bf3fd5e364',673,NULL),(2824,NULL,'The BRAF F294L mutation is located in an unknown functional domain of the protein. This mutation has been identified in colorectal cancer (PMID: 28481359). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF F294L demonstrate that the mutation is likely neutral as measured by cellular proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-28 09:52:38',NULL,NULL,NULL,NULL,'0bedcc6e-0b8d-4096-bbbf-f54063466e6f',673,NULL),(2825,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-11 18:08:19',NULL,NULL,NULL,NULL,'779a250f-390d-438d-95e3-403d9832886b',673,NULL),(2826,NULL,'The BRAF-PJA2 fusion protein results from exon 7 of PJA2 fusing with exon 11 of BRAF, retaining the BRAF kinase domain. This fusion has been identified in EGFR-mutant lung cancer (PMID: 30831205). Preclinical studies with H1975 and PC9 cells expressing BRAF-PJA2 demonstrated the fusion is resistant to treatment with osimertinib as measured by sustained cell viability following treatment (PMID: 30831205).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-11 18:08:20',NULL,NULL,NULL,NULL,'1aff8c74-2114-4c5c-86d2-e670c01eea63',673,NULL),(2827,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-29 13:10:11',NULL,NULL,NULL,NULL,'b4256573-9ddb-44b0-83ff-a8d1fe2d5706',673,NULL),(2828,NULL,'The BRAF I572F mutation is located in the kinase domain of the protein. This mutation has been identified in lung adenocarcinoma (PMID: 28481359). Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF I572F and KRAS G12V was treated with nivolumab and demonstrated progressive disease (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-29 13:10:18',NULL,NULL,NULL,NULL,'eeefc97d-4142-4f19-af7c-26048155a361',673,NULL),(2829,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-29 13:10:46',NULL,NULL,NULL,NULL,'6bb9b4e8-ed04-4c42-a793-432f4ee97299',673,NULL),(2830,NULL,'The BRAF M620I mutation is located in the kinase domain of the protein. This mutation has been identified in lung adenocarcinoma (PMID: 28481359). In silico analysis suggests that the mutation is activating and damaging (PMID: 34588906).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-29 13:10:51',NULL,NULL,NULL,NULL,'e2872a4c-bc9c-4349-979b-6b04b594df08',673,NULL),(2831,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-29 13:11:06',NULL,NULL,NULL,NULL,'4f26b6ee-0619-4a3e-8e4e-7a21baded348',673,NULL),(2832,NULL,'The BRAF P162S mutation is located in the RBD domain of the protein. This mutation has been identified in biliary tract cancer (PMID: 29360550). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF P162S demonstrate that the mutation is likely neutral as measured by cellular proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-06 14:23:03',NULL,NULL,NULL,NULL,'69cec9f1-2fa9-4d15-9fa2-3a854debe177',673,NULL),(2833,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-29 15:34:24',NULL,NULL,NULL,NULL,'86548f98-9509-40e8-9184-72020ba09c77',673,NULL),(2834,NULL,'The BRAF R239Q mutation is located in the zinc-finger domain of the protein. This mutation has been identified in melanoma (PMID: 28481359). Although not functionally characterized per se, preclinical studies with patient-derived melanoma cell lines harboring BRAF R239Q and BRAF L597S demonstrated sensitivity to treatment with binimetinib plus encorafenib, dabrafenib plus trametinib, single agent encorafenib and single agent binimetinib as measured by reduced cellular proliferation and tumor growth (PMID: 29903896).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-29 15:34:30',NULL,NULL,NULL,NULL,'9e1a9d85-4299-4e5d-b22b-a00933dd4196',673,NULL),(2835,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-29 15:34:51',NULL,NULL,NULL,NULL,'ff3ab9a1-702c-46c8-b759-4d3a5e00947e',673,NULL),(2836,NULL,'The class III BRAF R558Q mutation is located in the kinase domain of the protein (PMID: 31515458). This mutation has been identified in colorectal cancer (PMID: 28481359). In vitro studies with mouse embryonic fibroblasts expressing BRAF R558Q demonstrate that the mutation is activating as measured by increased ERK signaling compared to wildtype (PMID: 31515458). In a clinical trial, a patient with metastatic colorectal cancer harboring BRAF R558Q was treated with combination cetuximab plus irinotecan and demonstrated complete response with a progression-free survival of 37.7 months (PMID: 31515458).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-29 19:41:01',NULL,NULL,NULL,NULL,'40f7f0ec-543e-4254-b80a-713689ecf867',673,NULL),(2837,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-29 15:35:25',NULL,NULL,NULL,NULL,'6f2f0aef-d53a-4ab9-bc1b-6aca75ba711e',673,NULL),(2838,NULL,'The BRAF S363F mutation is located in an unknown functional domain of the protein. This mutation has been identified in melanoma (PMID: 28467829). Although not functionally characterized per se, preclinical studies with melanoma cell lines harboring BRAF S363F and BRAF K601E demonstrated sensitivity to treatment with binimetinib plus encorafenib, dabrafenib plus trametinib, trametinib plus vemurafenib, single agent encorafenib, single agent binimetinib, single agent trametinib and single agent cobimetinib as measured by reduced cellular proliferation (PMID: 29903896).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-29 15:35:30',NULL,NULL,NULL,NULL,'85f1e474-daa3-459e-9380-8e69a4441351',673,NULL),(2839,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-29 15:36:02',NULL,NULL,NULL,NULL,'8fa75258-b5ed-4a43-aba3-cc776ee00c5a',673,NULL),(2840,NULL,'The BRAF T310I mutation is located in an unknown functional domain of the protein. This mutation has been identified in glioma (PMID: 18772396). In vitro studies with SkBr3 cells expressing BRAF T310I demonstrate that the mutation is activating as measured by an increase in ERK phosphorylation compared to wildtype (PMID: 31158244). Preclinical studies with SkBr3 cells expressing this mutation demonstrate resistance to treatment with dabrafenib as measured by sustained ERK phosphorylation (PMID: 31158244).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-29 15:36:10',NULL,NULL,NULL,NULL,'aa4c01bb-ff19-4975-8d6c-2fc5047418a7',673,NULL),(2841,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Resistance','2023-03-29 15:36:35',NULL,NULL,NULL,NULL,'b5152b18-7699-4da6-9902-f83472ec5b55',673,NULL),(2842,NULL,'The BRAF L485Y mutation is located in the kinase domain of the protein. In vitro studies with HT29 cells expressing BRAF L485Y demonstrated resistance to treatment with GDC0879 as measured by sustained cellular proliferation following treatment (PMID: 19276360).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-30 12:02:36',NULL,NULL,NULL,NULL,'66e6caab-ad7f-4b9c-9a6f-cbed104e7823',673,NULL),(2843,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Resistance','2023-03-29 15:37:03',NULL,NULL,NULL,NULL,'65d99fa1-11e8-4693-bbbf-cea4ad0df770',673,NULL),(2844,NULL,'The BRAF L597P mutation is located in the kinase domain of the protein. This mutation has been identified in melanoma (PMID: 23890154). In vitro studies with patient-derived intrahepatic cholangiocarcinoma organoids expressing BRAF L597P demonstrated resistance to treatment with lenvatinib, temuterkib and sunitinib as measured by sustained cellular proliferation, and sensitivity to treatment with trametinib and cobimetinib, as measured by inhibition of cellular proliferation (PMID: 36867406).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-06 16:27:32',NULL,NULL,NULL,NULL,'69394227-ec1b-4915-8c3d-a7376f871c59',673,NULL),(2845,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-29 15:37:34',NULL,NULL,NULL,NULL,'a7379224-dad4-4f69-b3d2-eee291ab022c',673,NULL),(2846,NULL,'The BRAF V600A mutation (also known as V599A) is located in the activation segment of the kinase domain of the protein. This mutation has been identified in prostate cancer (PMID: 16721785). In vitro studies with HEK293 cells expressing BRAF V600A demonstrate that the mutation is likely neutral as measured by phosphorylation comparable to wildtype (PMID: 26744778).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-30 11:39:00',NULL,NULL,NULL,NULL,'c810a2c1-f5f4-45fb-9a9a-0251be778f03',673,NULL),(2847,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-29 15:37:52',NULL,NULL,NULL,NULL,'d46d6559-30db-46b1-8f06-9f80a184f35f',673,NULL),(2848,NULL,'The BRAF V600L mutation (also known as V599L) is located in the activation segment of the kinase domain of the protein. This mutation has been identified in colorectal cancer (PMID: 22586484). In vitro studies with MCF10A cells expressing BRAF V600L demonstrate that the mutation is activating as measured by increased cell proliferation and viability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-30 11:38:55',NULL,NULL,NULL,NULL,'6be2d4e5-848a-456c-a5bd-951bac12dc4f',673,NULL),(2849,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-30 11:56:51',NULL,NULL,NULL,NULL,'717e67f2-67aa-45b0-9201-be2a6d843eff',673,NULL),(2850,NULL,'The BRAF M117L mutation is located in an unknown functional domain of the protein. This mutation has been identified in lung adenocarcinoma (PMID: 28481359). Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF M117L was treated with ipilimumab plus nivolumab and demonstrated stable disease (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-06 16:30:34',NULL,NULL,NULL,NULL,'1b15b026-9ed1-40f0-82ae-8c44c62efcdf',673,NULL),(2851,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-30 11:57:24',NULL,NULL,NULL,NULL,'edefd28d-d764-4fca-82d2-46649b96ff5c',673,NULL),(2852,NULL,'The BRAF N412S mutation is located in an unknown functional domain of the protein. This mutation has been identified in lung adenocarcinoma (PMID: 28481359). Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF N412S was treated with nivolumab and demonstrated stable disease (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-30 11:57:32',NULL,NULL,NULL,NULL,'1ac32a66-825d-4a32-9a45-ae8626a2d5db',673,NULL),(2853,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-30 11:57:56',NULL,NULL,NULL,NULL,'0a5685f6-377d-4252-9c21-ea951c58d39f',673,NULL),(2854,NULL,'The BRAF S616Y mutation is located in the kinase domain of the protein. Although not functionally characterized per se, a patient with squamous cell carcinoma harboring BRAF S616Y was treated with nivolumab and demonstrated a complete response (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-06 16:31:22',NULL,NULL,NULL,NULL,'ace3ae6e-78bb-4efb-9f3d-4ea4ce34dfbf',673,NULL),(2855,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-30 11:58:33',NULL,NULL,NULL,NULL,'5d020698-93b7-40aa-8996-48d038b97f29',673,NULL),(2856,NULL,'The BRAF T274A mutation is located in the zinc finger domain of the protein. Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF T274A, KRAS G12D and mutant NF1 was treated with nivolumab and demonstrated progressive disease (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-30 11:58:36',NULL,NULL,NULL,NULL,'1a45db61-f9a1-47c2-8269-b59db2ea70d2',673,NULL),(2857,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-30 11:59:02',NULL,NULL,NULL,NULL,'3d789366-9270-403b-9ae9-0632cb3f21d6',673,NULL),(2858,NULL,'The BRAF S607F mutation is located in the kinase domain of the protein. This mutation has been identified in basal cell carcinoma (PMID: 26950094). In vitro studies with HEK293 cells expressing BRAF S607F demonstrate that the mutation is likely neutral as measured by ERK phosphorylation comparable to wildtype (PMID: 32059434).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-30 11:59:19',NULL,NULL,NULL,NULL,'a8ceb704-0924-4eba-a22f-4965fedf34a9',673,NULL),(2859,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-30 11:59:45',NULL,NULL,NULL,NULL,'add98c56-340e-45a8-9e9d-3e6cf140fc97',673,NULL),(2860,NULL,'The BRAF M517V mutation is located in the kinase domain of the protein. In silico analysis suggests that the mutation contributes to resistance against vemurafenib (PMID: 35215249).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-30 11:59:50',NULL,NULL,NULL,NULL,'892cb6e5-481c-4f59-960e-62b334192d44',673,NULL),(2861,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-30 12:00:10',NULL,NULL,NULL,NULL,'59b9d7dd-aa66-4ece-8c2e-66b8fa53f7fa',673,NULL),(2862,NULL,'The BRAF W604C mutation is located in the kinase domain of the protein. This mutation has been identified in melanoma (PMID: 28467829). Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF W604C and BRAF G469A was treated with dabrafenib plus trametinib and demonstrated stable disease with a progression-free survival of fifteen months (PMID: 30926357).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-06 16:47:09',NULL,NULL,NULL,NULL,'2e8a7d43-8453-4d53-9337-48e395b4a216',673,NULL),(2863,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-03-30 12:00:42',NULL,NULL,NULL,NULL,'27e36595-b7bd-4f06-a13b-2dbf005329dc',673,NULL),(2864,NULL,'The BRAF R389C mutation is located in an unknown functional domain of the protein. This mutation has been identified in colorectal cancer (PMID: 25344691). In vitro studies with Ba/F3 and MCF10A cells expressing BRAF R389C demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-03-30 12:00:46',NULL,NULL,NULL,NULL,'a7793917-0857-4385-ba53-00a84894c5cb',673,NULL),(2865,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-30 12:01:09',NULL,NULL,NULL,NULL,'64ce0979-890a-451a-86c2-22a3d58f0141',673,NULL),(2866,NULL,'The BRAF E451K mutation is located in an unknown functional domain of the protein. In vitro studies with SkBr3 cells expressing BRAF E451K demonstrate that the mutation is activating as measured by increased ERK phosphorylation compared to wildtype (PMID: 31158244). Preclinical studies with SkBr3 cells expressing this mutation demonstrate that the mutation is resistant to dabrafenib as measured by sustained ERK phosphorylation following treatment (PMID: 31158244).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-30 12:01:13',NULL,NULL,NULL,NULL,'313d8c8b-8f58-47fb-882c-3ae17edb3cfb',673,NULL),(2867,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-31 12:00:39',NULL,NULL,NULL,NULL,'47f68c6a-eb88-4b54-a2cf-bdcae474eceb',673,NULL),(2868,NULL,'The BRAF S122Y mutation is located in the SBC motif of the protein. This mutation has been identified in osteosarcoma (PMID: 28481359). In vitro studies with 293T cells expressing BRAF S122Y demonstrate that the mutation is activating as measured by reduced SPOP-mediated ubiquitination and increased MAPK/ERK cascade activation compared to wildtype (PMID: 36585710).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-03-31 12:00:41',NULL,NULL,NULL,NULL,'97510f77-deb4-4388-b0c2-df32cea34d23',673,NULL),(2869,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-03-31 12:01:16',NULL,NULL,NULL,NULL,'7cb50114-6671-44dd-8bc2-46e64960b0e7',673,NULL),(2870,NULL,'The BRAF R239L mutation is located in the zinc finger domain of the protein. Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF R239L and KRAS G12C was treated with nivolumab and demonstrated stable disease (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-03-31 12:01:21',NULL,NULL,NULL,NULL,'92b6ade4-8ffa-4a8d-bdda-989bd7573b38',673,NULL),(2871,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-03-31 12:02:27',NULL,NULL,NULL,NULL,'5577f1a6-a3b2-480a-96d1-1954ba542c30',673,NULL),(2872,NULL,'The ZNF767-BRAF fusion results from the fusion of exon 1 of ZNF767 with exon 11 of BRAF. In vivo studies with nude mice expressing ZNF767-BRAF demonstrate that the fusion is activating as measured by rapid tumor growth compared to wildtype (PMID: 28092667). Preclinical studies with PMM cells expressing ZNF767-BRAF demonstrated sensitivity to treatment with trametinib, BKM120, LEE011, trametinib plus BKM120 and trametinib plus LEE011 as measured by inhibition of cell growth (PMID: 28092667). A patient with melanoma harboring ZNF767-BRAF was treated with vemurafenib and demonstrated resistance, with rapid progression within two weeks of treatment (PMID: 28092667).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-06 16:48:36',NULL,NULL,NULL,NULL,'446c9369-f5ff-4924-a94e-35ff64a2b23c',673,NULL),(2873,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-06 16:49:26',NULL,NULL,NULL,NULL,'c7d1f27a-b106-4f21-a877-c7c7e9eda827',673,NULL),(2874,NULL,'The JHDM1D-BRAF fusion is a chromosomal rearrangement fusing exon 13 of JHDM1D and exon 4 of BRAF, resulting in the loss of part of the RAS-binding domain of RAF. In vitro studies with AZD-R cells expressing JHDM1D-BRAF demonstrate that the fusion is resistant to treatment with FGFR inhibitors AZD4547, NVP-BGJ398 and PD173074 as measured by sustained cellular proliferation following treatment (PMID: 30045926). Preclinical studies with AZD-R cells expressing this fusion demonstrated sensitivity to treatment with MEK inhibitors CH4987655, trametinib and PD0325901 as measured by decreased cell viability following treatment (PMID: 30045926).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-06 16:49:02',NULL,NULL,NULL,NULL,'d8f04f71-4c59-4b7a-b55d-5752bc0f6dbf',673,NULL),(2875,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-03 12:58:07',NULL,NULL,NULL,NULL,'35532f7e-0122-40a1-b016-cede13efc773',673,NULL),(2876,NULL,'The BRAF E46K mutation is located in the N-terminus of the protein. In vitro studies with 293FT cells expressing BRAF E46K demonstrate that the mutation is activating as measured by increased binding to HRAS compared to wildtype (PMID: 31606273).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-03 12:58:12',NULL,NULL,NULL,NULL,'30292363-1ec1-4b7c-873b-12cc8a629507',673,NULL),(2877,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 12:58:41',NULL,NULL,NULL,NULL,'73c87c18-45b8-43fd-abf0-bfa4411653a0',673,NULL),(2878,NULL,'The BRAF N49I mutation is located in the N-terminus of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF N49I demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 12:58:47',NULL,NULL,NULL,NULL,'0120a776-b0f3-48f7-a447-adcaccb19b7d',673,NULL),(2879,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 12:59:00',NULL,NULL,NULL,NULL,'a8590040-ef77-4f63-99db-a8b6e072331f',673,NULL),(2880,NULL,'The BRAF L64I mutation is located in the N-terminus of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF L64I demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 12:59:04',NULL,NULL,NULL,NULL,'cec3efcf-1a79-40fb-9054-4a3bdee7e437',673,NULL),(2881,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 14:22:09',NULL,NULL,NULL,NULL,'5a7516da-b77e-47d8-843b-1d14136e3a33',673,NULL),(2882,NULL,'The BRAF K205Q mutation is located in the RBD domain of the protein. This mutation has been identified in colorectal cancer (PMID: 22810696). In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF K205Q demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 14:22:20',NULL,NULL,NULL,NULL,'289de6f0-b37b-42d7-af58-539c44d08d7f',673,NULL),(2883,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 14:22:41',NULL,NULL,NULL,NULL,'04b0d077-b478-4ce1-a96e-d87b2e03cc3a',673,NULL),(2884,NULL,'The BRAF E228V mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF E228V demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 14:22:45',NULL,NULL,NULL,NULL,'6e81a81e-e0fe-4ed7-8647-a4f7bf5543d6',673,NULL),(2885,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-03 14:23:06',NULL,NULL,NULL,NULL,'e23a602c-edf3-4f5e-a8c1-cff49d93738b',673,NULL),(2886,NULL,'The BRAF L245F mutation is located in the zinc finger domain of the protein. In vitro studies with Ba/F3 cells expressing BRAF L245F demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-03 14:23:08',NULL,NULL,NULL,NULL,'3ae4dd07-0e7e-4e74-a3ed-caf2b539eada',673,NULL),(2887,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-03 14:24:10',NULL,NULL,NULL,NULL,'18dbbedf-a3e6-46c0-9249-daeee5495f4a',673,NULL),(2888,NULL,'The BRAF G258V mutation is located in the zinc finger domain of the protein. This mutation has been identified in glioma (PMID: 28852847). In vitro studies with Ba/F3 cells expressing BRAF G258V demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-03 14:24:15',NULL,NULL,NULL,NULL,'f3e01d78-f08a-4d1f-81a1-0a82077e0f5c',673,NULL),(2889,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 14:24:28',NULL,NULL,NULL,NULL,'2e3d64ae-4abf-4f68-99fa-5d91c113484e',673,NULL),(2890,NULL,'The BRAF F259L mutation is located in the zinc finger domain of the protein. This mutation has been identified in colorectal cancer (PMID: 28002797). In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF F259L demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 14:24:32',NULL,NULL,NULL,NULL,'f95c0b41-d824-4460-bcf6-f0464224b306',673,NULL),(2891,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:48:24',NULL,NULL,NULL,NULL,'b63943d4-3452-4128-baa7-f3d9d6afe98d',673,NULL),(2892,NULL,'The BRAF A320T mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF A320T demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:48:28',NULL,NULL,NULL,NULL,'8cb9844c-02e4-4b8b-8051-a9d340fc54a9',673,NULL),(2893,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:48:52',NULL,NULL,NULL,NULL,'613d6701-bae5-4ba2-b3df-c662498abd1a',673,NULL),(2894,NULL,'The BRAF P348T mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF P348T demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:48:54',NULL,NULL,NULL,NULL,'2724078f-4cda-4416-b014-91590a7e45c1',673,NULL),(2895,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:49:16',NULL,NULL,NULL,NULL,'a440d444-9a2c-434b-a496-b2dbc76d0c03',673,NULL),(2896,NULL,'The BRAF S364L mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF S364L demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:49:18',NULL,NULL,NULL,NULL,'a5671214-c40b-44b3-bbd5-9b460b4a928a',673,NULL),(2897,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:49:36',NULL,NULL,NULL,NULL,'e8d796e0-97d4-4f49-b3e7-ca9ffaa8d2e3',673,NULL),(2898,NULL,'The BRAF D380H mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF D380H demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:49:40',NULL,NULL,NULL,NULL,'7100d4a2-9bc3-4e0c-8450-b02a85a7212b',673,NULL),(2899,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-04-03 15:49:53',NULL,NULL,NULL,NULL,'5ddde0ec-af7b-4206-95e1-9c9f59b50f6a',673,NULL),(2900,NULL,'The BRAF X380_splice mutation is located in an unknown functional domain of the protein. Although not functionally characterized per se, a patient with lung adenocarcinoma harboring BRAF X380_splice and KRAS G12C was treated with atezolizumab and demonstrated progressive disease (PMID: 34963703).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-03 15:49:57',NULL,NULL,NULL,NULL,'065529f8-03d6-49c2-a4e8-fff568778d6b',673,NULL),(2901,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:50:54',NULL,NULL,NULL,NULL,'4ecc4b2c-5e36-4b89-8aaa-2b1cf604caac',673,NULL),(2902,NULL,'The BRAF T401I mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF T401I demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:50:56',NULL,NULL,NULL,NULL,'4723a368-e050-4086-8338-7ff38eb3df87',673,NULL),(2903,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:51:11',NULL,NULL,NULL,NULL,'917a7fc9-c634-42e1-a54e-6fdc3f15d893',673,NULL),(2904,NULL,'The BRAF S419Y mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF S419Y demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:51:16',NULL,NULL,NULL,NULL,'e41fd2ef-9ae7-4528-bc2f-5f1f3ef43492',673,NULL),(2905,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 15:51:30',NULL,NULL,NULL,NULL,'2fab0ff2-e1c7-4ca3-aa7a-cd6e23a05f2e',673,NULL),(2906,NULL,'The BRAF G421V mutation is located in an unknown functional domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF G421V demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 15:51:32',NULL,NULL,NULL,NULL,'41452405-532f-4132-9e1a-e75ffe7291ca',673,NULL),(2907,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-04-11 18:28:06',NULL,NULL,NULL,NULL,'b8f89baa-21b0-459c-8b48-f5bb084e0ca2',673,NULL),(2908,NULL,'The BRAF H568D mutation is located in the kinase domain of the protein. In vitro studies with HEK293T cells expressing BRAF H568D demonstrate that the mutation is inactivating as measured by a loss of kinase activity compared to wildtype (PMID: 29666306). BRAF mutations with low kinase activity or those that are kinase-dead are Class III mutations and can possibly activate CRAF as a scaffold, resulting in increased MEK-regulated signaling (PMID: 20141835).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-11 18:28:06',NULL,NULL,NULL,NULL,'bf2c98d7-eb95-4ffb-be4f-fb0b87204cd2',673,NULL),(2909,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-03 16:53:55',NULL,NULL,NULL,NULL,'f557e568-9d8a-41eb-9ef3-c6409db39b7e',673,NULL),(2910,NULL,'The BRAF L613F mutation is located in the zinc finger domain of the protein. In vitro studies with MCF10A cells expressing BRAF L613F demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-03 16:53:59',NULL,NULL,NULL,NULL,'add0243b-99de-41c7-b712-c8d73631f389',673,NULL),(2911,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 16:54:21',NULL,NULL,NULL,NULL,'293674ed-e09c-44cb-b934-0aadd29bc84d',673,NULL),(2912,NULL,'The BRAF V639I mutation is located in the kinase domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF V639I demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 16:54:53',NULL,NULL,NULL,NULL,'4964c813-f10d-499c-aa59-0eac1d608d31',673,NULL),(2913,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 16:54:39',NULL,NULL,NULL,NULL,'8720ada2-a22c-43e5-ba96-85a693f9b919',673,NULL),(2914,NULL,'The BRAF E648Q mutation is located in the kinase domain of the protein. In vitro studies with MCF10A cells expressing BRAF E648Q demonstrate that the mutation is inactivating as measured by decreased cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 16:54:47',NULL,NULL,NULL,NULL,'42c85711-44fc-49b1-b1ee-4cb67df682ec',673,NULL),(2915,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 16:55:17',NULL,NULL,NULL,NULL,'bd4933d9-e9e0-4687-881b-06f4549f3aab',673,NULL),(2916,NULL,'The BRAF L678I mutation is located in the kinase domain of the protein. In vitro studies with MCF10A cells expressing BRAF L678I demonstrate that the mutation is inactivating as measured by decreased cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-04-03 16:55:19',NULL,NULL,NULL,NULL,'224a26ee-92a1-4522-a467-ae857e42e607',673,NULL),(2917,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-03 16:55:39',NULL,NULL,NULL,NULL,'4f8a1198-7552-4f31-a641-eb43655431a1',673,NULL),(2918,NULL,'The BRAF E695K mutation is located in the kinase domain of the protein. This mutation has been identified in plasma cell myeloma (PMID: 30545397). In vitro studies with MCF10A cells expressing BRAF E695K demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-03 16:55:43',NULL,NULL,NULL,NULL,'8e2dd1e0-e643-41e9-8cb6-495353cab76a',673,NULL),(2919,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 16:55:53',NULL,NULL,NULL,NULL,'7c2ca349-5c77-4743-be79-38858be03bd7',673,NULL),(2920,NULL,'The BRAF K698R mutation is located in the kinase domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF K698R demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 16:55:56',NULL,NULL,NULL,NULL,'79b0cedd-a37f-461b-a66a-3eb1b170869a',673,NULL),(2921,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-03 16:56:22',NULL,NULL,NULL,NULL,'c8e4c286-8009-4d83-a2cd-d6bede55fb19',673,NULL),(2922,NULL,'The BRAF A712T mutation is located in the kinase domain of the protein. This mutation has been identified in colorectal cancer (PMID: 27149842). In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF A712T demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-03 16:56:27',NULL,NULL,NULL,NULL,'21adc71d-5d1b-47ac-bd91-fe36d9618603',673,NULL),(2923,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-03 16:56:43',NULL,NULL,NULL,NULL,'5153ddf7-c6dc-44e0-819b-3bf55f74a371',673,NULL),(2924,NULL,'The BRAF P731S mutation is located in an unknown functional domain of the protein. This mutation has been identified in melanoma (PMID: 22817889). In vitro studies with Ba/F3 cells expressing BRAF P731S demonstrate that the mutation is activating as measured by increased transformation ability compared to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-03 16:56:48',NULL,NULL,NULL,NULL,'4830d21c-4512-487d-907b-10302c3890c5',673,NULL),(2925,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-04-11 18:30:27',NULL,NULL,NULL,NULL,'3e5d66e4-deae-484d-a6aa-0f3bcc54b586',673,NULL),(2926,NULL,'The BRAF A762E mutation is located in an unknown functional domain of the protein. While this mutation has not been functionally validated per se, in vitro studies with A549 cells expressing BRAF A762E demonstrated that the mutation confers resistance to EGFR inhibitor erlotinib as measured by sustained cell viability following treatment (PMID: 27478040).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-04 15:09:22',NULL,NULL,NULL,NULL,'3406a615-022d-41d0-958a-a06d1afb505a',673,NULL),(2927,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-04 15:10:15',NULL,NULL,NULL,NULL,'8ad3ac76-7130-4524-a8a1-77e7dd030ca5',673,NULL),(2928,NULL,'The BRAF Q609H mutation is located in the kinase domain of the protein. In vitro studies with Ba/F3 cells and MCF10A cells expressing BRAF Q609H demonstrate that the mutation is likely neutral as measured by cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-04-04 15:10:17',NULL,NULL,NULL,NULL,'b62c016c-0682-4d8a-97fb-8046cc52e64b',673,NULL),(2929,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-04 15:23:32',NULL,NULL,NULL,NULL,'d2b878af-1b7d-427c-bd5d-a9d9309b5336',673,NULL),(2930,NULL,'The GNAI1-BRAF fusion results from a chromosomal rearrangement fusing exon 1 of GNAI1 and exon 10 of BRAF. This fusion has been identified in pilocytic astrocytoma (PMID: 23817572). In vitro studies with NIH3T3 cells expressing GNAI1-BRAF demonstrate that the fusion is activating as measured by increased phosphorylated ERK levels compared to vector control (PMID: 29156677). Preclinical studies with NIH3T3 cells expressing the fusion demonstrated sensitivity to treatment with trametinib as measured by inhibition of colony formation (PMID: 29156677).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-04 15:23:34',NULL,NULL,NULL,NULL,'036538d2-4a0f-4a70-869d-eaee537fa0e3',673,NULL),(2931,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-04 15:25:38',NULL,NULL,NULL,NULL,'2d32fafb-1395-4195-b20f-24766f8449bc',673,NULL),(2932,NULL,'The CLCN6-BRAF fusion results from a chromosomal rearrangement fusing exon 2 of CLCN6 and exon 11 of BRAF, retaining the BRAF kinase domain. This fusion has been identified in pilocytic astrocytoma (PMID: 23817572). In vitro studies with NIH3T3 cells expressing CLCN6-BRAF demonstrate that the fusion is activating as measured by increased phosphorylated ERK levels compared to vector control (PMID: 29156677). Preclinical studies with NIH3T3 cells expressing the fusion demonstrated sensitivity to treatment with trametinib as measured by inhibition of colony formation (PMID: 29156677).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-11 18:31:19',NULL,NULL,NULL,NULL,'f17ba4d6-8c93-4010-8e0b-4996e43ed556',673,NULL),(2933,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-04-04 15:27:11',NULL,NULL,NULL,NULL,'b510f68a-e682-4207-b814-4830e295b120',673,NULL),(2934,NULL,'The BRAF T470P mutation is located in the kinase domain of the protein. In silico analysis suggests that the mutation is damaging and activating (PMID: 24037001).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-04 15:27:13',NULL,NULL,NULL,NULL,'3da02e00-69cb-48fd-904a-32bf20bfa6c4',673,NULL),(2935,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-04 15:29:41',NULL,NULL,NULL,NULL,'92f96e90-5b2d-48ec-ae66-c3b752ff624c',673,NULL),(2936,NULL,'The BRAF M517I mutation is located in the kinase domain of the protein. In vitro studies with MCF10A cells expressing BRAF M517I demonstrate that the mutation is inactivating as measured by decreased cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-04-04 15:29:44',NULL,NULL,NULL,NULL,'0626731d-71e8-48cd-9e0d-8ef190e46b9c',673,NULL),(2937,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-11 18:32:33',NULL,NULL,NULL,NULL,'d2f65940-007d-407b-9d70-1320ada739d9',673,NULL),(2938,NULL,'The CSTF3-BRAF fusion results from a chromosomal rearrangement that retains the full kinase domain of BRAF. Although not functionally characterized per se, a patient with metastatic melanoma harboring BRAF V600E was treated with dabrafenib plus trametinib and acquired CSTF3-BRAF as a secondary resistance mutation following disease progression (PMID: 28539463).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-11 18:32:34',NULL,NULL,NULL,NULL,'1df85571-f803-4389-88c0-675a3500e905',673,NULL),(2939,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-04 15:31:57',NULL,NULL,NULL,NULL,'ad532864-471e-4a40-92e7-f56678fed421',673,NULL),(2940,NULL,'The SEPT3-BRAF fusion results from chromosomal rearrangement fusing exon 10 of SEPT3 and exon 8 of BRAF, retaining the full kinase domain and losing the inhibitory RBD domain of BRAF. In vivo studies with patient-derived xenograft models expressing SEPT3-BRAF demonstrate that the fusion is activating as measured by increased cellular proliferation and invasion compared to wildtype (PMID: 30254212).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-04 15:31:59',NULL,NULL,NULL,NULL,'4fc7feb3-cbcd-4ab0-951a-f14bdbaf7b13',673,NULL),(2941,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-04-04 15:54:21',NULL,NULL,NULL,NULL,'8d032552-13af-43b5-8eba-5b0cb0aed0a6',673,NULL),(2942,NULL,'The SLC45A3-BRAF fusion results from inter-chromosomal rearrangement fusing untranslated exon 1 of SLC45A3 and exon 8 of BRAF. In vivo studies with nude mice overexpressing SLC45A3-BRAF demonstrate that the fusion is activating as measured by rapid tumor growth compared to vector control (PMID: 20526349). Preclinical studies with RWPE cells expressing this fusion demonstrated sensitivity to treatment with RAF kinase inhibitor sorafenib and MEK inhibitor U0126 as measured by decreased colony growth following treatment (PMID: 20526349).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-04-04 15:54:28',NULL,NULL,NULL,NULL,'886fcc24-f518-44a6-ab7b-7e9a6de042b3',673,NULL),(2943,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-04-04 15:56:12',NULL,NULL,NULL,NULL,'32001b24-77d3-4579-b919-138d16a18050',673,NULL),(2944,NULL,'The BRAF G615R mutation is located in the kinase domain of the protein. In vitro studies with MCF10A cells expressing BRAF G615R demonstrate that the mutation is inactivating as measured by decreased cell proliferation and viability comparable to wildtype (PMID: 29533785).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-04-04 15:56:15',NULL,NULL,NULL,NULL,'8743d79d-d96d-42c8-bbdc-34ca15be3433',673,NULL),(2945,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Resistance','2023-04-04 15:57:19',NULL,NULL,NULL,NULL,'35fa69d8-33b3-4bdb-9d22-99baa9a53978',673,NULL),(2946,NULL,'The BRAF L485S mutation is located in the kinase domain of the protein. Although not functionally characterized per se, in vitro studies with A375 cells expressing BRAF L485S demonstrate that the mutation confers resistance to treatment with dabrafenib as measured by sustained cell growth (PMID: 31925410).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-04-04 15:57:28',NULL,NULL,NULL,NULL,'fd514185-c401-4ece-ae58-e8368656cdfc',673,NULL),(2947,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:49:07',NULL,NULL,NULL,NULL,NULL,673,NULL),(2948,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:49:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(2949,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:50:21',NULL,NULL,NULL,NULL,NULL,673,NULL),(2950,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:50:25',NULL,NULL,NULL,NULL,NULL,673,NULL),(2951,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:50:59',NULL,NULL,NULL,NULL,NULL,673,NULL),(2952,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:50:57',NULL,NULL,NULL,NULL,NULL,673,NULL),(2953,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:50:57',NULL,NULL,NULL,NULL,NULL,673,NULL),(2954,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:52:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(2955,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:52:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(2956,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:52:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(2957,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-28 09:52:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(2958,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(2959,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(2960,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(2961,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(2962,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(2963,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:50',NULL,NULL,NULL,NULL,NULL,673,NULL),(2964,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:09:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(2965,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(2966,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:33',NULL,NULL,NULL,NULL,NULL,673,NULL),(2967,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(2968,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(2969,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(2970,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(2971,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:10:28',NULL,NULL,NULL,NULL,NULL,673,NULL),(2972,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:11:27',NULL,NULL,NULL,NULL,NULL,673,NULL),(2973,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:11:25',NULL,NULL,NULL,NULL,NULL,673,NULL),(2974,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:11:24',NULL,NULL,NULL,NULL,NULL,673,NULL),(2975,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 13:11:23',NULL,NULL,NULL,NULL,NULL,673,NULL),(2976,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:34:12',NULL,NULL,NULL,NULL,NULL,673,NULL),(2977,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:34:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(2978,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:34:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(2979,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:07',NULL,NULL,NULL,NULL,NULL,673,NULL),(2980,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:08',NULL,NULL,NULL,NULL,NULL,673,NULL),(2981,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(2982,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(2983,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(2984,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(2985,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(2986,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:35:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(2987,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:36:23',NULL,NULL,NULL,NULL,NULL,673,NULL),(2988,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:36:25',NULL,NULL,NULL,NULL,NULL,673,NULL),(2989,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:36:22',NULL,NULL,NULL,NULL,NULL,673,NULL),(2990,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:36:21',NULL,NULL,NULL,NULL,NULL,673,NULL),(2991,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:37:21',NULL,NULL,NULL,NULL,NULL,673,NULL),(2992,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(2993,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:52:06',NULL,NULL,NULL,NULL,NULL,673,NULL),(2994,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(2995,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(2996,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:50',NULL,NULL,NULL,NULL,NULL,673,NULL),(2997,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(2998,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(2999,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3000,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:53',NULL,NULL,NULL,NULL,NULL,673,NULL),(3001,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3002,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3003,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3004,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:53',NULL,NULL,NULL,NULL,NULL,673,NULL),(3005,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3006,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3007,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3008,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3009,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3010,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(3011,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:55',NULL,NULL,NULL,NULL,NULL,673,NULL),(3012,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:35',NULL,NULL,NULL,NULL,NULL,673,NULL),(3013,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3014,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:33',NULL,NULL,NULL,NULL,NULL,673,NULL),(3015,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3016,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3017,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3018,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3019,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:28',NULL,NULL,NULL,NULL,NULL,673,NULL),(3020,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:27',NULL,NULL,NULL,NULL,NULL,673,NULL),(3021,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:26',NULL,NULL,NULL,NULL,NULL,673,NULL),(3022,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:20',NULL,NULL,NULL,NULL,NULL,673,NULL),(3023,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:22',NULL,NULL,NULL,NULL,NULL,673,NULL),(3024,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:19',NULL,NULL,NULL,NULL,NULL,673,NULL),(3025,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-29 15:38:18',NULL,NULL,NULL,NULL,NULL,673,NULL),(3026,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:56:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3027,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:12',NULL,NULL,NULL,NULL,NULL,673,NULL),(3028,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:16',NULL,NULL,NULL,NULL,NULL,673,NULL),(3029,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:14',NULL,NULL,NULL,NULL,NULL,673,NULL),(3030,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:13',NULL,NULL,NULL,NULL,NULL,673,NULL),(3031,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3032,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3033,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3034,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3035,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3036,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:57:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3037,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:58:21',NULL,NULL,NULL,NULL,NULL,673,NULL),(3038,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:58:22',NULL,NULL,NULL,NULL,NULL,673,NULL),(3039,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:58:22',NULL,NULL,NULL,NULL,NULL,673,NULL),(3040,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:58:20',NULL,NULL,NULL,NULL,NULL,673,NULL),(3041,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:58:19',NULL,NULL,NULL,NULL,NULL,673,NULL),(3042,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:58:18',NULL,NULL,NULL,NULL,NULL,673,NULL),(3043,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3044,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3045,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3046,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3047,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:33',NULL,NULL,NULL,NULL,NULL,673,NULL),(3048,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3049,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3050,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 11:59:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3051,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:00:01',NULL,NULL,NULL,NULL,NULL,673,NULL),(3052,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:00:00',NULL,NULL,NULL,NULL,NULL,673,NULL),(3053,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:00:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3054,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:00:58',NULL,NULL,NULL,NULL,NULL,673,NULL),(3055,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3056,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3057,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3058,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3059,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3060,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3061,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3062,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3063,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3064,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3065,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3066,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-30 12:01:28',NULL,NULL,NULL,NULL,NULL,673,NULL),(3067,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:21',NULL,NULL,NULL,NULL,NULL,673,NULL),(3068,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:23',NULL,NULL,NULL,NULL,NULL,673,NULL),(3069,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:00',NULL,NULL,NULL,NULL,NULL,673,NULL),(3070,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:02',NULL,NULL,NULL,NULL,NULL,673,NULL),(3071,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:06',NULL,NULL,NULL,NULL,NULL,673,NULL),(3072,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:58',NULL,NULL,NULL,NULL,NULL,673,NULL),(3073,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:59',NULL,NULL,NULL,NULL,NULL,673,NULL),(3074,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:02',NULL,NULL,NULL,NULL,NULL,673,NULL),(3075,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:05',NULL,NULL,NULL,NULL,NULL,673,NULL),(3076,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:57',NULL,NULL,NULL,NULL,NULL,673,NULL),(3077,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:58',NULL,NULL,NULL,NULL,NULL,673,NULL),(3078,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:03',NULL,NULL,NULL,NULL,NULL,673,NULL),(3079,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:04',NULL,NULL,NULL,NULL,NULL,673,NULL),(3080,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:56',NULL,NULL,NULL,NULL,NULL,673,NULL),(3081,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:00:55',NULL,NULL,NULL,NULL,NULL,673,NULL),(3082,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(3083,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:53',NULL,NULL,NULL,NULL,NULL,673,NULL),(3084,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3085,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3086,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(3087,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:54',NULL,NULL,NULL,NULL,NULL,673,NULL),(3088,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(3089,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3090,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3091,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:56',NULL,NULL,NULL,NULL,NULL,673,NULL),(3092,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3093,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:50',NULL,NULL,NULL,NULL,NULL,673,NULL),(3094,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3095,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3096,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3097,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3098,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3099,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3100,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3101,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3102,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3103,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3104,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3105,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:33',NULL,NULL,NULL,NULL,NULL,673,NULL),(3106,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3107,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:01:35',NULL,NULL,NULL,NULL,NULL,673,NULL),(3108,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:03:06',NULL,NULL,NULL,NULL,NULL,673,NULL),(3109,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:03:07',NULL,NULL,NULL,NULL,NULL,673,NULL),(3110,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:03:08',NULL,NULL,NULL,NULL,NULL,673,NULL),(3111,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:03:05',NULL,NULL,NULL,NULL,NULL,673,NULL),(3112,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:03:05',NULL,NULL,NULL,NULL,NULL,673,NULL),(3113,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-03-31 12:03:04',NULL,NULL,NULL,NULL,NULL,673,NULL),(3114,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3115,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3116,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3117,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3118,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3119,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3120,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3121,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3122,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:51:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3123,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:57:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3124,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:57:54',NULL,NULL,NULL,NULL,NULL,673,NULL),(3125,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:57:55',NULL,NULL,NULL,NULL,NULL,673,NULL),(3126,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:57:56',NULL,NULL,NULL,NULL,NULL,673,NULL),(3127,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:57:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3128,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:57:54',NULL,NULL,NULL,NULL,NULL,673,NULL),(3129,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:58:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3130,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3131,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:26',NULL,NULL,NULL,NULL,NULL,673,NULL),(3132,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3133,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:35',NULL,NULL,NULL,NULL,NULL,673,NULL),(3134,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3135,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3136,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3137,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3138,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3139,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3140,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3141,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3142,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3143,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3144,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:33',NULL,NULL,NULL,NULL,NULL,673,NULL),(3145,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3146,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3147,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:24',NULL,NULL,NULL,NULL,NULL,673,NULL),(3148,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3149,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3150,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 12:59:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3151,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 14:22:26',NULL,NULL,NULL,NULL,NULL,673,NULL),(3152,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 14:22:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3153,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 14:22:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(3154,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 14:24:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3155,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:08',NULL,NULL,NULL,NULL,NULL,673,NULL),(3156,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:10',NULL,NULL,NULL,NULL,NULL,673,NULL),(3157,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:12',NULL,NULL,NULL,NULL,NULL,673,NULL),(3158,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:14',NULL,NULL,NULL,NULL,NULL,673,NULL),(3159,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:09',NULL,NULL,NULL,NULL,NULL,673,NULL),(3160,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:11',NULL,NULL,NULL,NULL,NULL,673,NULL),(3161,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:12',NULL,NULL,NULL,NULL,NULL,673,NULL),(3162,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:14',NULL,NULL,NULL,NULL,NULL,673,NULL),(3163,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:09',NULL,NULL,NULL,NULL,NULL,673,NULL),(3164,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:35',NULL,NULL,NULL,NULL,NULL,673,NULL),(3165,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3166,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3167,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:48:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3168,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:49:02',NULL,NULL,NULL,NULL,NULL,673,NULL),(3169,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:49:03',NULL,NULL,NULL,NULL,NULL,673,NULL),(3170,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:49:04',NULL,NULL,NULL,NULL,NULL,673,NULL),(3171,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:49:25',NULL,NULL,NULL,NULL,NULL,673,NULL),(3172,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:50:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3173,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:50:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3174,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:50:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3175,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:50:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3176,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:50:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3177,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:01',NULL,NULL,NULL,NULL,NULL,673,NULL),(3178,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:54',NULL,NULL,NULL,NULL,NULL,673,NULL),(3179,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3180,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(3181,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:50',NULL,NULL,NULL,NULL,NULL,673,NULL),(3182,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:55',NULL,NULL,NULL,NULL,NULL,673,NULL),(3183,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3184,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3185,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:51',NULL,NULL,NULL,NULL,NULL,673,NULL),(3186,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:56',NULL,NULL,NULL,NULL,NULL,673,NULL),(3187,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:45',NULL,NULL,NULL,NULL,NULL,673,NULL),(3188,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3189,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3190,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:57',NULL,NULL,NULL,NULL,NULL,673,NULL),(3191,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3192,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3193,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3194,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:57',NULL,NULL,NULL,NULL,NULL,673,NULL),(3195,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3196,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 15:51:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3197,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:10',NULL,NULL,NULL,NULL,NULL,673,NULL),(3198,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:14',NULL,NULL,NULL,NULL,NULL,673,NULL),(3199,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:13',NULL,NULL,NULL,NULL,NULL,673,NULL),(3200,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:12',NULL,NULL,NULL,NULL,NULL,673,NULL),(3201,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3202,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3203,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:53:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3204,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:54:07',NULL,NULL,NULL,NULL,NULL,673,NULL),(3205,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:54:08',NULL,NULL,NULL,NULL,NULL,673,NULL),(3206,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:54:09',NULL,NULL,NULL,NULL,NULL,673,NULL),(3207,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:54:07',NULL,NULL,NULL,NULL,NULL,673,NULL),(3208,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:54:06',NULL,NULL,NULL,NULL,NULL,673,NULL),(3209,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:54:05',NULL,NULL,NULL,NULL,NULL,673,NULL),(3210,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:55:04',NULL,NULL,NULL,NULL,NULL,673,NULL),(3211,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:55:27',NULL,NULL,NULL,NULL,NULL,673,NULL),(3212,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:56:06',NULL,NULL,NULL,NULL,NULL,673,NULL),(3213,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:56:07',NULL,NULL,NULL,NULL,NULL,673,NULL),(3214,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:56:08',NULL,NULL,NULL,NULL,NULL,673,NULL),(3215,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:56:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3216,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:56:57',NULL,NULL,NULL,NULL,NULL,673,NULL),(3217,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:56:58',NULL,NULL,NULL,NULL,NULL,673,NULL),(3218,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:57:00',NULL,NULL,NULL,NULL,NULL,673,NULL),(3219,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:57:02',NULL,NULL,NULL,NULL,NULL,673,NULL),(3220,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-03 16:57:03',NULL,NULL,NULL,NULL,NULL,673,NULL),(3221,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:09:50',NULL,NULL,NULL,NULL,NULL,673,NULL),(3222,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:10:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3223,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:09:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3224,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:09:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3225,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3226,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3227,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3228,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:46',NULL,NULL,NULL,NULL,NULL,673,NULL),(3229,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3230,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3231,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3232,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(3233,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:52',NULL,NULL,NULL,NULL,NULL,673,NULL),(3234,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3235,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3236,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:47',NULL,NULL,NULL,NULL,NULL,673,NULL),(3237,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3238,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3239,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3240,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:53',NULL,NULL,NULL,NULL,NULL,673,NULL),(3241,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3242,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3243,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:48',NULL,NULL,NULL,NULL,NULL,673,NULL),(3244,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:53',NULL,NULL,NULL,NULL,NULL,673,NULL),(3245,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3246,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:43',NULL,NULL,NULL,NULL,NULL,673,NULL),(3247,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3248,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:53',NULL,NULL,NULL,NULL,NULL,673,NULL),(3249,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3250,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3251,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:49',NULL,NULL,NULL,NULL,NULL,673,NULL),(3252,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:54',NULL,NULL,NULL,NULL,NULL,673,NULL),(3253,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3254,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:44',NULL,NULL,NULL,NULL,NULL,673,NULL),(3255,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:50',NULL,NULL,NULL,NULL,NULL,673,NULL),(3256,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:55',NULL,NULL,NULL,NULL,NULL,673,NULL),(3257,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:26:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3258,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:23',NULL,NULL,NULL,NULL,NULL,673,NULL),(3259,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:33',NULL,NULL,NULL,NULL,NULL,673,NULL),(3260,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3261,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:29',NULL,NULL,NULL,NULL,NULL,673,NULL),(3262,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:23',NULL,NULL,NULL,NULL,NULL,673,NULL),(3263,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3264,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:39',NULL,NULL,NULL,NULL,NULL,673,NULL),(3265,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3266,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:24',NULL,NULL,NULL,NULL,NULL,673,NULL),(3267,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3268,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3269,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3270,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:25',NULL,NULL,NULL,NULL,NULL,673,NULL),(3271,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:35',NULL,NULL,NULL,NULL,NULL,673,NULL),(3272,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3273,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:30',NULL,NULL,NULL,NULL,NULL,673,NULL),(3274,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:25',NULL,NULL,NULL,NULL,NULL,673,NULL),(3275,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(3276,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:40',NULL,NULL,NULL,NULL,NULL,673,NULL),(3277,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3278,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:26',NULL,NULL,NULL,NULL,NULL,673,NULL),(3279,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3280,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:41',NULL,NULL,NULL,NULL,NULL,673,NULL),(3281,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:31',NULL,NULL,NULL,NULL,NULL,673,NULL),(3282,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:26',NULL,NULL,NULL,NULL,NULL,673,NULL),(3283,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3284,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3285,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3286,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:27',NULL,NULL,NULL,NULL,NULL,673,NULL),(3287,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:38',NULL,NULL,NULL,NULL,NULL,673,NULL),(3288,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:42',NULL,NULL,NULL,NULL,NULL,673,NULL),(3289,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:32',NULL,NULL,NULL,NULL,NULL,673,NULL),(3290,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:28:27',NULL,NULL,NULL,NULL,NULL,673,NULL),(3291,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:55:14',NULL,NULL,NULL,NULL,NULL,673,NULL),(3292,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:55:15',NULL,NULL,NULL,NULL,NULL,673,NULL),(3293,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:55:18',NULL,NULL,NULL,NULL,NULL,673,NULL),(3294,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:55:16',NULL,NULL,NULL,NULL,NULL,673,NULL),(3295,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:57:34',NULL,NULL,NULL,NULL,NULL,673,NULL),(3296,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:57:35',NULL,NULL,NULL,NULL,NULL,673,NULL),(3297,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:57:37',NULL,NULL,NULL,NULL,NULL,673,NULL),(3298,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-04-04 15:57:36',NULL,NULL,NULL,NULL,NULL,673,NULL),(26074,NULL,'ROS1, a receptor tyrosine kinase, is altered by mutation or chromosomal rearrangement in a diverse range of cancers, including lung cancer.','GENE_SUMMARY',NULL,_binary '\0',NULL,'2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'865dbc19-355b-4142-b48c-6baa4018aae2',6098,NULL),(26075,NULL,'The ROS1 gene encodes a transmembrane protein with intracellular tyrosine kinase activity (PMID: 18778756). ROS1 is a member of the sevenless subfamily of tyrosine kinase insulin receptor genes (PMID: 27256160). The normal physiological role and ligand of this protein in humans is currently unknown (PMID: 23814043). ROS1 rearrangements where the kinase domain is retained (PMID: 22327623) are implicated in a range of human epithelial cancers including cholangiocarcinoma (PMID: 21253578), ovarian carcinoma (PMID: 22163003), gastric carcinoma (PMID: 23400546), angiosarcoma (PMID: 23637631) and most commonly non-small cell lung cancer (PMID: 22215748). Although ROS1 rearrangements were first discovered in a human glioblastoma cell line (PMID: 2827175), there is a paucity of ROS1 rearrangements in human gliomas (PMID: 24999209). The mechanism by which ROS1 rearrangements leads to dysregulated kinase activity is not clear, as its ligand has not yet been deciphered; however, it is hypothesized to occur through constitutive kinase activation (PMID: 18083107).','GENE_BACKGROUND',NULL,_binary '\0',NULL,'2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'6ab05933-cfea-46bb-a467-998f93a46f77',6098,NULL),(26076,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'aeb5989e-4867-4046-8c60-d724510de642',6098,NULL),(26077,NULL,'Fusions of ROS1 arise from chromosomal translocations that fuse the N-terminal portion of a partner protein with the C-terminal kinase-domain-containing portion of ROS1. These fusion proteins are found in a variety of cancer types and result in constitutive activation of ROS1 kinase activity. Biological characterization of specific ROS1 fusion proteins demonstrates that they are activating as shown by enhanced proliferation and colony formation and that they are sensitive to treatment with the tyrosine kinase inhibitors crizotinib and entrectinib (PMID: 23719267, 22919003, 22215748, 28465216). Preliminary clinical data suggests that lorlatinib may also be efficacious in patients with ROS1 fusion-positive non-small cell lung cancer, including those with central nervous system disease (PMID: 31669155).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2020-01-06 00:53:53',NULL,NULL,NULL,NULL,'e8113bd5-9371-4afc-a563-3c8db56bb3d8',6098,NULL),(26078,NULL,'The tyrosine kinase inhibitors crizotinib, entrectinib, repotrectinib and taletrectinib are FDA-approved for the treatment of patients with ROS1 fusion-positive non-small cell lung cancer.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2025-06-13 17:40:48',NULL,NULL,NULL,NULL,'3aa7655f-3fd3-41d8-a883-43bfccc6d2a2',6098,NULL),(26079,NULL,'Crizotinib, an inhibitor of ALK, ROS1 and MET tyrosine kinases, is FDA-approved for the treatment of patients with metastatic non-small cell lung cancer (NSCLC) whose tumors are ROS1-positive. FDA approval is based on the results of a Phase I trial of crizotinib in 53 patients with advanced NSCLC positive for ROS1 rearrangements in which the objective response rate was 72% (95% CI= 58-84), including six complete responses, 32 partial responses (66%) and a median progression-free survival (PFS) of 19.3 months (95% CI= 15.2-39.1)(PMID: 25264305, 30980071). In a Phase II trial of crizotinib in 127 East Asian patients with ROS1-positive NSCLC, the overall response rate was 71.1% (95% CI= 63.0-79.3), including seventeen of 127 patients with complete responses (13.4%), 74 of 127 patients with partial responses (58.3%), and a median PFS of 15.9 months (95% CI= 12.9-24.0) (PMID: 29596029).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-23 11:33:59',NULL,'LEVEL_1','NO','LEVEL_3B','c89514ef-b096-442a-828d-6ed357e12d57',6098,NULL),(26080,NULL,'Entrectinib is a potent, orally available, ATP-competitive tyrosine kinase inhibitor that is highly effective against NTRK1/2/3, ROS1 and ALK and is FDA-approved for patients with ROS1-fusion-positive non-small cell lung cancer (NSCLC). FDA approval was based on the results of three multicenter, single-arm clinical trials (ALKA, STARTRK-1 and STARTRK-2) in which of 51 evaluable patients with ROS1-fusion-positive NSCLC who were treated with entrectinib, the overall response rate was 78% (95% CI = 65-89) and response duration was more than one year for 55% of patients (PMID: 28183697)(Abstract: Drilon et al. Abstract# CT060, AACR 2017. https://cancerres.aacrjournals.org/content/77/13_Supplement/CT060)(Abstract: Braud et al. Abstract# 64P, ESMO 2019. https://academic.oup.com/annonc/article/30/Supplement_9/mdz420.003/5638397).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-23 11:34:10',NULL,'LEVEL_1','NO','LEVEL_3B','3dbaf733-ca83-47d3-b8d6-f15162c4055f',6098,NULL),(26081,NULL,'Repotrectinib, an orally administered next-generation ROS1/TRK tyrosine kinase inhibitor (TKI), is FDA-approved for the treatment of patients with ROS1 fusion positive non-small cell lung cancer (NSCLC). FDA-approval is based on the results of the Phase I/II TRIDENT-1 trial of repotrectinib in patients with ROS1 fusion positive NSCLC that are ROS1-inhibitor-naive or pre-treated. ROS1-inhibitor-naive patients (n=71) demonstrated an objective response rate of 79% (95% CI=68%-88%), with a 6% complete response rate and a 73% partial response rate, and a median duration response of 34.1 months (95% CI=25.6-NE) whereas ROS1-inhibitor pre-treated patients (n=56) demonstrated an objective response rate of 38% (95% CI=25%-52%), with a 5% complete response rate and a 32% partial response rate, and a median duration response of 14.8 months (95% CI=7.6-NE) (Abstract: Cho et al. Abstract# OA03.06, J. of Thoracic Oncol. Nov. 2023. https://www.jto.org/article/S1556-0864%2823%2900837-7/fulltext).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-12-01 15:30:17',NULL,'LEVEL_1','NO','LEVEL_3B','8c2ac63c-bdc2-471f-8beb-89b57b36cf16',6098,NULL),(26082,NULL,'Ceritinib is a small molecule inhibitor of ALK that also binds to and inhibits ROS1 and is NCCN-listed as a category 2A treatment recommendation for patients with non-small cell lung cancer (NSCLC) who have ROS1 rearrangements. In a Phase II study of ceritinib in 32 patients with NSCLC harboring ROS1 rearrangement, the overall response rate in 28 evaluable patients was 62% (95% CI= 45 - 77) with one patient having a complete response and nineteen patients having a partial response (PMID: 28520527).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-24 13:16:49',NULL,'LEVEL_2','NO','LEVEL_3B','a24c4391-adab-4b50-9df2-5d4253e243e6',6098,NULL),(26083,NULL,'Lorlatinib is a third-generation small molecule inhibitor of ALK and ROS1 that is NCCN-listed as a category 2A recommended subsequent therapy option for patients with ROS1-rearrangement-positive non-small cell lung cancer (NSCLC) who progressed on entrectinib, crizotinib or ceritinib. In a phase I/II trial of lorlatinib in patients with ROS1-fusion-positive NSCLC, thirteen of 21 tyrosine kinase inhibitor-naive patients had a response to lorlatinib for an overall response rate of 62% (95% CI= 38-82), while fourteen of 40 crizotinib pre-treated patients had a response to lorlatinib with an overall response rate of 35% (95% CI= 21-52)(PMID: 31669155).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2023-02-23 11:34:32',NULL,'LEVEL_2','NO','LEVEL_3B','88b15a8c-c63f-4b50-a0ae-b0c1e08180a9',6098,NULL),(26084,NULL,'Taletrectinib is an orally available, small-molecule ROS1/ NTRK tyrosine kinase inhibitor (TKI) that is FDA-approved for the treatment of adult patients with locally advanced or metastatic ROS1-positive non-small cell lung cancer (NSCLC). FDA approval was based on the results of the Phase I TRUST-I (NCT04395677) and TRUST-II (NCT04919811) trials of taletrectinib in patients with ROS1-positive NSCLC. \nIn the Phase II TRUST-I (NCT04395677) and TRUST-II (NCT04919811) trials, TKI-nave patients (n=160) demonstrated an objective response rate (ORR) of 88.8% (95% CI=82.8-93.2), with eight complete responses (CR), 134 partial responses (PR) and ten patients demonstrating stable disease (SD), a disease control rate (DCR) of 95.0% (95% CI=90.4-97.8) and a median duration of response (DOR) of 44.2 months (95% CI=30.4-NR) (PMID: 40179330). Patients that were pretreated with a ROS1 TKI (n=113) demonstrated an ORR of 55.8% (95% CI=46.1-65.1), with five CRs, 58 PRs and 36 patients demonstrating stable disease, a DCR of 87.6% (95% CI=80.1-93.1) and a median DOR of 16.6 months (95% CI=10.6-27.3) (PMID: 40179330).','STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY','LEVEL_Fda2',_binary '\0','Sensitive','2025-06-13 17:47:40',NULL,'LEVEL_1','NO','LEVEL_3B','9682534d-aaea-4f23-9fb7-40e71fc32a44',6098,NULL),(26085,NULL,'While the tyrosine kinase inhibitors crizotinib, entrectinib, repotrectinib and taletrectinib are FDA-approved for the treatment of patients with ROS1 fusion-positive non-small cell lung cancer, the clinical utility of these agents in patients with ROS1 fusion-positive [[tumor type]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2025-06-13 17:47:54',NULL,NULL,NULL,NULL,'25c05c7e-a64e-4eec-8206-4a198462879f',6098,NULL),(26086,NULL,'While the tyrosine kinase inhibitors crizotinib, entrectinib, repotrectinib and taletrectinib are FDA-approved for the treatment of patients with ROS1 fusion-positive non-small cell lung cancer, the clinical utility of these agents in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2025-06-18 19:09:00',NULL,NULL,NULL,NULL,'df602908-e466-4f95-b006-c65382ca083a',6098,NULL),(26087,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-15 11:35:00',NULL,NULL,NULL,NULL,'9d636c58-c0f6-40f4-9b100-d61369f2d563',6098,NULL),(26088,NULL,'The FIG-ROS1 fusion is characterized by the fusion of the 3\' region of ROS1 (exon 35) to the 5\' region of FIG (exon 7); this fusion results from an intra-chromosomal deletion in chromosome 6 (PMID: 12661006). The FIG-ROS1 fusion results in a constitutively active ROS1 kinase (PMID: 2827175, 12661006).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-02-15 11:35:56',NULL,NULL,NULL,NULL,'9151ba5c-e692-49a5-b13f-b7ee0875eb8a',6098,NULL),(26089,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'e866e555-f8c5-44cd-9354-67152a0b84e6',6098,NULL),(26090,NULL,'CD74-ROS1 is a rearrangement mutation characterized by the fusion of CD74 (exon 6) and ROS1 (exon 34). This fusion generates a protein with constitutive kinase activity (PMID: 22215748, 22919003, 22327623, 18083107). CD74-ROS1 demonstrated transforming capabilities when tested in foci forming assays in vitro, and also demonstrated the ability to form subcutaneous tumors in nude mice (PMID: 22327623). A patient with non-small cell lung cancer harboring a CD74-ROS1 fusion responded to treatment with brigatinib and ceritinib after progressing on crizotinib (Abstract: Hegde et al. JCO PO, 2018. https://ascopubs.org/doi/full/10.1200/PO.18.00267).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2019-05-20 19:36:32',NULL,NULL,NULL,NULL,'bc796f57-eb77-4434-a67b-c83f986ccd53',6098,NULL),(26091,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'b5d0a7be-8f5d-48ca-8dc7-fccc54485b44',6098,NULL),(26092,NULL,'LRIG3-ROS1, a rearrangement mutation that results in the fusion of exon 16 of EZR with exon 35 of ROS1, results in a constitutively active protein containing the ROS1 kinase domain. This fusion demonstrated transforming capabilities when tested in foci forming assays in vitro, and also demonstrated the ability to form subcutaneous tumors in nude mice (PMID: 22327623).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'4192b084-39f0-4e6e-baed-2f099ebc5e3a',6098,NULL),(26093,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'5c3d5d86-c7ef-4a5f-ad33-02414d7a37ba',6098,NULL),(26094,NULL,'KDELR2-ROS1, a rearrangement mutation that results in the fusion of the 5\' end of KDELR2 with the 3\' end of with ROS1 (PMID: 22980976). The oncogenic potential of this fusion has not been functionally tested.','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'47a2ab73-a141-4768-8b29-205a346d447b',6098,NULL),(26095,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'c243f6b3-8a77-488a-b446-ae6d349aebe0',6098,NULL),(26096,NULL,'SLC34A2-ROS1 is a rearrangement mutation that results in the fusion of SLC34A2 (exon 4) and ROS1 (exon 32 and 34), creating a short and long transcript, respectively. This in-frame mutation generates a protein with constitutively active kinase activity (PMID: 22215748, 22919003). This fusion demonstrated transforming capabilities when tested in foci forming assays in vitro, and also demonstrated the ability to form subcutaneous tumors in nude mice (PMID: 22327623).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'19cbcfdb-7220-41f7-9db5-519c7aba7d14',6098,NULL),(26097,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'8bd71a92-568f-4a02-b316-68c6fc3f4a19',6098,NULL),(26098,NULL,'TPM3-ROS1, a rearrangement mutation that results in the fusion of exon 8 of TPM3 with exon 35 of ROS1, results in a constitutively active protein containing the ROS1 kinase domain (PMID: 22327623). This fusion demonstrated transforming capabilities when tested in foci forming assays in vitro, and also demonstrated the ability to form subcutaneous tumors in nude mice (PMID: 22327623).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'99fc9f8c-568d-48ea-a6a9-cd00e11509d6',6098,NULL),(26099,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'9c1e1dea-f7b5-4819-bb0a-2c14226842cb',6098,NULL),(26100,NULL,'SDC4-ROS1, a rearrangement mutation that results in the fusion of exon 2 of SDC4 with exon 33 or 34 of ROS1, results in a constitutively active protein containing the ROS1 kinase domain (PMID: 22327623). This fusion demonstrated transforming capabilities when tested in foci forming assays in vitro, and also demonstrated the ability to form subcutaneous tumors in nude mice (PMID: 22327623).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'f139c10b-60e3-4a9a-98e4-8c4059ebf637',6098,NULL),(26101,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'4829f927-214f-4cad-8e03-f865f2388eea',6098,NULL),(26102,NULL,'EZR-ROS1 is a rearrangement mutation that results in the fusion of exon 10 of EZR with exon 34 of ROS1, results in a constitutively active protein containing the ROS1 kinase domain (PMID: 22327623). This fusion demonstrated transforming capabilities when tested in foci forming assays in vitro, and also demonstrated the ability to form subcutaneous tumors in nude mice (PMID: 22327623).','MUTATION_EFFECT',NULL,_binary '\0','Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'223971d5-6d75-429c-8ee6-ff87292c52ba',6098,NULL),(26103,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'e11e0f79-6dbd-4662-90e1-f511b46f1bd9',6098,NULL),(26104,NULL,'The CCDC6-ROS1 fusion is a rearrangement resulting in the fusion of the 3\' region of ROS1 (exon 34) to the 5\' region of CCDC6 (exon 6). This fusion retains the coiled-coil domain of CCDC6 and the kinase domain of ROS1 (PMID: 22975805, 25870798). The oncogenic potential of this fusion has not been functionally tested.','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-10-21 18:23:30',NULL,NULL,NULL,NULL,'5bec6cce-2d4e-4d27-9435-e82c1c5afb4e',6098,NULL),(26105,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'1f433934-e5ee-4521-b22c-813311f97682',6098,NULL),(26106,NULL,'The CEP85L-ROS1 fusion protein combines the N terminal of CEP85L until exon 8 with the C-terminal of ROS1 starting at exon 36 and is found in CNS tumors, angiosarcoma and glioblastoma (PMID: 23637631, 25204415, 30171048). The breakpoint preserves the coiled-coil domain of CEP85L and the tyrosine kinase domain of ROS1, and is predicted to possess transforming abilities (PMID: 23637631, 25204415, 30171048). Expression of this fusion protein in BaF3, NIH3T3, and immortalized human astrocyte cells demonstrated that it is activating as shown by IL3 independent growth, pathway activation and increased colony formation compared to ectopic expression of wildtype ROS1. Expression of this fusion in BaF3 cells also demonstrated that it is sensitive to multiple TKIs including crizotinib, foretinib, cabozantinib, ceritinib, brigatinib, AZD3463, lorlatinib, and entrectinib proliferation, pathway activation and colony formation (PMID: 30171048).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2019-04-22 01:54:49',NULL,NULL,NULL,NULL,'2efa77d7-fce2-4db2-9a47-efd2d37a8442',6098,NULL),(26107,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'56ca090e-c0bb-4513-8807-d51bc9c88a9c',6098,NULL),(26108,NULL,'TFG-ROS1 is a rearrangement mutation that results in the fusion of TFG (exon 4) to ROS1 (exon 36) (PMID: 24875859). The oncogenic potential of this fusion has not been functionally tested.','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'189ca680-c125-47ce-a56b-8475d18ed537',6098,NULL),(26109,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'8912c735-23c3-4d05-9523-eac31ed076e0',6098,NULL),(26110,NULL,'YWHAE-ROS1 is a rearrangement mutation that results in the fusion of YWHAE (exon 4) to ROS1 (exon 36) (PMID: 24875859). The oncogenic potential of this fusion has not been functionally tested.','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'e877496a-b1bb-4cef-95ec-7dbf01296343',6098,NULL),(26111,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'d36eed89-50ac-4f42-841c-1020fe8b559a',6098,NULL),(26112,NULL,'LIMA1-ROS1 is a rearrangement mutation that results in the fusion of LIMA1 (exon 10) to ROS1 (exon 36). This novel fusion has been identified in a tumor sample from a patient with non-small cell lung cancer (NSCLC) treated with crizotinib (PMID: 25264305).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-02-24 13:21:30',NULL,NULL,NULL,NULL,'2526d944-55a3-407d-ae39-80db14f9ce98',6098,NULL),(26113,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'42b26a42-224d-4eed-a458-c9749f402c19',6098,NULL),(26114,NULL,'MSN-ROS1 is a rearrangement mutation that results in the fusion of MSN (exon 9) to ROS1 (exon 34). This novel fusion has been identified in a tumor sample from a patient with non-small cell lung cancer (NSCLC) treated with crizotinib (PMID: 25264305).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-02-24 13:21:39',NULL,NULL,NULL,NULL,'363f218b-1836-413c-bdc9-58212a6060f0',6098,NULL),(26115,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-10-16 13:59:45',NULL,NULL,NULL,NULL,'86467f41-2065-463d-9db7-c86f7d5a94ed',6098,NULL),(26116,NULL,'The ROS1 G2032R mutation is located in the protein\'s kinase domain and is associated with acquired resistance to crizotinib in ROS1-rearranged non-small cell lung cancer (PMID: 25688157, 24047073, 23724914). In vitro studies with A549 cells expressing ROS1 G2032R demonstrate that the mutation is activating as measured by increased cellular invasion and migration, increased expression of MMP-9 and TWIST1 and induced epithelial-mesenchymal transition compared to wildtype (PMID: 29477381). Preclinical studies and structural analysis of ROS1 G2032R demonstrate that the mutation confers resistance by interfering with crizotinib binding (PMID: 25351743, 23724914, 25033171). This mutation corresponds to the G1202R ALK kinase domain mutant (PMID: 23724914, 27401242). Preclinical studies of ROS1 G2032R demonstrate sensitivity to combination treatment of TWIST1 siRNA and crizotinib as measured by reduced cellular viability and invasion (PMID: 29477381).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-09-25 15:20:27',NULL,NULL,NULL,NULL,'aa803d57-5855-4e90-a480-c4362b3b482a',6098,NULL),(26117,NULL,'The ROS1 G2032R mutation has been identified in ROS1-mutant non-small cell lung cancer (NSCLC) collected following disease progression on the tyrosine kinase inhibitors crizotinib and lorlatinib. There are promising clinical data in patients with ROS1 G2032R mutant NSCLC treated with the ROS1/NTRK tyrosine kinase inhibitor taletrectinib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-09-25 14:45:58',NULL,NULL,NULL,NULL,'2516a3d9-b233-4440-b634-cb8b25796e39',6098,NULL),(26118,NULL,'Taletrectinib is an orally available, small molecule ROS1/ NTRK tyrosine kinase inhibitor. There are promising clinical data in patients with ROS1 G2032R mutant non-small cell lung cancer (NSCLC) treated with taletrectinib. In the Phase II TRUST-I (NCT04395677) trial of taletrectinib in 173 patients with ROS1-positive NSCLC, patients with ROS1 G2023R mutant NSCLC (n=12) demonstrated an objective response rate of 66.7% (95% CI=34.89-90.08) (PMID: 38822758). In vivo studies with ROS1 fusion-positive xenograft models expressing ROS1 G2023R demonstrate sensitivity to taletrectinib as measured by tumor regression and inhibition of ROS1 autophosphorylation (PMID: 31399568).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2024-09-25 15:00:14',NULL,'LEVEL_3A','NO','LEVEL_3B','e7caa0a1-9ee3-45dc-a71c-21d57bdbc3f7',6098,NULL),(26119,NULL,'Crizotinib is an orally available, small molecule multitarget kinase inhibitor that is FDA-approved for treatment of patients with metastatic non-small cell lung cancer (NSCLC) whose tumors are ALK or ROS1-positive, as detected by an FDA-approved test. ROS1-rearrangement for treatment with crizotinib were detected by the Oncomine Dx Target Test. ROS1 G2032R has been identified as an acquired resistance mutation in patients with ROS1-mutant NSCLC upon disease progression with crizotinib treatment. In multiple clinical studies, ROS1 G2032R was identified in 22 patients with ROS1 fusion-positive NSCLC following crizotinib treatment (PMID: 33685866, 29333528, 23724914). In vitro studies with ROS1 fusion-positive Ba/F3 cells expressing ROS1 G2032R demonstrate resistance to crizotinib as measured by sustained kinase activity (PMID: 32918045). In vitro enzymatic assays demonstrate that the crizotinib concentration needed to achieve 50% enzyme inhibition was increased by a factor of 270 for ROS1 G2032R compared to wildtype ROS1 (PMID: 23724914).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_RESISTANCE','LEVEL_Fda3',_binary '\0','Resistant','2024-09-25 15:42:34',NULL,'LEVEL_R2',NULL,NULL,'14ec85c2-12a8-4e75-8d89-0370fce464f4',6098,NULL),(26120,NULL,'Lorlatinib is an orally available, small molecule multi-targeted kinase inhibitor that is listed in the NCCN Non-Small Cell Lung Cancer Guidelines (v10.2024) under \"Subsequent therapy\" for patients with ROS1-rearranged non-small cell lung cancer (NSCLC). ROS1 G2032R has been identified as an acquired resistance mutation in patients with ROS1-mutant NSCLC upon disease progression with lorlatinib treatment. In multiple clinical studies, ROS1 G2032R was identified in ten patients with ROS1 fusion-positive NSCLC following lorlatinib treatment (PMID: 36135089, 33685866). In vitro studies with ROS1 fusion-positive Ba/F3 cells expressing ROS1 G2032R demonstrate moderate resistance to lorlatinib as measured by the IC50 values for lorlatinib increasing by a factor of 281 for ROS1 G2032R compared to wildtype ROS1 (PMID: 33685866).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_RESISTANCE','LEVEL_Fda3',_binary '\0','Resistant','2024-09-25 15:43:19',NULL,'LEVEL_R2',NULL,NULL,'86c0b34f-ac48-4ead-b6b2-e56aacc23c81',6098,NULL),(26121,NULL,'While there are promising clinical data in patients with ROS1 G2032R mutant non-small cell lung cancer treated with the ROS1/NTRK tyrosine kinase inhibitor taletrectinib, the clinical utility of this inhibitor in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-10-22 21:13:35',NULL,NULL,NULL,NULL,'2d79fca6-25d3-4232-8c71-28d6632ddb5f',6098,NULL),(26122,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Resistance','2022-12-09 12:52:29',NULL,NULL,NULL,NULL,'a98d8d63-c524-4fec-8aba-6e79097123f5',6098,NULL),(26123,NULL,'The ROS1 L1951R mutation lies in the tyrosine kinase domain of the ROS1 protein. In vitro studies in a CD74-ROS1-transformed cell line demonstrated that L1951R confers resistance to crizotinib and ceritinib as measured by cell viability and pathway activation assays. This effect is likely due to steric hindrance (PMID: 25351743).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'2b2b259b-8e00-4bf4-ad6d-079ad7a9cc6d',6098,NULL),(26124,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'9bf7d605-86c1-4fe3-b1de-700e4079c29b',6098,NULL),(26125,NULL,'The ROS1 D2033N mutation lies in the tyrosine kinase domain of the ROS1 protein. It has been detected in a patient with CD74-ROS1-fusion positive lung cancer with acquired resistance to crizotinib. In a murine B-cell line a D2033N-mutated CD74-ROS1 construct conferred reduced sensitivity to crizotinib while cabozantinib was equally effective as against the unmutated CD74-ROS, as measured in growth and downstream signaling assays. Experiments also supported the notion that D2033N confers resistance to ceritinib, brigatinib and lorlatinib. The analogous mutation in ALK (D1203N) confers crizotinib resistance in vitro and molecular studies indicate that mutations at these residues are important due to their being located at binding surface for ATP (PMID: 26673800).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'26191541-33bd-46d7-834b-288cf0f9a380',6098,NULL),(26126,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'5cc8c16c-c571-48dc-9448-0cb47a29623d',6098,NULL),(26127,NULL,'The ROS1 C2060G mutation lies in the tyrosine kinase domain of the ROS1 protein. In a murine B-cell line a C2060G-mutated FIG-ROS1 construct conferred increased resistance to crizotinib, as compared to the unmutated construct in cell-growth assays. The mutated construct showed slightly reduced sensitivity to the inhibitor foretinib (PMID: 24218589).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'ba413320-b38d-4bed-9dce-2cfd6c64f434',6098,NULL),(26128,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'990fad2a-aa8d-4650-9455-430979acdb9f',6098,NULL),(26129,NULL,'The ROS1 E1935G mutation occurs in an uncharacterized region ten amino acid residues prior to the tyrosine kinase domain of the ROS1 protein. In a murine B-cell line a E1935G-mutated FIG-ROS1 construct conferred increased resistance to crizotinib, as compared to the unmutated construct in cell-growth assays. The mutated construct showed slightly reduced sensitivity to the inhibitor foretinib (PMID: 24218589).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'55739fa0-b5b6-4f08-a975-e1769dfdf13e',6098,NULL),(26130,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'21bd828d-75e5-480e-ba46-f9c44f74205d',6098,NULL),(26131,NULL,'The ROS1 G1971E mutation lies in the tyrosine kinase domain of the ROS1 protein. In a murine B-cell line a G1971E-mutated FIG-ROS1 construct conferred increased resistance to crizotinib, as compared to the unmutated construct in cell-growth assays. The mutated construct showed slightly reduced sensitivity to the inhibitor foretinib (PMID: 24218589).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'9ab26a40-7f6f-4ffd-b850-74576f925345',6098,NULL),(26132,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'912e61e9-f663-4532-adf8-3fce18e85ac9',6098,NULL),(26133,NULL,'The ROS1 L1947R mutation occurs in the protein\'s kinase domain and is often identified in the context of a ROS1 fusion protein. FIG-ROS1 Ba/F3 cells expressing the L1947R mutation demonstrated increased resistance to crizotinib compared to control FIG-ROS1 Ba/F3 cells, as measured by sustained cell growth and protein activation upon drug treatment (PMID: 24218589).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'93af185b-5b37-4c69-b631-b40c9168468d',6098,NULL),(26134,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'890afecb-328f-42fe-9f35-0ded40d95c9e',6098,NULL),(26135,NULL,'The ROS1 V2098I mutation occurs in the tyrosine kinase domain of the ROS1 protein at a site implicated in inhibitor binding. In a murine B-cell line a V2098I -mutated FIG-ROS1 construct conferred increased resistance to crizotinib, as compared to the unmutated construct in cell-growth assays. The mutated construct showed slightly reduced sensitivity to the inhibitor foretinib (PMID: 24218589).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'89a9587b-402b-409d-8c46-f4c7c83d883e',6098,NULL),(26136,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'278e7a1f-c695-4df6-ad75-869b8462aec0',6098,NULL),(26137,NULL,'The ROS1 G2101A mutation lies in the tyrosine kinase domain of the ROS1 protein. In a murine B-cell line a G2101A-mutated CD74-ROS1 construct conferred increased resistance to crizotinib, as compared to the unmutated construct in cell-growth assays. The mutated construct showed slightly reduced sensitivity to the inhibitor foretinib. The variants is analogous to the previously described ALK G1269A resistance mutation (PMID: 25688157, 24675041, 22235099).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'a05a2f36-dfef-4578-a479-e291ed0355a7',6098,NULL),(26138,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'385c8437-b18a-4819-beab-20ae3cb25ae6',6098,NULL),(26139,NULL,'The ROS1 S1986Y mutation lies in tyrosine kinase domain the of the ROS1 protein. It has been detected in a patient with ROS1-fusion positive lung cancer in concert with mutation S1986F. The two alterations at the same residue were associated with resistance to crizotinib but subsequent sensitivity to lorlatinib. In a murine B-cell line expression of individual S1986F- and S1986Y-mutated ROS1-fusion constructs were resistant to crizotinib and ceritinib but not lorlatinib, as assessed by cell viability and pathway activation assays (PMID: 27401242).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'2e14998d-6a5e-4d9d-982d-90d7d47e51d8',6098,NULL),(26140,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'bcd08c5b-9fe6-46e6-aace-af80088b9703',6098,NULL),(26141,NULL,'The ROS1 S1986F mutation lies in the tyrosine kinase domain of the ROS1 protein. It has been detected in a patient with ROS1-fusion-positive lung cancer in concert with the S1986Y mutation. The two alterations at the same residue were associated with resistance to crizotinib but subsequent sensitivity to lorlatinib. In a murine B-cell line expression of individual S1986F- and S1986Y-mutated ROS1-fusion constructs were resistant to crizotinib and ceritinib but not lorlatinib, as assessed by cell viability and pathway activation assays (PMID: 27401242).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2022-12-07 17:31:30',NULL,NULL,NULL,NULL,'19acaaee-d519-4577-9b10-2ea228c116b3',6098,NULL),(26142,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'1d70db26-7ecb-4039-c0dc-4c97bd823c89',6098,NULL),(26143,NULL,'Amplification of ROS1 has been shown to result in an increase in ROS1 transcript and ROS1 protein levels in cancer cell lines that is sensitive to ROS1 inhibition demonstrating that ROS1 amplification is likely activating (PMID: 25978031). Studies in cancer cell lines have demonstrated that ROS1 amplification can lead to resistance to EGFR inhibitors in cells that are otherwise sensitive to EGFR inhibition (PMID: (PMID: 25978031).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-10 14:55:05',NULL,NULL,NULL,NULL,'58511c90-26db-438e-b897-0bdea61f61af',6098,NULL),(26144,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2019-04-22 01:53:11',NULL,NULL,NULL,NULL,'f34c6773-0f41-42b0-866d-8a7c3b3b694b',6098,NULL),(26145,NULL,'The GOPC-ROS1 fusion protein combines the N terminus of GOPC until exon 7 with the C-terminus of ROS1 starting at exon 35 and is found in CNS tumors (PMID: 30171048). This is the most frequent ROS1 fusion found in glioblastomas (GBMs). Expression of this fusion protein in BaF3 and immortalized human astrocyte cells demonstrated that it is activating as shown by IL3 independent growth and increased colony formation compared to ectopic expression of wildtype ROS1 (PMID: 30171048). Expression of this fusion in BaF3 cells also demonstrated that it is sensitive to multiple TKIs including crizotinib, foretinib, cabozantinib, ceritinib, brigatinib, AZD3463, lorlatinib, and entrectinib as measured by proliferation, pathway activation and colony formation with lorlatinib having the highest potency (PMID: 30171048). In a case report, a patient with ER-positive HER2-negative breast cancer harboring GOPC-ROS1 demonstrated an exceptionally response to crizotinib therapy and a subsequent modest response to cabozantinib after development of crizotinib resistance (PMID: 37487149).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2023-12-05 13:16:33',NULL,NULL,NULL,NULL,'a95bd9a0-ce83-4a4b-8782-c9cdff1175b1',6098,NULL),(26146,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2019-10-02 21:39:48',NULL,NULL,NULL,NULL,'21c81568-8f05-4de9-bed2-749a8b6bf283',6098,NULL),(26147,NULL,'The ROS1 F2004V mutation is located in the kinase domain of the protein. This mutation was found as a secondary mutation in a patient with lung adenocarcinoma who had progressive disease in the brain and bones after treatment with crizotinib, entrectinib and brigatinib but who had a dramatic response to treatment with lorlatinib (Abstract: Dimou et al. JCO PO, 2019. https://ascopubs.org/doi/full/10.1200/PO.19.00013). This mutation may confer resistance to entrectinib and brigatinib but remains sensitive to lorlatinib. Structural studies based on homology to ALK predict that ROS1 F2004V would confer resistance to ceritinib (PMID: 27401242).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2021-01-14 00:21:45',NULL,NULL,NULL,NULL,'c74bbb3d-42fc-42f1-b262-d02131bf4837',6098,NULL),(26148,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2021-01-14 00:57:23',NULL,NULL,NULL,NULL,'fbc05162-1285-449a-a3b7-54a791068d3f',6098,NULL),(26149,NULL,'The ROS1 F2004C mutation is located in the kinase domain of the protein. This mutation has been found in lung cancer (Abstract: Doebele et al. Abstract# LBA-28, ESMO 2019. https://www.sciencedirect.com/science/article/pii/S0923753419603777). Expression of this mutation in the context of a ROS1 fusion in Ba/F3 cells demonstrated decreased sensitivity to ROS1 inhibition (PMID: 26372962). Structural studies based on homology to ALK predict that ROS1 F2004C would confer resistance to ceritinib (PMID: 27401242). At least one patient with ROS1-fusion-positive non-small cell lung cancer acquired the ROS1 F2004C mutation upon disease progression on treatment with entrectinib (Abstract: Doebele et al. Abstract# LBA-28, ESMO 2019. https://www.sciencedirect.com/science/article/pii/S0923753419603777).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2021-02-02 23:45:06',NULL,NULL,NULL,NULL,'e558dfec-5cdb-431e-8f7a-9a3d30be0f41',6098,NULL),(26150,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Resistance','2024-10-21 16:11:40',NULL,NULL,NULL,NULL,'1b04cf14-2faf-41ea-b753-c45bec17b043',6098,NULL),(26151,NULL,'The ROS1 L2086F mutation is located in the kinase domain of the protein. This mutation has been identified in non-small cell lung cancer (NSCLC), and is associated with acquired resistance to lorlatinib and repotrectinib in ROS1-rearranged NSCLC (PMID: 32321774, 33685866, 39091594). Although this mutation has not been functionally characterized, patients with ROS1-rearranged NSCLC harboring ROS1 L2086F have demonstrated sensitivity to cabozantinib (PMID: 33685866, 37291202, 39117775, 39091594).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2025-03-19 12:25:29',NULL,NULL,NULL,NULL,'d796b0a7-6a7e-4a7d-bbad-37282de9abc7',6098,NULL),(26152,NULL,'The ROS1 L2086F mutation has been identified in ROS1-mutant non-small cell lung cancer (NSCLC) collected following disease progression on the tyrosine kinase inhibitors lorlatinib and repotrectinib. There are promising clinical data in patients with ROS1 L2086F mutant NSCLC treated with the tyrosine kinase inhibitor cabozantinib.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-09-25 16:06:24',NULL,NULL,NULL,NULL,'7e1f0e76-4191-4cac-8c3a-636a31e3efa3',6098,NULL),(26153,NULL,'Cabozantinib is an orally available, small molecule multi-kinase inhibitor. There are promising clinical data in patients with ROS1 L2086F mutant non-small cell lung cancer (NSCLC) treated with cabozantinib. Across multiple case studies, patients with ROS1 fusion-positive NSCLC expressing ROS1 L2086F were treated with cabozantinib and achieved responses ranging from partial response (PR) (n=2) (PMID: 39117775, 39091594) to stable disease (n=2) (PMID: 33685866, 37291202). For the 2 PR, one patient was previously treated with entrectinib and one patient was previously treated with repotrectinib, at which time the ROS1 L2086F mutation was detected and cabozantinib treatment was started (PMID: 39117775, 39091594). In vitro studies with ROS1 fusion-positive patient-derived cells expressing ROS1 L2086F demonstrate sensitivity to cabozantinib as measured by inhibition of cellular proliferation and ROS1 autophosphorylation (PMID: 33685866, 39117775, 39091594).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2024-10-10 15:09:44',NULL,'LEVEL_3A','NO','LEVEL_3B','13a6f1ba-3659-4324-9840-a69b20add57d',6098,NULL),(26154,NULL,'Lorlatinib is an orally available, small molecule multi-targeted kinase inhibitor that is listed in the NCCN Non-Small Cell Lung Cancer Guidelines (v10.2024) under \"Subsequent therapy\" for patients with ROS1-rearranged non-small cell lung cancer (NSCLC). ROS1 L2086F has been identified as an acquired resistance mutation in patients with ROS1-mutant NSCLC upon disease progression with lorlatinib treatment. In multiple clinical studies, ROS1 L2086F was identified in four patients with ROS1 fusion-positive NSCLC following lorlatinib treatment (PMID: 37291202, 33685866). In vitro studies with ROS1 fusion-positive Ba/F3 cells expressing ROS1 L2086F demonstrate resistance to lorlatinib as measured by the IC50 values for lorlatinib increasing by a factor of over 4286 for ROS1 L2086F compared to wildtype ROS1 (PMID: 33685866).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_RESISTANCE','LEVEL_Fda3',_binary '\0','Resistant','2024-09-25 16:27:42',NULL,'LEVEL_R2',NULL,NULL,'5504977a-2135-4f69-89e2-8c5c51829046',6098,NULL),(26155,NULL,'Repotrectinib is an orally available, small molecule next-generation ROS1/TRK tyrosine kinase inhibitor (TKI) that is FDA-approved for the treatment of adult patients with locally advanced or metastatic ROS1-positive non-small cell lung cancer (NSCLC). In a clinical study, ROS1 L2086F was identified in a patient with ROS1 fusion-positive NSCLC following repotrectinib treatment (PMID: 39091594). The patient had previously received crizotinib, ceritinib, carboplatin plus pemetrexed, and lorlatinib, with no other ROS1 mutations detected at the time of disease progression on repotrectinib (PMID: 39091594). In vitro studies with ROS1 fusion-positive Ba/F3 cells expressing ROS1 L2086F demonstrate resistance to repotrectinib as measured by the IC50 values for repotrectinib increasing by a factor of 840 for ROS1 L2086F compared to wildtype ROS1 (PMID: 33685866).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_RESISTANCE','LEVEL_Fda3',_binary '\0','Resistant','2024-10-10 15:20:55',NULL,'LEVEL_R2',NULL,NULL,'cb7f1965-25a7-4d7e-8be5-f841d75d4248',6098,NULL),(26156,NULL,'While there are promising clinical data in patients with ROS1 L2086F mutant non-small cell lung cancer treated with the tyrosine kinase inhibitor cabozantinib, the clinical utility of this inhibitor in patients with [[variant]] is unknown.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-10-22 21:15:30',NULL,NULL,NULL,NULL,'2a40401e-384f-45c9-b7ad-29a6c8d4d96f',6098,NULL),(26157,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-09 12:22:22',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26158,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-10 11:25:28',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26159,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-10 11:26:19',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26160,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-17 08:00:32',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26161,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-04-07 14:31:34',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26162,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-04-10 14:54:54',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26163,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-05-01 16:08:19',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26164,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2019-04-22 01:58:40',NULL,NULL,NULL,NULL,NULL,6098,NULL),(26165,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2020-06-10 00:28:04',NULL,NULL,NULL,NULL,NULL,6098,NULL),(28517,NULL,'TP53, a tumor suppressor in the DNA damage pathway, is the most frequently mutated gene in cancer.','GENE_SUMMARY',NULL,_binary '\0',NULL,'2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'1238efb3-3419-4254-a8ca-c810ff7b2696',7157,NULL),(28518,NULL,'TP53 encodes the p53 tumor suppressor protein, a transcription factor that responds to cellular stresses, including DNA damage and oncogenic activation, by inducing downstream anti-tumor responses such as DNA repair and apoptosis (PMID: 11099028). p53 levels are kept low in healthy cells due to negative regulation by MDM2/4, Cop1 and Trim24 and constant degradation by the ubiquitin-proteasome system (PMID: 36859359, 36207426). When DNA is damaged, a network of pathways is activated to detect and repair lesions in a cell- and context-specific manner (PMID: 36207426). p53 is rapidly phosphorylated by upstream regulators such as ATM, ATR, and CHL1/2, which results in the accumulation of stable p53 (PMID: 36207426). p53 then binds to specific DNA sequences to direct the expression of a wide variety of genes, including those involved in apoptosis, cell cycle arrest, DNA repair, senescence, stem cell differentiation, autophagy, cellular metabolism, and others (PMID: 27141080, 36859359, 36207426). Oncogenic mutations of TP53 often result in the dysregulation of p53 function, usually due to structural changes in the DNA binding domain (PMID: 36859359). Loss of p53 function can have various outcomes including tumorigenesis, invasion and metastasis, drug resistance, metabolic reprogramming, immune evasion and overall genomic instability (PMID: 36859359). TP53 is the most commonly mutated gene in human cancers, and germline mutations occur in the cancer predisposition syndrome Li-Fraumeni (PMID: 22713868, 21765642). Clinical and preclinical research into drugs that target TP53 is ongoing, notably with MDM2 inhibitors that aim to restore p53 function and are being tested in combination with other cancer therapies (PMID: 36859359, 37818252).','GENE_BACKGROUND',NULL,_binary '\0',NULL,'2024-06-05 14:33:01',NULL,NULL,NULL,NULL,'24a179db-26d2-45e2-80b9-71b1eb49a491',7157,NULL),(28519,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2018-12-12 12:41:44',NULL,NULL,NULL,NULL,'dd554ea2-478a-4ff5-c094-c2c74fa2032a',7157,NULL),(28520,NULL,'The TP53 P322R mutation is located in the oligomerization domain of the protein. In vitro studies have demonstrated that this mutation is likely neutral, as evidenced by its ability to oligomerize and transactivate downstream targets comparable to wildtype TP53 (PMID: 16007150).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2022-08-21 20:27:41',NULL,NULL,NULL,NULL,'682b6f29-3299-48fb-9d56-4449fb6aab66',7157,NULL),(28521,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-05-09 15:34:57',NULL,NULL,NULL,NULL,'cdcc69c1-9be4-4f56-a7d0-5e2c392561c9',7157,NULL),(28522,NULL,'The TP53 H214R mutation is located in the protein\'s DNA binding domain and is a known hotspot mutation (PMID: 26619011). This mutation has been found in breast cancers, among others (PMID: 20407015). In vivo studies with yeast expressing TP53 H214R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). Other in vivo studies with yeast expressing TP53 H214R demonstrate that the mutation is also inactivating as measured by partial loss of transactivational function compared to wildtype (PMID: 11920959). In vitro studies with various human cancer cell lines expressing TP53 H214R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644). In contrast, in vitro studies have demonstrated that this mutation is likely activating, as evidenced by increased TP53 transactivation activity compared to wildtype (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-05-09 15:34:59',NULL,NULL,NULL,NULL,'1d39ff8b-94cf-4c03-8a82-61886e2b9046',7157,NULL),(28523,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'10e5d16d-6ee4-4985-a46b-366b821f1ae8',7157,NULL),(28524,NULL,'The TP53 R175H mutation is located in the protein\'s DNA binding domain. This mutation has been found in acute myeloid leukemia (PMID: 31068365). In vitro and in vivo studies have demonstrated that this mutation is inactivating and oncogenic, as measured by the reduced ability to induce apoptosis and form colonies, as well as increased cell migration and faster progression of hematopoietic malignancy of the mutant compared to TP53 wildtype or deletion (PMID: 15781620, 25584008, 31068365). Structural studies have also shown that this mutant is defective in protein folding and DNA binding (PMID: 10713666, 21445056).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-08-09 21:59:59',NULL,NULL,NULL,NULL,'41c4e951-4d83-4e4a-9843-3f64039400ea',7157,NULL),(28525,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-15 12:21:38',NULL,NULL,NULL,NULL,'afc0e256-7933-4e36-9933-50e32b8a77d5',7157,NULL),(28526,NULL,'The TP53 S121F mutation is located in the protein\'s DNA binding domain. This mutation has been found in lymphoma (PMID: 22955915). In vitro studies have demonstrated that this mutation is likely activating, as evidenced by enhanced transcriptional activity, increased apoptosis and inhibition of cell cycle progression in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2022-08-24 14:21:19',NULL,NULL,NULL,NULL,'1a110d13-bfb1-4080-b89e-e1da6d04bb87',7157,NULL),(28527,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-15 11:41:41',NULL,NULL,NULL,NULL,'e5eb8bf2-4c0f-4432-b6f9-f9efc1020164',7157,NULL),(28528,NULL,'The TP53 H214Q mutation is located in the DNA binding domain of the protein. This mutation has been found in colorectal carcinoma and is a statistically significant hotspot (PMID: 26619011, 11044641). In vitro studies have demonstrated that this mutation is likely activating, as evidenced by significantly increased apoptosis, decreased cell viability and altered transcription of TP53 target genes in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-07-23 13:32:49',NULL,NULL,NULL,NULL,'204f9e50-8e83-45a7-a3ba-64c847535564',7157,NULL),(28529,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2022-09-29 11:15:05',NULL,NULL,NULL,NULL,'3a9cc731-75b8-4639-84d7-853aa820d2a1',7157,NULL),(28530,NULL,'The TP53 K291E mutation is located in the DNA binding domain of the protein. This mutation has been found in non-small cell lung cancer and is a statistically significant hotspot mutation (PMID: 26619011, 9416838). In vitro studies have demonstrated that this mutation is likely activating, as evidenced by significantly increased apoptosis, decreased cell viability and altered transcription of TP53 target genes in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-07-23 13:26:07',NULL,NULL,NULL,NULL,'5fc85227-bb09-490f-8adc-cbc68fa48ac1',7157,NULL),(28531,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2022-10-03 11:59:23',NULL,NULL,NULL,NULL,'d1b01d13-6e1d-4aee-a800-d70b602d3247',7157,NULL),(28532,NULL,'The TP53 K292T mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely activating, as evidenced by significantly increased apoptosis, decreased cell viability and altered transcription of TP53 target genes in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2022-10-03 11:59:34',NULL,NULL,NULL,NULL,'8698b568-758c-4318-8054-28515ca884c8',7157,NULL),(28533,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2022-10-03 12:02:33',NULL,NULL,NULL,NULL,'96f664ad-8758-4b98-9a1e-7648ce877649',7157,NULL),(28534,NULL,'The TP53 Q144R mutation is located in the protein\'s DNA binding domain and is a known hotspot mutation (PMID: 26619011). However, in vitro studies have demonstrated that this mutation is likely activating, as evidenced by significantly increased apoptosis, decreased cell viability and altered transcription of TP53 target genes in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2022-10-03 12:02:34',NULL,NULL,NULL,NULL,'e113fd62-9072-4307-9b2c-3f42dac5e306',7157,NULL),(28535,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2022-08-21 22:48:37',NULL,NULL,NULL,NULL,'3a8903db-dc5d-47e8-8962-98d13cdf24e8',7157,NULL),(28536,NULL,'The TP53 I162M mutation is located in the protein\'s DNA binding domain. This mutation has been found in lymphoma (PMID: 27101868). In vitro studies have demonstrated that this mutation has inconclusive effect, as evidenced by altered transcription of TP53 target genes as well as an insignificant increased apoptosis and cell cycle arrest in the mutant compared to wildtype TP53 (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2022-08-21 22:50:26',NULL,NULL,NULL,NULL,'a84c15a9-c0ef-4a97-a7bd-a737fecb67c4',7157,NULL),(28537,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2022-09-29 11:14:57',NULL,NULL,NULL,NULL,'48f340e6-0711-4a50-996a-f0c8d403b9c4',7157,NULL),(28538,NULL,'The TP53 R290G mutation is located in the protein\'s DNA binding domain (PMID: 26619011). In vitro studies have demonstrated that this mutation is likely activating, as evidenced by significantly increased apoptosis, decreased cell viability and altered transcription of TP53 target genes in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2022-10-03 14:28:10',NULL,NULL,NULL,NULL,'7e7a3ae5-6c0c-45a4-85c9-d3bc9042a818',7157,NULL),(28539,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8fb3a0c0-971c-445b-ab06-23c6d7b58f14',7157,NULL),(28540,NULL,'The TP53 L344P mutation is located in the protein\'s oligomerization domain. This mutation has been found in Li-Fraumeni syndromes (PMID: 9704930). In vitro and structural studies have demonstrated that this mutation is inactivating, as evidenced by the mutant\'s inability to oligomerize and form TP53 tetramers, leading to decreased transactivation of TP53 target genes, decreased TP53 acetylation and decreased recruitment of p300 in the mutant compared to wildtype (PMID: 19454241, 19106109, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-14 21:16:52',NULL,NULL,NULL,NULL,'b4f92878-cffd-4450-bfd4-7a584ef43fbc',7157,NULL),(28541,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c6d759ed-e514-4b33-9b60-28fd926b6ab1',7157,NULL),(28542,NULL,'The TP53 R337C mutation is located in the protein\'s oligomerization domain (PMID: 17311302). In vitro and structural studies have demonstrated that this mutation is inactivating, as evidenced by the mutant\'s ability to inhibit transcriptional activity by destabilizing the p53 tetramer, as well as its inability to oligomerize, leading to decreased transactivation of TP53 target genes, decreased TP53 acetylation and decreased recruitment of p300 in the mutant compared to wildtype (PMID: 19454241, 11753428, 16007150, 19106109).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-14 22:14:00',NULL,NULL,NULL,NULL,'65c2d4a3-1211-4f05-9543-d8c5173e0996',7157,NULL),(28543,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'48b1d3a6-c7c3-45b5-a3b9-a7c1dc912f34',7157,NULL),(28544,NULL,'The TP53 L330H mutation is located in the protein\'s oligomerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transcriptional activity, decreased protein stability and decreased p53 tetramer formation in the mutant compared to wildtype (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 21:38:00',NULL,NULL,NULL,NULL,'a0a3bd7b-79bd-4176-a99c-c5ea8aebdbcc',7157,NULL),(28545,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'78f23675-e3a1-4f6b-9d75-d413f954312a',7157,NULL),(28546,NULL,'The TP53 L330P mutation is located in the protein\'s oligomerization domain. Structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:04:04',NULL,NULL,NULL,NULL,'ea95c3b8-e97a-43c8-9dba-216a2853477f',7157,NULL),(28547,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5c990d0d-c6e1-4dc6-95f9-330aa3a1b01b',7157,NULL),(28548,NULL,'The TP53 L330R mutation is located in the protein\'s tetramerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as by the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:15:57',NULL,NULL,NULL,NULL,'13f49980-7683-47d3-992e-27ce5f82dd57',7157,NULL),(28549,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-07 12:44:54',NULL,NULL,NULL,NULL,'cc8ff4ec-c683-47e6-8e65-2545188bf3bf',7157,NULL),(28550,NULL,'The TP53 R337P mutation is located in the protein\'s oligomerization domain. This mutation has been found in the germline of patients with Li-Fraumeni syndrome and adrenocortical carcinoma (PMID: 11481490). In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-16 14:52:48',NULL,NULL,NULL,NULL,'f87dbe53-d8b1-4367-95f4-6bb3dd657587',7157,NULL),(28551,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9ce55e0f-9b63-4c75-9b13-11c230da8d2f',7157,NULL),(28552,NULL,'The TP53 R342P mutation is located in the protein\'s oligomerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as by the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-14 22:16:49',NULL,NULL,NULL,NULL,'124e09ba-4bf7-4853-98b6-b678eec9fbdc',7157,NULL),(28553,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8ca0b47c-31dd-4258-8b9f-8618f4e7c3d2',7157,NULL),(28554,NULL,'The TP53 F341C mutation is located in the protein\'s oligomerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as by the mutant\'s decreased ability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:18:31',NULL,NULL,NULL,NULL,'10cead43-d239-4486-a664-7dacc7401fdf',7157,NULL),(28555,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e0150809-ab63-452f-8ab3-ee6daccfbada',7157,NULL),(28556,NULL,'The TP53 L344R mutation is located in the protein\'s oligomerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as by the mutant\'s decreased ability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:19:21',NULL,NULL,NULL,NULL,'97dd3cee-daea-46cc-a9e4-5a7dc823273e',7157,NULL),(28557,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'031ed56b-d825-4cf6-8883-3e3cf0b12bbb',7157,NULL),(28558,NULL,'The TP53 A347T mutation is located in the protein\'s oligomerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as by the mutant\'s decreased ability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:33:44',NULL,NULL,NULL,NULL,'c50c6631-3785-4b6e-8d43-a6c2bed68d76',7157,NULL),(28559,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-07 12:45:54',NULL,NULL,NULL,NULL,'4a6fd479-7ecb-42b5-b271-7f217243a070',7157,NULL),(28560,NULL,'The TP53 R337H mutation is located in the protein\'s oligomerization domain. This mutation has been found in the germline of patients with Li-Fraumeni syndrome and adrenocortical carcinoma (PMID: 11481490). In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-16 14:55:10',NULL,NULL,NULL,NULL,'b14ddc8c-73d9-4fe1-b08c-e8fa522bd10a',7157,NULL),(28561,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-07 12:46:30',NULL,NULL,NULL,NULL,'46db9414-aee3-48c9-9a02-9e3e1821654a',7157,NULL),(28562,NULL,'The TP53 R337L mutation is located in the protein\'s oligomerization domain. This mutation has been found in the germlines of patients with Li-Fraumeni syndrome and adrenocortical carcinoma (PMID: 11481490). In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:37:23',NULL,NULL,NULL,NULL,'a9ed758b-0cc4-4168-b1fe-f0f98d6d92e0',7157,NULL),(28563,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'55b67ba9-3f52-4ed8-ae03-2406967fe4e1',7157,NULL),(28564,NULL,'The TP53 F338I mutation is located in the protein\'s oligomerization domain. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased transcriptional activity and protein stability in the mutant compared to wildtype, as well as by the mutant\'s inability to form p53 tetramers, which is required for TP53 DNA binding activity (PMID: 19454241, 20978130)','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:38:52',NULL,NULL,NULL,NULL,'06cd006b-82bd-4d4c-bf31-4ced5518df1a',7157,NULL),(28565,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'4da6625f-6dc7-474a-9f57-b0e74f44e1bd',7157,NULL),(28566,NULL,'The TP53 L348S mutation is located in the protein\'s oligomerization domain. Structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the mutant\'s decreased ability to form TP53 tetramers compared to wildtype TP53, which likely impacts TP53 DNA binding activity (PMID: 20978130).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:43:37',NULL,NULL,NULL,NULL,'2b5e984a-900d-4850-9dcd-cbe5d6cd0dfe',7157,NULL),(28567,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d2c2e9e7-d4cf-40ab-b1c9-38182840e967',7157,NULL),(28568,NULL,'The TP53 P47S mutation is located in the protein\'s transactivation domain. This mutation has been found in the germline of patients with pediatric adrenocortical carcinoma (PMID: 25584008). In vitro studies have demonstrated that this mutation is likely neutral, as evidenced by similar levels of colony formation and TP53 transactivation activity in the mutant compared to wildtype (PMID: 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Neutral','2022-09-16 14:56:26',NULL,NULL,NULL,NULL,'0d1f4bb9-af36-433b-8a35-3eca8f862227',7157,NULL),(28569,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-15 11:39:08',NULL,NULL,NULL,NULL,'2815bd4b-1619-4f67-8856-b5bfe65d1d94',7157,NULL),(28570,NULL,'The TP53 F134Y mutation is located in the protein\'s DNA binding domain and occurs at a known hotspot residue (PMID: 26619011). In vitro studies have demonstrated that this mutation has inconclusive effect, as evidenced by similar levels of colony formation and TP53 transactivation activity in the mutant compared to wildtype, even though its position at a hotspot residue suggests that it is likely oncogenic (PMID: 25584008, 26619011).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-02-15 11:39:09',NULL,NULL,NULL,NULL,'4bc2942b-10c2-4f3f-babe-bc76bbdb2b7d',7157,NULL),(28571,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'00393141-cdd1-4ab5-9741-1bb9feb5ece8',7157,NULL),(28572,NULL,'The TP53 P152L mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by loss of p53-mediated transactivation and reduced colony formation in the mutant compared to wildtype (PMID: 10229196, 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-14 22:56:46',NULL,NULL,NULL,NULL,'cb67c3e4-a16b-416c-a5ee-4a1a89cfcb19',7157,NULL),(28573,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'39b8f59b-cd28-402d-ab72-8b7e6fa4700a',7157,NULL),(28574,NULL,'The TP53 R158H mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by loss of p53-mediated transactivation, decreased ability to induce apoptosis and increased oncogenic colony transformation of the mutant compared to wildtype (PMID: 10229196, 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-10-03 14:29:52',NULL,NULL,NULL,NULL,'94bb01ea-f180-4306-8cd9-cab503e1bbcd',7157,NULL),(28575,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8536d498-deef-47ed-a102-1b94647aeb32',7157,NULL),(28576,NULL,'The TP53 R158L mutation is located in the protein\'s DNA binding domain. This mutation has been found in lung cancer (PMID: 31067569). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by loss of TP53-mediated transactivation, decreased ability to reduce colony formation and aberrant transcriptional regulation in the mutant compared to wildtype (PMID: 25584008, 16861262, 31067569).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-10-03 15:23:58',NULL,NULL,NULL,NULL,'e833d643-3b09-43c2-ba46-3c342042ddcf',7157,NULL),(28577,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8a3f9e27-ab57-4697-9fa4-97a1239fa9e6',7157,NULL),(28578,NULL,'The TP53 Y163C mutation is located in the protein\'s DNA-binding domain. In vitro studies have demonstrated that this mutation is inactivating, as measured by increased cellular growth, decreased levels of apoptosis and decreased transactivation of downstream p53 target genes in the mutant compared to wildtype (PMID: 23246812,15077194, 25584008, 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 22:34:19',NULL,NULL,NULL,NULL,'5d765c01-7b4d-40dc-a5a6-ad5f2eebeea9',7157,NULL),(28579,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c3faae77-9204-4160-9e91-470b2541935b',7157,NULL),(28580,NULL,'The TP53 H193P mutation is located in the protein\'s DNA-binding domain. This mutation has been found in Li-Fraumeni syndrome (PMID: 25584008). In vitro studies have demonstrated that this mutation is inactivating, as measured by decreased transactivation of p53 target genes and failure to suppress colony growth in a colony reduction assay in the mutant compared to wildtype (PMID: 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 22:40:00',NULL,NULL,NULL,NULL,'f7616b81-3c8c-488b-bfe3-f2f41f0981b4',7157,NULL),(28581,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'996354e0-9e2d-414a-a730-32538956157e',7157,NULL),(28582,NULL,'The TP53 P219S mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the loss of of p53-mediated transactivation, increased colony formation, decreased transactivation of genes important in cell cycle and apoptosis and decreased apoptosis compared to wildtype (PMID: 25584008, 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2023-03-23 15:34:54',NULL,NULL,NULL,NULL,'2f286414-2199-435b-be11-f12daee9f162',7157,NULL),(28583,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'27d46a94-299d-4d6e-adb0-58bfe98a5149',7157,NULL),(28584,NULL,'The TP53 C229R mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by increased p53-mediated transactivation and increased colony formation in the mutant compared to wildtype (PMID: 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-15 22:55:03',NULL,NULL,NULL,NULL,'2a27eee4-b03c-4090-8b70-4c063146286a',7157,NULL),(28585,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bc631aad-1012-40e9-bc63-4a2182c94274',7157,NULL),(28586,NULL,'The TP53 N235D mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by increased p53-mediated transactivation and increased colony formation in the mutant compared to wildtype (PMID: 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 22:54:55',NULL,NULL,NULL,NULL,'87426af5-e84a-4ed0-a5e1-6f2a394868c9',7157,NULL),(28587,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'4cd0dde0-3a2a-4e83-b1c5-fc852493ae90',7157,NULL),(28588,NULL,'The TP53 R248W mutation is located in the protein\'s DNA binding domain. This mutation has been found in Li-Fraumeni syndrome (PMID: 25584008). In vitro studies have demonstrated that this mutation is inactivating, as evidenced by increased p53-mediated transactivation and failure to suppress colony growth in the mutant compared to wildtype (PMID: 18762571, 10229196, 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-16 15:03:51',NULL,NULL,NULL,NULL,'2416b4ed-b136-4f3a-9fda-8a2f17032146',7157,NULL),(28589,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'b1d7458b-5abc-4467-a834-a83dc126f0cf',7157,NULL),(28590,NULL,'The TP53 I254T mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the loss of p53-mediated transactivation and increased colony formation in the mutant compared to wildtype (PMID: 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 22:58:19',NULL,NULL,NULL,NULL,'4afa0c35-0e9f-410e-8507-4dcc93431988',7157,NULL),(28591,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'41f37360-eee1-4f3a-bb83-76ff7f62d897',7157,NULL),(28592,NULL,'The TP53 R267Q mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by increased levels of p53-mediated transactivation (suggesting loss of negative feedback) and increased colony formation in the mutant compared to wildtype (PMID: 25584008, 10435620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-28 18:30:24',NULL,NULL,NULL,NULL,'c9ef7cfc-1216-4d66-ba5b-32c44a642b4f',7157,NULL),(28593,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'94b57c4a-5bb1-44d1-9819-4b0fd48c34b6',7157,NULL),(28594,NULL,'The TP53 R273H mutation is located in the protein\'s DNA binding domain. This mutation has been found in triple-negative breast cancer and Li-Fraumeni syndrome (PMID: 31776133, 25584008). Expression of this mutant resulted in cell survival by activating AKT signaling and suppressing the expression of BCL-modifying factor (BMF) (PMID: 26181206). Expression of this mutant in TP53-null cells resulted in decreased transactivation of p53 target genes, as measured by a luciferase reporter assay (PMID: 25584008, 15037740). Expression of this mutation in osteosarcoma and breast cancer cell lines demonstrated that it is inactivating as measured by moderately increased cell proliferation and failure to suppress colony growth compared to wildtype (PMID: 25584008, 31776133).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-28 18:32:56',NULL,NULL,NULL,NULL,'b54d2ced-283e-49fa-bfe2-8d3c01d34c73',7157,NULL),(28595,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'0b69cc7e-f05c-4306-9b34-f7620ad43bde',7157,NULL),(28596,NULL,'The TP53 E285V mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the loss of p53-mediated transactivation as well as defective growth suppression and apoptosis in the mutant compared to wildtype (PMID: 25584008, 18762572).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 23:05:17',NULL,NULL,NULL,NULL,'181e1d31-6d2e-49ee-bb6c-386b07892aea',7157,NULL),(28597,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'10cc2308-482b-41ad-8554-2a19e84a6a03',7157,NULL),(28598,NULL,'The TP53 R175L mutation is located in the protein\'s DNA binding domain and occurs at a hotspot residue (PMID: 26619011). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by colony formation and reduced ability to induce apoptosis in the mutant compared to wildtype (PMID: 25584008, 15781620, 18762572).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-16 15:54:52',NULL,NULL,NULL,NULL,'a288827a-589c-4f83-94a1-15027bd2329d',7157,NULL),(28599,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2022-09-15 23:14:17',NULL,NULL,NULL,NULL,'a2803d54-e3a5-4ae4-9f14-e290f9cfa059',7157,NULL),(28600,NULL,'The TP53 R175C mutation is located in the protein\'s DNA binding domain and occurs at a hotspot residue (PMID: 26619011). In vitro studies have demonstrated that this mutation has inconclusive effect, as evidenced by cell cycle arrest and apoptosis comparable to that of the wildtype (PMID: 22955915).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2022-09-16 15:56:08',NULL,NULL,NULL,NULL,'f1612c40-9be7-4d4b-9ee4-881ddc4dc72a',7157,NULL),(28601,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f67b970c-6d59-42ba-91d8-d178dc63659c',7157,NULL),(28602,NULL,'The TP53 R248Q mutation is located in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines resulted in aggregation of unfolded mutant proteins, likely leading to destabilization of TP53 and upregulation of heat shock proteins (PMID: 21445056). Expression of this mutation in osteosarcoma cell lines led to a reduced apoptotic response, including reduced expression of pro-apoptotic proteins (PMID: 16778209). In biochemical assays, this mutation was found to bind DNA with a lower affinity than wildtype TP53 (PMID: 24814347).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'4482b15e-4ec4-4d3b-8818-533d02e248ae',7157,NULL),(28603,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7cec6549-39de-4936-a81f-7462dc6cb172',7157,NULL),(28604,NULL,'The TP53 R273C mutation is located in the protein\'s DNA binding domain. Expression of this mutant in TP53 null mouse embryonic fibrobalsts (MEFs) resulted in a significant increase in cell migration relative to TP53 null cells (PMID: 23612969). Structural studies demonstrated that the R273C mutation leads to a dramatic reduction in the DNA binding affinity of p53, although the protein retains wildtype stability (PMID: 23863845, 11900253). Expression of this mutation in cell lines that lack TP53 expression did not enhance TP53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'1795102b-8bb7-4343-999c-44e20f282c49',7157,NULL),(28605,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2025-03-12 14:45:47',NULL,NULL,NULL,NULL,'3a9e6f10-5522-4a50-b939-26727c5abba6',7157,NULL),(28606,NULL,'The TP53 R273G mutation is located in the protein\'s DNA binding domain and occurs at a hotspot residue (PMID: 26619011). In vitro studies have demonstrated that the TP53 R273G is likely neutral as measured by similar levels of proliferation, colony formation and invasion in the mutant compared to wildtype (PMID: 24677579, 37697151).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2025-03-12 14:45:49',NULL,NULL,NULL,NULL,'8997fe78-40db-4812-8ead-2635615747c1',7157,NULL),(28607,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'88122f71-b84f-4681-bb49-6447aa9180cb',7157,NULL),(28608,NULL,'The TP53 R213Q mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is inactivating, as measured by reduced ability to induce cell cycle arrest and apoptosis, as well as increased induction of MDM2 expression (PMID: 10871862).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 23:21:16',NULL,NULL,NULL,NULL,'ca3a45e9-ba06-4bdf-8ff8-4cb85c1eb250',7157,NULL),(28609,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2798f6db-abab-48cf-af47-c72abcc3894a',7157,NULL),(28610,NULL,'The TP53 G245S mutation is located in the protein\'s DNA binding domain. Expression of this mutation in cell lines that lack TP53 expression did not enhance TP53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740). A mouse model harboring the TP53 G245S mutation displayed lack of p53 tumor suppressor abilities including cell cycle arrest, senescence and apoptosis. TP53 G245S-mutant mice display rapid tumor onset and accelerated death in comparison to p53-null mice (PMID: 2358418).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'25888452-3f91-4dfb-9ffa-b910ee02be2e',7157,NULL),(28611,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d334264d-fd19-4147-ac6c-73aed2517c29',7157,NULL),(28612,NULL,'The TP53 G245D mutation is located in the protein\'s DNA binding domain. Structural studies have shown that this mutation results in differences in conformational flexibility of the DNA-binding domain of p53 (PMID: 22803791). Expression of this mutant in hepatocellular carcinoma cell lines resulted in association with the transcription factor ZBP-89 in the cytoplasm. The ZBP-89/mutant TP53 interaction abrogated induction of the cell cycle gene p21 and other apoptotic proteins, leading to reduced apoptosis (PMID: 22214764). Additionally, expression of this mutation in head and neck cancer cell lines was associated with the inability of TP53 to suppress colony growth, as compared to wildtype TP53 and is associated with reduced expression of p21 suggesting that the G245D mutation inactivates the function of p53 (PMID: 25119136).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-05-18 18:39:34',NULL,NULL,NULL,NULL,'b1e75857-639f-4ad4-bf91-8819ae1925c7',7157,NULL),(28613,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2022-09-15 23:24:59',NULL,NULL,NULL,NULL,'5d65bf5b-ce94-495a-817d-ffcfe4d851fe',7157,NULL),(28614,NULL,'The TP53 G245C mutation is located in the protein\'s DNA binding domain. In vitro and structural studies have demonstrated that this mutation has inconclusive effect, as evidenced by the similar levels of TP53-mediated transcriptional activity in the mutant compared to wildtype, even though the mutant results in differences in conformational flexibility of the DNA-binding domain of p53 (PMID: 15037740, 22803791).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2022-09-28 18:35:52',NULL,NULL,NULL,NULL,'13d74785-c4da-4a68-b468-5773a14a5412',7157,NULL),(28615,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'633c047f-3593-41bd-9c66-d39f5be99226',7157,NULL),(28616,NULL,'The TP53 G245A mutation is located in the protein\'s DNA binding domain. This mutation destabilizes p53-DNA binding. However, acetylation was shown to restore the DNA-binding ability and growth-suppressive activity of this mutation (PMID: 20589832).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d26ce914-72cf-4652-905e-3f81efa3c871',7157,NULL),(28617,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'1ed8e010-a253-402f-b9fb-72c72ebd5f7a',7157,NULL),(28618,NULL,'The TP53 Y220C mutation is located in the protein\'s DNA binding domain. Structural studies demonstrated that this mutation destabilizes p53 due the loss of five electrostatic interactions with key residues (PMID: 23315175). A small molecule compound, PK7088, raises the melting temperature of this mutation, triggering cell-cycle arrest and apoptosis. This compound also increases the amount of correctly folded p53 in Y220C mutant cells and increases the expression of p53 target genes such as p21 and Noxa (PMID: 23630318). Expression of this mutation in cell lines that lack TP53 expression did not enhance TP53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e4c8f539-c654-4e1b-8b94-ca11569ef178',7157,NULL),(28619,NULL,'There is promising clinical data in patients with TP53 Y220C-mutant solid tumors treated with the TP53 structural corrector rezatapopt.','TUMOR_TYPE_SUMMARY',NULL,_binary '\0',NULL,'2024-07-24 12:37:06',NULL,NULL,NULL,NULL,'58cfdce6-2588-49a9-8f3c-76534fd4951c',7157,NULL),(28620,NULL,'Rezatapopt is an orally available, small-molecule structural corrector specific for the Y220C-mutant p53 protein. In the Phase I PYNNACLE (NCT04585750) trial of rezatapopt in 29 patients with advanced solid tumors harboring a TP53 Y220C mutation, one patient with small cell lung cancer and one patient with breast cancer each had a partial response to treatment (Abstract: Dumbrava et al. Abstract# 3003, ASCO 2022. https://ascopubs.org/doi/abs/10.1200/JCO.2022.40.16_suppl.3003). In an updated analysis of the Phase I/II PYNNACLE (NCT04585750) trial of rezatapopt in 61 patients with TP53 Y220C solid tumors, the overall response rate was 33.3% (n=12) in patients with TP53 Y220C and KRAS wildtype tumors, including ovarian, breast, prostate, small-cell lung and endometrial cancer, and the median duration of confirmed response was 7 months (Abstract: Schram et al. Abstract# LB_A25, AACR 2023. https://aacrjournals.org/mct/article/22/12_Supplement/LB_A25/730549/Abstract-LB-A25-Updated-Phase-1-results-from-the). The results of an updated subgroup analysis of the Phase I/II PYNNACLE (NCT04585750) trial of rezatapopt in heavily pretreated patients with advanced ovarian cancer (n=22), specifically those with TP53 Y220C and KRAS wildtype tumors, demonstrated that among fifteen evaluable patients, seven patients achieved partial response, seven had stable disease, and one had progressive disease, with an overall response rate of 47% and median duration of confirmed response of 7.0 months (Abstract: Schram et al. Abstract# LBA_26, SGO 2024. https://www.gynecologiconcology-online.net/article/S0090-8258(24)00453-0/fulltext). In vivo xenograft studies with TP53 Y220C-mutant gastric cancer cell line NUGC3 and Y220C-mutant patient-derived xenografts demonstrated sensitivity to rezatapopt treatment as measured by dose-dependent tumor shrinkage and conversion of TP53 Y220C to the wildtype conformation (Abstract: Dumble et al. Abstract# LB006, AACR 2021. https://aacrjournals.org/cancerres/article/81/13_Supplement/LB006/669897/Abstract-LB006-PC14586-The-first-orally).','INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY','LEVEL_Fda3',_binary '\0','Sensitive','2025-03-19 13:05:12',NULL,'LEVEL_3A','NO','NO','495ba473-e137-4343-b248-04acbea470bd',7157,NULL),(28621,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5afc6808-9dfc-4408-bbb3-2175f1c98cd1',7157,NULL),(28622,NULL,'The TP53 Y220S mutation is located in the protein\'s DNA binding domain. This mutation has been identified as a germline mutation and is commonly found in patients with Li-Fraumeni syndrome (LFS). Cells expressing TP53 Y220S displayed angiogenetic properties, andibroblasts expressing this mutation exhibited resistance to doxorubicin hydrochloride (DX) (PMID: 15977174).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bbdc7145-4b24-441d-8dbb-8100078c9759',7157,NULL),(28623,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5efbee72-241e-487a-b2e2-940bea3a256f',7157,NULL),(28624,NULL,'The TP53 R282Q mutation is located in the protein\'s DNA binding domain. This mutation has been identified as a statistically significant hotspot (PMID: 26619011). Structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the decreased flexibility of the L1 DNA-binding loop and two novel conformations of the L3 DNA-binding loop in the mutant compared to wildtype, all of which are predicted to deactivate the tumor suppressor functions of p53 (PMID: 18453682).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-10-03 14:34:31',NULL,NULL,NULL,NULL,'42c10a9f-79bc-4300-87bb-dc024eff1ef4',7157,NULL),(28625,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7d717015-2e80-46dc-bcaf-5c79cdf42a67',7157,NULL),(28626,NULL,'The TP53 H179R mutation is located in the protein\'s DNA binding domain. Expression of this mutant failed to induce expression of genes involved in apoptosis or cell cycle arrest (PMID: 17361096). Additionally, this mutation exhibited aberrant regulation of the p53 target gene, BTG2, resulting in elevated HRAS and a unique expression pattern of a cancer-related gene signature (PMID: 22427690).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d37c6c5b-21b8-47b4-a98a-8fae8955f01b',7157,NULL),(28627,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'12012306-2551-40fd-ba51-4967a0101d4a',7157,NULL),(28628,NULL,'The TP53 H179Y mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is inactivating, as evidenced by the increased G1 to S phase transition, enlarged cell size, as well as increased expression of cyclin A1 and CDK4 in the mutant compared to wildtype (PMID: 17530187).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 23:48:56',NULL,NULL,NULL,NULL,'4900f3b0-5437-4553-a4b8-b7a9978c107c',7157,NULL),(28629,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'900bfaea-0b3a-4a6a-a739-3562a76d32a9',7157,NULL),(28630,NULL,'The TP53 H179L mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is inactivating, as evidenced by enhanced adhesion to extracellular matrix and increased growth in suspension culture in the mutant compared to wildtype (PMID: 23150443).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 23:57:11',NULL,NULL,NULL,NULL,'5b08abf9-5d6f-499f-8c90-378c1a899f37',7157,NULL),(28631,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bc045567-d397-42de-a75c-dda7444199e5',7157,NULL),(28632,NULL,'The TP53 H179Q mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is inactivating and behaves as a dominant-negative allele, as evidenced by reduced expression of p21 and increased G1 to S phase cell cycle progression in the mutant compared to wildtype (PMID: 22540896).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-09-15 23:59:54',NULL,NULL,NULL,NULL,'4ee72828-764c-4310-90a1-4719796682ca',7157,NULL),(28633,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'fd6aa90a-36c5-48f7-a943-b7cb7fb0bfe9',7157,NULL),(28634,NULL,'The TP53 H179N mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by reduced DNA binding activity and similar exonuclease activity in the mutant compared to wildtype (PMID: 19462533).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-16 00:04:54',NULL,NULL,NULL,NULL,'89391554-fcfb-436b-9903-c4982afcaf7e',7157,NULL),(28635,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'58a24041-73ee-499e-91b0-cdf0fca27d79',7157,NULL),(28636,NULL,'The TP53 H193R mutation is located in the protein\'s DNA binding domain. This mutation has been found in sarcoma (PMID: 25669829). In vitro studies have demonstrated that this mutation is likely inactivating, as measured by decreased DNA binding and decreased transactivaction of TP53, likely leading to reduced p21 activation (PMID: 9627118). In a phase I clinical trial of pazopanib, a VEGF inhibitor, and vorinostat, a histone deacetylase inhibitor, eleven patients with sarcoma and colorectal cancer who harbor the TP53 H193R mutation as well as other TP53 mutations in exons 5-8 had longer median progression-free and overall survival compared to patients without TP53 mutations in response to treatment (PMID: 25669829).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-09-28 19:04:03',NULL,NULL,NULL,NULL,'19981225-8861-4254-8a15-b38ad0176529',7157,NULL),(28637,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e0acc9c2-c214-44b1-b98e-30e699ceead2',7157,NULL),(28638,NULL,'The TP53 H193N mutation is located in the protein\'s DNA binding domain. In vitro studies have suggested that this mutation is likely inactivating, as evidenced by increased exonuclease activity and significantly reduced DNA binding activity compared to the wildtype (PMID: 19462533).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-11-04 12:46:13',NULL,NULL,NULL,NULL,'a4c0defb-5dde-4e3c-86b4-514e487bbcbd',7157,NULL),(28639,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'41f3b0e6-56b8-4f6b-84fd-181f55838535',7157,NULL),(28640,NULL,'The TP53 C176F mutation is located in the protein\'s DNA binding domain. This mutation has been found in head and neck cancer (PMID: 22090360). In vitro studies have demonstrated that this mutation is inactivating, as evidenced by decreased transactivation of p21, decreased production of reactive oxygen species and decreased cell death when the mutant is treated with radiation compared to wildtype, suggesting an inhibition of radiation-induced senescence that results in treatment failure (PMID: 22090360).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-11-04 14:27:17',NULL,NULL,NULL,NULL,'d6a9728c-480b-4a9a-a8b2-48c3fd855b33',7157,NULL),(28641,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'31896bf1-ca99-4d2c-b09c-964af0ee3fa6',7157,NULL),(28642,NULL,'The TP53 C176R mutation is located in the protein\'s DNA binding domain. In vitro studies have suggested that this mutation is inactivating and acts as a dominant negative, as evidenced by the ability of mutant yeasts coexpressing both wild-type and C176R-mutated p53 to survive on Foa plates compared to wildtype (PMID: 8633021).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-11-04 14:49:52',NULL,NULL,NULL,NULL,'8be2e6a2-755f-4bd6-ab7d-88b448c6171a',7157,NULL),(28643,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'446fc820-d97c-445b-ae45-bcadb8d2331f',7157,NULL),(28644,NULL,'The TP53 I195T mutation is located in the protein\'s DNA binding domain. Structural studies have suggested that this mutation is likely inactivating as evidenced by the decreased half-life time, faster aggregation and faster unfolding time at body temperature of the mutant compared to wildtype protein, suggesting that the mutant is highly destabilized and could result in loss of p53 tumor suppressor function (PMID: 12700230).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-04 16:25:55',NULL,NULL,NULL,NULL,'9c47ac79-0f93-4901-8709-3d2666850fcf',7157,NULL),(28645,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'790f5f5e-b15a-449e-a777-cc631aa83b60',7157,NULL),(28646,NULL,'The TP53 G266E mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transcriptional transactivation of p53 target genes in the mutant compared to the wildtype, suggesting a loss of function to regulate transactivation of p53 target genes (PMID: 17724467).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-04 16:34:32',NULL,NULL,NULL,NULL,'a37213f6-cf16-40a6-b0bc-368d8530366f',7157,NULL),(28647,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'343aedad-e7f2-4766-97c6-e2b1d89249f0',7157,NULL),(28648,NULL,'The TP53 R249S mutation is located in the protein\'s DNA binding domain. Structural and in vitro studies have demonstrated that this mutation is likely inactivating as evidenced by distortion of the local binding surface and increased aggregation of the mutant proteins compared to wildtype, reduced TP53 transactivation activity as measured by luciferase reporter assays, defective cell cycle progression, and aberrant expression of apoptosis proteins in cells expressing the mutant protein compared to wildtype (PMID: 14741214, 21445056, 18996393).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-04 16:27:28',NULL,NULL,NULL,NULL,'9c07e707-9d36-4a29-90f3-c27a187e03da',7157,NULL),(28649,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'0102b352-c961-4002-a011-30f5682ef93d',7157,NULL),(28650,NULL,'The TP53 R249M mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the mutant\'s inability to inhibit the transcriptional activity of the p21 promoter and enhance TP53-mediated transcriptional activity in reporter assays compared to wildtype (PMID: 17999388, 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-04 16:18:29',NULL,NULL,NULL,NULL,'131c7f95-88bd-4981-a4c7-727fae35bef8',7157,NULL),(28651,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'34064c08-acba-4723-8dbb-cf91b2c70392',7157,NULL),(28652,NULL,'The TP53 R249W mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by increased colony formation and greater migration in the mutant compared to wildtype (PMID: 23612969).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-04 16:30:56',NULL,NULL,NULL,NULL,'ed12f229-01c9-4abd-9750-7d75ea473d06',7157,NULL),(28653,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6110b319-851e-494d-902a-298390b81b6d',7157,NULL),(28654,NULL,'The TP53 Y234C mutation is located in the protein\'s DNA binding domain. This mutation has been found in ovarian cancer (PMID: 26748848). In vitro studies have demonstrated that this mutation is inactivating, as evidenced by decreased transactivation of p53 target genes, increased cytoplasmic localization and increased apoptosis when treated with an amyloid aggregate TP53 inhibitor in the mutant compared to wildtype, suggesting that this mutation may cause amyloid aggregate formation that inactivates the p53 protein (PMID: 10229196, 16861262, 26748848).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-11-07 12:49:06',NULL,NULL,NULL,NULL,'96e2befc-af08-41f6-9e7c-1756b7ed3e47',7157,NULL),(28655,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8da5c56f-54ce-4e1e-aa44-bddeecd51b9c',7157,NULL),(28656,NULL,'The TP53 Y234H mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the reduced ability to transactivate p53 target genes of the mutant compared to wildtype (PMID: 22710932).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-05 11:23:01',NULL,NULL,NULL,NULL,'27311c4f-ae10-4f10-b49f-113d790baa00',7157,NULL),(28657,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'55add0c9-1b2d-4f4a-c019-260bcb83dd20',7157,NULL),(28658,NULL,'The TP53 C238F mutation is located in the protein\'s DNA binding domain. This mutation has been found in breast cancer (PMID: 20407015). In vivo and structural studies have demonstrated that this mutation is inactivating, as evidenced by the structural prediction that the mutation interferes with the histidine or cysteine side chains that are known to be involved in the coordination of the zinc ion, as well as the mutant\'s inability to transactivate p53 target genes compared to wildtype (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-11-07 12:49:39',NULL,NULL,NULL,NULL,'f651d617-a45b-4abb-92ea-e0b8ae80cf0f',7157,NULL),(28659,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'dbc135d1-8807-4a56-ac84-a2a9563d2ff5',7157,NULL),(28660,NULL,'The TP53 C238S mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transactivation of p53 target genes in the mutant compared to wildtype (PMID: 19681600).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-05 11:32:40',NULL,NULL,NULL,NULL,'49e1b7b9-d9d6-440c-b0a8-ac813f3a2740',7157,NULL),(28661,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'46d61767-80b8-4aca-994a-e533fea37822',7157,NULL),(28662,NULL,'The TP53 V173E mutation is located in the protein\'s DNA binding domain. This mutation has been found in lung carcinoma  (PMID: 9472631). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the mutant\'s inability to bind to the responsive elements of p53 and transactivate p53-dependent promoters compared to wildtype (PMID: 9472631).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-07 12:50:11',NULL,NULL,NULL,NULL,'3eec88ab-6e54-492b-8a02-36e7abed203e',7157,NULL),(28663,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7f521adf-8df7-45d8-b9c9-c8d310f46f15',7157,NULL),(28664,NULL,'The TP53 R158C mutation is located in the protein\'s DNA binding domain. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transactivation of p53 target genes in the mutant compared to wildtype (PMID: 17947339).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-11-05 12:59:29',NULL,NULL,NULL,NULL,'73d09b4a-9b64-4d4c-af06-008a67646710',7157,NULL),(28665,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d5c23995-9b5f-4448-8349-30f1642a88ab',7157,NULL),(28666,NULL,'The TP53 P151S mutation is located in the protein\'s DNA binding domain. This mutation has been found in head and neck squamous cell carcinoma (PMID: 21903770). In vitro and in vivo studies have demonstrated that this mutation is inactivating, as evidenced by the formation of larger tumor spheroids with longer migration and invasion distances in matrigel culture compared to wildtype, as well as increased tumor growth and decreased animal survival in xenograft models (PMID: 21903770).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-11-07 12:50:54',NULL,NULL,NULL,NULL,'1151e72c-dc3c-4a2c-ba52-c825b76238c2',7157,NULL),(28667,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3dbaa0fa-0c07-43a7-b18f-507cfeb27c1f',7157,NULL),(28668,NULL,'The TP53 P151H mutation is located in the DNA binding domain of the protein. In vivo studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased transactivation of p53 target genes in the mutant compared to wildtype (PMID: 22908277).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:12:31',NULL,NULL,NULL,NULL,'b88eb62c-c5da-4c01-8525-9e907028360a',7157,NULL),(28669,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'cfa0a2ab-09bb-4c97-ba4a-30f701297160',7157,NULL),(28670,NULL,'The TP53 P151A mutation is located in the DNA binding domain of the protein. In vivo studies have demonstrated that this mutation is likely inactivating, as evidenced by the complete lack of transactivation of p53 target genes in the mutant compared to wildtype (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:13:53',NULL,NULL,NULL,NULL,'89c7e0cc-517a-4bfd-a4db-20384cc53bb0',7157,NULL),(28671,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5460b591-1ea9-493e-a8ad-6cb2f5285272',7157,NULL),(28672,NULL,'The TP53 Y236C mutation is located in the DNA binding domain of the protein. This mutation has been found in colon cancer  (PMID: 16827139). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by increased cytoplasmic localization of the mutant, leading to reduced transcriptional activity in the mutant compared to wildtype (PMID: 16827139).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:14:15',NULL,NULL,NULL,NULL,'ee2e9c0c-5c03-4700-bfd2-07ae5859bead',7157,NULL),(28673,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'b652862c-9b8d-4cbf-8a89-b3e3bcb0bfe4',7157,NULL),(28674,NULL,'The TP53 Y236D mutation is located in the DNA binding domain of the protein. In vivo studies have demonstrated that this mutation is likely inactivating, as evidenced by the loss of binding ability of the p53 protein to its consensus sequence in the mutant compared to wildtype (PMID: 8633021).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:14:49',NULL,NULL,NULL,NULL,'0d29d38e-250f-4c6c-9b79-371450932c17',7157,NULL),(28675,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'52cd1643-06bb-4ac4-9f04-e05f0f602a66',7157,NULL),(28676,NULL,'The TP53 Y236S mutation is located in the DNA binding domain of the protein. In vivo studies have demonstrated that this mutation is likely inactivating, as evidenced by the loss of transcriptional activity for p53 target genes, p21 and RGC, in the mutant compared to wildtype (PMID: 11429705).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:15:19',NULL,NULL,NULL,NULL,'dbc51820-a8b0-4383-b1e3-8880dfaecd8f',7157,NULL),(28677,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'502cefa0-d28c-4506-a108-1035f51d8fe4',7157,NULL),(28678,NULL,'The TP53 P278S mutation is located in the DNA binding domain of the protein. In vitro and in vivo studies have demonstrated that this mutation is inactivating, as evidenced by failure to activate the p21 promoter in the mutant compared to wildtype (PMID: 22989750).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-12-07 18:15:41',NULL,NULL,NULL,NULL,'833b1afa-c712-4d02-ae7d-2fa94464b10d',7157,NULL),(28679,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'4c4e6134-e407-43f2-9485-7b7fc2865c9e',7157,NULL),(28680,NULL,'The TP53 P278L mutation is located in the DNA binding domain of the protein. In vivo studies have demonstrated that this mutation is likely inactivating, as evidenced by failure to transactivate p53 target genes in the mutant compared to wildtype (PMID: 16861262).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:16:01',NULL,NULL,NULL,NULL,'71cdbaeb-4d9a-422c-93be-3963b82b154d',7157,NULL),(28681,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'efbdd2eb-04d8-4744-9188-7e0541eef00f',7157,NULL),(28682,NULL,'The TP53 P278A mutation is located in the DNA binding domain of the protein. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by failure to transactivate p53-dependent genes in the mutant compared to wildtype (PMID: 24076587).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:17:55',NULL,NULL,NULL,NULL,'c75da0cb-1d2f-42be-9869-bfd46052d9c3',7157,NULL),(28683,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'32202420-35e5-4bd4-b05c-99a0ec5b4654',7157,NULL),(28684,NULL,'The TP53 P278R mutation is located in the DNA binding domain of the protein. This mutation has been found in hepatocellular carcinoma (PMID: 21760996). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the loss of transactivation activity of p53 target genes in the mutant compared to wildtype (PMID: 21760996).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:20:19',NULL,NULL,NULL,NULL,'f5736b95-d9c0-47d3-a5d3-bc56948d72ca',7157,NULL),(28685,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'64951c47-fa12-4c80-82bf-0a7e60644559',7157,NULL),(28686,NULL,'The TP53 C135Y mutation is located in the DNA binding domain of the protein. In vitro studies with p53-null endometrial cancer cells expressing TP53 C135Y have demonstrated that this mutation is inactivating as measured by loss of miRNA expression regulation, specifically the microRNA miR-130b and miR-200c, resulting in increased epithelial to mesenchymal transition and cell invasion in the mutant compared to wildtype (PMID: 22847613). In vivo studies with yeast expressing TP53 C135Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C135Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-19 14:35:45',NULL,NULL,NULL,NULL,'58ac6d98-c731-4ae6-adf3-fa75457ce195',7157,NULL),(28687,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6715ed2d-1b0b-4556-bb8d-e29a218e2d75',7157,NULL),(28688,NULL,'The TP53 C135R mutation is located in the DNA binding domain of the protein. This mutation has been found in non-small cell lung cancer (PMID: 17914575). Structural and in vitro studies have demonstrated that this mutation is inactivating, as evidenced by its predicted negative effect on p53-DNA interaction and the mutant protein\'s inability to bind to the TP53 response element from the p21 gene in an electrophoretic mobility shift assay compared to wildtype (PMID: 17914575).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:22:15',NULL,NULL,NULL,NULL,'47bcf836-b853-4dc6-8446-91840ffe30be',7157,NULL),(28689,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c611c09b-1c1b-4a04-9e2e-86211924af5f',7157,NULL),(28690,NULL,'The TP53 C135S mutation is located in the DNA binding domain of the protein. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by reduced binding capacity to TP53 response elements in the mutant compared to wildtype (PMID: 12034820).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:22:44',NULL,NULL,NULL,NULL,'927a3833-3c85-4286-bdb9-74a8cdeef3d6',7157,NULL),(28691,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'ddca5b22-8975-42c7-8261-ce9212412460',7157,NULL),(28692,NULL,'The TP53 C275S mutation is located in the DNA binding domain of the protein. Structural and in vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased affinity for the Gadd45 promoter site and increased attenuation of oxidative dissociation in the mutant compared to wildtype, suggesting that this mutation affects p53\'s function through altering p53-DNA contact (PMID: 25584637).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-12-07 18:23:25',NULL,NULL,NULL,NULL,'b19da370-891e-48e9-bf54-1b6d7b3e4bff',7157,NULL),(28693,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8932cc30-946c-4dcf-9b18-0ec6fd2f6c41',7157,NULL),(28694,NULL,'The TP53 M237K mutation is located in the DNA binding domain of the protein. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the transactivation of some p53 target genes, including bax and MDM2, but not p21 in the mutant compared to wildtype, suggesting that the mutant is partially but not fully functional (PMID: 20505364).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:24:28',NULL,NULL,NULL,NULL,'f633f499-f3cc-4229-874e-ff248e29efd0',7157,NULL),(28695,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'52522d39-f779-465b-b75c-ae42eaf5ee0a',7157,NULL),(28696,NULL,'The TP53 S241F mutation is located in the DNA binding domain of the protein. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased co-immunoprecipitation of p53 with the tumor suppressor BARD1 and the 3\' processing factor cleavage-stimulation factor 1 (CstF1), respectively, as well as decreased formation of the p53/BARD1/CstF1 complex, resulting in reduced inhibition of 3\' cleavage (PMID: 21383700).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:24:59',NULL,NULL,NULL,NULL,'dde9b7ee-3377-4b66-8840-abba2b97b783',7157,NULL),(28697,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'926c73a5-ef20-4e9e-b171-166a28553cf2',7157,NULL),(28698,NULL,'The TP53 S241T mutation is located in the DNA binding domain of the protein. In vitro studies have demonstrated that this mutation is inactivating and exhibits a dominant-negative phenotype, as evidenced by the lack of transcription of p53 target genes when both mutant and wildtype proteins are present, suggesting the mutant\'s ability to inhibit wildtype p53\'s transactivation ability (PMID: 11896595).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:25:23',NULL,NULL,NULL,NULL,'4fa7669a-0569-4358-898b-6cbaac8c2d03',7157,NULL),(28699,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'707351b7-2efe-4dad-9bae-38852823e1f7',7157,NULL),(28700,NULL,'The TP53 V272M mutation is located in the DNA binding domain of the protein. This mutation has been found in esophageal cancer (PMID: 9096669). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by the mutant\'s reduced reactivity with the wildtype conformation-specific antibody PAb1620, suggesting the adoption of a mutant conformation, as well as reduced accumulation in cells exposed to DNA-damaging agents and inability to rescue TP53-mediated transcriptional activity in TP53-null cells of the mutant compared to wildtype (PMID:15037740, 9096669, 11870884).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:25:50',NULL,NULL,NULL,NULL,'ea2f157e-44b1-4866-a2d6-31ab37cea043',7157,NULL),(28701,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-02-23 16:22:10',NULL,NULL,NULL,NULL,'348124be-1c9c-4f7f-9fde-07b0c77ac89c',7157,NULL),(28702,NULL,'The TP53 V272L mutation is located in the DNA binding domain of the protein. In silico and in vitro studies have demonstrated that this mutation has inconclusive effect, as evidenced by comparable levels of p53 target gene transactivation in the mutant compared to wildtype, despite its position at a hotspot residue which suggests that it is likely oncogenic (PMID: 20407015, 26619011).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2022-12-07 18:27:36',NULL,NULL,NULL,NULL,'bcadaf27-fa17-4978-bc4c-1efd42db3f0a',7157,NULL),(28703,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f75ff833-43d2-4a6a-954e-de1da2b6d03c',7157,NULL),(28704,NULL,'The TP53 V157F mutation is located in the DNA binding domain of the protein. In vitro and structural studies have demonstrated that this mutation is likely inactivating, as evidenced by the mutant protein\'s rearranged DNA-binding surface and reduced DNA-binding activity compared to wildtype, as well as reduced expression of pro-apoptotic proteins, reduced apoptotic response and impaired transcriptional activity in cells expressing the mutant compared to wildtype (PMID: 21561095, 16778209, 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-07 18:27:45',NULL,NULL,NULL,NULL,'b771c2bb-6b11-4575-83bb-63339f53b154',7157,NULL),(28705,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'4d5e948f-ead3-4b0b-a207-e93f30e4011e',7157,NULL),(28706,NULL,'The TP53 V157D mutation is located in the DNA binding domain of the protein. This mutation has been found in the germline of patients with familial cancer syndromes (PMID: 23981578). In vitro and in vivo studies have demonstrated that this mutation is inactivating, as evidenced by increased colony formation in soft agar assays and increased tumor formation in a transgenic Drosophila system expressing the mutant compared to wildtype (PMID: 23981578).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-12-07 18:28:17',NULL,NULL,NULL,NULL,'e139ddd4-b683-4b90-97ff-520a0e2eb969',7157,NULL),(28707,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2bf5a097-e62a-4e22-a62d-64cfabb6e576',7157,NULL),(28708,NULL,'The TP53 R280K mutation is located in the DNA binding domain of the protein. In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by decreased apoptosis and increased migration in TP53-null cells expressing the mutant compared to wildtype (PMID: 24714748).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2022-12-07 18:28:37',NULL,NULL,NULL,NULL,'e3cc7c18-1e4e-4911-8127-79b267f757c8',7157,NULL),(28709,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'66bf4f84-ee70-498f-9803-abb7e70b4497',7157,NULL),(28710,NULL,'The TP53 R280T mutation is located in the DNA binding domain of the protein. This mutation has been found in glioma (PMID: 22999923). In vitro studies have demonstrated that this mutation is likely inactivating, as evidenced by increased proliferation of cells expressing the mutant compared to wildtype, as well as increased p-GSK-3 levels, reduced PTEN and Akt phosphorylation and reduced cell growth upon knockdown of the mutant protein compared to wildtype, suggesting that this mutation promotes cell growth through GSK-3beta-PTEN pathway (PMID: 22999923).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:48:23',NULL,NULL,NULL,NULL,'dc9b6b17-0643-4fb9-bd35-26ee22eb93cf',7157,NULL),(28711,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'643a4091-2dcf-40cc-a97c-2825d661a41b',7157,NULL),(28712,NULL,'The TP53 C242F mutation is located in the protein\'s DNA binding domain. Structural studies have elucidated that this mutant is inactivating as it directly affects the zinc coordination sphere by substituting one of the zinc ligands and thus affects the DNA binding capabilities of p53 (PMID: 17401432).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-03-16 15:50:11',NULL,NULL,NULL,NULL,'d43188f4-7503-42c5-a155-6a0ec1b14000',7157,NULL),(28713,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'074c9875-1183-46a7-8c16-5cad43f53e90',7157,NULL),(28714,NULL,'The TP53 N239S mutation is located in the protein\'s DNA binding domain. This mutant has lost the wild type p53 tumor suppresive function because this mutant is unable to bind to DNA and p53 target promoters. Moreover, it fails to regulate the transcription of p21. Additionally, xenografts studies have shown that this mutant promotes tumorigenesis in p53 null MEFs (PMID: 22553460).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'73e866a2-4701-46a6-9f19-ba5eda7316f5',7157,NULL),(28715,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'55ee798d-973d-458c-93d1-fb0b4bd68c74',7157,NULL),(28716,NULL,'The TP53 S215C mutation is located in the protein\'s DNA binding domain. Expression of this mutant resulted in transactivation of p53 target genes comparable to wildtype p53, as a measured by yeast-based functional assay (PMID: 12917626). Expression of this mutation in human osteosarcoma cell lines resulted in slightly increased apoptosis, as evidenced by cell cycle progression studies, though the result was not statistically significant (PMID: 15781620). However, this mutation is a hotspot, suggesting that it is likely oncogenic (PMID: 26619011).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e6026de0-a248-4b16-a2b5-b46bd49f18da',7157,NULL),(28717,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3398e46b-20a4-4486-811d-eab0ddccd483',7157,NULL),(28718,NULL,'The TP53 C242S mutation is located in the protein\'s DNA binding domain. This mutation is known to interfere with the histidine or cysteine side chains involved in coordination of the zinc ion. This structural change makes this mutant protein nonfunctional in terms of transactivation (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'80b18f71-f2e7-4f63-afe5-5d153f40afe2',7157,NULL),(28719,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'a512b538-7474-49e2-9b80-ef6a9db53d95',7157,NULL),(28720,NULL,'The TP53 C242Y mutation is located in the protein\'s DNA binding domain. This mutation is known to interfere with the histidine or cysteine side chains involved in coordination of the zinc ion. This structural change makes this mutant protein nonfunctional in terms of transactivation (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'eea40f25-bcab-4be0-b0aa-06a6e3d2cc78',7157,NULL),(28721,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c04bb5bd-ddf6-4414-8970-2bb58ffb8333',7157,NULL),(28722,NULL,'The TP53 G244D mutation is located in the protein\'s DNA binding domain. When studied in a yeast model, this mutation acted as a dominant negative suggesting a loss of function mutation of p53 (PMID: 12901798).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'fee8e0dc-7b6a-410e-b3ce-e2b8d5010538',7157,NULL),(28723,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9fc7be5a-633c-49ca-a774-dea88e50cf25',7157,NULL),(28724,NULL,'The TP53 G244S mutation is located in the protein\'s DNA binding domain. This mutation was frequently found in patients with Lii Fraumeni syndrome (LFS). Functional analysis have indicated that this mutant is unable to transactivate, does not bind to DNA and does not inhibit cell growth like wild type p53 does (PMID: 26818906).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'06dd9563-c8d2-46b8-86f0-ccdfcea83b1c',7157,NULL),(28725,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'595be5b5-8c99-40ca-bef1-f19323553624',7157,NULL),(28726,NULL,'The TP53 G244R mutation is located in the protein\'s DNA binding domain. When tested in transactivation assays, this mutant displayed lower (1.8 to 3 fold) levels than wild type p53 indicating a loss of function. Additionally, this mutant was also deficient in inhibiting cell growth in proliferation assays (PMID: 11160722).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'0f17d51e-970d-4457-b9da-d48614774198',7157,NULL),(28727,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'a83fc154-023e-426c-a0a3-a5b16078595b',7157,NULL),(28728,NULL,'The TP53 V274F mutation is located in the protein\'s DNA binding domain. In cell line studies this mutation exhibited decreased transactivation of p53 target genes and an increase in colony formation (PMID: 12782597). Additionally, overexpression of this mutant increased the resistance of prostate cells to DNA damage-induced cell death (http://www.dtic.mil/dtic/tr/fulltext/u2/a447626.pdf).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'1a7c6f6b-efb4-4fdc-a843-4e7e56a00ac6',7157,NULL),(28729,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5efabdab-3081-4014-8fca-30138d16503a',7157,NULL),(28730,NULL,'The TP53 S121A mutation is located in the protein\'s DNA binding domain. Expression of this mutation in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53, as evidenced by cell cycle progression studies (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6c1a04aa-198f-4254-b724-a635bce1b553',7157,NULL),(28731,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'eaa9c697-4f97-4644-8370-4bdac3e8c2ad',7157,NULL),(28732,NULL,'The TP53 S121Y mutation is located in the protein\'s DNA binding domain. While expression of this mutation in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53, (PMID: 15781620), this experimental result was not statistically significant.','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'16487a38-70dd-4eff-9bda-83ebd4f9b1cd',7157,NULL),(28733,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7dbc66fd-09d0-4725-80b4-60466e86c744',7157,NULL),(28734,NULL,'The TP53 S121C mutation is located in the protein\'s DNA binding domain. Expression of this mutation in human osteosarcoma cell lines resulted in a significant increase in apoptotic ability compared to wildtype TP53 (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6e290257-712b-4c12-b00d-eb3e3c863db1',7157,NULL),(28735,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'4108b6dd-d774-46dc-83fb-f4e4c6181123',7157,NULL),(28736,NULL,'The TP53 P153H mutation is located in the protein\'s DNA binding domain. Changes in the DNA binding domain result in altered contact of p53 with its target DNA sequences, thereby altering its transcriptional function (PMID: 8023157, 11900253). Expression of this mutation in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53 (PMID: 15781620), though this experimental result was not statistically significant. TP53 P153H also had altered transcriptional activity of target genes in a luciferase assay compared to wildytype TP53 (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d051edb6-4619-4a26-8121-d09734c31c87',7157,NULL),(28737,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2025-03-12 14:38:30',NULL,NULL,NULL,NULL,'64b2e934-78aa-444c-9abf-eeaedb929eb5',7157,NULL),(28738,NULL,'The TP53 R156C mutation is located in the protein\'s DNA binding domain. There is conflicting evidence on the oncogenic effect of this mutation. Mutations in the DNA binding domain result in altered contact of p53 with its target DNA sequences, thereby affecting transcriptional function (PMID: 8023157, 11900253). Expression of TP53 R156C in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53, though this experimental result was not statistically significant (PMID: 15781620). TP53 R156C demonstrated altered transcriptional activity of target genes in a luciferase assay compared to wildtype TP53 (PMID: 15781620). Conversely, in vitro studies with HCT116 and U2OS cells expressing TP53 R156C demonstrate a gene expression phenotype similar to wildtype, suggesting minimal impact on the p53 pathway (PMID: 37697151).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2025-03-12 14:38:30',NULL,NULL,NULL,NULL,'e6fe0c84-a907-4043-ae0b-3fc85008535c',7157,NULL),(28739,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3fca7877-4ff0-4c29-a614-62a4f48062a9',7157,NULL),(28740,NULL,'The TP53 F212Y mutation is located in the protein\'s DNA binding domain. While expression of this mutation in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53, (PMID: 15781620), this experimental result was not statistically significant.','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c75e621f-6f4c-4812-844a-185ceba92676',7157,NULL),(28741,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c79e17dd-a29c-4522-9182-20d32adfbecd',7157,NULL),(28742,NULL,'The TP53 E221Q mutation is located in the protein\'s DNA binding domain. While expression of this mutation in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53, (PMID: 15781620), this experimental result was not statistically significant.','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'81e878e9-b8b8-41e3-a17d-3c1f4732f1ad',7157,NULL),(28743,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'63631662-6129-4033-985f-ed3cb10d6021',7157,NULL),(28744,NULL,'The TP53 K291Q mutation is located in the protein\'s DNA binding domain. Changes in the DNA binding domain result in altered contact of p53 with its target DNA sequences, thereby altering its transcriptional function (PMID: 8023157, 11900253). Expression of this mutation in human osteosarcoma cell lines resulted in increased apoptosis compared to wildtype TP53 (PMID: 15781620), though this experimental result was not statistically significant. TP53 K291Q also had altered transcriptional activity of target genes in a luciferase assay compared to wildytype TP53 (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'19f9bd96-2a87-4477-b545-a03e2a37563c',7157,NULL),(28745,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2024-07-23 14:54:52',NULL,NULL,NULL,NULL,'1a702d75-9979-4c4f-9def-be858dac644c',7157,NULL),(28746,NULL,'The TP53 K292I mutation is located in the DNA binding domain of the protein. Changes in the DNA binding domain result in altered contact of p53 with its target DNA sequences, thereby altering its transcriptional function (PMID: 8023157, 11900253). In vitro studies have demonstrated that this mutation is likely activating, as evidenced by significantly increased apoptosis, decreased cell viability and altered transcription of TP53 target genes in the mutant compared to wildtype (PMID: 15781620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2024-07-23 14:54:45',NULL,NULL,NULL,NULL,'39359a77-de12-40bf-a17b-557674374b58',7157,NULL),(28747,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'983ffafd-6427-4bcb-a07b-1bd4b7059780',7157,NULL),(28748,NULL,'The TP53 F328V mutation occurs in the protein\'s tetramerization domain. Expression of this mutation in human lung cancer cell lines resulted in decreased transcriptional activity and protein stability, as evidenced by a reporter assay (PMID: 19454241).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f70a5690-e28a-4629-a69b-7720bcabecec',7157,NULL),(28749,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8ea5737f-8374-4c12-8a4d-a7e6bac479d4',7157,NULL),(28750,NULL,'The TP53 L348F mutation occurs in the protein\'s tetramerization domain. Expression of this mutation in human lung cancer cell lines resulted in decreased transcriptional activity and protein stability compared to wildtype TP53, as evidenced by a reporter assay (PMID: 19454241).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8c18e082-f43c-47c6-afcd-7c3a3c42fd73',7157,NULL),(28751,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'95d899f6-b753-41fc-ba17-c10354abeb60',7157,NULL),(28752,NULL,'The TP53 S241Y mutation is located in the protein\'s DNA binding domain. Expression of this mutant in TP53-null cells resulted in decreased transactivation of p53 target genes, as a measured by a luciferase report assay (PMID: 25584008). Additionally, the TP53 S241Y mutation failed to suppress colony growth in a colony reduction assay (PMID: 25584008).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'62868117-37d1-4152-b4d5-b80cf493a2f7',7157,NULL),(28753,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'07a9a978-1806-4d47-b748-7c50e9dfd4be',7157,NULL),(28754,NULL,'The TP53 R248L mutation is located in the protein\'s DNA binding domain. Expression of this mutant in TP53-null cells resulted in decreased transactivation of p53 target genes, as a measured by a luciferase reporter assay (PMID: 25584008). Additionally, the TP53 R248L mutation failed to suppress colony growth in a colony reduction assay compared to wildtype TP53 (PMID: 25584008). Ovarian cancer patient samples and mouse models expressing this mutation were sensitive to an amyloid aggregate TP53 inhibitor as evidenced by apoptosis assays. Some TP53 mutations have been shown to form amyloid aggregates that leads to inactivation of the TP53 protein. Responsiveness of this variant to amyloid aggregate inhibitors suggest that this is a loss-of-function mutation (PMID: 26748848).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'de24f368-de4c-41f1-8c8d-9559e23bff2b',7157,NULL),(28755,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-04-11 16:03:28',NULL,NULL,NULL,NULL,'6bda7bec-479c-42c6-bf32-97df8bb7e15b',7157,NULL),(28756,NULL,'The TP53 G334R mutation occurs in the protein\'s oligomerization domain and has been found as a germline low-penetrance founder mutation in individuals of Ashkenazi Jewish descent who have evidence of Li-Fraumeni syndrome (PMID: 32675277). Expression of this mutation in TP53-null cells did not alter TP53 transactivation in luciferase assays or colony growth as compared to expression of wildtype TP53 (PMID: 25584008). However, in a separate study, cells expressing TP53 G334R demonstrated impaired suppression ability in a colony reduction assay compared to wildtype cells. Additionally, immortalized G334R-mutant lymphoblastoid cell lines demonstrated impaired activation of a subset of TP53 target genes (including PCLO, PLTP, PLXNB3 and LCN15) following treatment with Nutlin-3A.  Further analysis showed TP53 G334R had reduced thermal stability compared to wild type cells (PMID: 32675277).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2022-04-18 12:43:42',NULL,NULL,NULL,NULL,'bd776b5a-dfb1-4314-a8d3-5ef37e793e24',7157,NULL),(28757,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-07-23 11:34:32',NULL,NULL,NULL,NULL,'c8bbc98a-d6f3-4120-81eb-c92c3fdb212b',7157,NULL),(28758,NULL,'The TP53 R282W mutation is located in the DNA binding domain of the protein. This mutation has been identified in glioma and colorectal cancer and is a statistically significant hotspot (PMID: 25078279, 21458303, 26619011). There is conflicting evidence of the functional effect of this mutation. In vitro studies of TP53-null cell lines expressing TP53 R282W demonstrate that the mutation is likely inactivating as measured by destabilization of TP53, inhibited AMPK activation in response to glucose deprivation and increased cellular growth and invasion compared to wildtype (PMID: 17401432, 31395785, 24857548, 21445056). In contrast, other in vitro studies of TP53-null cell lines expressing TP53 R282W demonstrate that the mutation could be likely activating as measured by upregulated CYP3A4 levels, increased transactivation of p21, increased protein stability, increased MIR155HG expression and increased cellular growth and invasion compared to wildtype (PMID: 24603336, 15077194, 37030635, 22797073).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2024-07-23 11:35:22',NULL,NULL,NULL,NULL,'1e75a314-3d89-4e02-9292-5f3a5e1bc2f5',7157,NULL),(28759,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'30e07ac9-9569-4563-ab13-29d8d7c74d59',7157,NULL),(28760,NULL,'The TP53 P250L mutation is located in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines resulted in aggregation of mutant unfolded proteins, likely leading to inactivation of TP53 and upregulation of heat shock proteins (PMID: 21445056).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'506e75cd-a2e9-42bb-b2b4-67a6966c572f',7157,NULL),(28761,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e66907cb-2c8d-4177-8fa5-5384f62831c7',7157,NULL),(28762,NULL,'The TP53 E258V mutation is located in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines resulted in aggregation of mutant unfolded proteins, likely leading to inactivation of TP53 and upregulation of heat shock proteins (PMID: 21445056).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f6bc0d78-5a1d-4d01-ad6b-956b3592df46',7157,NULL),(28763,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'34a8e91b-0b5f-470a-aa87-0fdc8af3b9e7',7157,NULL),(28764,NULL,'The TP53 R110L mutation is located in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines resulted in aggregation of mutant unfolded proteins, likely leading to inactivation of TP53 and upregulation of heat shock proteins (PMID: 21445056). Breast cancer cell lines expressing this mutation were dependent of the expression of GEF-H1, a guanidine nucleotide exchange factor (PMID: 16778209).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6e3b9ea7-4819-4573-ab6a-42897c1f12c7',7157,NULL),(28765,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f12818a2-8ef1-42e9-912e-5e4d8bcf2209',7157,NULL),(28766,NULL,'The TP53 R110P mutation is located in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines resulted in aggregation of mutant unfolded proteins, likely leading to inactivation of TP53 and upregulation of heat shock proteins (PMID: 21445056).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9dbab568-5e51-425b-a92c-74e4e763b85b',7157,NULL),(28767,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6d42a140-887f-4685-a040-f4f6bcd1cc39',7157,NULL),(28768,NULL,'The TP53 H168R mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in TP53-null lung cancer cell lines results in reduced TP53 transactivation activity, as evidenced by luciferase reporter assays. Cell lines expressing this mutation also have a defect in cell cycle progression and aberrant expression of apoptosis proteins, as compared to cells expressing wildtype TP53 (PMID: 18996393).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'57cc120f-bba4-45da-952a-c728c064e5c7',7157,NULL),(28769,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'81e84053-bb4f-4799-aad0-4ec03e4a39b1',7157,NULL),(28770,NULL,'The TP53 T123A mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in TP53-null lung cancer cell lines did not significantly alter TP53-induced transactivation activity, apoptosis, or cell cycle progression as compared to cells expressing wildtype TP53 (PMID: 18996393). Other reports suggest that this mutation results in a gain-of-function, leading to enhanced apoptosis (PMID: 16687402).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3ab614df-855d-45dc-bef9-b9940929c473',7157,NULL),(28771,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'473e1586-aee3-40ac-b278-ccb486b9a7ed',7157,NULL),(28772,NULL,'The TP53 H168N mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this mutation had DNA binding ability approximately 70% of that of wildtype P53, but no difference in exonuclease activity (PMID: 19462533). Further functional assays need to be completed to determine if this alteration is oncogenic.','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'896cbb36-59a2-4d4c-bd85-92f94a52591c',7157,NULL),(28773,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-02-23 16:16:27',NULL,NULL,NULL,NULL,'fa2463e4-3c5e-4877-a352-6d0434cad166',7157,NULL),(28774,NULL,'The TP53 H233N mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this mutation did not alter the DNA binding ability of TP53 compared to wildtype TP53 (PMID: 19462533). However, this mutation is a hotspot, suggesting that it is likely oncogenic (PMID: 26619011).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'cec92b8a-1cb2-4db1-996e-7c896ce1d025',7157,NULL),(28775,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'89e26112-9021-49f1-8a9b-1e4ebb985344',7157,NULL),(28776,NULL,'The TP53 H297N mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this mutation did not alter the DNA binding ability of TP53 (PMID: 19462533). Further functional assays need to be completed to determine if this alteration is oncogenic.','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3a70d2c9-cad1-447c-b723-4d37b83dcdb6',7157,NULL),(28777,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'487a1829-f2f5-430e-c070-c21914273612',7157,NULL),(28778,NULL,'The TP53 H214N mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this mutant protein has reduced DNA binding activity, suggesting it is loss-of-function. Further functional assays are required to determine if this alteration is oncogenic (PMID: 19462533).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8d1e100f-e8d1-47fa-92f3-8bcbfa7341d8',7157,NULL),(28779,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'40665770-9057-44fc-841e-8592998a1b1f',7157,NULL),(28780,NULL,'The TP53 H115N mutation is located in the protein\'s DNA binding domain. In biochemical assays, this mutation has an enhanced ability to bind DNA compared to wildtype TP53, but reduced exonuclease activity. Expression of this mutation in lung cancer cell lines results in cell cycle arrest but this mutation cannot induce TP53-mediated apoptosis (PMID: 19462533).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'183c70aa-2587-44e9-923d-88efd0e1408c',7157,NULL),(28781,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c2c4b2c5-99bd-4df1-9318-540401b18d4e',7157,NULL),(28782,NULL,'The TP53 A161T mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines suggested that this mutation is inactivating as measured by increased protein expression compared to wildtype p53, which suggests lack of negative regulation characteristic of TP53 inactivating mutations (e.g., R175H) (PMID: 21232794, 16827139).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'cb0a7e75-bea1-4f82-b539-14d1f97a5e0a',7157,NULL),(28783,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d92d9394-98d2-4a67-9dbf-ca95437df6a4',7157,NULL),(28784,NULL,'The TP53 G266R mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cell lines resulted in decreased TP53-induced transactivation, as evidenced by luciferase assays (PMID: 16827139).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'85b319f4-43ac-48b5-9846-3e25c8dd0ec5',7157,NULL),(28785,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'66bc592d-7d16-49d1-90dd-4141eb37b932',7157,NULL),(28786,NULL,'The TP53 A161S mutation is located in the protein\'s DNA binding domain. Expression of this mutation in head and neck cancer cell lines results in the inability of TP53 to suppress colony growth, as compared to wildtype TP53. In addition, this alteration leads to reduced expression of p21 with implications for apoptosis (PMID: 25119136).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f110691a-6247-4b25-be7a-ba9c97938b7c',7157,NULL),(28787,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'12191136-e015-4953-8744-7c7dfa6c3c99',7157,NULL),(28788,NULL,'The TP53 V197E mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cells resulted in an increased interaction with PBF, an oncogene implicated in the regulation of TP53 target genes, such as p21 (PMID: 25408419).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'15c2897e-fc8d-4b79-8a2f-7076876796b0',7157,NULL),(28789,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'587e949b-673e-4f92-974e-e105135873ae',7157,NULL),(28790,NULL,'The TP53 M133K mutation occurs in the protein\'s DNA binding domain and likely confers loss of function to the expressed p53 protein. The TP53 M133K mutation is associated with loss of p53 transcriptional activity. Moreover, expression of this mutation in TP53-null cells was associated with an aberrant increased interaction with PBF, an oncogene that interferes in p53 transcriptional activity and consequent deregulation of its target genes, such as p21 (PMID: 25408419).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-05-18 19:36:09',NULL,NULL,NULL,NULL,'ca002c0f-42bc-4ac9-b9c9-eee83096a9e9',7157,NULL),(28791,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8339d09f-c7a8-4db0-b3f1-298a69f81ea2',7157,NULL),(28792,NULL,'The TP53 I255F mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in TP53-null cells resulted in an increased interaction with PBF, an oncogene implicated in the regulation of TP53 target genes, such as p21 (PMID: 25408419).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2b8cee54-b7ba-4e4c-b45e-b25824b8f695',7157,NULL),(28793,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'ad0d09f8-70be-414a-9ee6-87e3bb76dc89',7157,NULL),(28794,NULL,'The TP53 V197L mutation occurs in the protein\'s DNA binding domain. Endogenous expression of this mutation in a glioblastoma cell line resulted in decreased apoptosis and aberrant cell cycle progression, likely due to decreased TP53 transactivation (PMID: 11668476)','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'de75d745-2950-4466-aaa3-8dc2c2fbf549',7157,NULL),(28795,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'334f8e20-24e5-4a30-b2d7-dca0c275cf3f',7157,NULL),(28796,NULL,'The TP53 M133T mutation occurs in the protein\'s DNA binding domain. Endogenous germline expression of this mutation in breast cancer cells results in increased expression of the anti-apoptotic gene BIRC3, likely due to alerted TP53 transcriptional activation. Expression of this mutation also unregulated the inflammatory response, including increased expression of IL-1beta (PMID: 21311097).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'ea618d9d-5503-4381-8eb4-f3ee19b20fc2',7157,NULL),(28797,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'087e4522-ac70-41d1-bb2c-961fa38cb050',7157,NULL),(28798,NULL,'The TP53 C141Y mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in cell lines that lack TP53 expression did not enhance TP53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2b6e32cb-8db4-4dac-8bc7-110737b520fe',7157,NULL),(28799,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5b0c5073-9772-4497-98d8-3ce215950ab2',7157,NULL),(28800,NULL,'The TP53 V173L mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in cell lines that lack TP53 expression did not enhance TP53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'de48d259-cb71-42e1-87fe-5f46d1b47935',7157,NULL),(28801,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e635bb54-cf10-481c-ba9a-b74a65f4174a',7157,NULL),(28802,NULL,'The TP53 V173M mutation occurs in the protein\'s DNA binding domain and likely confers loss of function to the expressed p53 protein. When TP53 V173M was expressed in p53-null cells, a transient reporter gene assays demonstrated mutant p53 was impaired from binding to the promoters of its target genes and consequent lowered transcriptional activity, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-05-18 19:24:57',NULL,NULL,NULL,NULL,'e6705b65-cad4-47ce-8106-c2ec8e4ee65f',7157,NULL),(28803,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'b2d63dac-4119-4c87-a295-cb1d5da7be27',7157,NULL),(28804,NULL,'The TP53 Y205C mutation occurs in the protein\'s DNA binding domain and likely confers loss of function to the expressed p53 protein. When TP53 Y205C was expressed in p53-null cells, a transient reporter gene assays demonstrated mutant p53 was impaired from binding to the promoters of its target genes and consequent lowered transcriptional activity, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-05-18 19:24:01',NULL,NULL,NULL,NULL,'43d62edb-4151-466d-86cd-3c2055e93fd1',7157,NULL),(28805,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'868609a3-c2ff-430e-9927-676f426b3c00',7157,NULL),(28806,NULL,'The TP53 R273L mutation occurs in the protein\'s DNA binding domain (DBD). Mutations in the DBD result in altered contact of p53 with its target DNA sequences, thereby altering its transcriptional function (PMID: 8023157, 11900253). Given that p53 directs the transcription of proteins that enable apoptosis (PMID: 11900253), its inactivation results in cells harboring damaged DNA and overall genomic instability (PMID: 11900253). Expression of the R273L mutation in cell lines that lack p53 expression did not enhance p53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'cb4e3b94-9457-4208-b286-87908f41f6e8',7157,NULL),(28807,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bf7eadb3-637c-47f1-9e59-cdda77d7eb0d',7157,NULL),(28808,NULL,'The TP53 E286K mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in cell lines that lack TP53 expression did not enhance TP53-mediated transcriptional activity in reporter assays, as compared to expression of wildtype TP53 (PMID: 15037740).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9b80ab13-f126-40c6-bc44-98dc7fd22a90',7157,NULL),(28809,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e40a0483-c475-4af3-9f8b-c621def6afe8',7157,NULL),(28810,NULL,'The TP53 I251S mutation occurs in the protein\'s DNA binding domain. Ovarian cancer patient samples expressing this mutation were sensitive to an amyloid aggregate TP53 inhibitor, as evidenced by apoptosis assays. Some TP53 mutations have been shown to form amyloid aggregates that lead to inactivation of the TP53 protein. Responsiveness of this variant to amyloid aggregate inhibitors suggest that this is a loss-of-function mutation (PMID: 26748848).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f64f1d16-1bb7-4a21-8321-b41b971cc019',7157,NULL),(28811,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bc7d3729-c0f3-46a7-a742-a3a6bf468b54',7157,NULL),(28812,NULL,'The TP53 E326L mutation occurs in the protein\'s DNA binding domain. Ovarian cancer patient samples expressing this mutation were sensitive to an amyloid aggregate TP53 inhibitor as evidenced by apoptosis assays. Some TP53 mutations have been shown to form amyloid aggregates that lead to inactivation of the TP53 protein. Responsiveness of this variant to amyloid aggregate inhibitors suggest that this is a loss-of-function mutation (PMID: 26748848).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-05-02 15:56:53',NULL,NULL,NULL,NULL,'1cdf7555-cb23-4b2f-875f-0e187e1bfe89',7157,NULL),(28813,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'6701ca7c-9543-44a6-847a-084a82ccced7',7157,NULL),(28814,NULL,'The TP53 R181C mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in osteosarcoma cell lines that lack wildtype TP53 results in decreased transactivation of genes important for the cell cycle and apoptosis, including p21 and MDM2. Cell lines expressing this mutation exhibited decreased apoptosis and decreased DNA binding ability as compared to cell lines expressing wildtype TP53 (PMID: 10229196, 20471942).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8abe3752-f25a-47dc-9e3a-450a580e2f8b',7157,NULL),(28815,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d9d4aa47-93cf-4d98-8451-89cc44ad7ee1',7157,NULL),(28816,NULL,'The TP53 R202S mutation is located in the protein\'s DNA binding domain. Expression of this mutation in TP53-null osteosarcoma cell lines demonstrated that TP53 R202S is inactivating as measured by decreased transactivation of genes important in cell cycle and apoptosis, including p21 and BAX, compared to cell lines that express wildtype TP53 (PMID: 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-03-24 11:36:40',NULL,NULL,NULL,NULL,'e4790ef9-3603-44bc-aa7f-d07d1c34c324',7157,NULL),(28817,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8dcd4c8c-88b9-48e8-8328-0112ce40365f',7157,NULL),(28818,NULL,'The TP53 T150I mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in osteosarcoma cell lines that lack wildtype TP53 results in decreased transactivation of genes important in cell cycle and apoptosis, including p21 and Bax. Cell lines expressing this mutation have decreased apoptosis as compared to cell lines that express wildtype form of TP53 (PMID: 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'31eeba79-dd25-4024-85f2-564d6f862c1e',7157,NULL),(28819,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7c310f91-960c-410d-bf87-03e8aa8d8102',7157,NULL),(28820,NULL,'The TP53 S240R mutation occurs in the DNA binding domain of the protein. Expression of this mutation in osteosarcoma cell lines that lack wildtype p53 demonstrated that this mutation is likely inactivating as measured by decreased transactivation of genes important for cell cycle regulation and apoptosis, including p21 and Bax, as well as decreased apoptosis compared to cells expressing wildtype p53 (PMID: 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f3b9ec06-6ec6-404a-8789-db95f5809e6e',7157,NULL),(28821,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'59d77dea-2f64-4eaf-9270-551432c9aba1',7157,NULL),(28822,NULL,'The TP53 G199R mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in osteosarcoma cell lines that lack wildtype TP53 results in decreased transactivation of genes important in cell cycle and apoptosis, including p21 and Bax. Cell lines expressing this mutation have decreased apoptosis as compared to cell lines that express wildtype form of TP53 (PMID: 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bad89955-4825-4219-803b-6f11d4d016f1',7157,NULL),(28823,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'dbf32171-2b62-4a45-8ff3-ee8cbe0647f2',7157,NULL),(28824,NULL,'The TP53 Q144P mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in osteosarcoma cell lines that lack wildtype TP53 results in decreased transactivation of genes important in cell cycle and apoptosis. Cell lines expressing this mutation have decreased apoptosis and enhanced cell growth as compared to cell lines that express wildtype form of TP53 (PMID: 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'1b3bda9d-0a5f-4721-9fe2-41d430c8e859',7157,NULL),(28825,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'053f67d8-ad36-423d-b3c3-7787304d7d20',7157,NULL),(28826,NULL,'The TP53 Y163N mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in osteosarcoma cell lines that lack wildtype TP53 is associated with decreased transcriptional activation of the p53 target gene p21. Additionally, cell lines expressing this mutation have decreased apoptosis and enhanced cell growth as compared to cell lines that express wildtype form of TP53 (PMID: 10229196) suggesting that this mutation confers loss of p53 function.','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-05-18 19:16:45',NULL,NULL,NULL,NULL,'d039e7b7-0731-481b-8e82-9ba69812def5',7157,NULL),(28827,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'1f66b84a-d75d-489b-bec8-e6fa2652f374',7157,NULL),(28828,NULL,'The TP53 H168Y mutation occurs in the protein\'s DNA binding domain and likely confers loss of function to the p53 protein. Expression of this mutation in osteosarcoma cell lines that lack wildtype TP53 was associated with decreased transcription of p53 target genes. Cell lines expressing this mutation have decreased apoptosis and enhanced cell growth as compared to cell lines that express wildtype form of TP53 (PMID: 10229196).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-05-18 19:28:03',NULL,NULL,NULL,NULL,'17a2a1b1-4a50-4190-a8ee-4c4cc6736ca0',7157,NULL),(28829,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2fed257c-1ac2-4e09-9def-2ff0e1f3b16d',7157,NULL),(28830,NULL,'The TP53 R181H mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant was found to have defective DNA binding activity compared to wildtype TP53. Expression of this mutation in osteosarcoma cell lines resulted in decreased transactivation of TP53 target genes and reduced apoptosis (PMID: 20471942).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f25a8db0-d6db-4f12-9868-3257e32db0d5',7157,NULL),(28831,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'001f2b88-fb31-4d3e-a0d2-dea646391c92',7157,NULL),(28832,NULL,'The TP53 R181P mutation occurs in the protein\'s DNA binding domain and likely confers loss of function to the p53 protein. In biochemical assays, this TP53 mutant was found to completely lose DNA binding activity. Expression of this mutation in osteosarcoma cell lines resulted in decreased transactivation of TP53 target genes and reduced apoptosis (PMID: 20471942).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-05-18 19:40:42',NULL,NULL,NULL,NULL,'7f5362d2-5cc6-4a0b-a07d-52e0b3df046e',7157,NULL),(28833,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'564683ab-c62a-46b8-ba84-910976339f36',7157,NULL),(28834,NULL,'The TP53 E180K mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant was found to have defective DNA binding activity compared to wildtype TP53 (PMID: 24814347). Expression of this mutation in osteosarcoma cell lines resulted in decreased transactivation of TP53 target genes and reduced apoptosis (PMID: 20471942).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'e4542676-3410-4b9a-85fa-3afc44d88a05',7157,NULL),(28835,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7ea701e7-d356-4122-b740-c18b729170d7',7157,NULL),(28836,NULL,'The TP53 R181L mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant was found to have defective DNA binding activity compared to wildtype TP53. Expression of this mutation in osteosarcoma cell lines resulted in decreased transactivation of TP53 target genes and reduced apoptosis (PMID: 20471942).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9e42d6df-dbb2-49ae-be6d-f5ffd6064262',7157,NULL),(28837,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'cb05874e-484e-4f65-bde5-6f5bec6603d0',7157,NULL),(28838,NULL,'The TP53 K120R mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in lung cancer cell lines demonstrated that this mutant is defective for TP53 acetylation, compared to wildtype TP53. Loss of acetylation results in defective apoptosis, without impacting cell cycle arrest (PMID: 17189187, 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'40e1aac7-b29e-44bb-a6aa-bde06b842413',7157,NULL),(28839,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3fbb0295-0363-42be-abac-cb74f0269645',7157,NULL),(28840,NULL,'The TP53 K120M mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in lung cancer cell lines result in decreased cell cycle arrest, compared to expression of wildtype TP53 (PMID: 17189187).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'a431dbc8-2140-4cef-920c-ad433311b7c5',7157,NULL),(28841,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3d957e82-4398-4cf4-a5b4-3615ca706728',7157,NULL),(28842,NULL,'The TP53 K120E mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in lung cancer cell lines resulted in decreased cell cycle arrest, compared to expression of wildtype TP53 (PMID: 17189187, 24814347). In colony formation assays, expression of this mutation resulted in increased cell growth compared to wildtype TP53. Expression of this mutation resulted in decreased p21 and MDM2 expression and defective cytochrome c release from the mitochondria (PMID: 18524770).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'3bee033c-f24a-4c07-869c-251203a4b891',7157,NULL),(28843,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9809454f-de0b-4cf6-b098-849358bd0f84',7157,NULL),(28844,NULL,'The TP53 R273K mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'dbce0662-29d0-4b88-bf3c-49a4526b37be',7157,NULL),(28845,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2f740ed9-536a-4f19-be8d-bb9b5ab97f9e',7157,NULL),(28846,NULL,'The TP53 K120N mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'79869457-41eb-4e30-86d6-f5ccd09dc513',7157,NULL),(28847,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d7744220-e23f-4140-a619-26514d841fda',7157,NULL),(28848,NULL,'The TP53 C277Q mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'93b63170-5476-4728-ab27-c4617f921b46',7157,NULL),(28849,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'099645c0-c531-4b22-b6e0-f6e0a0a69630',7157,NULL),(28850,NULL,'The TP53 C277R mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c331d331-6191-41ae-9730-5b0ca48ecf61',7157,NULL),(28851,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'a4d9e08e-2675-4318-869f-696e2d2faf0b',7157,NULL),(28852,NULL,'The TP53 R248K mutation occurs in the protein\'s DNA binding domain. Biochemical assays have demonstrated that this TP53 mutation is inactivating as measured by decreased ability to bind DNA compared to wildtype TP53 (PMID: 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-13 15:41:16',NULL,NULL,NULL,NULL,'a2630e1b-9300-486c-8077-f37d23f085ba',7157,NULL),(28853,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c2d1a9ff-f41f-4896-96ef-b68381e1ea7a',7157,NULL),(28854,NULL,'The TP53 R283Q mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 7651437).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'a4a9d1be-14a2-4cfe-843a-3a422b412940',7157,NULL),(28855,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'7b5e897c-ab0d-46d0-bc07-fe5b93fb2d57',7157,NULL),(28856,NULL,'The TP53 H178D mutation occurs in the protein\'s DNA binding domain and likely confers loss of function to the p53 protein. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA of apoptosis effecters Puma and p21, but not anti-apoptotic mediators such as Bcl-xl compared to wildtype TP53 (PMID: 24814347).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-05-18 19:39:38',NULL,NULL,NULL,NULL,'adf3b436-da78-45c0-8582-fce8b3f1a868',7157,NULL),(28857,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'48494947-6086-43b6-9a57-baae32aa168f',7157,NULL),(28858,NULL,'The TP53 D186A mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 24814347).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'38fad032-750c-4204-9578-cb2c8c76b486',7157,NULL),(28859,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'9a3c3525-cce5-49c8-9d77-30e03bd577b5',7157,NULL),(28860,NULL,'The TP53 C277W mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 24814347).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'a0de3084-e532-43b3-8bc1-a14d7b990697',7157,NULL),(28861,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'18decb14-dbd7-4df7-8da2-d2b8836a7303',7157,NULL),(28862,NULL,'The TP53 K120A mutation occurs in the protein\'s DNA binding domain. In biochemical assays, this TP53 mutant had a decreased ability to bind DNA compared to wildtype TP53 (PMID: 24814347). Expression of this mutation in cell lines resulted in cell cycle defects and reduced apoptosis, consistent with loss of TP53 activity (PMID: 16687402).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'b105cd98-386a-4031-959f-ebd8b9d9a701',7157,NULL),(28863,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f8157c54-5106-4e26-869c-77fca2f5bba5',7157,NULL),(28864,NULL,'The TP53 R280A mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in lung cancer cell lines resulted in decreased cell cycle arrest, compared to expression of wildtype TP53 (PMID: 17189187, 24814347). In colony formation assays, expression of this mutation resulted in increased cell growth compared to wildtype TP53. Expression of this mutation resulted in decreased p21 and MDM2 expression, consistent with defects in cell cycle (PMID: 18524770).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'f48417f5-975b-424d-8f8f-b77e66dab9cb',7157,NULL),(28865,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c4a77df5-6243-487a-84b5-df937ddc7b4e',7157,NULL),(28866,NULL,'The TP53 V143A mutation occurs in the protein\'s DNA binding domain. Expression of this mutation in osteosarcoma TP53-null cell lines resulted in increased colony formation compared to cells expressing wildtype TP53 (PMID: 10435620).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5c6716f8-583a-42ec-97bf-a9c00ccd4ebc',7157,NULL),(28867,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2024-05-30 13:33:01',NULL,NULL,NULL,NULL,'0acca871-c27a-481c-a750-f3172f195b28',7157,NULL),(28868,NULL,'The TP53 R290H mutation occurs in the protein\'s DNA binding domain. In vitro studies with TP53-null Saos-2 cells expressing TP53 R290H demonstrate inconclusive evidence of the oncogenic function of the mutation as measured by slightly lower growth suppression and reduced p21 protein expression compared to wildtype and TP53 transcriptional activity being comparable to wildtype (PMID: 10435620). In contrast, in vitro studies with TP53-null HCT116 cells and lymphocytes expressing TP53 R290H demonstrate that the mutation is likely neutral as measured by DNA-binding activity and transcriptional activity being comparable to wildtype (PMID: 24076587, 28369373).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2024-05-30 13:33:03',NULL,NULL,NULL,NULL,'baab13ee-970b-4ef1-b408-718a71f4b577',7157,NULL),(28869,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2025-07-07 11:26:56',NULL,NULL,NULL,NULL,'41656930-7b09-48a0-8b39-dbf18ff23110',7157,NULL),(28870,NULL,'The TP53 R156H mutation is located in the DNA-binding domain of the protein. There is conflicting evidence on the oncogenic effect of this mutation. In vitro studies with TP53-null osteosarcoma cell lines expressing TP53 R156H demonstrate reduced transactivation of p21 and tumor cell growth suppression that is comparable to wildtype (PMID: 10435620). In vivo studies with yeast expressing TP53 R156H demonstrate that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). Other preclinical studies with TP53-null cells and yeast expressing TP53 R156H demonstrate that the mutation is likely neutral as measured by induction of apoptosis, transcriptional activity and growth suppression comparable to wildtype (PMID: 30840781).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2025-07-07 11:34:55',NULL,NULL,NULL,NULL,'3dc0d794-ce11-45dd-b380-b910685fd5ad',7157,NULL),(28871,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'74dec252-1b48-4ec4-acd8-156813aad0c4',7157,NULL),(28872,NULL,'The TP53 R267P mutation occurs in the protein\'s DNA binding domain. Lung cancer cell lines that have endogenous expression of this mutation are sensitive to knockdown of TP53 in growth assays. Expression of this mutation in cell lines results in enhanced expression of the tyrosine kinase Axl (PMID: 22989750, 23264849).','MUTATION_EFFECT',NULL,_binary '\0','Likely Gain-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'8ed97575-e8de-466e-937c-44743e090b2c',7157,NULL),(28873,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-02-23 16:13:18',NULL,NULL,NULL,NULL,'e942a3c9-f18d-46af-a716-00acd11193b3',7157,NULL),(28874,NULL,'The TP53 Q331H mutation occurs in the protein\'s tetramerization domain. In biochemical assays, this TP53 mutation enhanced the formation of TP53 tetramers (PMID: 20978130). However, this mutation is a hotspot, suggesting that it is likely oncogenic (PMID: 26619011).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'d896c2aa-a20c-46b4-a0dc-8b8dd6392e70',7157,NULL),(28875,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'b8b7d2b2-3dcc-484e-a76b-31d1f0ec49d9',7157,NULL),(28876,NULL,'The TP53 T329I mutation occurs in the protein\'s tetramerization domain. In biochemical assays, this TP53 mutation enhanced the formation of TP53 tetramers (PMID: 20978130). Further functional data is required to determine if this mutation is indeed gain-of-function.','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'c19b8f02-f19f-43b5-a019-4b777367ddcf',7157,NULL),(28877,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'53587bdf-fc1e-4550-bd94-880815ec8356',7157,NULL),(28878,NULL,'The TP53 Q331R mutation occurs in the protein\'s tetramerization domain. In biochemical assays, this TP53 mutation enhanced the formation of TP53 tetramers (PMID: 20978130), which could lead to alteration of p53 function','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'5375dc0c-395c-43ff-be5a-598d1f79a658',7157,NULL),(28879,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'47ec5f61-9067-43ff-89df-d8f1df0a6786',7157,NULL),(28880,NULL,'The TP53 G356A mutation occurs in the protein\'s tetramerization domain. In biochemical assays, this TP53 mutation enhanced the formation of TP53 tetramers (PMID: 20978130). Further functional data is required to determine if this mutation is indeed gain-of-function.','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2017-05-02 15:44:33',NULL,NULL,NULL,NULL,'74c7493c-85e4-4ee9-c02b-b493955e323a',7157,NULL),(28881,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'063c9847-e0fd-47e5-9261-b7d6407615f1',7157,NULL),(28882,NULL,'The TP53 P316L mutation is located in the protein\'s most N-terminal signal (NLS1) domain. Mutation in this domain has been shown to induce the synthesis of a cytoplasmic p53 protein, thus interfering with normal p53 protein localization (PMID: 2553699, 10618702, 20010306 ).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'bd32563e-1c00-463e-a143-132858c5e3f1',7157,NULL),(28883,NULL,'This assertion is supported by (PMID: 25860933, 25412851).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'ee6b8c2d-0e7a-42bb-ab8f-ef29344cd081',7157,NULL),(28884,NULL,'The WHO classification of hematologic tumors list TP53 alterations as a prognostic biomarker in [[tumor type]]. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'507da72f-9996-47e7-935f-2477a4b97c40',7157,NULL),(28885,NULL,'This assertion is supported by (PMID: 22186996, 18596741).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'e43a1d21-8fe5-454e-8200-34d72533a72e',7157,NULL),(28886,NULL,'The NCCN guidelines list TP53 alterations as a prognostic biomarker in acute myeloid leukemia. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'368ceece-c5c4-409b-8820-d6fad62beeb1',7157,NULL),(28887,NULL,'This assertion is supported by (PMID: 25412846, 25952993).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'17eadd11-3a15-408a-853b-121286199008',7157,NULL),(28888,NULL,'The NCCN guidelines list TP53 alterations as a prognostic biomarker in therapy-related myeloid neoplasms. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2021-01-07 14:12:15',NULL,NULL,NULL,NULL,'20ba01fb-bdf4-4de4-9c2a-afc806df84b9',7157,NULL),(28889,NULL,'This assertion is supported by (PMID: 19188171, 20697090, 24652989, 23243274, 24004666).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'8d78d7c1-2bf2-4199-9d9b-8fdb688bb9be',7157,NULL),(28890,NULL,'The NCCN guidelines list TP53 alterations as a prognostic biomarker in [[tumor type]]. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'c384db22-b61e-4c46-b4b6-623d01adcaa2',7157,NULL),(28891,NULL,'This assertion is supported by (PMID: 24220272, 21714648, 25092778).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'4e8b3dca-b8a1-43bf-986a-54d18b445ceb',7157,NULL),(28892,NULL,'The NCCN guidelines list TP53 alterations as a prognostic biomarker in myelodysplastic syndromes. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'a17bc9c0-d851-4c77-b257-e713827946b9',7157,NULL),(28893,NULL,'This assertion is supported by (PMID: 22887079, 29296692, 24478400).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'b6f54e8b-aa53-4c55-9cd8-5edc0c81fb77',7157,NULL),(28894,NULL,'The NCCN guidelines list TP53 alterations as a prognostic biomarker in [[tumor type]]. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'d9fa3911-27ea-47af-bc47-e02a74df75bf',7157,NULL),(28895,NULL,'This assertion is supported by (PMID: 22887079, 25516983, 22052707).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'b487b052-7db8-4a2c-8441-ee6b295823c6',7157,NULL),(28896,NULL,'The NCCN guidelines list TP53 alterations as a prognostic biomarker in primary myelofibrosis. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'6302e148-b612-4d7f-bcfb-537512d7b0df',7157,NULL),(28897,NULL,'This assertion is supported by (PMID: 22887079, 25516983, 22052707).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'7940b806-79c6-4cda-a468-befdd8806bb7',7157,NULL),(28898,NULL,'The WHO classification of hematologic tumors list TP53 alterations as a prognostic biomarker in [[tumor type]]. Well-powered studies find that patients with a TP53 alteration such as [[gene]] [[mutation]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'ee5ad677-d86e-4fb2-8324-b52072a2a559',7157,NULL),(28899,NULL,'This assertion is supported by (PMID: 28819011, 26022239, 24684350, 8639789, 7579380).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px3',NULL,NULL,'e05be7f2-30ba-4b12-84ab-aeba2c0e29ab',7157,NULL),(28900,NULL,'Clinical studies suggest that patients with TP53 altered [[tumor type]] have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'6f6f4c38-43a8-4ea3-a0a7-68a3d515b009',7157,NULL),(28901,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-03-18 20:38:38',NULL,NULL,NULL,NULL,'62b6cb66-9135-4a95-9032-caedfa3fe453',7157,NULL),(28902,NULL,'The TP53 S215G mutant is located in the DNA binding domain of the protein. Mutations in the DNA binding domain are associated with altered contact of p53 with its target DNA sequences, thereby altering its transcriptional function (PMID: 8023157, 11900253). Functional studies in yeast and human cell lines suggest this mutation is temperature sensitive and inactivating, as measured by functional analysis of separated alleles in yeast (FASAY) (PMID: 17724467, 22862161). This mutation is also a hotspot, suggesting that it is likely oncogenic (PMID: 26619011).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2023-01-18 19:04:06',NULL,NULL,NULL,NULL,'a3aead6a-7ad7-4381-9b21-81c71f99515a',7157,NULL),(28903,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'daf0222b-c413-49d6-b893-eb9b1e39951e',7157,NULL),(28904,NULL,'The TP53 L145R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in melanoma and is a statistically significant hotspot (PMID: 20571075). In vivo studies with yeast expressing TP53 L145R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L145R also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Loss-of-function','2023-03-23 15:34:28',NULL,NULL,NULL,NULL,'a65bca08-1b2c-4423-ab16-eabe9e9a14f7',7157,NULL),(28905,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'2e4fa506-8016-4475-9c80-0226dc103dd0',7157,NULL),(28906,NULL,'TP53 deletions are predicted to be inactivating and associated with poor prognosis (PMID: 11900253, 20697090, 21467160, 12584563). Experimental studies have revealed that deletions result in loss of TP53 tumor suppressive function, since TP53 cannot decrease cell viability in vitro (PMID: 27759562). Additionally, deletion of TP53 in the neurons of mice, while not sufficient to cause tumor development, allows the accumulation of other oncogenic mutations leading to tumorigenesis (PMID: 19477430). Further, loss of TP53 protein during 17p deletion events in leukemia/lymphoma, enhances cancer development through cooperation of linked tumor suppressive genes such as EIF5A and ALOX15B (PMID: 26982726).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2019-07-18 15:06:42',NULL,NULL,NULL,NULL,'2e3b9a82-adcc-4067-9ebc-e394a9ca3081',7157,NULL),(28907,NULL,'This assertion is supported by (PMID: 9680348, 18337559, 17209057 Abstract: Hofste op Bruinink, D. et al. Abstract# Blood 128:3271, 2017. http://www.bloodjournal.org/content/128/22/3271?sso-checked=true; http://www.bloodjournal.org/content/122/21/1846).','PROGNOSTIC_IMPLICATION',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,'LEVEL_Px1',NULL,NULL,'143077a9-08b8-4800-8c52-db80dad34385',7157,NULL),(28908,NULL,'The NCCN guidelines list deletion of TP53 as a prognostic biomarker in [[tumor type]]. Well-powered studies find that patients with deletion of TP53 have a less favorable prognosis.','PROGNOSTIC_SUMMARY',NULL,_binary '\0',NULL,'2020-09-16 17:52:15',NULL,NULL,NULL,NULL,'43ecf7b6-6d53-44b7-a99e-f47d87f9b9bc',7157,NULL),(28909,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:45:46',NULL,NULL,NULL,NULL,'74aec1d1-f984-430f-a431-fca85f5375e5',7157,NULL),(28910,NULL,'Truncating mutations of TP53 occur throughout the gene and lead to the production of several C-terminally truncated protein forms. These alterations are predicted to be inactivating and are associated with poor prognosis (PMID: 11900253, 11753428, 16007150, 21467160, 19336573). Experimental studies have revealed that truncating mutations promote cancer cell proliferation, survival and metastasis, since ectopic expression of these mutations in melanoma cells increased cell motility and tumor formation in vivo. This was due in part to aberrant localization of truncated proteins to the mitochondria, regulating genes involved in cell survival, including CypD (PMID: 27759562).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-06-13 14:53:20',NULL,NULL,NULL,NULL,'72ca72a6-e423-4ace-893e-b5e837f1c2c7',7157,NULL),(28911,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2017-04-20 17:39:17',NULL,NULL,NULL,NULL,'b459bc2f-df4e-4b37-95a1-98670b60bb2e',7157,NULL),(28912,NULL,'The exon 8 TP53 E285K mutation is located in the DNA binding domain of the protein. Expression of TP53 E285K in yeast and human cancer cells demonstrated that it is likely loss of function as evidenced by decreased expression and transactivation compared to wildtype TP53 (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2017-07-19 09:42:12',NULL,NULL,NULL,NULL,'c5aec539-492b-4e93-aab5-901680618a0c',7157,NULL),(28913,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2018-07-24 11:03:06',NULL,NULL,NULL,NULL,'f5224470-2da2-4825-a9ca-08d8a5bcb2b7',7157,NULL),(28914,NULL,'The 3 bp in-frame deletion leading to the p53 P191del mutation arises in the DNA-binding domain of p53. Mutations in the DNA-binding domain are likely loss-of-function, associated with the loss of p53 transcriptional activity (PMID: 8023157). Additionally, this mutation was identified as a recurrent hotspot (statistically significant) in a population-scale cohort of tumor samples of various cancer types using methodology (PMID: 26619011). Lastly, immunohistochemical analysis of a stage IV adrenocortical carcinoma harboring this mutation showed absence of p53 protein (PMID: 11454518).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2018-07-24 11:11:27',NULL,NULL,NULL,NULL,'4bbfb245-8242-4c5f-8ec3-58d0eb2f8fb1',7157,NULL),(28915,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2018-08-09 16:17:55',NULL,NULL,NULL,NULL,'0efd0be7-8d9e-4660-a7cf-88d83f26739c',7157,NULL),(28916,NULL,'The TP53 R196G mutation arises in the p53 DNA-binding domain of the protein. Human H1299 cell transactivation studies have demonstrated that this mutation is inactivating as demonstrated by significantly decreased transcriptional activity of p53 downstream factors bax, p21 and RGC relative to wildtype p53 at two different experimental temperature conditions (32 and 37 degrees Celsius) (PMID: 22710932).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:25:35',NULL,NULL,NULL,NULL,'0b4f34f8-aa1c-4663-b586-eca7ddd34fa1',7157,NULL),(28917,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-07 15:10:49',NULL,NULL,NULL,NULL,'ebf5de87-0bf9-43f1-bd6c-f944f1bf0d49',7157,NULL),(28918,NULL,'The TP53 R196L mutation arises in the p53 DNA-binding domain of the protein. When this mutation is expressed in human Saos cells at different experimental temperature conditions, a difference in transactivational activity of this TP53 mutant has been observed (PMID: 14559903). However, there is no specific data demonstrating whether the mutation has decreased transactivational activity relative to wildtype p53 (PMID: 14559903). Nonetheless, an alternate mutation at this position, the TP53 R196G mutation, has been demonstrated to decrease p53 transactivation activity relative to wildtype p53 (PMID: 22710932). The R196L mutation is also a 3D hotspot mutation and therefore TP53 R196L is considered likely oncogenic (PMID: 28115009).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-07 15:11:09',NULL,NULL,NULL,NULL,'e6263878-9cbb-468a-9837-ef2294db7409',7157,NULL),(28919,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2020-06-10 00:35:09',NULL,NULL,NULL,NULL,'16fba3bd-82c5-4dbf-be5f-b7952ed9f390',7157,NULL),(28920,NULL,'The TP53 C124R mutation is located in the DNA binding domain of the protein. Expression of this mutation in a yeast functional assay demonstrated that it is likely inactivating (PMID: 9115587.)','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2020-06-10 00:35:07',NULL,NULL,NULL,NULL,'1407e432-557d-49ce-b50d-ce74d65fb840',7157,NULL),(28921,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2020-06-10 00:38:38',NULL,NULL,NULL,NULL,'8cf023da-f5e8-4ade-879d-4068589d6971',7157,NULL),(28922,NULL,'The TP53 C141W mutation is located in the DNA binding domain of the protein. Expression of this mutation in biochemical assays demonstrated that it is likely inactivating as measured by function that is moderately altered compared to wildtype (PMID: 20407015).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2020-06-10 00:39:01',NULL,NULL,NULL,NULL,'794e66b7-6319-4180-a829-15f0b558b1b1',7157,NULL),(28923,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2020-06-10 00:52:33',NULL,NULL,NULL,NULL,'0b5519b6-cde0-4060-8f6d-bfdfb69b9f46',7157,NULL),(28924,NULL,'The TP53 C176W mutation is located in the DNA binding domain of the protein. Expression of this mutation in biochemical assays demonstrated that it is likely inactivating as measured by impaired ability to activate downstream target genes compared to wildtype (PMID: 19850740).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2020-06-10 00:54:39',NULL,NULL,NULL,NULL,'7b23803d-83e6-4294-9bf8-8810777a0527',7157,NULL),(28925,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2020-06-10 01:02:03',NULL,NULL,NULL,NULL,'1ec44108-57ab-498c-b1c3-768bec243758',7157,NULL),(28926,NULL,'The TP53 G154V mutation is located in the DNA binding domain of the protein. Biochemical experiments have suggested that this mutation is inactivating as it acts as a dominant-negative (PMID: 21289082).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:25:57',NULL,NULL,NULL,NULL,'58fc793e-8e9e-40ee-bc2c-b1c849a283f1',7157,NULL),(28927,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:55:31',NULL,NULL,NULL,NULL,'abe217c7-44c4-4b46-aeeb-ac64fa8a71db',7157,NULL),(28928,NULL,'The TP53 C277Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Ewing sarcoma and myeloproliferative neoplasms and is a statistically significant hotspot (PMID: 15192123, 35030630). In vitro studies with H1299 cells ectopically expressing TP53 C277Y demonstrated the mutation is inactivating as measured by the mutant\'s significant reduction in apoptosis and downregulation of p21 and BAX expression as compared to wildtype (PMID: 34282274).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:30:24',NULL,NULL,NULL,NULL,'61d808e4-78c4-4611-b571-f87316a5c3ff',7157,NULL),(28929,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:56:20',NULL,NULL,NULL,NULL,'763ff1b3-d610-4906-a9d7-e914bffd62c6',7157,NULL),(28930,NULL,'The TP53 Y236H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer, lung cancer and basaloid pancreatic carcinoma and is a statistically significant hotspot (PMID: 20682393, 32674491, 25432630). In vivo studies with yeast expressing TP53 Y236H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:30:51',NULL,NULL,NULL,NULL,'a225696a-e751-4a12-9603-e2c980449bad',7157,NULL),(28931,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:57:14',NULL,NULL,NULL,NULL,'a9aac0c1-e239-473d-98d4-437c05c03bd1',7157,NULL),(28932,NULL,'The TP53 Y236N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in B-cell precursor acute lymphoblastic leukemia, breast cancer, colon cancer and glioblastoma and is a statistically significant hotspot (PMID: 33095873, 8950983, 9301461, 10427138). In vivo studies with yeast expressing TP53 Y236N demonstrated that the mutation retains its transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 Y236N demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:31:26',NULL,NULL,NULL,NULL,'528eba0e-4a36-481e-be48-779287cec4d7',7157,NULL),(28933,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:57:48',NULL,NULL,NULL,NULL,'eb38de83-1cbb-472a-a5d8-17db280b3d76',7157,NULL),(28934,NULL,'The TP53 H193Q mutation is located in the DNA-binding domain of the protein.  This mutation has been identified in colorectal cancer, esophageal squamous cell carcinoma, bone sarcoma and squamous cell carcinoma and is a statistically significant hotspot (PMID: 12433709, 10414702, 1423262, 10414702). In vivo studies with yeast expressing TP53 H193Q demonstrated that mutation has partial inactivation of its transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:32:09',NULL,NULL,NULL,NULL,'c4462ae7-d47b-4388-88d6-8a7717109eca',7157,NULL),(28935,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:58:38',NULL,NULL,NULL,NULL,'460db1c2-76e6-428b-bb6c-b0a8f190d7a0',7157,NULL),(28936,NULL,'The TP53 H193D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma, diffuse large B-cell lymphoma, ovarian cancer and breast cancer and is a statistically significant hotspot (PMID: 17187432, 15370252, 9665415, 12032228). In vivo studies with yeast expressing TP53 H193D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H193D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:32:30',NULL,NULL,NULL,NULL,'3b9d6c9b-7a7a-48fc-91e7-ff0723870a95',7157,NULL),(28937,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:59:05',NULL,NULL,NULL,NULL,'ac419870-741c-45ac-a246-b564122fcdc3',7157,NULL),(28938,NULL,'The TP53 H193Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma, cholangiocarcinoma, malignant lymphoma and squamous cell carcinoma and is a statistically significant hotspot (PMID: 11504770, 10564952, 11552717, 12509970). In vivo studies with yeast expressing TP53 H193Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H193Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:33:02',NULL,NULL,NULL,NULL,'23fd0dbc-f22f-463e-b9ef-d8192ed42e58',7157,NULL),(28939,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 14:59:39',NULL,NULL,NULL,NULL,'0eae3dbf-83e1-491e-bd0d-f4f12e30caa3',7157,NULL),(28940,NULL,'The TP53 H193L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in diffuse large B-cell lymphoma, squamous cell carcinoma and breast cancer and is a statistically significant hotspot (PMID: 17278262, 18081876, 16818855). In vivo studies with yeast expressing TP53 H193L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H193L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:50:23',NULL,NULL,NULL,NULL,'ae9731c5-63f5-40ca-97ae-ffa807ee5899',7157,NULL),(28941,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:00:14',NULL,NULL,NULL,NULL,'cb045c02-a29e-4090-aadf-deaabdf9fb40',7157,NULL),(28942,NULL,'The TP53 Y163D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in B-cell chronic lymphocytic leukemia, breast cancer and ovarian cancer and is a statistically significant hotspot (PMID: 11180073, 10706127, 20229506). In vivo studies with yeast expressing TP53 Y163D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y163D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:50:49',NULL,NULL,NULL,NULL,'2b1eec05-a674-477c-a223-cf671c5010ea',7157,NULL),(28943,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:00:59',NULL,NULL,NULL,NULL,'8fba2def-8fa0-4a21-9ec3-0e15c555b150',7157,NULL),(28944,NULL,'The TP53 Y163H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma, hepatocellular carcinoma, colorectal cancer and diffuse large B-cell lymphoma, and is a statistically significant hotspot (PMID: 9683822, 1327523, 9000147, 9162193). In vivo studies with yeast expressing TP53 Y163H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y163H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 18:51:39',NULL,NULL,NULL,NULL,'00643e36-3d7c-4aaf-b19a-e33e2d136ef6',7157,NULL),(28945,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:11:40',NULL,NULL,NULL,NULL,'3b5bf0b9-c452-4e5e-b252-bde150dd476a',7157,NULL),(28946,NULL,'The TP53 V274A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer, hepatocellular carcinoma and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 8102535, 1330291, 8621246). In vivo studies with yeast expressing TP53 V274A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V274A also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-12 15:11:42',NULL,NULL,NULL,NULL,'a16380c0-e780-4b6c-b5f2-e7c44f62110e',7157,NULL),(28947,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:12:10',NULL,NULL,NULL,NULL,'761b53c8-f2dd-4788-ae44-39f995a96251',7157,NULL),(28948,NULL,'The TP53 V274D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer, low-grade mucosa-associated lymphoid tissue lymphomas, papillary carcinoma and acute myeloid leukemia and is a statistically significant hotspot (PMID: 21720365, 8579126, 15538112, 2009369). In vivo studies with yeast expressing TP53 V274D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V274D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:10:50',NULL,NULL,NULL,NULL,'3c57f2a6-81ef-40c9-919b-ff47ff5f8711',7157,NULL),(28949,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:12:32',NULL,NULL,NULL,NULL,'c62d1341-a69e-4eb7-9dec-e561671e6581',7157,NULL),(28950,NULL,'The TP53 C277F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer, splenic B-cell lymphoma and non-small cell lung cancer and is a statistically significant hotspot (PMID: 9504831, 8018922, 8826941). In vivo studies with yeast expressing TP53 C277F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C277F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:11:11',NULL,NULL,NULL,NULL,'6462663b-11d0-4f85-9fb7-82c702fb89e0',7157,NULL),(28951,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:12:58',NULL,NULL,NULL,NULL,'c5dd5259-ccd2-482b-813c-2de0f810d26c',7157,NULL),(28952,NULL,'The TP53 V274G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer, ovarian cancer, breast cancer and hepatocellular carcinoma and is a statistically significant hotspot (PMID: 16721749, 21720365, 12908823, 9463584). In vivo studies with yeast expressing TP53 V274G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V274G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:11:52',NULL,NULL,NULL,NULL,'a9064cc9-2dbb-41b1-a05b-bab623816855',7157,NULL),(28953,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:13:36',NULL,NULL,NULL,NULL,'07374489-060a-4056-afb4-1f17bcd6de7d',7157,NULL),(28954,NULL,'The TP53 V274L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer, glioblastoma, colorectal cancer and acute myeloid leukemia and is a statistically significant hotspot (PMID: 11329143, 15466178, 11044641, 12792784). In vivo studies with yeast expressing TP53 V274L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V274L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:12:20',NULL,NULL,NULL,NULL,'3c6acba0-e48c-4d88-8dfe-6c7ff8d884ff',7157,NULL),(28955,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:16:37',NULL,NULL,NULL,NULL,'4af89e61-602c-4d19-879b-cd96213ba682',7157,NULL),(28956,NULL,'The TP53 G244A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma, B-cell lymphoma and prostate cancer and is a statistically significant hotspot (PMID: 14675778, 9262496, 15541116). In vivo studies with yeast expressing TP53 G244A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G244A also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:12:52',NULL,NULL,NULL,NULL,'82a54398-cc0a-457b-b6c7-bbb391723a2a',7157,NULL),(28957,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:17:01',NULL,NULL,NULL,NULL,'10ce96a1-df6f-415f-ac24-2b7125173c0c',7157,NULL),(28958,NULL,'The TP53 G244C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma, colorectal cancer and bladder cancer and is a statistically significant hotspot (PMID: 15305417, 14534722, 16061860). In vivo studies with yeast expressing TP53 G244C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:13:51',NULL,NULL,NULL,NULL,'421495fa-6682-4408-a40d-b54fb3c422a6',7157,NULL),(28959,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:17:36',NULL,NULL,NULL,NULL,'beb5cb1f-7d8f-4fee-874c-6550fce390d2',7157,NULL),(28960,NULL,'The TP53 G244V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in stomach adenocarcinoma, colorectal cancer and ovarian cancer, and is a statistically significant hotspot (PMID: 12796400, 11406538, 14760384). In vivo studies with yeast expressing TP53 G244V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G244V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-12 15:17:37',NULL,NULL,NULL,NULL,'4c3b2db6-4546-4541-bc05-29c1cd6b30ef',7157,NULL),(28961,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-12 15:18:13',NULL,NULL,NULL,NULL,'326f11db-9309-496d-a034-d73bea629d8d',7157,NULL),(28962,NULL,'The TP53 G244F mutation is located in the DNA-binding domain of the protein and is a statistically significant hotspot. In vitro studies with various human cancer cell lines expressing TP53 G244F demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2022-12-13 19:14:25',NULL,NULL,NULL,NULL,'013358ff-b85e-4cdd-9738-064d31f647d1',7157,NULL),(28963,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:04:10',NULL,NULL,NULL,NULL,'6c4b9710-4567-449e-94ea-7e65dc9c4c64',7157,NULL),(28964,NULL,'The TP53 V173A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in stomach adenocarcinoma, breast cancer and ovarian cancer, and is a statistically significant hotspot (PMID: 1370612, 9569050, 9569050). In vivo studies with yeast expressing TP53 V173A demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V173A also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:04:45',NULL,NULL,NULL,NULL,'27e80858-2e5c-4992-a40f-241c60acdd01',7157,NULL),(28965,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:04:38',NULL,NULL,NULL,NULL,'00ead91e-68e3-4975-85cd-5d3a664f5623',7157,NULL),(28966,NULL,'The TP53 V173G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in stomach adenocarcinoma, breast cancer and ovarian cancer, and is a statistically significant hotspot (PMID: 1370612, 9569050, 9569050). In vivo studies with yeast expressing TP53 V173G demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V173G also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-03-23 15:35:21',NULL,NULL,NULL,NULL,'932dbcae-ac69-4e2e-9cbc-c583d35e7d4c',7157,NULL),(28967,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:04:59',NULL,NULL,NULL,NULL,'8cd0c2f9-bd62-4d62-b7a0-752c37c0cad2',7157,NULL),(28968,NULL,'The TP53 C277G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer, gastric cancer and malignant peripheral nerve sheath tumor, and is a statistically significant hotspot (PMID: 9516972, 8620428, 11406645). In vivo studies with yeast expressing TP53 C277G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C277G also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:05:35',NULL,NULL,NULL,NULL,'82e9c452-5ef5-4b69-89a3-2e663564a0a9',7157,NULL),(28969,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:07:17',NULL,NULL,NULL,NULL,'9dfd804b-abac-4319-99cf-751331ee4636',7157,NULL),(28970,NULL,'The TP53 Y205S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer, squamous cell carcinoma and diffuse large B-cell lymphoma, and is a statistically significant hotspot (PMID: 16489069, 1324797, 15533223). In vivo studies with yeast expressing TP53 Y205S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y205S also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:06:06',NULL,NULL,NULL,NULL,'4e476ffa-0a02-4770-84e0-edaa870133e8',7157,NULL),(28971,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:07:46',NULL,NULL,NULL,NULL,'83ccc781-ea2b-4d81-8b7c-12f875d9c7c0',7157,NULL),(28972,NULL,'The TP53 Y205D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer, colorectal cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 18725978, 15778432, 15958551). In vivo studies with yeast expressing TP53 Y205D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y205D also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:06:29',NULL,NULL,NULL,NULL,'deefda37-d4bd-418f-8ddc-0b0c61272eb6',7157,NULL),(28973,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:08:14',NULL,NULL,NULL,NULL,'a346638f-1f40-48ba-a1cb-0c99db56b811',7157,NULL),(28974,NULL,'The TP53 Y205F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in urothelial cancer, glioblastoma, colorectal cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 21413016, 15466178, 17410283, 15957194). In vivo studies with yeast expressing TP53 Y205F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y205F also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:12:01',NULL,NULL,NULL,NULL,'6508c60a-5f02-4287-b49d-7e3ed121bb84',7157,NULL),(28975,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:08:49',NULL,NULL,NULL,NULL,'c12195bb-1d27-4d71-88fe-b11e8dfca974',7157,NULL),(28976,NULL,'The TP53 Y205H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in urothelial cancer, glioblastoma, colorectal cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 21413016, 15466178, 17410283, 15957194). In vivo studies with yeast expressing TP53 Y205H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y205H also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:12:34',NULL,NULL,NULL,NULL,'2a793159-c60d-4bd2-9876-86fd0b548583',7157,NULL),(28977,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:09:13',NULL,NULL,NULL,NULL,'68686269-030c-4951-9c98-574bc58e04f7',7157,NULL),(28978,NULL,'The TP53 Y205N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 7852189, 15958551). In vivo studies with yeast expressing TP53 Y205N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y205N also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:12:57',NULL,NULL,NULL,NULL,'93b855a6-9c16-4327-9da5-716c71cd920b',7157,NULL),(28979,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:17:14',NULL,NULL,NULL,NULL,'650cb919-803d-4b78-a27a-0c279d483ec3',7157,NULL),(28980,NULL,'The TP53 V143E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer, colorectal cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 17683074, 9052405, 7767998). In vivo studies with yeast expressing TP53 V143E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V143E also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:13:21',NULL,NULL,NULL,NULL,'56d25060-f66b-48ae-9eee-d9f34b3f9d5a',7157,NULL),(28981,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:18:00',NULL,NULL,NULL,NULL,'7e07bb48-f05a-46e9-bb40-f49dbd41362c',7157,NULL),(28982,NULL,'The TP53 C242R mutation is located in exon 6 of the protein. This mutation has been identified in pancreatic cancer, prostate cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 15541116, 15940643, 17456604). In vivo studies with yeast expressing TP53 C242R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C242R also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:13:47',NULL,NULL,NULL,NULL,'9bd2e135-c776-4144-96a0-9e31da698e50',7157,NULL),(28983,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:18:11',NULL,NULL,NULL,NULL,'6aaaacbf-8fa9-4733-b4e0-85ddbe148aeb',7157,NULL),(28984,NULL,'The TP53 C242W mutation is located in exon 6 of the protein. This mutation has been identified in breast cancer, colorectal cancer and ovarian cancer, and is a statistically significant hotspot (PMID: 9815649, 8682586, 15221786). In vivo studies with yeast expressing TP53 C242W demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C242W also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:15:07',NULL,NULL,NULL,NULL,'8e37c5e7-5e92-4997-bfff-e5c9545d7b2a',7157,NULL),(28985,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2022-12-22 17:18:46',NULL,NULL,NULL,NULL,'aeba8487-c6f2-4fde-81e6-c7d83b0f6287',7157,NULL),(28986,NULL,'The TP53 C242G mutation is located in exon 6 of the protein. This mutation has been identified in glioblastoma, malignant melanoma and breast cancer, and is a statistically significant hotspot (PMID: 11453385, 17260012, 7928628). In vivo studies with yeast expressing TP53 C242G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C242G also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:15:27',NULL,NULL,NULL,NULL,'de5d81eb-6375-4753-a458-2bc8adf49c3f',7157,NULL),(28987,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-13 16:40:07',NULL,NULL,NULL,NULL,'b82252b2-0d6d-402c-ace6-5cbc2620a368',7157,NULL),(28988,NULL,'The TP53 V143G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in adenocarcinoma and is a statistically significant hotspot (PMID: 22551440). In vivo studies with yeast expressing TP53 V143G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V143G also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:16:01',NULL,NULL,NULL,NULL,'be7084f5-f6a3-4318-a484-9635db5720bf',7157,NULL),(28989,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-13 16:41:06',NULL,NULL,NULL,NULL,'f3c22abd-5c1b-4b87-a9e4-b3ef361e48cc',7157,NULL),(28990,NULL,'The TP53 G266V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in adenocarcinoma and is a statistically significant hotspot (PMID: 16143127). In vivo studies with yeast expressing TP53 G266V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G266V also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:16:20',NULL,NULL,NULL,NULL,'888c47c6-58c7-4e04-81e6-1536c2bd51f4',7157,NULL),(28991,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-13 16:41:26',NULL,NULL,NULL,NULL,'d65ce7c9-dc14-4490-8f29-94c8b06eb46a',7157,NULL),(28992,NULL,'The TP53 V143M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 8393371). In vivo studies with yeast expressing TP53 V143M demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V143M also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:16:40',NULL,NULL,NULL,NULL,'8b31e7bd-936b-4c15-9afb-c14006669945',7157,NULL),(28993,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-13 16:41:50',NULL,NULL,NULL,NULL,'fc4dfacd-c797-455d-861b-f231b35dff67',7157,NULL),(28994,NULL,'The TP53 G266K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Merkel cell carcinoma and is a statistically significant hotspot (PMID: 26238782). In vitro studies with various human cancer cell lines expressing TP53 G266K demonstrate the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:17:03',NULL,NULL,NULL,NULL,'cf20a5e1-6648-4545-b838-3371112fad49',7157,NULL),(28995,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:43:31',NULL,NULL,NULL,NULL,'ab56a570-4152-4e6c-9403-909fc59e24ad',7157,NULL),(28996,NULL,'The TP53 R158P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 15564288). In vivo studies with yeast expressing TP53 R158P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R158P also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:17:20',NULL,NULL,NULL,NULL,'8094f20d-a8f4-4245-ba06-676a2f91222f',7157,NULL),(28997,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:44:09',NULL,NULL,NULL,NULL,'446ca72f-2d19-452a-a08b-1262350e3d10',7157,NULL),(28998,NULL,'The TP53 R158S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in clear cell carcinoma and is a statistically significant hotspot (PMID: 17581423). In vivo studies with yeast expressing TP53 R158S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:44:14',NULL,NULL,NULL,NULL,'fe773070-a321-4d9a-b211-4ebef84ac21b',7157,NULL),(28999,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:44:35',NULL,NULL,NULL,NULL,'b64369a3-594d-4991-bdb9-cd26081048b6',7157,NULL),(29000,NULL,'The TP53 R158F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 11532872). In vitro studies with various human cancer cell lines expressing TP53 R158F demonstrate the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:17:50',NULL,NULL,NULL,NULL,'a8c66e8b-4784-41bf-bb09-4d5e3ba46b05',7157,NULL),(29001,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:44:59',NULL,NULL,NULL,NULL,'c5779258-d883-41b7-88b3-e0e4107f0176',7157,NULL),(29002,NULL,'The TP53 R158G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and B-cell chronic lymphocytic leukemia, and is a statistically significant hotspot (PMID: 10454262, 12149195). In vivo studies with yeast expressing TP53 R158G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R158G also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:18:13',NULL,NULL,NULL,NULL,'b3447c25-a8fd-4ad1-bcb6-7a937e79e3df',7157,NULL),(29003,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:48:42',NULL,NULL,NULL,NULL,'0ab67089-360f-4a29-8809-a393d32ba72d',7157,NULL),(29004,NULL,'The TP53 E286Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and adenocarcinoma and is a statistically significant hotspot (PMID: 9133458, 12509279). In vivo studies with yeast expressing TP53 E286Q demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E286Q also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:18:43',NULL,NULL,NULL,NULL,'f163803d-8d94-4124-b0e2-9a0a13ff82b6',7157,NULL),(29005,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:49:01',NULL,NULL,NULL,NULL,'3d0afeb1-828a-41ce-b331-2c72b2ff1718',7157,NULL),(29006,NULL,'The TP53 E286A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 9421363). In vivo studies with yeast expressing TP53 E286A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E286A also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:29:15',NULL,NULL,NULL,NULL,'ebc95219-3398-4acb-81f0-c7180bd45e6e',7157,NULL),(29007,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:50:33',NULL,NULL,NULL,NULL,'4f46c883-06e6-43ce-beb5-e4a357ce6a89',7157,NULL),(29008,NULL,'The TP53 H179P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 14580680). In vivo studies with yeast expressing TP53 H179P demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H179P also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:36:17',NULL,NULL,NULL,NULL,'050d257d-1769-4dca-b2f7-ccf49c8fb1a1',7157,NULL),(29009,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:50:55',NULL,NULL,NULL,NULL,'f69b3715-81a9-48c5-b6f8-36adf31a0af6',7157,NULL),(29010,NULL,'The TP53 H179D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 7585600). In vivo studies with yeast expressing TP53 H179D demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H179D also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:36:40',NULL,NULL,NULL,NULL,'02cf3ff0-4f12-46e4-98f4-4739f5b930be',7157,NULL),(29011,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:51:22',NULL,NULL,NULL,NULL,'52694828-715e-446e-936e-ad16a380dc54',7157,NULL),(29012,NULL,'The TP53 H168P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 12649174). In vivo studies with yeast expressing TP53 H168P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H168P also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:36:55',NULL,NULL,NULL,NULL,'403d7ec6-9a06-42af-ac22-2df71fae1a78',7157,NULL),(29013,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:51:59',NULL,NULL,NULL,NULL,'7e31d293-cf1c-4273-a570-2599a8f6722c',7157,NULL),(29014,NULL,'The TP53 H168L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and breast cancer and is a statistically significant hotspot (PMID: 1408139, 18348141). In vivo studies with yeast expressing TP53 H168L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H168L also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:41:42',NULL,NULL,NULL,NULL,'ed098ad7-3177-4344-a459-47b2330de882',7157,NULL),(29015,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:53:57',NULL,NULL,NULL,NULL,'7cc87df2-9e76-4815-b34e-307024318389',7157,NULL),(29016,NULL,'The TP53 C176S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in papillary serous cystadenocarcinoma and hepatocellular carcinoma and is a statistically significant hotspot (PMID: 9635683, 8407553). In vivo studies with yeast expressing TP53 C176S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C176S also demonstrated the mutation is inactivating as measured by reduced growth suppressive activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 19:42:20',NULL,NULL,NULL,NULL,'45ff9bcb-ef3c-4f8d-ab87-d515c640b219',7157,NULL),(29017,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:54:17',NULL,NULL,NULL,NULL,'ba3d8ace-aa2f-47a8-9089-d93d5d000d5e',7157,NULL),(29018,NULL,'The TP53 C176G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in diffuse large B-cell lymphoma and hepatocellular carcinoma, and is a statistically significant hotspot (PMID: 11113864, 8672994). In vivo studies with yeast expressing TP53 C176G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C176G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:54:20',NULL,NULL,NULL,NULL,'78e2f7ac-6c2e-46eb-8e82-f2dbfa111e29',7157,NULL),(29019,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:54:37',NULL,NULL,NULL,NULL,'cfd13aa8-4d85-476f-91c0-1234d553cde8',7157,NULL),(29020,NULL,'The TP53 C176Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and colorectal cancer, and is a statistically significant hotspot (PMID: 21798897, 9431782). In vivo studies with yeast expressing TP53 C176Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C176Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:54:41',NULL,NULL,NULL,NULL,'2652ccc2-d10f-4f63-b245-420e6d35a11c',7157,NULL),(29021,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:54:59',NULL,NULL,NULL,NULL,'42850a80-4674-4486-ae60-77c5f97c281e',7157,NULL),(29022,NULL,'The TP53 C238R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 10225439). In vivo studies with yeast expressing TP53 C238R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C238R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:55:03',NULL,NULL,NULL,NULL,'16fa7f0a-aee7-4c27-8d100-8075b1636587',7157,NULL),(29023,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:55:22',NULL,NULL,NULL,NULL,'5cce95dd-793a-4fea-af99-8f3410116ab1',7157,NULL),(29024,NULL,'The TP53 C238G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in adenocarcinoma and is a statistically significant hotspot (PMID: 16467080). In vivo studies with yeast expressing TP53 C238G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C238G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:55:30',NULL,NULL,NULL,NULL,'7627e773-9892-4d65-8c18-8de8cf9ad50c',7157,NULL),(29025,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:55:52',NULL,NULL,NULL,NULL,'d7cd8775-231e-4dca-86fb-6cc8ec905ea8',7157,NULL),(29026,NULL,'The TP53 C238W mutation is located in the DNA-binding domain of the protein. This mutation has been identified in adenocarcinoma and is a statistically significant hotspot (PMID: 11710597). In vivo studies with yeast expressing TP53 C238W demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C238W also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:55:57',NULL,NULL,NULL,NULL,'10dd559d-f9a4-4ee0-85a5-f91b47b13fd4',7157,NULL),(29027,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:56:15',NULL,NULL,NULL,NULL,'8f0e0342-3bd5-474d-969c-253737823789',7157,NULL),(29028,NULL,'The TP53 C238Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in osteosarcoma and is a statistically significant hotspot (PMID: 11596036). In vivo studies with yeast expressing TP53 C238Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C238Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:56:23',NULL,NULL,NULL,NULL,'e18a7835-42c3-4524-820e-0c729ca88cba',7157,NULL),(29029,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-01-17 15:56:42',NULL,NULL,NULL,NULL,'0a2be534-1140-42e0-9ad1-e81dba8f0ea9',7157,NULL),(29030,NULL,'The TP53 E285Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 11596036). In vivo studies with yeast expressing TP53 E285Q demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E285Q also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-01-26 15:05:54',NULL,NULL,NULL,NULL,'74d12aee-bed8-4d1b-9e7d-93d00d6aa01f',7157,NULL),(29031,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:57:11',NULL,NULL,NULL,NULL,'ba76abd6-2653-4f1e-8f18-2acbc431bf9a',7157,NULL),(29032,NULL,'The TP53 M237V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 15138567). In vivo studies with yeast expressing TP53 M237V demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M237V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:57:16',NULL,NULL,NULL,NULL,'c0d525f6-b2d7-4b3a-9710-743575f7adce',7157,NULL),(29033,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:57:33',NULL,NULL,NULL,NULL,'50baf7c6-ba89-4f15-a5d9-bfe3cc37b73e',7157,NULL),(29034,NULL,'The TP53 M237I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and colorectal cancer, and is a statistically significant hotspot (PMID: 8816892, 16696056). In vivo studies with yeast expressing TP53 M237I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M237I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:57:38',NULL,NULL,NULL,NULL,'8615bc12-457f-4116-9707-9e3cd67d5781',7157,NULL),(29035,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:57:54',NULL,NULL,NULL,NULL,'c42b3ec7-fe7d-406c-8682-1f677d5398e9',7157,NULL),(29036,NULL,'The TP53 G154C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 10709097). In vivo studies with yeast expressing TP53 G154C demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:57:59',NULL,NULL,NULL,NULL,'5ad6c562-bd20-45df-b62d-ecc55aab2e48',7157,NULL),(29037,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:58:15',NULL,NULL,NULL,NULL,'dbf06d8f-7f58-46b8-b137-cb602ed15124',7157,NULL),(29038,NULL,'The TP53 G154D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and adenocarcinoma, and is a statistically significant hotspot (PMID: 15138567, 11011123). In vivo studies with yeast expressing TP53 G154D demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:58:19',NULL,NULL,NULL,NULL,'06386cec-60b5-4b2d-a61a-d65265bb498b',7157,NULL),(29039,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:58:37',NULL,NULL,NULL,NULL,'28c9bc12-037a-4bcb-84fa-868ab04e9315',7157,NULL),(29040,NULL,'The TP53 S241P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in diffuse large B-cell lymphoma and malignant melanoma, and is a statistically significant hotspot (PMID: 9002964, 11216673). In vivo studies with yeast expressing TP53 S241P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S241P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:58:43',NULL,NULL,NULL,NULL,'1a2142bb-3cea-4bef-8c0a-824413de5f35',7157,NULL),(29041,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:59:01',NULL,NULL,NULL,NULL,'7a4d7dfa-8cb0-486a-9d04-b9cee79a1377',7157,NULL),(29042,NULL,'The TP53 S241A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 16489069). In vivo studies with yeast expressing TP53 S241A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S241A also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:59:06',NULL,NULL,NULL,NULL,'80607c3e-ed1d-4eee-9513-be9353f24587',7157,NULL),(29043,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 15:59:27',NULL,NULL,NULL,NULL,'ebfbd51a-6182-4ccb-ae89-095e4550c015',7157,NULL),(29044,NULL,'The TP53 S241C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in acute myeloid leukemia and is a statistically significant hotspot (PMID: 9854490). In vivo studies with yeast expressing TP53 S241C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S241C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 15:59:31',NULL,NULL,NULL,NULL,'70550ad9-88eb-47f4-b6c8-2652f6988fe2',7157,NULL),(29045,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:06:40',NULL,NULL,NULL,NULL,'6ca4bb32-5293-4c80-96d5-5d2d851c0d4b',7157,NULL),(29046,NULL,'The TP53 R282P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 7952630, 26619011). In vivo studies with yeast expressing TP53 R282P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R282P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-07-23 11:36:26',NULL,NULL,NULL,NULL,'8de3fd89-4727-4b43-acba-452cfbb9bdfd',7157,NULL),(29047,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:07:01',NULL,NULL,NULL,NULL,'9c593844-154a-4bb1-9cc2-eb3798b03c68',7157,NULL),(29048,NULL,'The TP53 R282G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and bladder cancer, and is a statistically significant hotspot (PMID: 17999388, 16061860, 26619011). In vivo studies with yeast expressing TP53 R282G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R282G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-07-23 11:35:29',NULL,NULL,NULL,NULL,'6d0bc3ca-4d3b-4b71-8b37-022f2a14adda',7157,NULL),(29049,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-01-17 16:07:19',NULL,NULL,NULL,NULL,'91a35633-1fcb-422c-c09c-07b1c59500b9',7157,NULL),(29050,NULL,'The TP53 R282L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 12365217, 26619011). In vivo studies with yeast expressing TP53 R282L demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R282L also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2024-07-23 11:36:52',NULL,NULL,NULL,NULL,'a3196366-dc81-4403-812a-5aa0581ce327',7157,NULL),(29051,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:07:47',NULL,NULL,NULL,NULL,'9a5808a1-55a5-4519-875f-469f3388fa45',7157,NULL),(29052,NULL,'The TP53 P250R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 31395065). In vivo studies with yeast expressing TP53 P250R demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:07:52',NULL,NULL,NULL,NULL,'81981412-3c58-4fb6-9017-558c1aa8f49e',7157,NULL),(29053,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:08:03',NULL,NULL,NULL,NULL,'ba4961b7-1fc2-4ca4-89bc-6b7e1822d2fc',7157,NULL),(29054,NULL,'The TP53 E286V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 11375957). In vivo studies with yeast expressing TP53 E286V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E286V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:08:09',NULL,NULL,NULL,NULL,'8bae3101-5a86-4238-9e9c-e83f2900f727',7157,NULL),(29055,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:14:02',NULL,NULL,NULL,NULL,'6d591281-a5f0-4975-adeb-466d5d9b6117',7157,NULL),(29056,NULL,'The TP53 N239D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 16271069). In vivo studies with yeast expressing TP53 N239D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 N239D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:14:06',NULL,NULL,NULL,NULL,'a304b726-13be-4bfe-9bf3-1cb645e8b744',7157,NULL),(29057,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:14:22',NULL,NULL,NULL,NULL,'d277ee96-c1cd-487f-8623-204858ba652b',7157,NULL),(29058,NULL,'The TP53 N239T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 8611423, 10225439). In vivo studies with yeast expressing TP53 N239T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 N239T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:14:28',NULL,NULL,NULL,NULL,'3ae412f1-f80b-4baf-8982-975298ca7a61',7157,NULL),(29059,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-01-17 16:14:44',NULL,NULL,NULL,NULL,'70e8d6d6-145f-4a89-b57f-32ec035b9c50',7157,NULL),(29060,NULL,'The TP53 N239H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in prostate cancer and is a statistically significant hotspot (PMID: 8978408). In vivo studies with yeast expressing TP53 N239H demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 N239H also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-01-26 15:23:10',NULL,NULL,NULL,NULL,'81ddeb7d-43c6-4842-9efa-23940a64ef54',7157,NULL),(29061,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:15:06',NULL,NULL,NULL,NULL,'b6a2cb2b-f4a6-4c46-9221-dc653e222bc8',7157,NULL),(29062,NULL,'The TP53 N239K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 9816044, 8407553). In vivo studies with yeast expressing TP53 N239K demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:15:09',NULL,NULL,NULL,NULL,'97436b67-9fcc-45aa-986e-91853a8b2972',7157,NULL),(29063,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:15:20',NULL,NULL,NULL,NULL,'22fd1422-1be9-4b96-9d6f-7cc20184800e',7157,NULL),(29064,NULL,'The TP53 E286G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and squamous cell carcinoma, and is a statistically significant hotspot (PMID: 1945416, 8453641). In vivo studies with yeast expressing TP53 E286G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E286G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:15:24',NULL,NULL,NULL,NULL,'374c669b-c134-4aea-8613-ec494eff6d98',7157,NULL),(29065,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:15:47',NULL,NULL,NULL,NULL,'947a5872-dba5-4662-b29c-6a56e258a5ea',7157,NULL),(29066,NULL,'The TP53 R280S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 10761706). In vivo studies with yeast expressing TP53 R280S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R280S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:15:51',NULL,NULL,NULL,NULL,'bf1d10b8-d608-4207-b7a8-1f60cf29ed23',7157,NULL),(29067,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:16:06',NULL,NULL,NULL,NULL,'42c02efc-a9c0-4eda-aa86-b02e969718ba',7157,NULL),(29068,NULL,'The TP53 R280G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 19909015). In vivo studies with yeast expressing TP53 R280G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R280G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:16:12',NULL,NULL,NULL,NULL,'7a821b22-dfa0-4037-8e80-9717c714e052',7157,NULL),(29069,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-17 16:16:28',NULL,NULL,NULL,NULL,'2d11d8aa-612f-447e-a4b3-7a049f821c21',7157,NULL),(29070,NULL,'The TP53 R280I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 1324794). In vivo studies with yeast expressing TP53 R280I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R280I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-17 16:16:33',NULL,NULL,NULL,NULL,'91b5592b-31c9-4a31-b5b4-9b31489f7b22',7157,NULL),(29071,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-18 13:50:28',NULL,NULL,NULL,NULL,'d46bcd9c-c3ce-445f-acce-408297679691',7157,NULL),(29072,NULL,'The TP53 P278T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in renal cell carcinoma and is a statistically significant hotspot (PMID: 17133269). In vivo studies with yeast expressing TP53 P278T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P278T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 13:50:34',NULL,NULL,NULL,NULL,'01075238-df33-4c1e-adde-01b0f1e2d6d4',7157,NULL),(29073,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-18 13:50:52',NULL,NULL,NULL,NULL,'c4157341-9d1d-4324-b160-06906cbb2c0b',7157,NULL),(29074,NULL,'The TP53 P278F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 20025891). In vitro studies with various human cancer cell lines expressing TP53 P278F demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 13:50:57',NULL,NULL,NULL,NULL,'54974bc2-65f2-487e-9630-f71138fd2b45',7157,NULL),(29075,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-18 13:51:18',NULL,NULL,NULL,NULL,'39d0d7b2-16a9-461a-a723-cd84b8175eac',7157,NULL),(29076,NULL,'The TP53 P278H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 9097977). In vivo studies with yeast expressing TP53 P278H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P278H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 13:51:22',NULL,NULL,NULL,NULL,'89625149-b587-401c-a80e-67e14db09453',7157,NULL),(29077,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-18 13:58:54',NULL,NULL,NULL,NULL,'02a31bbb-d4d9-406b-b5ea-da9abb61a369',7157,NULL),(29078,NULL,'The TP53 H178P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in serous cystadenocarcinoma and is a statistically significant hotspot (PMID: 20229506). In vivo studies with yeast expressing TP53 H178P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H178P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 13:58:58',NULL,NULL,NULL,NULL,'55b2eac5-e699-4b17-b760-11bfb897c72c',7157,NULL),(29079,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-18 13:59:09',NULL,NULL,NULL,NULL,'6046d1e2-de17-4f05-a5fc-39f82511577e',7157,NULL),(29080,NULL,'The TP53 H178Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in B-cell chronic lymphocytic leukemia and is a statistically significant hotspot (PMID: 16572201). In vivo studies with yeast expressing TP53 H178Q demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 13:59:14',NULL,NULL,NULL,NULL,'10f4417b-a677-4686-8c24-c69e2500454f',7157,NULL),(29081,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-18 13:59:26',NULL,NULL,NULL,NULL,'a6744c8b-26f5-4a24-b44b-164749e13c51',7157,NULL),(29082,NULL,'The TP53 H178N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in basal cell carcinoma and is a statistically significant hotspot (PMID: 10568172). In vivo studies with yeast expressing TP53 H178P demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-18 13:59:31',NULL,NULL,NULL,NULL,'63b5a81f-016e-47a4-89d3-9b736c65a320',7157,NULL),(29083,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:31:07',NULL,NULL,NULL,NULL,'def3f178-df33-4a6b-b588-6c60c6a237a3',7157,NULL),(29084,NULL,'The TP53 V197G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 12713560). In vivo studies with yeast expressing TP53 V197G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V197G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:31:13',NULL,NULL,NULL,NULL,'7f9a7013-42ad-4331-b465-67f02359809c',7157,NULL),(29085,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:31:31',NULL,NULL,NULL,NULL,'7167f4bb-5d19-4f2f-a345-b912fd3f521e',7157,NULL),(29086,NULL,'The TP53 V197M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11494027). In vivo studies with yeast expressing TP53 V197M demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V197M also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:31:37',NULL,NULL,NULL,NULL,'c2e5cb1e-8ff1-4521-b384-0468f08b5a46',7157,NULL),(29087,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:32:41',NULL,NULL,NULL,NULL,'143ee9ac-5399-4bee-b152-c4ed6e3e9717',7157,NULL),(29088,NULL,'The TP53 M133R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in adenocarcinoma and is a statistically significant hotspot (PMID: 12637159). In vivo studies with yeast expressing TP53 M133R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M133R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:32:46',NULL,NULL,NULL,NULL,'4cd1b262-00fb-4833-b047-55cb9a712e77',7157,NULL),(29089,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-01-20 15:33:19',NULL,NULL,NULL,NULL,'f70e61ea-8810-4de9-b2a1-06bc2501e9fe',7157,NULL),(29090,NULL,'The TP53 M133I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 9627113). In vivo studies with yeast expressing TP53 M133I demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M133I also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-01-26 15:27:21',NULL,NULL,NULL,NULL,'18096315-2f08-4c8b-833e-9f8c72ab487f',7157,NULL),(29091,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:34:07',NULL,NULL,NULL,NULL,'1cb002be-7b60-4603-8a2e-0d54e4e3324d',7157,NULL),(29092,NULL,'The TP53 P152Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 21567059). In vivo studies with yeast expressing TP53 P152Q demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:34:12',NULL,NULL,NULL,NULL,'d6a45691-d84e-4ffa-825f-eee7cedd1736',7157,NULL),(29093,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:34:36',NULL,NULL,NULL,NULL,'cba6c559-1d91-42e4-a0b4-13897b68c6ef',7157,NULL),(29094,NULL,'The TP53 P152R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11044641). In vivo studies with yeast expressing TP53 P152R demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:34:41',NULL,NULL,NULL,NULL,'2c1945a4-4fdd-4cc6-8a60-d5ab1c4c921c',7157,NULL),(29095,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:34:57',NULL,NULL,NULL,NULL,'703d1f9b-992c-4cfd-bb89-051f68a30460',7157,NULL),(29096,NULL,'The TP53 P152S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in basal cell carcinoma and is a statistically significant hotspot (PMID: 15656799). In vivo studies with yeast expressing TP53 P152S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P152S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:35:01',NULL,NULL,NULL,NULL,'e1f8468d-7d2b-4dc2-8828-80151c435c3b',7157,NULL),(29097,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:35:32',NULL,NULL,NULL,NULL,'cc308d1e-e5d4-4916-9a29-f235dd0e501d',7157,NULL),(29098,NULL,'The TP53 P152T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 11939587). In vivo studies with yeast expressing TP53 P152T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:35:38',NULL,NULL,NULL,NULL,'52370fa8-899b-49af-93e1-518ed06250f5',7157,NULL),(29099,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:35:55',NULL,NULL,NULL,NULL,'aca2f602-237e-4c78-ab9c-d8cdd7a3aeb7',7157,NULL),(29100,NULL,'The TP53 P152K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant melanoma and is a statistically significant hotspot (PMID: 28481359). In vitro studies with various human cancer cell lines expressing TP53 P152K demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:36:00',NULL,NULL,NULL,NULL,'741cb3e5-d294-47d1-a15c-74c464dfe9a5',7157,NULL),(29101,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:36:18',NULL,NULL,NULL,NULL,'64fa6d5d-76c3-4647-aaca-da873bc204c6',7157,NULL),(29102,NULL,'The TP53 C141R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in B-cell chronic lymphocytic leukemia and is a statistically significant hotspot (PMID: 12149195). In vivo studies with yeast expressing TP53 C141R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C141R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:36:23',NULL,NULL,NULL,NULL,'e53397a1-c57c-4ce9-b54c-ae93d7fc9260',7157,NULL),(29103,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:36:43',NULL,NULL,NULL,NULL,'494a2748-93ea-49ea-80b1-4fba8748f256',7157,NULL),(29104,NULL,'The TP53 C141S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 15316933). In vivo studies with yeast expressing TP53 C141S demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:36:47',NULL,NULL,NULL,NULL,'1bd5d093-b333-4443-a5e4-2f5aba67434c',7157,NULL),(29105,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:36:59',NULL,NULL,NULL,NULL,'7e6ff467-0a1a-4519-9c60-b77c6033ffd7',7157,NULL),(29106,NULL,'The TP53 C141F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 11801555). In vivo studies with yeast expressing TP53 C141F demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:37:02',NULL,NULL,NULL,NULL,'b278bb0c-6f0c-4bd3-ba0a-ed38ea6ec4a4',7157,NULL),(29107,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:37:16',NULL,NULL,NULL,NULL,'146aa4dd-281b-4624-aee9-6d129ce53216',7157,NULL),(29108,NULL,'The TP53 C141G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 15957194). In vivo studies with yeast expressing TP53 C141G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C141G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:37:21',NULL,NULL,NULL,NULL,'c31909bb-405b-4337-b0ce-327676a0496a',7157,NULL),(29109,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:38:43',NULL,NULL,NULL,NULL,'a6d42bc7-543b-4234-b9ae-9769b0fe9e46',7157,NULL),(29110,NULL,'The TP53 C275F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hairy cell leukemia and is a statistically significant hotspot (PMID: 10764158). In vivo studies with yeast expressing TP53 C275F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C275F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:38:54',NULL,NULL,NULL,NULL,'caa0493c-828d-445a-9631-522d0c15b92c',7157,NULL),(29111,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:39:01',NULL,NULL,NULL,NULL,'0d95dc0c-f514-41c4-a39e-273455796221',7157,NULL),(29112,NULL,'The TP53 C275R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 10679659). In vivo studies with yeast expressing TP53 C275R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C275R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:39:09',NULL,NULL,NULL,NULL,'1429cece-4ec2-4d73-8e1e-fcd4e3ba3dd3',7157,NULL),(29113,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:39:23',NULL,NULL,NULL,NULL,'9a74f303-d112-4551-a854-1e51445ea4ed',7157,NULL),(29114,NULL,'The TP53 C275W mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 12921629). In vivo studies with yeast expressing TP53 C275W demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C275W also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:39:33',NULL,NULL,NULL,NULL,'020e361a-5c50-4f24-af99-c12689f55001',7157,NULL),(29115,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:39:48',NULL,NULL,NULL,NULL,'512c2091-5262-4f70-be4b-97b2b9cb842d',7157,NULL),(29116,NULL,'The TP53 C275G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in serous cystadenocarcinoma and is a statistically significant hotspot (PMID: 9891239). In vivo studies with yeast expressing TP53 C275G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:39:52',NULL,NULL,NULL,NULL,'2a5b28b6-82be-4b02-b220-460d72a3cb0e',7157,NULL),(29117,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-20 15:40:05',NULL,NULL,NULL,NULL,'e84a610f-87d6-4ab9-866d-0d8e7f17e443',7157,NULL),(29118,NULL,'The TP53 C275Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 8761140). In vivo studies with yeast expressing TP53 C275Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C275Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-20 15:40:09',NULL,NULL,NULL,NULL,'186215d3-1dde-4066-a52d-0776558f0ff7',7157,NULL),(29119,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-23 16:28:15',NULL,NULL,NULL,NULL,'e1f3bbbd-8b1f-4516-8df9-1ab57311b303',7157,NULL),(29120,NULL,'The TP53 I251F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in transitional cell carcinoma and is a statistically significant hotspot (PMID: 7777479). In vivo studies with yeast expressing TP53 I251F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I251F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-23 16:28:21',NULL,NULL,NULL,NULL,'f1cb9c4e-f809-487c-9978-ec1eed0fe453',7157,NULL),(29121,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-23 16:28:37',NULL,NULL,NULL,NULL,'b5d3a5f4-383c-4d67-8d19-a427c2a20990',7157,NULL),(29122,NULL,'The TP53 I251M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in leiomyosarcoma and is a statistically significant hotspot (PMID: 28481359). In vivo studies with yeast expressing TP53 I251M demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-23 16:28:43',NULL,NULL,NULL,NULL,'48cd2bcd-e92a-4e15-8417-6d2af3d0c164',7157,NULL),(29123,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-23 16:29:12',NULL,NULL,NULL,NULL,'2f81974c-a1b6-4f33-96be-535cc280c2f7',7157,NULL),(29124,NULL,'The TP53 I251N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 12711820). In vivo studies with yeast expressing TP53 I251N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I251N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-23 16:29:17',NULL,NULL,NULL,NULL,'1a20e487-e83e-4916-a572-a0e7a2da1a37',7157,NULL),(29125,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:36:43',NULL,NULL,NULL,NULL,'20e50e28-b8bd-48b0-946d-cae76f115540',7157,NULL),(29126,NULL,'The TP53 E258A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 16818615). In vivo studies with yeast expressing TP53 E258A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E258A also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:36:52',NULL,NULL,NULL,NULL,'42d17869-d6f0-486d-a62c-f448e4e59313',7157,NULL),(29127,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:37:02',NULL,NULL,NULL,NULL,'3e56e426-5c35-4a60-9134-cbcdd72e5b16',7157,NULL),(29128,NULL,'The TP53 E258Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 8339293). In vivo studies with yeast expressing TP53 E258Q demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E258Q also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:37:10',NULL,NULL,NULL,NULL,'7359c7c8-75c1-4d1a-a8d7-9ce4b55b18ce',7157,NULL),(29129,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:37:26',NULL,NULL,NULL,NULL,'e33546e9-be7f-4ad1-a37a-38de7b6ef70c',7157,NULL),(29130,NULL,'The TP53 E258D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 12093899). In vivo studies with yeast expressing TP53 E258D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E258D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:37:34',NULL,NULL,NULL,NULL,'296911f8-d9bb-4a09-a6bf-08e45bd232a4',7157,NULL),(29131,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:37:50',NULL,NULL,NULL,NULL,'93930cca-ffe6-4d51-b585-89c9bf71a4d6',7157,NULL),(29132,NULL,'The TP53 E258G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 11051249). In vivo studies with yeast expressing TP53 E258G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E258G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:37:58',NULL,NULL,NULL,NULL,'3a00f49c-c590-4a39-9bef-a89965edb009',7157,NULL),(29133,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:38:10',NULL,NULL,NULL,NULL,'dcb95978-8667-45e9-9407-c48d7280d7c1',7157,NULL),(29134,NULL,'The TP53 E258K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 8878455). In vivo studies with yeast expressing TP53 E258K demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E258K also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:38:19',NULL,NULL,NULL,NULL,'7e5472e2-33d1-437f-b139-fee9e22354b9',7157,NULL),(29135,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:42:59',NULL,NULL,NULL,NULL,'e732b84e-2160-460b-95f0-c29dcb779fc7',7157,NULL),(29136,NULL,'The TP53 I255S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 9736425). In vivo studies with yeast expressing TP53 I255S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I255S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:43:09',NULL,NULL,NULL,NULL,'bbbfa8aa-dd87-4456-b62a-c557fcf83401',7157,NULL),(29137,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:43:18',NULL,NULL,NULL,NULL,'29b9864d-a84a-4a72-97b2-cfeb8fff666b',7157,NULL),(29138,NULL,'The TP53 I255T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 12118339). In vivo studies with yeast expressing TP53 I255T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I255T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:43:26',NULL,NULL,NULL,NULL,'8a7676f5-a723-4110-92c8-d4e4c25d3dc3',7157,NULL),(29139,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:43:39',NULL,NULL,NULL,NULL,'ef1bf2c7-6c3d-4284-8def-f330f174d38c',7157,NULL),(29140,NULL,'The TP53 I255N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in adenocarcinoma and is a statistically significant hotspot (PMID: 8187092). In vivo studies with yeast expressing TP53 I255N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I255N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:43:44',NULL,NULL,NULL,NULL,'799d989f-3b74-4eac-bf02-eda4058a6bf5',7157,NULL),(29141,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:43:55',NULL,NULL,NULL,NULL,'ba4af5ef-67df-4456-91e6-a8497a64aba6',7157,NULL),(29142,NULL,'The TP53 I254S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 9262496). In vivo studies with yeast expressing TP53 I254S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I254S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:44:03',NULL,NULL,NULL,NULL,'b354a9e5-9966-4eb6-b59e-d058c820c936',7157,NULL),(29143,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-01-24 14:44:20',NULL,NULL,NULL,NULL,'7d748519-325f-4637-91ec-91fc3685803a',7157,NULL),(29144,NULL,'The TP53 I254V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 15370252). In vivo studies with yeast expressing TP53 I254V demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I254V also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-01-26 15:32:57',NULL,NULL,NULL,NULL,'42087c14-2ab6-4db9-a9dc-35d7e881d66e',7157,NULL),(29145,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:44:44',NULL,NULL,NULL,NULL,'a69be9b5-0059-4284-97f5-b438c57aa6dd',7157,NULL),(29146,NULL,'The TP53 I254F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 17557246). In vivo studies with yeast expressing TP53 I254F demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:44:54',NULL,NULL,NULL,NULL,'ca5454d7-7726-417b-9e7e-23e3266dbdc9',7157,NULL),(29147,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-01-24 14:45:04',NULL,NULL,NULL,NULL,'64395ac8-49d5-42f7-91f6-3193ee178584',7157,NULL),(29148,NULL,'The TP53 I254L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant fibrous histiocytoma and is a statistically significant hotspot (PMID: 9645763). In vivo studies with yeast expressing TP53 I254L demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I254L also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-01-26 15:33:42',NULL,NULL,NULL,NULL,'d5656eb9-237f-4f99-a34e-4c77fb946894',7157,NULL),(29149,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:45:26',NULL,NULL,NULL,NULL,'66d97394-5062-4abb-9069-bffe31aafb77',7157,NULL),(29150,NULL,'The TP53 I254M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in B-cell chronic lymphocytic leukemia and is a statistically significant hotspot (PMID: 18980985). In vivo studies with yeast expressing TP53 I254M demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:45:34',NULL,NULL,NULL,NULL,'efeecb2d-1352-4480-bce7-44d3ba484bf0',7157,NULL),(29151,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:45:55',NULL,NULL,NULL,NULL,'45276fc1-c8ff-4a5c-9dd0-0edf00c048dc',7157,NULL),(29152,NULL,'The TP53 I254N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Burkitt lymphoma and is a statistically significant hotspot (PMID: 9172810). In vivo studies with yeast expressing TP53 I254N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I254N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:46:03',NULL,NULL,NULL,NULL,'39ec4068-10b3-476f-b957-b874221f2a1e',7157,NULL),(29153,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 14:56:33',NULL,NULL,NULL,NULL,'bfb839e4-b32d-4321-8182-fd3aac6d601a',7157,NULL),(29154,NULL,'The TP53 V272G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 10355250). In vivo studies with yeast expressing TP53 V272G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V272G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 14:56:42',NULL,NULL,NULL,NULL,'e5510754-9209-4205-baca-0946b6eba1fc',7157,NULL),(29155,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 15:14:28',NULL,NULL,NULL,NULL,'38143c1e-3a9c-48f3-b1a9-e65d13277113',7157,NULL),(29156,NULL,'The TP53 R337S mutation is located in the tetramerization motif of the protein. This mutation has been identified in biliary tract cancer and is a statistically significant hotspot (PMID: 29360550). In vivo studies with yeast expressing TP53 R337S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 15:14:43',NULL,NULL,NULL,NULL,'c0d6f6d4-1951-4031-a46c-02c8df64edc0',7157,NULL),(29157,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 15:23:12',NULL,NULL,NULL,NULL,'06e146bc-daac-41d5-a743-c9ad74de9f05',7157,NULL),(29158,NULL,'The TP53 V272A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in pancreatic cancer and is a statistically significant hotspot (PMID: 1630814). In vivo studies with yeast expressing TP53 V272A demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 15:23:18',NULL,NULL,NULL,NULL,'72a188d1-fb9d-4dff-8ac1-8a318f873aeb',7157,NULL),(29159,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 15:23:31',NULL,NULL,NULL,NULL,'710f3ab3-5dbc-4bda-89a4-583aed3e0f19',7157,NULL),(29160,NULL,'The TP53 V272E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in diffuse large B-cell lymphoma and is a statistically significant hotspot (PMID: 17881637). In vivo studies with yeast expressing TP53 V272E demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V272E also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 15:23:36',NULL,NULL,NULL,NULL,'15eccbf2-5330-4665-907f-c76730dcddfe',7157,NULL),(29161,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 15:23:49',NULL,NULL,NULL,NULL,'1434c6ab-4456-43ec-bf1c-c93c95a0dfdb',7157,NULL),(29162,NULL,'The TP53 V272K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung adenocarcinoma and is a statistically significant hotspot (PMID: 22980975). In vitro studies with various human cancer cell lines expressing TP53 V272K demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 15:23:54',NULL,NULL,NULL,NULL,'b90f144d-9b18-43a3-bc2d-51844b17e4f8',7157,NULL),(29163,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 15:25:54',NULL,NULL,NULL,NULL,'c7ebf3fa-4b02-474b-8add-b740d4008952',7157,NULL),(29164,NULL,'The TP53 R267W mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 17881637). In vivo studies with yeast expressing TP53 R267W demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R267W also demonstrated the mutation is inactivating as measured by moderately reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 15:25:59',NULL,NULL,NULL,NULL,'046003c6-7fc7-40c9-88f6-38b5b02c7294',7157,NULL),(29165,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-24 15:26:12',NULL,NULL,NULL,NULL,'df42cc7f-2f89-48eb-9ff6-d47fe5c48a8f',7157,NULL),(29166,NULL,'The TP53 R267L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in precursor cell lymphoblastic leukemia and is a statistically significant hotspot (PMID: 16844765). In vivo studies with yeast expressing TP53 R267L demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-24 15:26:14',NULL,NULL,NULL,NULL,'3f0056f9-3dd1-43bb-aea8-98fb06612804',7157,NULL),(29167,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-25 15:06:47',NULL,NULL,NULL,NULL,'c5ab1822-d072-4cec-a088-5b0edaa169c9',7157,NULL),(29168,NULL,'The TP53 V157A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in cutaneous T-cell lymphoma and is a statistically significant hotspot (PMID: 10551408). In vivo studies with yeast expressing TP53 V157A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V157A also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-25 15:06:51',NULL,NULL,NULL,NULL,'ece7f7bb-a624-4ce5-8da4-5896943b4160',7157,NULL),(29169,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-25 15:07:06',NULL,NULL,NULL,NULL,'6e0da137-96f3-4804-b72f-ab00632c74bc',7157,NULL),(29170,NULL,'The TP53 V157G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 7852189). In vivo studies with yeast expressing TP53 V157G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V157G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-25 15:07:10',NULL,NULL,NULL,NULL,'31d7862d-729c-4c70-ae52-5add10a422d2',7157,NULL),(29171,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-25 15:07:26',NULL,NULL,NULL,NULL,'f97a845e-2be1-4d51-bb72-02153855edbe',7157,NULL),(29172,NULL,'The TP53 V157L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 7694948). In vivo studies with yeast expressing TP53 V157L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V157L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-25 15:07:31',NULL,NULL,NULL,NULL,'d0b12068-c8cc-4d99-87e9-fc54ce42eee6',7157,NULL),(29173,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-25 15:07:54',NULL,NULL,NULL,NULL,'2ac95d22-081e-44a1-932b-c498b3d1f357',7157,NULL),(29174,NULL,'The TP53 R110C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in rectal cancer and is a statistically significant hotspot (PMID: 10884918). In vivo studies with yeast expressing TP53 R110C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-25 15:07:56',NULL,NULL,NULL,NULL,'9cd1932e-06c7-4386-8a8f-935b4c43394b',7157,NULL),(29175,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-01-25 15:08:14',NULL,NULL,NULL,NULL,'1f078e45-d318-4515-8697-31a44a637053',7157,NULL),(29176,NULL,'The TP53 R110H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 14612556). In vivo studies with yeast expressing TP53 R110H demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-01-25 15:08:16',NULL,NULL,NULL,NULL,'f3680fd7-29f9-4f01-b2ea-1bbcaccb0849',7157,NULL),(29177,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:20:18',NULL,NULL,NULL,NULL,'b89aeac5-a0ee-4a31-8032-d3fa3ed45165',7157,NULL),(29178,NULL,'The TP53 R249T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 14697642). In vivo studies with yeast expressing TP53 R249T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R249T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:20:24',NULL,NULL,NULL,NULL,'245b550b-772b-4077-99e6-949d96ea7b6c',7157,NULL),(29179,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:20:52',NULL,NULL,NULL,NULL,'388e5e69-f4bb-4767-a922-6e77c5dbcfc8',7157,NULL),(29180,NULL,'The TP53 R249G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 8673929). In vivo studies with yeast expressing TP53 R249G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R249G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:20:58',NULL,NULL,NULL,NULL,'32294a96-2e1d-4cfd-a31b-b29257426666',7157,NULL),(29181,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:21:22',NULL,NULL,NULL,NULL,'4fbf575b-521d-4e15-9acd-bd3d03d9f1d5',7157,NULL),(29182,NULL,'The TP53 R249I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in small cell lung cancer and is a statistically significant hotspot (PMID: 10551408). In vitro studies with various human cancer cell lines expressing TP53 R249I demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:21:27',NULL,NULL,NULL,NULL,'4279dbe0-3c2c-4b9f-b12e-d3c929007326',7157,NULL),(29183,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:21:56',NULL,NULL,NULL,NULL,'dbff5f06-2c96-4789-9b8b-ab997429521b',7157,NULL),(29184,NULL,'The TP53 R249K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11737306). In vivo studies with yeast expressing TP53 R249K demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:22:02',NULL,NULL,NULL,NULL,'0a6c8ee3-d351-4de9-8d22-6f879932f9b9',7157,NULL),(29185,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-01 16:22:40',NULL,NULL,NULL,NULL,'8fa3286a-1b6c-4df7-9257-8234b32d8969',7157,NULL),(29186,NULL,'The TP53 R337G mutation is located in the tetramerization motif of the protein and is a statistically significant hotspot. In vivo studies with yeast expressing TP53 R337G demonstrated that the mutation is likely neutral as measured by wildtype comparable transactivational activity (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R337G also demonstrated the mutation likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-02-01 16:22:49',NULL,NULL,NULL,NULL,'082ac28d-d3d3-48ee-867d-ea14ea7e7761',7157,NULL),(29187,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:35:14',NULL,NULL,NULL,NULL,'ca5291da-1dda-408f-8529-8eba0a1c290a',7157,NULL),(29188,NULL,'The TP53 A161P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in prolymphocytic leukemia and is a statistically significant hotspot (PMID: 9058723). In vivo studies with yeast expressing TP53 A161P demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:35:19',NULL,NULL,NULL,NULL,'4bd6ffda-0b3c-4d4b-a29a-709c216ea6cc',7157,NULL),(29189,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:35:37',NULL,NULL,NULL,NULL,'aedf58d8-9ef0-49bb-9474-8c5abd9e20ea',7157,NULL),(29190,NULL,'The TP53 A161D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 8522300). In vivo studies with yeast expressing TP53 A161D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:35:44',NULL,NULL,NULL,NULL,'f9f94d08-f6f3-4c8c-97be-692eda9ae9c7',7157,NULL),(29191,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:36:01',NULL,NULL,NULL,NULL,'c5b6fb1d-3044-4a97-9efe-7652edb12c99',7157,NULL),(29192,NULL,'The TP53 A161V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Burkitt lymphoma and is a statistically significant hotspot (PMID: 12890146). In vivo studies with yeast expressing TP53 A161V demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:36:03',NULL,NULL,NULL,NULL,'e20b4445-4b9b-49e3-9c76-569dc9c94a9e',7157,NULL),(29193,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:40:49',NULL,NULL,NULL,NULL,'50eaef1b-1fdc-4406-af22-e0bb9ab84efa',7157,NULL),(29194,NULL,'The TP53 R267G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 7869472). In vivo studies with yeast expressing TP53 R267G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R267G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:40:53',NULL,NULL,NULL,NULL,'d5c82ebf-3558-4fbb-956e-cb38a26f059b',7157,NULL),(29195,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:41:08',NULL,NULL,NULL,NULL,'0203e349-1d23-456d-bc5e-822a3fccc8a6',7157,NULL),(29196,NULL,'The TP53 Y220D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 9617346). In vivo studies with yeast expressing TP53 Y220D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y220D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:41:13',NULL,NULL,NULL,NULL,'9767cfd5-551e-4b61-8c2e-fff057f3f72f',7157,NULL),(29197,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:41:25',NULL,NULL,NULL,NULL,'02a04796-5a29-4073-8908-b0de455d4c18',7157,NULL),(29198,NULL,'The TP53 Y220H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 17683074). In vivo studies with yeast expressing TP53 Y220H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y220H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:41:30',NULL,NULL,NULL,NULL,'5e237df7-8b5e-4697-ab51-bc2db9ea3b8d',7157,NULL),(29199,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:41:45',NULL,NULL,NULL,NULL,'2218463b-3ccc-4ad6-b4a2-ea3dcb567fa0',7157,NULL),(29200,NULL,'The TP53 Y220N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 11375957). In vivo studies with yeast expressing TP53 Y220N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y220N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:41:47',NULL,NULL,NULL,NULL,'0d3ca595-7b31-4e46-9c9a-2e073439e79c',7157,NULL),(29201,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:42:03',NULL,NULL,NULL,NULL,'6e84e1ec-8946-454e-a1fb-7f10321c38d8',7157,NULL),(29202,NULL,'The TP53 F134C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 9014203). In vivo studies with yeast expressing TP53 F134C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F134C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:42:05',NULL,NULL,NULL,NULL,'084c437e-b0de-46fa-9312-9a9b90caea0f',7157,NULL),(29203,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:42:39',NULL,NULL,NULL,NULL,'b278cb0c-0b35-447d-95bc-f5d0820ee770',7157,NULL),(29204,NULL,'The TP53 R273P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 1734086). In vivo studies with yeast expressing TP53 R273P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R273P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:42:41',NULL,NULL,NULL,NULL,'1cdfd7c2-7257-46ed-a235-5c41c1bae05d',7157,NULL),(29205,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:42:55',NULL,NULL,NULL,NULL,'e5fdcc40-ceda-4592-8226-8af45e39fcac',7157,NULL),(29206,NULL,'The TP53 R273S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung cancer and is a statistically significant hotspot (PMID: 7767998). In vivo studies with yeast expressing TP53 R273S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R273S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:42:58',NULL,NULL,NULL,NULL,'b1a40a87-bf09-45f9-9e2c-e7939869f399',7157,NULL),(29207,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:43:09',NULL,NULL,NULL,NULL,'e14fc53a-b7c1-4be6-9ea9-dcfb1853bcd8',7157,NULL),(29208,NULL,'The TP53 F134S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in astrocytoma and is a statistically significant hotspot (PMID: 17917588). In vivo studies with yeast expressing TP53 F134S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F134S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:43:11',NULL,NULL,NULL,NULL,'416915fa-8990-4e65-972b-dd437adfd9cd',7157,NULL),(29209,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:45:13',NULL,NULL,NULL,NULL,'1c624fa5-591a-4b8d-910e-aa7a475c64a8',7157,NULL),(29210,NULL,'The TP53 F134V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 14633673). In vivo studies with yeast expressing TP53 F134V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F134V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:45:15',NULL,NULL,NULL,NULL,'4e6e07f3-f801-4f51-a15a-e6be1a2a48d9',7157,NULL),(29211,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:45:26',NULL,NULL,NULL,NULL,'b929b8c0-f570-4e5b-ac50-ea4b2dc8a3ab',7157,NULL),(29212,NULL,'The TP53 F134I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioma and is a statistically significant hotspot (PMID: 8561860). In vivo studies with yeast expressing TP53 F134I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F134I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:45:30',NULL,NULL,NULL,NULL,'9cbb3d84-44f9-4322-b0db-a648e0f9134c',7157,NULL),(29213,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:49:39',NULL,NULL,NULL,NULL,'74235ff2-0114-4c1e-bc48-8993e2922a11',7157,NULL),(29214,NULL,'The TP53 F134L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 8260732). In vivo studies with yeast expressing TP53 F134L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F134L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:49:41',NULL,NULL,NULL,NULL,'b48bbd6f-f3a6-4f09-8cf6-bc942a49bfb7',7157,NULL),(29215,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:50:19',NULL,NULL,NULL,NULL,'9a0a90c5-547d-4c73-9802-9f04900e5f11',7157,NULL),(29216,NULL,'The TP53 G245R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung adenocarcinoma and is a statistically significant hotspot (PMID: 9627113). In vivo studies with yeast expressing TP53 G245R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G245R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:50:24',NULL,NULL,NULL,NULL,'ecd4a668-f550-4660-ace4-9454b1d06bc9',7157,NULL),(29217,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:50:39',NULL,NULL,NULL,NULL,'c2fa718a-6009-4255-b9ea-3e7937463feb',7157,NULL),(29218,NULL,'The TP53 G245V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung squamous cell carcinoma and is a statistically significant hotspot (PMID: 7903781). In vivo studies with yeast expressing TP53 G245V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G245V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:50:42',NULL,NULL,NULL,NULL,'9938a0f5-5421-4a24-90ed-c51a7b0b1e9f',7157,NULL),(29219,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:50:56',NULL,NULL,NULL,NULL,'1b5a1617-b23f-45e2-81f5-fbbea04025b8',7157,NULL),(29220,NULL,'The TP53 G245N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 1946433). In vitro studies with various human cancer cell lines expressing TP53 G245N demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:51:01',NULL,NULL,NULL,NULL,'1ecd4ec9-ea8f-4295-991a-481a7c53a1c6',7157,NULL),(29221,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:55:40',NULL,NULL,NULL,NULL,'f91185fe-c6ad-4204-8e9f-8e5a2c0ac98b',7157,NULL),(29222,NULL,'The TP53 S215R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 16907706). In vivo studies with yeast expressing TP53 S215R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S215R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:55:43',NULL,NULL,NULL,NULL,'3766c612-8d4e-4c22-b23a-f1d2ccb811e6',7157,NULL),(29223,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:55:56',NULL,NULL,NULL,NULL,'138a36fc-81ca-47b0-ac78-5c07b1296d81',7157,NULL),(29224,NULL,'The TP53 S215T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in giant cell glioblastoma and is a statistically significant hotspot (PMID: 9284834). In vivo studies with yeast expressing TP53 S215T demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S215T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:56:00',NULL,NULL,NULL,NULL,'b5da045b-88ca-4f07-ac9c-3e90ddb17222',7157,NULL),(29225,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:56:12',NULL,NULL,NULL,NULL,'562dd2d0-492f-42e8-b7f9-d82a3d17bdeb',7157,NULL),(29226,NULL,'The TP53 S215I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 10557071). In vivo studies with yeast expressing TP53 S215I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S215I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:56:17',NULL,NULL,NULL,NULL,'28f17009-dfe8-48d1-b7c3-d54b8cb6ac15',7157,NULL),(29227,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:56:29',NULL,NULL,NULL,NULL,'465f18bb-d5fa-4afd-b0c0-9d0f0d3aee1a',7157,NULL),(29228,NULL,'The TP53 S215N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 10499618). In vivo studies with yeast expressing TP53 S215N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S215N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:56:35',NULL,NULL,NULL,NULL,'01503d51-6730-43ea-8bbd-be8753c852ec',7157,NULL),(29229,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:57:58',NULL,NULL,NULL,NULL,'c1acc1db-c0c7-441a-9fb9-9932a49a5028',7157,NULL),(29230,NULL,'The TP53 Q144H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 10615230). In vivo studies with yeast expressing TP53 Q144H demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-01 16:58:06',NULL,NULL,NULL,NULL,'1726f039-4618-47d6-ba34-1dc5db8a4259',7157,NULL),(29231,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-01 16:58:21',NULL,NULL,NULL,NULL,'7c539d2d-5a5c-40fc-8349-12caca6a5bb6',7157,NULL),(29232,NULL,'The TP53 Q144L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 12457032). In vivo studies with yeast expressing TP53 Q144L demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-28 15:17:26',NULL,NULL,NULL,NULL,'22ee7a33-f556-475b-a579-17d1de3003c9',7157,NULL),(29233,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:32:34',NULL,NULL,NULL,NULL,'27c93ca8-a87e-4ace-aa16-9f95b6824db5',7157,NULL),(29234,NULL,'The TP53 Y234S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 1923532). In vivo studies with yeast expressing TP53 Y234S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y234S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:32:38',NULL,NULL,NULL,NULL,'ba81abfe-5f89-4c70-9bbf-5a7527e1a925',7157,NULL),(29235,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:32:53',NULL,NULL,NULL,NULL,'ce18fe6a-8600-4490-abe2-2b46f7fc6049',7157,NULL),(29236,NULL,'The TP53 Y234D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in acute myeloid leukemia and is a statistically significant hotspot (PMID: 8824725). In vivo studies with yeast expressing TP53 Y234D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y234D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:32:59',NULL,NULL,NULL,NULL,'ec4aa232-55f5-4e80-91f5-a30626170c94',7157,NULL),(29237,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:33:15',NULL,NULL,NULL,NULL,'1d0c3c9f-b74c-4b1d-9387-cf45e770d32b',7157,NULL),(29238,NULL,'The TP53 Y234N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in osteosarcoma and is a statistically significant hotspot (PMID: 9824113). In vivo studies with yeast expressing TP53 Y234N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y234N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:33:19',NULL,NULL,NULL,NULL,'3f6956dc-0b82-4a6d-82da-092feabe22bc',7157,NULL),(29239,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:33:36',NULL,NULL,NULL,NULL,'0b2fd09b-905e-4061-a7ba-810fb5426af8',7157,NULL),(29240,NULL,'The TP53 R248P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Burkitt cell leukemia and is a statistically significant hotspot (PMID: 9379679). In vivo studies with yeast expressing TP53 R248P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R248P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:33:40',NULL,NULL,NULL,NULL,'e5ef4964-08a3-4f4c-a293-5383562a2ca6',7157,NULL),(29241,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:33:54',NULL,NULL,NULL,NULL,'86fee14e-d26d-4a9a-84f6-059fe6a8c53e',7157,NULL),(29242,NULL,'The TP53 R248G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in prostate adenocarcinoma and is a statistically significant hotspot (PMID: 8610059). In vivo studies with yeast expressing TP53 R248G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R248G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:33:58',NULL,NULL,NULL,NULL,'ce936ea3-b7d6-460d-9a61-4ec9d59836a8',7157,NULL),(29243,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:48:50',NULL,NULL,NULL,NULL,'de69d2ed-55b9-4529-bc34-8efdade5c4b6',7157,NULL),(29244,NULL,'The TP53 K132Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 8314342). In vivo studies with yeast expressing TP53 K132Q demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K132Q also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:48:53',NULL,NULL,NULL,NULL,'83053575-5b65-48cd-b557-44a68cd84b5b',7157,NULL),(29245,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:49:06',NULL,NULL,NULL,NULL,'e77df0e2-d12d-4d5c-892d-8082703ea364',7157,NULL),(29246,NULL,'The TP53 K132R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 17201907). In vivo studies with yeast expressing TP53 K132R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K132R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:49:10',NULL,NULL,NULL,NULL,'03219687-723f-4e61-a6fc-e6b1ba0d48c2',7157,NULL),(29247,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:49:22',NULL,NULL,NULL,NULL,'ea17a4da-e3fe-4d61-8a38-1858b966807d',7157,NULL),(29248,NULL,'The TP53 K132T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 11221842). In vivo studies with yeast expressing TP53 K132T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K132T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:49:26',NULL,NULL,NULL,NULL,'183c2bcc-2277-41fe-b941-66085bd03a65',7157,NULL),(29249,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:49:44',NULL,NULL,NULL,NULL,'f728ad35-4588-496d-971a-c89e788b5553',7157,NULL),(29250,NULL,'The TP53 K132E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 15188009). In vivo studies with yeast expressing TP53 K132E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K132E also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:49:48',NULL,NULL,NULL,NULL,'97347d7b-6750-4a93-a1ad-bc0128fd2046',7157,NULL),(29251,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:50:03',NULL,NULL,NULL,NULL,'009fd9ca-841e-4e74-99fb-ecbc75b20a27',7157,NULL),(29252,NULL,'The TP53 K132M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Ewing sarcoma and is a statistically significant hotspot (PMID: 11295075). In vivo studies with yeast expressing TP53 K132M demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K132M also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:50:05',NULL,NULL,NULL,NULL,'96b3975a-ef50-4802-8d72-e0466dd1f879',7157,NULL),(29253,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:50:22',NULL,NULL,NULL,NULL,'99196e51-528a-46e4-b706-1c785358e7ca',7157,NULL),(29254,NULL,'The TP53 K132N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 17557246). In vivo studies with yeast expressing TP53 K132N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K132N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:50:24',NULL,NULL,NULL,NULL,'58e13214-d535-41c0-9b6f-28d30ca8f398',7157,NULL),(29255,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:50:38',NULL,NULL,NULL,NULL,'0fe179c9-e3a3-4cd0-9212-32a640e9652b',7157,NULL),(29256,NULL,'The TP53 H214L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 26873401). In vivo studies with yeast expressing TP53 H214L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 H214L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:50:41',NULL,NULL,NULL,NULL,'a605492e-aa49-49a8-ab3f-68fd59c8b9f9',7157,NULL),(29257,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:51:00',NULL,NULL,NULL,NULL,'fe118a46-f238-41ea-b4a4-34dd9d76180c',7157,NULL),(29258,NULL,'The TP53 L130P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 15875732). In vivo studies with yeast expressing TP53 L130P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L130P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:51:01',NULL,NULL,NULL,NULL,'62987a47-2d20-423c-9d3f-4938f4c367cc',7157,NULL),(29259,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:51:14',NULL,NULL,NULL,NULL,'5a5155df-61ae-4459-9abd-58eca898d3bc',7157,NULL),(29260,NULL,'The TP53 L130R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in cholangiocarcinoma and is a statistically significant hotspot (PMID: 12210082). In vivo studies with yeast expressing TP53 L130R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L130R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:51:18',NULL,NULL,NULL,NULL,'bd0c288f-590c-470f-8f61-a57ebbcbaac3',7157,NULL),(29261,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:51:32',NULL,NULL,NULL,NULL,'6b67fadb-1206-4e66-8251-00e98395512b',7157,NULL),(29262,NULL,'The TP53 L130V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 17171684). In vivo studies with yeast expressing TP53 L130V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L130V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:51:37',NULL,NULL,NULL,NULL,'ef200fc3-095d-4b0f-c00b-6da9e50b8130',7157,NULL),(29263,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:51:57',NULL,NULL,NULL,NULL,'55785a9b-820e-4270-8396-85829e39d1f1',7157,NULL),(29264,NULL,'The TP53 L130F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 11051239). In vivo studies with yeast expressing TP53 L130F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L130F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:52:01',NULL,NULL,NULL,NULL,'e0dde03b-410b-4825-8376-609306f0a132',7157,NULL),(29265,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:52:17',NULL,NULL,NULL,NULL,'37705421-7861-43cc-8869-b5fb9203bd1f',7157,NULL),(29266,NULL,'The TP53 L130H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in prostate cancer and is a statistically significant hotspot (PMID: 19020536). In vivo studies with yeast expressing TP53 L130H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L130H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:52:23',NULL,NULL,NULL,NULL,'854360f1-49f1-4e36-af09-54313c43f672',7157,NULL),(29267,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:52:37',NULL,NULL,NULL,NULL,'b5d501d2-ed64-43e8-b225-b5f40aea3263',7157,NULL),(29268,NULL,'The TP53 K164E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 9699640). In vivo studies with yeast expressing TP53 K164E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 K164E also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:52:39',NULL,NULL,NULL,NULL,'a5928c78-fc9b-4e13-bf1e-d9053d994709',7157,NULL),(29269,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:52:54',NULL,NULL,NULL,NULL,'9dc305dc-3c89-43c1-a223-6fba3ec9772f',7157,NULL),(29270,NULL,'The TP53 F109S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in acute lymphoblastic leukemia and is a statistically significant hotspot (PMID: 10573131). In vivo studies with yeast expressing TP53 F109S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F109S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:53:00',NULL,NULL,NULL,NULL,'a9ff347b-b34a-42ee-9b43-4a151b57bc22',7157,NULL),(29271,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:53:19',NULL,NULL,NULL,NULL,'d0daea4f-4300-4794-8e0b-0e09eec91b43',7157,NULL),(29272,NULL,'The TP53 F109C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 21720365). In vivo studies with yeast expressing TP53 F109C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F109C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:53:21',NULL,NULL,NULL,NULL,'53c403ce-c4c8-4b7a-ad34-713319746af1',7157,NULL),(29273,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:53:40',NULL,NULL,NULL,NULL,'af277a58-8921-488a-9429-5f110517eb36',7157,NULL),(29274,NULL,'The TP53 F109V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 16489069). In vivo studies with yeast expressing TP53 F109V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F109V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:53:42',NULL,NULL,NULL,NULL,'a54425dc-550c-4e09-9d02-1292718a6dd8',7157,NULL),(29275,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:53:59',NULL,NULL,NULL,NULL,'523633ac-495c-4e76-972b-8c1fc30ae83d',7157,NULL),(29276,NULL,'The TP53 F109I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in penile squamous cell carcinoma and is a statistically significant hotspot (PMID: 26670561). In vivo studies with yeast expressing TP53 F109I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F109I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:54:03',NULL,NULL,NULL,NULL,'2ba2d557-5471-415f-93bc-849a30776898',7157,NULL),(29277,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:54:19',NULL,NULL,NULL,NULL,'09bac2c2-4a8f-4dc8-a1a8-11dd405dd692',7157,NULL),(29278,NULL,'The TP53 G105R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 9699640). In vivo studies with yeast expressing TP53 G105R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G105R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:54:22',NULL,NULL,NULL,NULL,'2e55007e-8646-44bd-bc6d-418ff4127029',7157,NULL),(29279,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:54:39',NULL,NULL,NULL,NULL,'326059a9-478c-4802-8833-35f25f05c5a8',7157,NULL),(29280,NULL,'The TP53 G105C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 18772890). In vivo studies with yeast expressing TP53 G105C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:54:44',NULL,NULL,NULL,NULL,'798b8061-c885-47bb-8729-b450f95152eb',7157,NULL),(29281,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:55:08',NULL,NULL,NULL,NULL,'75b8ee0c-7b3a-4200-95b7-05f232c7bf94',7157,NULL),(29282,NULL,'The TP53 G105S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 15035290). In vivo studies with yeast expressing TP53 G105S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:55:15',NULL,NULL,NULL,NULL,'927b4ab9-b20d-462e-81f8-8df84bcbdced',7157,NULL),(29283,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:55:27',NULL,NULL,NULL,NULL,'b51de8d6-3137-4573-a8a1-3ea365f6d8cb',7157,NULL),(29284,NULL,'The TP53 G105D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 15466178). In vivo studies with yeast expressing TP53 G105D demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G105D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:55:33',NULL,NULL,NULL,NULL,'a8b36404-c919-4661-b5a7-477c9265199e',7157,NULL),(29285,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:55:51',NULL,NULL,NULL,NULL,'61f60bc1-737b-4721-955e-4799fc3212ff',7157,NULL),(29286,NULL,'The TP53 G105V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 14519637). In vivo studies with yeast expressing TP53 G105V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:55:56',NULL,NULL,NULL,NULL,'3a8dc31c-5c99-406c-9a13-175a7e549646',7157,NULL),(29287,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:56:10',NULL,NULL,NULL,NULL,'9fb92a6f-d944-467e-a3ba-5bb3c19ff398',7157,NULL),(29288,NULL,'The TP53 E271Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 11929815). In vivo studies with yeast expressing TP53 E271Q demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:56:20',NULL,NULL,NULL,NULL,'40ecec3a-9808-4671-8f9d-6fa10aa135c6',7157,NULL),(29289,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:56:33',NULL,NULL,NULL,NULL,'3acc0172-dee2-4f85-b2cb-a5b66c8b2ddf',7157,NULL),(29290,NULL,'The TP53 E271V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 7628866). In vivo studies with yeast expressing TP53 E271V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E271V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:56:37',NULL,NULL,NULL,NULL,'8a8f6b3a-61d6-4929-bbd4-32b819d22fac',7157,NULL),(29291,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:56:57',NULL,NULL,NULL,NULL,'2f34edc2-655b-4b9c-b919-d3d2124b5959',7157,NULL),(29292,NULL,'The TP53 V172D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 7712430). In vivo studies with yeast expressing TP53 V172D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V172D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:57:03',NULL,NULL,NULL,NULL,'7b5883c3-3fff-450e-81bf-c2955375127b',7157,NULL),(29293,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:57:22',NULL,NULL,NULL,NULL,'91b64288-435a-4997-8d75-b836d8126890',7157,NULL),(29294,NULL,'The TP53 V172G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 8108145). In vivo studies with yeast expressing TP53 V172G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V172G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:57:24',NULL,NULL,NULL,NULL,'06dd1bbd-e484-4b63-912f-dd60f0b7f257',7157,NULL),(29295,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:57:41',NULL,NULL,NULL,NULL,'5d4710c9-4e32-4c9a-9a87-1fb63274fe13',7157,NULL),(29296,NULL,'The TP53 T155I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung adenocarcinoma and is a statistically significant hotspot (PMID: 11400116). In vivo studies with yeast expressing TP53 T155I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 T155I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:57:48',NULL,NULL,NULL,NULL,'5aee1f19-8652-466c-be94-521faa3608e6',7157,NULL),(29297,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:58:06',NULL,NULL,NULL,NULL,'d675b61e-2df6-4e75-8c28-f28c077b6705',7157,NULL),(29298,NULL,'The TP53 L257P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 1375111). In vivo studies with yeast expressing TP53 L257P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L257P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:58:09',NULL,NULL,NULL,NULL,'963a78f3-0a2f-4ace-ba6b-6e79a09afa07',7157,NULL),(29299,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:58:26',NULL,NULL,NULL,NULL,'2947f759-fb3e-4c1a-bd05-9a7ce55db031',7157,NULL),(29300,NULL,'The TP53 L257Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 20229506). In vivo studies with yeast expressing TP53 L257Q demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L257Q also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:58:31',NULL,NULL,NULL,NULL,'c410c596-c32f-4b39-8691-c8f8c035c043',7157,NULL),(29301,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:58:51',NULL,NULL,NULL,NULL,'c74bc26e-0ec8-4bfb-9c2d-151574046249',7157,NULL),(29302,NULL,'The TP53 L257R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 8700537). In vivo studies with yeast expressing TP53 L257R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L257R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:58:53',NULL,NULL,NULL,NULL,'eaf95a60-da3e-4109-8270-a7005ab78001',7157,NULL),(29303,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:59:09',NULL,NULL,NULL,NULL,'b9b52009-122d-4cf5-941a-ec3d5791b385',7157,NULL),(29304,NULL,'The TP53 L257V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioma and is a statistically significant hotspot (PMID: 12635658). In vivo studies with yeast expressing TP53 L257V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L257V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:59:13',NULL,NULL,NULL,NULL,'97bae24c-add8-4ad9-b793-0410baa2f451',7157,NULL),(29305,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:59:29',NULL,NULL,NULL,NULL,'384bf3a8-0dbb-4723-a34f-5463e4dededc',7157,NULL),(29306,NULL,'The TP53 N131I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 10492244). In vivo studies with yeast expressing TP53 N131I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 N131I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:59:35',NULL,NULL,NULL,NULL,'f90e9c7c-a7ed-45a9-b0ed-34e4d7bc6706',7157,NULL),(29307,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 15:59:52',NULL,NULL,NULL,NULL,'a8fe2092-4636-47b6-8f95-4fd09633a1f8',7157,NULL),(29308,NULL,'The TP53 N131K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 16271069). In vivo studies with yeast expressing TP53 N131K demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 15:59:57',NULL,NULL,NULL,NULL,'f98965dd-f01e-4474-a464-c8df17f361da',7157,NULL),(29309,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:00:10',NULL,NULL,NULL,NULL,'3c3aade7-7c32-4361-893f-25eec3101688',7157,NULL),(29310,NULL,'The TP53 L265P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 14760384). In vivo studies with yeast expressing TP53 L265P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L265P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:00:16',NULL,NULL,NULL,NULL,'15dfb079-b60c-418b-b80c-c492000af976',7157,NULL),(29311,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:03:19',NULL,NULL,NULL,NULL,'b8dee627-770b-4213-82e6-383a73f72a16',7157,NULL),(29312,NULL,'The TP53 L265R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11307149). In vivo studies with yeast expressing TP53 L265R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L265R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:03:24',NULL,NULL,NULL,NULL,'a0e182ae-d4a4-4b34-a4c2-167e24e1a16e',7157,NULL),(29313,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:03:37',NULL,NULL,NULL,NULL,'946a1d0f-f5d7-47f5-ae85-1e83e68449d4',7157,NULL),(29314,NULL,'The TP53 L265M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 15050734). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 L265M demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 L265M demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:03:41',NULL,NULL,NULL,NULL,'cfb2f296-d497-420e-c0ec-24e9a3deeacd',7157,NULL),(29315,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:03:57',NULL,NULL,NULL,NULL,'9ec6d731-1746-4e30-b05e-49da6c1f5028',7157,NULL),(29316,NULL,'The TP53 G262V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11307149). In vivo studies with yeast expressing TP53 G262V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 G262V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:04:02',NULL,NULL,NULL,NULL,'5aa75f46-53ed-4ff9-b3e7-ee7d1b7874a0',7157,NULL),(29317,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:04:19',NULL,NULL,NULL,NULL,'18a5ba78-fa38-4d3c-a450-f68c57b44bcf',7157,NULL),(29318,NULL,'The TP53 F113C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung adenocarcinoma and is a statistically significant hotspot (PMID: 1312896). In vivo studies with yeast expressing TP53 F113C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F113C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:04:21',NULL,NULL,NULL,NULL,'898ffa44-411c-4208-9881-91cd83542a76',7157,NULL),(29319,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:04:35',NULL,NULL,NULL,NULL,'ee003982-469f-495e-a566-0a1ed2e775dd',7157,NULL),(29320,NULL,'The TP53 F113S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 12242668). In vivo studies with yeast expressing TP53 F113S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F113S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:04:39',NULL,NULL,NULL,NULL,'be540e00-6fa8-4778-8330-4bd8b38851fa',7157,NULL),(29321,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:04:52',NULL,NULL,NULL,NULL,'51db914f-e8f9-40ad-b12f-6cfa0b8a6161',7157,NULL),(29322,NULL,'The TP53 F113V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in astrocytoma and is a statistically significant hotspot (PMID: 15188009). In vivo studies with yeast expressing TP53 F113V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F113V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:04:58',NULL,NULL,NULL,NULL,'ddc53f3e-c2cf-4780-9dba-6b85f4645a8d',7157,NULL),(29323,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:05:12',NULL,NULL,NULL,NULL,'1c2b0cfd-de90-4681-866f-1aaf090f1bae',7157,NULL),(29324,NULL,'The TP53 F113L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 16532037). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 F113L demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 F113L demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-23 19:17:50',NULL,NULL,NULL,NULL,'a3f6813b-7927-42fb-bb7c-4c4357892c64',7157,NULL),(29325,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:05:37',NULL,NULL,NULL,NULL,'c7c3a2f0-cf14-43c2-b32b-8c31ea9e51a6',7157,NULL),(29326,NULL,'The TP53 L111P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 16821082). In vivo studies with yeast expressing TP53 L111P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L111P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:05:39',NULL,NULL,NULL,NULL,'1d0c9077-7f90-4bd6-835f-109487f97226',7157,NULL),(29327,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:05:54',NULL,NULL,NULL,NULL,'e4c1984a-22dc-4617-b3ce-8eb50ceb0f1b',7157,NULL),(29328,NULL,'The TP53 L111Q mutation is located in the DNA-binding domain of the protein. This mutation has been identified in squamous cell carcinoma and is a statistically significant hotspot (PMID: 9400993). In vivo studies with yeast expressing TP53 L111Q demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L111Q also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:05:58',NULL,NULL,NULL,NULL,'a1e2aa3a-ed4d-44c4-bd94-43aeb7dd18df',7157,NULL),(29329,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:06:10',NULL,NULL,NULL,NULL,'4f86f8ad-0fad-443d-ba73-230d1ce1d673',7157,NULL),(29330,NULL,'The TP53 L111R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 14687797). In vivo studies with yeast expressing TP53 L111R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L111R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:06:13',NULL,NULL,NULL,NULL,'e19bed62-9066-4e78-bac6-8e4697fdba39',7157,NULL),(29331,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:06:28',NULL,NULL,NULL,NULL,'f57ddce8-5774-4eeb-86f7-7d65a26efb33',7157,NULL),(29332,NULL,'The TP53 L111M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 15126338). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 L111M demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 L111M demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:06:32',NULL,NULL,NULL,NULL,'1acb11dc-5b0d-4967-8274-65e77090aadc',7157,NULL),(29333,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:06:47',NULL,NULL,NULL,NULL,'41c18b1e-a95c-4ffd-b7e5-fd12b4276f77',7157,NULL),(29334,NULL,'The TP53 D259V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 12209975). In vivo studies with yeast expressing TP53 D259V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D259V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:06:51',NULL,NULL,NULL,NULL,'9a1c776f-f67e-4c06-a6d2-dbfae1f194e1',7157,NULL),(29335,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:07:04',NULL,NULL,NULL,NULL,'86bdad87-e2d6-420d-b990-b232db54968c',7157,NULL),(29336,NULL,'The TP53 D259H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 20824703). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D259H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D259H demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-23 19:25:29',NULL,NULL,NULL,NULL,'3fcd13c1-b177-47ac-b140-09b51c30a046',7157,NULL),(29337,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:07:22',NULL,NULL,NULL,NULL,'2b5ccef5-270b-40d6-b16e-6b842e252788',7157,NULL),(29338,NULL,'The TP53 D259Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 17171684). In vivo studies with yeast expressing TP53 D259Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D259Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:07:27',NULL,NULL,NULL,NULL,'1f3b2798-0751-446e-b403-7bea641feba1',7157,NULL),(29339,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:07:39',NULL,NULL,NULL,NULL,'48dffcf2-7720-4e11-a93a-0668328d4ca1',7157,NULL),(29340,NULL,'The TP53 D259N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 18348141). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D259N demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D259N demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:07:43',NULL,NULL,NULL,NULL,'9c7de2a0-2295-4d12-86cd-93b6c2680ae9',7157,NULL),(29341,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:07:58',NULL,NULL,NULL,NULL,'0c8ae8f6-6849-4f54-87bd-c13b122350f7',7157,NULL),(29342,NULL,'The TP53 I232T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in follicular lymphoma and is a statistically significant hotspot (PMID: 12217802). In vivo studies with yeast expressing TP53 I232T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I232T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:08:00',NULL,NULL,NULL,NULL,'0a8ce906-0365-45c9-8e6d-9b1bfbecae5d',7157,NULL),(29343,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:08:10',NULL,NULL,NULL,NULL,'7ed7f2b3-a71d-4b6d-ae97-c0de73aeba70',7157,NULL),(29344,NULL,'The TP53 I232N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in prostate cancer and is a statistically significant hotspot (PMID: 8688333). In vivo studies with yeast expressing TP53 I232N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I232N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:08:15',NULL,NULL,NULL,NULL,'894e6d0e-c12d-4b52-91a6-7a5402eb77f6',7157,NULL),(29345,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:08:27',NULL,NULL,NULL,NULL,'f863c871-c5eb-4f9b-9d9b-649844d13dfe',7157,NULL),(29346,NULL,'The TP53 C135F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Ewing sarcoma and is a statistically significant hotspot (PMID: 11406642). In vivo studies with yeast expressing TP53 C135F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C135F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:08:32',NULL,NULL,NULL,NULL,'1d4da7e4-9d7a-4b01-9a24-150c229faa64',7157,NULL),(29347,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:08:43',NULL,NULL,NULL,NULL,'f5ca4c1e-6162-4db3-87a4-dd7ea5b95114',7157,NULL),(29348,NULL,'The TP53 H214P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in astrocytoma and is a statistically significant hotspot (PMID: 7882284). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vitro studies with various human cancer cell lines expressing TP53 H214P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644). However, other in vivo studies with yeast expressing TP53 H214P demonstrated that the mutation is likely neutral as measured by wildtype comparable transactivational activity (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:08:47',NULL,NULL,NULL,NULL,'9587a5bc-bdc2-42d2-ab32-fc70968a1f60',7157,NULL),(29349,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:09:06',NULL,NULL,NULL,NULL,'de4a6cc4-2c4b-4a64-8b7d-165986495b24',7157,NULL),(29350,NULL,'The TP53 C135W mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 11309337). In vivo studies with yeast expressing TP53 C135W demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C135W also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:09:08',NULL,NULL,NULL,NULL,'9d137f25-ac6e-45b4-b284-922e467f7f2d',7157,NULL),(29351,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:09:26',NULL,NULL,NULL,NULL,'dfdf2cf0-70ff-4060-9667-dbfb2e4d35c3',7157,NULL),(29352,NULL,'The TP53 D281P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 24120142). In vitro studies with various human cancer cell lines expressing TP53 D281P demonstrated that the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:09:33',NULL,NULL,NULL,NULL,'8803add6-6bf7-4885-bc5e-bfce7dba782d',7157,NULL),(29353,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:09:49',NULL,NULL,NULL,NULL,'116fd63d-6a9b-4932-b425-ba040598c7fe',7157,NULL),(29354,NULL,'The TP53 P151T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 15316933). In vivo studies with yeast expressing TP53 P151T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P151T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:09:52',NULL,NULL,NULL,NULL,'a0ade61a-1273-4816-a789-880d7f066c74',7157,NULL),(29355,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:10:09',NULL,NULL,NULL,NULL,'8d949ed9-5cda-4cae-9fd4-7ff5ae1df088',7157,NULL),(29356,NULL,'The TP53 P151F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 14618621). In vitro studies with various human cancer cell lines expressing TP53 P151F demonstrated that the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:10:15',NULL,NULL,NULL,NULL,'20df2fcf-6008-461e-810a-31a563311b42',7157,NULL),(29357,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:10:33',NULL,NULL,NULL,NULL,'e631ad98-eabf-4a0b-9e5b-8318375004ff',7157,NULL),(29358,NULL,'The TP53 P151L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in NK/T-cell lymphoma and is a statistically significant hotspot (PMID: 15643509). In vivo studies with yeast expressing TP53 P151L demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:10:38',NULL,NULL,NULL,NULL,'e9429b6e-ada6-4662-a455-782d894ca67f',7157,NULL),(29359,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:10:47',NULL,NULL,NULL,NULL,'a3c5d9fd-1eef-4299-b589-f5461cc92094',7157,NULL),(29360,NULL,'The TP53 D281A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant peripheral nerve sheath tumors and is a statistically significant hotspot (PMID: 20010306). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D281A demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D281A demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-23 19:31:07',NULL,NULL,NULL,NULL,'cc9a3a96-431f-4a38-9bc7-0fee7434a13e',7157,NULL),(29361,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:13:13',NULL,NULL,NULL,NULL,'53651d7b-840a-4016-9144-99e21f93f8bc',7157,NULL),(29362,NULL,'The TP53 R175G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in acute T-cell lymphoblastic leukemia and is a statistically significant hotspot (PMID: 8486778). In vivo studies with yeast expressing TP53 R175G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R175G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:13:16',NULL,NULL,NULL,NULL,'cd3b7655-6ec8-4964-b518-73ee35f26f8a',7157,NULL),(29363,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:13:27',NULL,NULL,NULL,NULL,'6dc59b1e-6f96-4430-959b-30c233e27f03',7157,NULL),(29364,NULL,'The TP53 D281E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 11244334). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D281E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D281E demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:13:31',NULL,NULL,NULL,NULL,'fa55640a-c5bd-4c2b-92ec-b68dc144add7',7157,NULL),(29365,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:16:03',NULL,NULL,NULL,NULL,'63f0a0ec-88dd-4f9d-a584-5e4f23b4e6cf',7157,NULL),(29366,NULL,'The TP53 D281V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in Ewing sarcoma and is a statistically significant hotspot (PMID: 8168993). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D281V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D281V demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:16:08',NULL,NULL,NULL,NULL,'525a5306-5319-44cd-bf21-fa6ad17797cf',7157,NULL),(29367,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:16:17',NULL,NULL,NULL,NULL,'6a9964a4-e846-49f2-920b-240a00cbf1a7',7157,NULL),(29368,NULL,'The TP53 D281G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 8878455). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D281G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D281G demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-23 19:33:28',NULL,NULL,NULL,NULL,'86167b42-0d99-4be1-a2cd-9e7e4b8ac132',7157,NULL),(29369,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:16:34',NULL,NULL,NULL,NULL,'10a87fe3-21e4-49fd-8dc5-93a58a1616c6',7157,NULL),(29370,NULL,'The TP53 D281H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in clear cell renal cell carcinoma and is a statistically significant hotspot (PMID: 20137853). In vivo studies with yeast expressing TP53 D281H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D281H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:16:39',NULL,NULL,NULL,NULL,'cf731ee9-f5cc-4e6e-b5f9-5b52f12dda77',7157,NULL),(29371,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:20:49',NULL,NULL,NULL,NULL,'733caa1a-063f-4413-a5ba-698334247f7d',7157,NULL),(29372,NULL,'The TP53 D281Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in follicular lymphoma and is a statistically significant hotspot (PMID: 10854221). In vivo studies with yeast expressing TP53 D281Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D281Y also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:20:54',NULL,NULL,NULL,NULL,'33f1ff16-6118-4758-9a34-340d3692efc4',7157,NULL),(29373,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:21:03',NULL,NULL,NULL,NULL,'b3f62b84-50a5-4d24-83d6-39d8d5915f47',7157,NULL),(29374,NULL,'The TP53 D281N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant lymphoma and is a statistically significant hotspot (PMID: 8916968). In vivo studies with yeast expressing TP53 D281N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D281N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:21:09',NULL,NULL,NULL,NULL,'18ba68ee-ff40-43c2-b093-2e4c246c12de',7157,NULL),(29375,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:21:18',NULL,NULL,NULL,NULL,'01ba08ab-524c-4a73-8c73-d80f61955589',7157,NULL),(29376,NULL,'The TP53 V216E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 7928628). In vivo studies with yeast expressing TP53 V216E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V216E also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:21:22',NULL,NULL,NULL,NULL,'6bf2ce6b-961c-4e39-aca0-65a91c1a943f',7157,NULL),(29377,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:23:02',NULL,NULL,NULL,NULL,'b3e29b9d-841c-4ebb-83f7-3ec8c72799f0',7157,NULL),(29378,NULL,'The TP53 V216G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 16459017). In vivo studies with yeast expressing TP53 V216G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V216G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:23:04',NULL,NULL,NULL,NULL,'98d87d75-6f48-48b7-8b48-3aef97375f1c',7157,NULL),(29379,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:30:30',NULL,NULL,NULL,NULL,'220d411c-edad-48f1-a8cb-c658deee8688',7157,NULL),(29380,NULL,'The TP53 V216L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 9516972). In vivo studies with yeast expressing TP53 V216L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V216L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:30:34',NULL,NULL,NULL,NULL,'16c51556-6cf3-41fa-b0df-c8723320c0e9',7157,NULL),(29381,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:31:55',NULL,NULL,NULL,NULL,'0b8dee69-81ac-4481-b462-a101899e3bca',7157,NULL),(29382,NULL,'The TP53 V216M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 22551440). In vivo studies with yeast expressing TP53 V216M demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V216M also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:31:57',NULL,NULL,NULL,NULL,'f02226ef-970e-4c1d-8aae-2cd440c2dbc3',7157,NULL),(29383,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:32:04',NULL,NULL,NULL,NULL,'630511d2-1995-409f-8cfa-248f2583eaa1',7157,NULL),(29384,NULL,'The TP53 Y126C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 15188009). In vivo studies with yeast expressing TP53 Y126C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y126C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:32:09',NULL,NULL,NULL,NULL,'c6a32ee8-65e0-4724-9b87-142235e14b9d',7157,NULL),(29385,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:32:16',NULL,NULL,NULL,NULL,'aed17154-7dfe-46e1-a0bc-d826f14801fc',7157,NULL),(29386,NULL,'The TP53 Y126S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in gallbladder cancer and is a statistically significant hotspot (PMID: 15154647). In vivo studies with yeast expressing TP53 Y126S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y126S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:32:22',NULL,NULL,NULL,NULL,'9b55570d-3118-46cb-8b0f-0a75928e06b4',7157,NULL),(29387,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:35:27',NULL,NULL,NULL,NULL,'2c3cd489-28ad-4a8b-9326-f91333ef6353',7157,NULL),(29388,NULL,'The TP53 Y126D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 10896202). In vivo studies with yeast expressing TP53 Y126D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y126D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:35:32',NULL,NULL,NULL,NULL,'1e272251-a388-4c3e-b6fa-8fc02cf2014e',7157,NULL),(29389,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:39:50',NULL,NULL,NULL,NULL,'6009dab8-5c6e-4376-8cfe-285907db9069',7157,NULL),(29390,NULL,'The TP53 Y126H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11307149). In vivo studies with yeast expressing TP53 Y126H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y126H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:39:55',NULL,NULL,NULL,NULL,'ebdda3ba-c34b-4b28-89b8-25edcc0ebad5',7157,NULL),(29391,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:40:03',NULL,NULL,NULL,NULL,'dc8958c2-3a4a-461a-8446-d92fe988a5f7',7157,NULL),(29392,NULL,'The TP53 Y126N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 16818615). In vivo studies with yeast expressing TP53 Y126N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y126N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:40:08',NULL,NULL,NULL,NULL,'5ab2fe4b-718d-449e-a24e-ad062c47e9bc',7157,NULL),(29393,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:46:54',NULL,NULL,NULL,NULL,'580e70f5-bb71-428d-90e6-976efc38e60d',7157,NULL),(29394,NULL,'The TP53 A159P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 10023676). In vivo studies with yeast expressing TP53 A159P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 A159P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:46:58',NULL,NULL,NULL,NULL,'26f46a83-e5a7-4df7-aae6-91e2c91c74b5',7157,NULL),(29395,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-02 16:54:11',NULL,NULL,NULL,NULL,'914c65f6-e451-40cb-aff4-db054a82c131',7157,NULL),(29396,NULL,'The TP53 A159D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 7651727). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 A159D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 A159D demonstrated that the mutation is likely neutral as measured by wildtype comparable growth suppression activity (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-02 16:54:15',NULL,NULL,NULL,NULL,'fd759c19-416c-41db-ba73-9d0efc7186b2',7157,NULL),(29397,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:54:24',NULL,NULL,NULL,NULL,'719fc285-6202-40d3-9d8c-f494199263cf',7157,NULL),(29398,NULL,'The TP53 A159V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in astrocytoma and is a statistically significant hotspot (PMID: 8285583). In vivo studies with yeast expressing TP53 A159V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 A159V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:54:28',NULL,NULL,NULL,NULL,'885cbe35-bb5d-40c9-8a09-0d0ebc59f0ef',7157,NULL),(29399,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:56:05',NULL,NULL,NULL,NULL,'5a5f7829-5ea3-44b9-a5ec-7753464e4ab0',7157,NULL),(29400,NULL,'The TP53 L194P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioma and is a statistically significant hotspot (PMID: 1358438). In vivo studies with yeast expressing TP53 L194P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L194P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:56:07',NULL,NULL,NULL,NULL,'968b2bf2-c31a-43c3-a5f6-188c03bee3b4',7157,NULL),(29401,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-02 16:58:27',NULL,NULL,NULL,NULL,'51035597-bc2b-4ddd-9b3d-20e64f1436b5',7157,NULL),(29402,NULL,'The TP53 L194R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 15221786). In vivo studies with yeast expressing TP53 L194R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L194R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-02 16:58:32',NULL,NULL,NULL,NULL,'f2343ddd-085a-464b-976b-0b7e7a87916e',7157,NULL),(29403,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:19:43',NULL,NULL,NULL,NULL,'8b579f75-5877-4d08-a810-9c21a7999717',7157,NULL),(29404,NULL,'The TP53 L194F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 7619219). In vivo studies with yeast expressing TP53 L194F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L194F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:19:47',NULL,NULL,NULL,NULL,'7badfaf4-2bb6-4bfe-8455-13d6256394e8',7157,NULL),(29405,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:23:58',NULL,NULL,NULL,NULL,'c323b2b7-ba86-4e3c-b3cd-93c643838107',7157,NULL),(29406,NULL,'The TP53 L194H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in marginal zone B-cell lymphoma and is a statistically significant hotspot (PMID: 8541549). In vivo studies with yeast expressing TP53 L194H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 L194H also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:24:02',NULL,NULL,NULL,NULL,'14492913-e6e0-46cf-a8eb-25dce4e182f3',7157,NULL),(29407,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:24:13',NULL,NULL,NULL,NULL,'8236ac91-1bec-4e79-b09c-458c39bccedd',7157,NULL),(29408,NULL,'The TP53 F270S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 11588905). In vivo studies with yeast expressing TP53 F270S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F270S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:24:18',NULL,NULL,NULL,NULL,'bf5fca38-1f8f-4146-b678-23e83faed9af',7157,NULL),(29409,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:24:29',NULL,NULL,NULL,NULL,'4980d373-162a-43f7-a213-4a700a7a0cde',7157,NULL),(29410,NULL,'The TP53 F270C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 10830574). In vivo studies with yeast expressing TP53 F270C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F270C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:24:33',NULL,NULL,NULL,NULL,'43308441-8249-4edd-8e64-0237d15cd80f',7157,NULL),(29411,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:24:43',NULL,NULL,NULL,NULL,'5f49c760-8c12-47e0-bd43-ffc1069f5779',7157,NULL),(29412,NULL,'The TP53 F270V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 8731915). In vivo studies with yeast expressing TP53 F270V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F270V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:24:47',NULL,NULL,NULL,NULL,'69277319-51a5-4773-a5a9-76a30098759c',7157,NULL),(29413,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:24:55',NULL,NULL,NULL,NULL,'d8bf0ae8-6c4a-454f-be8f-6c06977b1c12',7157,NULL),(29414,NULL,'The TP53 F270I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 1551126). In vivo studies with yeast expressing TP53 F270I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F270I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:25:00',NULL,NULL,NULL,NULL,'36a6e20f-5ad9-49ac-a862-5054d0792c5e',7157,NULL),(29415,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:37:17',NULL,NULL,NULL,NULL,'1726fb77-736a-4d91-b73d-91572e09a92f',7157,NULL),(29416,NULL,'The TP53 F270L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in gallbladder cancer and is a statistically significant hotspot (PMID: 9376195). In vivo studies with yeast expressing TP53 F270L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:37:22',NULL,NULL,NULL,NULL,'97e3757d-94f5-41d5-b441-d0bd0de04bd3',7157,NULL),(29417,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:37:33',NULL,NULL,NULL,NULL,'785c6cd2-36dd-48fb-bc8e-a3dcbf55c140',7157,NULL),(29418,NULL,'The TP53 M246R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 11929815). In vivo studies with yeast expressing TP53 M246R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M246R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:37:37',NULL,NULL,NULL,NULL,'b3f763db-313c-4b38-ad21-96b245214c43',7157,NULL),(29419,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:37:48',NULL,NULL,NULL,NULL,'12a88a53-5a23-471f-9de0-ee16215cfc8b',7157,NULL),(29420,NULL,'The TP53 M246T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 9699530). In vivo studies with yeast expressing TP53 M246T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M246T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:37:49',NULL,NULL,NULL,NULL,'4b463a40-3237-4756-8584-3011015569c9',7157,NULL),(29421,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:42:28',NULL,NULL,NULL,NULL,'05c84ffd-8dac-41a1-8a8f-f883f4b2ed3d',7157,NULL),(29422,NULL,'The TP53 M246V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in hepatocellular carcinoma and is a statistically significant hotspot (PMID: 18567082). In vivo studies with yeast expressing TP53 M246V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M246V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 12:42:33',NULL,NULL,NULL,NULL,'635dba92-d445-493e-8dfa-80fa1f21b0d8',7157,NULL),(29423,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 12:49:58',NULL,NULL,NULL,NULL,'3d6191ce-50df-4bfc-8d0c-50ee9700ecb1',7157,NULL),(29424,NULL,'The TP53 M246I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 10761705). In vivo studies with yeast expressing TP53 M246I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 F270I also demonstrated the mutation is inactivating as measured by moderately reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-23 19:59:17',NULL,NULL,NULL,NULL,'053d5f8d-ada8-41df-82c6-88d4100ac1c0',7157,NULL),(29425,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:27:53',NULL,NULL,NULL,NULL,'d964e438-f0d2-4077-ac92-accc6102c4d2',7157,NULL),(29426,NULL,'The TP53 M246K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in larynx squamous cell carcinoma and is a statistically significant hotspot (PMID: 8001261). In vivo studies with yeast expressing TP53 M246K demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M246K also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:27:58',NULL,NULL,NULL,NULL,'fba619a7-30dd-4295-89dc-e7da6cf903a5',7157,NULL),(29427,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:28:07',NULL,NULL,NULL,NULL,'6e88791d-9f99-449f-ae8f-ba9116a85eb6',7157,NULL),(29428,NULL,'The TP53 M246L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 11304577). In vivo studies with yeast expressing TP53 M246L demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 M246L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:28:10',NULL,NULL,NULL,NULL,'9505a0ea-defc-463e-b372-6370f42ac2b4',7157,NULL),(29429,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:28:20',NULL,NULL,NULL,NULL,'6b33e8bd-ef32-4478-8ead-67723848fcbd',7157,NULL),(29430,NULL,'The TP53 S127P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 9569050). In vivo studies with yeast expressing TP53 S127P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S127P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:28:22',NULL,NULL,NULL,NULL,'e13428cf-9c32-4bf0-91ea-8d57e6ea789b',7157,NULL),(29431,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:28:32',NULL,NULL,NULL,NULL,'6acb9924-3548-46db-aa26-4c8a9066cce0',7157,NULL),(29432,NULL,'The TP53 S127C mutation is located in the DNA-binding domain of the protein. This mutation has been identified in acute myeloid leukemia and is a statistically significant hotspot (PMID: 16467864). In vivo studies with yeast expressing TP53 S127C demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S127C also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:28:34',NULL,NULL,NULL,NULL,'6e5e00c7-dc0b-4bbf-86ae-98e738b9fcac',7157,NULL),(29433,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:28:46',NULL,NULL,NULL,NULL,'cd6b9c85-b853-45ba-970e-c83c13a3c98e',7157,NULL),(29434,NULL,'The TP53 S127T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 18617062). In vivo studies with yeast expressing TP53 S127T demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S127T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:28:51',NULL,NULL,NULL,NULL,'939d682a-d260-4ecb-bb7e-6ce9fbabc73c',7157,NULL),(29435,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:29:33',NULL,NULL,NULL,NULL,'1c192e0b-5633-45e0-8fb9-c76cf7886b2d',7157,NULL),(29436,NULL,'The TP53 S127F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in xeroderma pigmentosum and is a statistically significant hotspot (PMID: 9766670). In vivo studies with yeast expressing TP53 S127F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 S127F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:29:37',NULL,NULL,NULL,NULL,'ba1c9947-5158-42e9-b15b-0ad21debfaf7',7157,NULL),(29437,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:29:47',NULL,NULL,NULL,NULL,'fe3315d5-5278-4ee9-bfd6-821913741eb0',7157,NULL),(29438,NULL,'The TP53 S127Y mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 10667596). In vivo studies with yeast expressing TP53 S127Y demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:29:51',NULL,NULL,NULL,NULL,'8b471e89-8a67-4ec7-999c-59b3686d332c',7157,NULL),(29439,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:51:18',NULL,NULL,NULL,NULL,'ce1b2e56-9642-49fb-bf6d-9a4cd7caa7e4',7157,NULL),(29440,NULL,'The TP53 Q136P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 14678961). In vivo studies with yeast expressing TP53 Q136P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Q136P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:51:22',NULL,NULL,NULL,NULL,'b995a5d3-48a2-45f5-acb9-64758d7c5c1e',7157,NULL),(29441,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:51:31',NULL,NULL,NULL,NULL,'74381502-85c4-4685-af73-405c6792142b',7157,NULL),(29442,NULL,'The TP53 Q136E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bronchioloalveolar carcinoma and is a statistically significant hotspot (PMID: 9614374). In vivo studies with yeast expressing TP53 Q136E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Q136E also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:51:33',NULL,NULL,NULL,NULL,'7800cf4c-df3c-49d2-827c-69cdaf849ce5',7157,NULL),(29443,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 13:51:44',NULL,NULL,NULL,NULL,'0bb58b04-87fc-4da6-a565-d591ae461827',7157,NULL),(29444,NULL,'The TP53 Q136H mutation is located in the DNA-binding domain of the protein. This mutation has been identified in acute lymphoblastic B-cell leukemia and is a statistically significant hotspot (PMID: 20575032). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 Q136H demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 Q136H demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:24:20',NULL,NULL,NULL,NULL,'5f377e67-dfe4-4937-b523-6d553a466b0c',7157,NULL),(29445,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 13:56:21',NULL,NULL,NULL,NULL,'991be246-c755-47ab-88a3-5434819b595d',7157,NULL),(29446,NULL,'The TP53 Q136L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in urothelial cancer and is a statistically significant hotspot (PMID: 20575032). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 Q136L demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 Q136L demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:24:48',NULL,NULL,NULL,NULL,'15963a90-50fe-43cb-bc22-d967f9592014',7157,NULL),(29447,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:56:35',NULL,NULL,NULL,NULL,'45e0259c-c370-453d-9674-46cf06cb9f9b',7157,NULL),(29448,NULL,'The TP53 T125P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 15073856). In vivo studies with yeast expressing TP53 T125P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 T125P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:56:42',NULL,NULL,NULL,NULL,'63d98a7c-a766-4cd0-ac39-2843e3dfc383',7157,NULL),(29449,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:56:49',NULL,NULL,NULL,NULL,'a51ec5d9-02c0-40d9-9f92-5db8b7b0b298',7157,NULL),(29450,NULL,'The TP53 T125A mutation is located in the DNA-binding domain of the protein. This mutation has been identified in NK/T-cell lymphoma and is a statistically significant hotspot (PMID: 15643509). In vivo studies with yeast expressing TP53 T125A demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:56:54',NULL,NULL,NULL,NULL,'adb3705e-d606-41f5-8fed-9cf58df66a59',7157,NULL),(29451,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:57:03',NULL,NULL,NULL,NULL,'1540fa71-b49d-448c-8be4-23dd1156a546',7157,NULL),(29452,NULL,'The TP53 T125R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 11229518). In vivo studies with yeast expressing TP53 T125R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 T125R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:57:10',NULL,NULL,NULL,NULL,'5d53b278-5c6a-4ca8-b75e-8cabb2f0c8a4',7157,NULL),(29453,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 13:57:19',NULL,NULL,NULL,NULL,'52fb3702-48c8-4df5-b125-5a309a923fc5',7157,NULL),(29454,NULL,'The TP53 T125K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in pancreatic cancer and is a statistically significant hotspot (PMID: 18772397). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 T125K demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 T125K demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:25:51',NULL,NULL,NULL,NULL,'bf09b51f-e32d-42cb-9374-3f0d9f6a4aa2',7157,NULL),(29455,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 13:57:36',NULL,NULL,NULL,NULL,'cd59f350-e019-457f-812d-1e04529ab432',7157,NULL),(29456,NULL,'The TP53 T125M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in sinonasal NK/T-cell lymphoma and is a statistically significant hotspot (PMID: 15643509). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 T125M demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 T125M demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:26:19',NULL,NULL,NULL,NULL,'bdfd89d3-5771-476b-ad05-43b0f18318b8',7157,NULL),(29457,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:57:54',NULL,NULL,NULL,NULL,'d0fb5975-5239-4032-bc31-7504d0bf1e4e',7157,NULL),(29458,NULL,'The TP53 E271K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 15538112). In vivo studies with yeast expressing TP53 E271K demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 E271K also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:57:56',NULL,NULL,NULL,NULL,'b62ab704-2888-4338-93ea-f13b49e3c2c0',7157,NULL),(29459,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 13:59:43',NULL,NULL,NULL,NULL,'4c88bf6a-0ded-40cc-9ff5-3488a22bd100',7157,NULL),(29460,NULL,'The TP53 V172F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 10216487). In vivo studies with yeast expressing TP53 V172F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 13:59:47',NULL,NULL,NULL,NULL,'3decd904-eff8-49eb-885b-b01dc7eff6ad',7157,NULL),(29461,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:03:16',NULL,NULL,NULL,NULL,'f67e9632-6c59-465c-a96b-a58020d331a6',7157,NULL),(29462,NULL,'The TP53 T155P mutation is located in the DNA-binding domain of the protein. This mutation has been identified in small-cell lung cancer and is a statistically significant hotspot (PMID: 14660794). In vivo studies with yeast expressing TP53 T155P demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 T155P also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-24 12:29:14',NULL,NULL,NULL,NULL,'e6c1bf80-6bcc-4409-aa85-74fcb415a8f9',7157,NULL),(29463,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:03:25',NULL,NULL,NULL,NULL,'626a49dc-2691-4d79-9a07-fbfaf3e4a4d7',7157,NULL),(29464,NULL,'The TP53 T155N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 11275993). In vivo studies with yeast expressing TP53 T155N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:03:29',NULL,NULL,NULL,NULL,'177fb046-82ed-43e1-c0ae-3518de4c2414',7157,NULL),(29465,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:03:38',NULL,NULL,NULL,NULL,'e4326aa3-87d9-4d17-8af0-4176d7cd8fc3',7157,NULL),(29466,NULL,'The TP53 N131S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 16271069). In vivo studies with yeast expressing TP53 N131S demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:03:42',NULL,NULL,NULL,NULL,'60e562fd-d877-461d-869b-e740034061c0',7157,NULL),(29467,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:06:00',NULL,NULL,NULL,NULL,'7e143a9e-f655-43c8-a24e-516d6485122c',7157,NULL),(29468,NULL,'The TP53 F113P mutation is located in the DNA-binding domain of the protein and is a statistically significant hotspot. In vitro studies with various human cancer cell lines expressing TP53 F113P demonstrated that the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:06:05',NULL,NULL,NULL,NULL,'c883a618-2c0c-42b3-aeff-355c40bf37a5',7157,NULL),(29469,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 14:06:12',NULL,NULL,NULL,NULL,'42a6a062-57d4-40ea-92f3-0b0c704b7bbd',7157,NULL),(29470,NULL,'The TP53 D208E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 8156519). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 D208E demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 D208E demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:30:15',NULL,NULL,NULL,NULL,'04520f9d-a519-4aa3-b4f3-4f76a209545a',7157,NULL),(29471,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:06:31',NULL,NULL,NULL,NULL,'8831cd7b-c361-4471-ae40-7cf40380c0fc',7157,NULL),(29472,NULL,'The TP53 D208V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 8099903). In vivo studies with yeast expressing TP53 D208V demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D208V also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:06:33',NULL,NULL,NULL,NULL,'4fd2507a-a9db-4bbd-8fd5-2dc729b2c4a4',7157,NULL),(29473,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-03 14:09:44',NULL,NULL,NULL,NULL,'4ef8ed91-a16d-4a1d-9ba6-269555c01d4f',7157,NULL),(29474,NULL,'The TP53 D208G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 9516972). In vivo studies with yeast expressing TP53 D208G demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-02-24 12:30:52',NULL,NULL,NULL,NULL,'c6e71f6f-a9f7-4ca3-acc8-371557aa109b',7157,NULL),(29475,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-03 14:10:03',NULL,NULL,NULL,NULL,'defb094a-e6e2-4519-8910-323a17da1f35',7157,NULL),(29476,NULL,'The TP53 D208N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in thyroid papillary carcinoma and is a statistically significant hotspot (PMID: 8768871). In vivo studies with yeast expressing TP53 D208N demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 D208N also demonstrated the mutation likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-02-24 12:33:03',NULL,NULL,NULL,NULL,'557f4470-8fc5-4b4d-908a-65e3ab5e67b2',7157,NULL),(29477,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:10:19',NULL,NULL,NULL,NULL,'1bb7a6af-382d-4b6f-aa5c-6d1c9a081d4e',7157,NULL),(29478,NULL,'The TP53 N247I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in non-small cell lung cancer and is a statistically significant hotspot (PMID: 10023676). In vivo studies with yeast expressing TP53 N247I demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 N247I also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:10:23',NULL,NULL,NULL,NULL,'1a108a3e-dd3d-44b5-ad90-ffb307bd499e',7157,NULL),(29479,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:10:32',NULL,NULL,NULL,NULL,'6ebd857d-066a-4469-9544-ee08576473a6',7157,NULL),(29480,NULL,'The TP53 Y107D mutation is located in the DNA-binding domain of the protein. This mutation has been identified in bladder cancer and is a statistically significant hotspot (PMID: 16061860). In vivo studies with yeast expressing TP53 Y107D demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 Y107D also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:10:37',NULL,NULL,NULL,NULL,'65bf9dac-bd1f-4a55-90ad-3ad2859152fd',7157,NULL),(29481,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:12:58',NULL,NULL,NULL,NULL,'d0285f72-66b1-4b52-ae57-fed4512188d2',7157,NULL),(29482,NULL,'The TP53 I232S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 1617667). In vivo studies with yeast expressing TP53 I232S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I232S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:13:02',NULL,NULL,NULL,NULL,'8f77f06a-8bd0-40d6-82c7-f63b292fb735',7157,NULL),(29483,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:15:22',NULL,NULL,NULL,NULL,'cc45b460-9570-4cf7-910e-20f97e7cebdb',7157,NULL),(29484,NULL,'The TP53 I232F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 10753186). In vivo studies with yeast expressing TP53 I232F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:15:27',NULL,NULL,NULL,NULL,'d8939e03-2292-4989-8507-f52959238c38',7157,NULL),(29485,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:15:35',NULL,NULL,NULL,NULL,'fd374789-14a3-437e-ab4f-98a87de04109',7157,NULL),(29486,NULL,'The TP53 P190R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 16337994). In vivo studies with yeast expressing TP53 P190R demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P190R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:15:39',NULL,NULL,NULL,NULL,'647e0a17-ec15-427c-9cbe-0481b8cea498',7157,NULL),(29487,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 14:15:47',NULL,NULL,NULL,NULL,'ba64a42c-128a-4c7a-a27c-7c8f4c0c7265',7157,NULL),(29488,NULL,'The TP53 P190S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 17692090). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 P190S demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 P190S demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:34:45',NULL,NULL,NULL,NULL,'db419730-f891-4cfe-95e3-b292baed66ae',7157,NULL),(29489,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:23:52',NULL,NULL,NULL,NULL,'c86b790d-e7f0-4cd4-8e53-b4d7b2242b51',7157,NULL),(29490,NULL,'The TP53 P190T mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 12713560). In vivo studies with yeast expressing TP53 P190T demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P190T also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:23:58',NULL,NULL,NULL,NULL,'19b1f4cc-c835-49cc-820b-d63ae5c94bff',7157,NULL),(29491,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:24:05',NULL,NULL,NULL,NULL,'8237bb02-95db-4afc-b3df-bdaf68c1bfdb',7157,NULL),(29492,NULL,'The TP53 P190L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 12124823). In vivo studies with yeast expressing TP53 P190L demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P190L also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:24:09',NULL,NULL,NULL,NULL,'ac798566-2677-488d-a824-29f91c3ae87a',7157,NULL),(29493,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:24:17',NULL,NULL,NULL,NULL,'18180fbe-e0b2-4c39-824d-78dc766fd4c9',7157,NULL),(29494,NULL,'The TP53 V218E mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 9569050). In vivo studies with yeast expressing TP53 V218E demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V218E also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:24:21',NULL,NULL,NULL,NULL,'c51ab0c9-9b4d-4d3b-b015-40f66ef0b308',7157,NULL),(29495,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-03 14:24:30',NULL,NULL,NULL,NULL,'933dbd39-4919-4e88-a210-3cea240dde03',7157,NULL),(29496,NULL,'The TP53 V218G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in malignant melanoma and is a statistically significant hotspot (PMID: 8826861). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vitro studies with various human cancer cell lines expressing TP53 V218G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644). However, other in vivo studies with yeast expressing TP53 V218G demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:35:40',NULL,NULL,NULL,NULL,'f8508ea9-8b21-4599-9157-be8835c2ee62',7157,NULL),(29497,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-02-03 14:30:11',NULL,NULL,NULL,NULL,'0d0866f9-dab4-4f3c-a45a-4a5d02a52bef',7157,NULL),(29498,NULL,'The TP53 V218L mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer and is a statistically significant hotspot (PMID: 8556707). In vivo studies with yeast expressing TP53 V218L demonstrated that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 V218L also demonstrated the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2023-02-24 12:36:29',NULL,NULL,NULL,NULL,'51dcc022-b757-4ad0-8170-bafdd3bd2492',7157,NULL),(29499,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-03 14:30:39',NULL,NULL,NULL,NULL,'254c91ed-cf33-4bb2-a416-92bb0a057757',7157,NULL),(29500,NULL,'The TP53 V218M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in papillary serous carcinoma and is a statistically significant hotspot (PMID: 10728699). In vivo studies with yeast expressing TP53 V218M demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-03 14:30:44',NULL,NULL,NULL,NULL,'775ab59c-b3a0-4d5d-a6c8-9983acb18cf6',7157,NULL),(29501,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-24 12:37:18',NULL,NULL,NULL,NULL,'a6f49b52-a289-432c-9efe-da0a75438529',7157,NULL),(29502,NULL,'The TP53 N268_R273del mutation is located in the DNA-binding domain of the protein and is a statistically significant hotspot. Although not functionally characterized per se, an advanced non-small cell lung cancer patient harboring both TP53 N268_R273del and EGFR D770_N771insG treated with poziotinib demonstrated sensitivity and later acquired CDK6 amplification as a secondary resistance mutation (PMID: 35820397).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-03 15:22:08',NULL,NULL,NULL,NULL,'e28f77db-5897-4413-8593-f665b8309830',7157,NULL),(29503,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-06 14:38:04',NULL,NULL,NULL,NULL,'57036612-9792-4ca9-a5f0-eb74c77026bc',7157,NULL),(29504,NULL,'The TP53 I195S mutation is located in the DNA-binding domain of the protein. This mutation has been identified in esophageal squamous cell carcinoma and is a statistically significant hotspot (PMID: 9133458). In vivo studies with yeast expressing TP53 I195S demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I195S also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-06 14:38:08',NULL,NULL,NULL,NULL,'3dd7fb97-061a-421b-89f1-341009371c7b',7157,NULL),(29505,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-06 14:38:26',NULL,NULL,NULL,NULL,'8d5534d3-2067-4b77-a8fd-941c57549700',7157,NULL),(29506,NULL,'The TP53 I195F mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 22460905). In vivo studies with yeast expressing TP53 I195F demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I195F also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-06 14:38:30',NULL,NULL,NULL,NULL,'9ae310a7-f0a9-4645-bfeb-63c6dbd9ae3f',7157,NULL),(29507,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-06 14:38:43',NULL,NULL,NULL,NULL,'6098083f-be56-461b-99ee-3357a5f4860e',7157,NULL),(29508,NULL,'The TP53 I195K mutation is located in the DNA-binding domain of the protein. This mutation has been identified in lung adenocarcinoma and is a statistically significant hotspot (PMID: 28481359). In vitro studies with various human cancer cell lines expressing TP53 I195K demonstrate that the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-06 14:38:48',NULL,NULL,NULL,NULL,'662d0b85-cc03-428c-bd8a-d6bd90ce229c',7157,NULL),(29509,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Inconclusive','2023-02-06 14:38:56',NULL,NULL,NULL,NULL,'f7e117c6-292e-4a55-8fd2-53d6f2e95813',7157,NULL),(29510,NULL,'The TP53 I195M mutation is located in the DNA-binding domain of the protein. This mutation has been identified in plasma cell myeloma and is a statistically significant hotspot (PMID: 25302557). There is conflicting evidence about the biological and oncogenic effect of the mutation. In vivo studies with yeast expressing TP53 I195M demonstrated that the mutation is inactivating as measured by the partial loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). However, other in vitro studies with various human cancer cell lines expressing TP53 I195M demonstrated that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-02-24 12:38:32',NULL,NULL,NULL,NULL,'0ccd7729-e3c8-494c-b2cd-5c371b007db3',7157,NULL),(29511,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-06 14:39:10',NULL,NULL,NULL,NULL,'469be8ee-8d28-4e39-9bdc-3a4ebf9437a7',7157,NULL),(29512,NULL,'The TP53 I195N mutation is located in the DNA-binding domain of the protein. This mutation has been identified in ovarian cancer and is a statistically significant hotspot (PMID: 16337994). In vivo studies with yeast expressing TP53 I195N demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 I195N also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-06 14:39:14',NULL,NULL,NULL,NULL,'5d209b32-8275-406f-9a3f-8741f5b3eb47',7157,NULL),(29513,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-06 14:39:26',NULL,NULL,NULL,NULL,'cd08472f-d535-4a87-b98a-61f6677ea879',7157,NULL),(29514,NULL,'The TP53 C135G mutation is located in the DNA-binding domain of the protein. This mutation has been identified in breast cancer and is a statistically significant hotspot (PMID: 10690522). In vivo studies with yeast expressing TP53 C135G demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 C135G also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-06 14:39:31',NULL,NULL,NULL,NULL,'19b005cb-ad9d-4f99-87b4-30e729d37307',7157,NULL),(29515,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2023-02-06 14:39:41',NULL,NULL,NULL,NULL,'9bd83fdd-ab5a-4052-b223-d69cd2ff4957',7157,NULL),(29516,NULL,'The TP53 P151R mutation is located in the DNA-binding domain of the protein. This mutation has been identified in glioblastoma and is a statistically significant hotspot (PMID: 15466178). In vivo studies with yeast expressing TP53 P151R demonstrated that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 P151R also demonstrated the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2023-02-06 14:39:46',NULL,NULL,NULL,NULL,'35a34952-6e05-46e4-aec7-3f000d0303a5',7157,NULL),(29517,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2023-07-11 18:18:44',NULL,NULL,NULL,NULL,'5da991e0-79f8-4a2a-a55c-953956ee76ca',7157,NULL),(29518,NULL,'The TP53 G360V mutation is located in the CARM1, HIPK1, HIPK2 and USP7 interacting regions of the protein. This mutation has been identified in biliary tract cancer (PMID: 34810298). There is conflicting evidence about the biological effect of this mutation. In vitro studies with H1299 cells expressing TP53 G360V demonstrate that the mutation is likely neutral as measured by colony growth comparable to wildtype (PMID: 33257846). Other in vitro studies with various human cancer cell lines expressing TP53 G360V also demonstrate that the mutation is likely neutral as measured by growth suppression activity comparable to wildtype (PMID: 29979965, 30224644). In vivo studies with yeast expressing TP53 G360V demonstrate that the mutation is likely neutral as measured by transactivational activity comparable to wildtype (PMID: 27328919, 12826609). In contrast, in vitro studies with H1299 cells expressing TP53 G360V demonstrate that the mutation is activating as measured by increased transactivational activity, increased apoptosis and decreased colony growth compared to wildtype (PMID: 27297285).','MUTATION_EFFECT',NULL,_binary '\0','Inconclusive','2023-07-11 18:18:53',NULL,NULL,NULL,NULL,'b3b7f8fa-2917-468e-b3f9-d14d3b0cee6c',7157,NULL),(29519,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-09-13 17:32:31',NULL,NULL,NULL,NULL,'34869853-1be8-4adf-840a-42d49c9cdc5a',7157,NULL),(29520,NULL,'The DNA-binding domain (DBD) of TP53 allows for contact with target DNA sequences to transactivate downstream genes. Mutations in the DBD result in conformational changes of the protein, altering contact of TP53 with its target DNA sequences and thereby altering transcriptional function (PMID: 8023157, 11900253). Given that TP53 directs the transcription of proteins that enable apoptosis, the inactivation of TP53 results in cells harboring damaged DNA and overall genomic instability (PMID: 11900253, 11900253).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-09-13 17:32:34',NULL,NULL,NULL,NULL,'5f74dc05-e878-4499-aa0d-f1b933bc7684',7157,NULL),(29521,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-09-13 17:33:16',NULL,NULL,NULL,NULL,'7fc66870-e2bd-4ac1-9668-98540bb912bf',7157,NULL),(29522,NULL,'The DNA-binding domain (DBD) of TP53 allows for contact with target DNA sequences to transactivate downstream genes. Mutations in the DBD result in conformational changes of the protein, altering contact of TP53 with its target DNA sequences and thereby altering transcriptional function (PMID: 8023157, 11900253). Given that TP53 directs the transcription of proteins that enable apoptosis, the inactivation of TP53 results in cells harboring damaged DNA and overall genomic instability (PMID: 11900253, 11900253).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-09-13 17:33:18',NULL,NULL,NULL,NULL,'4a10e9a6-97f9-4a45-a0ee-36eed16c8e64',7157,NULL),(29523,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-09-13 17:33:52',NULL,NULL,NULL,NULL,'835e65eb-6c26-4980-9516-9e2664205773',7157,NULL),(29524,NULL,'The DNA-binding domain (DBD) of TP53 allows for contact with target DNA sequences to transactivate downstream genes. Mutations in the DBD result in conformational changes of the protein, altering contact of TP53 with its target DNA sequences and thereby altering transcriptional function (PMID: 8023157, 11900253). Given that TP53 directs the transcription of proteins that enable apoptosis, the inactivation of TP53 results in cells harboring damaged DNA and overall genomic instability (PMID: 11900253, 11900253).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-09-13 17:33:55',NULL,NULL,NULL,NULL,'e5053096-2220-44e8-bbe7-6c961b3d0142',7157,NULL),(29525,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-09-13 17:34:36',NULL,NULL,NULL,NULL,'3d9d1781-26ab-4b3c-b3c5-7f369dc1bfcc',7157,NULL),(29526,NULL,'The DNA-binding domain (DBD) of TP53 allows for contact with target DNA sequences to transactivate downstream genes. Mutations in the DBD result in conformational changes of the protein, altering contact of TP53 with its target DNA sequences and thereby altering transcriptional function (PMID: 8023157, 11900253). Given that TP53 directs the transcription of proteins that enable apoptosis, the inactivation of TP53 results in cells harboring damaged DNA and overall genomic instability (PMID: 11900253, 11900253).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-09-13 17:34:40',NULL,NULL,NULL,NULL,'a5ccbc05-9942-4193-8e34-e661cd652a6c',7157,NULL),(29527,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-09-13 17:50:01',NULL,NULL,NULL,NULL,'dd56cbc3-0354-47e7-b8fb-d988cad226e4',7157,NULL),(29528,NULL,'The tetramerization domain of TP53 is critical for the proper oligomerization and function of the p53 tumor suppressor protein. Mutations of the tetramerization domain, specifically those around G334 that affect the hinge region between the alpha helix and the beta-strand, have been shown to result in structural changes that preclude proper oligomerization of the protein and therefore disrupt proper functioning (PMID: 16007150, 16460008).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-09-13 17:58:20',NULL,NULL,NULL,NULL,'c8b78bd6-3d11-45e8-b5db-0a5bbfe501b2',7157,NULL),(29529,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2024-09-13 17:58:58',NULL,NULL,NULL,NULL,'cd09d6ee-1b5e-49c4-bcd3-1b43d7a05d03',7157,NULL),(29530,NULL,'The tetramerization domain of TP53 is critical for the proper oligomerization and function of the p53 tumor suppressor protein. Mutations of the tetramerization domain, specifically those around G334 that affect the hinge region between the alpha helix and the beta-strand, have been shown to result in structural changes that preclude proper oligomerization of the protein and therefore disrupt proper functioning (PMID: 16007150, 16460008).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2024-09-13 17:59:05',NULL,NULL,NULL,NULL,'dea787f3-045d-49e6-836b-1f50a830a749',7157,NULL),(29531,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2025-03-12 10:42:51',NULL,NULL,NULL,NULL,'7c39e655-93cd-4c86-8f2b-87ab5f3e58ab',7157,NULL),(29532,NULL,'The TP53 V157I mutation is located in the DNA-binding domain of the protein. This mutation has been identified in vulvar squamous cell carcinoma (PMID: 17539962). In vitro studies with various human cancer cell lines expressing TP53 V157I demonstrate that the mutation is likely neutral as measured by the lack of dominant-negative function, growth suppression activity being comparable to wildtype and the gene expression profile being similar to wildtype (PMID: 29979965, 30224644, 37697151). In vivo studies with yeast expressing TP53 V157I demonstrate that the mutation retains partial functionality of transactivational activity compared to wildtype (PMID: 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2025-03-12 11:05:59',NULL,NULL,NULL,NULL,'3e31f62f-72f2-4316-b40a-07215d252020',7157,NULL),(29533,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2025-03-12 14:48:38',NULL,NULL,NULL,NULL,'add021b0-0b2f-4f86-be09-c502dfc6d7e0',7157,NULL),(29534,NULL,'The TP53 A276V mutation is located in the DNA-binding domain of the protein. This mutation has been identified in colorectal cancer (PMID: 26109429). In vitro studies with various human cancer cell lines expressing TP53 A276V demonstrate that the mutation is likely neutral as measured by the lack of dominant-negative function, growth suppression activity being comparable to wildtype and the gene expression profile being similar to wildtype (PMID: 29979965, 30224644, 37697151). In vivo studies with yeast expressing TP53 A276V demonstrate that the mutation retains partial functionality of transactivational activity compared to wildtype (PMID: 12826609).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2025-03-12 14:48:43',NULL,NULL,NULL,NULL,'c12e1b67-c70a-4d50-ba21-c217d05f5156',7157,NULL),(29535,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Neutral','2025-06-05 16:23:12',NULL,NULL,NULL,NULL,'e5cbbc37-aec6-4176-adca-79e435e973f4',7157,NULL),(29536,NULL,'The TP53 R290C mutation is located in the DNA binding domain of the protein. In vitro studies with transactivation assays demonstrate that the mutation is likely neutral as measured by transactivation functionality being comparable to wildtype (PMID: 12826609). In vitro studies with A549 TP53 wildtype and null cells expressing TP53 R290C also demonstrate that the mutation is likely neutral as measured by the lack of dominant-negative effect (PMID: 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Neutral','2025-06-05 16:35:44',NULL,NULL,NULL,NULL,'07b3a209-a4f4-4d88-ad82-efc8eba52edf',7157,NULL),(29537,NULL,NULL,'ONCOGENIC',NULL,_binary '\0','Likely Oncogenic','2025-07-07 11:36:09',NULL,NULL,NULL,NULL,'cfb9ca69-48d0-4a48-8f76-2a5922c1a00b',7157,NULL),(29538,NULL,'The TP53 R156P mutation is located in the DNA-binding domain of the protein. In vivo studies with yeast expressing TP53 R156P demonstrate that the mutation is inactivating as measured by the loss of transactivational activity as compared to wildtype (PMID: 27328919, 12826609). In vitro studies with various human cancer cell lines expressing TP53 R156P also demonstrate the mutation is inactivating as measured by reduced growth suppression activity as compared to wildtype (PMID: 29979965, 30224644).','MUTATION_EFFECT',NULL,_binary '\0','Likely Loss-of-function','2025-07-07 11:37:54',NULL,NULL,NULL,NULL,'a2a7a451-66f3-4746-af4b-1764b3025160',7157,NULL),(29539,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:56:41',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29540,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:56:56',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29541,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:57:07',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29542,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:57:20',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29543,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:57:30',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29544,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:57:40',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29545,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:57:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29546,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:58:13',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29547,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:58:23',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29548,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:58:32',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29549,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:58:42',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29550,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:58:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29551,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 13:59:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29552,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 14:00:48',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29553,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 14:00:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29554,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 14:01:11',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29555,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 14:01:22',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29556,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-09-26 14:01:32',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29557,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-02 21:29:56',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29558,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-02 22:03:21',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29559,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-02 22:06:16',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29560,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-02 22:07:33',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29561,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 12:32:30',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29562,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 14:13:58',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29563,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 14:45:19',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29564,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 14:45:43',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29565,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 21:58:10',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29566,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 22:12:35',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29567,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-04 22:38:27',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29568,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-07 15:04:25',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29569,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-23 18:06:34',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29570,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-23 18:07:27',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29571,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-10-23 18:07:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29572,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-11-06 23:30:29',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29573,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-11-06 23:31:20',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29574,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-11-06 23:31:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29575,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-12-04 21:31:54',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29576,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-12-04 22:10:02',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29577,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2016-12-04 22:10:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29578,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-02 12:07:09',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29579,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-02 12:07:12',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29580,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-02 12:06:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29581,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-02 12:07:05',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29582,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-02 12:07:03',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29583,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2024-07-02 11:17:23',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29584,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-10 14:38:16',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29585,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-10 14:58:50',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29586,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-03-10 15:20:12',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29587,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-06-06 15:02:00',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29588,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-06-06 15:02:09',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29589,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-11-07 08:18:48',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29590,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2017-11-07 08:18:49',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29591,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2020-06-10 00:32:45',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29592,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2020-06-10 01:02:21',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29593,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-01-24 15:13:52',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29594,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-01-24 15:13:58',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29595,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2024-07-02 11:22:23',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29596,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-01-24 15:14:12',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29597,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-01-24 15:14:18',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29598,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-01-26 15:55:24',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29599,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:37:04',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29600,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:37:09',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29601,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:37:14',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29602,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:37:18',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29603,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:39:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29604,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:39:57',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29605,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:40:06',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29606,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 14:55:49',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29607,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:00:42',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29608,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:00:46',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29609,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:00:49',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29610,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:00:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29611,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:00:57',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29612,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:01',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29613,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:05',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29614,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:09',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29615,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:13',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29616,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:17',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29617,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:21',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29618,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:01:26',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29619,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:06:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29620,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:07:03',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29621,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:07:07',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29622,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-03 15:07:11',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29623,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:29',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29624,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:34',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29625,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:38',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29626,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:42',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29627,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:47',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29628,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:53',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29629,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:31:57',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29630,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:03',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29631,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:08',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29632,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:14',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29633,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:17',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29634,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:22',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29635,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:27',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29636,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:31',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29637,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:34',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29638,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:38',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29639,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:44',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29640,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:48',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29641,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:52',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29642,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:32:56',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29643,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:00',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29644,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:04',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29645,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:09',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29646,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:13',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29647,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:17',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29648,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:22',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29649,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:27',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29650,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:32',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29651,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:36',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29652,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:40',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29653,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:44',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29654,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:50',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29655,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:54',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29656,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:33:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29657,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:03',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29658,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:09',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29659,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:13',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29660,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:18',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29661,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:21',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29662,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:26',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29663,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:30',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29664,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:34',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29665,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:38',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29666,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:43',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29667,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:49',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29668,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:54',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29669,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:34:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29670,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:03',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29671,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:08',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29672,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:13',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29673,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:18',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29674,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:24',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29675,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:30',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29676,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:35',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29677,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:42',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29678,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:46',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29679,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:50',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29680,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:55',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29681,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:35:59',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29682,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:06',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29683,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:12',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29684,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:16',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29685,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:21',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29686,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:28',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29687,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:32',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29688,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:36',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29689,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:42',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29690,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:46',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29691,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:50',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29692,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:36:54',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29693,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:00',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29694,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:04',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29695,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:10',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29696,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:15',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29697,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:20',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29698,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:24',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29699,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:28',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29700,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:35',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29701,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:39',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29702,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:44',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29703,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:37:51',NULL,NULL,NULL,NULL,NULL,7157,NULL),(29704,NULL,NULL,'VUS',NULL,_binary '\0',NULL,'2023-02-06 14:39:56',NULL,NULL,NULL,NULL,NULL,7157,NULL);
/*!40000 ALTER TABLE `evidence` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `evidence_alteration`
--

DROP TABLE IF EXISTS `evidence_alteration`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `evidence_alteration` (
  `evidence_id` int NOT NULL,
  `alteration_id` int NOT NULL,
  PRIMARY KEY (`evidence_id`,`alteration_id`),
  KEY `FKCF414387AAD5975` (`alteration_id`),
  KEY `FKCF414387F7262935` (`evidence_id`),
  CONSTRAINT `FKCF414387AAD5975` FOREIGN KEY (`alteration_id`) REFERENCES `alteration` (`id`),
  CONSTRAINT `FKCF414387F7262935` FOREIGN KEY (`evidence_id`) REFERENCES `evidence` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `evidence_alteration`
--

LOCK TABLES `evidence_alteration` WRITE;
/*!40000 ALTER TABLE `evidence_alteration` DISABLE KEYS */;
INSERT INTO `evidence_alteration` VALUES (2447,1474),(2448,1474),(2449,1474),(2450,1474),(2451,1474),(2452,1474),(2453,1474),(2454,1474),(2455,1474),(2456,1474),(2457,1474),(2458,1474),(2459,1474),(2460,1474),(2461,1474),(2462,1474),(2463,1474),(2464,1474),(2465,1474),(2466,1474),(2467,1474),(2468,1474),(2469,1474),(2470,1474),(2471,1474),(2472,1474),(2473,1474),(2474,1474),(2475,1474),(2476,1474),(2477,1474),(2478,1474),(2479,1474),(2480,1474),(2481,1474),(2482,1474),(2483,1474),(2484,1474),(2507,1474),(2508,1474),(2509,1474),(2510,1474),(2485,1475),(2486,1475),(2487,1476),(2488,1476),(2489,1477),(2490,1477),(2491,1477),(2492,1477),(2493,1477),(2494,1477),(2495,1477),(2496,1477),(2497,1477),(2498,1477),(2499,1477),(2500,1477),(2501,1477),(2502,1478),(2503,1478),(2504,1478),(2505,1478),(2506,1478),(2507,1478),(2508,1478),(2509,1478),(2510,1478),(2511,1479),(2512,1479),(2513,1480),(2514,1480),(2515,1481),(2516,1481),(2517,1482),(2518,1482),(2519,1483),(2520,1483),(2521,1484),(2522,1484),(2523,1485),(2524,1485),(2525,1485),(2526,1485),(2527,1486),(2528,1486),(2529,1486),(2530,1486),(2531,1487),(2532,1487),(2533,1488),(2534,1488),(2535,1489),(2536,1489),(2537,1490),(2538,1490),(2539,1491),(2540,1491),(2541,1492),(2542,1492),(2543,1493),(2544,1493),(2545,1494),(2546,1494),(2547,1495),(2548,1495),(2549,1496),(2550,1496),(2551,1497),(2552,1497),(2553,1498),(2554,1498),(2555,1499),(2556,1499),(2557,1500),(2558,1500),(2559,1501),(2560,1501),(2561,1502),(2562,1502),(2563,1503),(2564,1503),(2746,1503),(2747,1503),(2565,1504),(2566,1504),(2567,1505),(2568,1505),(2569,1506),(2570,1506),(2746,1506),(2747,1506),(2571,1507),(2572,1507),(2746,1507),(2747,1507),(2573,1508),(2574,1508),(2575,1509),(2576,1509),(2577,1510),(2578,1510),(2579,1511),(2580,1511),(2581,1512),(2582,1512),(2583,1513),(2584,1513),(2585,1514),(2586,1514),(2587,1515),(2588,1515),(2589,1516),(2590,1516),(2591,1517),(2592,1517),(2593,1518),(2594,1518),(2595,1519),(2596,1519),(2597,1520),(2598,1520),(2599,1521),(2600,1521),(2601,1522),(2602,1522),(2603,1523),(2604,1523),(2605,1524),(2606,1524),(2607,1525),(2608,1525),(2609,1526),(2610,1526),(2611,1527),(2612,1527),(2613,1528),(2614,1528),(2615,1529),(2616,1529),(2617,1530),(2618,1530),(2619,1531),(2620,1531),(2621,1532),(2622,1532),(2623,1533),(2624,1533),(2625,1534),(2626,1534),(2627,1535),(2628,1535),(2629,1536),(2630,1536),(2631,1537),(2632,1537),(2633,1538),(2634,1538),(2635,1539),(2636,1539),(2637,1540),(2638,1540),(2639,1541),(2640,1541),(2641,1542),(2642,1542),(2643,1543),(2644,1543),(2645,1544),(2646,1544),(2647,1545),(2648,1545),(2649,1546),(2650,1546),(2651,1547),(2652,1547),(2653,1548),(2654,1548),(2655,1549),(2656,1549),(2657,1550),(2658,1550),(2659,1551),(2660,1551),(2661,1552),(2662,1552),(2663,1553),(2664,1553),(2665,1554),(2666,1554),(2667,1555),(2668,1555),(2669,1556),(2670,1556),(2671,1557),(2672,1557),(2673,1558),(2674,1558),(2675,1559),(2676,1559),(2677,1560),(2678,1560),(2679,1561),(2680,1561),(2681,1562),(2682,1562),(2683,1563),(2684,1563),(2685,1564),(2686,1564),(2687,1565),(2688,1565),(2689,1566),(2690,1566),(2691,1567),(2692,1567),(2693,1567),(2694,1567),(2695,1567),(2696,1567),(2697,1567),(2698,1567),(2699,1567),(2700,1567),(2701,1567),(2702,1568),(2703,1568),(2704,1569),(2705,1569),(2706,1570),(2707,1570),(2708,1571),(2709,1571),(2710,1572),(2711,1572),(2712,1573),(2713,1573),(2714,1574),(2715,1574),(2716,1575),(2717,1575),(2718,1576),(2719,1576),(2720,1577),(2721,1577),(2722,1578),(2723,1578),(2724,1579),(2725,1579),(2726,1580),(2727,1580),(2728,1581),(2729,1581),(2730,1582),(2731,1582),(2732,1583),(2733,1583),(2734,1584),(2735,1584),(2736,1585),(2737,1585),(2738,1586),(2739,1586),(2740,1587),(2741,1587),(2742,1588),(2743,1588),(2744,1589),(2745,1589),(2746,1590),(2747,1590),(2748,1591),(2749,1591),(2750,1592),(2751,1592),(2752,1593),(2753,1593),(2754,1594),(2755,1594),(2756,1595),(2757,1595),(2758,1596),(2759,1596),(2760,1597),(2761,1597),(2762,1598),(2763,1598),(2764,1599),(2765,1599),(2766,1600),(2767,1600),(2768,1601),(2769,1601),(2770,1602),(2771,1602),(2772,1603),(2773,1603),(2774,1604),(2775,1604),(2776,1605),(2777,1605),(2778,1606),(2779,1606),(2780,1607),(2781,1607),(2782,1607),(2783,1607),(2784,1607),(2785,1608),(2786,1608),(2787,1608),(2788,1608),(2789,1608),(2790,1608),(2791,1608),(2792,1608),(2793,1609),(2794,1609),(2795,1609),(2796,1609),(2797,1610),(2798,1610),(2799,1611),(2800,1611),(2801,1612),(2802,1612),(2803,1613),(2804,1613),(2805,1614),(2806,1614),(2807,1615),(2808,1615),(2809,1616),(2810,1616),(2811,1617),(2812,1617),(2813,1618),(2814,1618),(2815,1619),(2816,1619),(2817,1620),(2818,1620),(2819,1621),(2820,1621),(2821,1622),(2822,1622),(2823,1623),(2824,1623),(2825,1624),(2826,1624),(2827,1625),(2828,1625),(2829,1626),(2830,1626),(2831,1627),(2832,1627),(2833,1628),(2834,1628),(2835,1629),(2836,1629),(2837,1630),(2838,1630),(2839,1631),(2840,1631),(2841,1632),(2842,1632),(2843,1633),(2844,1633),(2845,1634),(2846,1634),(2847,1635),(2848,1635),(2849,1636),(2850,1636),(2851,1637),(2852,1637),(2853,1638),(2854,1638),(2855,1639),(2856,1639),(2857,1640),(2858,1640),(2859,1641),(2860,1641),(2861,1642),(2862,1642),(2863,1643),(2864,1643),(2865,1644),(2866,1644),(2867,1645),(2868,1645),(2869,1646),(2870,1646),(2871,1647),(2872,1647),(2873,1648),(2874,1648),(2875,1649),(2876,1649),(2877,1650),(2878,1650),(2879,1651),(2880,1651),(2881,1652),(2882,1652),(2883,1653),(2884,1653),(2885,1654),(2886,1654),(2887,1655),(2888,1655),(2889,1656),(2890,1656),(2891,1657),(2892,1657),(2893,1658),(2894,1658),(2895,1659),(2896,1659),(2897,1660),(2898,1660),(2899,1661),(2900,1661),(2901,1662),(2902,1662),(2903,1663),(2904,1663),(2905,1664),(2906,1664),(2907,1665),(2908,1665),(2909,1666),(2910,1666),(2911,1667),(2912,1667),(2913,1668),(2914,1668),(2915,1669),(2916,1669),(2917,1670),(2918,1670),(2919,1671),(2920,1671),(2921,1672),(2922,1672),(2923,1673),(2924,1673),(2925,1674),(2926,1674),(2927,1675),(2928,1675),(2929,1676),(2930,1676),(2931,1677),(2932,1677),(2933,1678),(2934,1678),(2935,1679),(2936,1679),(2937,1680),(2938,1680),(2939,1681),(2940,1681),(2941,1682),(2942,1682),(2943,1683),(2944,1683),(2945,1684),(2946,1684),(2947,1685),(2948,1686),(2949,1687),(2950,1688),(2951,1689),(2952,1690),(2953,1691),(2954,1692),(2955,1693),(2956,1694),(2957,1695),(2958,1696),(2959,1697),(2960,1698),(2961,1699),(2962,1700),(2963,1701),(2964,1702),(2965,1703),(2966,1704),(2967,1705),(2968,1706),(2969,1707),(2970,1708),(2971,1709),(2972,1710),(2973,1711),(2974,1712),(2975,1713),(2976,1714),(2977,1715),(2978,1716),(2979,1717),(2980,1718),(2981,1719),(2982,1720),(2983,1721),(2984,1722),(2985,1723),(2986,1724),(2987,1725),(2988,1726),(2989,1727),(2990,1728),(2991,1729),(2992,1730),(2993,1731),(2994,1732),(2995,1733),(2996,1734),(2997,1735),(2998,1736),(2999,1737),(3000,1738),(3001,1739),(3002,1740),(3003,1741),(3004,1742),(3005,1743),(3006,1744),(3007,1745),(3008,1746),(3009,1747),(3010,1748),(3011,1749),(3012,1750),(3013,1751),(3014,1752),(3015,1753),(3016,1754),(3017,1755),(3018,1756),(3019,1757),(3020,1758),(3021,1759),(3022,1760),(3023,1761),(3024,1762),(3025,1763),(3026,1764),(3027,1765),(3028,1766),(3029,1767),(3030,1768),(3031,1769),(3032,1770),(3033,1771),(3034,1772),(3035,1773),(3036,1774),(3037,1775),(3038,1776),(3039,1777),(3040,1778),(3041,1779),(3042,1780),(3043,1781),(3044,1782),(3045,1783),(3046,1784),(3047,1785),(3048,1786),(3049,1787),(3050,1788),(3051,1789),(3052,1790),(3053,1791),(3054,1792),(3055,1793),(3056,1794),(3057,1795),(3058,1796),(3059,1797),(3060,1798),(3061,1799),(3062,1800),(3063,1801),(3064,1802),(3065,1803),(3066,1804),(3067,1805),(3068,1806),(3069,1807),(3070,1808),(3071,1809),(3072,1810),(3073,1811),(3074,1812),(3075,1813),(3076,1814),(3077,1815),(3078,1816),(3079,1817),(3080,1818),(3081,1819),(3082,1820),(3083,1821),(3084,1822),(3085,1823),(3086,1824),(3087,1825),(3088,1826),(3089,1827),(3090,1828),(3091,1829),(3092,1830),(3093,1831),(3094,1832),(3095,1833),(3096,1834),(3097,1835),(3098,1836),(3099,1837),(3100,1838),(3101,1839),(3102,1840),(3103,1841),(3104,1842),(3105,1843),(3106,1844),(3107,1845),(3108,1846),(3109,1847),(3110,1848),(3111,1849),(3112,1850),(3113,1851),(3114,1852),(3115,1853),(3116,1854),(3117,1855),(3118,1856),(3119,1857),(3120,1858),(3121,1859),(3122,1860),(3123,1861),(3124,1862),(3125,1863),(3126,1864),(3127,1865),(3128,1866),(3129,1867),(3130,1868),(3131,1869),(3132,1870),(3133,1871),(3134,1872),(3135,1873),(3136,1874),(3137,1875),(3138,1876),(3139,1877),(3140,1878),(3141,1879),(3142,1880),(3143,1881),(3144,1882),(3145,1883),(3146,1884),(3147,1885),(3148,1886),(3149,1887),(3150,1888),(3151,1889),(3152,1890),(3153,1891),(3154,1892),(3155,1893),(3156,1894),(3157,1895),(3158,1896),(3159,1897),(3160,1898),(3161,1899),(3162,1900),(3163,1901),(3164,1902),(3165,1903),(3166,1904),(3167,1905),(3168,1906),(3169,1907),(3170,1908),(3171,1909),(3172,1910),(3173,1911),(3174,1912),(3175,1913),(3176,1914),(3177,1915),(3178,1916),(3179,1917),(3180,1918),(3181,1919),(3182,1920),(3183,1921),(3184,1922),(3185,1923),(3186,1924),(3187,1925),(3188,1926),(3189,1927),(3190,1928),(3191,1929),(3192,1930),(3193,1931),(3194,1932),(3195,1933),(3196,1934),(3197,1935),(3198,1936),(3199,1937),(3200,1938),(3201,1939),(3202,1940),(3203,1941),(3204,1942),(3205,1943),(3206,1944),(3207,1945),(3208,1946),(3209,1947),(3210,1948),(3211,1949),(3212,1950),(3213,1951),(3214,1952),(3215,1953),(3216,1954),(3217,1955),(3218,1956),(3219,1957),(3220,1958),(3221,1959),(3222,1960),(3223,1961),(3224,1962),(3225,1963),(3226,1964),(3227,1965),(3228,1966),(3229,1967),(3230,1968),(3231,1969),(3232,1970),(3233,1971),(3234,1972),(3235,1973),(3236,1974),(3237,1975),(3238,1976),(3239,1977),(3240,1978),(3241,1979),(3242,1980),(3243,1981),(3244,1982),(3245,1983),(3246,1984),(3247,1985),(3248,1986),(3249,1987),(3250,1988),(3251,1989),(3252,1990),(3253,1991),(3254,1992),(3255,1993),(3256,1994),(3257,1995),(3258,1996),(3259,1997),(3260,1998),(3261,1999),(3262,2000),(3263,2001),(3264,2002),(3265,2003),(3266,2004),(3267,2005),(3268,2006),(3269,2007),(3270,2008),(3271,2009),(3272,2010),(3273,2011),(3274,2012),(3275,2013),(3276,2014),(3277,2015),(3278,2016),(3279,2017),(3280,2018),(3281,2019),(3282,2020),(3283,2021),(3284,2022),(3285,2023),(3286,2024),(3287,2025),(3288,2026),(3289,2027),(3290,2028),(3291,2029),(3292,2030),(3293,2031),(3294,2032),(3295,2033),(3296,2034),(3297,2035),(3298,2036),(26076,16804),(26077,16804),(26078,16804),(26079,16804),(26080,16804),(26081,16804),(26082,16804),(26083,16804),(26084,16804),(26085,16804),(26086,16805),(26087,16806),(26088,16806),(26089,16807),(26090,16807),(26091,16808),(26092,16808),(26093,16809),(26094,16809),(26095,16810),(26096,16810),(26097,16811),(26098,16811),(26099,16812),(26100,16812),(26101,16813),(26102,16813),(26103,16814),(26104,16814),(26105,16815),(26106,16815),(26107,16816),(26108,16816),(26109,16817),(26110,16817),(26111,16818),(26112,16818),(26113,16819),(26114,16819),(26115,16820),(26116,16820),(26117,16820),(26118,16820),(26119,16820),(26120,16820),(26121,16820),(26122,16821),(26123,16821),(26124,16822),(26125,16822),(26126,16823),(26127,16823),(26128,16824),(26129,16824),(26130,16825),(26131,16825),(26132,16826),(26133,16826),(26134,16827),(26135,16827),(26136,16828),(26137,16828),(26138,16829),(26139,16829),(26140,16830),(26141,16830),(26142,16831),(26143,16831),(26144,16832),(26145,16832),(26146,16833),(26147,16833),(26148,16834),(26149,16834),(26150,16835),(26151,16835),(26152,16835),(26153,16835),(26154,16835),(26155,16835),(26156,16835),(26157,16836),(26158,16837),(26159,16838),(26160,16839),(26161,16840),(26162,16841),(26163,16842),(26164,16843),(26165,16844),(28519,18246),(28520,18246),(28521,18247),(28522,18247),(28523,18248),(28524,18248),(28525,18249),(28526,18249),(28527,18250),(28528,18250),(28529,18251),(28530,18251),(28531,18252),(28532,18252),(28533,18253),(28534,18253),(28535,18254),(28536,18254),(28537,18255),(28538,18255),(28539,18256),(28540,18256),(28541,18257),(28542,18257),(28543,18258),(28544,18258),(28545,18259),(28546,18259),(28547,18260),(28548,18260),(28549,18261),(28550,18261),(28551,18262),(28552,18262),(28553,18263),(28554,18263),(28555,18264),(28556,18264),(28557,18265),(28558,18265),(28559,18266),(28560,18266),(28561,18267),(28562,18267),(28563,18268),(28564,18268),(28565,18269),(28566,18269),(28567,18270),(28568,18270),(28569,18271),(28570,18271),(28571,18272),(28572,18272),(28573,18273),(28574,18273),(28575,18274),(28576,18274),(28577,18275),(28578,18275),(28579,18276),(28580,18276),(28581,18277),(28582,18277),(28583,18278),(28584,18278),(28585,18279),(28586,18279),(28587,18280),(28588,18280),(28589,18281),(28590,18281),(28591,18282),(28592,18282),(28593,18283),(28594,18283),(28595,18284),(28596,18284),(28597,18285),(28598,18285),(28599,18286),(28600,18286),(28601,18287),(28602,18287),(28603,18288),(28604,18288),(28605,18289),(28606,18289),(28607,18290),(28608,18290),(28609,18291),(28610,18291),(28611,18292),(28612,18292),(28613,18293),(28614,18293),(28615,18294),(28616,18294),(28617,18295),(28618,18295),(28619,18295),(28620,18295),(28621,18296),(28622,18296),(28623,18297),(28624,18297),(28625,18298),(28626,18298),(28627,18299),(28628,18299),(28629,18300),(28630,18300),(28631,18301),(28632,18301),(28633,18302),(28634,18302),(28635,18303),(28636,18303),(28637,18304),(28638,18304),(28639,18305),(28640,18305),(28641,18306),(28642,18306),(28643,18307),(28644,18307),(28645,18308),(28646,18308),(28647,18309),(28648,18309),(28649,18310),(28650,18310),(28651,18311),(28652,18311),(28653,18312),(28654,18312),(28655,18313),(28656,18313),(28657,18314),(28658,18314),(28659,18315),(28660,18315),(28661,18316),(28662,18316),(28663,18317),(28664,18317),(28665,18318),(28666,18318),(28667,18319),(28668,18319),(28669,18320),(28670,18320),(28671,18321),(28672,18321),(28673,18322),(28674,18322),(28675,18323),(28676,18323),(28677,18324),(28678,18324),(28679,18325),(28680,18325),(28681,18326),(28682,18326),(28683,18327),(28684,18327),(28685,18328),(28686,18328),(28687,18329),(28688,18329),(28689,18330),(28690,18330),(28691,18331),(28692,18331),(28693,18332),(28694,18332),(28695,18333),(28696,18333),(28697,18334),(28698,18334),(28699,18335),(28700,18335),(28701,18336),(28702,18336),(28703,18337),(28704,18337),(28705,18338),(28706,18338),(28707,18339),(28708,18339),(28709,18340),(28710,18340),(28711,18341),(28712,18341),(28713,18342),(28714,18342),(28715,18343),(28716,18343),(28717,18344),(28718,18344),(28719,18345),(28720,18345),(28721,18346),(28722,18346),(28723,18347),(28724,18347),(28725,18348),(28726,18348),(28727,18349),(28728,18349),(28729,18350),(28730,18350),(28731,18351),(28732,18351),(28733,18352),(28734,18352),(28735,18353),(28736,18353),(28737,18354),(28738,18354),(28739,18355),(28740,18355),(28741,18356),(28742,18356),(28743,18357),(28744,18357),(28745,18358),(28746,18358),(28747,18359),(28748,18359),(28749,18360),(28750,18360),(28751,18361),(28752,18361),(28753,18362),(28754,18362),(28755,18363),(28756,18363),(28757,18364),(28758,18364),(28759,18365),(28760,18365),(28761,18366),(28762,18366),(28763,18367),(28764,18367),(28765,18368),(28766,18368),(28767,18369),(28768,18369),(28769,18370),(28770,18370),(28771,18371),(28772,18371),(28773,18372),(28774,18372),(28775,18373),(28776,18373),(28777,18374),(28778,18374),(28779,18375),(28780,18375),(28781,18376),(28782,18376),(28783,18377),(28784,18377),(28785,18378),(28786,18378),(28787,18379),(28788,18379),(28789,18380),(28790,18380),(28791,18381),(28792,18381),(28793,18382),(28794,18382),(28795,18383),(28796,18383),(28797,18384),(28798,18384),(28799,18385),(28800,18385),(28801,18386),(28802,18386),(28803,18387),(28804,18387),(28805,18388),(28806,18388),(28807,18389),(28808,18389),(28809,18390),(28810,18390),(28811,18391),(28812,18391),(28813,18392),(28814,18392),(28815,18393),(28816,18393),(28817,18394),(28818,18394),(28819,18395),(28820,18395),(28821,18396),(28822,18396),(28823,18397),(28824,18397),(28825,18398),(28826,18398),(28827,18399),(28828,18399),(28829,18400),(28830,18400),(28831,18401),(28832,18401),(28833,18402),(28834,18402),(28835,18403),(28836,18403),(28837,18404),(28838,18404),(28839,18405),(28840,18405),(28841,18406),(28842,18406),(28843,18407),(28844,18407),(28845,18408),(28846,18408),(28847,18409),(28848,18409),(28849,18410),(28850,18410),(28851,18411),(28852,18411),(28853,18412),(28854,18412),(28855,18413),(28856,18413),(28857,18414),(28858,18414),(28859,18415),(28860,18415),(28861,18416),(28862,18416),(28863,18417),(28864,18417),(28865,18418),(28866,18418),(28867,18419),(28868,18419),(28869,18420),(28870,18420),(28871,18421),(28872,18421),(28873,18422),(28874,18422),(28875,18423),(28876,18423),(28877,18424),(28878,18424),(28879,18425),(28880,18425),(28881,18426),(28882,18426),(28883,18427),(28884,18427),(28885,18427),(28886,18427),(28887,18427),(28888,18427),(28889,18427),(28890,18427),(28891,18427),(28892,18427),(28893,18427),(28894,18427),(28895,18427),(28896,18427),(28897,18427),(28898,18427),(28899,18427),(28900,18427),(28901,18428),(28902,18428),(28903,18429),(28904,18429),(28905,18430),(28906,18430),(28907,18430),(28908,18430),(28909,18431),(28910,18431),(28911,18432),(28912,18432),(28913,18433),(28914,18433),(28915,18434),(28916,18434),(28917,18435),(28918,18435),(28919,18436),(28920,18436),(28921,18437),(28922,18437),(28923,18438),(28924,18438),(28925,18439),(28926,18439),(28927,18440),(28928,18440),(28929,18441),(28930,18441),(28931,18442),(28932,18442),(28933,18443),(28934,18443),(28935,18444),(28936,18444),(28937,18445),(28938,18445),(28939,18446),(28940,18446),(28941,18447),(28942,18447),(28943,18448),(28944,18448),(28945,18449),(28946,18449),(28947,18450),(28948,18450),(28949,18451),(28950,18451),(28951,18452),(28952,18452),(28953,18453),(28954,18453),(28955,18454),(28956,18454),(28957,18455),(28958,18455),(28959,18456),(28960,18456),(28961,18457),(28962,18457),(28963,18458),(28964,18458),(28965,18459),(28966,18459),(28967,18460),(28968,18460),(28969,18461),(28970,18461),(28971,18462),(28972,18462),(28973,18463),(28974,18463),(28975,18464),(28976,18464),(28977,18465),(28978,18465),(28979,18466),(28980,18466),(28981,18467),(28982,18467),(28983,18468),(28984,18468),(28985,18469),(28986,18469),(28987,18470),(28988,18470),(28989,18471),(28990,18471),(28991,18472),(28992,18472),(28993,18473),(28994,18473),(28995,18474),(28996,18474),(28997,18475),(28998,18475),(28999,18476),(29000,18476),(29001,18477),(29002,18477),(29003,18478),(29004,18478),(29005,18479),(29006,18479),(29007,18480),(29008,18480),(29009,18481),(29010,18481),(29011,18482),(29012,18482),(29013,18483),(29014,18483),(29015,18484),(29016,18484),(29017,18485),(29018,18485),(29019,18486),(29020,18486),(29021,18487),(29022,18487),(29023,18488),(29024,18488),(29025,18489),(29026,18489),(29027,18490),(29028,18490),(29029,18491),(29030,18491),(29031,18492),(29032,18492),(29033,18493),(29034,18493),(29035,18494),(29036,18494),(29037,18495),(29038,18495),(29039,18496),(29040,18496),(29041,18497),(29042,18497),(29043,18498),(29044,18498),(29045,18499),(29046,18499),(29047,18500),(29048,18500),(29049,18501),(29050,18501),(29051,18502),(29052,18502),(29053,18503),(29054,18503),(29055,18504),(29056,18504),(29057,18505),(29058,18505),(29059,18506),(29060,18506),(29061,18507),(29062,18507),(29063,18508),(29064,18508),(29065,18509),(29066,18509),(29067,18510),(29068,18510),(29069,18511),(29070,18511),(29071,18512),(29072,18512),(29073,18513),(29074,18513),(29075,18514),(29076,18514),(29077,18515),(29078,18515),(29079,18516),(29080,18516),(29081,18517),(29082,18517),(29083,18518),(29084,18518),(29085,18519),(29086,18519),(29087,18520),(29088,18520),(29089,18521),(29090,18521),(29091,18522),(29092,18522),(29093,18523),(29094,18523),(29095,18524),(29096,18524),(29097,18525),(29098,18525),(29099,18526),(29100,18526),(29101,18527),(29102,18527),(29103,18528),(29104,18528),(29105,18529),(29106,18529),(29107,18530),(29108,18530),(29109,18531),(29110,18531),(29111,18532),(29112,18532),(29113,18533),(29114,18533),(29115,18534),(29116,18534),(29117,18535),(29118,18535),(29119,18536),(29120,18536),(29121,18537),(29122,18537),(29123,18538),(29124,18538),(29125,18539),(29126,18539),(29127,18540),(29128,18540),(29129,18541),(29130,18541),(29131,18542),(29132,18542),(29133,18543),(29134,18543),(29135,18544),(29136,18544),(29137,18545),(29138,18545),(29139,18546),(29140,18546),(29141,18547),(29142,18547),(29143,18548),(29144,18548),(29145,18549),(29146,18549),(29147,18550),(29148,18550),(29149,18551),(29150,18551),(29151,18552),(29152,18552),(29153,18553),(29154,18553),(29155,18554),(29156,18554),(29157,18555),(29158,18555),(29159,18556),(29160,18556),(29161,18557),(29162,18557),(29163,18558),(29164,18558),(29165,18559),(29166,18559),(29167,18560),(29168,18560),(29169,18561),(29170,18561),(29171,18562),(29172,18562),(29173,18563),(29174,18563),(29175,18564),(29176,18564),(29177,18565),(29178,18565),(29179,18566),(29180,18566),(29181,18567),(29182,18567),(29183,18568),(29184,18568),(29185,18569),(29186,18569),(29187,18570),(29188,18570),(29189,18571),(29190,18571),(29191,18572),(29192,18572),(29193,18573),(29194,18573),(29195,18574),(29196,18574),(29197,18575),(29198,18575),(29199,18576),(29200,18576),(29201,18577),(29202,18577),(29203,18578),(29204,18578),(29205,18579),(29206,18579),(29207,18580),(29208,18580),(29209,18581),(29210,18581),(29211,18582),(29212,18582),(29213,18583),(29214,18583),(29215,18584),(29216,18584),(29217,18585),(29218,18585),(29219,18586),(29220,18586),(29221,18587),(29222,18587),(29223,18588),(29224,18588),(29225,18589),(29226,18589),(29227,18590),(29228,18590),(29229,18591),(29230,18591),(29231,18592),(29232,18592),(29233,18593),(29234,18593),(29235,18594),(29236,18594),(29237,18595),(29238,18595),(29239,18596),(29240,18596),(29241,18597),(29242,18597),(29243,18598),(29244,18598),(29245,18599),(29246,18599),(29247,18600),(29248,18600),(29249,18601),(29250,18601),(29251,18602),(29252,18602),(29253,18603),(29254,18603),(29255,18604),(29256,18604),(29257,18605),(29258,18605),(29259,18606),(29260,18606),(29261,18607),(29262,18607),(29263,18608),(29264,18608),(29265,18609),(29266,18609),(29267,18610),(29268,18610),(29269,18611),(29270,18611),(29271,18612),(29272,18612),(29273,18613),(29274,18613),(29275,18614),(29276,18614),(29277,18615),(29278,18615),(29279,18616),(29280,18616),(29281,18617),(29282,18617),(29283,18618),(29284,18618),(29285,18619),(29286,18619),(29287,18620),(29288,18620),(29289,18621),(29290,18621),(29291,18622),(29292,18622),(29293,18623),(29294,18623),(29295,18624),(29296,18624),(29297,18625),(29298,18625),(29299,18626),(29300,18626),(29301,18627),(29302,18627),(29303,18628),(29304,18628),(29305,18629),(29306,18629),(29307,18630),(29308,18630),(29309,18631),(29310,18631),(29311,18632),(29312,18632),(29313,18633),(29314,18633),(29315,18634),(29316,18634),(29317,18635),(29318,18635),(29319,18636),(29320,18636),(29321,18637),(29322,18637),(29323,18638),(29324,18638),(29325,18639),(29326,18639),(29327,18640),(29328,18640),(29329,18641),(29330,18641),(29331,18642),(29332,18642),(29333,18643),(29334,18643),(29335,18644),(29336,18644),(29337,18645),(29338,18645),(29339,18646),(29340,18646),(29341,18647),(29342,18647),(29343,18648),(29344,18648),(29345,18649),(29346,18649),(29347,18650),(29348,18650),(29349,18651),(29350,18651),(29351,18652),(29352,18652),(29353,18653),(29354,18653),(29355,18654),(29356,18654),(29357,18655),(29358,18655),(29359,18656),(29360,18656),(29361,18657),(29362,18657),(29363,18658),(29364,18658),(29365,18659),(29366,18659),(29367,18660),(29368,18660),(29369,18661),(29370,18661),(29371,18662),(29372,18662),(29373,18663),(29374,18663),(29375,18664),(29376,18664),(29377,18665),(29378,18665),(29379,18666),(29380,18666),(29381,18667),(29382,18667),(29383,18668),(29384,18668),(29385,18669),(29386,18669),(29387,18670),(29388,18670),(29389,18671),(29390,18671),(29391,18672),(29392,18672),(29393,18673),(29394,18673),(29395,18674),(29396,18674),(29397,18675),(29398,18675),(29399,18676),(29400,18676),(29401,18677),(29402,18677),(29403,18678),(29404,18678),(29405,18679),(29406,18679),(29407,18680),(29408,18680),(29409,18681),(29410,18681),(29411,18682),(29412,18682),(29413,18683),(29414,18683),(29415,18684),(29416,18684),(29417,18685),(29418,18685),(29419,18686),(29420,18686),(29421,18687),(29422,18687),(29423,18688),(29424,18688),(29425,18689),(29426,18689),(29427,18690),(29428,18690),(29429,18691),(29430,18691),(29431,18692),(29432,18692),(29433,18693),(29434,18693),(29435,18694),(29436,18694),(29437,18695),(29438,18695),(29439,18696),(29440,18696),(29441,18697),(29442,18697),(29443,18698),(29444,18698),(29445,18699),(29446,18699),(29447,18700),(29448,18700),(29449,18701),(29450,18701),(29451,18702),(29452,18702),(29453,18703),(29454,18703),(29455,18704),(29456,18704),(29457,18705),(29458,18705),(29459,18706),(29460,18706),(29461,18707),(29462,18707),(29463,18708),(29464,18708),(29465,18709),(29466,18709),(29467,18710),(29468,18710),(29469,18711),(29470,18711),(29471,18712),(29472,18712),(29473,18713),(29474,18713),(29475,18714),(29476,18714),(29477,18715),(29478,18715),(29479,18716),(29480,18716),(29481,18717),(29482,18717),(29483,18718),(29484,18718),(29485,18719),(29486,18719),(29487,18720),(29488,18720),(29489,18721),(29490,18721),(29491,18722),(29492,18722),(29493,18723),(29494,18723),(29495,18724),(29496,18724),(29497,18725),(29498,18725),(29499,18726),(29500,18726),(29501,18727),(29502,18727),(29503,18728),(29504,18728),(29505,18729),(29506,18729),(29507,18730),(29508,18730),(29509,18731),(29510,18731),(29511,18732),(29512,18732),(29513,18733),(29514,18733),(29515,18734),(29516,18734),(29517,18735),(29518,18735),(29519,18736),(29520,18736),(29521,18737),(29522,18737),(29523,18738),(29524,18738),(29525,18739),(29526,18739),(29527,18740),(29528,18740),(29529,18741),(29530,18741),(29531,18742),(29532,18742),(29533,18743),(29534,18743),(29535,18744),(29536,18744),(29537,18745),(29538,18745),(29539,18746),(29540,18747),(29541,18748),(29542,18749),(29543,18750),(29544,18751),(29545,18752),(29546,18753),(29547,18754),(29548,18755),(29549,18756),(29550,18757),(29551,18758),(29552,18759),(29553,18760),(29554,18761),(29555,18762),(29556,18763),(29557,18764),(29558,18765),(29559,18766),(29560,18767),(29561,18768),(29562,18769),(29563,18770),(29564,18771),(29565,18772),(29566,18773),(29567,18774),(29568,18775),(29569,18776),(29570,18777),(29571,18778),(29572,18779),(29573,18780),(29574,18781),(29575,18782),(29576,18783),(29577,18784),(29578,18785),(29579,18786),(29580,18787),(29581,18788),(29582,18789),(29583,18790),(29584,18791),(29585,18792),(29586,18793),(29587,18794),(29588,18795),(29589,18796),(29590,18797),(29591,18798),(29592,18799),(29593,18800),(29594,18801),(29595,18802),(29596,18803),(29597,18804),(29598,18805),(29599,18806),(29600,18807),(29601,18808),(29602,18809),(29603,18810),(29604,18811),(29605,18812),(29606,18813),(29607,18814),(29608,18815),(29609,18816),(29610,18817),(29611,18818),(29612,18819),(29613,18820),(29614,18821),(29615,18822),(29616,18823),(29617,18824),(29618,18825),(29619,18826),(29620,18827),(29621,18828),(29622,18829),(29623,18830),(29624,18831),(29625,18832),(29626,18833),(29627,18834),(29628,18835),(29629,18836),(29630,18837),(29631,18838),(29632,18839),(29633,18840),(29634,18841),(29635,18842),(29636,18843),(29637,18844),(29638,18845),(29639,18846),(29640,18847),(29641,18848),(29642,18849),(29643,18850),(29644,18851),(29645,18852),(29646,18853),(29647,18854),(29648,18855),(29649,18856),(29650,18857),(29651,18858),(29652,18859),(29653,18860),(29654,18861),(29655,18862),(29656,18863),(29657,18864),(29658,18865),(29659,18866),(29660,18867),(29661,18868),(29662,18869),(29663,18870),(29664,18871),(29665,18872),(29666,18873),(29667,18874),(29668,18875),(29669,18876),(29670,18877),(29671,18878),(29672,18879),(29673,18880),(29674,18881),(29675,18882),(29676,18883),(29677,18884),(29678,18885),(29679,18886),(29680,18887),(29681,18888),(29682,18889),(29683,18890),(29684,18891),(29685,18892),(29686,18893),(29687,18894),(29688,18895),(29689,18896),(29690,18897),(29691,18898),(29692,18899),(29693,18900),(29694,18901),(29695,18902),(29696,18903),(29697,18904),(29698,18905),(29699,18906),(29700,18907),(29701,18908),(29702,18909),(29703,18910),(29704,18911);
/*!40000 ALTER TABLE `evidence_alteration` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `evidence_article`
--

DROP TABLE IF EXISTS `evidence_article`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `evidence_article` (
  `evidence_id` int NOT NULL,
  `article_id` int NOT NULL,
  PRIMARY KEY (`evidence_id`,`article_id`),
  KEY `FK947FCB0E981D93BF` (`article_id`),
  KEY `FK947FCB0EF7262935` (`evidence_id`),
  CONSTRAINT `FK947FCB0E981D93BF` FOREIGN KEY (`article_id`) REFERENCES `article` (`id`),
  CONSTRAINT `FK947FCB0EF7262935` FOREIGN KEY (`evidence_id`) REFERENCES `evidence` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `evidence_article`
--

LOCK TABLES `evidence_article` WRITE;
/*!40000 ALTER TABLE `evidence_article` DISABLE KEYS */;
INSERT INTO `evidence_article` VALUES (2538,137),(2542,137),(2548,137),(2556,137),(2558,137),(2562,137),(2731,137),(2802,137),(2810,137),(2818,137),(2822,137),(2824,137),(2828,137),(2830,137),(2834,137),(2836,137),(2850,137),(2852,137),(2868,137),(29100,137),(29122,137),(29508,137),(2552,197),(2632,197),(2749,197),(2802,197),(2806,197),(2808,197),(2810,197),(2812,197),(2816,197),(2820,197),(2822,197),(2824,197),(2832,197),(2848,197),(2864,197),(2878,197),(2880,197),(2882,197),(2884,197),(2886,197),(2888,197),(2890,197),(2892,197),(2894,197),(2896,197),(2898,197),(2902,197),(2904,197),(2906,197),(2910,197),(2912,197),(2914,197),(2916,197),(2918,197),(2920,197),(2922,197),(2924,197),(2928,197),(2936,197),(2944,197),(2737,198),(2739,198),(2741,198),(2814,198),(2926,198),(2448,242),(2503,242),(26137,246),(26137,248),(26079,251),(2812,393),(2446,451),(2792,454),(2755,455),(2788,455),(2790,455),(2924,508),(28883,577),(28891,583),(28891,585),(28893,592),(28872,778),(2446,1084),(2446,1085),(2448,1085),(2488,1085),(2503,1085),(2512,1085),(2514,1085),(2518,1085),(2522,1085),(2534,1085),(2536,1085),(2550,1085),(2554,1085),(2564,1085),(2570,1085),(2572,1085),(2588,1085),(2692,1085),(2745,1085),(2753,1085),(2446,1086),(2448,1086),(2512,1086),(2516,1086),(2522,1086),(2534,1086),(2536,1086),(2554,1086),(2556,1086),(2564,1086),(2670,1086),(2446,1087),(2448,1087),(2488,1087),(2503,1087),(2512,1087),(2514,1087),(2522,1087),(2536,1087),(2550,1087),(2554,1087),(2564,1087),(2570,1087),(2572,1087),(2624,1087),(2446,1088),(2562,1088),(2692,1089),(2707,1089),(2709,1089),(2711,1089),(2717,1089),(2719,1089),(2725,1089),(2692,1090),(2711,1090),(2753,1090),(2692,1091),(2694,1091),(2696,1091),(2715,1091),(2719,1091),(2723,1091),(2725,1091),(2753,1091),(2692,1092),(2942,1092),(2692,1093),(2753,1093),(2692,1094),(2715,1094),(2717,1094),(2753,1094),(2448,1095),(2488,1095),(2503,1095),(2512,1095),(2514,1095),(2516,1095),(2518,1095),(2522,1095),(2526,1095),(2530,1095),(2534,1095),(2536,1095),(2538,1095),(2542,1095),(2544,1095),(2548,1095),(2550,1095),(2554,1095),(2556,1095),(2558,1095),(2560,1095),(2562,1095),(2564,1095),(2566,1095),(2570,1095),(2572,1095),(2588,1095),(2592,1095),(2636,1095),(2644,1095),(2648,1095),(2692,1095),(2731,1095),(2733,1095),(2735,1095),(2743,1095),(2745,1095),(2747,1095),(2692,1096),(2696,1096),(2703,1096),(2705,1096),(2713,1096),(2717,1096),(2727,1096),(2692,1097),(2694,1097),(2709,1097),(2717,1097),(2692,1098),(2709,1098),(2723,1098),(2753,1098),(2692,1099),(2692,1100),(2709,1100),(2723,1100),(2696,1101),(2715,1101),(2721,1101),(2448,1102),(2503,1102),(2512,1102),(2448,1103),(2503,1103),(2512,1103),(2514,1103),(2516,1103),(2522,1103),(2534,1103),(2536,1103),(2546,1103),(2550,1103),(2554,1103),(2556,1103),(2558,1103),(2560,1103),(2562,1103),(2566,1103),(2596,1103),(2598,1103),(2612,1103),(2636,1103),(2642,1103),(2644,1103),(2648,1103),(2656,1103),(2662,1103),(2448,1104),(2476,1104),(2478,1104),(2448,1105),(2503,1105),(2512,1105),(2448,1106),(2503,1106),(2448,1107),(2503,1107),(2544,1107),(2564,1107),(2574,1107),(2682,1107),(2450,1108),(2450,1109),(2488,1109),(2450,1110),(2451,1111),(2451,1112),(2480,1112),(2482,1112),(2507,1112),(2508,1112),(2795,1112),(2451,1113),(2453,1114),(2453,1115),(2480,1115),(2482,1115),(2456,1116),(2458,1117),(2459,1117),(2459,1118),(2461,1119),(2467,1119),(2461,1120),(2464,1121),(2464,1122),(2464,1123),(2464,1124),(2464,1125),(2464,1126),(2464,1127),(2464,1128),(2464,1129),(2464,1130),(2466,1131),(2467,1132),(2467,1133),(2467,1134),(2470,1135),(2476,1136),(2478,1136),(2490,1136),(2516,1136),(2470,1137),(2478,1137),(2470,1138),(2478,1138),(2783,1139),(2486,1141),(2520,1141),(2528,1141),(2486,1142),(2488,1142),(2520,1142),(2522,1142),(2534,1142),(2486,1143),(2488,1143),(2520,1143),(2522,1143),(2524,1143),(2528,1143),(2534,1143),(2544,1143),(2745,1143),(2486,1144),(2524,1144),(2528,1144),(2834,1144),(2838,1144),(2566,1145),(2570,1145),(2490,1146),(2492,1146),(2510,1147),(2794,1147),(2480,1148),(2482,1148),(2507,1148),(2795,1148),(2507,1149),(2508,1149),(2795,1149),(2507,1150),(2795,1150),(2507,1151),(2795,1151),(2507,1152),(2795,1152),(2507,1153),(2795,1153),(2507,1154),(2795,1154),(2507,1155),(2795,1155),(2507,1156),(2508,1156),(2795,1156),(2509,1157),(2796,1157),(2509,1158),(2796,1158),(2495,1159),(2503,1160),(2503,1161),(2503,1162),(2508,1163),(2509,1164),(2512,1165),(2518,1165),(2512,1166),(2514,1166),(2516,1167),(2516,1168),(2514,1169),(2518,1169),(2518,1171),(2526,1172),(2530,1172),(2747,1172),(2526,1173),(2530,1173),(2747,1173),(2488,1174),(2526,1174),(2530,1174),(2536,1174),(2562,1174),(2648,1174),(2747,1174),(2488,1176),(2528,1176),(2488,1177),(2528,1177),(2532,1178),(2580,1178),(2586,1178),(2632,1178),(2658,1178),(2536,1179),(2536,1180),(2536,1181),(2538,1182),(2542,1182),(2612,1182),(2644,1182),(2731,1182),(2908,1182),(2540,1183),(2542,1186),(2544,1186),(2562,1186),(2566,1186),(2570,1186),(2636,1186),(2648,1186),(2743,1186),(2546,1187),(2582,1187),(2584,1187),(2600,1187),(2650,1187),(2546,1188),(2562,1188),(2564,1188),(2566,1188),(2584,1188),(2600,1188),(2614,1188),(2616,1188),(2618,1188),(2650,1188),(2546,1189),(2648,1189),(2550,1190),(2560,1190),(2562,1190),(2648,1190),(2550,1191),(2554,1191),(2550,1192),(2554,1192),(2566,1192),(2596,1192),(2598,1192),(2602,1192),(2644,1192),(2552,1193),(2558,1193),(2560,1193),(2562,1193),(2554,1194),(2558,1195),(2558,1196),(2562,1197),(2606,1197),(2564,1198),(2564,1199),(2566,1199),(2568,1200),(2570,1200),(2568,1201),(2570,1202),(28522,1202),(28528,1202),(28530,1202),(28534,1202),(28538,1202),(28570,1202),(28598,1202),(28600,1202),(28606,1202),(28624,1202),(28702,1202),(28716,1202),(28758,1202),(28774,1202),(28874,1202),(28902,1202),(28914,1202),(29046,1202),(29048,1202),(29050,1202),(2570,1203),(2644,1203),(2570,1204),(2804,1204),(2572,1205),(2572,1206),(2572,1207),(2576,1208),(2582,1209),(2584,1209),(2614,1209),(2616,1209),(2618,1209),(2620,1209),(2634,1209),(2590,1210),(2620,1210),(2592,1211),(2594,1212),(2604,1213),(2608,1214),(2652,1214),(2610,1215),(2612,1216),(2715,1216),(2618,1218),(2622,1219),(2622,1220),(2622,1221),(2626,1222),(2628,1222),(2630,1222),(2626,1223),(2628,1223),(2630,1223),(2638,1224),(2638,1225),(2842,1225),(2638,1226),(2640,1227),(2644,1228),(2654,1229),(2658,1230),(2660,1231),(2798,1232),(2536,1233),(2664,1233),(2666,1233),(2668,1233),(2670,1233),(2674,1233),(2578,1234),(2672,1234),(2680,1235),(2717,1235),(2930,1235),(2932,1235),(2682,1236),(2682,1237),(2682,1238),(2684,1238),(2686,1239),(2688,1240),(2690,1242),(2703,1243),(2938,1243),(2703,1244),(2759,1244),(2705,1245),(2705,1246),(2705,1247),(2826,1247),(2711,1248),(2713,1249),(2474,1250),(2699,1250),(2715,1250),(2723,1251),(2729,1252),(2729,1253),(2729,1254),(2729,1255),(2729,1256),(2733,1257),(2733,1258),(2733,1259),(2735,1260),(2735,1261),(2743,1262),(2743,1263),(2743,1264),(2745,1265),(2745,1266),(2749,1267),(2751,1268),(2757,1269),(2798,1269),(2757,1270),(28948,1801),(28952,1801),(29272,1801),(2882,2261),(29352,2816),(2488,3225),(2534,3225),(2554,3225),(2564,3225),(2636,3225),(2840,3243),(29280,3244),(26075,3322),(26090,3322),(26092,3322),(26096,3322),(26098,3322),(26100,3322),(26102,3322),(2534,5379),(2564,5379),(2802,5379),(2816,5677),(28918,5682),(29506,6367),(2662,6660),(29020,6691),(26080,6865),(26106,6894),(2922,7385),(28706,7526),(2858,7755),(26075,8073),(26106,8073),(2814,8158),(29162,8158),(26094,8237),(2864,8403),(26075,8431),(26090,8431),(26075,8432),(26075,8433),(26075,8434),(26088,8434),(26075,8435),(26075,8436),(26075,8437),(26075,8438),(26077,8438),(26090,8438),(26096,8438),(26075,8439),(26075,8440),(26077,8441),(26090,8441),(26096,8441),(26077,8442),(26077,8443),(26077,8444),(26083,8444),(26079,8445),(26112,8445),(26114,8445),(26079,8446),(26088,8449),(26104,8451),(26104,8452),(26106,8453),(26145,8453),(26108,8454),(26110,8454),(26116,8455),(26123,8455),(26116,8457),(26119,8457),(26116,8458),(26116,8459),(26139,8459),(26141,8459),(26147,8459),(26149,8459),(26116,8460),(26137,8460),(26125,8461),(26127,8462),(26129,8462),(26131,8462),(26133,8462),(26135,8462),(26143,8463),(28518,9939),(28518,9940),(28518,9941),(28520,9942),(28542,9942),(28910,9942),(29528,9942),(29530,9942),(28522,9943),(28658,9943),(28670,9943),(28702,9943),(28718,9943),(28720,9943),(28912,9943),(28922,9943),(28524,9945),(28526,9945),(28528,9945),(28530,9945),(28532,9945),(28534,9945),(28536,9945),(28538,9945),(28598,9945),(28716,9945),(28730,9945),(28732,9945),(28734,9945),(28736,9945),(28738,9945),(28740,9945),(28742,9945),(28744,9945),(28746,9945),(28524,9946),(28524,9947),(28524,9948),(28568,9948),(28570,9948),(28572,9948),(28574,9948),(28576,9948),(28578,9948),(28580,9948),(28582,9948),(28584,9948),(28586,9948),(28588,9948),(28590,9948),(28592,9948),(28594,9948),(28596,9948),(28598,9948),(28752,9948),(28754,9948),(28756,9948),(28524,9949),(28602,9949),(28648,9949),(28758,9949),(28760,9949),(28762,9949),(28764,9949),(28766,9949),(28736,9950),(28738,9950),(28744,9950),(28746,9950),(28806,9950),(28902,9950),(28914,9950),(29520,9950),(29522,9950),(29524,9950),(29526,9950),(28604,9951),(28736,9951),(28738,9951),(28744,9951),(28746,9951),(28806,9951),(28902,9951),(28906,9951),(28910,9951),(29520,9951),(29522,9951),(29524,9951),(29526,9951),(28540,9952),(28542,9952),(28544,9952),(28548,9952),(28550,9952),(28552,9952),(28554,9952),(28556,9952),(28558,9952),(28560,9952),(28562,9952),(28564,9952),(28748,9952),(28750,9952),(28540,9953),(28542,9953),(28540,9954),(28544,9954),(28546,9954),(28548,9954),(28550,9954),(28552,9954),(28554,9954),(28556,9954),(28558,9954),(28560,9954),(28562,9954),(28564,9954),(28566,9954),(28874,9954),(28876,9954),(28878,9954),(28880,9954),(28542,9955),(28910,9955),(28542,9956),(28550,9957),(28560,9957),(28562,9957),(28572,9958),(28574,9958),(28582,9958),(28588,9958),(28654,9958),(28814,9958),(28816,9958),(28818,9958),(28820,9958),(28822,9958),(28824,9958),(28826,9958),(28828,9958),(28578,9959),(28594,9959),(28604,9959),(28610,9959),(28614,9959),(28618,9959),(28650,9959),(28700,9959),(28704,9959),(28798,9959),(28800,9959),(28802,9959),(28804,9959),(28806,9959),(28808,9959),(28576,9960),(28654,9960),(28680,9960),(28578,9961),(28758,9961),(28578,9962),(28588,9963),(28592,9964),(28866,9964),(28868,9964),(28870,9964),(28594,9965),(28594,9966),(28596,9967),(28598,9967),(28526,9969),(28600,9969),(28602,9970),(28704,9970),(28764,9970),(28602,9971),(28834,9971),(28842,9971),(28856,9971),(28858,9971),(28860,9971),(28862,9971),(28864,9971),(28604,9972),(28604,9973),(28652,9973),(28606,9974),(28608,9975),(28610,9976),(28612,9977),(28614,9977),(28612,9978),(28612,9979),(28786,9979),(28616,9980),(28618,9981),(28618,9982),(28622,9983),(28624,9984),(28626,9985),(28626,9986),(28628,9987),(28630,9988),(28632,9989),(28634,9990),(28638,9990),(28772,9990),(28774,9990),(28776,9990),(28778,9990),(28780,9990),(28636,9991),(28640,9992),(28642,9993),(28674,9993),(28644,9994),(28646,9995),(28902,9995),(28648,9996),(28648,9997),(28768,9997),(28770,9997),(28650,9998),(29048,9998),(28654,9999),(28754,9999),(28810,9999),(28812,9999),(28656,10000),(28916,10000),(28918,10000),(28660,10001),(28662,10002),(28664,10003),(28668,10005),(28672,10006),(28782,10006),(28784,10006),(28676,10007),(28678,10008),(28872,10008),(28682,10009),(28868,10009),(28684,10010),(28686,10011),(28688,10012),(28692,10013),(28694,10014),(28696,10015),(28698,10016),(28700,10017),(28704,10018),(28708,10019),(28710,10020),(28712,10021),(28758,10021),(28714,10022),(28716,10023),(28722,10024),(28724,10025),(28726,10026),(28728,10027),(28770,10028),(28862,10028),(28782,10029),(28788,10030),(28790,10030),(28792,10030),(28794,10031),(28796,10032),(28814,10033),(28830,10033),(28832,10033),(28834,10033),(28836,10033),(28838,10034),(28840,10034),(28842,10034),(28864,10034),(28838,10035),(28844,10035),(28846,10035),(28848,10035),(28850,10035),(28852,10035),(28854,10035),(28842,10036),(28864,10036),(28882,10037),(28882,10038),(28882,10039),(29360,10039),(28883,10040),(28885,10041),(28885,10042),(28887,10043),(28887,10044),(28889,10045),(28889,10046),(28889,10047),(28889,10048),(28906,10048),(28889,10049),(28891,10050),(28893,10051),(28895,10051),(28897,10051),(28893,10052),(28895,10053),(28897,10053),(28895,10054),(28897,10054),(28899,10055),(28899,10056),(28899,10057),(28899,10058),(28899,10059),(28902,10060),(28906,10061),(28910,10061),(28906,10062),(28910,10062),(28906,10063),(28906,10064),(28906,10065),(28907,10066),(28907,10067),(28907,10068),(28910,10070),(28914,10071),(28918,10072),(29528,10073),(29530,10073),(28920,10074),(28924,10075),(28926,10076),(2514,10701),(2480,10807),(2482,10807),(2640,10808),(2540,10809),(2544,10809),(2646,10809),(2648,10809),(2804,10809),(2652,10810),(2858,10810),(2660,10811),(2664,10812),(2672,10812),(2676,10812),(2678,10812),(2761,10812),(2763,10812),(2765,10812),(2767,10812),(2769,10812),(2771,10812),(2773,10812),(2775,10812),(2777,10812),(2672,10813),(2488,10814),(2520,10814),(2558,10814),(2562,10814),(2564,10814),(2572,10814),(2614,10814),(2636,10814),(2644,10814),(2682,10814),(2733,10814),(2749,10814),(2806,10814),(2818,10814),(2836,10814),(28936,10874),(29054,10893),(29200,10893),(2832,10956),(29156,10956),(2946,11059),(26149,11190),(2779,11204),(2800,11204),(2786,11227),(2672,11228),(2790,11228),(2476,11230),(2476,11231),(2476,11232),(2476,11233),(2476,11234),(2476,11235),(2478,11235),(2478,11236),(2478,11237),(2478,11238),(2478,11239),(2478,11240),(2478,11241),(2478,11242),(2499,11243),(2499,11244),(2499,11245),(2703,11246),(2705,11246),(2717,11246),(2512,11376),(28756,11454),(26082,11461),(2480,11480),(2482,11480),(2480,11498),(2482,11498),(2548,11500),(2757,11500),(28528,11512),(28954,11512),(29094,11512),(28530,11513),(28536,11514),(28540,11548),(28636,11560),(28576,11561),(28666,11623),(28690,11872),(28700,11873),(2510,11874),(2838,11910),(2862,11910),(28928,11918),(28928,11919),(28928,11920),(28522,11921),(28686,11921),(28870,11921),(28904,11921),(28930,11921),(28932,11921),(28934,11921),(28936,11921),(28938,11921),(28940,11921),(28942,11921),(28944,11921),(28946,11921),(28948,11921),(28950,11921),(28952,11921),(28954,11921),(28956,11921),(28958,11921),(28960,11921),(28964,11921),(28966,11921),(28968,11921),(28970,11921),(28972,11921),(28974,11921),(28976,11921),(28978,11921),(28980,11921),(28982,11921),(28984,11921),(28986,11921),(28988,11921),(28990,11921),(28992,11921),(28996,11921),(28998,11921),(29002,11921),(29004,11921),(29006,11921),(29008,11921),(29010,11921),(29012,11921),(29014,11921),(29016,11921),(29018,11921),(29020,11921),(29022,11921),(29024,11921),(29026,11921),(29028,11921),(29030,11921),(29032,11921),(29034,11921),(29036,11921),(29038,11921),(29040,11921),(29042,11921),(29044,11921),(29046,11921),(29048,11921),(29050,11921),(29052,11921),(29054,11921),(29056,11921),(29058,11921),(29060,11921),(29062,11921),(29064,11921),(29066,11921),(29068,11921),(29070,11921),(29072,11921),(29076,11921),(29078,11921),(29080,11921),(29082,11921),(29084,11921),(29086,11921),(29088,11921),(29090,11921),(29092,11921),(29094,11921),(29096,11921),(29098,11921),(29102,11921),(29104,11921),(29106,11921),(29108,11921),(29110,11921),(29112,11921),(29114,11921),(29116,11921),(29118,11921),(29120,11921),(29122,11921),(29124,11921),(29126,11921),(29128,11921),(29130,11921),(29132,11921),(29134,11921),(29136,11921),(29138,11921),(29140,11921),(29142,11921),(29144,11921),(29146,11921),(29148,11921),(29150,11921),(29152,11921),(29154,11921),(29156,11921),(29158,11921),(29160,11921),(29164,11921),(29166,11921),(29168,11921),(29170,11921),(29172,11921),(29174,11921),(29176,11921),(29178,11921),(29180,11921),(29184,11921),(29186,11921),(29188,11921),(29190,11921),(29192,11921),(29194,11921),(29196,11921),(29198,11921),(29200,11921),(29202,11921),(29204,11921),(29206,11921),(29208,11921),(29210,11921),(29212,11921),(29214,11921),(29216,11921),(29218,11921),(29222,11921),(29224,11921),(29226,11921),(29228,11921),(29230,11921),(29232,11921),(29234,11921),(29236,11921),(29238,11921),(29240,11921),(29242,11921),(29244,11921),(29246,11921),(29248,11921),(29250,11921),(29252,11921),(29254,11921),(29256,11921),(29258,11921),(29260,11921),(29262,11921),(29264,11921),(29266,11921),(29268,11921),(29270,11921),(29272,11921),(29274,11921),(29276,11921),(29278,11921),(29280,11921),(29282,11921),(29284,11921),(29286,11921),(29288,11921),(29290,11921),(29292,11921),(29294,11921),(29296,11921),(29298,11921),(29300,11921),(29302,11921),(29304,11921),(29306,11921),(29308,11921),(29310,11921),(29312,11921),(29314,11921),(29316,11921),(29318,11921),(29320,11921),(29322,11921),(29324,11921),(29326,11921),(29328,11921),(29330,11921),(29332,11921),(29334,11921),(29336,11921),(29338,11921),(29340,11921),(29342,11921),(29344,11921),(29346,11921),(29348,11921),(29350,11921),(29354,11921),(29358,11921),(29360,11921),(29362,11921),(29364,11921),(29366,11921),(29368,11921),(29370,11921),(29372,11921),(29374,11921),(29376,11921),(29378,11921),(29380,11921),(29382,11921),(29384,11921),(29386,11921),(29388,11921),(29390,11921),(29392,11921),(29394,11921),(29396,11921),(29398,11921),(29400,11921),(29402,11921),(29404,11921),(29406,11921),(29408,11921),(29410,11921),(29412,11921),(29414,11921),(29416,11921),(29418,11921),(29420,11921),(29422,11921),(29424,11921),(29426,11921),(29428,11921),(29430,11921),(29432,11921),(29434,11921),(29436,11921),(29438,11921),(29440,11921),(29442,11921),(29444,11921),(29446,11921),(29448,11921),(29450,11921),(29452,11921),(29454,11921),(29456,11921),(29458,11921),(29460,11921),(29462,11921),(29464,11921),(29466,11921),(29470,11921),(29472,11921),(29474,11921),(29476,11921),(29478,11921),(29480,11921),(29482,11921),(29484,11921),(29486,11921),(29488,11921),(29490,11921),(29492,11921),(29494,11921),(29496,11921),(29498,11921),(29500,11921),(29504,11921),(29506,11921),(29510,11921),(29512,11921),(29514,11921),(29516,11921),(29518,11921),(29538,11921),(28522,11922),(28686,11922),(28870,11922),(28904,11922),(28930,11922),(28932,11922),(28934,11922),(28936,11922),(28938,11922),(28940,11922),(28942,11922),(28944,11922),(28946,11922),(28948,11922),(28950,11922),(28952,11922),(28954,11922),(28956,11922),(28958,11922),(28960,11922),(28964,11922),(28966,11922),(28968,11922),(28970,11922),(28972,11922),(28974,11922),(28976,11922),(28978,11922),(28980,11922),(28982,11922),(28984,11922),(28986,11922),(28988,11922),(28990,11922),(28992,11922),(28996,11922),(28998,11922),(29002,11922),(29004,11922),(29006,11922),(29008,11922),(29010,11922),(29012,11922),(29014,11922),(29016,11922),(29018,11922),(29020,11922),(29022,11922),(29024,11922),(29026,11922),(29028,11922),(29030,11922),(29032,11922),(29034,11922),(29036,11922),(29038,11922),(29040,11922),(29042,11922),(29044,11922),(29046,11922),(29048,11922),(29050,11922),(29052,11922),(29054,11922),(29056,11922),(29058,11922),(29060,11922),(29062,11922),(29064,11922),(29066,11922),(29068,11922),(29070,11922),(29072,11922),(29076,11922),(29078,11922),(29080,11922),(29082,11922),(29084,11922),(29086,11922),(29088,11922),(29090,11922),(29092,11922),(29094,11922),(29096,11922),(29098,11922),(29102,11922),(29104,11922),(29106,11922),(29108,11922),(29110,11922),(29112,11922),(29114,11922),(29116,11922),(29118,11922),(29120,11922),(29122,11922),(29124,11922),(29126,11922),(29128,11922),(29130,11922),(29132,11922),(29134,11922),(29136,11922),(29138,11922),(29140,11922),(29142,11922),(29144,11922),(29146,11922),(29148,11922),(29150,11922),(29152,11922),(29154,11922),(29156,11922),(29158,11922),(29160,11922),(29164,11922),(29166,11922),(29168,11922),(29170,11922),(29172,11922),(29174,11922),(29176,11922),(29178,11922),(29180,11922),(29184,11922),(29186,11922),(29188,11922),(29190,11922),(29192,11922),(29194,11922),(29196,11922),(29198,11922),(29200,11922),(29202,11922),(29204,11922),(29206,11922),(29208,11922),(29210,11922),(29212,11922),(29214,11922),(29216,11922),(29218,11922),(29222,11922),(29224,11922),(29226,11922),(29228,11922),(29230,11922),(29232,11922),(29234,11922),(29236,11922),(29238,11922),(29240,11922),(29242,11922),(29244,11922),(29246,11922),(29248,11922),(29250,11922),(29252,11922),(29254,11922),(29256,11922),(29258,11922),(29260,11922),(29262,11922),(29264,11922),(29266,11922),(29268,11922),(29270,11922),(29272,11922),(29274,11922),(29276,11922),(29278,11922),(29280,11922),(29282,11922),(29284,11922),(29286,11922),(29288,11922),(29290,11922),(29292,11922),(29294,11922),(29296,11922),(29298,11922),(29300,11922),(29302,11922),(29304,11922),(29306,11922),(29308,11922),(29310,11922),(29312,11922),(29314,11922),(29316,11922),(29318,11922),(29320,11922),(29322,11922),(29324,11922),(29326,11922),(29328,11922),(29330,11922),(29332,11922),(29334,11922),(29336,11922),(29338,11922),(29340,11922),(29342,11922),(29344,11922),(29346,11922),(29348,11922),(29350,11922),(29354,11922),(29358,11922),(29360,11922),(29362,11922),(29364,11922),(29366,11922),(29368,11922),(29370,11922),(29372,11922),(29374,11922),(29376,11922),(29378,11922),(29380,11922),(29382,11922),(29384,11922),(29386,11922),(29388,11922),(29390,11922),(29392,11922),(29394,11922),(29396,11922),(29398,11922),(29400,11922),(29402,11922),(29404,11922),(29406,11922),(29408,11922),(29410,11922),(29412,11922),(29414,11922),(29416,11922),(29418,11922),(29420,11922),(29422,11922),(29424,11922),(29426,11922),(29428,11922),(29430,11922),(29432,11922),(29434,11922),(29436,11922),(29438,11922),(29440,11922),(29442,11922),(29444,11922),(29446,11922),(29448,11922),(29450,11922),(29452,11922),(29454,11922),(29456,11922),(29458,11922),(29460,11922),(29462,11922),(29464,11922),(29466,11922),(29470,11922),(29472,11922),(29474,11922),(29476,11922),(29478,11922),(29480,11922),(29482,11922),(29484,11922),(29486,11922),(29488,11922),(29490,11922),(29492,11922),(29494,11922),(29496,11922),(29498,11922),(29500,11922),(29504,11922),(29506,11922),(29510,11922),(29512,11922),(29514,11922),(29516,11922),(29518,11922),(29532,11922),(29534,11922),(29536,11922),(29538,11922),(28930,11923),(28930,11924),(28930,11925),(28932,11926),(28522,11927),(28686,11927),(28904,11927),(28932,11927),(28936,11927),(28938,11927),(28940,11927),(28942,11927),(28944,11927),(28946,11927),(28948,11927),(28950,11927),(28952,11927),(28954,11927),(28956,11927),(28960,11927),(28962,11927),(28964,11927),(28966,11927),(28968,11927),(28970,11927),(28972,11927),(28974,11927),(28976,11927),(28978,11927),(28980,11927),(28982,11927),(28984,11927),(28986,11927),(28988,11927),(28990,11927),(28992,11927),(28994,11927),(28996,11927),(29000,11927),(29002,11927),(29004,11927),(29006,11927),(29008,11927),(29010,11927),(29012,11927),(29014,11927),(29016,11927),(29018,11927),(29020,11927),(29022,11927),(29024,11927),(29026,11927),(29028,11927),(29030,11927),(29032,11927),(29034,11927),(29040,11927),(29042,11927),(29044,11927),(29046,11927),(29048,11927),(29050,11927),(29054,11927),(29056,11927),(29058,11927),(29060,11927),(29064,11927),(29066,11927),(29068,11927),(29070,11927),(29072,11927),(29074,11927),(29076,11927),(29078,11927),(29084,11927),(29086,11927),(29088,11927),(29090,11927),(29096,11927),(29100,11927),(29102,11927),(29108,11927),(29110,11927),(29112,11927),(29114,11927),(29118,11927),(29120,11927),(29124,11927),(29126,11927),(29128,11927),(29130,11927),(29132,11927),(29134,11927),(29136,11927),(29138,11927),(29140,11927),(29142,11927),(29144,11927),(29148,11927),(29152,11927),(29154,11927),(29160,11927),(29162,11927),(29164,11927),(29168,11927),(29170,11927),(29172,11927),(29178,11927),(29180,11927),(29182,11927),(29186,11927),(29194,11927),(29196,11927),(29198,11927),(29200,11927),(29202,11927),(29204,11927),(29206,11927),(29208,11927),(29210,11927),(29212,11927),(29214,11927),(29216,11927),(29218,11927),(29220,11927),(29222,11927),(29224,11927),(29226,11927),(29228,11927),(29234,11927),(29236,11927),(29238,11927),(29240,11927),(29242,11927),(29244,11927),(29246,11927),(29248,11927),(29250,11927),(29252,11927),(29254,11927),(29256,11927),(29258,11927),(29260,11927),(29262,11927),(29264,11927),(29266,11927),(29268,11927),(29270,11927),(29272,11927),(29274,11927),(29276,11927),(29278,11927),(29284,11927),(29290,11927),(29292,11927),(29294,11927),(29296,11927),(29298,11927),(29300,11927),(29302,11927),(29304,11927),(29306,11927),(29310,11927),(29312,11927),(29314,11927),(29316,11927),(29318,11927),(29320,11927),(29322,11927),(29324,11927),(29326,11927),(29328,11927),(29330,11927),(29332,11927),(29334,11927),(29336,11927),(29338,11927),(29340,11927),(29342,11927),(29344,11927),(29346,11927),(29348,11927),(29350,11927),(29352,11927),(29354,11927),(29356,11927),(29360,11927),(29362,11927),(29364,11927),(29366,11927),(29368,11927),(29370,11927),(29372,11927),(29374,11927),(29376,11927),(29378,11927),(29380,11927),(29382,11927),(29384,11927),(29386,11927),(29388,11927),(29390,11927),(29392,11927),(29394,11927),(29396,11927),(29398,11927),(29400,11927),(29402,11927),(29404,11927),(29406,11927),(29408,11927),(29410,11927),(29412,11927),(29414,11927),(29418,11927),(29420,11927),(29422,11927),(29424,11927),(29426,11927),(29428,11927),(29430,11927),(29432,11927),(29434,11927),(29436,11927),(29440,11927),(29442,11927),(29444,11927),(29446,11927),(29448,11927),(29452,11927),(29454,11927),(29456,11927),(29458,11927),(29462,11927),(29468,11927),(29470,11927),(29472,11927),(29476,11927),(29478,11927),(29480,11927),(29482,11927),(29486,11927),(29488,11927),(29490,11927),(29492,11927),(29494,11927),(29496,11927),(29498,11927),(29504,11927),(29506,11927),(29508,11927),(29510,11927),(29512,11927),(29514,11927),(29516,11927),(29518,11927),(29532,11927),(29534,11927),(29536,11927),(29538,11927),(28932,11928),(28932,11929),(28522,11930),(28686,11930),(28904,11930),(28932,11930),(28936,11930),(28938,11930),(28940,11930),(28942,11930),(28944,11930),(28946,11930),(28948,11930),(28950,11930),(28952,11930),(28954,11930),(28956,11930),(28960,11930),(28962,11930),(28964,11930),(28966,11930),(28968,11930),(28970,11930),(28972,11930),(28974,11930),(28976,11930),(28978,11930),(28980,11930),(28982,11930),(28984,11930),(28986,11930),(28988,11930),(28990,11930),(28992,11930),(28994,11930),(28996,11930),(29000,11930),(29002,11930),(29004,11930),(29006,11930),(29008,11930),(29010,11930),(29012,11930),(29014,11930),(29016,11930),(29018,11930),(29020,11930),(29022,11930),(29024,11930),(29026,11930),(29028,11930),(29030,11930),(29032,11930),(29034,11930),(29040,11930),(29042,11930),(29044,11930),(29046,11930),(29048,11930),(29050,11930),(29054,11930),(29056,11930),(29058,11930),(29060,11930),(29064,11930),(29066,11930),(29068,11930),(29070,11930),(29072,11930),(29074,11930),(29076,11930),(29078,11930),(29084,11930),(29086,11930),(29088,11930),(29090,11930),(29096,11930),(29100,11930),(29102,11930),(29108,11930),(29110,11930),(29112,11930),(29114,11930),(29118,11930),(29120,11930),(29124,11930),(29126,11930),(29128,11930),(29130,11930),(29132,11930),(29134,11930),(29136,11930),(29138,11930),(29140,11930),(29142,11930),(29144,11930),(29148,11930),(29152,11930),(29154,11930),(29160,11930),(29162,11930),(29164,11930),(29168,11930),(29170,11930),(29172,11930),(29178,11930),(29180,11930),(29182,11930),(29186,11930),(29194,11930),(29196,11930),(29198,11930),(29200,11930),(29202,11930),(29204,11930),(29206,11930),(29208,11930),(29210,11930),(29212,11930),(29214,11930),(29216,11930),(29218,11930),(29220,11930),(29222,11930),(29224,11930),(29226,11930),(29228,11930),(29234,11930),(29236,11930),(29238,11930),(29240,11930),(29242,11930),(29244,11930),(29246,11930),(29248,11930),(29250,11930),(29252,11930),(29254,11930),(29256,11930),(29258,11930),(29260,11930),(29262,11930),(29264,11930),(29266,11930),(29268,11930),(29270,11930),(29272,11930),(29274,11930),(29276,11930),(29278,11930),(29284,11930),(29290,11930),(29292,11930),(29294,11930),(29296,11930),(29298,11930),(29300,11930),(29302,11930),(29304,11930),(29306,11930),(29310,11930),(29312,11930),(29314,11930),(29316,11930),(29318,11930),(29320,11930),(29322,11930),(29324,11930),(29326,11930),(29328,11930),(29330,11930),(29332,11930),(29334,11930),(29336,11930),(29338,11930),(29340,11930),(29342,11930),(29344,11930),(29346,11930),(29348,11930),(29350,11930),(29352,11930),(29354,11930),(29356,11930),(29360,11930),(29362,11930),(29364,11930),(29366,11930),(29368,11930),(29370,11930),(29372,11930),(29374,11930),(29376,11930),(29378,11930),(29380,11930),(29382,11930),(29384,11930),(29386,11930),(29388,11930),(29390,11930),(29392,11930),(29394,11930),(29396,11930),(29398,11930),(29400,11930),(29402,11930),(29404,11930),(29406,11930),(29408,11930),(29410,11930),(29412,11930),(29414,11930),(29418,11930),(29420,11930),(29422,11930),(29424,11930),(29426,11930),(29428,11930),(29430,11930),(29432,11930),(29434,11930),(29436,11930),(29440,11930),(29442,11930),(29444,11930),(29446,11930),(29448,11930),(29452,11930),(29454,11930),(29456,11930),(29458,11930),(29462,11930),(29468,11930),(29470,11930),(29472,11930),(29476,11930),(29478,11930),(29480,11930),(29482,11930),(29486,11930),(29488,11930),(29490,11930),(29492,11930),(29494,11930),(29496,11930),(29498,11930),(29504,11930),(29506,11930),(29508,11930),(29510,11930),(29512,11930),(29514,11930),(29516,11930),(29518,11930),(29532,11930),(29534,11930),(29538,11930),(28932,11931),(28934,11932),(28934,11933),(28934,11934),(28936,11935),(28936,11936),(29144,11936),(28936,11937),(28938,11938),(28938,11939),(28938,11940),(28938,11941),(28940,11942),(28940,11943),(28940,11944),(28942,11945),(28942,11946),(29078,11946),(29300,11946),(28942,11947),(28944,11948),(28944,11949),(28944,11950),(28944,11951),(28946,11952),(28946,11953),(28946,11954),(28948,11955),(28948,11956),(28948,11957),(29458,11957),(28950,11958),(28950,11959),(28950,11960),(28952,11961),(28952,11962),(28952,11963),(28954,11964),(28974,11964),(28976,11964),(29284,11964),(29516,11964),(28954,11965),(28954,11966),(28956,11967),(29142,11967),(28956,11968),(28982,11968),(28956,11969),(28958,11970),(28958,11971),(29048,11971),(29480,11971),(28958,11972),(28960,11973),(28960,11974),(29310,11974),(28960,11975),(2800,11989),(29502,12018),(2806,12029),(2918,12042),(29256,12096),(28964,12110),(28966,12110),(29430,12110),(29494,12110),(28964,12111),(28966,12111),(28968,12112),(28968,12113),(29380,12113),(29474,12113),(28968,12114),(28970,12115),(28970,12116),(28970,12117),(29042,12117),(29274,12117),(28972,12118),(28978,12118),(28972,12119),(28972,12120),(28974,12121),(28976,12121),(28974,12122),(28976,12122),(28974,12123),(28976,12123),(29108,12123),(28978,12124),(29170,12124),(28980,12125),(29198,12125),(28980,12126),(28980,12127),(29206,12127),(28982,12128),(28982,12129),(28984,12130),(28984,12131),(29402,12131),(28984,12132),(28986,12133),(28986,12134),(28986,12135),(29376,12135),(28988,12136),(29382,12136),(28990,12137),(28992,12138),(28996,12139),(28994,12140),(29002,12141),(29102,12141),(29002,12142),(29000,12143),(28998,12144),(29004,12145),(29504,12145),(29004,12146),(29006,12147),(29008,12148),(29010,12149),(29012,12150),(29014,12151),(29340,12151),(29014,12152),(29016,12153),(29062,12153),(29016,12154),(29018,12174),(29018,12175),(29022,12176),(29058,12176),(29020,12177),(29028,12178),(29030,12178),(29026,12179),(29024,12180),(29034,12181),(29034,12182),(29032,12183),(29038,12183),(29044,12184),(29040,12185),(29040,12186),(29038,12187),(29036,12188),(29052,12189),(29050,12190),(29046,12191),(29058,12192),(29056,12193),(29308,12193),(29466,12193),(29060,12194),(29062,12195),(29066,12196),(29064,12197),(29064,12198),(29072,12199),(29070,12200),(29068,12201),(29082,12202),(29080,12203),(29076,12204),(29074,12205),(29092,12206),(29090,12207),(29216,12207),(29088,12208),(29086,12209),(29084,12210),(29490,12210),(29098,12211),(29096,12212),(29112,12213),(29110,12214),(29106,12215),(29104,12216),(29354,12216),(29120,12217),(29118,12218),(29116,12219),(29114,12220),(29132,12221),(29130,12222),(29128,12223),(29126,12224),(29392,12224),(29124,12225),(29140,12226),(29138,12227),(29136,12228),(29134,12229),(29368,12229),(29152,12230),(29150,12231),(29148,12232),(29146,12233),(29254,12233),(29160,12234),(29164,12234),(29158,12235),(29154,12236),(29172,12237),(29168,12238),(29182,12238),(29166,12239),(29176,12240),(29174,12241),(29178,12383),(29188,12384),(29184,12385),(29180,12386),(29196,12387),(29194,12388),(29192,12389),(29190,12390),(29208,12391),(29204,12392),(29202,12393),(29214,12394),(29212,12395),(29210,12396),(29218,12397),(29220,12398),(29224,12399),(29222,12400),(29226,12401),(29228,12402),(29230,12403),(29236,12404),(29234,12405),(29240,12406),(29238,12407),(29244,12408),(29242,12409),(29246,12410),(29248,12411),(29250,12412),(29322,12412),(29384,12412),(29252,12413),(29258,12414),(29260,12415),(29262,12416),(29338,12416),(29264,12417),(29266,12418),(29268,12419),(29278,12419),(29270,12420),(29276,12421),(29282,12422),(29288,12423),(29418,12423),(29286,12424),(29290,12425),(29294,12426),(29292,12427),(29298,12428),(29296,12429),(29304,12430),(29302,12431),(29306,12432),(29312,12433),(29316,12433),(29390,12433),(29314,12434),(29318,12435),(29320,12436),(29324,12437),(29326,12438),(29328,12439),(29330,12440),(29332,12441),(29334,12442),(29336,12443),(29342,12444),(29346,12445),(29344,12446),(29350,12447),(29348,12448),(29358,12449),(29450,12449),(29456,12449),(29356,12450),(29364,12451),(29362,12452),(29370,12453),(29366,12454),(29372,12455),(29374,12456),(29378,12457),(29386,12458),(29396,12459),(29394,12460),(29478,12460),(29388,12461),(29404,12462),(29400,12463),(29398,12464),(29406,12465),(29408,12466),(29412,12467),(29410,12468),(29416,12469),(29414,12470),(29420,12471),(29422,12472),(29426,12473),(29424,12474),(29432,12475),(29428,12476),(29436,12477),(29434,12478),(29440,12479),(29438,12480),(29442,12481),(29444,12482),(29446,12482),(29452,12483),(29448,12484),(29460,12485),(29454,12486),(29464,12487),(29462,12488),(29472,12489),(29470,12490),(29476,12491),(29488,12492),(29486,12493),(29512,12493),(29484,12494),(29482,12495),(29500,12496),(29498,12497),(29496,12498),(29492,12499),(29510,12500),(29514,12501),(29232,12556),(26151,12628),(2540,12653),(2540,12654),(2542,12655),(2542,12656),(2544,12657),(2486,12658),(2520,12658),(2532,12658),(2546,12658),(2552,12658),(2584,12658),(2598,12658),(2604,12658),(2610,12658),(2616,12658),(2622,12658),(2642,12658),(2658,12658),(2670,12658),(2672,12658),(2548,12659),(2548,12660),(2638,12660),(2522,12661),(2550,12661),(2554,12661),(2556,12661),(2562,12661),(2564,12661),(2566,12661),(2636,12661),(2552,12662),(2556,12664),(2512,12665),(2536,12665),(2556,12665),(2562,12665),(2648,12665),(2512,12666),(2514,12666),(2558,12666),(2562,12666),(2560,12668),(2560,12669),(2560,12670),(2560,12671),(2560,12672),(2562,12673),(2634,12673),(2648,12673),(2562,12674),(2520,12675),(2522,12675),(2564,12675),(2572,12675),(2638,12675),(2582,12676),(2620,12677),(2642,12678),(2642,12679),(2652,12680),(2654,12681),(2654,12682),(2711,12683),(2930,12683),(2932,12683),(2759,12684),(2940,12684),(2488,12753),(2488,12754),(2512,12755),(2512,12756),(2512,12757),(2512,12758),(2514,12759),(2514,12760),(2520,12761),(2532,12762),(2534,12763),(2524,12764),(2534,12764),(2570,12764),(2696,12764),(2590,12765),(2596,12766),(2600,12766),(2610,12767),(2606,12768),(2614,12769),(2680,12769),(2632,12770),(2634,12771),(2636,12772),(2650,12773),(2650,12774),(2808,12774),(2656,12776),(2664,12777),(2672,12778),(2672,12779),(2682,12780),(2735,12781),(2840,12781),(2866,12781),(2739,12782),(2739,12783),(2844,12799),(2765,12855),(2802,12856),(2804,12858),(2804,12859),(2808,12860),(2860,12860),(28904,12940),(2820,12942),(2828,12943),(2850,12943),(2852,12943),(2854,12943),(2856,12943),(2870,12943),(2900,12943),(2830,12944),(2844,12949),(2848,12950),(2846,12951),(2846,12952),(2862,12953),(2872,12954),(2868,12955),(2874,12959),(2876,12960),(2890,12961),(2888,12962),(2908,12963),(2934,12964),(28522,13042),(2480,13117),(29518,13983),(29518,13984),(29518,13985),(26116,14001),(26116,14002),(2454,14086),(26145,14275),(2802,14710),(2802,14711),(2501,14822),(2701,14822),(28518,15040),(28518,15041),(28518,15042),(28518,15043),(28868,15044),(2484,15075),(28758,15214),(28758,15215),(28758,15216),(28758,15217),(28758,15218),(28758,15219),(28758,15220),(2480,15255),(2482,15255),(2482,15256),(2526,15257),(2530,15257),(2747,15257),(2530,15258),(2747,15258),(2536,15259),(2556,15259),(2560,15260),(2656,15261),(2676,15261),(2678,15261),(2804,15262),(26080,15449),(26080,15450),(26081,15451),(26090,15452),(26147,15453),(26149,15454),(28620,15465),(28907,15466),(2682,15521),(26118,15676),(26118,15677),(26119,15679),(26119,15680),(26120,15680),(26151,15680),(26153,15680),(26154,15680),(26155,15680),(26119,15681),(26120,15682),(26151,15683),(26153,15683),(26155,15683),(26151,15684),(26153,15684),(26151,15685),(26153,15685),(26154,15685),(2460,15851),(2492,15924),(2492,15925),(28606,16000),(28738,16000),(29532,16000),(29534,16000),(28620,16001),(28620,16002),(28620,16003),(29532,16004),(29534,16005),(26084,16240),(28870,16273);
/*!40000 ALTER TABLE `evidence_article` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `evidence_cancer_type`
--

DROP TABLE IF EXISTS `evidence_cancer_type`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `evidence_cancer_type` (
  `evidence_id` int NOT NULL,
  `cancer_type_id` int NOT NULL,
  PRIMARY KEY (`evidence_id`,`cancer_type_id`),
  KEY `FKE65A5BB14C5916CF` (`cancer_type_id`),
  KEY `FKE65A5BB1F7262935` (`evidence_id`),
  CONSTRAINT `FKE65A5BB14C5916CF` FOREIGN KEY (`cancer_type_id`) REFERENCES `cancer_type` (`id`),
  CONSTRAINT `FKE65A5BB1F7262935` FOREIGN KEY (`evidence_id`) REFERENCES `evidence` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `evidence_cancer_type`
--

LOCK TABLES `evidence_cancer_type` WRITE;
/*!40000 ALTER TABLE `evidence_cancer_type` DISABLE KEYS */;
INSERT INTO `evidence_cancer_type` VALUES (28883,17),(28884,17),(2469,32),(2470,32),(2477,54),(2478,54),(2496,83),(2505,83),(28895,185),(28896,185),(2483,259),(2484,259),(2500,259),(2501,259),(2700,259),(2701,259),(2463,276),(2464,276),(2465,276),(2466,276),(28891,355),(28892,355),(2791,367),(2792,367),(2479,393),(2480,393),(2469,407),(2470,407),(2473,407),(2474,407),(2698,407),(2699,407),(2455,409),(2456,409),(28887,418),(28888,418),(2449,453),(2450,453),(2451,453),(2494,453),(2495,453),(2504,453),(2507,453),(2508,453),(2509,453),(2510,453),(2523,453),(2524,453),(2527,453),(2528,453),(2695,453),(2696,453),(2781,453),(2793,453),(2794,453),(2795,453),(2796,453),(28885,527),(28886,527),(28889,533),(28890,533),(28899,586),(28900,586),(2461,611),(2462,611),(2489,611),(2490,611),(2787,611),(2788,611),(2475,619),(2476,619),(28897,632),(28898,632),(28907,645),(28908,645),(2469,669),(2470,669),(28893,695),(28894,695),(2783,857),(2784,857),(2467,862),(2468,862),(2491,862),(2492,862),(2789,862),(2790,862),(2452,876),(2453,876),(2454,876),(2780,876),(26078,876),(26079,876),(26080,876),(26081,876),(26082,876),(26083,876),(26084,876),(26117,876),(26118,876),(26119,876),(26120,876),(26152,876),(26153,876),(26154,876),(26155,876),(2693,880),(2694,880),(2479,897),(2480,897),(2471,934),(2498,934),(2499,934),(2785,934),(2786,934),(2457,935),(2458,935),(2459,935),(2460,935),(2493,935),(26086,982),(2481,984),(2482,984),(2526,984),(2530,984),(2746,984),(2747,984),(28619,984),(28620,984),(2472,986),(2497,986),(2506,986),(2525,986),(2529,986),(2697,986),(2782,986),(26085,986),(26121,986),(26156,986);
/*!40000 ALTER TABLE `evidence_cancer_type` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `evidence_excluded_cancer_type`
--

DROP TABLE IF EXISTS `evidence_excluded_cancer_type`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `evidence_excluded_cancer_type` (
  `evidence_id` int NOT NULL,
  `cancer_type_id` int NOT NULL,
  PRIMARY KEY (`evidence_id`,`cancer_type_id`),
  KEY `FK3FC7AF8C4C5916CF` (`cancer_type_id`),
  KEY `FK3FC7AF8CF7262935` (`evidence_id`),
  CONSTRAINT `FK3FC7AF8C4C5916CF` FOREIGN KEY (`cancer_type_id`) REFERENCES `cancer_type` (`id`),
  CONSTRAINT `FK3FC7AF8CF7262935` FOREIGN KEY (`evidence_id`) REFERENCES `evidence` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `evidence_excluded_cancer_type`
--

LOCK TABLES `evidence_excluded_cancer_type` WRITE;
/*!40000 ALTER TABLE `evidence_excluded_cancer_type` DISABLE KEYS */;
INSERT INTO `evidence_excluded_cancer_type` VALUES (2475,223),(2476,223),(2475,296),(2476,296),(2475,332),(2476,332),(2475,635),(2476,635),(2475,753),(2476,753),(2475,855),(2476,855),(2481,935),(2482,935);
/*!40000 ALTER TABLE `evidence_excluded_cancer_type` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `evidence_relevant_cancer_type`
--

DROP TABLE IF EXISTS `evidence_relevant_cancer_type`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `evidence_relevant_cancer_type` (
  `evidence_id` int NOT NULL,
  `cancer_type_id` int NOT NULL,
  PRIMARY KEY (`evidence_id`,`cancer_type_id`),
  KEY `FK20D24FDF4C5916CF` (`cancer_type_id`),
  KEY `FK20D24FDFF7262935` (`evidence_id`),
  CONSTRAINT `FK20D24FDF4C5916CF` FOREIGN KEY (`cancer_type_id`) REFERENCES `cancer_type` (`id`),
  CONSTRAINT `FK20D24FDFF7262935` FOREIGN KEY (`evidence_id`) REFERENCES `evidence` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `evidence_relevant_cancer_type`
--

LOCK TABLES `evidence_relevant_cancer_type` WRITE;
/*!40000 ALTER TABLE `evidence_relevant_cancer_type` DISABLE KEYS */;
/*!40000 ALTER TABLE `evidence_relevant_cancer_type` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `gene`
--

DROP TABLE IF EXISTS `gene`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `gene` (
  `entrez_gene_id` int NOT NULL,
  `gene_type` varchar(30) DEFAULT NULL,
  `grch37_isoform` varchar(100) DEFAULT NULL,
  `grch37_ref_seq` varchar(100) DEFAULT NULL,
  `grch38_isoform` varchar(100) DEFAULT NULL,
  `grch38_ref_seq` varchar(100) DEFAULT NULL,
  `hugo_symbol` varchar(50) DEFAULT NULL,
  PRIMARY KEY (`entrez_gene_id`),
  UNIQUE KEY `hugo_symbol` (`hugo_symbol`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `gene`
--

LOCK TABLES `gene` WRITE;
/*!40000 ALTER TABLE `gene` DISABLE KEYS */;
INSERT INTO `gene` VALUES (673,'ONCOGENE','ENST00000288602','NM_004333.4','ENST00000646891','NM_004333.4','BRAF'),(6098,'ONCOGENE','ENST00000368508','NM_002944.2','ENST00000368508','NM_002944.2','ROS1'),(7157,'TSG','ENST00000269305','NM_000546.5','ENST00000269305','NM_000546.5','TP53');
/*!40000 ALTER TABLE `gene` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `gene_alias`
--

DROP TABLE IF EXISTS `gene_alias`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `gene_alias` (
  `entrez_gene_id` int NOT NULL,
  `alias` varchar(255) DEFAULT NULL,
  KEY `FKF1C8F0A6C602BC92` (`entrez_gene_id`),
  CONSTRAINT `FKF1C8F0A6C602BC92` FOREIGN KEY (`entrez_gene_id`) REFERENCES `gene` (`entrez_gene_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `gene_alias`
--

LOCK TABLES `gene_alias` WRITE;
/*!40000 ALTER TABLE `gene_alias` DISABLE KEYS */;
INSERT INTO `gene_alias` VALUES (673,'BRAF1'),(6098,'MCF3'),(6098,'ROS'),(6098,'c-ros-1'),(7157,'LFS1'),(7157,'p53');
/*!40000 ALTER TABLE `gene_alias` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `geneset`
--

DROP TABLE IF EXISTS `geneset`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `geneset` (
  `id` int NOT NULL AUTO_INCREMENT,
  `name` varchar(255) NOT NULL,
  `uuid` varchar(40) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `geneset`
--

LOCK TABLES `geneset` WRITE;
/*!40000 ALTER TABLE `geneset` DISABLE KEYS */;
/*!40000 ALTER TABLE `geneset` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `geneset_gene`
--

DROP TABLE IF EXISTS `geneset_gene`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `geneset_gene` (
  `geneset_id` int NOT NULL,
  `entrez_gene_id` int NOT NULL,
  PRIMARY KEY (`geneset_id`,`entrez_gene_id`),
  KEY `FKA19C3AE7C602BC92` (`entrez_gene_id`),
  KEY `FKA19C3AE7C94E945F` (`geneset_id`),
  CONSTRAINT `FKA19C3AE7C602BC92` FOREIGN KEY (`entrez_gene_id`) REFERENCES `gene` (`entrez_gene_id`),
  CONSTRAINT `FKA19C3AE7C94E945F` FOREIGN KEY (`geneset_id`) REFERENCES `geneset` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `geneset_gene`
--

LOCK TABLES `geneset_gene` WRITE;
/*!40000 ALTER TABLE `geneset_gene` DISABLE KEYS */;
/*!40000 ALTER TABLE `geneset_gene` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `info`
--

DROP TABLE IF EXISTS `info`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `info` (
  `id` int NOT NULL AUTO_INCREMENT,
  `data_version` varchar(255) DEFAULT NULL,
  `data_version_date` datetime DEFAULT NULL,
  `ncit_version` varchar(255) DEFAULT NULL,
  `oncotree_version` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `info`
--

LOCK TABLES `info` WRITE;
/*!40000 ALTER TABLE `info` DISABLE KEYS */;
INSERT INTO `info` VALUES (1,'v5.0','2025-07-23 00:00:00','19.03d','oncotree_2019_12_01');
/*!40000 ALTER TABLE `info` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `portal_alteration`
--

DROP TABLE IF EXISTS `portal_alteration`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `portal_alteration` (
  `id` int NOT NULL AUTO_INCREMENT,
  `alteration_type` longtext,
  `cancer_study` longtext NOT NULL,
  `cancer_type` longtext NOT NULL,
  `protein_change` varchar(255) DEFAULT NULL,
  `protein_end` int DEFAULT NULL,
  `protein_start` int DEFAULT NULL,
  `sample_id` varchar(255) DEFAULT NULL,
  `entrez_gene_id` int DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `FK29E0AA52C602BC92` (`entrez_gene_id`),
  CONSTRAINT `FK29E0AA52C602BC92` FOREIGN KEY (`entrez_gene_id`) REFERENCES `gene` (`entrez_gene_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `portal_alteration`
--

LOCK TABLES `portal_alteration` WRITE;
/*!40000 ALTER TABLE `portal_alteration` DISABLE KEYS */;
/*!40000 ALTER TABLE `portal_alteration` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `portal_alteration_oncokb_alteration`
--

DROP TABLE IF EXISTS `portal_alteration_oncokb_alteration`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `portal_alteration_oncokb_alteration` (
  `oncokb_alteration_id` int NOT NULL,
  `portal_alteration_id` int NOT NULL,
  PRIMARY KEY (`oncokb_alteration_id`,`portal_alteration_id`),
  KEY `FKB822F32F7BF2942E` (`portal_alteration_id`),
  KEY `FKB822F32F56653D18` (`oncokb_alteration_id`),
  CONSTRAINT `FKB822F32F56653D18` FOREIGN KEY (`oncokb_alteration_id`) REFERENCES `alteration` (`id`),
  CONSTRAINT `FKB822F32F7BF2942E` FOREIGN KEY (`portal_alteration_id`) REFERENCES `portal_alteration` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `portal_alteration_oncokb_alteration`
--

LOCK TABLES `portal_alteration_oncokb_alteration` WRITE;
/*!40000 ALTER TABLE `portal_alteration_oncokb_alteration` DISABLE KEYS */;
/*!40000 ALTER TABLE `portal_alteration_oncokb_alteration` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `treatment`
--

DROP TABLE IF EXISTS `treatment`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `treatment` (
  `id` int NOT NULL AUTO_INCREMENT,
  `priority` int DEFAULT NULL,
  `uuid` varchar(40) DEFAULT NULL,
  `evidence_id` int DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `FKFC397878F7262935` (`evidence_id`),
  CONSTRAINT `FKFC397878F7262935` FOREIGN KEY (`evidence_id`) REFERENCES `evidence` (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=648 DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `treatment`
--

LOCK TABLES `treatment` WRITE;
/*!40000 ALTER TABLE `treatment` DISABLE KEYS */;
INSERT INTO `treatment` VALUES (85,1,NULL,2450),(86,2,NULL,2451),(87,1,NULL,2453),(88,2,NULL,2454),(89,1,NULL,2456),(90,1,NULL,2458),(91,2,NULL,2459),(92,1,NULL,2460),(93,1,NULL,2466),(94,1,NULL,2470),(95,1,NULL,2474),(96,1,NULL,2476),(97,1,NULL,2478),(98,1,NULL,2480),(99,1,NULL,2482),(100,1,NULL,2484),(101,1,NULL,2490),(102,1,NULL,2492),(103,2,NULL,2492),(104,1,NULL,2495),(105,1,NULL,2499),(106,2,NULL,2499),(107,1,NULL,2501),(108,1,NULL,2507),(109,2,NULL,2508),(110,3,NULL,2509),(111,4,NULL,2510),(112,1,NULL,2524),(113,1,NULL,2526),(114,1,NULL,2528),(115,1,NULL,2530),(116,1,NULL,2694),(117,2,NULL,2694),(118,2,NULL,2696),(119,1,NULL,2696),(120,1,NULL,2699),(121,1,NULL,2701),(122,1,NULL,2747),(123,1,NULL,2786),(124,2,NULL,2786),(125,2,NULL,2788),(126,1,NULL,2788),(127,1,NULL,2790),(128,2,NULL,2790),(129,2,NULL,2792),(130,1,NULL,2792),(131,1,NULL,2794),(132,2,NULL,2795),(133,3,NULL,2796),(619,1,NULL,26079),(620,2,NULL,26080),(621,3,NULL,26081),(622,4,NULL,26082),(623,5,NULL,26083),(624,1,NULL,26084),(625,1,NULL,26118),(626,2,NULL,26119),(627,3,NULL,26120),(628,1,NULL,26153),(629,2,NULL,26154),(630,3,NULL,26155),(640,1,NULL,28620);
/*!40000 ALTER TABLE `treatment` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `treatment_approved_indications`
--

DROP TABLE IF EXISTS `treatment_approved_indications`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `treatment_approved_indications` (
  `treatment_id` int NOT NULL,
  `approved_indications` longtext,
  KEY `FK333C46769D586F3F` (`treatment_id`),
  CONSTRAINT `FK333C46769D586F3F` FOREIGN KEY (`treatment_id`) REFERENCES `treatment` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `treatment_approved_indications`
--

LOCK TABLES `treatment_approved_indications` WRITE;
/*!40000 ALTER TABLE `treatment_approved_indications` DISABLE KEYS */;
INSERT INTO `treatment_approved_indications` VALUES (85,'Dabrafenib is FDA-approved for BRAF V600E mutant unresectable or metastatic melanoma.'),(86,'Vemurafenib is FDA-approved for BRAF V600E mutant unresectable or metastatic melanoma'),(87,'Dabrafenib and trametinib are FDA-approved for use in combination in the treatment of patients with metastatic non-small cell lung cancer (NSCLC) with BRAF V600E mutation'),(88,'Encorafenib and binimetinib are FDA-approved for use in combination in the treatment of patients with metastatic non-small cell lung cancer (NSCLC) with BRAF V600E mutation'),(89,'Dabrafenib in combination with trametinib is FDA-approved for the treatment of patients with locally advanced or metastatic anaplastic thyroid cancer (ATC) with BRAF V600E mutation and with no satisfactory locoregional treatment options.'),(101,'Vemurafenib is FDA-approved for the treatment of patients with unresectable or metastatic melanoma with the BRAF V600E mutation as detected by an FDA-approved test.'),(108,'Dabrafenib + Trametinib is FDA-approved for BRAF V600E or V600K mutant unresectable or metastatic melanoma'),(109,'Trametinib is FDA-approved for BRAF V600E or V600K mutant unresectable or metastatic melanoma'),(110,'Cobimetinib is FDA-approved for the treatment of patients with unresectable or metastatic melanoma with a BRAF V600E or V600K mutation, in combination with vemurafenib. Cobimetinib is not indicated for treatment of patients with wild-type BRAF melanoma.'),(111,'In combination for patients with unresectable or metastatic melanoma with a BRAF V600E or V600K mutation'),(619,'Crizotinib is FDA approved for the treatment of patients with metastatic non-small cell lung cancer (NSCLC) whose tumors are ROS1-positive.');
/*!40000 ALTER TABLE `treatment_approved_indications` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `treatment_drug`
--

DROP TABLE IF EXISTS `treatment_drug`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `treatment_drug` (
  `priority` int DEFAULT NULL,
  `drug_id` int NOT NULL,
  `treatment_id` int NOT NULL,
  PRIMARY KEY (`drug_id`,`treatment_id`),
  KEY `FK5720FB0744975E15` (`drug_id`),
  KEY `FK5720FB079D586F3F` (`treatment_id`),
  CONSTRAINT `FK5720FB0744975E15` FOREIGN KEY (`drug_id`) REFERENCES `drug` (`id`),
  CONSTRAINT `FK5720FB079D586F3F` FOREIGN KEY (`treatment_id`) REFERENCES `treatment` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `treatment_drug`
--

LOCK TABLES `treatment_drug` WRITE;
/*!40000 ALTER TABLE `treatment_drug` DISABLE KEYS */;
INSERT INTO `treatment_drug` VALUES (1,9,619),(1,9,626),(1,10,622),(1,13,623),(1,13,627),(1,13,629),(2,17,94),(2,17,96),(2,17,97),(3,17,104),(2,17,110),(1,17,117),(1,17,118),(1,17,123),(1,17,126),(1,17,127),(1,17,130),(2,17,133),(2,18,87),(2,18,89),(2,18,98),(2,18,99),(2,18,100),(2,18,108),(1,18,109),(1,18,112),(1,18,114),(1,18,116),(1,18,119),(1,18,124),(1,18,125),(1,18,128),(1,18,129),(2,18,132),(1,24,85),(1,24,87),(1,24,89),(1,24,98),(1,24,99),(1,24,100),(1,24,103),(1,24,106),(1,24,108),(1,24,132),(1,25,86),(1,25,93),(1,25,94),(1,25,96),(1,25,97),(1,25,101),(1,25,102),(1,25,104),(1,25,105),(1,25,110),(1,25,133),(1,26,88),(1,26,90),(1,26,91),(1,26,92),(1,26,111),(1,26,131),(2,27,88),(2,27,111),(2,27,131),(2,28,90),(2,28,92),(2,29,91),(3,30,92),(1,31,95),(1,31,120),(2,32,104),(1,33,107),(1,33,121),(1,34,113),(1,34,115),(1,34,122),(1,119,628),(1,130,620),(1,131,621),(1,131,630),(1,150,624),(1,150,625),(1,155,640);
/*!40000 ALTER TABLE `treatment_drug` ENABLE KEYS */;
UNLOCK TABLES;

--
-- Table structure for table `variant_consequence`
--

DROP TABLE IF EXISTS `variant_consequence`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `variant_consequence` (
  `term` varchar(100) NOT NULL,
  `description` varchar(255) NOT NULL,
  `is_generally_truncating` bit(1) NOT NULL,
  PRIMARY KEY (`term`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Dumping data for table `variant_consequence`
--

LOCK TABLES `variant_consequence` WRITE;
/*!40000 ALTER TABLE `variant_consequence` DISABLE KEYS */;
INSERT INTO `variant_consequence` VALUES ('3_prime_UTR_variant','A UTR variant of the 3\' UTR',_binary '\0'),('5_prime_UTR_variant','A UTR variant of the 5\' UTR',_binary '\0'),('any','any variant',_binary '\0'),('coding_sequence_variant','A sequence variant that changes the coding sequence',_binary '\0'),('coding_transcript_variant','A transcript variant of a protein coding gene',_binary '\0'),('downstream_gene_variant','A sequence variant located 3\' of a gene',_binary '\0'),('feature_elongation','A sequence variant that causes the extension of a genomic feature, with regard to the reference sequence',_binary '\0'),('feature_truncation','A sequence variant that causes the reduction of a genomic feature, with regard to the reference sequence',_binary ''),('frameshift_variant','A sequence variant which causes a disruption of the translational reading frame, because the number of nucleotides inserted or deleted is not a multiple of three',_binary ''),('fusion','Fusion',_binary ''),('incomplete_terminal_codon_variant','A sequence variant where at least one base of the final codon of an incompletely annotated transcript is changed',_binary ''),('inframe_deletion','An inframe non synonymous variant that deletes bases from the coding sequence',_binary '\0'),('inframe_insertion','An inframe non synonymous variant that inserts bases into in the coding sequence',_binary '\0'),('intergenic_variant','A sequence variant located in the intergenic region, between genes',_binary '\0'),('intron_variant','A transcript variant occurring within an intron',_binary '\0'),('mature_miRNA_variant','A transcript variant located with the sequence of the mature miRNA',_binary '\0'),('missense_variant','A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved',_binary '\0'),('NA','NA',_binary '\0'),('NMD_transcript_variant','A variant in a transcript that is the target of NMD',_binary '\0'),('non_coding_transcript_exon_variant','A sequence variant that changes non-coding exon sequence in a non-coding transcript',_binary '\0'),('non_coding_transcript_variant','A transcript variant of a non coding RNA gene',_binary '\0'),('non_truncating_variant','A change in the DNA that does not truncate or shorten the protein',_binary '\0'),('protein_altering_variant','A sequence_variant which is predicted to change the protein encoded in the coding sequence',_binary '\0'),('regulatory_region_ablation','A feature ablation whereby the deleted region includes a regulatory region',_binary '\0'),('regulatory_region_amplification','A feature amplification of a region containing a regulatory region',_binary '\0'),('regulatory_region_variant','A sequence variant located within a regulatory region',_binary '\0'),('sequence_variant','A sequence_variant is a non exact copy of a sequence_feature or genome exhibiting one or more sequence_alteration',_binary '\0'),('splice_acceptor_variant','A splice variant that changes the 2 base region at the 3\' end of an intron',_binary ''),('splice_donor_5th_base_variant','A sequence variant that causes a change at the 5th base pair after the start of the intron in the orientation of the transcript',_binary ''),('splice_donor_region_variant','A sequence variant that falls in the region between the 3rd and 6th base after splice junction (5\' end of intron)',_binary ''),('splice_donor_variant','A splice variant that changes the 2 base region at the 5\' end of an intron',_binary ''),('splice_polypyrimidine_tract_variant','A sequence variant that falls in the polypyrimidine tract at 3\' end of intron between 17 and 3 bases from the end (acceptor -3 to acceptor -17)',_binary ''),('splice_region_variant','A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron',_binary ''),('start_lost','A codon variant that changes at least one base of the canonical start codon',_binary ''),('start_retained_variant','A sequence variant where at least one base in the start codon is changed, but the start remains',_binary '\0'),('stop_gained','A sequence variant whereby at least one base of a codon is changed, resulting in a premature stop codon, leading to a shortened transcript',_binary ''),('stop_lost','A sequence variant where at least one base of the terminator codon (stop) is changed, resulting in an elongated transcript',_binary '\0'),('stop_retained_variant','A sequence variant where at least one base in the terminator codon is changed, but the terminator remains',_binary '\0'),('synonymous_variant','A sequence variant where there is no resulting change to the encoded amino acid',_binary '\0'),('TFBS_ablation','A feature ablation whereby the deleted region includes a transcription factor binding site',_binary ''),('TFBS_amplification','A feature amplification of a region containing a transcription factor binding site',_binary '\0'),('TF_binding_site_variant','A sequence variant located within a transcription factor binding site',_binary '\0'),('transcript_ablation','A feature ablation whereby the deleted region includes a transcript feature',_binary ''),('transcript_amplification','A feature amplification of a region containing a transcript',_binary '\0'),('upstream_gene_variant','A sequence variant located 5\' of a gene',_binary '\0');
/*!40000 ALTER TABLE `variant_consequence` ENABLE KEYS */;
UNLOCK TABLES;
/*!40103 SET TIME_ZONE=@OLD_TIME_ZONE */;

/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40014 SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;

-- Dump completed on 2025-09-02 14:19:33
